# Introduction
```{r setup, include = FALSE}
library(tidyverse)
library(knitr)
library(kableExtra)
```
\glsresetall

This thesis investigates the role that positive genetic selection has played in causing the health disparities of metabolic disease in Polynesian populations. 

This chapter provides the background and rationale for the research presented in this thesis. It covers the settlement of the Pacific, and highlights the health disparities facing Polynesian populations and the releveance of genetics. It then introduces natural selection and the methodologies for detecting 'signals of selection', followed by how genetic selection may have played a role in creating the health disparities caused by metabolic disease in Polynesian populations.


## Settlement of Polynesia

The Polynesian triangle is defined as the the geographic region in the Pacific Ocean bounded by \Gls{nz} in the southwest, Hawaii in the north, and Easter Island in the east [@Barcham2009]. The settlement history of this region begins with the Out of Africa migration approximately 50-100 thousand \gls{ya} [@Nielsen2017]. This migration flowed up the Levant and Arabian peninsula (45-55 k\gls{ya}) and then split into a migration to South Asia, Indonesia, and Australia (50 k\gls{ya}, @Kivisild1999; @Quintana-Murci1999), with a migration into Europe under way 45 k\gls{ya}. From South Asia, the migration continued into East Asia (20 k\gls{ya}, @Groucutt2015). 

Near Oceania is the geographic region from New Guinea and the Bismark Archipelago, through to the Solomon Islands in the east, with the remainder of the Pacific Islands east from the Reef/Santa Cruz group, southeast of the Solomon Island being Remote Oceania [@Matisoo-Smith2018]. The settlement of Near Oceania occurred first (40 k\gls{ya}), and was followed by the settlement of Remote Oceania, which reached as far as Samoan and Tonga (Western Polynesia) as part of the Lapita expansion (3 k\gls{ya}, @Matisoo-Smith2015; @Skoglund2016). After this settlement, came the peopling of the Eastern Polynesia (including the Cook Islands) (1-1.2 k\gls{ya}; @Wilmshurst2011) and then finally, the settlement of \glsdesc{nz} by \gls{nz} M\tex{\={a}}ori 800 \gls{ya} [@Duggan2014;@Matisoo-Smith2015].

In the 1950's and 60's there was a rural to urban migration of the Polynesian populations, for mostly economic reasons. As part of this, many individuals and families relocated from Samoa, Tonga, and the Cook Islands to New Zealand in search of work, and now the populations in New Zealand largely out-number the populations of their islands of origin [@MatisooSmith2012highway]. Within \glsdesc{nz}, the largest number of Polynesian people is found within the Auckland region [@Barcham2009].


## Metabolic syndrome

The World Health Organisation has previously defined Metabolic Syndrome to be glucose intolerance, or diabetes mellitus, and or insulin resistance combined with at least two of the following: impaired glucose regulation or diabetes, insulin resistance, raised arterial pressure, raised plasma triglycerides or low high-density lipoproteins to cholesterol levels, central obesity, and microalbuminuria [@Alberti1998]. \Gls{hu} can also be considered a component but is not required for the condition to be recognised. \gls{hu} is defined as a serum urate $\geq$ 0.42 mmol/L in males and $\geq$ 0.36 mmol/L in females [@Choi2007b]. There has been a slight revision and refinement in the criteria for a diagnosis of metabolic syndrome [@Alberti2009]. The requirement for diabetes mellitus and or insulin resistance is no longer used, instead a requirement of having 3 of the 5 co-morbidities is used.

Obesity is defined as a \gls{bmi} of >30 kg/m^2^ and is a contributing risk factor for many other diseases such as \gls{t2d} and \gls{cvd} [@Haslam2005]. Obesity has been a key focus for research as the prevalence continues to increase and is posing a major threat to health systems in many countries [@Wang2011]. With respect to Pacific people obesity is a major problem, with an obesity rate of 68%. New Zealand M\tex{\={a}}ori have an obesity prevalence of 48% [@Health2013]. From 2012 to 2013 the prevalence of obesity increased by 6% in Pacific populations. 

\Gls{t2d} is diagnosed by a fasting blood glucose of >7.0 mmol/L or 2-hr post blood glucose load of  >11.1 mmol/L [@Alberti1998]. The risk of developing \gls{t2d} is strongly correlated with excess weight [@Rana2007]. The prevalence of diabetes is 13% in Pacific people and 7% in NZ M\tex{\={a}}ori [@Health2013] compared to 5.5% in Europeans [@Winnard2013]. In New Zealand, the rate of diabetes is twice that in M\tex{\={a}}ori compared to non-M\tex{\={a}}ori, and the rate is 3.6 fold higher in Pacific compared to non-Pacific ethnicities. 

### Hyperuricaemia and gout

Hyperuricaemia can be caused by diet, genetic predisposition or under-excretion of urate. The elevated uric acid levels in the blood contributes to the forming of urate crystals in the joints, triggering an inflammatory response by stimulating the synthesis and release of humoral and cellular inflammatory mediators [@Choi2005a]. There is a clear relationship between hyperuricaemia and the causal role in gout [@Snaith2004]. There is also an increased prevalence in gout in different ethnicities. New Zealand M\tex{\={a}}ori and Pacific ethnicities have a prevalence of 12% and 14%, respectively, in males, compared to 4% in both European and Asian males [@Winnard2012; @Winnard2013]. While New Zealand M\tex{\={a}}ori and Pacific people have an increased prevalence of hyperuricaemia, they also have a genetic predisposition to elevated urate and risk of gout  [@Hollis-Moffatt2009; @Hollis-Moffatt2011; @Phipps-Green2010]. The prevalence of metabolic syndrome in US gout patients is ~60%, or nearly 3 fold higher than US adults without gout [@Choi2007b].




## Genetic variation

As with many complex diseases, such as gout, \gls{t2d}, obesity, kidney disease, and metabolic syndrome, there is an environmental component and a genetic component. Genetic variation at a single locus, or at many, influences the genetic risk of an individual for developing disease. Genetic variation refers to the differences in DNA sequence that occur between individuals of a species. Types of genetic variation include, \gls{snp} which are single base changes (or polymorphisms), and \glspl{indel}. \Glspl{indel} are either additional DNA, or DNA that had been lost, and can be short (a single base, up to approximately 100 base pairs) or long (kilobases to megabases) at which point that can also become structural variants. Other types of genetic variation include copy number variation (alterations in the number of copies of a gene from what would be expected from the ploidy of the organism), and re-arrangements such as translocations - segments of DNA moving to different chromosomes, or inversions - segments of DNA that are excised, inverted, and reinserted at the same location on the chromosome. Within population based genetic studies, such as case-control association studies, the prominent type of genetic variation that is studied are \glspl{snp}. 

Common methodologies of genotyping, to determine the genetic variation of an individual, are \gls{snp} arrays  or sequencing. \Gls{snp} arrays, such as the Illumina Human CoreExome beadchip contain probes designed for specific genetic variants that bind DNA samples, and after a single base extension step with a labelled nucleotide, enable a genotype to be determined from a fluorence sensitive scanner [@Wang1998;@Gunderson2005; @Shen2005]. Sequencing technologies, such as the short read sequencing performed by the next-generation sequencers, take DNA and randomly fragment it, then determine the bases sequentially in the fragment to approximately 150 bp from the start to create a 'read' [@Metzker2010]. The reads are then aligned to a reference genome and at each base of the genome a genotype is called by using the distribution of bases from reads overlapping that position [@Li2008; @mckenna2010genome]. One of the key differences between \gls{snp} arrays and sequencing is that \gls{snp} arrays capture known variation, whereas sequencing allows for unknown genetic variants to be genotyped.

## Natural selection

Charles @darwin1909origin in The Origin of Species declared: _“[The] preservation of favourable individual differences and variations, and the destruction of those which are injurious, I have called Natural Selection or Survival of the Fittest. Variations neither useful or injurious would not be affected by natural selection, and would be left either a fluctuating element, as perhaps we see in certain polymorphic species, or would ultimately become fixed, owing to the nature of the organism and the nature of the conditions.”_ This theory set the scene for research in population biology. Selection acts upon the phenotype and this pressure then transfers through to act upon the genetic contribution of that phenotype. Most important to Darwin’s Theory was it created models for what natural selection should look like and the methods for being able to detect selective events on the genetics of organisms. A focus on selection in humans has added insight into past events since migration out of Africa [@Soares2012] and yielded possibilities of the influences on human disease.


### Neutrality and genetic drift

The random sampling of alleles can cause an increase or decrease in allele frequency of a population leading to genetic drift. @Wright1931 proposed genetic drift as an evolutionary model. He put forward that by chance, in a finite population without selection acting, an allele could increase in frequency over generations and time, changing significantly from the original frequency values. Fisher believed that “Natural Selection depends on the succession of favourable chances”, and thought that genetic drift was insignificant compared to selection [@Fisher1930]. In Fisher's model, genetic variation would disappear at an extremely slow rate in a population without selection, and a moderate rate of new mutations would be enough to maintain the population variability [@Fisher1922]. @Feller1951 unified the population models from both Wright and Fisher into what is now known as the Wright-Fisher model. The Wright-Fisher model in its simplest form treats a diploid population of size $N$ as a haploid population of $2N$. In unifying the population models Feller noted that the assumption of constant population size contributes more than was previously thought, and this had been crucial in Wright’s model for gene frequency to satisfy a diffusion equation.

Adding further to the genetic drift side of the discussion was the neutral theory of molecular evolution. The neutral theory of molecular evolution put forward by @Kimura1979a states that most evolutionary change is not due to selection but is due to random genetic drift. The term “drift” was used because the variants under neutrality are expected to convey no advantage, and rise and fall (or drift) randomly in prevalence over time. Random genetic drift plays a role in population structure by keeping the number of co-existing alleles down [@Kimura1955]. One of the problems with the neutral theory was how the diversity of species could be accounted for. The solution to this can be explained through a positive correlation between increased genetic diversity and an increase in recombination, hinting at a rise in mutation rate as frequency of genetic exchange increases [@Hellmann2003]. This link provided an explanation of a neutral process leading to increased genetic diversity.

Crucial to both the neutral theory and the theory of natural selection is the estimation of the population mutation rate, commonly referred to as $\theta$. The Watterson estimator, $\theta$ = 2 $\times$ ploidy $\times$ N~e~ $\times$ $\mu$, represents the expected number of new mutations for a population per generation [@Watterson1975], where $\mu$ is the mutation rate and N~e~ is the effective population size. Effective population size is the size of an idealised population with the same amount of inbreeding or random gene frequency drift as the population under consideration [@KimuraCrow1963]. Other estimators of $\theta$ include: $\hat{\theta}$ = F~ST~ [@Weir1984], and $\hat{\theta}$ = E($\pi$), in the infinite sites model, where $\pi$ is the average pair-wise nucleotide differences per site [@Tajima1996a]. The infinite sites model assumes only one mutation occurs per site.

Natural selection seems to be the more popular model to the neutral hypothesis, however the methodology for detecting selection utilises the neutral hypothesis for many of the methods mentioned later (section \@ref(selMethods)). These methods use a model for neutrality, and if the statistic reaches a level that is considered to be beyond that of the neutral model, the model is rejected. Typically the neutral hypothesis is a composite hypothesis using assumptions for which it would be difficult to find a human population that fulfils them, such as: “the population is in equilibrium at constant size with no population subdivision or gene flow from other populations” [@Nielsen2005a].

### Selection

Possible types of selection that can act upon a population include balancing, background, or directional selection. Balancing selection maintains both alleles in a population and is often the case when the heterozygote has an advantage over the homozygote and can be referred to as "over-dominant selection". Two examples of this are MHC class 1 [@Hughes1988] and sickle-cell anaemia [@Allison1956]. With sickle-cell anaemia, individuals homozygous for normal haemoglobin alpha are susceptible to malaria, individuals homozygote for the 'sickling' mutation develop sickle-cell disease, and the heterozygote has a protective effect against malarial infection [Aidoo2002]. "Background selection" is a reduction of genetic diversity at a locus because of deleterious selection acting on a separate but linked locus [@Charlesworth1993]. Background selection can be thought of as the opposite effect to genetic hitch-hiking. Genetic hitch-hiking occurs when a locus is under selection, linked sites on the same chromosome will increase in frequency, causing a change in frequency of the alleles that are not on the same chromosome [@Smith1974]. \Gls{ld} can arise at sites that are under directional selection, even when the loci are unlinked [@Felsenstein1965] but, recombination will act to reduce this \gls{ld}. 

Directional selection operates in two fashions, positive or negative. Positive selection is where an allele conveys an advantage in fitness that enhances reproductive success, resulting in the beneficial allele increasing in frequency within the population. An example of positive selection is the lactase gene in European populations which enables the absorption of lactose [@Bersaglieri2004]. Negative selection occurs when the allele conveys a fitness disadvantage and ends with reduced reproductive success, decreasing the frequency of the allele in the population. The cystic fibrosis gene _CFTR_, is an example of negative selection, where in males splicing mutations can lead to sterility [@Pagani2005].


The way selection can affect allele frequencies can be thought of as a sweep: as an allele is selected it “sweeps” through the population. The intensity of this sweep can be classified as, hard, soft, and partial or incomplete. A hard sweep, also known as a classical sweep, is when a beneficial allele rapidly rises in frequency and through hitch-hiking the neutral variation is simultaneously reduced at linked sites, as shown in Figure \@ref(fig:sweepDiagram) [@Hermisson2005;@Hermisson2017; @Nielsen2005; @Smith1974].

(ref:sweepDiagram) Representation of alleles before and after a hard sweep.  Each rectangle represents a haplotype within a population. A new mutation is shown in red and neutral alleles are shown in black. After a hard sweep the new mutation becomes fixed in the population, along with the neutal allele that is linked to it.


```{r sweepDiagram, echo = FALSE, fig.cap='(ref:sweepDiagram)', fig.width=3, fig.asp=1, out.width='50%', fig.align='center'}
knitr::include_graphics('images/sweep_diagram.png')
```

A soft sweep occurs when an initially fitness-neutral allele evolving under drift experiences a beneficial environmental change and increases in frequency until fixation. The initial neutral nature of the allele prior to becoming beneficial means there is a range of surrounding haplotypes and there is not as much reduction in genetic diversity as happens with a hard sweep [@Hermisson2005; @Schrider2015b; @Hermisson2017]. The result of a soft sweep is multiple haplotypes of intermediate frequency around the selected site. If the selective environment for a population changes frequently enough, soft sweeps could be the main method of adaptation [@Schrider2015b].

A partial or incomplete sweep occurs when part way through a sweep the selective pressure is changed/removed resulting in an increase in frequency for the allele that then returns to random genetic drift.

In order to determine if a selective event has occurred, one can attempt to find what are termed “signatures of selection”. Population differentiation can sometimes be indicative of a selective event. If extreme levels of differentiation are observed at a locus between populations, this can be interpreted as evidence of positive selection. Regions of the genome that have a reduction in variability in a population can be the effect of a selective sweep. A third example of a signature of selection is deviation in the frequency spectrum for alleles beyond what would be considered possible under neutrality [@Przeworski2002]. A key signature of positive selection is the increase in proportion of high frequency variants [@fay2000hitchhiking].

Assuming a background of neutrality means that there needs to be a way to identify variants under selection as opposed to population demographic events, such as bottlenecks or expansions. In the case of Polynesian settlement, there is evidence for a founder effect which needs to be considered when interpreting selection results [@Kayser2006a]. It is for this reason that the unique features of a selective event need to be able to be identified from other possible features such as population demography or admixture. Population events and demography affect the entire genome, whereas selection acts on a localised region [@Stajich2005]. To be able to recreate demography, the choice in the \glspl{snp} used in the calculation of neutrality and selection statistics is important as there is chance for a large ascertainment bias if \glspl{snp} are only selected from a small number of populations [@Wall2008]. The \glspl{snp} used also need to be sampled away from genic areas.

## Methods for detecting selection {#selMethods}

Since 2005 there have been multiple reviews published (see @Nielsen2005, @sabeti2006positive, @Utsunomiya2015, @Vitti2013 and @Haasl2016) on methods of identifying selection, as there has been a renewed interest to examine selection at a genome wide scale due to increased availability of whole genome genotype data. Methods that have been developed to detect signatures of selection can be placed into two main categories, those that only work on a single population and those that compare between populations. There are also specific underlying genetic features that these methods can use, such as allele frequency (frequency spectrum methods) or differences in haplotype (linkage methods). There are also methods that combine different approaches (composite methods). A summary of selection methods is provided in Table \@ref(tab:selectionMethods).


The different methods for detecting selection work best over different time periods. Recent soft sweeps are able to be detected well with \gls{ld}-based methods, but frequency spectrum tests struggle [@Hermisson2005]. The opposite applies in the case of hard sweeps [@Hermisson2005]. In humans, the method that is able to detect the most ancient selection, ranging to millions of years ago is the K~a~/K~s~ ratio, however if a strong background of neutrality is present this can undermine the effectiveness of the test. \Gls{td} is able to detect selection events as old as 250,000 years ago in regions of low diversity containing an excess of rare alleles. \Gls{fwh} is able to detect events less than 80,000 years ago based on derived alleles having arisen through mutation and then risen to a high frequency via hitch-hiking. F~ST~, in the range of less than 50,000 to 70,000 years ago, picks up geographic separation of populations that have been subject to different physical and cultural environments based on the changes in allele frequency in one population versus another. Methods relying on long haplotypes such as \gls{ihs} have the shortest range at less than 30,000 years ago for a long haplotype that hasn't had time to break down through recombination and contains a high frequency variant. Partial sweeps can also be detected down to a \gls{maf} of approximately 10% [@sabeti2006positive].

```{r, eval = TRUE, echo = FALSE}
tibble(
  stats = c("Ka/Ks","HKA","iHS", "XPEHH", "nSL", "Tajima's \\textit{D}", "Fay and Wu's \\textit{H}", "Fu and Li's \\textit{F}", "Zeng's \\textit{E}", "F\\textsubscript{ST}", "$\\Delta$DAF", "SCCT","CLR","XP-CLR","CMS", "Meta-SS", "CSS"),
  intra_inter  = c(rep("Within and Between",2), "Within","Between",rep("Within", 5), "Between","Between","Within","Within","Between", rep("Within and Between",3)),
  timeframe = c("Ancient selection", "","< 30,000 years", "","", "< 250,000 years", "< 80,000 years", "", "", "< 50-70,000 years", rep("",7)),
  ref = c("\\citet{Hughes1988}", "\\citet{Hudson1987}","\\citet{voight2006map}", "\\citet{tang2007new}", "\\citet{Ferrer-Admetlla2014}", "\\citet{Tajima1989}", "\\citet{fay2000hitchhiking}", "\\citet{Fu1993}", "\\citet{Zeng2006}", "\\citet{Weir1984}", "\\citet{Grossman2010}", "\\citet{Wang2014}", "\\citet{Kim2002}", "\\citet{Chen2010}", "\\citet{Grossman2010}", "\\citet{Utsunomiya2015}", "\\citet{Randhawa2014}")
) %>% kable(., escape = FALSE, col.names = c("Method","Detects selection within or between populations", "Timeframe", "Key reference"), caption = " (\\#tab:selectionMethods) Summary of Selection Methods", booktabs = TRUE, format = 'latex') %>% group_rows("Macroevolutionary", 1, 2) %>% group_rows("Haplotypic", 3, 5) %>% group_rows("Frequency spectrum", 6, 12) %>% group_rows("Composite", 13, 17) %>% column_spec(2, width = '10em')
```

### Macroevolutionary methods

#### K~a~/K~s~
The ratio of non-synonymous to synonymous mutations is called the K~a~/K~s~ ratio (or d~N~/d~S~ ratio as it is often referred to), and is used to infer functional impact from selection, since synonymous mutations are thought of as silent mutations with a neutral effect on the protein [@Hughes1988]. Comparing the baseline rate of mutation (synonymous) to the non-synonymous rate gives an understanding of the tolerance of amino acid alternatives in the protein, with an excess of non-synonymous mutations indicating the novel protein structures are being favoured and the protein is being positively selected for. A K~a~/K~s~ ratio greater than 1 is indicative of positive selection, while a value less than 1 is indicative of negative selection against deleterious mutations and for keeping the protein conserved.

#### HKA

The HKA test [@Hudson1987] compares between and within populations the polymorphism and divergence of two or more loci, and uses this to establish if a locus has been under selection, since variation within, and diversity between, species under neutrality should be based only on the mutation rate. The assumption that mutation rates at neutral loci is constant over time is important, as if it changes, it will affect the difference between the polymorphism and divergence. In practice the HKA test is powered for recent selection events, however, in practice it is difficult to find a putatively neutral locus [@Zhai2009].

### Haplotypic methods

There are a few methods that are different derivations and applications of looking at haplotype decay around a core \gls{snp}. @sabeti2006positive put forward the concept of using \gls{ehh}. The haplotypes that had unusually long haplotype homozygosity and a high frequency in the population indicated that the polymorphism had risen in prominence faster than it would be expected under the neutral model and therefore under selection. The main concept is that a beneficial mutation wouldn't solely be selected for, but the surrounding region would initially be selected and therefore have a conservation effect. As time and generations progressed, this haplotype would decay due to recombination, and this haplotype decay can be used to assist in the ageing of the selection event. In order to use haplotypic methods, there is a requirement to establish the alleles that were inherited from the maternal side and those that were inherited from the paternal side constituting the two haplotypes (phasing).

#### Integrated haplotype homozygosity score

The \gls{ihs} [@voight2006map; @Szpiech2014] calculation starts with \gls{ehh}, which is the probability that 2 chromosomes carrying a core haplotype are homogenous to the distance $x$. A value of 0 represents no homozygosity, meaning all haplotypes are different, while a value of 1 is complete homozygosity  meaning all haplotypes are the same. The \gls{ehh} value versus distance is integrated to find the area under \gls{ehh} curve until \gls{ehh} decays to 0.05. This \gls{ihh} is calculated based on the ancestral or derived allele and annotated as \gls{ihh}~A~ or \gls{ihh}~D~. The unstandardised \gls{ihs} is calculated in equation \@ref(eq:ihs), which is the \texttt{selscan} version [@Szpiech2014]. Standardisation of \gls{ihs} is shown in equation \@ref(eq:stdihs), where the expectations and standard deviation of equation \@ref(eq:ihs) are estimated from the empirical distribution of SNPs whose derived allele frequency p matches the frequency at the core \gls{snp} [@voight2006map]. Large positive values indicate long haplotypes carrying the derived allele, whereas negative values indicate haplotypes carrying the ancestral allele. The calculation in @voight2006map has \gls{ihh}~A~ and \gls{ihh}~D~ switched. The @Szpiech2014 \gls{ihs} calculation is used in this thesis (equations \@ref(eq:ihs) and \@ref(eq:stdihs)). \Gls{ihs} identifies variants under selection driven to intermediate frequencies.

\begin{equation} 
iHS_{unstd} = \ln{\frac{iHH_D}{iHH_A}}
(\#eq:ihs)
\end{equation}

\begin{equation} 
iHS_{std} = \ln{\frac{iHH_D}{iHH_A}} - \frac{E_p [\ln{\frac{iHH_D}{iHH_A}}] }{ SD_p [\ln{\frac{iHH_D}{iHH_A}}] }
(\#eq:stdihs)
\end{equation}


#### Cross population extended haplotype homozygosity

\Gls{xpehh}, created by @tang2007new, is also known as Rsb. \Gls{xpehh} was designed for use with \gls{snp} data rather than sequence. \Gls{xpehh} lacks power to detect intermediate frequency variants but is designed to detect variants that are near fixation or completely fixed. The \gls{ehh} of \gls{snp} site (EHHS), integrates to sum the area under the decay by distance in \gls{ehh} (iHH) which is then used as a log ratio between populations at a single site (equation \@ref(eq:xpehh)) and in a standardised form (equation \@ref(eq:xpehhstd)), with 1 and 2 indicating the population. Recombination rate is mostly conserved within populations, so provides an internal control in \gls{xpehh} for effects of heterogeneous recombination rates. Extreme values of \gls{xpehh} indicate a slower decay of \gls{ehh} in one population compared to the other. \Gls{xpehh} relies on the breakdown of \gls{ld} over time and has weak power to detect selective sweeps that were historical and ended thousands of generations ago [@Chen2010]. F~ST~ is better than \gls{xpehh} for differential selection between closely related populations because haplotype-based signals are mostly shared between geographically similar populations [@pickrell2009signals]. 

\begin{equation} 
 XPEHH_{unstd} = \ln{\frac{iHH_1}{iHH_2}}
(\#eq:xpehh)
\end{equation}

\begin{equation} 
 XPEHH_{std} = \frac{\ln{\frac{iHH_1}{iHH_2}} - E [ \ln{\frac{iHH_1}{iHH_2}} ]}{ SD [ \ln{\frac{iHH_1}{iHH_2}} ]}
(\#eq:xpehhstd)
\end{equation}

#### Number of segragating sites by length

\Glsdesc{nsl} is another single population haplotypic method that is extremely similar to \gls{ihs}, however \gls{nsl} also measures the length of a segment of haplotype homozygosity between a pair of haplotypes in terms of number of mutations in the remaining haplotypes in the data set in the same region [@Ferrer-Admetlla2014]. Demographic events affect \gls{nsl} less than \gls{ihs}. Under simulations for different demographic events, \gls{nsl} had a smaller total difference in variation between the standard distribution of the test statistic and the distribution of the modelled demographic event [@Ferrer-Admetlla2014]. 

### Frequency spectrum methods

The frequency spectrum methods are based on finding deviations in the frequency spectra of alleles that are outside of what would be expected under neutrality. These can involve an increase in particular types of frequencies such as low, intermediate, high variant frequencies, or a difference in allele frequencies between populations. 

#### Tajima's _D_

\gls{td} [@Tajima1989] counts the number of segregating sites of individuals in a population and finds the ratio between polymorphisms and pair-wise individual comparisons. \Gls{td} is a test of the neutral hypothesis. The hypothesis is rejected when there is an excess of low frequency variants. A negative \gls{td} is indicative of positive selection or weak negative selection but a negative value can also be attributed to population events such as population expansion. Positive values of \gls{td} indicate an excess of intermediate frequency variants. An excess of intermediate frequency variants could also be attributed to balancing selection, population structure, or population bottlenecks [@Kreitman2000; @Nielsen2005; @nielsen2007recent]. The comparison of other polymorphisms, such as changes in \glspl{snp} versus changes in \gls{indel} numbers can provide further evidence for either selection or a population event, since population events should affect both \glspl{snp} and \glspl{indel} in an equal fashion. The direction (positive, zero, or negative) of \gls{td} values after a population bottleneck depends on the strength and duration of the bottleneck and can generate similar \gls{td} values, and can be positive, negative or zero [@Fay1999]. After a population reduction or selective event it is expected the population will have excess rare alleles (recent, low frequency) as the population recovers, therefore \gls{td} will be negative [@Barton1998]. \Gls{td} is sensitive to sequencing errors that are called as \glspl{snp} because they appear equivalent to low frequency variants [@Achaz2008].

#### Fu and Li's F

\gls{flf} is the comparison of the number of derived singleton mutations and the mean pair-wise difference between sequences [@Fu1993]. It can be calculated both with or without an out-group (_F_^\*^), with Fu and Li's _F_^\*^ having the greatest power of the Fu and Li statistics [@Fu1993; @Ramirez-Soriano2008].



#### Fay and Wu's _H_

\gls{fwh} is sensitive to \glspl{snp} rising to moderate to high frequency and uses an out-group (such as the ancestral reference sequence) to determine the derived/ancestral state of the allele. A positive value indicates a deficit in derived moderate to high frequency \glspl{snp}. A negative value indicates an excess of derived moderate to high frequency \glspl{snp}. \Gls{fwh} rejects neutrality when there is an excess of high but not low frequency variants. An increase in the proportion of high frequency variants compared to intermediate frequency variants is a unique signature of positive selection [@fay2000hitchhiking]. \Gls{fwh} can be used in regions of low recombination to distinguish hitch-hiking from neutral or background selection where it is expected there will be the same level of intermediate to high frequency variants.

#### Zeng's E

In contrast to \gls{fwh} and \gls{td}, which compare either high or low frequency variants with intermediate frequency variants, \gls{ze} compares the low and high frequency variants [@Zeng2006]. After a selective sweep, the lower frequency variants are expected to 'bounce back' faster than the higher frequency variants. The _E_ test is not sensitive to population subdivision. A negative value for \gls{ze} is indicative of a selective sweep, but a rapid population expansion can also produce a negative value.

<!-- Why is it not sensitive to population subdivision? -->

<!-- > equation? -->

<!-- > @Ferretti2017 explains how to interpret in the the form of trees -->

#### F~ST~

F~ST~ is a population differentiation metric that measures the differences in heterogeneity of a sub-population compared to a global population. Initially created by @Wright1951, as an inbreeding coefficient for inbreeding as a whole in a population (equation \@ref(eq:fst)), F~IT~ is the inbreeding coefficient for the individual in the total population and F~IS~ is the inbreeding coefficient for inbreeding of the individual in the subpopulation. These fixation indices provide ways of summarising population structure. Selection will act to drive an allele to fixation, whereas immigration usually reduces the frequency. When the selective advantage is strong, selection will be the dominant force on the locus, however, when the selective advantage is weak the allele will reduce in frequency and disappear. Cockerham [-@Cockerham1969; -@Cockerham1973] equated $\theta$ (population genetic differentiation) with F~ST~, extending F~ST~ from calculating population structure to population differentiation too. The fixation indices were further developed by @Weir1984 to add population size weightings to the calculation. F~ST~ values fall between 0 and 1, each representing the fixation of an allele. Large differences in F~ST~ values are indicative of population differentiation, suggesting directional selection at that locus, whereas small differences in F~ST~ indicate that the populations are similar. F~ST~ can have high variation in neighbouring loci under neutrality [@Weir2005]. Unless populations are closely related, such as northwest and southeast Europeans [@Price2008a], the noise will drown out the signal, and identification of genome-wide significance in F~ST~ will be difficult. @Weir2005 suggested taking a sliding window approach to help improve the signal to noise ratio.

\begin{equation} 
F_{ST} = \frac{F_{IT}{-}F_{IS}}{1{-}F_{IS}}
(\#eq:fst)
\end{equation}

F~ST~ has been used in multiple studies as evidence for natural selection [@Akey2012; @myles2007identification; @pickrell2009signals]. @Akey2012 used F~ST~ at individual loci and compared this to the distributions for the genome, chromosome and gene levels. The use of the empirical distribution to compare to F~ST~ for each \gls{snp} controlled for the effects of demography. F~ST~ as a method for detecting selection was developed further into d~_i_~, which is a function of pair-wise F~ST~ between population _i_ and the remaining populations [@Akey2010]. Outside of the human context F~ST~ has been a very popular method of reporting both population structure and evidence of selection [@Akey2010; @Hancock2011; @Qanbari2011; @Wei2015b]. The population branching statistic is a F~ST~ based statistic that involves the use of an out-group to compare estimates in divergence time between populations [@Yi2010b].

<!-- #### Population branching statistic -->

#### Selection by conditional coalescent tree

\Gls{scct} (@Wang2014) detects recent positive selection by searching for an imbalance of genetic variants and conditions these on the allele frequencies of the candidate loci. \Gls{scct} pinpoints the causal variant more accurately by using a method that is based on the conditional coalescent tree method from @Wiuf1999, where haplotypes are partitioned into two subgroups according to the allelic state at a particular locus. It is assumed one subgroup carries the derived allele at the locus and the other subgroup carries the ancestral allele. The coalescence of a sample of haplotypes, conditioning on the particular mutation, means each group should coalesce together individually before the two subgroups coalesce. This is a similar idea to that of \gls{ehh} used by \gls{ihs}, and the results of \gls{ihs} and \gls{scct} tend to be similar. The selection statistic (S) is based on comparing the lineage lengths of the two groups. Under neutrality S approaches $\theta$, with significant departures from $\theta$ being considered evidence of positive selection. When natural selection has a strong hitch hiking effect on a selectively favoured allele, \gls{scct} (and \gls{ihs}) is more powerful than \gls{td}, Fay and Wu’s _H_, and measuring change in derived allele frequency (see \@ref(dafsection)). In most cases the power of \gls{scct} is equivalent to \gls{ihs} except when selection is very strong, in which case \gls{ihs} has more power [@Wang2014]. The power of \gls{scct} can be improved by increasing the sample size. \Gls{scct} performs moderately with population demographic events (such as bottlenecks) and out performs \gls{td} and \gls{ihs} in false discovery rate, although it does not perform as well as  \gls{fwh} [@Wang2014].

#### Change in derived allele frequency {#dafsection}

Similar to F~ST~, \gls{ddaf} looks at population differentiation. The derived allele is determined through the use of an ancestral out-group such as chimpanzees when used in human populations. \gls{ddaf} measures the absolute difference in the derived allele frequency between two populations [@Grossman2010] and has greater power to detect sweeps, both partial and complete, and outperforms \gls{xpehh} for partial sweeps [@Colonna2014]. \gls{ddaf} is part of the composite of multiple sites statistic [@Grossman2010] but has also been used as a standalone metric in humans [@Colonna2014; @Gudbjartsson2015] and in other species such as cattle [@Randhawa2014] to measure differentiation.

### Composite methods

Composite methods combine multiple methods with the goal of increasing overall power and/or spatial resolution for detecting selection. The combination of methods also gives greater resilience against demographic events.

#### Composite likelihood ratio

The \gls{clr}, created by @Kim2002, is a test for local
hitch-hiking selection along a recombining chromosome. It calculates a null distribution of variation from neutral coalescent simulations involving recombination, and uses this to calculate the probability of observing a threshold ratio of segregating variants. The @Nielsen2005a version of \gls{clr} is similar to Kim and Stephan's test but creates the null distribution from the data itself, instead of using the neutral population model. A key consideration for the use of \gls{clr} is that it is very sensitive to \gls{snp} ascertainment bias [@Chen2010].

#### Cross population composite likelihoods ratio

The cross population composite likelihoods ratio (XP-CLR) is an extension of the \gls{clr} test developed by @Nielsen2009 that scans for multi-locus allele frequency differentiation between two populations and then tests if these frequency differences are too recent to be consistent with neutrality based on the \gls{ld} in the region [@Chen2010]. A down-weighting of \glspl{snp} that are in perfect or high \gls{ld} is performed because they essentially provide the same information. By doing this it is hoped that false positives are reduced, since the correlation of marginal likelihood terms in the composite likelihood function is often overlooked and leads to false positives. Normalised XP-CLR scores should be resistant to variation in demographic events, however a conservative approach of using rank-order of scores across the genome is usually taken. Unlike single population \gls{clr} and similar to \gls{xpehh}, XP-CLR is resistant to \gls{snp} ascertainment bias because population differentiation is not affected by ascertainment bias. There is also a strong correlation between genes identified with XP-CLR and \gls{xpehh}. XP-CLR is also considered an extension of F~ST~ and is beginning to be used as a replacement for investigating selection in animal studies (e.g., Cattle – @Lee2014; Goats - @Benjelloun2015)

#### Composite of multiple signals

Composite of Multiple Signals (CMS) is a combination metric composed of \gls{ihs}, \gls{xpehh}, F~ST~, \gls{ddaf}, and $\Delta$\gls{ihh}. $\Delta$\gls{ihh} is used to compare the actual length of the haplotype. CMS was created to take advantage of the combined power gained from using all metrics together to be able to detect the source of the selective signal better than any test individually could [@Grossman2010]. In CMS it was found that \gls{xpehh} and F~ST~ contributed mostly to spatially locating the causal variant whereas \gls{ihs}, \gls{ddaf}, and $\Delta$\gls{ihh} did poorly in spatially locating the selective signal but performed well at identifying the precise causal variant.

#### Meta-analysis of multiple tests

Meta-analysis of Multiple Tests (meta-SS; @Utsunomiya2013; @Utsunomiya2015) is only able to combine selection statistics that generate P-values. It uses an adapted version of a weighted Stouffer method [@stouffer1949american;@Zaykin2011] to combine Z-transformed P-values. For each marker from each test _i_ the respective P-value is transformed into a Z score by Z~i~ = - $\phi$ -1(1 - $\pi$), where $\phi$ is the normal cumulative density function. Each test is weighted (w~i~) to 1/n where n is the number of comparisons. The combined statistic of $k$ tests, for each \gls{snp} in each population is defined in equation \@ref(eq:metass).

\begin{equation} 
meta\-SS = \frac{\sum\limits_{i =1}^k {\omega_i Z_i}}{\sqrt{\sum\limits_{i =1}^k {\omega_i}^2}}
(\#eq:metass)
\end{equation}


#### Composite of selection signals

\Gls{css} is similar to meta-SS, in that it is a method for combining the results from multiple selection tests but is not limited to tests that produce P-values [@Randhawa2014]. \Gls{css} currently uses F~ST~, \gls{xpehh}, and \gls{ddaf} or change in selected allele frequency, and combines them by taking the test statistic for each method across each \gls{snp}(1,…,n) then obtaining the rank of each statistic across each \gls{snp}(1 … n) and converting these into fractional ranks so they lie between 0 and 1, that is, 1/(n+1) to n/(n+1). By doing this the magnitudes of the original statistics are not used, thus increasing the robustness. The fractional ranks are converted to Z-statistics and the average z-values are calculated at each \gls{snp} position. P-values are obtained from the distribution of the means with the -log10(p) being the \gls{css} value. The \gls{css} is plotted against genomic position with an excess in \gls{css} showing a common signal between the multiple test statistics.

### Power

Power to detect selected sites can be very limited if the site is not an outlier from empirical methods, such as simulations or using permutations of the data, as they won't 'stand out' [@Teshima2006]. A key consideration when selecting a particular method for detecting selection is the limitations the method has on power to detect a selective event, such as starting and final allele frequency of the variant after a sweep. \Gls{td} is powered for low and intermediate frequency variants [@Simonsen1995] whereas \gls{fwh} is powered for intermediate and high frequency variants. Some methods such as \gls{ihs}, are good for detection of middle frequency variants. \Gls{xpehh}, on the other hand is powered at the fixation ends of the spectrum. \Gls{xpehh}, because it is correlated with \gls{ld}, lacks power to detect ancient selective events [@Chen2010].  Sample size is another factor to consider. \Gls{ihs} has modest reduction of power when reducing samples until approximately 40 samples of single chromosomes. \Gls{xpehh} can maintain power down to about 20 samples of single chromosomes, so long as the reference population is a fixed size. The grouping of similar genetic populations may be an option to increase power with low sample sizes [@pickrell2009signals]). For F~ST~, power to detect selection should be sufficient if sample size is > 1/F~ST~ [@Bhatia2013a; @Bhatia2011]

Power of a statistic also needs to be considered for the type of selective sweep that may have occurred, such is the case with \gls{nsl}, which has similar power to detect selection to \gls{ihs}, but performs better for larger starting allele frequencies. It also outperforms \gls{ihs} in power in both hard and soft sweep scenarios [@Ferrer-Admetlla2014]. When it comes to picking up demographic events, haplotype tests are useful at detecting recent and more moderate bottlenecks, frequency spectrum tests (such as \gls{fwh}/ \gls{td}) have the best power for detecting moderately ancestral severe bottlenecks [@Depaulis2003], whereas \gls{ze} and \gls{flf} have the most power during population expansion [@Zeng2006;@Ramirez-Soriano2008].

### Challenges

When scanning genome-wide in a population (such as in a \gls{gwas}, or genome-wide selection scan), the variability in the genome also has population demographic events that have acted upon it, such as population bottlenecks, migration, or growth. Mutation and recombination are also acting on the genome. Methods based on comparing non-synonymous to synonymous variants are relatively unaffected, but, methods based on allele frequency are susceptible to detecting population demographic events, along with mutation and recombination rate changes [@Nielsen2009].  A meta population is a population which can be subdivided into many different sub-populations among which there is some pattern of migration, extinction, and recolonization [@Wakeley2001]. Bottlenecks with a meta-population model can lead to a high frequency of derived alleles, greater than would be expected to arise through a neutral model [@Jensen2005] and as previously mentioned by @fay2000hitchhiking about high frequency derived alleles being a unique signature of selection, it not actually unique [@Przeworski2002].

\Gls{snp} ascertainment bias is caused by the non-random sampling of SNPs on array chips. This is a problem that affects the selection detection methods to different extents, with frequency spectrum methods being the most susceptible. \Gls{td} is biased upwards due to the \gls{snp} discovery process having ascertainment biases, which leads to an excess of intermediate frequency alleles in the sample [@Kelley2006]. Tests that rely on long haplotypes are less susceptible to ascertainment bias [@sabeti2006positive] but because long haplotypes can be quickly broken down through recombination, they are only useful for short time periods. \Gls{snp} ascertainment bias can be overcome with 
the use of whole genome sequencing [@Albrechtsen2010]. 

Many of these statistics were developed to investigate selection at a particular locus, at a time when applying the statistics across a genome was not possible. Up until recently, genome scans have been performed using \gls{snp} genotypes identified through a \gls{snp} discovery process, which means they are not random samples – which would include non \gls{snp} genotypes (such as \glspl{indel}) and create an ascertainment bias [@Nielsen2005a]

Identification of exactly what is deemed as being "significantly selected"" has issues. The current practice is to take outlier loci, usually from an empirical distribution of the selection statistic and report these as significant. This is not necessarily indicative of selection, as cut off levels are created subjectively rather than derived from the model [@Qanbari2012a]. Using an empirical approach will result in many false positives, however this is considered an acceptable approach for selecting candidate genes so long as it is realised that the false positive rate is high [@Kelley2006].

In order to use haplotypic methods, (re)construction of haplotypes is required. Currently this involves the need for the phasing of haplotypes from genomic data. Phasing is the process of determining which alleles are found together on the same physical chromosomes. There are a number of ways that this can be done, such as through a pedigree, through the use of large population data, or using the reads from next generation sequencing. Phasing through pedigree information is utilised in most agricultural species, whereas for human populations the population genotypes are largely from unrelated individuals. Phasing of unrelated individuals currently relies on probabilistic methods [@Browning2009; @Delaneau2012; @Delaneau2013] in order to make the best guess at each individual haplotype. Newer methods have been developed to use the reads from next generation sequencing, and are able to trace a haplotype across multiple overlapping reads and be used to complement other probabilistic phasing methods, however these read-backed methods are dependent on the depth of coverage and read quality [@Delaneau2014].


## Improving GWAS

\Gls{gwas} are a method to scan genomes, usually \gls{snp} genotype data, for association with a phenotype. The strength of \gls{gwas} is that prior knowledge of loci is not required, and can be used as a tool for hypothesis generation. \Gls{gwas}, because of the large number of association tests, require large cohorts of samples to achieve sufficient power for statistical significance. As the effect size for a variant decreases, larger sample sizes are also needed, as power is proportional to the square of the effect size [@Korte2013; @Hemani2013]. Limitations in recruiting sufficient numbers of participants for \gls{gwas} therefore requires new ways of increasing power to detect smaller effects without needing to increase sample size, as outlined in this section. The goal of being able to prioritise \gls{gwas} results by using additional information, such as incorporating \gls{ld} or selection statistics by using a weighting system to increase the power of a \gls{gwas}, has been suggested [@Ayodo2007; @Roeder2006]. When \gls{ld} between observed \glspl{snp} and causal variants is high, such as in the case of selection, greater power in \gls{gwas} can be achieved by focusing on searching for non-additive variance [@Hemani2013].

The basis of \gls{gwas} studies is to identify \glspl{snp} that are associated with differences in the mean of a trait. These studies don't necessarily identify the causal \gls{snp} but instead identify a marker for the causal variant, usually in \gls{ld}. @Fisher1919 showed that total genetic variance could be split into components of additive, dominance and epistatic effects. The additive component has been shown to contribute the most to the overall genetic variance, with epistatic variance contributing little in an outbred population [@Hill2008]. Complex traits gain their complexity from being the sum of interactions from many small effect loci, or the product of non-additive interactions between loci and the environment, making it challenging to find the exact contribution of a locus to a trait [@Fu2013a]. These small, true effect loci can end up non-significant due to the noise in the data, but might be able to retrieved using methods that find non-additive effects. 

Explained heritability is defined as the proportion of heritability (or phenotypic variance) explained by a collection of genetic variants, with the "missing" portion referring to the unaccounted for genetic variance that has been estimated through family studies. When the ratio of explained heritability to total heritability is < 1, the unexplained component is defined as missing heritability [@Zuk2012]. The use of methods that capture additional loci with non-additive effects should be able to explain more of missing heritability.

<!-- > The total heritability of a trait is inferred from population data and twin studies, with what is know as missing heritability -->
<!-- infers there are additional loci that contribute to the phenotypic variance that are yet to be discovered.  -->


<!-- ### Non-additive Variance -->

<!--  The incorporation of selection statistics or a \gls{vgwas} results has the potential to inform \gls{gwas} since these extra results capture some of the non-additive effects and variance that may be influencing a locus.  To attempt to capture this non-additive variance, \gls{vgwas} can be performed. This uses the phenotypic variance, instead of mean, in the phenotype by genotype regression model. This method captures not just additive effects but also alternative methods of genetic influence [@Pare2010]. Additional work is required to determine if it is gene-gene or gene-environment interactions that are the cause of this variation, such as a gene by gene or gene by environment interaction analysis focussing on the top results from the \gls{vgwas}. -->

<!-- Important issues to bear in mind when using \gls{vgwas} include the size of a dataset, imbalance between cases and controls, and test statistics becoming inflated when minor allele frequency is small [@Yang2012]. A large dataset is often needed in order to be able to detect small differences in variance between groups. Imbalance in the data can appear as too many cases or controls but also in allele frequency. If there are not enough samples in an affection or allele group then the test statistic gets inflated due to sampling bias. For rare alleles this can require very large sample sizes to compensate. -->

<!-- #### Transformation of Variance -->

<!-- A key consideration when investigating the variance of a dataset is the importance of ensuring that the differences in variance cannot be explained by performing transformations on the data, such as log or inverse normal transformations [@Sun2013]. @Shen2013 agreed with the principle of doing transformations but note that it is also very important for the transformations to make biological sense. For example, even though the scales are arbitrary -->
<!-- biologically for \gls{bmi} or height it doesn't make sense to have a larger difference between a \gls{bmi} of 21 and 22 than between 24 and 25. -->

### Selection and GWAS

The significance of \gls{gwas} results is greatly increased when selection statistics and association studies are combined, this has been applied in a \gls{gwas} for malarial resistance variants [@Ayodo2007]. A smaller sample size can be used if there is evidence of positive selection because the beneficial allele will rise in prevalence, and will be in high \gls{ld} with other variants (due to genetic hitch-hiking) which can act as proxy markers [@Karlsson2014]. Complex disease traits compared to Mendelian disease associated traits have a bias towards larger mouse-human K~a~/K~s~ ratios, by comparing the K~a~/K~s~ ratios for ortholog genes between mouse and human, which suggests that evidence of positive selection could be utilised in identifying variation associated with complex disease [@Thomas2004].

### Pathway analysis

A biological pathway is a dependency graph detailing the genes or proteins involved in a series of biochemical reactions, such as a metabolic pathway like the Krebs cycle. There are curated databases of many of the biological pathways used by living organisms, such as \gls{kegg} [@Kanehisa2017] or Reactome [@Fabregat2018]. Pathways have been hypothesised to be involved in the genic adaptation of populations since altering a gene in a biochemical pathway is likely to alter different genes in the same pathway [@Bigham2010]. The use of pathway analysis rather than focusing on individual \glspl{snp} is another way to increase the power of a \gls{gwas} [@Jia2011]. This increase in power is largely obtained through a reduction in dimensionality in ways that attempt to remove noise from the underlying biological signal. There are however limitations to the effectiveness of using such an approach, such as the selection of pathway database, density of \glspl{snp} per gene, or gene length and boundaries. The pathway database is important because pathways can differ based on curator, age of the database - with older databases not incorporating current knowledge, or differing focuses. \Glspl{snp} per gene, and gene length or boundaries all affect the pathway analysis as smaller genes or \gls{snp} density reduces the observations for the gene compared to large genes or genes with high \gls{snp} density. Pathways that involve many large genes also have a higher chance of a \gls{snp} having an extreme statistic value under a null model, because overall they are more likely to have more \glspl{snp} than pathways containing small length genes.

## Applications

### Genome-wide selection

Genome wide selection in humans has been investigated by @sabeti2006positive, @pickrell2009signals,@voight2006map, @Hancock2008, the 1000 Genomes Project Consortium [@1KGP2010;@1KGP2012;@1KGP2015snp], @Grossman2010, @Colonna2014, and @Mallick2016. The population datasets used involved the HapMap Project [@Hapmap2005], the Human Genome Diversity Project [@Cann2002; @Rosenburg2002] and the 1000 Genomes Project (populations are defined in section \@ref(datasets); @1KGP2010). The 1000 Genomes Project Consortium [@1KGP2010; @1KGP2012; @1KGP2015snp] reported genome wide diversification between main population groups based on F~ST~ and \gls{ddaf}. Genome wide mean F~ST~ was found to only be a maximum of 8%, indicating overall little genetic differentiation between populations, however several thousand SNPs had large F~ST~ indicating local population adaptation [@1KGP2010]. The analysis also found 139 non-synonymous SNPs with large frequency differences between populations. Investigation of the \gls{1kgp} data set was also performed by @Colonna2014. Using \gls{ddaf}, it was found that sites of high population differentiation (\gls{ddaf} > 0.7 between and > 0.25 within continents) clustered together, the same was found for sites of low differentiation. Genes containing highly differentiated sites were likely to also have additional evidence of positive selection in a third. There was also a strong association of highly differentiated sites with genes and gene regulatory elements.

Metabolic syndrome and gout have associations and interactions with genes in metabolic pathways and the immune system [@Osborn2012]. The categories of genes that have displayed evidence of selection from the previously mentioned genome-wide selection scans include: metabolism of sugars [@tang2007new; @voight2006map], drug metabolism [@tang2007new], immune system [@Grossman2010; @tang2007new], and climate adaptation [@Hancock2008]. Genes that play a role in adaptation to climate extremes are likely to play a role in metabolic disorders that make up metabolic syndrome [@Hancock2008]. In contrast, variation in metabolic related genes can lead to elite phenotypes in athletes, showing that metabolic genes also impact on fitness [@Ahmetov2009a].

#### Selection in Polynesia

The inclusion of Oceanic populations in multi-population studies has had a focus on genetic structure, measured via F~ST~ for differentiation to investigate population ancestry, rather than selection [@Friedlaender2008; @Tennessen2011]. The migratory history of Polynesian populations suggests that the points of differentiation are likely the events possibly occurring since the out of Africa dispersal ~60,000 years ago [@Soares2012] with settlement of Near Oceania ~40,000 years ago, and Remote Oceania ~3,100 years ago [@Matisoo-Smith2004] and the Polynesian Triangle (bounded by New Zealand, Easter Island, and Hawaii) ~1,000-1,200 years ago [@Wilmshurst2011]. The time frame for these events means that nearly all of the previously mentioned methods would still be within their powered time frames to detect selective events. The haplotypic methods would possibly not be able to detect selective events as old as the Africa dispersal and may be at the fringe of the time period for the Near Oceania settlement. 

Large population comparison studies have used Oceanic populations, but analyses were limited to Melanesian and Papuan populations because the cohort used was the Human Genome Diversity Project [@Cann2002] which didn't include Polynesian populations. The Melanesian and Papuan populations are often used to infer the genetics of Polynesian populations. This extrapolation doesn't take into account the current ancestry predictions of Polynesians populations being more closely related to Asian/Taiwanese Aboriginal populations than Melanesian groups, or the presence of European admixture [@Friedlaender2008]. Genomic DNA and mitochondrial DNA suggest Polynesians have East Asian ancestry, with substantial male Melanesian admixture [@Kayser2008a]. There has been limited analysis of selection in Polynesian populations beyond @Kimura2008, where 24 Tongan samples in conjunction with other populations were analysed using both F~ST~ and a modified \gls{ehh} test. Genomic regions with evidence of having been selected in the Tongan samples were potential candidates for increased fat, muscle and bone masses [@Kimura2008].

A few candidate-genes for selection which also have been associated with obesity and \gls{t2d} in Polynesian populations have been investigated, such as the _PPARGC1A_ [@Myles2011] and _CREBRF_ [@Minster2016] genes. The "Thrifty-gene hypothesis" [@Neel1962] has been the reasoning behind why these loci may have undergone selection. The hypothesis was originally conceived to attempt to explain the difference in prevalence of \gls{t2d}, that is, given that there should be a strong selective pressure against it, there must be some genetic advantage. This hypothesis has been challenged in the Polynesian context as fitting a particular narrative [@Gosling2014; @Gosling2015; @Cadzow2016] with an alternative hypothesis being a link between the metabolic diseases and the innate immune system that is under selective pressure via an infectious agent, such as malaria, leading to disease susceptibility in these populations [@Gosling2015].


### Selection of metabolic syndrome and urate {#goutMetSyn}


The genetic influence of urate and gout is found in the under-excretion of serum urate. There are 28 loci that have been identified as having a role in \gls{hu} and gout, of which, two (_ABCG2_ and _SLC2A9_) are responsible 3.4% of the variability of urate [@Kottgen2013]. _SLC2A9_ encodes a urate transporter expressed on both the apical and baso-lateral membranes in the proximal tubule of the kidney and reabsorbs urate [@Mandal2015]. _ABCG2_ encodes an ATP powered urate transporter and is responsible for urate excretion mostly in the gut [@Maliepaard2001; @Mandal2015]. Twenty-six other loci associated with serum urate explain a further 3.6% of the variance [@Kottgen2013]. The heritability of serum uric acid ranges from 40 - 70% [@Nath2007; @Yang2005]. This difference in explained variability and total heritability leaves a large component yet to be explained. This missing heritability could be reduced through identifying additional variance attributed to non-additive effects.

<!-- Heritability of \gls{bmi} ranges from 45% to 84% in twin studies [@Schousboe2003]. -->

There have been multiple genetic loci associated with \gls{t2d} [@Billings2010] and \gls{bmi} [@Locke2015], however, @Chen2010 found no evidence for enrichment in selection for \gls{bmi}, but found significant enrichment for \gls{t2d}. The genes associated with \gls{t2d} have been found to be differentiated by population, however the individual \glspl{snp} are often not [@pickrell2009signals]. @Koh2014 identified six \glspl{snp} in or near nine genes that had been reported as \gls{t2d} associated and overlapped with evidence of positive selection in East Asian populations. Common variants that have been found in European populations associating with obesity don't replicate in the Samoan population [@Karns2012], although this is likely due to power. The non-replication of European variants in Polynesians and the high prevalence of obesity in Polynesian populations suggest there is likely to be some population specific variants that are yet to be identified, similar to the Polynesian specific variant in _CREBRF_ (rs373863828), associated with obesity and protection for \gls{t2d}, identified in the Samoan population [@Minster2016]. Rare variants tend to be recent and therefore geographically restricted [@1KGP2012].

The benefits of urate are in line with the categories found to be enriched by selection in genome wide scans. Urate is an anti-oxidant in the blood, acting to protect not just erythrocytes but also T and B lymphocytes and macrophages [@Ames1981]. Urate has also been hypothesised to have had a survival advantage during the Miocene period, when a series of mutations in urate oxidase occurred, leading to elevated urate. \Gls{hu} enables blood pressure to be maintained under low salt dietary conditions [@Watanabe2002]. Serum urate is also a potent anti-oxidant which accounts for >50% of the anti-oxidant activity of the blood [@Glantzounis2005; @Parmar2009]. Later in life elevated serum urate is associated with increased cognitive function [@Euser2009]. Urate crystals activate the NLRP3 inflammasome which plays an important role in systemic infection and sepsis [@Opitz2009]. The role of urate in these situations shows that while \gls{hu} is now considered as contributing to disease (such as gout), in the past it may have influenced longevity and survival, therefore increasing reproductive success and being a phenotype that has undergone selection. 

An elevated prevalence of metabolic syndrome co-morbidities and hyperuricaemia against a unique genetic background of Polynesian populations provides an important opportunity to study the role and effects that selection might have played on these metabolic conditions.

## Research purpose and aims

This thesis seeks to understand selection in the context of Polynesian populations, and the effect it may have played in the metabolic disease burden that is present in these populations. There are three research chapters, each with their own aims. In Chapter \@ref(selectionResults): "Positive Selection in Polynesian Populations", the aims were to discover and characterise regions that have undergone selection in Polynesian populations. In Chapter \@ref(clustering): "Clustering of Selection Statistics", the aims were to investigate potential shared ancestry regions that are relevant to urate and metabolic disease. And in Chapter \@ref(selGwas): "Selection and Association Studies", the aims were to investigate the use of gout definition in genetic association studies, and the use of selection statistics in conjunction with \gls{gwas} results. 

It is hoped that the investigation of genetic selection in Polynesian populations will further understanding of the cause of health disparities due to metabolic disease in these populations.


<!-- ## Extras -->

<!-- A nice little summary of some of the stats http://citeseerx.ist.psu.edu/viewdoc/download?rep=rep1&type=pdf&doi=10.1.1.210.4017 -->






<!--
A negative D is means there is an excess of external branches, whereas a positive means there is a deficency

without an outgroup you can't lcoate the root of the tree

Based on a 'geneology' of genes. older part will have internal branches, younger part will be mostly external branches

Negative/purifying selection:
-excess in mutations in external branches because deleterious alleles are present in low freq

excess of mutations in external branches if advantageous allele recently fixed then the majority of mutations are expected to be young

F can be thought of as the normalised difference between the average number of segregating sites ($\pi$) between 2 sequences and the number of singletons 
-->


<!-- TD and FLD + haplotype homozygosity have lower power to detect thanfwh in admixted populations - Lohmueller2011 \cite{Lohmueller2011b} -->

<!-- ### Urate -->

<!-- Uric acid is a by product of purine metabolism.  -->
<!-- Inflammatory response is triggered when the monosodium urate crystals interact with the innate and adaptive immune response and result in the generation of IL-1 @Rock2013 -->


<!-- ### extra papers -->

<!-- - \cite{Blekhman2008} -->
<!-- - \cite{Blekhman2008} -->

<!-- ### things to update -->

<!-- [] need some newer references eg cell paper about diving adapation, minster2017 -->

<!-- [x] update to have the appropriate statistics that were actually used -->

<!-- [] section about introgression? -->

<!-- [x] remove vgwas? -->

<!-- [x] zengs e -->

<!-- [] cadzow2016 and minster2017 paper for metabolic selection in polynesian -->

<!-- [] new papers -->


<!-- papers that have used empirical distribution: -->

<!-- There is precedent in using the empirical distribution to determine significance, instead of relying purely on the numerical value of the statistic being used. This approach however does need to acknowledge false positives, and also that true selection signals may be lost in the noise. The empirical distribution approach has been used for multiple studies [@ref; @ref; @ref]. -->

<!-- - Teshima2006 -->
<!-- - Xue2009 -->
<!-- - Hider2013 (fwh,fld,flf,td,ihs) -->
<!-- - Joannalagadda2017 (fwh,td,ihs) -->
<!-- - akey2002interrogating (Fst) -->
<!-- - Akey2010 (d~i~ Fst derived stat) -->


<!--
variant prioritzation:  Eilbeck2017
effect size vs allele frequency: Park2011
-->


<!--
Neutrality vs Selection
TD = Neutrality
FWh = Neutrality?
Fu Li D  = Neutrality?
Fu Li F = Neutrality?
Zeng E = Neutrality?
iHS = selection
nsl = selection
xpehh = selection

fst = pop differentiation


-->