# Methods/Data
```{r, echo = FALSE, message = FALSE, warning= FALSE}
source("scripts/methods_tables.R")
```

## Methods

### Phasing

-axiom
-omni
-coreExome

### SelectionTools



Selection statistics that are generated:

Window based
- Tajima's D (vcftools) [@Tajima1989]
- Fay and Wu's H [@fay2000hitchhiking] (variscan)
- Fu and Li's D [@Fu1993] (variscan)
- Fu and Li's F [@Fu1993] (variscan)
- Eta (variscan) total (minimum number) number of mutations ($\eta$) 
- Eta E (variscan) number of external mutations ie number of derived singletons
- Pi (variscan) nucleotide diversity ($\pi$)
- S (variscan) number of segregating sites
- mean F~ST~ (vcftools)

Haplotype based
- sL0
- sL1
- unstandardised nSL
- standardised nSL
- ihh0
- ihh1
- unstandardised iHS
- standardised iHS
- iHH population 1
- iHH population 2
- unstandardised XP-EHH
- standardised XP-EHH

Frequency
- Minor allele frequency
- Derived allele frequency

Gene
- Ka/Ks




#### Selection database creation

#### PCA - eigenstrat
```
#combine the chromosomes
bcftools concat -O z -o combined_super_all.vcf.gz --threads 10 $(ls /Volumes/BiochemXsan/scratch/merrimanlab/murray/working_dir/coreExome_selection/NZ_coreExome_1kgp/data/dbload/fixed_vcf/combined/combined_super_chr*vcf.gz | sort -t r -k12n)

#find the independent markers
plink2 --vcf combined_super_all.vcf.gz --maf 0.1 --indep-pairwise 50 5 0.2 --out combined_super_all
touch cases.txt
plink2 --vcf combined_super_all.vcf.gz --extract combined_super_all.prune.in --recode --out combined_super_all_indep --make-pheno cases.txt '*'
# create the eigenstrat file
echo -e "genotype: combined_super_all_indep.ped\nsnpname: combined_super_all_indep.map\nindivname: combined_super_all_indep.ped\noutputformat: EIGENSTRAT\ngenotypeoutputname: combined_super_all_indep.eigenstratgeno\nsnpoutname: combined_super_all_indep.snp\nindivoutname: combined_super_all_indep.ind\nfamilynames: NO" > par.PED.EIGENSTRAT

convertf -p par.PED.EIGENSTRAT > eigen.log
smartpca.perl -i combined_super_all_indep.eigenstratgeno -a combined_super_all_indep.snp -b combined_super_all_indep.ind  -o combined_super_all_indep_eigen.pca -p smartpca.perl -i combined_super_all_indep.eigenstratgeno -a combined_super_all_indep.snp -b combined_super_all_indep.ind  -o combined_super_all_indep_eigen.pca -p combined_super_all_indep_eigen -e combined_super_all_indep_eigen.eval -l combined_all_indep_eigen.log -m 0
```
#### admixture analysis
```
bcftools concat -O z -o combined_super_all.vcf.gz --threads 10 $(ls /Volumes/BiochemXsan/scratch/merrimanlab/murray/working_dir/coreExome_selection/NZ_coreExome_1kgp/data/dbload/fixed_vcf/combined/combined_super_chr*vcf.gz | sort -t r -k12n)
plink2 --vcf combined_super_all.vcf.gz --indep-pairwise 50 5 0.1 --out combined_super_all_admix

plink2 --vcf combined_super_all.vcf.gz --make-bed --extract combined_super_all_admix.prune.in --out core_exome_admix
for K in $(seq 1 15) ; do admixture -s 123456 --cv core_exome_admix.bed $K -j20 | tee log${K}.out ; done
grep CV log* | cut -d':' -f1,3 |tr -d ':' | sed 's/log\|\.out//g' > CV_error.txt
```

### GWAS

Uk Biobank GWAS


####

#### Heritability Analysis

10000 controls were randomly selected. The same 10000 samples were then combined with each different gout criteria and heritiability estimates were obtained by first creating the genetic relationship matrix and then the variance of gout 
```{r, engine='bash', eval=FALSE}
# load fam into R
# sample(10000, fam[AFF == controls]) -> controls.txt
# awk '{if($6 == 2){print $1"\t"$2}}' < condition.fam | cat - controls.txt > ids_keep.txt

# Subset samples
for i in $(seq 1 22)
do 
  plink2 --bed /Volumes/BiocArchive/archive/merrimanlab/central_datasets/ukbiobank/plink/chr${i}impv1.bed \
            --bim /Volumes/BiocArchive/archive/merrimanlab/central_datasets/ukbiobank/plink/chr${i}impv1.bim_1kg_marker \
            --fam /Volumes/BiocArchive/archive/merrimanlab/central_datasets/ukbiobank/plink/chrallimpv1.fam_allgout_allcontrols \
            --hwe 0.000001 \
            --maf 0.01 \
            --keep ult_ids_keep.txt \
            --make-bed \
            --out gout_chr${i} \
done
```

Calculate the genetic relationship matrix
```{r, engine = 'bash', eval = FALSE}
for i in $(seq 1 22)
do 
  gcta1.26.0 \
    --bfile gout_chr${i} \
    --chr ${i} \
    --make-grm-bin \
    --out ukbio_10ksample_grm_chr${i} \
    --thread-num 16 > grm_${i}.log  \
done
```

Calculate the genetic variance explained using a general population gout prevalence of 2%
```{r, engine = 'bash',eval = FALSE}
for i in $(seq 1 22) 
do 
  gcta1.26.0 \
    --grm ukbio_10ksample_grm_chr${i} \
    --pheno phenos.txt  \
    --prevalence 0.02 \
    --out var_chr${i} \
    --reml \
    --chr ${i} \
    --thread-num 16 \
done

```

### Variant Annotation
snpEff


### Pathway Analysis


## Datasets

### 1000 Genomes Project Phase 3


@misc{1kgp_ftp,
keywords = {1KGP},
mendeley-groups = {PPARGC1Apaper,Datasets},
mendeley-tags = {1KGP},
title = {{1000 Genomes Project FTP}},
url = {ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/},
urldate = {2015-08-18}
}

### Genetics of Gout in Aoteroa

The genetics of Gout in Aoteroa study is a case-control cohort with recruitment mainly from the Auckland, Wellington, Christchurch and Dunedin regions of New Zealand. Individuals are asked to fill in a questionaire regarding demographics information, clinical information, and food consumption. 

Individuals from this study have been genotyped on at least one of the following platforms: an affymetrix AXIOM SNP chip, an Illumina OMNIexpress SNP chip, an Illumina CoreExome SNP chip, whole genome low-coverage resequenced, or exome resequence data.

#### Axiom

- about
- population

| | obs missing (n) | Māori AXIOM
---|---|---
Number | - | 71 
Sex (% male) | - | 64.7
Gout cases (%) | - | 49.3
Mean BMI (range) | 1 | 34.5 (22.1 to 56.1)
Diabetes (%) | 3 | 21.1
Mean grandparent ancestry (range) | - | 0.916 (0.625 to 1.00)

Mean grandparent ancestry is the proportion of grandparents self-reported as belonging to the ethnicity group BMI body mass index


- markers
- density


#### Omni


| | obs missing (n) | Samoan OMNIexpress
---|---|---
Number | - | 96
Sex (% male) | - | 74
Gout cases (%) | - | 50.0
Mean BMI (range) | - | 36.1 (20.5 to 93.4)
Diabetes (%) | - | 12.5
Mean grandparent ancestry (range) | - | 0.997 (0.875 to 1.00)

Mean grandparent ancestry is the proportion of grandparents self-reported as belonging to the ethnicity group BMI body mass index


- about
- population
- markers
- density

#### CoreExome

The Infinium CoreExome-24 beadchip is a genotyping platform available from Illumina. It comprises a core set of 551,839 markers with a mean density of XXX.  Genotyping quality control was performed following the protocol from @guo2014. Illumina’s Genome Studio vXXX genotyping module vXXXwas used for the initial calling of genotypes. Samples were exported from Genome Studio that had a call rate > 98%, markers were zeroed if they were < 95%. 

- removed gender mismatches
- checked relatedness of samples
- checked ancestry with PCA
- HWE by population
- checked heterozygosity and inbreeding outliers
- checked genotype consistency


Selection of samples to form the populations used in the selection analyses
Gout affection was determined by...
A single individual was randomly selected from family groups identified through identity by state analysis. Individuals who had a mismatch between self reported and genetic ancestry were also removed. Principle components 2 and 4 were found to provide the greatest separation between Europeans, East Polynesians, and West Polynesians. Individuals reporting 4 grandparents of Caucasian, NZ Maori, CI Maori, Samoan, and Tongan ancestry were used to define the threshold values. Thresholds were calculated using mean +/- 2 SD for PC2 and PC4 for each population group. Once thresholds were calculated all samples were filtered so that only those that lay within a boundary remained. Remaining samples were then randomly sampled to form groups of 100 individuals, where possible trying to match general population prevalence for gout.

Marker Categories | Number of Markers
---|---
In RefSeq Genes | 374,258 (399,009 within 10 kb)
In RefSeq Exons | 269,186
In RefSeq Promoter Regions | 13,990
In ADME Genes | 11,276 (12,455 within 10 kb)
In ADME Exons | 9,314
MHC (Extended MHC) | 3,940 (5,628)
Overlap with Genes in COSMIC | 365,135
Overlap with Genes in Gene Ontology | 96,331
Nonsense Markers | 14,994
Missense Markers | 219,162
Synonymous Markers | 14,747
Silent Markers | 8,850
Mitochondrial Markers |351
Indels | 12,436

mean call rate
mean missing
number of snps passing qc
number of snps used for selection


(http://www.illumina.com/content/dam/illumina-marketing/documents/products/datasheets/datasheet_human_core_exome_beadchip.pdf, accessed 16 Jan 2017)


Clinical Information


\newpage
\blandscape
```{r, echo = FALSE, results = 'asis', message = FALSE, warning= FALSE}
 knitr::kable(ce_clinical_table(), longtable = TRUE, format = 'pandoc', booktabs = TRUE , caption = 'Clinical information for each core Exome population')
```



\elandscape
\newpage

```{r, echo = FALSE, results = 'asis', message = FALSE, warning=FALSE}
knitr::kable(ce_populations_table(), longtable = TRUE, format = 'pandoc', booktabs = TRUE , caption = 'Clinical information for New Zealand Populations in selection analysis')
```


- about
- population
- markers
- density



#### Polynesian Sequences

n | ~90
mean coverage |  Y x


### Reference datasets

\newpage
\blandscape
```{r, echo = FALSE, results = 'asis'}
tab <- data.frame(Name = c("1000 Genomes Project Phase 3 release 5","1000 Genomes Project Phase 3 release 5","1000 Genomes Phase 3 phased haplotypes", "Human Genome Reference FASTA", "Human Ancestral Allele FASTA"),
                  URL = c("","http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/","https://mathgen.stats.ox.ac.uk/impute/1000GP_Phase3.html", "ftp://ftp.1000genomes.ebi.ac.uk:21/vol1/ftp/technical/reference/phase2_reference_assembly_sequence/hs37d5.fa.gz", "ftp://ftp.ensembl.org/pub/release-66/fasta/ancestral_alleles/homo_sapiens_ancestor_GRCh37_e66.tar.bz" ), Date = c("17 Sept 2014","20 Mar 2017","14 Dec 2014", "5 Aug 2014", "6 Aug 2014"))
knitr::kable(tab, longtable=TRUE, caption = "Reference Datasets")
```



\elandscape
\newpage