# SelectionTools Pipeline

```{r, include = FALSE, echo = FALSE}
knitr::opts_chunk$set(echo = FALSE)
library(xtable)
library(tidyverse)
source('scripts/multiplot.R')
source('scripts/03_selectionpipeline.R')

```

Part of this work was originally published in @Cadzow2016 but I have updated the selection analysis to make use of a new dataset that became available after publication.


## Introduction



In this chapter I investigate regions of the genome that exhibit signatures of selection in Polynesian populations. 

Much of the early work in looking at selection was focused at the gene level [@Wright1951;@Fisher] and the methods that were developed were intended for few loci, or limited sequence from a single region [@ref;@ref;@ref]. This was largely due to the high cost and limited availability of whole genome chip or sequence data. It wasn't until after the completion of the human genome project in 2001 and the advent of 'next generation sequencing' where the cost of full genome data rapidly dropped and there became an abundance of whole genome sequence data [@ref]. There were a few large efforts to characterise human variation, the first main project was the HapMap project [@Hapmap2005], which was followed on by the 1000 Genomes Project [@1KGP2010; @1KGP2012; @1KGP2015snp]. Alongside these variation projects were a few projects that aimed to focus on the diversity of humans, this was the Human Genome Diversity Project [@Cann2002; @Rosenburg2002], and later the Simons Genome Diversity Project [@Mallick2016].

One of the methods to still use the selection and neutrality tests with the genome-wide data was to use the empirical distribution [@ref; @Clarke?]

over the last decade there has been a steady flow of studies into genome-wide selection. The main samples for this have been a part of the 1000 Genomes Project, which sample over 2500 individuals for full genome (re)sequencing from five 'super' population groups. This dataset, while capturing a large representation of the human population variation, does not have representation for the people's of the pacific [@ref]. 

More recently there have been a studies that have looked at selection in Polynesian populations but with a focus on a few candidate loci, not at a genome-wide scale [@Cadzow2016; @Minster2016].

Thrifty hypothesis

- @Neel1962
- @Gosling2014
- @Gosling2015
- @Minster2016
- @Cadzow2016


Loci associated with T2D have been found to have evidence of having undergone selection [@pickrell2009; @vought2009signals?].

@Minster2016 with CREBRF found a particular variant that was unique to Samoans that had evidence of having under gone selection

@Zang2014 had evidence that urate was under selection



<!--
motivation: seemed to be little in the way of calculating stats genome wide and wasn't sure how sensitive these stats were to snp density



This chapter is about trying to first of all establish how sensitive the various statistics were to window/slide variation

from the literature there doesn't seem to be any recommendation as to how to compute these statistics genome wide. Up until recently the various statistics were computed at the 'feature' level, usually the gene.

Fst has a paper in early 2000's from Bruce Weir that touched on what sort of size to use howver it largely just came down to 'this window size seems to have enough of a smoothing effect as well as display some of the variation in the data' type approach and was largely just based on a this looks good enough.

Pybus et al specified their window sizes which was roughly 30 kb window and 3kb slide for most statistics however this was decided on unshown data as a trade off between resolution and density.

In cadzow 2014 I tried had tried a few sizes and it it seemed that there was little difference between them other than the smaller the window the more noisy the data looked.

While I was playing around with this for Tajima's D there does seem to be at least some relation to the density of the snps and the window size, which does intuitively make sense, in that if there are none or only 1 snp per window then the stats are going to be pretty poor.

From the core exome work it looks as though 30 kb was ok and at least when the distribution of TD was binned by number of snps per window it didn't look to bad and the overall total distribution of TD looks not too bad. Increasing from 30 kb didn't really change TD that much. the extremes become more extreme but the means increaseas window size increases.
-->

<!-- ascertainment bias will skew selection statistics that are site frequency spectrum dependant in chip data due to the over-representation of intermediate frequency SNPs, these include Tajima's D (upward) and Fay and Wu's H [@Clark2005]


Carlson2005 did windows of 100kb, slide 10kb and mean snps were ~50/window. ascertainment bias esp in the +ve end. continous region on >20 windows where 75% were in bottom 1% was defined as contiguous regions of Tajimas D reduction.
RafajloviÄ‡2014 largely followed Carlson2005, they also discarded windows with less than 5 snps.

Bigman2010 has a custom Tajima's D that is created to be similar to iHS in terms of calculation
-->

literature
Cadzow2014
Cadzow2016
Pybus2014
Ramos2017
Minster2017

### objectives

- Investigate regions that are under selection in the Polynesian populations

## Methods

### Selection dataset creation


The selection of individuals, and preliminary data preparation for the selection data set is detailed in Section \@ref(selectionDataset). It consisted of combining 500 individuals -100 per population - from the Genetics of Gout in Aotearoa who had been genoypted on the Illumina CoreExome v24 mircoarray. Analysis groups were created by using genetic ancestry for the following populations: \glsdesc{nzm}, \glsdesc{cim}, \glsdesc{sam}, \glsdesc{ton}, and \glsdesc{nzc} and unenriching for gout. To unenrich for gout, population specific prevalences based on @Winnard2013 of 2.3% for \gls{nzc}, 7.7% for \gls{nzm}, and 8.6% for \gls{cim}, \gls{sam}, and \gls{ton} were used and individuals were randomly sampled.

Sample population prevalences of 2.0% (\gls{nzc}), 7.0% (\gls{nzm}), 45.0% (\gls{cim}), 12.0% (\gls{sam}), and 54.0% (\gls{ton}) were obtained.



The phased \gls{vcf} files for the combined \gls{nz} and marker matched 1000 Genome Project individuals were normalised

The selection statistics were calculated for individuals populations using a combination of two main software packages, PopGenome in R, and selectionTools v1.1. PopGenome was used to calculate the intra population frequency spectrum based tests for selection and neutrality.


Haplotypic stats of \gls{ihs}, \gls{nsl}, and \gls{xpehh} were calculated using the selectionTools pipeline with significance set as being the 99th percentile for the absolute normalised score. Frequency based statistics were also generated such as \gls{daf} and F~ST~.

#### PopGenome

\Gls{td}, \gls{fwh}, \gls{fld}, \gls{flf}, and \gls{ze} were calculated using the Popgenome package in R (see section \@ref(popgenomeMethods) for further details). The effect of window and slide size was investigated using windows sizes ranging from 30 kb through to 150 kb (Figure \@ref(fig:windowSizes)), and slide sizes ranging from 3 kb through to 35 kb, using the \gls{gbr} and \gls{nzc} populations. The general trend was as window size increased the entire distribution was shifted to the right and also widened. Similar to @Pybus2014, @Rafajlovic2014 and @Carlson2005 a trade off between increasing the number of \gls{snp}s per window and adequate resolution was made and the window size of 100 kb and slide of 10 kb was chosen. Windows that had fewer than 4 segregating sites were filtered out and window co-ordinates were altered to be +/- 5 kb from the centre of the window. The empirical distribution for each population was used to create a significance threshold. The lower threshold was the 1st percentile and less than zero. The upper significance threshold was the 99th percentile and greater than zero. The greater or less than zero condition was to enable the interpretation of the selection statistic results.

Regions of XXX were created using a rolling window approach, whereby a condition of 7 or more windows out of 10 consecutive 10 kb windows were required to have met a significance threshold in the same tail of the distribution.

#### selectionTools

The calculation of \gls{ihs}, \gls{nsl}, and \gls{xpehh} was done using the selectionTools pipeline as detailed in \@ref(selectionTools). The default setting were used which was a maximum gap extension of 200 kb. Significance was defined as the most extreme 1% of values after normalisation using frequency bins of 0.05.


Derived allele frequency

F~ST~

iHS and nSL
tested gap width to try and bring in SLC2A9 for coreExome data
Don't actually want the comparisons between NAD and EPN or WPN, or the ones for EPN/WPN with NZM/CIM/SAM/TONDon't actually want the comparisons between NAD and EPN or WPN, or the ones for EPN/WPN with NZM/CIM/SAM/TON



<!--
## SelectionTools pipeline

for the selection pipeline we need to split into populations

```{r, engine = 'bash', eval = FALSE}
# working dir: /Volumes/BiochemXsan/scratch/merrimanlab/murray/working_dir/coreExome_selection/NZ_coreExome_1kgp/data/
# make a combined panel file
cut -f1-3 /Volumes/BiocArchive/archive/merrimanlab/reference_files/VCF/1000Genomes_vcf_files/Phase3_March2017/integrated_call_samples_v3.20130502.ALL.panel |\
cat - NZ_coreExome/*panel > nz_1kgp.panel

mkdir Indiv_pops

parallel '
  bcftools view \
    -s $(grep {1} nz_1kgp.panel |\
      cut -f 1 |\
      tr "\n" "," |\
      sed "s/,$//g") \
    -O z \
    -o Indiv_pops/{1}.chr{2}.phased.vcf.gz NZ_1KGP.chr{2}.phased.sample_updated.vcf.gz
  ' ::: $(cut -f 2 nz_1kgp.panel |grep -v "pop" | sort |uniq) ::: $(seq 1 22)
```

Break into super populations to analyse the Polynesians (NZM,CIM,TON,SAM) - make sure to exclude the NZC from the EUR super pop and also exclude the admixed NZM (NAD), Eastern Polynesians (EPN), and Western Polynesians (WPN).

```{r, engine = 'bash', eval = FALSE}
# working dir: /Volumes/BiochemXsan/scratch/merrimanlab/murray/working_dir/coreExome_selection/NZ_coreExome_1kgp/data/
mkdir Super_pops

parallel '
  bcftools view \
    -s $(grep {1} nz_1kgp.panel |\
      grep -v "NAD\|EPN\|NZC\|WPN" |\
      cut -f 1 |\
      tr "\n" "," |\
      sed "s/,$//g") \
    -O z \
    -o Super_pops/{1}.chr{2}.phased.vcf.gz NZ_1KGP.chr{2}.phased.sample_updated.vcf.gz
  ' ::: $(cut -f 3 nz_1kgp.panel |grep -v "pop" | sort |uniq) ::: $(seq 1 22)
```

Now send to NeSI pan cluster for the selectionTools pipeline to take place

Individual populations are first run through selectionTools with iHS, nSL, and XP-EHH calculated later

```{r, engine = 'bash', eval = FALSE}
# on nesi
mkdir -p /home/murray.cadzow/uoo00008/NZ_1KGP_unimputed/{Indiv_pops,Indiv_pops_results}

# locally
parallel '
  scp {} murray.cadzow@login.uoa.nesi.org.nz:~/uoo00008/NZ_1KGP_unimputed/Indiv_pops/ 
  ' ::: $(ls /media/xsan/scratch/merrimanlab/murray/working_dir/coreExome_selection/NZ_coreExome_1kgp/data/Indiv_pops/*gz)

# on nesi
# working dir: /home/murray.cadzow/uoo00008/NZ_1KGP_unimputed/Indiv_pops

# find all the vcfs and use the first prefix to use as the population code
for p in $(ls * | cut -d'.' -f1 | sort | uniq)
do 
  sbatch ../unimputed_selection_pipeline.sl $p ; sleep 2; 
done
```

Then run run_selscan.sl and run_nsl_selscan on each population results file
```{r, engine = 'bash', eval = FALSE}
# on nesi
# working dir: Indiv_pops_results
for p in $(find ./[A-Z]* -type d | grep -v xpehh | sed 's/\.\///g')
do 
  cd $p 
  sbatch -J ihs_$p ../../run_selscan.sl $p
  sbatch -J nsl_$p ../../run_nsl_selscan.sl $p
  cd ../
  sleep 5
done
```



```{r, engine = 'bash', eval = FALSE}
# on nesi
# working dir: /home/murray.cadzow/uoo00008/NZ_1KGP_unimputed/Indiv_pops_results
# individual pops
for p1 in CIM NZC NZM SAM TON
do
  for p2 in ACB ASW BEB CDX CEU CHB CHS CIM CLM EPN ESN FIN GBR GIH GWD IBS ITU JPT KHV LWK MSL MXL NAD NZC NZM PEL PJL PUR SAM STU TON TSI WPN YRI
  do
    if [[ $p1 != $p2 ]]
    then
      echo $p1 $p2
      sbatch ../run_xpehh.sl $p1 $p2
      sleep 5
    fi
  done
done


for p1 in EPN NAD WPN
do
  for p2 in ACB ASW BEB CDX CEU CHB CHS CIM CLM EPN ESN FIN GBR GIH GWD IBS ITU JPT KHV LWK MSL MXL NAD NZC NZM PEL PJL PUR SAM STU TON TSI WPN YRI
  do
    if [[ $p1 != $p2 ]]
    then
      echo $p1 $p2
      sbatch ../run_xpehh.sl $p1 $p2
      sleep 5
    fi
  done
done




# super pops
for p in EUR EAS SAS AMR AFR
do
  sbatch ../run_xpehh.sl POL $p
  sleep 5
done

# Don't actually want the comparisons between NAD and EPN or WPN, or the ones for EPN/WPN with NZM/CIM/SAM/TON
rm -r NAD_{WPN,EPN}
rm -r {EPN,WPN}_{NZM,CIM,SAM,TON}
```

-->


<!--
This is at least covered for TD in the R notebook

window slide was equal to the window size for each size tested.

The rough results are that because of the density of the SNPs on the chip the windows tested 1,5,10,30,50,100 KB for 1,5,10 there is not a great number of SNPs per window. 

The 1 kb window provides a very large number of windows with no snps present and when plotted looks like there is much fewer windows than the 5kb


the window size of 30kb starts to offer a 

table: summary of n segreating sites by window size
-->

```{r,echo= FALSE }
test_windows <- data.frame(
  Window = c(30,30,50,50,100,100,150,150),
  Slide = c(3,10,5,10,10,25,10,35) )

```





table: 

```{r, results = 'asis'}
#pop: 1 = GBR, 2 = NZC 

summary_results <- readRDS('~/data/NZ_coreExome_1kgp/popgenome_window_tests_filtered_3ns_summary-19-7-2017.RDS')
#summary_results %>% ungroup()%>% mutate(pop = ifelse(pop == 1, "GBR", "NZC"), range = max - min) %>% group_by(pop) %>% mutate_at(vars(c('mean', 'sd', 'lower_1', 'upper_99', 'median', 'min', 'max','range')), funs(round(., digits = 3))) %>% ungroup()  %>% left_join(., test_windows %>% mutate(ws = as.numeric(rownames(.))), by = 'ws') %>% select(stat, Window, Slide, pop, min, lower_1, median, mean, sd, upper_99, max, range) %>% data.frame #%>%  pander::pandoc.table(.)
```

(ref:windowSizes) Effect of window size on selection statistics using a slide 10 kb. The vertical grey line indicates the the chosen window size of 100 kb.

```{r windowSizes, fig.cap = '(ref:windowSizes)'}
summary_results %>% mutate(range = max - min)%>% mutate_at(vars(c('mean', 'sd', 'lower_1', 'upper_99', 'median', 'min', 'max', 'range')), funs(round(., digits = 3))) %>% ungroup() %>% mutate(pop = ifelse(pop == 1, "GBR", "NZC")) %>% left_join(., test_windows %>% mutate(ws = as.numeric(rownames(.))), by = 'ws') %>% select(stat, Window, Slide, pop, min, lower_1, median, mean, sd, upper_99, max, range) %>% gather("sumStat", 'value', 5:NCOL(.))%>% filter(Slide == 10) %>% ggplot(., aes(x = Window, y = value, by = pop, colour = sumStat )) + geom_line()  + geom_vline(xintercept = 100, colour = "grey") + facet_wrap(~stat, scales = 'free_y') + theme_bw()
```



<!--
want to have the

decided on 100kb with 10 kb slide
number of segregating sites
-->











<!--
## Simulations

### Bottleneck

### Migration

### Expansion

### Positive Selection

### Balancing Selection

### Negative Selection


-->

## Results
<!--
Look at what windows/genes for each population are interesting
Super populations - 1KGP vs POL
Super Populations - 1KGP vs EPN/WPN
Individual populations - CIM/NZM/SAM/TON/NZC
-->



## method based


### Intra
```{r}

```


For the frequency spectrum based intra population statistics of \gls{td}, \gls{fwh}, \gls{fld}, \gls{flf}, and \gls{ze} there were population specific distributions. There was similarity however between the populations of a super population.

There each population had a different distribution of values

the super populations had similar min, mean, sd, median, max for the stats


```{r}
lower_sig_stats %>% mutate(statname = paste0(statname,"_lo")) %>%  select(contains('chrom'), pop,statname) %>% distinct() %>% group_by(pop,statname) %>% tally() %>% spread(statname,n) %>% full_join(.,upper_sig_stats %>% mutate(statname = paste0(statname,"_up")) %>%  select(contains('chrom'), pop,statname) %>% distinct() %>% group_by(pop,statname) %>% tally() %>% spread(statname,n) , by = "pop" ) %>% knitr::kable(., caption = "number of significant windows by selection stat", booktabs = TRUE, format = 'latex') %>% kableExtra::kable_styling(latex_options = 'scale_down') %>% kableExtra::add_header_above(., c("", "Lower Tail" = 5, "Upper Tail" = 5))
```



<!--
table for each stat that give global mins/means/sd/max for each population
-->





#### Tajima's D

<!--
describe the overall tajima's d and where the polynesian populations fell at the summary level
-->

```{r, warning=FALSE, message=FALSE}
a <- list()
for(p in sort(unique(lower_sig_stats$pop))){
  a[[p]] <-  lower_sig_stats %>% filter(statname == 'Tajima.D', pop == p) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Tajima.D', pop != p), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()
}
td_pop_unique_genes_lower <- bind_cols(a) %>% gather("pop","n")
rm(a)

a <- list()
for(p in sort(unique(upper_sig_stats$pop))){
  a[[p]] <-  upper_sig_stats %>% filter(statname == 'Tajima.D', pop == p) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Tajima.D', pop != p), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()
}
td_pop_unique_genes_upper <- bind_cols(a) %>% gather("pop","n")
rm(a)

```


The mean number of windows that met the significance threshold per population for the lower tail was  `r lower_sig_stats %>% filter(statname == 'Tajima.D') %>% group_by(pop) %>% tally() %>% summarise(mean(n)) %>% round(digits = 0)` (SD = `r lower_sig_stats %>% filter(statname == 'Tajima.D') %>% group_by(pop) %>% tally() %>% summarise(sd(n)) %>% round(digits = 0)`) with a minimum of `r lower_sig_stats %>% filter(statname == 'Tajima.D') %>% group_by(pop) %>% tally() %>% summarise(min(n)) %>% round(digits = 0)` and a maximum of `r lower_sig_stats %>% filter(statname == 'Tajima.D') %>% group_by(pop) %>% tally() %>% summarise(max(n)) %>% round(digits = 0)`. The mean number of genes that were unique to a population was `r td_pop_unique_genes_lower %>% summarise(mean(n)) %>% round(digits = 0)` (SD = `r td_pop_unique_genes_lower %>% summarise(sd(n)) %>% round(digits = 0)`) and minimum of `r td_pop_unique_genes_lower %>% summarise(min(n)) %>% round(digits = 0)` and maximum of `r td_pop_unique_genes_lower %>% summarise(max(n)) %>% round(digits = 0)`.

The mean number of windows per population that met the significance threshold for the upper tail was `r upper_sig_stats %>% filter(statname == 'Tajima.D') %>% group_by(pop) %>% tally() %>% summarise(mean(n)) %>% round(digits = 0)` (SD = `r upper_sig_stats %>% filter(statname == 'Tajima.D') %>% group_by(pop) %>% tally() %>% summarise(sd(n)) %>% round(digits = 0)`), with a minimum of `r upper_sig_stats %>% filter(statname == 'Tajima.D') %>% group_by(pop) %>% tally() %>% summarise(min(n)) %>% round(digits = 0)` and a maximum of `r upper_sig_stats %>% filter(statname == 'Tajima.D') %>% group_by(pop) %>% tally() %>% summarise(max(n)) %>% round(digits = 0)`. The mean number of genes that were unique to a population was `r td_pop_unique_genes_upper %>% summarise(mean(n)) %>% round(digits = 0)` (SD = `r td_pop_unique_genes_upper %>% summarise(sd(n)) %>% round(digits = 0)`) and minimum of `r td_pop_unique_genes_upper %>% summarise(min(n)) %>% round(digits = 0)` and maximum of `r td_pop_unique_genes_upper %>% summarise(max(n)) %>% round(digits = 0)`.





```{r}
global_summary <- readRDS("~/data/NZ_coreExome_1kgp/100kbWindow_intra/filtered/100kbwindows_summary_popgenome_filtered_3ns_resized.31-8-2017.RDS") %>% filter(!pop %in% c("WPN","EPN","POL","NAD"))

td_super_pop_means <- global_summary %>% filter(stat == 'Tajima.D') %>% left_join(., panel %>% select(pop, super_pop) %>% distinct(), by = 'pop') %>% arrange(super_pop) %>% group_by(super_pop) %>%  summarise(mean = mean(mean))
```

```{r, results='asis'}
gs_td_table <- global_summary %>% ungroup%>% filter(stat == 'Tajima.D') %>% left_join(., panel %>% select(pop, super_pop) %>% distinct(), by = 'pop') %>% arrange(super_pop) %>% select(super_pop, pop, mean, sd, min, lower_1,median,upper_99, max)  %>%  mutate_at(vars('mean','sd','min','lower_1','upper_99','median','max'), funs(sprintf("%.3f",. ))) %>%  data.frame() 
#gs_td_table %>% pander::pandoc.table(., justify = c('left', rep('right', NCOL(.)-1)))
gs_td_table %>% select(-super_pop) %>%  knitr::kable(., format = 'latex', booktabs = TRUE, col.names = c("Population","Mean","Std. Dev","Min","Lower 1%","Median","Upper 99%", "Max"), align = "lrrrrrrr", caption = "(\\#tab:gsTdTable) \\gls{td} summary statistics by population") %>% kableExtra::group_rows("AFR", 1, 7) %>% kableExtra::group_rows("AMR", 8, 11) %>% kableExtra::group_rows("EAS", 12, 16) %>% kableExtra::group_rows("EUR", 17, 22) %>% kableExtra::group_rows("POL", 23, 26) %>% kableExtra::group_rows("SAS", 27,31) 
```


population summary stats in table \@ref(tab:gsTdTable)

The mean Tajima's D for each chromosome can be seen in Table \@ref(tab:meanTDtable) and is found in the appendix. The mean \gls{td} for the Polynesian populations was `r global_summary %>% filter(pop %in% c('CIM','NZM','SAM','TON'), stat == 'Tajima.D') %>% summarise(m = mean(mean)) %>% .[["m"]] %>% sprintf('%.3f', .)` with  `r gs_td_table %>% filter(pop == 'CIM') %>% .[["mean"]]` for \gls{cim}, `r gs_td_table %>% filter(pop == 'NZM') %>% .[["mean"]]` for \gls{nzm}, `r gs_td_table %>% filter(pop == 'SAM') %>% .[["mean"]]` for \gls{sam}, and `r gs_td_table %>% filter(pop == 'TON') %>% .[["mean"]]` for \gls{ton}. The means of the Polynesian populations were lower than all of the other populations except \gls{pel} and \gls{mxl}.

```{r, warning =  FALSE, message = FALSE}
pol_td_t.test <- bind_rows(
  lapply(c("AFR","AMR", "EAS","EUR","SAS"), function(sp){
    cbind(SuperPop = sp, 
          global_summary %>% filter(stat == 'Tajima.D') %>% left_join(., panel %>% select(pop, super_pop) %>% distinct(), by = 'pop') %>% filter(super_pop %in% c("POL",sp)) %>% ungroup() %>% mutate(super_pop = as.character(super_pop)) %>% select(super_pop, mean) %>% do(broom::tidy(t.test( mean ~ super_pop, data = .))))
})) %>% mutate(adjusted_p = p.adjust(p.value, "bonferroni"))
```

The minimum \gls{td} value for a Polynesian population was `r gs_td_table %>% filter(super_pop == "POL") %>% filter(min(min) == min) %>% .[["min"]]` and was from `r gs_td_table %>% filter(super_pop == "POL") %>% filter(min(min) == min) %>% .[["pop"]]`, whereas the maximum value was `r gs_td_table %>% filter(super_pop == "POL") %>% filter(max(max) == max) %>% .[["max"]]` and from `r gs_td_table %>% filter(super_pop == "POL") %>% filter(max(max) == max) %>% .[["pop"]]`. The Polynesian mean was significantly different to the other means for the super population using a t-test. The means were \gls{afr}: `r td_super_pop_means %>% filter(super_pop == "AFR") %>% .[["mean"]] %>% sprintf('%.3f',.) ` (P = `r pol_td_t.test %>% filter(SuperPop == 'AFR') %>% select(p.value) %>% sprintf('%.3f',.)`), \gls{amr}: `r td_super_pop_means %>% filter(super_pop == "AMR") %>% .[["mean"]] %>% sprintf('%.3f',.) ` (P = `r pol_td_t.test %>% filter(SuperPop == 'AMR') %>% select(p.value) %>% sprintf('%.3f',.)`), \gls{eur}: `r td_super_pop_means %>% filter(super_pop == "EUR") %>% .[["mean"]] %>% sprintf('%.3f',.) ` (P = `r pol_td_t.test %>% filter(SuperPop == 'EUR') %>% select(p.value) %>% sprintf('%.3f',.)`), \gls{eas}: `r td_super_pop_means %>% filter(super_pop == "EAS") %>% .[["mean"]] %>% sprintf('%.3f',.) ` (P = `r pol_td_t.test %>% filter(SuperPop == 'EAS') %>% select(p.value) %>% sprintf('%.3f',.)`), and \gls{sas}: `r td_super_pop_means %>% filter(super_pop == "SAS") %>% .[["mean"]] %>% sprintf('%.3f',.) ` (P = `r pol_td_t.test %>% filter(SuperPop == 'SAS') %>% select(p.value) %>% sprintf('%.3f',.)`). 


CIM



<!--
- `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM')) %>% select(contains('chrom'))  %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions 
- `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp of region 
- `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes 
- `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Tajima.D', !pop %in% c('CIM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only from CIM
- `r paste(lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Tajima.D', !pop %in% c('CIM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% .[["SYMBOL"]], collapse = " ")`
-->
```{r}
gr <- lower_sig_stats %>% filter(pop == "CIM", statname == 'Tajima.D') %>%  GenomicRanges::GRanges() %>% GenomicRanges::reduce()
gr[GenomicRanges::width(gr) >= 20 * 10e3]
```


NZM



<!--
- `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions 
- `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp of region 
- `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes 
- `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('NZM')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Tajima.D', !pop %in% c('NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only from NZM
- `r paste(lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('NZM')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Tajima.D', !pop %in% c('NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% .[["SYMBOL"]], collapse = " ")`
-->
```{r}
gr <- lower_sig_stats %>% filter(pop == "NZM", statname == 'Tajima.D') %>%  GenomicRanges::GRanges() %>% GenomicRanges::reduce()
gr[GenomicRanges::width(gr) >= 20 * 10e3]
```


SAM


<!--
- `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions 
- `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp of region 
- `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes 
- `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Tajima.D', !pop %in% c('SAM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only from SAM
- `r paste(lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Tajima.D', !pop %in% c('SAM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% .[["SYMBOL"]], collapse = " ")`
-->

```{r}
gr <- lower_sig_stats %>% filter(pop == "SAM", statname == 'Tajima.D') %>%  GenomicRanges::GRanges() %>% GenomicRanges::reduce()
gr[GenomicRanges::width(gr) >= 20 * 10e3]
```


TON




<!--
- `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions 
- `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp of region 
- `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes 
- `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('TON')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Tajima.D', !pop %in% c('TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only from TON
- `r paste(lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('TON')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Tajima.D', !pop %in% c('TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% .[["SYMBOL"]], collapse = " ")`
-->

```{r}
gr <- lower_sig_stats %>% filter(pop == "TON", statname == 'Tajima.D') %>%  GenomicRanges::GRanges() %>% GenomicRanges::reduce()
gr[GenomicRanges::width(gr) >= 20 * 10e3]
```



EP

- `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM', 'NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions 
- `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM', 'NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp of region 
- `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM', 'NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes 
- `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM', 'NZM')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Tajima.D', !pop %in% c('CIM','NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only from EP
- `r paste(lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM','NZM')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Tajima.D', !pop %in% c('CIM','NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% .[["SYMBOL"]], collapse = " ")`



WP

- `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions 
- `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp of region 
- `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM','TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes 
- `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM','TON')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Tajima.D', !pop %in% c('SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only from WP
- `r paste(lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM','TON')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Tajima.D', !pop %in% c('SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% .[["SYMBOL"]], collapse = " ")`

POL



<!--
- `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions 
- `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp of region 
- `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes 
- `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM','NZM','SAM','TON')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Tajima.D', !pop %in% c('CIM','NZM','SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only from POL
- `r paste(lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM','NZM','SAM','TON')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Tajima.D', !pop %in% c('CIM','NZM','SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% .[["SYMBOL"]], collapse = " ")`
-->

<!-- sort into numerical order for chromosome -->
Genes Unique to Polynesian Populations

Table \@ref(tab:lowersigtdPol)



#### Fay and Wu's H

```{r, warning=FALSE, message=FALSE}
a <- list()
for(p in sort(unique(lower_sig_stats$pop))){
  a[[p]] <-  lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop == p) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fay.Wu.H', pop != p), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()
}
fwh_pop_unique_genes_lower <- bind_cols(a) %>% gather("pop","n")
rm(a)

a <- list()
for(p in sort(unique(upper_sig_stats$pop))){
  a[[p]] <-  upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop == p) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fay.Wu.H', pop != p), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()
}
fwh_pop_unique_genes_upper <- bind_cols(a) %>% gather("pop","n")
rm(a)

```


The mean number of windows that met the significance threshold per population for the lower tail was  `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H') %>% group_by(pop) %>% tally() %>% summarise(mean(n)) %>% round(digits = 0)` (SD = `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H') %>% group_by(pop) %>% tally() %>% summarise(sd(n)) %>% round(digits = 0)`) with a minimum of `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H') %>% group_by(pop) %>% tally() %>% summarise(min(n)) %>% round(digits = 0)` and a maximum of `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H') %>% group_by(pop) %>% tally() %>% summarise(max(n)) %>% round(digits = 0)`. The mean number of genes that were unique to a population was `r fwh_pop_unique_genes_lower %>% summarise(mean(n)) %>% round(digits = 0)` (SD = `r fwh_pop_unique_genes_lower %>% summarise(sd(n)) %>% round(digits = 0)`) and minimum of `r fwh_pop_unique_genes_lower %>% summarise(min(n)) %>% round(digits = 0)` and maximum of `r fwh_pop_unique_genes_lower %>% summarise(max(n)) %>% round(digits = 0)`.

The mean number of windows per population that met the significance threshold for the upper tail was `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H') %>% group_by(pop) %>% tally() %>% summarise(mean(n)) %>% round(digits = 0)` (SD = `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H') %>% group_by(pop) %>% tally() %>% summarise(sd(n)) %>% round(digits = 0)`), with a minimum of `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H') %>% group_by(pop) %>% tally() %>% summarise(min(n)) %>% round(digits = 0)` and a maximum of `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H') %>% group_by(pop) %>% tally() %>% summarise(max(n)) %>% round(digits = 0)`.



```{r, results='asis'}
global_summary %>% ungroup%>% filter(stat == 'Fay.Wu.H') %>% select(pop, mean, sd, min, lower_1,median,upper_99, max) %>% mutate_at(vars('mean','sd','min','lower_1','upper_99','median','max'), funs(sprintf("%.3f",. ))) %>%  data.frame() %>% pander::pandoc.table(. , justify = c('left', rep('right', NCOL(.)-1)))
```

CIM



<!--
- `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions 
- `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp of region 
- `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes 
- `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Fay.Wu.H', !pop %in% c('CIM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only from CIM
-->

```{r}
gr <- lower_sig_stats %>% filter(pop == "CIM", statname == 'Fay.Wu.H') %>%  GenomicRanges::GRanges() %>% GenomicRanges::reduce()
gr[GenomicRanges::width(gr) >= 20 * 10e3]
```

NZM


<!--
- `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions 
- `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp of region 
- `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes 
- `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('NZM')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Fay.Wu.H', !pop %in% c('NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only from NZM
-->

```{r}
gr <- lower_sig_stats %>% filter(pop == "NZM", statname == 'Fay.Wu.H') %>%  GenomicRanges::GRanges() %>% GenomicRanges::reduce()
gr[GenomicRanges::width(gr) >= 20 * 10e3]
```


SAM



<!--
- `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions 
- `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp of region 
- `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes 
- `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('SAM')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Fay.Wu.H', !pop %in% c('SAM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only from SAM
-->

```{r}
gr <- lower_sig_stats %>% filter(pop == "SAM", statname == 'Fay.Wu.H') %>%  GenomicRanges::GRanges() %>% GenomicRanges::reduce()
gr[GenomicRanges::width(gr) >= 20 * 10e3]
```


TON



<!--
- `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions 
- `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp of region 
- `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes 
- `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('TON')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Fay.Wu.H', !pop %in% c('TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only from TON
-->

```{r}
gr <- lower_sig_stats %>% filter(pop == "TON", statname == 'Fay.Wu.H') %>%  GenomicRanges::GRanges() %>% GenomicRanges::reduce()
gr[GenomicRanges::width(gr) >= 20 * 10e3]
```



EP

- `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM', 'NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions 
- `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM', 'NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp of region 
- `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM', 'NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes 
- `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM', 'NZM')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Fay.Wu.H', !pop %in% c('CIM','NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only from EP

WP

- `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions 
- `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp of region 
- `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('SAM','TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes 
- `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('SAM','TON')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Fay.Wu.H', !pop %in% c('SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only from WP

POL



<!--
- `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions 
- `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp of region 
- `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM','NZM','SAM','TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes 
- `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM','NZM','SAM','TON')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Fay.Wu.H', !pop %in% c('CIM','NZM','SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only from POL
-->

Genes Unique to Polynesian populations

Table \@ref(tab:lowersigfwhPol)

#### Fu and Li's D

```{r, warning=FALSE, message=FALSE}
a <- list()
for(p in sort(unique(lower_sig_stats$pop))){
  a[[p]] <-  lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop == p) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.D', pop != p), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()
}
fld_pop_unique_genes_lower <- bind_cols(a) %>% gather("pop","n")
rm(a)

a <- list()
for(p in sort(unique(upper_sig_stats$pop))){
  a[[p]] <-  upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop == p) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.D', pop != p), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()
}
fld_pop_unique_genes_upper <- bind_cols(a) %>% gather("pop","n")
rm(a)

```


The mean number of windows that met the significance threshold per population for the lower tail was  `r lower_sig_stats %>% filter(statname == 'Fu.Li.D') %>% group_by(pop) %>% tally() %>% summarise(mean(n)) %>% round(digits = 0)` (SD = `r lower_sig_stats %>% filter(statname == 'Fu.Li.D') %>% group_by(pop) %>% tally() %>% summarise(sd(n)) %>% round(digits = 0)`) with a minimum of `r lower_sig_stats %>% filter(statname == 'Fu.Li.D') %>% group_by(pop) %>% tally() %>% summarise(min(n)) %>% round(digits = 0)` and a maximum of `r lower_sig_stats %>% filter(statname == 'Fu.Li.D') %>% group_by(pop) %>% tally() %>% summarise(max(n)) %>% round(digits = 0)`. The mean number of genes that were unique to a population was `r fld_pop_unique_genes_lower %>% summarise(mean(n)) %>% round(digits = 0)` (SD = `r fld_pop_unique_genes_lower %>% summarise(sd(n)) %>% round(digits = 0)`) and minimum of `r fld_pop_unique_genes_lower %>% summarise(min(n)) %>% round(digits = 0)` and maximum of `r fld_pop_unique_genes_lower %>% summarise(max(n)) %>% round(digits = 0)`.

The mean number of windows per population that met the significance threshold for the upper tail was `r upper_sig_stats %>% filter(statname == 'Fu.Li.D') %>% group_by(pop) %>% tally() %>% summarise(mean(n)) %>% round(digits = 0)` (SD = `r upper_sig_stats %>% filter(statname == 'Fu.Li.D') %>% group_by(pop) %>% tally() %>% summarise(sd(n)) %>% round(digits = 0)`), with a minimum of `r upper_sig_stats %>% filter(statname == 'Fu.Li.D') %>% group_by(pop) %>% tally() %>% summarise(min(n)) %>% round(digits = 0)` and a maximum of `r upper_sig_stats %>% filter(statname == 'Fu.Li.D') %>% group_by(pop) %>% tally() %>% summarise(max(n)) %>% round(digits = 0)`. The mean number of genes that were unique to a population was `r fld_pop_unique_genes_upper %>% summarise(mean(n)) %>% round(digits = 0)` (SD = `r fld_pop_unique_genes_upper %>% summarise(sd(n)) %>% round(digits = 0)`) and minimum of `r fld_pop_unique_genes_upper %>% summarise(min(n)) %>% round(digits = 0)` and maximum of `r fld_pop_unique_genes_upper %>% summarise(max(n)) %>% round(digits = 0)`.



```{r, results='asis'}
global_summary %>% ungroup%>% filter(stat == 'Fu.Li.D') %>% select(pop, mean, sd, min, lower_1,median,upper_99, max) %>% mutate_at(vars('mean','sd','min','lower_1','upper_99','median','max'), funs(sprintf("%.3f",. ))) %>%  data.frame() %>% pander::pandoc.table(. , justify = c('left', rep('right', NCOL(.)-1)))
```

CIM



<!--
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions 
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp of region 
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes 
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Fu.Li.D', !pop %in% c('CIM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only from CIM
-->
```{r}
gr <- lower_sig_stats %>% filter(pop == "CIM", statname == 'Fu.Li.D') %>%  GenomicRanges::GRanges() %>% GenomicRanges::reduce()
gr[GenomicRanges::width(gr) >= 20 * 10e3]
```

NZM



<!--
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions 
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp of region 
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes 
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('NZM')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Fu.Li.D', !pop %in% c('NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only from NZM
-->
```{r}
gr <- lower_sig_stats %>% filter(pop == "NZM", statname == 'Fu.Li.D') %>%  GenomicRanges::GRanges() %>% GenomicRanges::reduce()
gr[GenomicRanges::width(gr) >= 20 * 10e3]
```




SAM



<!--
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions 
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp of region 
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes 
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('SAM')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Fu.Li.D', !pop %in% c('SAM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only from SAM
-->
```{r}
gr <- lower_sig_stats %>% filter(pop == "SAM", statname == 'Fu.Li.D') %>%  GenomicRanges::GRanges() %>% GenomicRanges::reduce()
gr[GenomicRanges::width(gr) >= 20 * 10e3]
```


TON



<!--
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions 
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp of region 
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes 
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('TON')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Fu.Li.D', !pop %in% c('TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only from TON
-->
```{r}
gr <- lower_sig_stats %>% filter(pop == "TON", statname == 'Fu.Li.D') %>%  GenomicRanges::GRanges() %>% GenomicRanges::reduce()
gr[GenomicRanges::width(gr) >= 20 * 10e3]
```



EP

- `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM', 'NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions 
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM', 'NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp of region 
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM', 'NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes 
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM', 'NZM')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Fu.Li.D', !pop %in% c('CIM','NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only from EP

WP

- `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions 
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp of region 
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('SAM','TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes 
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('SAM','TON')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Fu.Li.D', !pop %in% c('SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only from WP

POL



<!--
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions 
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp of region 
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes 
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM','NZM','SAM','TON')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Fu.Li.D', !pop %in% c('CIM','NZM','SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only from POL
-->

Genes Unique to Polynesian populations

Table \@ref(tab:lowersigfldPol)

#### Fu and Li's F

```{r, warning=FALSE, message=FALSE}
a <- list()
for(p in sort(unique(lower_sig_stats$pop))){
  a[[p]] <-  lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop == p) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.F', pop != p), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()
}
flf_pop_unique_genes_lower <- bind_cols(a) %>% gather("pop","n")
rm(a)

a <- list()
for(p in sort(unique(upper_sig_stats$pop))){
  a[[p]] <-  upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop == p) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.F', pop != p), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()
}
flf_pop_unique_genes_upper <- bind_cols(a) %>% gather("pop","n")
rm(a)

```


The mean number of windows that met the significance threshold per population for the lower tail was  `r lower_sig_stats %>% filter(statname == 'Fu.Li.F') %>% group_by(pop) %>% tally() %>% summarise(mean(n)) %>% round(digits = 0)` (SD = `r lower_sig_stats %>% filter(statname == 'Fu.Li.F') %>% group_by(pop) %>% tally() %>% summarise(sd(n)) %>% round(digits = 0)`) with a minimum of `r lower_sig_stats %>% filter(statname == 'Fu.Li.F') %>% group_by(pop) %>% tally() %>% summarise(min(n)) %>% round(digits = 0)` and a maximum of `r lower_sig_stats %>% filter(statname == 'Fu.Li.F') %>% group_by(pop) %>% tally() %>% summarise(max(n)) %>% round(digits = 0)`. The mean number of genes that were unique to a population was `r flf_pop_unique_genes_lower %>% summarise(mean(n)) %>% round(digits = 0)` (SD = `r flf_pop_unique_genes_lower %>% summarise(sd(n)) %>% round(digits = 0)`) and minimum of `r flf_pop_unique_genes_lower %>% summarise(min(n)) %>% round(digits = 0)` and maximum of `r flf_pop_unique_genes_lower %>% summarise(max(n)) %>% round(digits = 0)`.

The mean number of windows per population that met the significance threshold for the upper tail was `r upper_sig_stats %>% filter(statname == 'Fu.Li.F') %>% group_by(pop) %>% tally() %>% summarise(mean(n)) %>% round(digits = 0)` (SD = `r upper_sig_stats %>% filter(statname == 'Fu.Li.F') %>% group_by(pop) %>% tally() %>% summarise(sd(n)) %>% round(digits = 0)`), with a minimum of `r upper_sig_stats %>% filter(statname == 'Fu.Li.F') %>% group_by(pop) %>% tally() %>% summarise(min(n)) %>% round(digits = 0)` and a maximum of `r upper_sig_stats %>% filter(statname == 'Fu.Li.F') %>% group_by(pop) %>% tally() %>% summarise(max(n)) %>% round(digits = 0)`. The mean number of genes that were unique to a population was `r flf_pop_unique_genes_upper %>% summarise(mean(n)) %>% round(digits = 0)` (SD = `r flf_pop_unique_genes_upper %>% summarise(sd(n)) %>% round(digits = 0)`) and minimum of `r flf_pop_unique_genes_upper %>% summarise(min(n)) %>% round(digits = 0)` and maximum of `r flf_pop_unique_genes_upper %>% summarise(max(n)) %>% round(digits = 0)`.



```{r, results='asis'}
global_summary %>% ungroup%>% filter(stat == 'Fu.Li.F') %>% select(pop, mean, sd, min, lower_1,median,upper_99, max) %>% mutate_at(vars('mean','sd','min','lower_1','upper_99','median','max'), funs(sprintf("%.3f",. ))) %>%  data.frame() %>% pander::pandoc.table(. , justify = c('left', rep('right', NCOL(.)-1)))
```

CIM



<!--
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions 
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp of region 
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes 
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Fu.Li.F', !pop %in% c('CIM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only from CIM
-->
```{r}
gr <- lower_sig_stats %>% filter(pop == "CIM", statname == 'Fu.Li.F') %>%  GenomicRanges::GRanges() %>% GenomicRanges::reduce()
gr[GenomicRanges::width(gr) >= 20 * 10e3]
```

NZM



<!--
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions 
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp of region 
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes 
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('NZM')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Fu.Li.F', !pop %in% c('NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only from NZM
-->

```{r}
gr <- lower_sig_stats %>% filter(pop == "NZM", statname == 'Fu.Li.F') %>%  GenomicRanges::GRanges() %>% GenomicRanges::reduce()
gr[GenomicRanges::width(gr) >= 20 * 10e3]
```


SAM




<!--
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions 
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp of region 
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes 
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('SAM')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Fu.Li.F', !pop %in% c('SAM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only from SAM
-->
```{r}
gr <- lower_sig_stats %>% filter(pop == "SAM", statname == 'Fu.Li.F') %>%  GenomicRanges::GRanges() %>% GenomicRanges::reduce()
gr[GenomicRanges::width(gr) >= 20 * 10e3]
```


TON




<!--
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions 
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp of region 
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes 
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('TON')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Fu.Li.F', !pop %in% c('TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only from TON
-->
```{r}
gr <- lower_sig_stats %>% filter(pop == "TON", statname == 'Fu.Li.F') %>%  GenomicRanges::GRanges() %>% GenomicRanges::reduce()
gr[GenomicRanges::width(gr) >= 20 * 10e3]
```



EP

- `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM', 'NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions 
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM', 'NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp of region 
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM', 'NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes 
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM', 'NZM')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Fu.Li.F', !pop %in% c('CIM','NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only from EP

WP

- `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions 
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp of region 
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('SAM','TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes 
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('SAM','TON')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Fu.Li.F', !pop %in% c('SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only from WP

POL




<!--
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions 
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp of region 
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM','NZM','SAM','TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes 
- `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM','NZM','SAM','TON')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Fu.Li.F', !pop %in% c('CIM','NZM','SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only from POL
-->

Genes Unique to Polynesian populations

Table \@ref(tab:lowersigflfPol)

#### Zeng's E

```{r, warning=FALSE, message=FALSE}
a <- list()
for(p in sort(unique(lower_sig_stats$pop))){
  a[[p]] <-  lower_sig_stats %>% filter(statname == 'Zeng.E', pop == p) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Zeng.E', pop != p), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()
}
ze_pop_unique_genes_lower <- bind_cols(a) %>% gather("pop","n")
rm(a)

a <- list()
for(p in sort(unique(upper_sig_stats$pop))){
  a[[p]] <-  upper_sig_stats %>% filter(statname == 'Zeng.E', pop == p) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Zeng.E', pop != p), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()
}
ze_pop_unique_genes_upper <- bind_cols(a) %>% gather("pop","n")
rm(a)

```


The mean number of windows that met the significance threshold per population for the lower tail was  `r lower_sig_stats %>% filter(statname == 'Zeng.E') %>% group_by(pop) %>% tally() %>% summarise(mean(n)) %>% round(digits = 0)` (SD = `r lower_sig_stats %>% filter(statname == 'Zeng.E') %>% group_by(pop) %>% tally() %>% summarise(sd(n)) %>% round(digits = 0)`) with a minimum of `r lower_sig_stats %>% filter(statname == 'Zeng.E') %>% group_by(pop) %>% tally() %>% summarise(min(n)) %>% round(digits = 0)` and a maximum of `r lower_sig_stats %>% filter(statname == 'Zeng.E') %>% group_by(pop) %>% tally() %>% summarise(max(n)) %>% round(digits = 0)`. The mean number of genes that were unique to a population was `r ze_pop_unique_genes_lower %>% summarise(mean(n)) %>% round(digits = 0)` (SD = `r ze_pop_unique_genes_lower %>% summarise(sd(n)) %>% round(digits = 0)`) and minimum of `r ze_pop_unique_genes_lower %>% summarise(min(n)) %>% round(digits = 0)` and maximum of `r ze_pop_unique_genes_lower %>% summarise(max(n)) %>% round(digits = 0)`.

The mean number of windows per population that met the significance threshold for the upper tail was `r upper_sig_stats %>% filter(statname == 'Zeng.E') %>% group_by(pop) %>% tally() %>% summarise(mean(n)) %>% round(digits = 0)` (SD = `r upper_sig_stats %>% filter(statname == 'Zeng.E') %>% group_by(pop) %>% tally() %>% summarise(sd(n)) %>% round(digits = 0)`), with a minimum of `r upper_sig_stats %>% filter(statname == 'Zeng.E') %>% group_by(pop) %>% tally() %>% summarise(min(n)) %>% round(digits = 0)` and a maximum of `r upper_sig_stats %>% filter(statname == 'Zeng.E') %>% group_by(pop) %>% tally() %>% summarise(max(n)) %>% round(digits = 0)`. The mean number of genes that were unique to a population was `r ze_pop_unique_genes_upper %>% summarise(mean(n)) %>% round(digits = 0)` (SD = `r ze_pop_unique_genes_upper %>% summarise(sd(n)) %>% round(digits = 0)`) and minimum of `r ze_pop_unique_genes_upper %>% summarise(min(n)) %>% round(digits = 0)` and maximum of `r ze_pop_unique_genes_upper %>% summarise(max(n)) %>% round(digits = 0)`.


```{r, results='asis'}
global_summary %>% ungroup%>% filter(stat == 'Zeng.E') %>% select(pop, mean, sd, min, lower_1,median,upper_99, max) %>% mutate_at(vars('mean','sd','min','lower_1','upper_99','median','max'), funs(sprintf("%.3f",. ))) %>%  data.frame() %>% pander::pandoc.table(., justify = c('left', rep('right', NCOL(.)-1)))
```

CIM



<!--
- `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions 
- `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp of region 
- `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes 
- `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Zeng.E', !pop %in% c('CIM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only from CIM
-->
```{r}
gr <- lower_sig_stats %>% filter(pop == "CIM", statname == 'Zeng.E') %>%  GenomicRanges::GRanges() %>% GenomicRanges::reduce()
gr[GenomicRanges::width(gr) >= 20 * 10e3]
```

NZM




<!--
- `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions 
- `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp of region 
- `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes 
- `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('NZM')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Zeng.E', !pop %in% c('NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only from NZM
-->
```{r}
gr <- lower_sig_stats %>% filter(pop == "NZM", statname == 'Zeng.E') %>%  GenomicRanges::GRanges() %>% GenomicRanges::reduce()
gr[GenomicRanges::width(gr) >= 20 * 10e3]
```


SAM




<!--
- `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions 
- `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp of region 
- `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes 
- `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('SAM')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Zeng.E', !pop %in% c('SAM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only from SAM
-->
```{r}
gr <- lower_sig_stats %>% filter(pop == "SAM", statname == 'Zeng.E') %>%  GenomicRanges::GRanges() %>% GenomicRanges::reduce()
gr[GenomicRanges::width(gr) >= 20 * 10e3]
```

TON




<!--
- `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions 
- `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp of region 
- `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes 
- `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('TON')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Zeng.E', !pop %in% c('TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only from TON
-->
```{r}
gr <- lower_sig_stats %>% filter(pop == "TON", statname == 'Zeng.E') %>%  GenomicRanges::GRanges() %>% GenomicRanges::reduce()
gr[GenomicRanges::width(gr) >= 20 * 10e3]
```


EP

- `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM', 'NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions 
- `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM', 'NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp of region 
- `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM', 'NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes 
- `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM', 'NZM')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Zeng.E', !pop %in% c('CIM','NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only from EP

WP

- `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions 
- `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp of region 
- `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('SAM','TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes 
- `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('SAM','TON')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Zeng.E', !pop %in% c('SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only from WP

POL



<!--
- `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions 
- `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp of region 
- `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM','NZM','SAM','TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes 
- `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM','NZM','SAM','TON')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Zeng.E', !pop %in% c('CIM','NZM','SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only from POL
-->

Genes Unique to Polynesian populations

Table \@ref(tab:lowerzePol)

#### iHS/nSL

```{r}
sig_ihs <- readRDS('~/data/NZ_coreExome_1kgp/haplotype/sig_ihs_clus-14-7-2017.RDS')
sig_ihs %>% group_by(pop) %>% tally()
```

The Polynesians populations had a mean of `r sig_ihs %>% group_by(pop,superpop) %>% tally() %>% group_by(superpop) %>% summarise(mn = mean(n)) %>% filter(superpop == 'POL') %>% .[['mn']] %>% sprintf("%.0f",.)` SNPs with a \gls{ihs} value that was significant.<!-- This compared to `r sig_ihs %>% group_by(pop,superpop) %>% tally() %>% group_by(superpop) %>% summarise(mn = mean(n)) %>% filter(superpop == 'AFR') %>% .[['mn']]` for the \gls{afr}, `r sig_ihs %>% group_by(pop,superpop) %>% tally() %>% group_by(superpop) %>% summarise(mn = mean(n)) %>% filter(superpop == 'AMR') %>% .[['mn']]` from \gls{amr}, `r sig_ihs %>% group_by(pop,superpop) %>% tally() %>% group_by(superpop) %>% summarise(mn = mean(n)) %>% filter(superpop == 'EAS') %>% .[['mn']]` for \gls{eas}, `r sig_ihs %>% group_by(pop,superpop) %>% tally() %>% group_by(superpop) %>% summarise(mn = mean(n)) %>% filter(superpop == 'EUR') %>% .[['mn']]` for \gls{eur}, and `r sig_ihs %>% group_by(pop,superpop) %>% tally() %>% group_by(superpop) %>% summarise(mn = mean(n)) %>% filter(superpop == 'SAS') %>% .[['mn']]` for \gls{sas}. Found only in the \gls{cim} population were `r sig_ihs %>% anti_join(., sig_ihs %>% filter(pop != 'POL_CIM'), by = c('genename')) %>% tally()` \gls{snp}s from `r sig_ihs %>% anti_join(., sig_ihs %>% filter(pop != 'POL_CIM'), by = c('genename')) %>% select(genename) %>% distinct() %>% NROW()` distinct genes. 
-->

There were `r sig_ihs %>% anti_join(., sig_ihs %>% filter(superpop != 'POL'), by = c('genename')) %>% select(pop, genename) %>% distinct() %>% group_by(genename) %>% tally() %>% filter( n == 4) %>% NROW()` genes that all Polynesian populations and no other super populations had a \gls{snp} with a significant \gls{ihs} marker, and there were `r sig_ihs %>% anti_join(., sig_ihs %>% filter(superpop != 'POL'), by = c('genename')) %>% select(pop, genename) %>% distinct() %>% group_by(genename) %>% tally() %>% filter(n > 0) %>% NROW()` genes that at least one Polynesian population and no other super populations had.


```{r}
sig_ihs %>% anti_join(., sig_ihs %>% filter(superpop != 'POL'), by = c('genename')) %>% select(pop, genename) %>% distinct() %>% group_by(genename) %>% tally() %>% filter( n == 4)
```

```{r}
sig_ihs %>% anti_join(., sig_ihs %>% filter(superpop != 'POL'), by = c('genename')) %>% select(pop, genename) %>% distinct() %>% group_by(genename) %>% tally()

sig_ihs %>% anti_join(., sig_ihs %>% filter(!pop %in% c('POL_CIM', 'POL_NZM')), by = c('genename')) %>% select(pop, genename) %>% distinct() %>% group_by(genename) %>% tally() %>% filter( n == 2)

sig_ihs %>% anti_join(., sig_ihs %>% filter(!pop %in% c('POL_SAM', 'POL_TON')), by = c('genename')) %>% select(pop, genename) %>% distinct() %>% group_by(genename) %>% tally() %>% filter( n == 2)
```


Gout/urate specific:
```{r}
sig_ihs %>% filter(genename %in% urate_goutGR$SYMBOL) %>% group_by(pop, genename) %>% tally() %>% arrange(genename)
```





There were a total of XXX regions that had significant
```{r}
ihs_clus <-readRDS('~/data/NZ_coreExome_1kgp/haplotype/ihs_clus_regions-14-7-2017.RDS')
ihs_clus %>% group_by(pop) %>% tally()
```

```{r}
sig_nsl <- readRDS('~/data/NZ_coreExome_1kgp/haplotype/sig_nsl_clus-14-7-2017.RDS')
sig_nsl %>% group_by(pop) %>% tally()
```



```{r}
nsl_clus <-readRDS('~/data/NZ_coreExome_1kgp/haplotype/nsl_clus_regions-14-7-2017.RDS')
nsl_clus %>% group_by(pop) %>% tally()
```

- overlap in regions between ihs and nsl
- how many significant snps

ihs
- total snps significant
- mean per population

```{r}

```



nsl
- total snps significant
- mean per population



difference between ihs and nsl:
```{r}
sig_ihs %>% select(chrom, chrom_start, chrom_end, pop)%>% inner_join(., sig_nsl, by =c("chrom","chrom_start", "chrom_end","pop")) %>% group_by(pop, genename) %>% tally() %>%  arrange(genename)
```



#### Summary of Intra

```{r}
lower_sig_stats %>% filter(statname %in% c('Tajima.D','Fay.Wu.H','Zeng.E', 'Fu.Li.F', 'Fu.Li.D'), pop %in% c('CIM','NZM','SAM','TON')) %>% select(pop,statname ,SYMBOL) %>% distinct() %>% group_by(pop, SYMBOL) %>% tally() %>%filter(n > 1) %>% group_by(SYMBOL) %>% tally()  %>% arrange(desc(nn))
```


```{r}
lower_sig_stats %>% filter(SYMBOL %in% urate_goutGR$SYMBOL) %>% select(SYMBOL, pop) %>% anti_join(., lower_sig_stats %>% filter(!pop %in% c("CIM","NZM","SAM","TON")) %>% select(SYMBOL, pop), by = "SYMBOL")%>% distinct() %>% spread(SYMBOL, SYMBOL, fill = "")
```

Of the windows that intersected a significance threshold in the lower tail there were only two genes from the urate associated loci that were significant in only the \gls{sam} and \gls{ton} was _PRKAG2_


```{r}
upper_sig_stats %>% filter(statname %in% c('Tajima.D','Fay.Wu.H','Zeng.E', 'Fu.Li.F', 'Fu.Li.D'), pop %in% c('CIM','NZM','SAM','TON')) %>% select(pop,statname ,SYMBOL) %>% distinct() %>% group_by(pop, SYMBOL) %>% tally() %>%filter(n > 1) %>% group_by(SYMBOL) %>% tally()  %>% arrange(desc(nn))
```

regions that differed between the 2 stats?

### Inter

#### F~ST~
<!-- need fst at a windowed level -->
```{r}
fst <- readRDS('~/data/NZ_coreExome_1kgp/whole_chr_fst_popgenome.3-7-2017.RDS')
  a <- bind_rows(lapply(fst, function(x){x %>% mutate(pop1 = as.character(pop1), pop2 = as.character(pop2))%>% select(-p1,-p2) %>% filter(!pop1 %in% c('WPN','EPN','NAD'), !pop2 %in% c('WPN','EPN','NAD')) %>% left_join(., panel %>% mutate(super_pop = paste(super_pop, pop, sep = '_')) %>% select("pop1" = pop, super_pop) %>% distinct(), by = "pop1" ) %>% select(-pop1) %>% spread(., "super_pop", fst )})) %>% group_by(pop2) %>% summarise_all(mean) %>% left_join(., panel %>% mutate(super_pop = paste(super_pop, pop, sep = '_')) %>% select("pop2" = pop, super_pop) %>% distinct(), by = "pop2" ) %>% select(Population = super_pop, everything(),-chrom, -pop2) %>% arrange(Population) %>% mutate_if(is.numeric, funs(sprintf('%.4f',.))) %>% data.frame(., row.names = NULL)
  
```

```{r, results = 'asis'}
xtable::print.xtable(xtable::xtable(a, caption = "Fst"),floating = TRUE,  rotate.colnames = TRUE, floating.environment = "sidewaystable" , scalebox = 0.5, include.rownames = FALSE, comment = FALSE, booktabs = TRUE)
```


```{r, results = 'asis', eval = FALSE}
knitr::kable(a, booktabs = TRUE) %>% kableExtra::kable_styling(latex_options = 'scale_down')%>%  kableExtra::group_rows("African", 1, 7) %>%
kableExtra::group_rows("American", 8, 11) %>% kableExtra::group_rows("East Asian", 12,16) %>% kableExtra::group_rows("European", 17,22) %>% kableExtra::group_rows("Polynesian",23, 26)%>% kableExtra::group_rows("South Asian", 27, 31) %>% kableExtra::landscape()
```

<!--
table
-->

supplemental table 5 @1KGP2015snp

#### $\Delta$ DAF

<!--
summary of allele frequencies
-->

#### XP-EHH

```{r, eval = FALSE}
xpehh <- read_delim('~/data/NZ_coreExome_1kgp/haplotype/sig_xpehh_5-6-2017.csv', delim = ',')

# only genes that were significant for CIM with POL populations and not found in any other pops
xpehh %>% select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% filter(pop1 == 'CIM', pop2 %in% c('NZM','SAM','TON')) %>%  group_by(pop1,pop2,genename) %>% tally() %>% anti_join(., xpehh %>% select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% filter(pop1 == 'CIM', !pop2 %in% c('NZM','SAM','TON')) %>%  group_by(pop1,pop2,genename) %>% tally(), by = c('genename') )
```



<!--
homozygosity and hetersozygosity
correlation with histone marks/DNAse hypersensitivity
look at selected regions/snps and expression in GTex
-->







## populations

- the mean number of genes that were in significant windows for a population was...
- the mean number of genes that were only found in a single population was...
- the mean number of regions that were significant was...

```{r, warning = FALSE, message = FALSE}
sfs_windows_table<- bind_rows(lapply(c("Tajima.D","Fay.Wu.H","Fu.Li.D","Fu.Li.F","Zeng.E"), function(s){
full_join(bind_rows(lapply(c("CIM","NZM","SAM","TON"), function(p){
data.frame(pop = p,stat = s, tail= "Lower",
           window = lower_sig_stats %>% filter(statname == s, pop == p) %>% select(contains('chrom')) %>% distinct() %>% NROW(),
           indep_regions = lower_sig_stats %>% filter(statname == s, pop == p) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length(), 
  sig_genes = lower_sig_stats %>% filter(statname == s, pop == p) %>% select('SYMBOL') %>% distinct() %>% NROW(), 
  pop_only_sig_genes = lower_sig_stats %>% filter(statname == s, pop == p) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == s, pop != p), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW())
})),

bind_rows(lapply(c("CIM","NZM","SAM","TON"), function(p){
data.frame(pop = p,stat = s, tail= "Upper",
           window = upper_sig_stats %>% filter(statname == s, pop == p) %>% select(contains('chrom')) %>% distinct() %>% NROW(),
           indep_regions = upper_sig_stats %>% filter(statname == s, pop == p) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length(), 
  sig_genes = upper_sig_stats %>% filter(statname == s, pop == p) %>% select('SYMBOL') %>% distinct() %>% NROW(), 
  pop_only_sig_genes = upper_sig_stats %>% filter(statname == s, pop == p) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == s, pop != p), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW())
})), by = "pop")
})
) 

```


```{r, results = 'asis'}
sfs_windows_table %>% select(-tail.x, -tail.y, -stat.x, -stat.y) %>% knitr::kable(., format = 'latex', col.names = c("Pop",rep(c("Windows","Num. Regions","Num. Genes","Num. Pop. Genes"),2)), booktab = TRUE, caption = "(\\#tab:sfsWindowsTable) Number of significant regions and genes in Polynesian populations") %>% kableExtra::kable_styling(latex_options = 'scale_down') %>% kableExtra::add_header_above(., c("" ,"Lower Tail" = 4,"Upper Tail" = 4)) %>% kableExtra::group_rows("Tajima's D", 1,4) %>% kableExtra::group_rows("Fay and Wu's H",5 ,8) %>% kableExtra::group_rows("Fu and Li's D", 9, 12) %>% kableExtra::group_rows("Fu and Li's F", 13, 16) %>% kableExtra::group_rows("Zeng's E", 17, 20)
```
<!--

This section is about reporting how a population went, maybe add in a general comparison of the 1000 genomes populations, eg average number of genes specific to a population

The general format that I want for this section is to get:
- the genes that were reported in multiple stats as significant
- genes that were only reported in a particular population
- runs of depressed regions (eg 7/10 windows) that were significant in a row
- pathway/ontology terms


-->


### Polynesian super population


As shown in table \@ref(tab:sfsWindowsTable), there were similar numbers of genes that intersected the windows that met the significance threshold in either tail between the Polynesian populations of \gls{cim}, \gls{nzm}, \gls{sam}, and \gls{ton}. 

```{r, eval = FALSE}
# sfs sig genes for cim with count of all sfs stats
lower_sig_stats %>% filter(pop == 'CIM') %>% select(-contains("chrom"), -pop) %>% distinct() %>% spread(statname, statname) %>% left_join(., lower_sig_stats %>% filter(pop == 'CIM') %>% select(-contains("chrom"), -pop) %>% distinct() %>% group_by(SYMBOL) %>% tally(), by = "SYMBOL")
```

```{r, eval = FALSE}
# sfs lower significant plus ihs/nsl
lower_sig_stats %>% filter(pop == 'CIM') %>% select(-contains("chrom"), -pop) %>% bind_rows(., sig_ihs %>% filter(pop == "POL_CIM") %>% select(SYMBOL = genename)%>% mutate(statname = 'ihs'), sig_nsl %>% filter(pop == "POL_CIM") %>% select(SYMBOL = genename)%>% mutate(statname = 'nsl')) %>% distinct() %>% spread(statname, statname) %>% left_join(., lower_sig_stats %>% filter(pop == 'CIM') %>% select(-contains("chrom"), -pop) %>% bind_rows(., sig_ihs %>% filter(pop == "POL_CIM") %>% select(SYMBOL = genename)%>% mutate(statname = 'ihs'), sig_nsl %>% filter(pop == "POL_CIM") %>% select(SYMBOL = genename)%>% mutate(statname = 'nsl')) %>% distinct() %>% group_by(SYMBOL) %>% tally(), by = "SYMBOL")
```

```{markdown, echo = FALSE, eval = FALSE}
_Tajima's D_

With the populations that made up the \gls{pol} super population there were `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower tail significance threshold, these windows formed `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions totaling `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. These regions contained `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM','NZM','SAM','TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Tajima.D', !pop %in% c('CIM','NZM','SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant with the \gls{pol} populations with \gls{td}. In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that reached significance and made `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions covering a total of `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. There were `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes with `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM','NZM','SAM','TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Tajima.D', !pop %in% c('CIM','NZM','SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` that were only found in the \gls{pol} populations for \gls{td} within these regions.

_Fay and Wu's H_

Within all of the Polynesian populations used, there were  `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM','NZM','SAM','TON')) %>% NROW()` windows that met the significance threshold in the lower tail. These windows formed `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions, and were at total of `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. There were `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM','NZM','SAM','TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes that intersected these regions and `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM','NZM','SAM','TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fay.Wu.H', !pop %in% c('CIM','NZM','SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes were only significant in the \gls{pol} populations for \gls{fwh}.

For the upper tail there were  `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM','NZM','SAM','TON')) %>% NROW()` windows that were reduced to `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions totaling `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp that met the threshold for significance. In these regions there were `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM','NZM','SAM','TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes and `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM','NZM','SAM','TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fay.Wu.H', !pop %in% c('CIM','NZM','SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes were only significant in the \gls{pol} populations in \gls{fwh}.

_Fu and Li's D_

With the populations that made up the \gls{pol} super population there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower tail significance threshold, these windows formed `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions totaling `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. These regions contained `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM','NZM','SAM','TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.D', !pop %in% c('CIM','NZM','SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant with the \gls{pol} populations with \gls{fld}. In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that reached significance and made `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions covering a total of `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. There were `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes with `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM','NZM','SAM','TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.D', !pop %in% c('CIM','NZM','SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` that were only found in the \gls{pol} populations for \gls{fld} within these regions.

_Fu and Li's F_

With the populations that made up the \gls{pol} super population there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower tail significance threshold, these windows formed `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions totaling `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. These regions contained `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM','NZM','SAM','TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM','NZM','SAM','TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.F', !pop %in% c('CIM','NZM','SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant with the \gls{pol} populations with \gls{flf}. In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that reached significance and made `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions covering a total of `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. There were `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM','NZM','SAM','TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes with `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM','NZM','SAM','TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.F', !pop %in% c('CIM','NZM','SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` that were only found in the \gls{pol} populations for \gls{flf} within these regions.

_Zeng's E_

With the populations that made up the \gls{pol} super population there were `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower tail significance threshold, these windows formed `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions totaling `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. These regions contained `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM','NZM','SAM','TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM','NZM','SAM','TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Zeng.E', !pop %in% c('CIM','NZM','SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant with the \gls{pol} populations with \gls{ze}. In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that reached significance and made `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions covering a total of `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. There were `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM','NZM','SAM','TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes with `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM','NZM','SAM','TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Zeng.E', !pop %in% c('CIM','NZM','SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` that were only found in the \gls{pol} populations for \gls{ze} within these regions.
```


A complete list of all of the significant genes for the Polynesian populatoins from the lower tail can be found in tables \@ref(tab:lowersigtdPol), \@ref(tab:lowersigfwhPol), \@ref(tab:lowersigfldPol), \@ref(tab:lowersigflfPol), and \@ref(tab:lowersigzePol)

### Cook Island Maori


The \gls{cim} population had similar numbers of windows that were significant in both tails of the distribution compared to the other Polynesian populations but slightly more genes intersected these windows, and were also only significant in \gls{cim} for the lower tail. The upper tail had slightly fewer genes compared to the other Polynesian populations.

In the lower tail there were  `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes for \gls{td},  `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` for \gls{fwh},  `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` for \gls{fld},  `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` for \gls{flf}, and  `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` for \gls{ze}, that intersected windows that met the significance threshold. There were only two genes - _FANCM_ and _RGS3_ - that had windows that met the significance threshold for each of \gls{td}, \gls{fwh}, \gls{fld}, \gls{flf}, and \gls{ze}. There were `r lower_sig_stats %>% filter(pop == 'CIM') %>% select(-contains("chrom"), -pop) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% arrange(desc(n)) %>% filter(n == 4) %>% NROW()` genes that met the threshold in four of the selection statistics, `r lower_sig_stats %>% filter(pop == 'CIM') %>% select(-contains("chrom"), -pop) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% arrange(desc(n)) %>% filter(n == 3) %>% NROW()`  in three selection statistics, `r lower_sig_stats %>% filter(pop == 'CIM') %>% select(-contains("chrom"), -pop) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% arrange(desc(n)) %>% filter(n == 2) %>% NROW()` in two selection statistives, and `r lower_sig_stats %>% filter(pop == 'CIM') %>% select(-contains("chrom"), -pop) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% arrange(desc(n)) %>% filter(n == 1) %>% NROW()` with only one selection statistic meeting the threshold for the lower tail.

\Gls{cim} also had the most independent consecutive regions for \gls{td} and \gls{fwh} out of the Polynesian populations.






```{r}
tmp_td <- lower_sig_stats %>% filter(pop == "CIM", statname == "Tajima.D") %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce(., ignore.strand = TRUE)
tmp_td$stat <- "td"

tmp_fwh <- lower_sig_stats %>% filter(pop == "CIM", statname == "Fay.Wu.H") %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce(., ignore.strand = TRUE)
tmp_fwh$stat <- "fwh"

tmp <- IRanges::slice(GenomicRanges::coverage(c(tmp_td, tmp_fwh))[[4]], 2)
tmp@ranges
```

There were fewer genes that intersected the windows that met the threshold for significance in the upper tail for each of \gls{td} (`r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()`), \gls{fwh} (`r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()`), \gls{fld} (`r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()`), \gls{flf} (`r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()`), and \gls{ze} (`r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()`) compared with the lower tail. In the upper tail there was only a single gene - _CSMD1_ - that had windows that met the significance threshold in \gls{td}, \gls{fwh}, \gls{fld}, \gls{flf}, and \gls{ze}. There were  `r upper_sig_stats %>% filter(pop == 'CIM') %>% select(-contains("chrom"), -pop) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% arrange(desc(n)) %>% filter(n == 4) %>% NROW()` genes that met the threshold with four of the selection statistics,  `r upper_sig_stats %>% filter(pop == 'CIM') %>% select(-contains("chrom"), -pop) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% arrange(desc(n)) %>% filter(n == 3) %>% NROW()` genes that met the threshold in three selection statistics,  `r upper_sig_stats %>% filter(pop == 'CIM') %>% select(-contains("chrom"), -pop) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% arrange(desc(n)) %>% filter(n == 2) %>% NROW()` genes with two selection statistics, and  `r upper_sig_stats %>% filter(pop == 'CIM') %>% select(-contains("chrom"), -pop) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% arrange(desc(n)) %>% filter(n == 1) %>% NROW()` genes with only one selection statistic that met the upper tail significance threshold.

\gls{cim} had the most independent regions for \gls{flf} and \gls{ze}, and the most genes that were population specific, out of the the Polynesian populations.

- fwh most pop specific
- flf most regions
- ze most regions


```{r, eval = FALSE}
upper_sig_stats %>% filter(pop == 'CIM') %>% select(-contains("chrom"), -pop) %>% distinct() %>% spread(statname, statname) %>% left_join(., upper_sig_stats %>% filter(pop == 'CIM') %>% select(-contains("chrom"), -pop) %>% distinct() %>% group_by(SYMBOL) %>% tally(), by = "SYMBOL")
```

From the lower tail there were `r consec_lower_regions$CIM$Tajima.D %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce(ignore.strand = TRUE) %>% length()` regions of extended significant windows for \gls{td}, `r consec_lower_regions$CIM$Fay.Wu.H %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce(ignore.strand = TRUE) %>% length()` for \gls{fwh}, `r consec_lower_regions$CIM$Fu.Li.D %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce(ignore.strand = TRUE) %>% length()` for \gls{fld}, `r consec_lower_regions$CIM$Fu.Li.F %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce(ignore.strand = TRUE) %>% length()` for \gls{flf}, and `r consec_lower_regions$CIM$Zeng.E %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce(ignore.strand = TRUE) %>% length()` for \gls{ze}.

From the haplotypic tests for selection (\gls{ihs} and \gls{nsl}) there were `r sig_ihs %>% filter(pop == "POL_CIM") %>% select(genename) %>% anti_join(., sig_ihs %>% filter(pop !="POL_CIM") %>% select(genename), by = 'genename') %>% distinct() %>% NROW()` genes that were only significant in \gls{cim} for \gls{ihs}, and `r sig_nsl %>% filter(pop == "POL_CIM") %>% select(genename) %>% anti_join(., sig_nsl %>% filter(pop !="POL_CIM") %>% select(genename), by = 'genename') %>% distinct() %>% NROW()` for \gls{nsl}.





```{markdown, echo = FALSE, eval = FALSE}
_Tajima's D_

The \gls{cim} had a total of `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the significance threshold in the lower tail. This represented `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` non-continuous regions representing `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp of th genome. Within these windows were `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which only `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Tajima.D', !pop %in% c('CIM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes were distinct for \gls{cim} out of the \gls{td} results. In the upper tail there were  `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the upper significance threshold, these windows formed `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM')) %>% select(contains('chrom'))  %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` non-continuous regions covering `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. These regions contained `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Tajima.D', !pop %in% c('CIM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only from \gls{cim} for \gls{td}.

_Fay and Wu's H_

There were `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the significance threshold in the lower tail. These windows made up `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions with a total of  `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. Within these regions were `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fay.Wu.H', !pop %in% c('CIM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant in the \gls{cim} population for \gls{fwh}

In the upper tail there were  `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that me the significance threshold, consisting of `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions that totaled `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. There were  `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes within these regions and `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fay.Wu.H', !pop %in% c('CIM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes were only significant in \gls{cim} for \gls{fwh}.

_Fu and Li's D_

For the \gls{cim} population there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower tail threshold for significance and made `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions covering `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. In these regions there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which  `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.D', !pop %in% c('CIM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant in \gls{cim} for \gls{fld}. In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the significance threshold. These windows made `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions with a total of `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp containing `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, with  `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.D', !pop %in% c('CIM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only meeting significance in \gls{cim} for \gls{fld}.

_Fu and Li's F_

For the \gls{cim} population there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower tail threshold for significance and made `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions covering `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. In these regions there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which  `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.F', !pop %in% c('CIM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant in \gls{cim} for \gls{flf}. In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the significance threshold. These windows made `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions with a total of `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp containing `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, with  `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.F', !pop %in% c('CIM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only meeting significance in \gls{cim} for \gls{flf}.

_Zeng's E_

For the \gls{cim} population there were `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower tail threshold for significance and made `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions covering `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. In these regions there were `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which  `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Zeng.E', !pop %in% c('CIM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant in \gls{cim} for \gls{ze}. In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the significance threshold. These windows made `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions with a total of `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp containing `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, with  `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Zeng.E', !pop %in% c('CIM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only meeting significance in \gls{cim} for \gls{ze}.
```

were there any genes that were significant from the gene lists and related to the thrifty gene hypothesis?


### New Zealand Maori

The \gls{nzm} population had the largest number of genes intersecting windows that met the lower tail significance threshold that were only reported in a single population for \gls{fwh}, \gls{fld}, \gls{flf}, and \gls{ze} out of the Polynesian populations.

There were `r lower_sig_stats %>% filter(pop == 'NZM') %>% select(SYMBOL) %>% distinct() %>% anti_join(.,bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(pop !="NZM") %>% select(SYMBOL), by = "SYMBOL") %>% NROW()` genes that were had windows that intersected the lower tail significance threshold and `r upper_sig_stats %>% filter(pop == 'NZM') %>% select(SYMBOL) %>% distinct() %>% anti_join(.,bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(pop !="NZM") %>% select(SYMBOL), by = "SYMBOL") %>% NROW()` for the upper tail in at least one selection statistic and were not found in any of the other populations.


There were `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes for \gls{td} that intersected with windows that met the significance threshold in the lower tail. There were a similar numbers for \gls{fld} (`r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW()`), and \gls{flf} (`r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW()`). There were more genes for \gls{fwh} (`r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW()`), and \gls{ze} (`r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW()`).



- td most genes
- fld most genes
- flf most genes
- ze most regions, genes, and pop specific genes

From the upper tail, there were fewer genes overall that intersected windows that met the significance threshold than for the lower threshold. \gls{td} had `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, \gls{fwh} `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW()`, \gls{fld} `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW()`, \gls{flf} `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW()`, and \gls{ze} had `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW()`. Within the Polynesian populations, \gls{nzm} had the most independent consecutive regions for \gls{fwh}.\gls{fld} and \gls{flf} also had the most genes that intersected significant windows and the highest number of genes that were only found in a single population, of the Polynesian populations.

- fwh most regions
- fld most regions, genes, and pop specific genes
- flf most regions, genes, and pop specific genes

```{markdown, echo = FALSE, eval = FALSE}
<!--
_Tajimas D_

The \gls{nzm} had `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower significance threshold. These windows made `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions covering `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. Theses regions contained `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('NZM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Tajima.D', !pop %in% c('NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes were only from \gls{nzm}. The upper tail had `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that formed `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions covering `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. There were `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('NZM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Tajima.D', !pop %in% c('NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were unique to \gls{nzm} for \gls{td}.

_Fay and Wu's H_

There were `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows forming  `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions that met the significance threshold for the lower tail. These regions were a total of `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp and had `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('NZM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fay.Wu.H', !pop %in% c('NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` of which were only significant in the \gls{nzm} population for \gls{fwh}.

In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that formed  `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` continuous regions that met the significance threshold. These regions covered `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp and had `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('NZM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fay.Wu.H', !pop %in% c('NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant in the \gls{nzm} population for \gls{fwh}


_Fu and Li's D_

For the \gls{nzm} population there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower tail threshold for significance and made `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions covering `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. In these regions there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which  `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('NZM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.D', !pop %in% c('NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant in \gls{nzm} for \gls{fld}. In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the significance threshold. These windows made `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions with a total of `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp containing `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, with  `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('NZM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.D', !pop %in% c('NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only meeting significance in \gls{nzm} for \gls{fld}.

_Fu and Li's F_

For the \gls{nzm} population there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower tail threshold for significance and made `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions covering `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. In these regions there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which  `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('NZM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.F', !pop %in% c('NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant in \gls{nzm} for \gls{flf}. In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the significance threshold. These windows made `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions with a total of `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp containing `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, with  `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('NZM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.F', !pop %in% c('NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only meeting significance in \gls{nzm} for \gls{flf}.

_Zeng's E_

For the \gls{nzm} population there were `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower tail threshold for significance and made `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions covering `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. In these regions there were `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which  `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('NZM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Zeng.E', !pop %in% c('NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant in \gls{nzm} for \gls{ze}. In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the significance threshold. These windows made `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions with a total of `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp containing `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, with  `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('NZM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Zeng.E', !pop %in% c('NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only meeting significance in \gls{nzm} for \gls{ze}.
-->
```

### Samoan

There were `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes that intersected the windows that met the significance threshold in the lower tail for \gls{td}. For \gls{fwh}, there were `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, \gls{fld} had `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()`, \gls{flf} had `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()`, and \gls{ze} had `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()`. \gls{sam} had the most (XXX) independent consecutive regions of the Polynesian populations for \gls{fld}.

lower

- td most specific
- fld most regions

In the upper tail, there were `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes for \gls{td} that intersected windows that met the significance threshold. \gls{fwh} had `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()`, \gls{fld} had `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()`, \gls{flf} had `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()`, and \gls{ze} had `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()`.

upper

- td most specific

\gls{sam} had the most genes that were specific to \gls{sam} for \gls{td} out of the Polynesian populations for both significance thresholds.

```{markdown, echo = FALSE, eval = FALSE}
_Tajima's D_

In the \gls{sam} population there were `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that made up `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` continuous regions covering `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. In these regions there were`r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Tajima.D', !pop %in% c('SAM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were unique to \gls{sam}. For the upper tail there were  `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that formed  `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions with a total of `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. There were `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes that were covered by these regions, `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Tajima.D', !pop %in% c('SAM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` that were only found in \gls{sam} for \gls{td}.

_Fay and Wu's H_

There were  `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that formed  `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions, totaling  `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp that met the significance threshold for the lower tail. These regions had `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('SAM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fay.Wu.H', !pop %in% c('SAM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant in the \gls{sam} population for \gls{fwh}.

In the upper tail there were  `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the threshold for significance. These windows formed `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions with a total of  `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. There were  `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('SAM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fay.Wu.H', !pop %in% c('SAM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only from the \gls{sam} population for \gls{fwh}. 

_Fu and Li's D_

For the \gls{sam} population there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower tail threshold for significance and made `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions covering `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. In these regions there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which  `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('SAM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.D', !pop %in% c('SAM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant in \gls{sam} for \gls{fld}. In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the significance threshold. These windows made `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions with a total of `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp containing `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, with  `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('SAM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.D', !pop %in% c('SAM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only meeting significance in \gls{sam} for \gls{fld}.

_Fu and Li's F_

For the \gls{sam} population there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower tail threshold for significance and made `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions covering `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. In these regions there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which  `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('SAM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.F', !pop %in% c('SAM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant in \gls{sam} for \gls{flf}. In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the significance threshold. These windows made `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions with a total of `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp containing `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, with  `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('SAM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.F', !pop %in% c('SAM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only meeting significance in \gls{sam} for \gls{flf}.

_Zeng's E_

For the \gls{sam} population there were `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower tail threshold for significance and made `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions covering `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. In these regions there were `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which  `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('SAM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Zeng.E', !pop %in% c('SAM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant in \gls{sam} for \gls{ze}. In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the significance threshold. These windows made `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions with a total of `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp containing `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, with  `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('SAM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Zeng.E', !pop %in% c('SAM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only meeting significance in \gls{sam} for \gls{ze}.
```


### Tongan

lower

- fwh most genes, and most specific
- flf most regions, and most specific


upper

- td most regions, and most genes
- fwh most genes
- flf most genes and most specific
- ze most genes and most specific


```{markdown, echo = FALSE, eval = FALSE}
_Tajima's D_

For the \gls{ton} population there were `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower tail threshold for significance and made `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions covering `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. In these regions there were `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which  `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Tajima.D', !pop %in% c('TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant in \gls{ton} for \gls{td}. In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the significance threshold. These windows made `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions with a total of `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp containing `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, with  `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Tajima.D', !pop %in% c('TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only meeting significance in \gls{ton} for \gls{td}.

_Fay and Wu's H_

There were  `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that reached the significance threshold in the lower tail. These windows made `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()`independent continuous regions totaling `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6`. Within these regions there were  `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fay.Wu.H', !pop %in% c('TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes that were only significant in \gls{ton} for \gls{fwh}, from `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes that were significant.

In the upper tail there were  `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the significance threshold making up  `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions that were independent and continuous with a total length of `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. Within these regions there were `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which  `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fay.Wu.H', !pop %in% c('TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were significant in only \gls{ton} for \gls{fwh}.

_Fu and Li's D_

For the \gls{ton} population there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower tail threshold for significance and made `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions covering `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. In these regions there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which  `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.D', !pop %in% c('TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant in \gls{ton} for \gls{fld}. In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the significance threshold. These windows made `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions with a total of `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp containing `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, with  `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.D', !pop %in% c('TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only meeting significance in \gls{ton} for \gls{fld}.

_Fu and Li's F_

For the \gls{ton} population there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower tail threshold for significance and made `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions covering `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. In these regions there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which  `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.F', !pop %in% c('TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant in \gls{ton} for \gls{flf}. In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the significance threshold. These windows made `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions with a total of `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp containing `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, with  `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.F', !pop %in% c('TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only meeting significance in \gls{ton} for \gls{flf}.

_Zeng's E_

For the \gls{ton} population there were `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower tail threshold for significance and made `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions covering `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. In these regions there were `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which  `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Zeng.E', !pop %in% c('TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant in \gls{ton} for \gls{ze}. In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the significance threshold. These windows made `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions with a total of `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp containing `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, with  `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Zeng.E', !pop %in% c('TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only meeting significance in \gls{ton} for \gls{ze}.
```



## Chapter Discussion
<!--
Important to remember that the modern day populations esp AFR won't necessarily have the same results as AFR pops did 100 kya

Pipeline discussion - how were the results?
  - Are the genes that are picked up similar to what has been discovered?

Is there anything special about the population comparisons?
  - shared genes between similar populations?
  - East/West split seen?
  - NZC vs GBR similarity?
  
  @Rafajlovic2014 has a good talk about the use of the empirical distribution for sfs stats.
  
  
Because fld and flf are based on the number of singletons, how is this going to be affected by using chip data that should have essentially 0. presumable this will dramatically affect the interpretability of the statistic and push everything over to the right but is it acceptable to recentre the distribution and look that the extremes from there? does it have biological plausibility?  

what does it mean to have a significant window of one tail neighboured by the other tail? Or on the broad scale what does it mean to have a gene classed as significant in both tails?

effect of phasing when polynesian populations aren't represented in the 1000 genomes populations

window size meant that snps that snps not in the window might not even be near the actual window ie -45 and +45 kb away from the edge of the defined window in the worst case. This would lead to some difficulty in definitive location. There was a trade off between snp density and window size.


limitations of using the chip data versus imputed data
- ascertainment bias of snps on the chip
- imputation from the chip might not work well given the polys aren't (unconfirmed) well represented in the reference panel
- phasing of haplotypes doesn't have polynesian haplotypes in the panel

-->

The shift in the distributions due to having chip data for the selection tests that rely on frequency spectrum is seen here. Mostly the African populations have shifted so far to the right with Tajima's D that the first percentile is > 0. 
XXX The intermediate frequency variants are over represented

With the adaptation of the selection and neutrality tests into genome-wide usage, specifically the sliding window approach, there was the need to balance the number of segregating sites with the window size [@Pybus2014]. This balance is entirely influenced by \gls{snp} density with whole genome (re)sequence data offering high density XXX snps/kb, but \gls{snp} chip data being less dense XXX snps/kb. This means that window size needs to increase for \gls{snp} chip data in order to have the same number of segregating sites. However, as window size increases it also reduces the resolution of the genome and with it the ability to associate a window with particular genomic features.




