# Positive Selection in Polynesian Populations

```{r source, include = FALSE, echo = FALSE}
knitr::opts_chunk$set(echo = FALSE)
library(phdFunctions)
#devtools::install_github('murraycadzow/UpSetR')
library(UpSetR)
library(Homo.sapiens)
library(ggbio)
library(GenomicRanges)
library(knitr)
library(kableExtra)
library(xtable)
library(stringr)
data(genesymbol, package = "biovizBase")
library(tidyverse)

source('scripts/multiplot.R')
source('scripts/03_selectionpipeline.R')
poly_pop <- c('CIM','NZM','TON','SAM')
```


In this chapter I investigate regions of the genome that exhibit signatures of selection in Polynesian populations with a particular focus on loci associated with serum urate and co-morbidities.

Part of this work was originally published in @Cadzow2016 but I have updated the selection analysis to make use of a new dataset that became available after publication. The selection analyses are in this chapter.


## Introduction

<!-- Much of the early work in looking at selection was focused at the gene level [@Wright1951;@Fisher] and the methods that were developed were intended for few loci, or limited sequence from a single region [@ref;@ref;@ref]. This was largely due to the high cost and limited availability of whole genome chip or sequence data. It wasn't until after the completion of the human genome project in 2001 and the advent of 'next generation sequencing' where the cost of full genome data rapidly dropped and there became an abundance of whole genome sequence data [@ref]. -->
At the turn of the millenium the draft reference human genome was released [@ref]. Following from this, there were a few large projects to characterise human variation. The first main project was the HapMap project [@Hapmap2005] and was based on microarray data. This was followed on by the 1000 Genomes Project [@1KGP2010; @1KGP2012; @1KGP2015snp] which was the first large project to based on a whole-genome (re)sequence data. Alongside these variation projects were a few projects that aimed to focus on the genomic diversity of humans, this was the Human Genome Diversity Project [@Cann2002; @Rosenburg2002], and later the Simons Genome Diversity Project [@Mallick2016]. 

Over the last decade there has been a steady flow of studies into genome-wide selection [@voight2006map; @sabeti2006positive; @Coop2009; @tang2007new; @pickrell2009signals; @Grossman2010]. The main population samples for this have been part of the 1000 Genomes Project, which sampled over 2500 individuals for full genome (re)sequencing from five 'super' population groups [@1KGP2015snp]. This dataset, while capturing a large representation of the human population variation, does not have representation for the populations of Polynesia. 


### Positive selection

### Selection in Polynesian populations

Projects to capture the genetic diversity of humans have sampled populations worldwide, but outside the populations of the 1000 Genomes Project are with small numbers for some populations [@Cann2002; @Rosenburg2002; @Mallick2016]. Past studies that have included Oceanic populations have had minimal representation of Polynesian populations, @Kimura2008 only had 24 Tongan individuals. The latest project for human diversity, the Simons Human Diversity Project, did include the New Zealand M$\bar{a}$ori but they were only represented by a single individual and did not feature in many of the analyses [@Mallick2016].

More recently there have been studies that have looked at selection in Polynesian populations but with a focus on a few candidate loci, not at a genome-wide scale [@Myles2011; @Cadzow2016; @Minster2016]. The main feature of these studies has been investigating 'thrifty genes'.

#### Thrifty Gene

The "Thrifty genotype hypothesis" was originally hypothesised by @Neel1962, whereby it was posed that due to the differences between individuals with diabetes and without that how could a disease that should have strong selection against it continue to have high prevalence if there was not some genetic advantage [@Neel1962]. In the Polynesian context, this hypothesis has been invoked in an attempt to explain the higher prevalence of metabolic related diseases such as \gls{t2d} and obesity [@Myles2011; @Minster2016]. Although there are several refutations of the thrifty genetype for Polynesian populations as a being 'just-so' stories [@Cadzow2016; @Gosling2014; @Gosling2015]. Uricase, the enzyme for metabolising urate, was lost through multiple mutations, a reduction in promoter activity followed by 2 further loss of function mutations and occurred before the divergence of humans from the great apes [@Kratzer2014]. The evolutionary history of uricase however has also been discribed as the original thrifty-gene [@Kratzer2014].

### Selection in urate/gout

The benefits of urate include the its ability to increase blood pressure and is thought to have helped with early hominins to stand [@Wantannabe2002] and increases blood flow to the brain. Serum urate is also a potent anti-oxidant which accounts for >50% of the anti-oxidant activity of the blood [@Glantzounis2005; @Parmar2009]. Mono-sodium urate crystals are a potent adjuvant for the immune system and enhance the immune response, providing a protective effect against cancer due to the turnover of cells [@ref; @Ames1981].

<!-- - Possible reasons for the loss @Alvares-Lario2010 -->
<!--     - urate helps raise bp, this would have been beneficial in low salt environments, helped early hominins stand @Wantannabe? -->
<!--     - increases blood flow to the brain -->
<!--     - potent anti-oxidant which accounts for > 50% of the bloods anti-oxidant activity [refs 26,27 alvares-lario2010] -->
<!--     - crystals enhance local immune response - protective against cancer due to the turnover of cells? -->

<!-- ?? The fructose response for leads to insulin desensitivity @Johnston2007? -->
<!--     - fructose blocks glyconeogeneis -> increased glycogen storage -->

The prevalence of \gls{hu} in Polynesia is also higher, with it are the higher prevalences of the co-morbidities of gout, diabetes, obesity, dyslipidemia, kidney disease, and \gls{cvd} [@Winnard2013]. In New Zealand, 40% of individuals with gout also had diabetes or \gls{cvd}.  

\Gls{hu} has two main components: over production of urate, or under excretion. Over production is influenced by diet, such as high purine or fructose content, or under genetic control. Under excretion is the main cause of \gls{hu}, caused by either increased reabsorption or decreased secretion. The ATP powered transporter ABCG2 is responsible for excretion of urate in the gut. There are Polynesian specific variants that lead to a reduction of function in ABCG2 [@phippsgreen2011]. The kidneys are the organ where the majority of secretion and excretion of urate is performed, with about 90% being reabsorbed [@Kutzing2008]. In the proximal tuble, on the apical membrane, the urate transporters of URAT1, OAT4, and SLC2A9 reabsorb urate, whereas ABCC4, NPT4, and NPT1 are responsible for secretion [@ref]. On the basolateral membrane, OAT1-3, increase intracellular uric acid, but SLC2A9 moves uric acid from intracellular to extracellular [@ref]. 

Gout is an immune response to the uric acid crystals that are formed from the precipitation of urate from the blood and \gls{hu} is a prerequisite for the formation of these crystals.

\Gls{hu} in the Pacific is seen in Eastern and Western Polynesian, but the prevalence in the ancestral populations of Near and Remote Oceania is reduced. However, Taiwanese aboriginals also have the higher prevalence of \gls{hu} [@ref].

There is prior evidence of selection at urate associated loci. In the review from @Ramos2017 on natural selection in rheumatic disease, one of the main loci for gout, _SLC2A9_, in the \gls{yri} population had evidence. The region of _BCAS3_ and _NACA2_ also had evidence, originally found by @Grossman2013 in the Asian population from Hapmap. @Zhang2013a had evidence for the associated loci of _ATXN2_ (rs653178) and _RREB1_ (rs675209)  having undergone selection when usnig a F~ST~ based method. The co-morbidities of \gls{hu} also have evidence of selection with loci associated with \gls{t2d} have been found to have evidence of having undergone selection [@pickrell2009signals; @voight2006map; @ref]. Most recently a Samoan specific variant in _CREBRF_ that is protective for \gls{t2d} but causal for obesity has also been identified as having evidence of selection [@Minster2016].

Uric acid has also been identified as being an important part of the malarial response, contributing to cytokine secretion and response of dendritic and T cells [@GallegoDelgado2014].

- jha2015 rs589691?



> mention the out of africa migration again and the polynesian expansion. East and west are different genetically


<!--
motivation: seemed to be little in the way of calculating stats genome wide and wasn't sure how sensitive these stats were to snp density

This chapter is about trying to first of all establish how sensitive the various statistics were to window/slide variation

from the literature there doesn't seem to be any recommendation as to how to compute these statistics genome wide. Up until recently the various statistics were computed at the 'feature' level, usually the gene.

Fst has a paper in early 2000's from Bruce Weir that touched on what sort of size to use howver it largely just came down to 'this window size seems to have enough of a smoothing effect as well as display some of the variation in the data' type approach and was largely just based on a this looks good enough.

Pybus et al specified their window sizes which was roughly 30 kb window and 3kb slide for most statistics however this was decided on unshown data as a trade off between resolution and density.

In cadzow 2014 I tried had tried a few sizes and it it seemed that there was little difference between them other than the smaller the window the more noisy the data looked.

While I was playing around with this for Tajima's D there does seem to be at least some relation to the density of the snps and the window size, which does intuitively make sense, in that if there are none or only 1 snp per window then the stats are going to be pretty poor.

From the core exome work it looks as though 30 kb was ok and at least when the distribution of TD was binned by number of snps per window it didn't look to bad and the overall total distribution of TD looks not too bad. Increasing from 30 kb didn't really change TD that much. the extremes become more extreme but the means increaseas window size increases.
-->

<!-- ascertainment bias will skew selection statistics that are site frequency spectrum dependant in chip data due to the over-representation of intermediate frequency SNPs, these include Tajima's D (upward) and Fay and Wu's H [@Clark2005]


Carlson2005 did windows of 100kb, slide 10kb and mean snps were ~50/window. ascertainment bias esp in the +ve end. continous region on >20 windows where 75% were in bottom 1% was defined as contiguous regions of Tajimas D reduction.
RafajloviÄ‡2014 largely followed Carlson2005, they also discarded windows with less than 5 snps.

Bigman2010 has a custom Tajima's D that is created to be similar to iHS in terms of calculation
-->



### Objectives

- Investigate regions that are under selection in the Polynesian populations
- Investigate regions previously associated with urate, \gls{t2d}, obesity, kidney disease, and metabolic syndrome for signatures of selection

## Methods

The selection of individuals, and preliminary data preparation for the selection data set is detailed in Section \@ref(selectionDataset). It consisted of combining 500 individuals (100 per population) from the Genetics of Gout in Aotearoa study, who had been genoypted on the Illumina CoreExome v24 mircoarray with 2504 individuals from the 1000 Genomes Project. 

The selection and neutrality statistics were calculated for individuals populations using a combination of two main software packages, PopGenome in R, and selectionTools v1.1. PopGenome was used to calculate the intra population frequency spectrum based tests for selection and neutrality. Haplotypic stats of \gls{ihs}, \gls{nsl}, and \gls{xpehh} were calculated using the selectionTools pipeline with significance set as being the 99th percentile for the absolute normalised score. Frequency based statistics were also generated such as \gls{daf} and F~ST~.

#### Calculation of Selection and Neutrality Statistics



<!--
Similar to @Pybus2014, @Rafajlovic2014 and @Carlson2005 a trade off between increasing the number of \gls{snp}s per window and adequate resolution was made and the window size of 100 kb and slide of 10 kb was chosen.
-->
The R package PopGenome v2.2.3 [@Pfeifer2014] was used to calculate the site-frequency spectrum statistics \gls{td}, \gls{fwh}, \gls{fld}, \gls{flf}, \gls{ze}, and F~ST~ over non-overlapping windows of 10 kb (see section \@ref(popgenomeMethods) for further details). Except for F~ST~, the empirical distribution for each population was used to create an upper and  a lower significance threshold. The lower threshold was the 1st percentile and less than zero. The upper significance threshold was the 99th percentile and greater than zero. The greater or less than zero condition was to enable the interpretation of the selection and neutrality statistic results. Contiguous regions of score reduction or elevation were created using a rolling window approach, whereby a condition of 15 or more windows out of a minimum of 20 consecutive 10 kb windows were required to have met a significance threshold in the same tail of the distribution.

The calculation of \gls{ihs}, \gls{nsl}, and \gls{xpehh} were calculated and normalised using selscan v1.1.0b [@Szpiech2014] as part of the selectionTools 1.1 [@Cadzow2014] pipeline as detailed in \@ref(selectionTools). The default setting were used which was a maximum gap extension of 200 kb. Significance was defined as the most extreme 1% of values after normalisation using frequency bins of 0.05.


## Results

```{r}
fdr_low_tab <- fdr_low %>% 
  filter(lowest_quant == lower_per, super_pop == 'POL') %>% 
  arrange(stat, pop) %>%
  ungroup() %>% 
  select(stat, pop, FDR, lowerCI, upperCI, lowest_quant)

fdr_high_tab <- fdr_high %>% 
  filter(highest_quant == upper_per, super_pop == 'POL') %>% 
  arrange(stat, pop) %>% 
  ungroup() %>% 
  select(stat, pop, FDR , lowerCI, upperCI, highest_quant)
```




In this section I will report on each statistic, describing the general distributions. This is followed by more specific report of the individual Polynesian population results. Finally, the results for specific disease associated genes will be reported.

<!--
Look at what windows/genes for each population are interesting
Super populations - 1KGP vs POL
Super Populations - 1KGP vs EPN/WPN
Individual populations - CIM/NZM/SAM/TON/NZC
-->

### Intra-population Selection and Neutrality Statistics


For the frequency spectrum based intra population statistics of \gls{td}, \gls{fwh}, \gls{fld}, \gls{flf}, and \gls{ze} there were population specific distributions. There was similarity however between the populations of a super population. This observation was the basis of the investigation that formed chapter \@ref(clustering). 



```{r numSigWindows}
# add on ihs/nsl
ihs_nsl_sum_tab <- sig_ihs_val %>% 
  group_by(pop) %>% 
  tally() %>% 
  select(pop, n_ihs_snps = n) %>% 
  left_join(., sig_nsl_val %>% 
              group_by(pop) %>% 
              tally() %>% 
              select(pop, n_nsl_snps = n) , by = 'pop') %>% 
  left_join(., sig_ihs_val %>% 
              select(pop, chrom, chrom_start, chrom_end) %>% 
              inner_join(., sig_nsl_val %>% 
                           select(pop, chrom, chrom_start, chrom_end), by = c('pop','chrom','chrom_start','chrom_end') ) %>% 
              group_by(pop) %>% 
              tally(), by = "pop") 

all_sfs_windows_table <- lower_sig_stats %>% 
  mutate(statname = paste0(statname,"_lo")) %>%  
  select(contains('chrom'), pop,statname) %>% 
  distinct() %>% 
  group_by(pop,statname) %>% 
  tally() %>% 
  spread(statname,n) %>% 
  full_join(.,upper_sig_stats %>% 
              mutate(statname = paste0(statname,"_up")) %>% 
              select(contains('chrom'), pop,statname) %>% 
              distinct() %>% 
              group_by(pop,statname) %>%
              tally() %>% 
              spread(statname,n) , by = "pop" ) %>% 
  left_join(., panel %>% 
              select(pop, super_pop) %>% 
              distinct(), by = 'pop') %>% 
  arrange(super_pop) %>%
  select(-super_pop)

left_join(all_sfs_windows_table, ihs_nsl_sum_tab %>% select(-n), by = 'pop') %>%
 knitr::kable(., caption = "(\\#tab:numSigWindows) Number of significant windows/markers by selection stat", booktabs = TRUE, format = 'latex', col.names = c("Population",rep(c("\\gls{fwh}", "\\gls{fld}","\\gls{flf}","\\gls{td}","\\gls{ze}"), 2), 'iHS SNPs', 'nSL SNPs'), escape = FALSE) %>%
  kableExtra::kable_styling(latex_options = 'scale_down') %>%
  kableExtra::add_header_above(., c("", "Lower Tail" = 5, "Upper Tail" = 5, "", "") )%>% 
  kableExtra::group_rows("AFR", 1, 7) %>% 
  kableExtra::group_rows("AMR", 8, 11) %>% 
  kableExtra::group_rows("EAS", 12, 16) %>% 
  kableExtra::group_rows("EUR", 17, 22) %>% 
  kableExtra::group_rows("POL", 23, 26) %>% 
  kableExtra::group_rows("SAS", 27,31) 
```


<!--
table for each stat that give global mins/means/sd/max for each population
-->

#### Tajima's D

<!--
describe the overall tajima's d and where the polynesian populations fell at the summary level
-->

```{r td_main, warning=FALSE, message=FALSE}
a <- list()
for(p in sort(unique(lower_sig_stats$pop))){
  a[[p]] <- lower_sig_stats %>% 
    filter(statname == 'Tajima.D', pop == p) %>% 
    anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>%
                filter(statname == 'Tajima.D', pop != p), by = "SYMBOL") %>%
    select(SYMBOL) %>%
    distinct() %>% 
    NROW()
}

td_pop_unique_genes_lower <- bind_cols(a) %>% gather("pop","n")
rm(a)

a <- list()
for(p in sort(unique(upper_sig_stats$pop))){
  a[[p]] <- upper_sig_stats %>% 
    filter(statname == 'Tajima.D', pop == p) %>% 
    anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>%
                filter(statname == 'Tajima.D', pop != p), by = "SYMBOL") %>%
    select(SYMBOL) %>% 
    distinct() %>% 
    NROW()
}

td_pop_unique_genes_upper <- bind_cols(a) %>% gather("pop","n")
rm(a)

td_super_pop_means <- global_super_summary %>% 
  filter(stat == 'Tajima.D') %>% 
  mutate(range = max -min)

gs_td_table <- create_sel_summary_table("Tajima.D")

pol_td_t.test <- bind_rows(
  lapply(c("AFR","AMR", "EAS","EUR","SAS"), function(sp){
    cbind(SuperPop = sp, 
          global_summary %>% 
            filter(stat == 'Tajima.D') %>% 
            left_join(., panel %>% 
                        select(pop, super_pop) %>% 
                        distinct(), by = 'pop') %>% 
            filter(super_pop %in% c("POL",sp)) %>% 
            ungroup() %>% mutate(super_pop = as.character(super_pop)) %>%
            select(super_pop, mean) %>% 
            do(broom::tidy(t.test( mean ~ super_pop, data = .))))
  })) %>% 
  mutate(adjusted_p = p.adjust(p.value, "bonferroni"))
```

The results for the genome-wide \gls{td} showed that there were population differences in the distribution of \gls{td} values. The mean \gls{td} for the Polynesian populations were the smallest of the super populations at `r global_super_summary %>% filter(super_pop == 'POL', stat == 'Tajima.D') %>% .[["mean"]] %>% three_dp()`  (SD `r global_super_summary %>% filter(super_pop == 'POL', stat == 'Tajima.D') %>% .[["sd"]] %>% three_dp()`) with  `r gs_td_table %>% filter(pop == 'CIM') %>% .[["mean"]] %>% three_dp()` (SD `r gs_td_table %>% filter(pop == 'CIM') %>% .[["sd"]] %>% three_dp()`) for \gls{cim}, `r gs_td_table %>% filter(pop == 'NZM') %>% .[["mean"]] %>% three_dp()` (SD `r gs_td_table %>% filter(pop == 'NZM') %>% .[["sd"]] %>% three_dp()`) for \gls{nzm}, `r gs_td_table %>% filter(pop == 'SAM') %>% .[["mean"]] %>% three_dp()` (SD `r gs_td_table %>% filter(pop == 'SAM') %>% .[["sd"]] %>% three_dp()`) for \gls{sam}, and `r gs_td_table %>% filter(pop == 'TON') %>% .[["mean"]] %>% three_dp()` (SD `r gs_td_table %>% filter(pop == 'TON') %>% .[["sd"]] %>% three_dp()`) for \gls{ton} (Table \@ref(tab:gsTdTable)). The means of the Polynesian populations were lower than all of the other populations except \gls{pel} and \gls{mxl}. The \gls{sas} populations had the highest mean \gls{td} overall at `r global_super_summary %>% filter(super_pop == 'SAS', stat == 'Tajima.D') %>% .[["mean"]] %>% three_dp()`  (SD `r global_super_summary %>% filter(super_pop == 'SAS', stat == 'Tajima.D') %>% .[["sd"]] %>% three_dp()`). The \gls{afr} populations were the only populations with the 1st percentile above zero, but also had the smallest overall maximum value of  `r global_super_summary %>% filter(super_pop == 'AFR', stat == 'Tajima.D') %>% .[["max"]] %>% three_dp()`. The \gls{eur} populations had the highest maximum \gls{td} value at  `r global_super_summary %>% filter(super_pop == 'EUR', stat == 'Tajima.D') %>% .[["max"]] %>% three_dp()`. The smallest \gls{td} value of `r gs_td_table %>% filter(min(min) == min) %>% .[['min']] %>% three_dp()` came from the \gls{khv} population of the \gls{eas} super population. The \gls{afr} population had the smallest range (`r td_super_pop_means %>% filter(super_pop == 'AFR') %>% .[['range']] %>% three_dp()`), whereas the \gls{eas} super population had the largest (`r td_super_pop_means %>% filter(super_pop == 'EAS') %>% .[['range']] %>% three_dp()`). The \gls{pol} population had the second largest (`r td_super_pop_means %>% filter(super_pop == 'POL') %>% .[['range']] %>% three_dp()`). A right shifted distribution of \gls{td} can be attributed to using ascertainment bias from a chip, where there is a bias against low frequency markers.

<!-- better check that the t.test is ok in this way 
The minimum \gls{td} value for a Polynesian population was `r gs_td_table %>% filter(super_pop == "POL") %>% filter(min(min) == min) %>% .[["min"]] %>% three_dp` and was from `r gs_td_table %>% filter(super_pop == "POL") %>% filter(min(min) == min) %>% .[["pop"]]`, whereas the maximum value was `r gs_td_table %>% filter(super_pop == "POL") %>% filter(max(max) == max) %>% .[["max"]] %>% three_dp()` and from `r gs_td_table %>% filter(super_pop == "POL") %>% filter(max(max) == max) %>% .[["pop"]]`. The Polynesian mean was significantly different to the other means for the super population using a t-test. The means were \gls{afr}: `r td_super_pop_means %>% filter(super_pop == "AFR") %>% .[["mean"]] %>% three_dp() ` (P = `r pol_td_t.test %>% filter(SuperPop == 'AFR') %>% .[['p.value']] %>% three_dp()`), \gls{amr}: `r td_super_pop_means %>% filter(super_pop == "AMR") %>% .[["mean"]] %>% three_dp() ` (P = `r pol_td_t.test %>% filter(SuperPop == 'AMR') %>% .[['p.value']] %>% three_dp()`), \gls{eur}: `r td_super_pop_means %>% filter(super_pop == "EUR") %>% .[["mean"]] %>% three_dp() ` (P = `r pol_td_t.test %>% filter(SuperPop == 'EUR') %>% .[['p.value']] %>% three_dp()`), \gls{eas}: `r td_super_pop_means %>% filter(super_pop == "EAS") %>% .[["mean"]] %>% three_dp() ` (P = `r pol_td_t.test %>% filter(SuperPop == 'EAS') %>% .[['p.value']] %>% three_dp()`), and \gls{sas}: `r td_super_pop_means %>% filter(super_pop == "SAS") %>% .[["mean"]] %>% three_dp() ` (P = `r pol_td_t.test %>% filter(SuperPop == 'SAS') %>% .[['p.value']] %>% three_dp()`). 
-->

The mean number of windows that met the significance threshold per population for the lower tail was  `r lower_sig_stats %>% filter(statname == 'Tajima.D') %>% group_by(pop) %>% tally() %>% summarise(mean(n)) %>% round(digits = 0)` (SD `r lower_sig_stats %>% filter(statname == 'Tajima.D') %>% group_by(pop) %>% tally() %>% summarise(sd(n)) %>% round(digits = 0)`) with a minimum of `r lower_sig_stats %>% filter(statname == 'Tajima.D') %>% group_by(pop) %>% tally() %>% summarise(min(n)) %>% round(digits = 0)` and a maximum of `r lower_sig_stats %>% filter(statname == 'Tajima.D') %>% group_by(pop) %>% tally() %>% summarise(max(n)) %>% round(digits = 0)`. The mean number of genes that were unique to a population was `r td_pop_unique_genes_lower %>% summarise(mean(n)) %>% round(digits = 0)` (SD `r td_pop_unique_genes_lower %>% summarise(sd(n)) %>% round(digits = 0)`) with a minimum of `r td_pop_unique_genes_lower %>% summarise(min(n)) %>% round(digits = 0)` and maximum of `r td_pop_unique_genes_lower %>% summarise(max(n)) %>% round(digits = 0)`. The \gls{afr} populations had less windows that were significant because their 1st percentiles were above zero. In the upper tail, the mean number of windows per population that met the significance threshold was `r upper_sig_stats %>% filter(statname == 'Tajima.D') %>% group_by(pop) %>% tally() %>% summarise(mean(n)) %>% round(digits = 0)` (SD `r upper_sig_stats %>% filter(statname == 'Tajima.D') %>% group_by(pop) %>% tally() %>% summarise(sd(n)) %>% round(digits = 0)`), with a minimum of `r upper_sig_stats %>% filter(statname == 'Tajima.D') %>% group_by(pop) %>% tally() %>% summarise(min(n)) %>% round(digits = 0)` and a maximum of `r upper_sig_stats %>% filter(statname == 'Tajima.D') %>% group_by(pop) %>% tally() %>% summarise(max(n)) %>% round(digits = 0)`. The mean number of genes that were unique to a population was `r td_pop_unique_genes_upper %>% summarise(mean(n)) %>% round(digits = 0)` (SD `r td_pop_unique_genes_upper %>% summarise(sd(n)) %>% round(digits = 0)`) and minimum of `r td_pop_unique_genes_upper %>% summarise(min(n)) %>% round(digits = 0)` and maximum of `r td_pop_unique_genes_upper %>% summarise(max(n)) %>% round(digits = 0)`.

<!-- - FDR for Tajimas D is in Figure \@ref(fig:TajimaDfdr) -->
<!-- - population summary stats in table \@ref(tab:gsTdTable) -->
<!-- - test crossref \@ref(tab:gsTDtest) -->
<!-- - list of regions unique to Pol populations Table \@ref(tab:lowersigtdPol) -->

(ref:TajimaDfdr) The False discovery rate for Tajima's D by population from 5000 permutations. The dashed line indicates the 1st percentile from the observed data.

```{r TajimaDfdr, fig.cap='(ref:TajamaDfdr)'}
include_graphics('images/03_selectionpipeline/Tajima.D_fdr_low.pdf')
```



```{r td_ranges, eval = FALSE, include = FALSE}
#CIM
lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM')) %>% select(contains('chrom'))  %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length() # regions 
lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6 # Mbp of region 
lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW() # genes 
lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Tajima.D', !pop %in% c('CIM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW() # genes only from CIM

paste(lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Tajima.D', !pop %in% c('CIM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% .[["SYMBOL"]], collapse = " ")

gr <- lower_sig_stats %>% filter(pop == "CIM", statname == 'Tajima.D') %>%  GenomicRanges::GRanges() %>% GenomicRanges::reduce()
gr[GenomicRanges::width(gr) >= 20 * 10e3]

#NZM
lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length() # regions 
lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6 # Mbp of region 
lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW() # genes 
lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('NZM')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Tajima.D', !pop %in% c('NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW() # genes only from NZM

paste(lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('NZM')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Tajima.D', !pop %in% c('NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% .[["SYMBOL"]], collapse = " ")

gr <- lower_sig_stats %>% filter(pop == "NZM", statname == 'Tajima.D') %>%  GenomicRanges::GRanges() %>% GenomicRanges::reduce()
gr[GenomicRanges::width(gr) >= 20 * 10e3]

#SAM
lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length() # regions 
lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6 # Mbp of region 
lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW() # genes 
lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Tajima.D', !pop %in% c('SAM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW() # genes only from SAM

paste(lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Tajima.D', !pop %in% c('SAM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% .[["SYMBOL"]], collapse = " ")

gr <- lower_sig_stats %>% filter(pop == "SAM", statname == 'Tajima.D') %>%  GenomicRanges::GRanges() %>% GenomicRanges::reduce()
gr[GenomicRanges::width(gr) >= 20 * 10e3]

# TON
lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length() # regions 
lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6 # Mbp of region 
lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('TON')) %>% select('SYMBOL') %>% distinct() %>% NROW() # genes 
lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('TON')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Tajima.D', !pop %in% c('TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW() # genes only from TON
paste(lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('TON')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Tajima.D', !pop %in% c('TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% .[["SYMBOL"]], collapse = " ")


gr <- lower_sig_stats %>% filter(pop == "TON", statname == 'Tajima.D') %>%  GenomicRanges::GRanges() %>% GenomicRanges::reduce()
gr[GenomicRanges::width(gr) >= 20 * 10e3]

#EP
lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM', 'NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length() # regions 
lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM', 'NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6 # Mbp of region 
lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM', 'NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW() # genes 
lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM', 'NZM')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Tajima.D', !pop %in% c('CIM','NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW() # genes only from EP

paste(lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM','NZM')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Tajima.D', !pop %in% c('CIM','NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% .[["SYMBOL"]], collapse = " ")

#WP
lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length() # regions 
lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6 # Mbp of region 
lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM','TON')) %>% select('SYMBOL') %>% distinct() %>% NROW() # genes 
lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM','TON')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Tajima.D', !pop %in% c('SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW() # genes only from WP

paste(lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM','TON')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Tajima.D', !pop %in% c('SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% .[["SYMBOL"]], collapse = " ")

#POL
lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions 
lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp of region 
lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select('SYMBOL') %>% distinct() %>% NROW() # genes 
lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM','NZM','SAM','TON')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Tajima.D', !pop %in% c('CIM','NZM','SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW() # genes only from POL

paste(lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM','NZM','SAM','TON')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Tajima.D', !pop %in% c('CIM','NZM','SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% .[["SYMBOL"]], collapse = " ")
```



#### Fay and Wu's H

```{r fwh_main, warning=FALSE, message=FALSE}
a <- list()
for(p in sort(unique(lower_sig_stats$pop))){
  a[[p]] <-  lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop == p) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fay.Wu.H', pop != p), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()
}
fwh_pop_unique_genes_lower <- bind_cols(a) %>% gather("pop","n")
rm(a)

a <- list()
for(p in sort(unique(upper_sig_stats$pop))){
  a[[p]] <-  upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop == p) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fay.Wu.H', pop != p), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()
}
fwh_pop_unique_genes_upper <- bind_cols(a) %>% gather("pop","n")
rm(a)

fwh_super_pop_means <- global_super_summary %>% filter(stat == 'Fay.Wu.H') %>% mutate(range = max - min)
gs_fwh_table <- create_sel_summary_table("Fay.Wu.H")
```

Comparing genome-wide \gls{fwh} revealed that the \gls{pol} super population had the lowest mean (`r fwh_super_pop_means %>% filter(super_pop == 'POL') %>% .[['mean']] %>% three_dp()`), 1st percentile (`r fwh_super_pop_means %>% filter(super_pop == 'POL') %>% .[['lower_1']] %>% three_dp()`), median (`r fwh_super_pop_means %>% filter(super_pop == 'POL') %>% .[['median']] %>% three_dp()`), 99th percentile (`r fwh_super_pop_means %>% filter(super_pop == 'POL') %>% .[['upper_99']] %>% three_dp()`), and maximum values (`r fwh_super_pop_means %>% filter(super_pop == 'POL') %>% .[['max']] %>% three_dp()`). 
Within the \gls{pol} group, the minimum \gls{fwh} value for a Polynesian population was from `r gs_fwh_table %>% filter(super_pop == "POL") %>% filter(min(min) == min) %>% .[["pop"]]`, whereas the maximum value was from `r gs_fwh_table %>% filter(super_pop == "POL") %>% filter(max(max) == max) %>% .[["pop"]]`. \gls{eas} was the second lowest for mean (`r fwh_super_pop_means %>% filter(super_pop == 'EAS') %>% .[['mean']] %>% three_dp()`), 1st percentile (`r fwh_super_pop_means %>% filter(super_pop == 'EAS') %>% .[['lower_1']] %>% three_dp()`), median (`r fwh_super_pop_means %>% filter(super_pop == 'EAS') %>% .[['median']] %>% three_dp()`), and 99th percentile (`r fwh_super_pop_means %>% filter(super_pop == 'EAS') %>% .[['upper_99']] %>% three_dp()`), but had the largest maximum (`r fwh_super_pop_means %>% filter(super_pop == 'EAS') %>% .[['max']] %>% three_dp()`). The \gls{eur} super population had both the smallest minimum and largest range (`r fwh_super_pop_means %>% filter(super_pop == 'EUR') %>% .[['range']] %>% three_dp()`). The \gls{afr} super population were the only group to have a positive mean (`r fwh_super_pop_means %>% filter(super_pop == 'AFR') %>% .[['mean']] %>% three_dp()`), they also had the smallest range and variation (SD `r fwh_super_pop_means %>% filter(super_pop == 'AFR') %>% .[['sd']] %>% three_dp()`).

Looking at the number of windows that met the significance threshold of being in the 1st percentile and < 0, the mean number of windows that met this threshold per population for the lower tail was  `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H') %>% group_by(pop) %>% tally() %>% summarise(mean(n)) %>% round(digits = 0)` (SD `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H') %>% group_by(pop) %>% tally() %>% summarise(sd(n)) %>% round(digits = 0)`) with a minimum of `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H') %>% group_by(pop) %>% tally() %>% summarise(min(n)) %>% round(digits = 0)` and a maximum of `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H') %>% group_by(pop) %>% tally() %>% summarise(max(n)) %>% round(digits = 0)`. Of the windows that met the significance threshold, the  mean number of genes intersecting the windows, that were unique to a population was `r fwh_pop_unique_genes_lower %>% summarise(mean(n)) %>% round(digits = 0)` (SD `r fwh_pop_unique_genes_lower %>% summarise(sd(n)) %>% round(digits = 0)`) and minimum of `r fwh_pop_unique_genes_lower %>% summarise(min(n)) %>% round(digits = 0)` and maximum of `r fwh_pop_unique_genes_lower %>% summarise(max(n)) %>% round(digits = 0)`. The mean number of windows per population that met the significance threshold of being in the 99th percentile and > 0 was `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H') %>% group_by(pop) %>% tally() %>% summarise(mean(n)) %>% round(digits = 0)` (SD `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H') %>% group_by(pop) %>% tally() %>% summarise(sd(n)) %>% round(digits = 0)`), with a minimum of `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H') %>% group_by(pop) %>% tally() %>% summarise(min(n)) %>% round(digits = 0)` and a maximum of `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H') %>% group_by(pop) %>% tally() %>% summarise(max(n)) %>% round(digits = 0)`.


<!-- - summary stats for each population Table \@ref(tab:gsFwhTable) -->
<!-- - Genes Unique to Polynesian populations lower tail Table \@ref(tab:lowersigfwhPol) -->

```{r, warning =  FALSE, message = FALSE}
pol_fwh_t.test <- bind_rows(
  lapply(c("AFR","AMR", "EAS","EUR","SAS"), function(sp){
    cbind(SuperPop = sp, 
          global_summary %>% filter(stat == 'Fay.Wu.H') %>% left_join(., panel %>% select(pop, super_pop) %>% distinct(), by = 'pop') %>% filter(super_pop %in% c("POL",sp)) %>% ungroup() %>% mutate(super_pop = as.character(super_pop)) %>% select(super_pop, mean) %>% do(broom::tidy(t.test( mean ~ super_pop, data = .))))
})) %>% mutate(adjusted_p = p.adjust(p.value, "bonferroni"))
```


<!--
The Eastern Polynesian population with the minimum \gls{fwh} value of `r gs_fwh_table %>% filter(pop %in% c("CIM","NZM")) %>% filter(min(as.numeric(min)) == as.numeric(min)) %>% .[["min"]] %>% three_dp()` was `r gs_fwh_table %>% filter(pop %in% c("CIM","NZM")) %>% filter(min(min) == min) %>% .[["pop"]]`, however, values for \gls{cim} were overall lower. The Western Polynesian population with the minimum was `r gs_fwh_table %>% filter(pop %in% c("SAM","TON")) %>% filter(min(min) == min) %>% .[["pop"]]`, with a value of `r gs_fwh_table %>% filter(pop %in% c("SAM","TON")) %>% filter(min(min) == min) %>% .[["min"]] %>% three_dp()` and also had the maximum Polynesian \gls{fwh} value of `r gs_fwh_table %>% filter(super_pop == "POL") %>% filter(max(max) == max) %>% .[["max"]] %>% three_dp()`.  The maximum \gls{fwh} value from the Eastern Polynesian populations was `r gs_fwh_table %>% filter(pop %in% c("CIM","NZM")) %>% filter(max(max) == max) %>% .[["max"]] %>% three_dp()` with `r gs_fwh_table %>% filter(pop %in% c("CIM","NZM")) %>% filter(max(max) == max) %>% .[["pop"]]`. 

The Polynesian mean was significantly different to the other means for the super population using a t-test. The means were \gls{afr}: `r fwh_super_pop_means %>% filter(super_pop == "AFR") %>% .[["mean"]] %>% three_dp() ` (P = \num{`r pol_fwh_t.test %>% filter(SuperPop == 'AFR') %>% select(p.value) %>% format(., scientific = TRUE, digits = 3)`}), \gls{amr}: `r fwh_super_pop_means %>% filter(super_pop == "AMR") %>% .[["mean"]] %>% three_dp() ` (P = `r pol_fwh_t.test %>% filter(SuperPop == 'AMR') %>% .[['p.value']] %>% three_dp()`), \gls{eur}: `r fwh_super_pop_means %>% filter(super_pop == "EUR") %>% .[["mean"]] %>% three_dp() ` (P = `r pol_fwh_t.test %>% filter(SuperPop == 'EUR') %>% .[['p.value']] %>% three_dp()`), \gls{eas}: `r fwh_super_pop_means %>% filter(super_pop == "EAS") %>% .[["mean"]] %>% three_dp() ` (P = \num{`r pol_fwh_t.test %>% filter(SuperPop == 'EAS') %>% select(p.value) %>% format(., scientific = TRUE, digits = 3)`}), and \gls{sas}: `r fwh_super_pop_means %>% filter(super_pop == "SAS") %>% .[["mean"]] %>% three_dp() ` (P = \num{`r pol_fwh_t.test %>% filter(SuperPop == 'SAS') %>% select(p.value) %>% format(., scientific = TRUE, digits = 3)`}). -->


```{r, fwh_ranges, eval = FALSE, include = FALSE}
#CIM
lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length() # regions lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6 # Mbp of region 
lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW() # genes 
lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Fay.Wu.H', !pop %in% c('CIM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW() # genes only from CIM

gr <- lower_sig_stats %>% filter(pop == "CIM", statname == 'Fay.Wu.H') %>%  GenomicRanges::GRanges() %>% GenomicRanges::reduce()
gr[GenomicRanges::width(gr) >= 20 * 10e3]

#NZM
lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length() # regions lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6 # Mbp of region 
lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW() # genes 
lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('NZM')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Fay.Wu.H', !pop %in% c('NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW() # genes only from NZM

gr <- lower_sig_stats %>% filter(pop == "NZM", statname == 'Fay.Wu.H') %>%  GenomicRanges::GRanges() %>% GenomicRanges::reduce()
gr[GenomicRanges::width(gr) >= 20 * 10e3]

#SAM
lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length() # regions lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6 # Mbp of region 
lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW() # genes 
lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('SAM')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Fay.Wu.H', !pop %in% c('SAM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW() # genes only from SAM

gr <- lower_sig_stats %>% filter(pop == "SAM", statname == 'Fay.Wu.H') %>%  GenomicRanges::GRanges() %>% GenomicRanges::reduce()
gr[GenomicRanges::width(gr) >= 20 * 10e3]

#TON
lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length() # regions lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6 # Mbp of region 
lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('TON')) %>% select('SYMBOL') %>% distinct() %>% NROW() # genes 
lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('TON')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Fay.Wu.H', !pop %in% c('TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW() # genes only from TON

gr <- lower_sig_stats %>% filter(pop == "TON", statname == 'Fay.Wu.H') %>%  GenomicRanges::GRanges() %>% GenomicRanges::reduce()
gr[GenomicRanges::width(gr) >= 20 * 10e3]

#EP
lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM', 'NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length() # regions 
lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM', 'NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6 # Mbp of region 
lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM', 'NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW() # genes 
lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM', 'NZM')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Fay.Wu.H', !pop %in% c('CIM','NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW() # genes only from EP

#WP
lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length() # regions 
lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6 # Mbp of region 
lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('SAM','TON')) %>% select('SYMBOL') %>% distinct() %>% NROW() # genes 
lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('SAM','TON')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Fay.Wu.H', !pop %in% c('SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW() # genes only from WP

#POL

lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()# regions 
lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6 # Mbp of region 
lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM','NZM','SAM','TON')) %>% select('SYMBOL') %>% distinct() %>% NROW() # genes 
lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM','NZM','SAM','TON')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Fay.Wu.H', !pop %in% c('CIM','NZM','SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW() # genes only from POL
```



#### Fu and Li's D

```{r fld_main, warning=FALSE, message=FALSE}
a <- list()
for(p in sort(unique(lower_sig_stats$pop))){
  a[[p]] <-  lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop == p) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.D', pop != p), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()
}
fld_pop_unique_genes_lower <- bind_cols(a) %>% gather("pop","n")
rm(a)

a <- list()
for(p in sort(unique(upper_sig_stats$pop))){
  a[[p]] <-  upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop == p) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.D', pop != p), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()
}
fld_pop_unique_genes_upper <- bind_cols(a) %>% gather("pop","n")
rm(a)

gs_fld_table <- create_sel_summary_table("Fu.Li.D")
fld_super_pop_means <- global_super_summary %>% filter(stat == 'Fu.Li.D') %>% mutate(range = max - min)


pol_fld_t.test <- bind_rows(
  lapply(c("AFR","AMR", "EAS","EUR","SAS"), function(sp){
    cbind(SuperPop = sp, 
          global_summary %>% filter(stat == 'Fu.Li.D') %>% left_join(., panel %>% select(pop, super_pop) %>% distinct(), by = 'pop') %>% filter(super_pop %in% c("POL",sp)) %>% ungroup() %>% mutate(super_pop = as.character(super_pop)) %>% select(super_pop, mean) %>% do(broom::tidy(t.test( mean ~ super_pop, data = .))))
})) %>% mutate(adjusted_p = p.adjust(p.value, "bonferroni"))
```

Similar to \gls{td}, for \gls{fld} the \gls{pol} super population was overall lower than the other super populations and had the lowest mean (`r fld_super_pop_means %>% filter(super_pop == 'POL') %>% .[['mean']] %>% three_dp()`), 1st percentile (`r fld_super_pop_means %>% filter(super_pop == 'POL') %>% .[['lower_1']] %>% three_dp()`), median (`r fld_super_pop_means %>% filter(super_pop == 'POL') %>% .[['median']] %>% three_dp()`), and 99th percentile (`r fld_super_pop_means %>% filter(super_pop == 'POL') %>% .[['upper_99']] %>% three_dp()`). \gls{eas} too had the smallest min (`r fld_super_pop_means %>% filter(super_pop == 'EAS') %>% .[['min']] %>% three_dp()`), and \gls{eur} had the largest max (`r fld_super_pop_means %>% filter(super_pop == 'EUR') %>% .[['max']] %>% three_dp()`). \gls{afr} had the largest min (`r fld_super_pop_means %>% filter(super_pop == 'AFR') %>% .[['min']] %>% three_dp()`) and also had the highest mean (`r fld_super_pop_means %>% filter(super_pop == 'AFR') %>% .[['mean']] %>% three_dp()`).

The mean number of windows that met the significance threshold per population for the lower tail was  `r lower_sig_stats %>% filter(statname == 'Fu.Li.D') %>% group_by(pop) %>% tally() %>% summarise(mean(n)) %>% round(digits = 0)` (SD `r lower_sig_stats %>% filter(statname == 'Fu.Li.D') %>% group_by(pop) %>% tally() %>% summarise(sd(n)) %>% round(digits = 0)`) with a minimum of `r lower_sig_stats %>% filter(statname == 'Fu.Li.D') %>% group_by(pop) %>% tally() %>% summarise(min(n)) %>% round(digits = 0)` and a maximum of `r lower_sig_stats %>% filter(statname == 'Fu.Li.D') %>% group_by(pop) %>% tally() %>% summarise(max(n)) %>% round(digits = 0)`. The mean number of genes that were unique to a population was `r fld_pop_unique_genes_lower %>% summarise(mean(n)) %>% round(digits = 0)` (SD `r fld_pop_unique_genes_lower %>% summarise(sd(n)) %>% round(digits = 0)`) and minimum of `r fld_pop_unique_genes_lower %>% summarise(min(n)) %>% round(digits = 0)` and maximum of `r fld_pop_unique_genes_lower %>% summarise(max(n)) %>% round(digits = 0)`. The conservative estimate of the \gls{fdr} ranged from `r fdr_low %>% filter(stat == 'Fu.Li.D', lower_per == lowest_quant, super_pop == 'POL') %>% filter(max(FDR) == FDR) %>% mutate_at(vars(FDR, lowerCI, upperCI), funs(three_dp)) %>% mutate(fdr = paste0(FDR, ' (95% CI ',lowerCI, ', ',upperCI, ')')) %>% .[['fdr']]` to `r fdr_low %>% filter(stat == 'Fu.Li.D', lower_per == lowest_quant, super_pop == 'POL') %>% filter(min(FDR) == FDR) %>% mutate_at(vars(FDR, lowerCI, upperCI), funs(three_dp)) %>% mutate(fdr = paste0(FDR, ' (95% CI ',lowerCI, ', ',upperCI, ')')) %>% .[['fdr']]` for the \gls{pol} populations and from `r fdr_low %>% filter(stat == 'Fu.Li.D', lower_per == lowest_quant) %>% filter(min(FDR) == FDR) %>% mutate_at(vars(FDR, lowerCI, upperCI), funs(three_dp)) %>% mutate(fdr = paste0(FDR, ' (95% CI ',lowerCI, ', ',upperCI, ')')) %>% .[['fdr']]` to `r fdr_low %>% filter(stat == 'Fu.Li.D', lower_per == lowest_quant, super_pop == 'POL') %>% filter(max(FDR) == FDR) %>% mutate_at(vars(FDR, lowerCI, upperCI), funs(three_dp)) %>% mutate(fdr = paste0(FDR, ' (95% CI ',lowerCI, ', ',upperCI, ')')) %>% .[['fdr']]` from all populations. 

For the 99th percentile and > 0 threshold, the mean number of windows per population that met this threshold was `r upper_sig_stats %>% filter(statname == 'Fu.Li.D') %>% group_by(pop) %>% tally() %>% summarise(mean(n)) %>% round(digits = 0)` (SD `r upper_sig_stats %>% filter(statname == 'Fu.Li.D') %>% group_by(pop) %>% tally() %>% summarise(sd(n)) %>% round(digits = 0)`), with a minimum of `r upper_sig_stats %>% filter(statname == 'Fu.Li.D') %>% group_by(pop) %>% tally() %>% summarise(min(n)) %>% round(digits = 0)` and a maximum of `r upper_sig_stats %>% filter(statname == 'Fu.Li.D') %>% group_by(pop) %>% tally() %>% summarise(max(n)) %>% round(digits = 0)`. The mean number of genes that were unique to a population was `r fld_pop_unique_genes_upper %>% summarise(mean(n)) %>% round(digits = 0)` (SD `r fld_pop_unique_genes_upper %>% summarise(sd(n)) %>% round(digits = 0)`) and minimum of `r fld_pop_unique_genes_upper %>% summarise(min(n)) %>% round(digits = 0)` and maximum of `r fld_pop_unique_genes_upper %>% summarise(max(n)) %>% round(digits = 0)`. The \gls{fdr} estimated for the \gls{pol} populations ranged from `r fdr_high %>% filter(stat == 'Fu.Li.D', upper_per == highest_quant, super_pop == 'POL') %>% filter(min(FDR) == FDR) %>% mutate_at(vars(FDR, lowerCI, upperCI), funs(three_dp)) %>% mutate(fdr = paste0(FDR, ' (95% CI ',lowerCI, ', ',upperCI, ')')) %>% .[['fdr']]` to `r fdr_high %>% filter(stat == 'Fu.Li.D', upper_per == highest_quant, super_pop == 'POL') %>% filter(max(FDR) == FDR) %>% mutate_at(vars(FDR, lowerCI, upperCI), funs(three_dp)) %>% mutate(fdr = paste0(FDR, ' (95% CI ',lowerCI, ', ',upperCI, ')')) %>% .[['fdr']]`, and from `r fdr_high %>% filter(stat == 'Fu.Li.D', upper_per == highest_quant) %>% filter(min(FDR) == FDR) %>% mutate_at(vars(FDR, lowerCI, upperCI), funs(three_dp)) %>% mutate(fdr = paste0(FDR, ' (95% CI ',lowerCI, ', ',upperCI, ')')) %>% .[['fdr']]` to `r fdr_high %>% filter(stat == 'Fu.Li.D', upper_per == highest_quant) %>% filter(max(FDR) == FDR) %>% arrange(desc(upperCI,lowerCI)) %>% slice(1)%>% mutate_at(vars(FDR, lowerCI, upperCI), funs(three_dp)) %>% mutate(fdr = paste0(FDR, ' (95% CI ',lowerCI, ', ',upperCI, ')')) %>% .[['fdr']]` for all populations. The higher \gls{fdr} is likely due there being a bias against singletons, of which form the basis for the \gls{fld} calculation.


<!--
The mean \gls{fld} for the Polynesian populations was `r global_super_summary %>% filter(super_pop == 'POL', stat == 'Fu.Li.D') %>% .[["mean"]] %>% three_dp()`  (SD `r global_super_summary %>% filter(super_pop == 'POL', stat == 'Tajima.D') %>% .[["sd"]] %>% three_dp()`) with  `r gs_fld_table %>% filter(pop == 'CIM') %>% .[["mean"]] %>% three_dp()` (SD `r gs_fld_table %>% filter(pop == 'CIM') %>% .[["sd"]] %>% three_dp()`) for \gls{cim}, `r gs_fld_table %>% filter(pop == 'NZM') %>% .[["mean"]] %>% three_dp()` (SD `r gs_fld_table %>% filter(pop == 'NZM') %>% .[["sd"]] %>% three_dp()`) for \gls{nzm}, `r gs_fld_table %>% filter(pop == 'SAM') %>% .[["mean"]] %>% three_dp()` (SD `r gs_fld_table %>% filter(pop == 'SAM') %>% .[["sd"]] %>% three_dp()`) for \gls{sam}, and `r gs_fld_table %>% filter(pop == 'TON') %>% .[["mean"]] %>% three_dp()` (SD `r gs_fld_table %>% filter(pop == 'TON') %>% .[["sd"]] %>% three_dp()`) for \gls{ton}


The minimum \gls{fld} value for a Polynesian population was `r gs_fld_table %>% filter(super_pop == "POL") %>% filter(min(min) == min) %>% .[["min"]] %>% three_dp()` and was from `r gs_fld_table %>% filter(super_pop == "POL") %>% filter(min(min) == min) %>% .[["pop"]]`, whereas the maximum value was `r gs_fld_table %>% filter(super_pop == "POL") %>% filter(max(max) == max) %>% .[["max"]] %>% three_dp()` and from `r gs_fld_table %>% filter(super_pop == "POL") %>% filter(max(max) == max) %>% .[["pop"]]`.

 The Polynesian mean was significantly different to the other means for the super population using a t-test. The means were \gls{afr}: `r fld_super_pop_means %>% filter(super_pop == "AFR") %>% .[["mean"]] %>% three_dp() ` (P = `r pol_fld_t.test %>% filter(SuperPop == 'AFR') %>% .[['p.value']] %>% three_dp()`), \gls{amr}: `r fld_super_pop_means %>% filter(super_pop == "AMR") %>% .[["mean"]] %>% three_dp() ` (P = `r pol_fld_t.test %>% filter(SuperPop == 'AMR') %>% .[['p.value']] %>% three_dp()`), \gls{eur}: `r fld_super_pop_means %>% filter(super_pop == "EUR") %>% .[["mean"]] %>% three_dp() ` (P = `r pol_fld_t.test %>% filter(SuperPop == 'EUR') %>% .[['p.value']] %>% three_dp()`), \gls{eas}: `r fld_super_pop_means %>% filter(super_pop == "EAS") %>% .[["mean"]] %>% three_dp() ` (P = `r pol_fld_t.test %>% filter(SuperPop == 'EAS') %>% .[['p.value']] %>% three_dp()`), and \gls{sas}: `r fld_super_pop_means %>% filter(super_pop == "SAS") %>% .[["mean"]] %>% three_dp() ` (P = `r pol_fld_t.test %>% filter(SuperPop == 'SAS') %>% .[['p.value']] %>% three_dp()`). 
-->

<!-- - summary for each population Table \@ref(tab:gsFldTable)). -->
<!-- - Genes Unique to Polynesian populations Table \@ref(tab:lowersigfldPol) -->

```{r fld_ranges, eval = FALSE, include = FALSE}
#CIM

lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length() # regions 
lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6 # Mbp of region 
lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW() # genes 
lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Fu.Li.D', !pop %in% c('CIM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()# genes only from CIM

gr <- lower_sig_stats %>% filter(pop == "CIM", statname == 'Fu.Li.D') %>%  GenomicRanges::GRanges() %>% GenomicRanges::reduce()
gr[GenomicRanges::width(gr) >= 20 * 10e3]


#NZM
lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length() # regions 
lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6 # Mbp of region 
lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW() # genes 
lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('NZM')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Fu.Li.D', !pop %in% c('NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW() # genes only from NZM

gr <- lower_sig_stats %>% filter(pop == "NZM", statname == 'Fu.Li.D') %>%  GenomicRanges::GRanges() %>% GenomicRanges::reduce()
gr[GenomicRanges::width(gr) >= 20 * 10e3]



#SAM
lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length() # regions 
lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6 # Mbp of region 
lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW() # genes 
lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('SAM')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Fu.Li.D', !pop %in% c('SAM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW() # genes only from SAM

gr <- lower_sig_stats %>% filter(pop == "SAM", statname == 'Fu.Li.D') %>%  GenomicRanges::GRanges() %>% GenomicRanges::reduce()
gr[GenomicRanges::width(gr) >= 20 * 10e3]

#TON
lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length() # regions 
lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6 # Mbp of region 
lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('TON')) %>% select('SYMBOL') %>% distinct() %>% NROW() # genes 
lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('TON')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Fu.Li.D', !pop %in% c('TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()# genes only from TON

gr <- lower_sig_stats %>% filter(pop == "TON", statname == 'Fu.Li.D') %>%  GenomicRanges::GRanges() %>% GenomicRanges::reduce()
gr[GenomicRanges::width(gr) >= 20 * 10e3]


#EP
lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM', 'NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length() # regions
lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM', 'NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6 # Mbp of region 
lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM', 'NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW() # genes 
lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM', 'NZM')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Fu.Li.D', !pop %in% c('CIM','NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW() # genes only from EP

#WP

lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length() # regions 
lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6 # Mbp of region 
lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('SAM','TON')) %>% select('SYMBOL') %>% distinct() %>% NROW() # genes 
lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('SAM','TON')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Fu.Li.D', !pop %in% c('SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW() # genes only from WP

#POL
lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length() # regions 
lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6 # Mbp of region 
lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select('SYMBOL') %>% distinct() %>% NROW() # genes 
lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM','NZM','SAM','TON')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Fu.Li.D', !pop %in% c('CIM','NZM','SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW() # genes only from POL
```



#### Fu and Li's F

```{r flf_main, warning=FALSE, message=FALSE}
a <- list()
for(p in sort(unique(lower_sig_stats$pop))){
  a[[p]] <-  lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop == p) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.F', pop != p), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()
}
flf_pop_unique_genes_lower <- bind_cols(a) %>% gather("pop","n")
rm(a)

a <- list()
for(p in sort(unique(upper_sig_stats$pop))){
  a[[p]] <-  upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop == p) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.F', pop != p), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()
}
flf_pop_unique_genes_upper <- bind_cols(a) %>% gather("pop","n")
rm(a)


gs_flf_table <- create_sel_summary_table("Fu.Li.F") 
flf_super_pop_means <- global_super_summary %>% filter(stat == 'Fu.Li.F') %>% mutate(range = max - min)
```

The distributions of \gls{flf} were all centred above zero, with the lowest mean being that of \gls{pol} at `r flf_super_pop_means %>% filter(super_pop == 'POL') %>% .[['mean']] %>% three_dp()`. The \gls{pol} populations also had the lowest 1st percentile (`r flf_super_pop_means %>% filter(super_pop == 'POL') %>% .[['lower_1']] %>% three_dp()`), median (`r flf_super_pop_means %>% filter(super_pop == 'POL') %>% .[['median']] %>% three_dp()`), and 99th percentile (`r flf_super_pop_means %>% filter(super_pop == 'POL') %>% .[['upper_99']] %>% three_dp()`). Conversely, the \gls{sas} populations had the highest mean (`r flf_super_pop_means %>% filter(super_pop == 'SAS') %>% .[['mean']] %>% three_dp()`), median (`r flf_super_pop_means %>% filter(super_pop == 'SAS') %>% .[['median']] %>% three_dp()`), and were equal with the \gls{afr} populations with the highest minimum (`r flf_super_pop_means %>% filter(super_pop == 'SAS') %>% .[['min']] %>% three_dp()`). The \gls{afr} populations were the only group to have a positive 1st percentile (`r flf_super_pop_means %>% filter(super_pop == 'AFR') %>% .[['lower_1']] %>% three_dp()`). The \gls{eas} populations had the lowest minimum (`r flf_super_pop_means %>% filter(super_pop == 'EAS') %>% .[['min']] %>% three_dp()`), but the highest 99th percentile (`r flf_super_pop_means %>% filter(super_pop == 'EAS') %>% .[['upper_99']] %>% three_dp()`) and the largest range (`r flf_super_pop_means %>% filter(super_pop == 'EAS') %>% .[['range']] %>% three_dp()`).

The mean number of windows per population in the 1st percentile and with a value < 0 was  `r lower_sig_stats %>% filter(statname == 'Fu.Li.F') %>% group_by(pop) %>% tally() %>% summarise(mean(n)) %>% round(digits = 0)` (SD `r lower_sig_stats %>% filter(statname == 'Fu.Li.F') %>% group_by(pop) %>% tally() %>% summarise(sd(n)) %>% round(digits = 0)`) with a minimum of `r lower_sig_stats %>% filter(statname == 'Fu.Li.F') %>% group_by(pop) %>% tally() %>% summarise(min(n)) %>% round(digits = 0)` and a maximum of `r lower_sig_stats %>% filter(statname == 'Fu.Li.F') %>% group_by(pop) %>% tally() %>% summarise(max(n)) %>% round(digits = 0)`. The mean number of genes that were unique to a population was `r flf_pop_unique_genes_lower %>% summarise(mean(n)) %>% round(digits = 0)` (SD `r flf_pop_unique_genes_lower %>% summarise(sd(n)) %>% round(digits = 0)`), with a minimum of `r flf_pop_unique_genes_lower %>% summarise(min(n)) %>% round(digits = 0)` and maximum of `r flf_pop_unique_genes_lower %>% summarise(max(n)) %>% round(digits = 0)`. The mean number of windows per population in the 99th percentile and with a value > 0 was `r upper_sig_stats %>% filter(statname == 'Fu.Li.F') %>% group_by(pop) %>% tally() %>% summarise(mean(n)) %>% round(digits = 0)` (SD `r upper_sig_stats %>% filter(statname == 'Fu.Li.F') %>% group_by(pop) %>% tally() %>% summarise(sd(n)) %>% round(digits = 0)`), with a minimum of `r upper_sig_stats %>% filter(statname == 'Fu.Li.F') %>% group_by(pop) %>% tally() %>% summarise(min(n)) %>% round(digits = 0)` and a maximum of `r upper_sig_stats %>% filter(statname == 'Fu.Li.F') %>% group_by(pop) %>% tally() %>% summarise(max(n)) %>% round(digits = 0)`. The mean number of genes that were unique to a population was `r flf_pop_unique_genes_upper %>% summarise(mean(n)) %>% round(digits = 0)` (SD `r flf_pop_unique_genes_upper %>% summarise(sd(n)) %>% round(digits = 0)`), with a minimum of `r flf_pop_unique_genes_upper %>% summarise(min(n)) %>% round(digits = 0)` and maximum of `r flf_pop_unique_genes_upper %>% summarise(max(n)) %>% round(digits = 0)`. The \gls{flf} regions in the 1st percentile had `r round_dp(inner_join(lower_sig_stats %>% filter(statname == 'Fu.Li.D') %>% select(-statname), lower_sig_stats %>% filter(statname == 'Fu.Li.F') %>% select(-statname), by = c("chrom", "chrom_start", "chrom_end", "SYMBOL", "pop")) %>% NROW()/lower_sig_stats %>% filter(statname == 'Fu.Li.F') %>% NROW() *100, 1)`% also found in the 1st percentile of \gls{fld}, and likewise, there were `r round_dp(inner_join(upper_sig_stats %>% filter(statname == 'Fu.Li.D') %>% select(-statname), upper_sig_stats %>% filter(statname == 'Fu.Li.F') %>% select(-statname), by = c("chrom", "chrom_start", "chrom_end", "SYMBOL", "pop")) %>% NROW()/upper_sig_stats %>% filter(statname == 'Fu.Li.F') %>% NROW() *100, 1)`% of regions from the 99th percentile in common.

> perhaps the reason for the difference between fld and flf is that flf is normalised to an extent and that smooths out the lower tail?

<!-- - summary for each population Table \@ref(tab:gsFlfTable)). -->
<!-- - Genes Unique to Polynesian populations Table \@ref(tab:lowersigflfPol) -->

<!--
The mean \gls{flf} for the Polynesian populations was `r global_super_summary %>% filter(super_pop == 'POL', stat == 'Fu.Li.F') %>% .[["mean"]] %>% three_dp()`  (SD `r global_super_summary %>% filter(super_pop == 'POL', stat == 'Tajima.D') %>% .[["sd"]] %>% three_dp()`) with  `r gs_flf_table %>% filter(pop == 'CIM') %>% .[["mean"]] %>% three_dp()` (SD `r gs_flf_table %>% filter(pop == 'CIM') %>% .[["sd"]] %>% three_dp()`) for \gls{cim}, `r gs_flf_table %>% filter(pop == 'NZM') %>% .[["mean"]] %>% three_dp()` (SD `r gs_flf_table %>% filter(pop == 'NZM') %>% .[["sd"]] %>% three_dp()`) for \gls{nzm}, `r gs_flf_table %>% filter(pop == 'SAM') %>% .[["mean"]] %>% three_dp()` (SD `r gs_flf_table %>% filter(pop == 'SAM') %>% .[["sd"]] %>% three_dp()`) for \gls{sam}, and `r gs_flf_table %>% filter(pop == 'TON') %>% .[["mean"]] %>% three_dp()` (SD `r gs_flf_table %>% filter(pop == 'TON') %>% .[["sd"]] %>% three_dp()`) for \gls{ton} (Table \@ref(tab:gsFlfTable)).
-->


```{r flf_ranges, eval = FALSE, include = FALSE}
#CIM

lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length() # regions 
lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6 # Mbp of region 
lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW() # genes 
lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Fu.Li.F', !pop %in% c('CIM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW() # genes only from CIM
gr <- lower_sig_stats %>% filter(pop == "CIM", statname == 'Fu.Li.F') %>%  GenomicRanges::GRanges() %>% GenomicRanges::reduce()
gr[GenomicRanges::width(gr) >= 20 * 10e3]


#NZM
lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length() # regions 
lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6 # Mbp of region 
lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW() # genes 
lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('NZM')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Fu.Li.F', !pop %in% c('NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW() # genes only from NZM
gr <- lower_sig_stats %>% filter(pop == "NZM", statname == 'Fu.Li.F') %>%  GenomicRanges::GRanges() %>% GenomicRanges::reduce()
gr[GenomicRanges::width(gr) >= 20 * 10e3]



#SAM
lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length() # regions 
lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6 # Mbp of region 
lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW() # genes 
lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('SAM')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Fu.Li.F', !pop %in% c('SAM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW() # genes only from SAM
gr <- lower_sig_stats %>% filter(pop == "SAM", statname == 'Fu.Li.F') %>%  GenomicRanges::GRanges() %>% GenomicRanges::reduce()
gr[GenomicRanges::width(gr) >= 20 * 10e3]

#TON
lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length() # regions 
lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6 # Mbp of region 
lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('TON')) %>% select('SYMBOL') %>% distinct() %>% NROW() # genes 
lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('TON')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Fu.Li.F', !pop %in% c('TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW() # genes only from TON
gr <- lower_sig_stats %>% filter(pop == "TON", statname == 'Fu.Li.F') %>%  GenomicRanges::GRanges() %>% GenomicRanges::reduce()
gr[GenomicRanges::width(gr) >= 20 * 10e3]


#EP
lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM', 'NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length() # regions 
lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM', 'NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6 # Mbp of region 
lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM', 'NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW() # genes 
lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM', 'NZM')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Fu.Li.F', !pop %in% c('CIM','NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW() # genes only from EP


#WP

lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length() # regions 
lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6 # Mbp of region 
lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('SAM','TON')) %>% select('SYMBOL') %>% distinct() %>% NROW() # genes 
lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('SAM','TON')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Fu.Li.F', !pop %in% c('SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW() # genes only from WP

#POL
lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length() # regions 
lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6 # Mbp of region 
lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM','NZM','SAM','TON')) %>% select('SYMBOL') %>% distinct() %>% NROW() # genes 
lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM','NZM','SAM','TON')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Fu.Li.F', !pop %in% c('CIM','NZM','SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW() # genes only from POL
```



#### Zeng's E

```{r, results='asis', warning=FALSE}
a <- list()
for(p in sort(unique(lower_sig_stats$pop))){
  a[[p]] <-  lower_sig_stats %>% filter(statname == 'Zeng.E', pop == p) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Zeng.E', pop != p), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()
}
ze_pop_unique_genes_lower <- bind_cols(a) %>% gather("pop","n")
rm(a)

a <- list()
for(p in sort(unique(upper_sig_stats$pop))){
  a[[p]] <-  upper_sig_stats %>% filter(statname == 'Zeng.E', pop == p) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Zeng.E', pop != p), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()
}
ze_pop_unique_genes_upper <- bind_cols(a) %>% gather("pop","n")
rm(a)

gs_ze_table <- create_sel_summary_table("Zeng.E")
ze_super_pop_means <- global_super_summary %>% filter(stat == 'Zeng.E') %>% mutate(range = max - min)
```

```{r, include = FALSE, eval = FALSE}
fdr_low %>% filter(pop %in% poly_pop, stat == 'Zeng.E') %>% group_by(stat,pop) %>% filter(min(FDR) == FDR) %>% ungroup() %>% left_join(., perm_median %>% select(stat, pop, contains('lower')) %>% gather(bin, bin_value, 3:NCOL(.)) %>% mutate(bin = sapply(bin, function(x){as.numeric(strsplit(x, '_')[[1]][2])})), by = c('pop', 'stat')) %>% filter(lower_per == bin) %>% left_join(., lower_sig_stats_binned %>% select('stat' = statname, everything()), by = c('pop','stat')) %>% filter(bin_value > value)
```

Similar to \gls{td}, all of the distributions for \gls{ze} appear to have been shifted to the right. The \gls{afr} populations had the lowest overall distribution, with the lowest mean (`r ze_super_pop_means %>% filter(super_pop == 'AFR') %>% .[['mean']] %>% three_dp()`), 1st percentile (`r ze_super_pop_means %>% filter(super_pop == 'AFR') %>% .[['lower_1']] %>% three_dp()`), median (`r ze_super_pop_means %>% filter(super_pop == 'AFR') %>% .[['median']] %>% three_dp()`), 99th percentile (`r ze_super_pop_means %>% filter(super_pop == 'AFR') %>% .[['upper_99']] %>% three_dp()`), maximum (`r ze_super_pop_means %>% filter(super_pop == 'AFR') %>% .[['max']] %>% three_dp()`), and range (`r ze_super_pop_means %>% filter(super_pop == 'AFR') %>% .[['range']] %>% three_dp()`). Although it was the \gls{eur} populations that had the lowest minimum (`r ze_super_pop_means %>% filter(super_pop == 'EUR') %>% .[['min']] %>% three_dp()`) and the highest maximum (`r ze_super_pop_means %>% filter(super_pop == 'EUR') %>% .[['max']] %>% three_dp()`), which also meant the largest range (`r ze_super_pop_means %>% filter(super_pop == 'EUR') %>% .[['range']] %>% three_dp()`). The highest mean of `r ze_super_pop_means %>% filter(super_pop == 'EAS') %>% .[['mean']] %>% three_dp()` was from \gls{eas}. The \gls{pol} populations had the second highest mean (`r ze_super_pop_means %>% filter(super_pop == 'POL') %>% .[['mean']] %>% three_dp()`), median (`r ze_super_pop_means %>% filter(super_pop == 'POL') %>% .[['median']] %>% three_dp()`), 99th percentile (`r ze_super_pop_means %>% filter(super_pop == 'POL') %>% .[['upper_99']] %>% three_dp()`), and maximum (`r ze_super_pop_means %>% filter(super_pop == 'POL') %>% .[['max']] %>% three_dp()`). 


Zeng's E for the Polynesian populations had an estimated \gls{fdr} of 100% at the 1st percentile in all four populations, which was the equivalent of the `r fdr_low_tab %>% filter(stat == 'Zeng.E', pop %in% poly_pop) %>% summarise(round_dp(max(lowest_quant), 0))`nd percentile from the permutations. In order to have the minimum \gls{fdr} for the \gls{pol} populations from the gradations generated, this would be the 0.01 percentile, which was 6 fold smaller than  the first non-Polynesian population. The mean number of windows that met the significance threshold per population for the lower tail was  `r lower_sig_stats %>% filter(statname == 'Zeng.E') %>% group_by(pop) %>% tally() %>% summarise(mean(n)) %>% round(digits = 0)` (SD `r lower_sig_stats %>% filter(statname == 'Zeng.E') %>% group_by(pop) %>% tally() %>% summarise(sd(n)) %>% round(digits = 0)`) with a minimum of `r lower_sig_stats %>% filter(statname == 'Zeng.E') %>% group_by(pop) %>% tally() %>% summarise(min(n)) %>% round(digits = 0)` and a maximum of `r lower_sig_stats %>% filter(statname == 'Zeng.E') %>% group_by(pop) %>% tally() %>% summarise(max(n)) %>% round(digits = 0)`. The mean number of genes that were unique to a population was `r ze_pop_unique_genes_lower %>% summarise(mean(n)) %>% round(digits = 0)` (SD `r ze_pop_unique_genes_lower %>% summarise(sd(n)) %>% round(digits = 0)`) and minimum of `r ze_pop_unique_genes_lower %>% summarise(min(n)) %>% round(digits = 0)` and maximum of `r ze_pop_unique_genes_lower %>% summarise(max(n)) %>% round(digits = 0)`.

The 99th percentile for the four Polynesian populations had an estimated \gls{fdr} maximum of `r fdr_high_tab %>% filter(stat == 'Zeng.E', pop %in% poly_pop) %>% filter(max(FDR) == FDR) %>% mutate(a = paste0(three_dp(FDR), ' (95% CI ', three_dp(lowerCI),' - ', three_dp(upperCI), ')')) %>% .[['a']] ` and minimum of `r fdr_high_tab %>% filter(stat == 'Zeng.E', pop %in% poly_pop) %>% filter(min(FDR) == FDR) %>% mutate(a = paste0(three_dp(FDR), ' (95% CI ', three_dp(lowerCI),' - ', three_dp(upperCI), ')')) %>% .[['a']] ` and was conservatively equivalent to the `r fdr_high_tab %>% filter(stat == 'Zeng.E', pop %in% poly_pop) %>% summarise(round_dp(min(highest_quant), 2))`th percentile from the permutations. The mean number of windows per population that met the significance threshold for the upper tail was `r upper_sig_stats %>% filter(statname == 'Zeng.E') %>% group_by(pop) %>% tally() %>% summarise(mean(n)) %>% round(digits = 0)` (SD `r upper_sig_stats %>% filter(statname == 'Zeng.E') %>% group_by(pop) %>% tally() %>% summarise(sd(n)) %>% round(digits = 0)`), with a minimum of `r upper_sig_stats %>% filter(statname == 'Zeng.E') %>% group_by(pop) %>% tally() %>% summarise(min(n)) %>% round(digits = 0)` and a maximum of `r upper_sig_stats %>% filter(statname == 'Zeng.E') %>% group_by(pop) %>% tally() %>% summarise(max(n)) %>% round(digits = 0)`. The mean number of genes that were unique to a population was `r ze_pop_unique_genes_upper %>% summarise(mean(n)) %>% round(digits = 0)` (SD `r ze_pop_unique_genes_upper %>% summarise(sd(n)) %>% round(digits = 0)`) and minimum of `r ze_pop_unique_genes_upper %>% summarise(min(n)) %>% round(digits = 0)` and maximum of `r ze_pop_unique_genes_upper %>% summarise(max(n)) %>% round(digits = 0)`.
 

Again, the bias against lower frequency variants looks like it has led to the entire distribution being shifted right. Unlike the other frequency based statistics, it was the lower threshold that had the worse expected \gls{fdr}.


<!-- - summary of pol zeng e (Table \@ref(tab:gsZeTable)). -->
<!-- - Genes Unique to Polynesian populations Table \@ref(tab:lowersigzePol) -->

<!--
The mean \gls{ze} for the Polynesian populations was `r global_super_summary %>% filter(super_pop == 'POL', stat == 'Zeng.E') %>% .[["mean"]] %>% three_dp()`  (SD `r global_super_summary %>% filter(super_pop == 'POL', stat == 'Tajima.D') %>% .[["sd"]] %>% three_dp()`) with  `r gs_ze_table %>% filter(pop == 'CIM') %>% .[["mean"]] %>% three_dp()` (SD `r gs_ze_table %>% filter(pop == 'CIM') %>% .[["sd"]] %>% three_dp()`) for \gls{cim}, `r gs_ze_table %>% filter(pop == 'NZM') %>% .[["mean"]] %>% three_dp()` (SD `r gs_ze_table %>% filter(pop == 'NZM') %>% .[["sd"]] %>% three_dp()`) for \gls{nzm}, `r gs_ze_table %>% filter(pop == 'SAM') %>% .[["mean"]] %>% three_dp()` (SD `r gs_ze_table %>% filter(pop == 'SAM') %>% .[["sd"]] %>% three_dp()`) for \gls{sam}, and `r gs_ze_table %>% filter(pop == 'TON') %>% .[["mean"]] %>% three_dp()` (SD `r gs_ze_table %>% filter(pop == 'TON') %>% .[["sd"]] %>% three_dp()`) for \gls{ton}.
-->

```{r ze_ranges, eval = FALSE, include = FALSE}
#CIM
lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length() # regions 
lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6 # Mbp of region 
lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW() # genes 
lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Zeng.E', !pop %in% c('CIM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW() # genes only from CIM
gr <- lower_sig_stats %>% filter(pop == "CIM", statname == 'Zeng.E') %>%  GenomicRanges::GRanges() %>% GenomicRanges::reduce()
gr[GenomicRanges::width(gr) >= 20 * 10e3]


#NZM
lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length() # regions 
lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6 # Mbp of region 
lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW() # genes 
lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('NZM')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Zeng.E', !pop %in% c('NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW() # genes only from NZM


gr <- lower_sig_stats %>% filter(pop == "NZM", statname == 'Zeng.E') %>%  GenomicRanges::GRanges() %>% GenomicRanges::reduce()
gr[GenomicRanges::width(gr) >= 20 * 10e3]

#SAM
lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length() # regions 
lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6 # Mbp of region 
lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW() # genes 
lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('SAM')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Zeng.E', !pop %in% c('SAM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW() # genes only from SAM
gr <- lower_sig_stats %>% filter(pop == "SAM", statname == 'Zeng.E') %>%  GenomicRanges::GRanges() %>% GenomicRanges::reduce()
gr[GenomicRanges::width(gr) >= 20 * 10e3]


#TON
lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length() # regions 
lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6 # Mbp of region 
lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('TON')) %>% select('SYMBOL') %>% distinct() %>% NROW() # genes 
lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('TON')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Zeng.E', !pop %in% c('TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW() # genes only from TON
gr <- lower_sig_stats %>% filter(pop == "TON", statname == 'Zeng.E') %>%  GenomicRanges::GRanges() %>% GenomicRanges::reduce()
gr[GenomicRanges::width(gr) >= 20 * 10e3]


#EP
lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM', 'NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length() # regions 
lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM', 'NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6 # Mbp of region 
lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM', 'NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes 
lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM', 'NZM')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Zeng.E', !pop %in% c('CIM','NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW() # genes only from EP

#WP

lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length() # regions 
lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6 # Mbp of region 
lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('SAM','TON')) %>% select('SYMBOL') %>% distinct() %>% NROW() # genes 
lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('SAM','TON')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Zeng.E', !pop %in% c('SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW() # genes only from WP

#POL
lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length() # regions 
lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6 # Mbp of region 
lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM','NZM','SAM','TON')) %>% select('SYMBOL') %>% distinct() %>% NROW() # genes 
lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM','NZM','SAM','TON')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Zeng.E', !pop %in% c('CIM','NZM','SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW() # genes only from POL
```


\FloatBarrier

#### iHS and nSL

```{r}
ihs_nsl_clus_tab <- left_join(ihs_clus %>% filter(statid == 26) %>% select(pop,ClusterID, chrom) %>% distinct() %>% group_by(pop) %>% tally() %>% rename(n_ihs_clus = n), nsl_clus %>% filter(statid == 20) %>% select(pop,ClusterID, chrom) %>% distinct() %>% group_by(pop) %>% tally() %>% rename(n_nsl_clus = n), by = 'pop')
  
ihs_nsl_clus_tab <- ihs_nsl_clus_tab %>% left_join(., ihs_nsl_sum_tab %>% left_join(., panel %>% select(pop, super_pop) %>% distinct(), by = 'pop') %>% mutate(pop = paste0(super_pop,'_',pop), by = 'pop'), by = 'pop')  %>% select(-super_pop) %>% select(pop, n_ihs_snps, n_nsl_snps, n, n_ihs_clus, n_nsl_clus)

ihs_nsl_clus_tab %>% 
  mutate(pop = substr(pop, 5,7)) %>% 
  kable(., booktabs = TRUE,
        col.names = c("Population", "iHS SNPs (n)", "nSL SNPs (n)", "Both (n)", "iHS Clusters (n)", 'nSL Clusters (n)'), 
        caption = '(\\#tab:iHSnSLClusSum) Number of significant SNPs and clustered regions for iHS and nSL by population.') %>%
  kableExtra::group_rows("AFR", 1, 7) %>% 
  kableExtra::group_rows("AMR", 8, 11) %>% 
  kableExtra::group_rows("EAS", 12, 16) %>% 
  kableExtra::group_rows("EUR", 17, 22) %>% 
  kableExtra::group_rows("POL", 23, 26) %>% 
  kableExtra::group_rows("SAS", 27,31)

# add on number that were shared
#add on number of single snp clusters?

#sig_ihs %>% select(pop, chrom, chrom_start, chrom_end) %>% inner_join(., sig_nsl %>% select(pop, chrom, chrom_start, chrom_end), by = c('pop','chrom','chrom_start','chrom_end') ) %>% group_by(pop) %>% tally()
```



Both \gls{ihs} and the related \gls{nsl} were calculated and significance determined after normalising and taking the 1% most extreme. The markers having been identified as significant were then clustered using the DBSCAN package in R. For more details refer to Section \@ref(selectionTools). This was done to identify regions that contained multiple significant \gls{snp}s.

The Polynesian populations had the least number of significant \gls{snp}s for both \gls{ihs} and \gls{nsl} (Table \@ref(tab:iHSnSLClusSum)). The number of clusters followed the same trend as the number of significant \gls{snp}s in that there were both more \gls{snp}s and clusters for \gls{ihs} than for \gls{nsl}. The minimum number of \gls{snp}s that had a significant \gls{ihs} was `r sig_ihs_val %>% group_by(pop) %>% tally() %>% ungroup() %>% filter(min(n) == n) %>% .[['n']]` and was from the \gls{`r sig_ihs_val %>% group_by(pop) %>% tally() %>% ungroup() %>% filter(min(n) == n) %>% .[['pop']] %>% tolower()`} population. For \gls{nsl} this was also \gls{`r sig_nsl_val %>% group_by(pop) %>% tally() %>% ungroup() %>% filter(min(n) == n)  %>% .[['pop']] %>% tolower`} with `r sig_nsl_val %>% group_by(pop) %>% tally() %>% ungroup() %>% filter(min(n) == n) %>% .[['n']]` \gls{snp}s. The maximum number of \gls{snp}s was for \gls{`r sig_ihs_val %>% group_by(pop) %>% tally() %>% ungroup() %>% filter(max(n) == n) %>% .[['pop']] %>% tolower()`} with `r sig_ihs_val %>% group_by(pop) %>% tally() %>% ungroup() %>% filter(max(n) == n) %>% .[['n']]` for \gls{ihs} and \gls{`r sig_nsl_val %>% group_by(pop) %>% tally() %>% ungroup() %>% filter(max(n) == n) %>% .[['pop']] %>% tolower()`} with `r sig_nsl_val %>% group_by(pop) %>% tally() %>% ungroup() %>% filter(max(n) == n) %>% .[['n']]` \gls{snp}s. The mean number of significant \gls{snp}s for \gls{ihs} was `r sig_ihs_val %>% group_by(pop) %>% tally() %>% ungroup() %>% summarise(round_dp(mean(n), 1))` (SD  `r sig_ihs_val %>% group_by(pop) %>% tally() %>% ungroup() %>% summarise(round_dp(sd(n), 1))`) and `r sig_nsl_val %>% group_by(pop) %>% tally() %>% ungroup() %>% summarise(round_dp(mean(n), 1))`  (SD `r sig_nsl_val %>% group_by(pop) %>% tally() %>% ungroup() %>% summarise(round_dp(sd(n), 1))`) for \gls{nsl} (Table \@ref(tab:iHSnSLClusSum)). The Polynesian populations had a mean of `r sig_ihs_val %>% group_by(pop,superpop) %>% tally() %>% group_by(superpop) %>% summarise(mn = mean(n)) %>% filter(superpop == 'POL') %>% .[['mn']] %>% sprintf("%.0f",.)` SNPs with a \gls{ihs} value that was significant. Compared to  `r sig_nsl_val %>% group_by(pop,superpop) %>% tally() %>% group_by(superpop) %>% summarise(mn = mean(n)) %>% filter(superpop == 'POL') %>% .[['mn']] %>% sprintf("%.0f",.)` for \gls{nsl}. Between significant \gls{snp}s for \gls{ihs} and \gls{nsl} there was a mean of `r ihs_nsl_sum_tab %>% left_join(., panel %>% select(pop, super_pop) %>% distinct(), by = 'pop') %>% filter(super_pop == 'POL') %>% summarise(round_dp(mean(n), 1))` \gls{snp}s in common for the \gls{pol} populations, `r ihs_nsl_sum_tab %>% left_join(., panel %>% select(pop, super_pop) %>% distinct(), by = 'pop') %>% filter(super_pop == 'EAS') %>% summarise(round_dp(mean(n), 1))` for \gls{eas}, `r ihs_nsl_sum_tab %>% left_join(., panel %>% select(pop, super_pop) %>% distinct(), by = 'pop') %>% filter(super_pop == 'AFR') %>% summarise(round_dp(mean(n), 1))` \gls{afr}, `r ihs_nsl_sum_tab %>% left_join(., panel %>% select(pop, super_pop) %>% distinct(), by = 'pop') %>% filter(super_pop == 'EUR') %>% summarise(round_dp(mean(n), 1))` for \gls{eur}, `r ihs_nsl_sum_tab %>% left_join(., panel %>% select(pop, super_pop) %>% distinct(), by = 'pop') %>% filter(super_pop == 'AMR') %>% summarise(round_dp(mean(n), 1))` for \gls{amr}, and `r ihs_nsl_sum_tab %>% left_join(., panel %>% select(pop, super_pop) %>% distinct(), by = 'pop') %>% filter(super_pop == 'SAS') %>% summarise(round_dp(mean(n), 1))` from \gls{sas}.




<!-- NEED TO UPDATE -->
<!-- - poly ihs had the least number of snps but not too diff from CHS -->
<!-- - snps:cluster was ~4.4 for poly, compared to most others of 3.5 -->
<!-- - nsl had less snps, pol had more clusters -->
<!-- - nsl eas and pol had similar number of snps, the ratio isn't as different -->
<!-- - more clusters indicates  -->

```{r, include = FALSE, eval = FALSE}
# calculate mean number of snps, total number of clusters, and mean width of clusters by
# population after filtering out single base clusters
ihs_clus %>%select(pop, chrom, ClusterID, chrom_start, chrom_end) %>% distinct() %>% group_by(pop, chrom, ClusterID) %>% summarise(start = min(chrom_start), end = max(chrom_end), snps = n()) %>% distinct() %>% mutate(width = (end - start)/1e6, snp_per_mb = snps/width) %>% filter(!(end - start == 1 & snps == 1)) %>% group_by(pop) %>% summarise(ms = mean(snps), n= n(), mw = mean(width)) %>% arrange(desc(ms, mw))


nsl_clus %>%select(pop, chrom, ClusterID, chrom_start, chrom_end) %>% distinct() %>% group_by(pop, chrom, ClusterID) %>% summarise(start = min(chrom_start), end = max(chrom_end), snps = n()) %>% distinct() %>% mutate(width = (end - start)/1e6, snp_per_mb = snps/width) %>% filter(!(end - start == 1 & snps == 1)) %>% group_by(pop) %>% summarise(ms = mean(snps), n= n(), mw = mean(width)) %>% arrange(desc( ms))
```


<!-- > Was there anything specific about the genes? -->


```{r, eval = FALSE}
sig_ihs_val %>% left_join(., panel %>% select(pop, super_pop) %>% distinct(), by = 'pop')%>% anti_join(., sig_ihs_val %>%  left_join(., panel %>% select(pop, super_pop) %>% distinct(), by = 'pop')%>% filter(super_pop != 'POL'), by = c('genename')) %>% select(pop, genename) %>% distinct() %>% group_by(genename) %>% tally()

sig_ihs_val %>% anti_join(., sig_ihs_val %>% filter(!pop %in% c('CIM', 'NZM')), by = c('genename')) %>% select(pop, genename) %>% distinct() %>% group_by(genename) %>% tally() %>% filter( n == 2)

sig_ihs_val %>% anti_join(., sig_ihs_val %>% filter(!pop %in% c('SAM', 'TON')), by = c('genename')) %>% select(pop, genename) %>% distinct() %>% group_by(genename) %>% tally() %>% filter( n == 2)
```



```{r}
# Gout specific
# number of ihs snps by gout gene by population
sig_ihs_val %>% 
  filter(genename %in% urate_goutGR$SYMBOL) %>% 
  group_by(pop, genename) %>% tally() %>%
  arrange(genename) %>% 
  spread(genename, n, fill= "") %>% 
  arrange(pop) %>% 
  ungroup() %>%  
  right_join(., panel %>% select(pop, super_pop) %>% distinct() %>% filter(!pop %in% c('WPN','EPN','NAD')), by = 'pop', fill = '') %>% arrange(super_pop) %>% 
   select('Population' = pop, everything(), -super_pop) %>% mutate_all(funs(replace(., is.na(.), ""))) %>% 
  kable(., booktabs = TRUE, caption  = 'Number of SNPs with significant iHS in urate and gout associated genes') %>%
  kableExtra::row_spec(0, angle = 45) %>%
  kableExtra::group_rows("AFR", 1, 7) %>%
  kableExtra::group_rows("AMR", 8, 11) %>%
  kableExtra::group_rows("EAS", 12, 16) %>% kableExtra::group_rows("EUR", 17, 22) %>%
  kableExtra::group_rows("POL", 23, 26) %>%
  kableExtra::group_rows("SAS", 27,31) %>% kable_styling(latex_options = c("scale_down"))
```




```{r nSharedHaplo, eval = TRUE}
join_ihs_nsl <- function(p1, p2){
  inner_join(bind_rows(sig_ihs_val, sig_nsl_val) %>% filter(pop == p1) %>% select(chrom, chrom_start, chrom_end, statid), bind_rows(sig_ihs_val, sig_nsl_val) %>% filter(pop == p2) %>% select(chrom, chrom_start, chrom_end, statid), by = c("chrom", "chrom_start", "chrom_end", "statid")) %>% group_by(statid) %>% tally() %>% mutate(p1 = p1, p2 = p2, statid = case_when(statid == 20 ~ 'nsl', statid == 26 ~ 'ihs')) %>% spread(statid, n)
}
p <- panel %>% select(pop) %>% distinct() %>% pull(pop)

n_shared_haplo <- map(poly_pop, function(x){map2(x, p, ~ join_ihs_nsl(.x, .y)) %>% bind_rows()}) %>% bind_rows()

inner_join(n_shared_haplo %>% select(p1, p2, ihs) %>% spread(p1, ihs), n_shared_haplo %>% select(p1, p2, nsl) %>% spread(p1, nsl), by = 'p2') %>% kable(., col.names = c("Population",rep(c("CIM","NZM","SAM","TON"),2)), format = 'latex', booktabs = TRUE, caption = '(\\#tab:nSharedHaplo) Number of significant SNPs that were in common between populations.') %>% kableExtra::add_header_above(., c("", "iHS" = 4,"nSL" =4))
rm(p)
```


On average there were `r n_shared_haplo %>% filter(!p2 %in% poly_pop) %>% summarise(mean(ihs)) %>% round(., 1)` \gls{snp}s in common between the Polynesian populations and the other populations for \gls{ihs} and `r n_shared_haplo %>% filter(!p2 %in% poly_pop) %>% summarise(mean(nsl)) %>% round(., 1)` \gls{snp}s for \gls{nsl}. Whereas there was a higher number of significant \gls{snp}s in common between the Polynesian populations with a mean of `r n_shared_haplo %>% filter(p2 %in% poly_pop, p1 != p2) %>% summarise(mean(ihs)) %>% round(., 1)` for \gls{ihs} and `r n_shared_haplo %>% filter(p2 %in% poly_pop, p1 != p2) %>% summarise(mean(nsl)) %>% round(., 1)` for \gls{nsl} (Table \@ref(nSharedHaplo)).


```{r, eval = TRUE}
# gout specific
# number of nsl snps by gout gene by poplation
sig_nsl_val %>% 
  filter(genename %in% urate_goutGR$SYMBOL) %>% 
  group_by(pop, genename) %>% tally() %>%
  arrange(genename) %>% 
  spread(genename, n, fill= "") %>% 
  arrange(pop) %>% 
  ungroup() %>%  
  right_join(., panel %>% select(pop, super_pop) %>% distinct() %>% filter(!pop %in% c('WPN','EPN','NAD')), by = 'pop', fill = '') %>% arrange(super_pop) %>% 
   select('Population' = pop, everything(), -super_pop) %>% mutate_all(funs(replace(., is.na(.), ""))) %>% 
  kable(., booktabs = TRUE, caption  = 'Number of SNPs with significant nSL in urate and gout associated genes') %>%
  kableExtra::row_spec(0, angle = 45) %>%
  kableExtra::group_rows("AFR", 1, 7) %>%
  kableExtra::group_rows("AMR", 8, 11) %>%
  kableExtra::group_rows("EAS", 12, 16) %>% kableExtra::group_rows("EUR", 17, 22) %>%
  kableExtra::group_rows("POL", 23, 26) %>%
  kableExtra::group_rows("SAS", 27,31) %>% kable_styling(latex_options = c("scale_down"))
```

```{r, eval = FALSE}
# difference between ihs and nsl:
full_join(sig_ihs_val %>% select(chrom, chrom_start, chrom_end, pop, genename)%>% anti_join(., sig_nsl_val, by =c("chrom","chrom_start", "chrom_end","pop", 'genename')) %>% group_by(pop, genename) %>% tally() %>% arrange(genename) %>% select(pop, genename, n_ihs = n),

sig_nsl_val %>% select(chrom, chrom_start, chrom_end, pop, genename)%>% anti_join(., sig_ihs_val, by =c("chrom","chrom_start", "chrom_end","pop","genename")) %>% group_by(pop, genename) %>% tally() %>% arrange(genename) %>% select(pop, genename, n_nsl = n), by = c('pop', 'genename'))
```


```{r, eval = FALSE}
bind_rows(sig_ihs_val %>% 
            mutate(statname = 'ihs'), sig_nsl_val %>% 
            mutate(statname = 'nsl')) %>% 
  filter(pop %in% poly_pop, genename %in% c(urate_goutGR$SYMBOL, t2d_GR$SYMBOL, obesity_GR$SYMBOL, metsyn_GR$SYMBOL)) %>% 
  mutate(pop = paste(statname,pop)) %>% 
  group_by(genename, pop) %>% 
  tally() %>% 
  arrange(desc(n)) %>% 
  spread(pop, n, fill = "") %>% 
  kable(., longtable = TRUE, caption = '(\\#ref:iHSnSLPoltable) The number of significant SNPs across associated genes for Polynsians populations for iHS and nSL')
```

```{r n_ihs_gout_snps, include = FALSE}
# calculate the number of sig snps for ihs/nsl for all metabolic related genes for Polynesians
bind_rows(sig_ihs_val %>% 
            mutate(statname = 'ihs'), sig_nsl_val %>% 
            mutate(statname = 'nsl')) %>%
  filter(pop %in% poly_pop, genename %in% c(urate_goutGR$SYMBOL, t2d_GR$SYMBOL, obesity_GR$SYMBOL, metsyn_GR$SYMBOL)) %>% 
  mutate(pop = paste(statname,pop)) %>% 
  group_by(genename, pop) %>% tally() %>% 
  arrange(desc(n)) %>% 
  anti_join(., bind_rows(sig_ihs_val %>% # remove genes that are found in non-Pol populations
                           mutate(statname = 'ihs'), sig_nsl_val %>% 
                           mutate(statname = 'nsl')) %>% 
              filter(!pop %in% poly_pop) %>% 
              select(pop, genename), by = c('pop', 'genename') ) %>% 
  spread(pop, n, fill = '') %>% 
  kable(., caption = 'Disease genes with significant hit in only Polynesian populations from metabolic lists', longtable = TRUE)
```


#### False discovery rate of windowed statistics

```{r fdrpol}
 

fdr_low_tab %>% 
  left_join(., fdr_high_tab, by = c('stat','pop')) %>%
  mutate_at(vars(contains('.x'), contains('.y')), funs(three_dp)) %>%
  mutate(fdr_low = paste0(FDR.x, " (", lowerCI.x,', ', upperCI.x,')'), fdr_high = paste0(FDR.y, " (", lowerCI.y,', ', upperCI.y,')')) %>%
  select(stat, pop, lowest_quant, fdr_low, highest_quant, fdr_high) %>%
  kable(., col.names = c("Stat","Population","Percentile", "FDR (95% CI)", "Percentile", "FDR (95% CI)"), caption = "(\\#tab:fdrpol) False discovery rate for windowed frequency based statistics by population.", booktabs = TRUE, format = 'latex') %>% 
  kableExtra::add_header_above(., c("","","Lower tail" = 2, "Upper tail" = 2) ) %>% 
  kableExtra::group_rows("Fay and Wu's H",1, 4) %>%
  kableExtra::group_rows("Fu and Li's D", 5, 8) %>%
  kableExtra::group_rows("Fu and Li's F", 9, 12) %>%
  kableExtra::group_rows("Tajima's D", 13, 16) %>%
  kableExtra::group_rows("Zeng's E", 17, 20) %>%
  kableExtra::add_footnote(c("Percentile is the more conservative FDR quantile bin of the permuted data that was equivalent to either the 1st (lower tail) or 99th (upper tail) percentile in the observed data" ), notation = "symbol")
```

To calculate the false discovery rate, 5000 permutations were generated, each randomly shuffling the genotypes of markers by marker position (i.e shuffling rows only). For each permutation by statistic, the values for quantiles 'bins' ranging from 0.01 to 0.1% at 0.01 increments,  0.1 to 1% at 0.1% increments, 1 to 5% at 1% increments, and 10% for both tails of the distribution. The permutations were then combined and the median and 95% confidence intervals were calculated for each of the quantile 'bins'. The enrichment for each quantile bin was then calculated by comparing the number of windows that were observed in the data compared to the number expected from the permutations. \Gls{fdr} was then estimated by using the value for the 1st or 99th percentile and finding the nearest quantile bin using the median, that was either larger for the lower tail, or smaller for the upper tail and taking the inverse of enrichment for the bin (Table \@ref(tab:fdrpol)). This estimate will be conservative compared to the true \gls{fdr}.

The \gls{fdr} for \gls{ze} was 1.0 for all Polynesian populations in the lower end of the distribution and the 1st percentile of the data fell within the 2nd percentile bin from the permutations. Whereas, for \gls{td} and \gls{fwh} the quantile bin was `r fdr_low_tab %>%filter(stat %in% c('Tajima.D','Fay.Wu.H')) %>% summarise(min(lowest_quant))`%, with the maximum upper confidence interval being `r fdr_low_tab %>%filter(stat %in% c('Tajima.D','Fay.Wu.H')) %>% summarise(three_dp(max(upperCI)))`. 


#### Comparison with prior publications


```{r priorGenes, include = FALSE}
# chr_unfiltered_stats <- readRDS('~/Murray/Thesis_backup/NZ_coreExome_1kgp/100kbWindow_intra/unfiltered/100kbwindows_chr_stats_popgenome_long.31-8-2017.RDS')
# head(chr_unfiltered_stats)
# library(GenomicRanges)
# library(TxDb.Hsapiens.UCSC.hg19.knownGene)
# library(org.Hs.eg.db)
# library(tidyverse)
# ns <- chr_unfiltered_stats %>% filter(stat == 'n.segregating.sites') %>% ungroup() %>% mutate(chrom = paste0('chr',chrom))
# 
# hider_genes <- unique(unlist(list(
#   td = c("EDAR","MITF","ATRN","TRPM1", "TMEM33","BNC2","OPRM1","ADAM17"), 
#   fwh = c("MITF","BNC2","TYRP1","MLPH","PDIA6"), 
#   fld = c("MITF","ATRN","KLHL7","TMEM33","PMEL","SILV","BNC2","LYST","DCT","CTSD"), 
#   flf = c("MITF","ATRN","KLHL7","TMEM33","PMEL","SILV","BNC2","LYST","DCT","CTSD"),
#   ihs = c("EDAR","TH","MC1R") 
# )))
# 
# genes_entrez <- na.omit(AnnotationDbi::select(org.Hs.eg.db, keys = hider_genes, columns = c('SYMBOL','ENTREZID'), keytype="SYMBOL"))
# 
# genes_ucsc <- merge(AnnotationDbi::select(TxDb.Hsapiens.UCSC.hg19.knownGene , columns = c("TXID","TXCHROM","TXSTART","TXEND","TXSTRAND"),keys = genes_entrez$ENTREZID, keytype="GENEID"), genes_entrez, by.x ='GENEID', by.y = "ENTREZID")
# genes_ucscGR <- GRanges(genes_ucsc[!is.na(genes_ucsc$TXID) ,])
# genes_ucscGR <- genes_ucscGR[which(genes_ucscGR@seqnames %in% paste0('chr',1:22))]
# 
# ns_GR <- GRanges(ns)
# ol <- findOverlaps(query = genes_ucscGR, subject = ns_GR)
# saveRDS(chr_unfiltered_stats %>% ungroup() %>% filter(stat == 'n.segregating.sites', value < 4) %>% select(contains('chrom'), stat,value) %>% distinct() %>% mutate(chrom_start = as.numeric(chrom_start) + 45000, chrom_end = as.numeric(chrom_end) - 45000 ), file = '~/Murray/Thesis_backup/NZ_coreExome_1kgp/100kbWindow_intra/filtered/100kbwindows_chr_stats_popgenome_removed_windows_resized.25-9-2017.RDS')


#EAS (CHB, CHS, and JPN) regions paper
hider <- read_csv('data/hider2013_S1.csv', col_names = TRUE) %>% mutate(fld = ifelse(is.na(stringr::str_extract(Test, fixed('D*-'))),FALSE, TRUE), flf = ifelse(is.na(stringr::str_extract(Test, fixed('F*-'))),FALSE, TRUE), fwh = ifelse(is.na(stringr::str_extract(Test, fixed('H-'))),FALSE, TRUE), td = ifelse(is.na(stringr::str_extract(Test, fixed('D-'))),FALSE, TRUE) )


# from hider2013 table 1
hider_genes <- list(
  td = c("EDAR","MITF","ATRN","TRPM1", "TMEM33","BNC2","OPRM1","ADAM17"), 
  fwh = c("MITF","BNC2","TYRP1","MLPH","PDIA6"), 
  fld = c("MITF","ATRN","KLHL7","TMEM33","PMEL","SILV","BNC2","LYST","DCT","CTSD"), 
  flf = c("MITF","ATRN","KLHL7","TMEM33","PMEL","SILV","BNC2","LYST","DCT","CTSD"),
  ihs = c("EDAR","TH","MC1R") 
    )

# SAS (GIH and ITU) from Joannalagadda2017
joannalagadda_genes <- list(
td = c("DST","PDGFC","PDPK1","SLC24A5","MITF"),
fwh = c("BLOC1S5","MITF","MLPH","PTS","MBD4"),
ihs = c("HS2ST1","MLPH","SEMA3C","TYR","ADAM17")
)

lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CHB','CHS','JPN'), SYMBOL %in% hider_genes$td) %>% group_by(pop, SYMBOL) %>% tally()

lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CHB','CHS','JPN'), SYMBOL %in% hider_genes$fwh) %>% group_by(pop, SYMBOL) %>% tally()

lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CHB','CHS','JPN'), SYMBOL %in% hider_genes$fld) %>% group_by(pop, SYMBOL) %>% tally()

lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CHB','CHS','JPN'), SYMBOL %in% hider_genes$flf) %>% group_by(pop, SYMBOL) %>% tally()

lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('GIH','ITU'), SYMBOL %in% joannalagadda_genes$td) %>% group_by(pop, SYMBOL) %>% tally()

lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('GIH','ITU'), SYMBOL %in% joannalagadda_genes$fwh) %>% group_by(pop, SYMBOL) %>% tally()



#compare the regions between the Hiders regions and those that I identified as significant

h_tdGR <- hider %>% mutate(chrom =  paste0('chr',chr)) %>% filter(td == TRUE) %>% select(chrom, start,end) %>% GRanges() 

lo_sig_tdGR <- lower_sig_stats %>% filter(pop %in% c("CHB","CHS","JPT"), statname == 'Tajima.D') %>% select(contains('chrom')) %>% GRanges()

hits <- findOverlaps(query = lo_sig_tdGR, subject = h_tdGR)
h_tdGR[hits@to] %>% data.frame() %>% distinct() %>% NROW()
```

```{r ramos}
ramos <- list()
ramos[["ceu"]]  <- c("ALMS1P", "DGUOK", "PARD3B", "KCNIP4", "HLA", "PRKCH", "HIF1A", "IRF4", "TECRL", "CTNNA3", "WWOX", "RABEP1", "NUP88", "COMMD10", "SEMA6A", "ALDH7A1", "SNRPC", "UHRF1BP1", "OSBPL8", "GRM5", "NTNG2", "SETX", "GRHL2")
ramos[["yri"]] <- c("TNFRSF14", "MMEL1", "MAGI3", "PTPN22", "TLR5", "XCR1", "CCR3", "CCDC66", "ARHGEF3", "SLC2A9", "KCNIP4", "ARHGAP31", "CD80", "IL2", "IL21", "TPR3", "HLA", "CCDC167", "MIR4462", "PRSS54", "PRDM1", "ATG5", "TRAF3IP2", "PAK7PRKCH", "HIF1A", "HIP1")
ramos[["asi"]] <- c("NCF2", "RGL1", "PELI1", "HLA", "BCAS3", "NACA2", "CNTN6", "BTLA", "MRPS22", "COMMD10", "SEMA6A", "GTF2I", "KIAA0391", "FAM171A1", "KDM4C", "XKR6", "BLK", "LSMEM1", "NPM1P14")
ramos[["unknown"]] <- c("IL23R", "FCGR2B", "TNFSF4", "CR1", "ANTRX2", "TNIP1", "PTTG1", "VARS", "LSM2", "IKZF1", "IRF8", "TYK2", "CLEC16A", "CIITA", "ITGAM", "ITGAX", "SH2B3", "NAA25", "CD5", "PTGER4")
```

The selection and neutrality statistics that were in the 1st percentile were compared to regions and genes identified in @Hider2013 for \gls{eas} and @Jonnalagadda2017 for \gls{sas} as these papers had used a similar, but not identical methodology to identify regions under selection. @Hider2013 investigated selection in the East Asian populations (\gls{chb}, \gls{chs}, and \gls{jpt}) of the 1000 genomes project in the Phase 1 release. For this they used a non-overlapping window size of 25kb and excluded windows with less than 10 segregrating sites and then looked at the top 1% of results. They reported regions that were significant for \gls{td}, \gls{fwh}, \gls{fld}, \gls{flf}, and \gls{ihs}. Comparing the regions from @Hider2013 with regions that met the significance threshold for this project there were 118 of 206 regions for \gls{td}, 15 of 151 for \gls{fwh}, 65 of 261 for \gls{fld}, 83 of 255 for \gls{flf}, and 5 of 85 for \gls{ihs}.

Of the Genes reported in @Jonnalagadda2017, there were 2 of 5 for \gls{td}, 0 of 5 for \gls{fwh}, and 2 of 5 for \gls{ihs} that were also significant in this project. @Jonnalagadda2017 used the same window set up as @Hider2013, but used the 1000 Genomes Project Phase 3 release for the \gls{gih} and \gls{itu} populations. 


Inconjunction with the studies from @Hider2013 and @Jonnalagadda2017, @Ramos2017 looked at evidence of selection in rheumatic disease, including gout, using the \gls{ihs} results from @voight2006map from the Hapmap phase II dataset. There were three populations used which were Asian, European, and African. The equivalent populations of \gls{chb} and \gls{jpt} were used to compare to the Asian population, \gls{ceu} for European, and \gls{yri} for African, and unknown was any of the specified populations, from the 1000 genomes individuals when comparing the loci reported in @Ramos2017 with the \gls{ihs} results. Loci reported for European had  `r (sig_ihs_val %>% filter(pop == 'CEU', genename %in% ramos[['ceu']])%>% mutate(pop = 'CEU', statname = case_when(statid == 20 ~ "nSL", statid == 26 ~ "iHS"), chrom = paste0('chr',chrom)) %>%  select(SYMBOL = genename, chrom, statname) %>% distinct() %>% NROW() / length(ramos[["ceu"]] ) *100) %>% round_dp(., 1)`% overlap with \gls{ceu}. The Asian reported loci matched `r (sig_ihs_val %>% filter(pop == 'JPT' | pop == 'CHB', genename %in% ramos[['asi']])%>% mutate(pop = 'ASI', statname = case_when(statid == 20 ~ "nSL", statid == 26 ~ "iHS"), chrom = paste0('chr',chrom)) %>%  select(SYMBOL = genename, chrom, statname) %>% distinct() %>% NROW() / length(ramos[["asi"]] )*100) %>% round_dp(., 1)`% with \gls{chb} and \gls{jpt}, and for African there was `r (sig_ihs_val %>% filter(pop == 'YRI', genename %in% ramos[['yri']])%>% mutate(pop = 'YRI', statname = case_when(statid == 20 ~ "nSL", statid == 26 ~ "iHS"), chrom = paste0('chr',chrom)) %>%  select(SYMBOL = genename, chrom, statname) %>% distinct() %>% NROW() / length(ramos[["yri"]] ) *100) %>% round_dp(., 1)`% of the reported loci also found in \gls{yri}. For the loci reported with an unknown population, `r (sig_ihs_val %>% filter(pop %in% c('CEU','YRI','JPT','CHB' ), genename %in% ramos[['ceu']])%>% mutate(pop = 'ceu_yri_asi', statname = case_when(statid == 21 ~ "nSL", statid == 27 ~ "iHS"), chrom = paste0('chr',chrom)) %>%  select(SYMBOL = genename, chrom, statname) %>% distinct() %>% NROW() / length(ramos[["unknown"]] )*100) %>% round_dp(.,1)`% were found to match in any of \gls{ceu}, \gls{yri}, \gls{chb}, or \gls{jpt}.



<!--
NEED TO RELOOK AT THE HAPLOTYPIC RESULTS 15/12/2017 - DONE
results are from ~/Git_repos/Thesis/selectionpipeline/prior_pubs_comparison_SFS_results.Rmd

values are for the corrected 

results from looking at all of the hider regions

- td
    * 206 / 243 regions with as least 4 sites in my results
    * found 118/206 (0.573) windows from hider regions
- fwh
    * 151 / 158
    * 15 / 151 (0.0993)
- fld
    * 261 / 283
    * 65 / 261  (0.249)
- flf
    * 255 / 281
    * 83 / 255 (0.325)
- ihs (regions are done slightly differently so not a good comparison)
    * 85 / 100
    * 5 / 85 (0.059)

joannalagga genes

@Joannalagga2017 used a similar approach as @Hider2013, in that they used the same window size 
td 2/5
fwh 0/5
ihs 2/5
-->

<!-- #### Summary of Intra -->

```{r, eval = FALSE}
lower_sig_stats %>% 
  filter(statname %in% c('Tajima.D','Fay.Wu.H','Zeng.E', 'Fu.Li.F', 'Fu.Li.D'), pop %in% c('CIM','NZM','SAM','TON')) %>% 
  select(pop,statname ,SYMBOL) %>% 
  distinct() %>% 
  group_by(pop, SYMBOL) %>%
  tally() %>%
  filter(n >=4) %>% 
  arrange(desc(n)) %>% 
  spread(pop, n) %>% 
  mutate(rs = select(., NZM, SAM,CIM,TON) %>% 
           rowSums(., na.rm=TRUE)) %>% 
  arrange(desc(rs)) %>% 
  filter(rs > 5)
```


```{r, eval = FALSE}
lower_sig_stats %>% filter(SYMBOL %in% urate_goutGR$SYMBOL) %>% select(SYMBOL, pop) %>% anti_join(., lower_sig_stats %>% filter(!pop %in% c("CIM","NZM","SAM","TON")) %>% select(SYMBOL, pop), by = "SYMBOL")%>% distinct() %>% spread(SYMBOL, SYMBOL, fill = "")
```

```{r, eval = FALSE}
lower_sig_stats %>% filter(SYMBOL %in% c(urate_goutGR$SYMBOL, obesity_GR$SYMBOL, t2d_GR$SYMBOL, kd_GR$SYMBOL, metsyn_GR$SYMBOL)) %>% select(SYMBOL, pop) %>% anti_join(., lower_sig_stats %>% filter(!pop %in% c("CIM","NZM","SAM","TON")) %>% select(SYMBOL, pop), by = "SYMBOL")%>% distinct() %>% spread(SYMBOL, SYMBOL, fill = "")
```



<!-- Out of all the genes from urate/obesity/t2d/kd/metsyn, _MAP2K5_ has a significant result for a stat in all 4 populations. _APOA5_ and _BUD13_ in all except \gls{nzm}, _PRKAG2_ and _IDE_ in only \gls{sam} and \gls{ton}. _BDNF_ and _FNDC3B_ were only significant in \gls{sam}, and _CAMK2B_ in only \gls{ton}  -->

```{r, eval = FALSE}
upper_sig_stats %>% 
  filter(statname %in% c('Tajima.D','Fay.Wu.H','Zeng.E', 'Fu.Li.F', 'Fu.Li.D'), pop %in% c('CIM','NZM','SAM','TON')) %>% 
  select(pop,statname ,SYMBOL) %>% 
  distinct() %>% 
  group_by(pop, SYMBOL) %>% 
  tally() %>%
  filter(n > 1) %>% 
  group_by(SYMBOL) %>% 
  tally()  %>% 
  arrange(desc(nn))
```

<!-- regions that differed between the 2 stats? -->

### Inter-population Selection and Neutrality Statistics

#### F~ST~

```{r readFst}
fst <- readRDS('~/data/NZ_coreExome_1kgp/whole_chr_fst_popgenome.3-7-2017.RDS')
fst <- bind_rows(lapply(fst, function(x){x %>% mutate(pop1 = as.character(pop1), pop2 = as.character(pop2))%>% select(-p1,-p2) %>% filter(!pop1 %in% c('WPN','EPN','NAD'), !pop2 %in% c('WPN','EPN','NAD')) %>% left_join(., panel %>% mutate(super_pop = paste(super_pop, pop, sep = '_')) %>% select("pop1" = pop, super_pop) %>% distinct(), by = "pop1" ) %>% select(-pop1) %>% spread(., "super_pop", fst )})) %>% group_by(pop2) %>% summarise_all(mean) %>% left_join(., panel %>% mutate(super_pop = paste(super_pop, pop, sep = '_')) %>% select("pop2" = pop, super_pop) %>% distinct(), by = "pop2" ) %>% select(Population = super_pop, everything(),-chrom, -pop2) %>% arrange(Population) %>% mutate_if(is.numeric, funs(sprintf('%.4f',.))) %>% data.frame(., row.names = NULL)

intra_pol_max_fst <- pol_fst %>% gather("pop",'fst', -contains('chrom'))%>% filter(!is.na(fst))  %>% separate(pop, c("pop1","pop2"), sep = '_') %>% group_by(pop1,pop2) %>% filter(max(fst) == fst, (pop1 %in% poly_pop) & (pop2 %in% poly_pop))
pol_fstGR <- pol_fst %>% select(contains('chrom')) %>% mutate(chrom = paste0("chr",chrom)) %>% GRanges()
```



```{r, eval = FALSE}
pol_fst %>% gather("pop",'fst', -contains('chrom'))%>% filter(!is.na(fst)) %>% separate(pop, c("p1", "p2"), "_") %>% left_join(., panel %>% select(p1 = pop, super_pop1 = super_pop) %>% distinct(), by = 'p1') %>% left_join(., panel %>% select(p2 = pop, super_pop2 = super_pop) %>% distinct(), by = 'p2') %>% group_by(p1,p2, super_pop1, super_pop2) %>% summarise(mean_fst = mean(fst), sd_fst = sd(fst),min_fst = min(fst), max_fst = max(fst)) %>% arrange(desc(max_fst))
```



F~ST~ was calculated in sliding windows of 10 kb across the genome, pairwise between the Polynesian populations and all other populations. Similar to the results in section \@ref(wholeFst) for the F~ST~ calculated using entire chromosomes at once, the mean F~ST~ using windows had the Polynesian populations most differentiated from the \gls{afr} populations, with a range of mean F~ST~ from 0.148 to 0.203. The Polynesian populations were least differentiated from the \gls{eas} populations with a range of F~ST~ means from 0.051 to 0.076. Between the Polynesian populations the mean F~ST~ ranged from 0.004, between the Western Polynesian populations of \gls{sam} and \gls{ton}, to 0.029 between \gls{nzm} and \gls{sam}. The Eastern Polynesian populations had a marginally higher mean F~ST~ than the Western Polynesian populations at 0.006. The largest maximum F~ST~ of 0.888, across four windows, and was between \gls{ton} and \gls{msl} at chr17:62455002-62495001. This same region was the maximum F~ST~ for the \gls{cim} and \gls{nzm}, and the second largest for \gls{sam}, all with \gls{msl}. The maximum F~ST~ between the Polynesian populations was between `r intra_pol_max_fst %>% ungroup() %>% filter(max(fst)==fst) %>% .[['pop1']]` and `r intra_pol_max_fst %>% ungroup() %>% filter(max(fst)==fst) %>% .[['pop2']]`, with a value of `r intra_pol_max_fst %>% ungroup() %>% filter(max(fst)==fst) %>% .[['fst']] %>% three_dp()` at `r as.character(intra_pol_max_fst %>% ungroup() %>% filter(max(fst)==fst) %>% .[['chrom']])`:`r as.character(intra_pol_max_fst %>% ungroup() %>% filter(max(fst)==fst) %>% .[['chrom_start']])`-`r as.character(intra_pol_max_fst %>% ungroup() %>% filter(max(fst)==fst) %>% .[['chrom_end']])`. The smallest maximum F~ST~ of the Polynesian populations was between `r intra_pol_max_fst %>% ungroup() %>% filter((min(fst))==fst) %>% .[['pop1']]` and `r intra_pol_max_fst %>% ungroup() %>% filter(min(fst)==fst) %>% .[['pop2']]`, with a value of `r intra_pol_max_fst %>% ungroup() %>% filter(min(fst)==fst) %>% .[['fst']] %>% three_dp()` at `r as.character(intra_pol_max_fst %>% ungroup() %>% filter(min(fst)==fst) %>% .[['chrom']])`:`r as.character(intra_pol_max_fst %>% ungroup() %>% filter(min(fst)==fst) %>% .[['chrom_start']])`-`r as.character(intra_pol_max_fst %>% ungroup() %>% filter(min(fst)==fst) %>% .[['chrom_end']])`.

<!-- > summarise the number of genes that had a significant hit by population -->
<!-- ie NZM | obesity = 30/200 | urate = 2/30 | t2d  -->


<!-- > interpreting pop diff in terms of drift and selection - Cabe1994 -->

<!-- need fst at a windowed level -->

```{r, results = 'asis', eval = FALSE}
kable(fst, booktabs = TRUE, caption  = '(tab:Fst) F~st~') %>%
  kableExtra::row_spec(0, angle = 90) %>% kableExtra::landscape() %>% 
  kable_styling(latex_options = c("scale_down"))

#xtable::print.xtable(xtable::xtable(fst, caption = "(tab:Fst) Fst"),floating = TRUE,  rotate.colnames = TRUE, floating.environment = "sidewaystable" , scalebox = 0.5, include.rownames = FALSE, comment = FALSE, booktabs = TRUE)
```


```{r Fst, eval = FALSE}
fst %>% mutate_at(., .vars =  vars(-Population), .funs= function(x){three_dp(as.numeric(x))}) %>%  rename_at(., .vars = vars(-Population), .funs = function(x){substring(x, 5, 8)} ) %>% mutate(Population = substring(Population, 5, 8)) %>%  knitr::kable(., booktabs = TRUE, caption  = '(\\#tab:Fst) $F_{ST}$', format = 'latex') %>% kableExtra::kable_styling(latex_options = 'scale_down') %>% 
  #kableExtra::row_spec(0, angle = 90) %>%  
  kableExtra::add_header_above(., c("" , "African" = 7, "American" = 4, "East Asian" = 5, "European" = 6, "Polynesian" = 4, "South Asian" = 5)) %>% 
  kableExtra::group_rows("AFR", 1, 7) %>%
  kableExtra::group_rows("AMR", 8, 11) %>% 
  kableExtra::group_rows("EAS", 12,16) %>% 
  kableExtra::group_rows("EUR", 17,22) %>% 
  kableExtra::group_rows("POL",23, 26)%>% 
  kableExtra::group_rows("SAS", 27, 31) %>% kableExtra::landscape()
```



```{r, eval = FALSE}
# regions of max fst between populations
pol_fst %>% gather("pop",'fst', -contains('chrom'))%>% filter(!is.na(fst)) %>% separate(pop, c("p1", "p2"), "_") %>% left_join(., panel %>% select(p1 = pop, super_pop) %>% distinct(), by = 'p1') %>%  group_by(p1, p2) %>%  filter(max(fst)  == fst) %>% arrange(fst)

# number of regions that have fst = 0 between the pairs of populations
pol_fst %>% gather("pop",'fst', -contains('chrom'))%>% filter(!is.na(fst)) %>% separate(pop, c("p1", "p2"), "_") %>% left_join(., panel %>% select(p1 = pop, super_pop) %>% distinct(), by = 'p1') %>%  group_by(p1, p2) %>%  filter(min(fst)  == fst) %>% tally() %>% arrange(n)
```


supplemental table 5 @1KGP2015snp

> In this section look at regions that are highly differentiated
what are the regions that differentiate the polynesians from the rest of the populations
what are the regions that differentiate the ep from wp
any regions that are population specific?


regions that differentiated the Eastern Polynesian from the Western Polynesian




#### XP-EHH



<!--
homozygosity and hetersozygosity
correlation with histone marks/DNAse hypersensitivity
look at selected regions/snps and expression in GTex
-->




look at the number of urate genes that were significant for the polynesian populations
```{r}
xpehh %>%  select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% filter(pop1 %in% c("CIM","NZM","SAM","TON"), xpehh_value > 0) %>% filter(genename %in% urate_goutGR$SYMBOL) %>% group_by(pop1,pop2, genename) %>%tally() %>% spread(pop1, n, fill = "") %>% left_join(., panel %>% select(pop2 = pop, super_pop) %>% distinct(), by = "pop2" ) %>% select(super_pop, everything()) %>% arrange(genename,super_pop,pop2) %>% kable(., caption = 'urate genes significant in Polynesians', longtable =  TRUE)
```

```{r xpehh urate, results = 'asis', eval = FALSE}
xpehh %>%  select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% filter(pop1 %in% c("CIM","NZM","SAM","TON")) %>% filter(genename %in% urate_goutGR$SYMBOL) %>% select(pop1,pop2, genename) %>% distinct() %>% group_by(pop1, genename)%>% tally() %>%  spread(pop1, n, fill = "") %>% kable(., caption = 'number of populations with significant xpehh for urate/gout genes', booktabs = TRUE, longtable = TRUE)
```



```{r, eval = FALSE, include = FALSE}
# DOESN'T WORK THE WAY I WANT IT TO

# all genes that have both significanct results in nsl/ihs and xpehh for Pol popululations
# calculate the mean number of sig SNPs for EP and WP by gene then look at the diff
xpehh %>% select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% 
  filter(pop1 %in% c("CIM","NZM","SAM","TON"), xpehh_value > 0) %>% 
  select(pop = pop1,pop2, genename) %>% distinct() %>% filter(genename == 'COL22A1') %>%   group_by(pop, genename)%>% 
  tally() %>% 
  mutate(ep = ifelse(pop %in% c("CIM","NZM"), TRUE, FALSE)) %>% 
  group_by(ep, genename) %>% summarise(m = mean(n, na.rm = TRUE)) %>% 
  mutate(pol = ifelse(ep == TRUE, "EP","WP")) %>%
  ungroup() %>%  
  select(-ep) %>% 
  spread(pol, value = m, fill = 0) %>% 
  mutate(diff = abs(EP - WP)) %>% 
  filter(diff > quantile(diff, 0.75))%>% 
  arrange(desc(diff)) %>% head() %>% 
  kable(., caption = 'Difference between Eastern and Western Polynesian populations XP-EHH - All genes', longtable = TRUE, booktabs = TRUE)
```


```{r}
# all genes that have both significanct results in nsl/ihs and xpehh for Pol popululations
# calculate the mean number of sig SNPs for EP and WP by gene then look at the diff
xpehh %>%  
  select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% 
  filter(pop1 %in% c("CIM","NZM","SAM","TON"), xpehh_value > 0) %>% 
  select(pop = pop1,pop2, genename) %>% 
  distinct() %>% 
  group_by(pop, genename)%>% tally() %>% 
  inner_join(., bind_rows(sig_ihs_val, sig_nsl_val) %>% 
               select(pop,genename) %>% 
               distinct(), by = c('pop','genename')) %>%  
  spread(pop, n, fill = 0) %>% 
  mutate(EP = (CIM+ NZM)/2, WP = (SAM + TON)/2, diff = abs( (CIM+ NZM)/2 - (SAM + TON)/2  ))%>%
  filter(diff > quantile(diff, 0.75)) %>% 
  arrange(desc(diff)) %>% filter(EP == 0 | WP == 0) %>% head() %>% 
  kable(., caption = 'Difference between Eastern and Western Polynesian populations with significance in iHS or nSL, and XP-EHH - All genes', longtable = TRUE, booktabs = TRUE)
```

The gene that had the biggest difference in between the East and West Polynesians in terms of the number of populations 

COL22A1 (https://www.ncbi.nlm.nih.gov/pubmed/20222955), which is has been associated with serum creatinine levels


```{r}
xpehh %>%  
  select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% 
  filter(pop1 %in% c("CIM","NZM","SAM","TON"), xpehh_value > 0) %>% 
  filter(genename %in% c(urate_goutGR$SYMBOL, t2d_GR$SYMBOL, obesity_GR$SYMBOL, metsyn_GR$SYMBOL)) %>% 
  select(pop = pop1,pop2, genename) %>% 
  distinct() %>% 
  group_by(pop, genename)%>% tally() %>% 
  inner_join(., bind_rows(sig_ihs_val, sig_nsl_val) %>%
               select(pop,genename) %>% 
               distinct(), by = c('pop','genename')) %>%  
  spread(pop, n, fill = 0) %>% 
  mutate(diff = abs( (CIM+ NZM)/2 - (SAM + TON)/2  ))%>% 
  #filter(diff > quantile(diff, 0.25)) %>% 
  kable(., caption = '(\\#ref:xpehhPopEpWp) number of populations with significant xpehh and significant ihs/nsl', longtable = TRUE, booktabs = TRUE)
```

SAM didn't have PTPRD, LPP
TON didn't have ERRB4

SAM only had TP53INP
TON only for FHIT, JAZF1, NRXN3

From table \@ref(tab:xpehhPopEpWp) in western polynesian populations there were 

- IGFR

Eastern Polynesian populations

- ADCY9
- ARL15
- SSR1



```{r}
xpehh %>%  
  select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% 
  filter(pop1 %in% poly_pop, xpehh_value > 0) %>% 
  filter(genename %in% c(urate_goutGR$SYMBOL, t2d_GR$SYMBOL, obesity_GR$SYMBOL, metsyn_GR$SYMBOL)) %>% 
  select(pop = pop1,pop2, genename) %>% 
  distinct() %>% 
  group_by(pop, genename)%>% tally() %>% 
  inner_join(., bind_rows(sig_ihs_val, sig_nsl_val) %>% 
               select(pop,genename) %>% distinct(), by = c('pop','genename')) %>% 
  spread(pop, n, fill = "") %>% 
  anti_join(., bind_rows(sig_ihs_val, sig_nsl_val) %>% 
              select(pop, genename) %>% 
              filter(!pop %in% poly_pop) %>% 
              distinct() %>% 
              select(genename), by = "genename") %>% 
  kable(., caption = 'number of populations with significant xpehh, and significant ihs/nsl only in Polynesian', longtable = TRUE, booktabs = TRUE)
```
> Genes associated with gout/metsyn/t2d/obesity/kd with significant hit in ihs/nsl in only pol pops

<!--
XXX CHECK XXX

When using the significant loci of \gls{ihs} and \gls{nsl} as a filter on the significant xpehh loci, there were 2 genes that had a significant ihs or nsl value in only Polynesian populations populations that were also significant for XPEHH with Polynesian populations. These genes were _ATP8B2_ and _IFNGR1_.

The \gls{t2d} associated gene, _ATP8B2_, had 4 markers that were significant: rs6702754, rs11586379, rs2988721, and rs200305485. With rs6702754 being the marker with the most number of significant results for _ATP8B2_. Rs200305485 was only significant between the Western and Eastern Polynesian populations. The \gls{ton} population had a significant \gls{xpehh} with the Eastern Polynesian populations of \gls{cim} and \gls{nzm}, and also with all of the \gls{eas} populations of \gls{cdx}, \gls{chb}, \gls{chs}, \gls{khv}, \gls{jpt}, and the \gls{sas} population of \gls{beb}. The \gls{sam} population only had a significant \gls{xpehh} with a subset of the populations that \gls{ton} did. These were \gls{nzm}, \gls{chs}, and \gls{cdx}.  

_IFNGR1_ was from the obesity associated gene list, and only the \gls{nzm} population had both a significant \gls{ihs} and \gls{nsl}, and significant \gls{xpehh}. There were two markers that were significant for \gls{xpehh} with \gls{nzm} and \gls{asw}, rs1887415 and rs9376267.

XXXXX
-->
```{r, eval = FALSE}
bind_rows(sig_ihs_val, sig_nsl_val) %>% filter(genename %in%  c("ADCY9", "CCNJL","EYA2","LY86","PSMD6","TP53INP1")) %>% select(pop, genename, statid, statvalue) %>% arrange(genename)
```


\clearpage

## Selection in Polynesian Populations

There were numerous genes that were associated with urate and co-morbidities, with a large number having at least a single window or \gls{snp} meeting a threshold for a single statistic.  Genes reported were selected based on having a significant haplotypic result in multiple Polynesian populations, or having multiple statistics meeting the significance threshold in at least a single Polynesian population. Another criteria was the locus only having main significant result in Polynesian populations, with minimal signal in other populations.


<!-- - the mean number of genes that were in significant windows for a population was... -->
<!-- - the mean number of genes that were only found in a single population was... -->
<!-- - the mean number of regions that were significant was... -->

```{r, eval = FALSE}
lower_sig_stats %>% filter(pop %in% poly_pop) %>% select(pop, SYMBOL) %>% bind_rows(., bind_rows(sig_ihs_val, sig_nsl_val) %>% filter(pop %in% poly_pop) %>% select(pop, "SYMBOL"= genename)) %>% distinct() %>% NROW()

lower_sig_stats %>% select(pop, SYMBOL) %>% bind_rows(., bind_rows(sig_ihs_val, sig_nsl_val) %>% select(pop, "SYMBOL"= genename)) %>% distinct() %>% group_by(pop) %>% tally() %>% ungroup() %>% summarise(mean(n))
```





<!-- Table \@ref(tab:sfsWindowsTable) shows the total number of windows for each of the Polynesian populations and the number of regions that met the significance thresholds in the upper and lower tails of the distribution. The number of genes relates to the number of unique genes that intersected with a window that had met the significance threshold. The number of population genes shows the number of genes that intersected the regions meeting the significance threshold but were only significant for in the respective population. -->
```{r upsetPol, fig.caption = 'POL Upset Plot', eval = FALSE}
UpSetR::upset(fromList(list(`Tajima's D` = lower_sig_stats %>% filter(pop %in% poly_pop, statname == 'Tajima.D', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(), 
                            `Fay and Wu's H` = lower_sig_stats %>% filter(pop %in% poly_pop, statname == 'Fay.Wu.H', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(), 
                            `Fu and Li's D` = lower_sig_stats %>% filter(pop %in% poly_pop, statname == 'Fu.Li.D', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(), 
                            `Fu and Li's F` = lower_sig_stats %>% filter(pop %in% poly_pop, statname == 'Fu.Li.F', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(),
                            `Zeng's E` = lower_sig_stats %>% filter(pop %in% poly_pop, statname == 'Zeng.E', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(),
                            iHS = sig_ihs_val %>% filter( pop %in% poly_pop, !is.na(genename) & genename != "") %>% .[['genename']] %>% unique(), 
                            nSL =sig_nsl_val %>% filter(pop %in% poly_pop, !is.na(genename) & genename != "") %>% .[['genename']] %>% unique())), 
              order.by = c("freq","degree"), nset = 7, number.angles = 90, set_size.angles = 90, hjust = -0.25, vjust = 0.5, nintersects = NA)
```


```{r, warning = FALSE, message = FALSE}
# create the 
sfs_windows_table <- bind_rows(lapply(c("Tajima.D","Fay.Wu.H","Fu.Li.D","Fu.Li.F","Zeng.E"), function(s){
  full_join(bind_rows(lapply(c("CIM","NZM","SAM","TON"), function(p){
    data.frame(pop = p,stat = s, tail= "Lower",
               window = lower_sig_stats %>% filter(statname == s, pop == p) %>% select(contains('chrom')) %>% distinct() %>% NROW(),
               indep_regions = lower_sig_stats %>% filter(statname == s, pop == p) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length(), 
               sig_genes = lower_sig_stats %>% filter(statname == s, pop == p) %>% select('SYMBOL') %>% distinct() %>% NROW(), 
               pop_only_sig_genes = lower_sig_stats %>% filter(statname == s, pop == p) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == s, pop != p), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW())
  })),
  bind_rows(lapply(c("CIM","NZM","SAM","TON"), function(p){
    data.frame(pop = p,stat = s, tail= "Upper",
               window = upper_sig_stats %>% filter(statname == s, pop == p) %>% select(contains('chrom')) %>% distinct() %>% NROW(),
               indep_regions = upper_sig_stats %>% filter(statname == s, pop == p) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length(), 
               sig_genes = upper_sig_stats %>% filter(statname == s, pop == p) %>% select('SYMBOL') %>% distinct() %>% NROW(), 
               pop_only_sig_genes = upper_sig_stats %>% filter(statname == s, pop == p) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == s, pop != p), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW())
  })), by = "pop")
})
) 

```


```{r sfsWindowsTable, results = 'asis'}
sfs_windows_table %>% 
  select(-tail.x, -tail.y, -stat.x, -stat.y) %>% 
  knitr::kable(., format = 'latex', col.names = c("Pop",rep(c("Windows","Num. Regions","Num. Genes","Num. Pop. Genes"),2)), booktab = TRUE, caption = "(\\#tab:sfsWindowsTable) Number of significant regions and genes in Polynesian populations") %>% 
  kableExtra::kable_styling(latex_options = 'scale_down') %>%
  kableExtra::add_header_above(., c("" ,"Lower Tail" = 4,"Upper Tail" = 4)) %>% 
  kableExtra::group_rows("Tajima's D", 1,4) %>%
  kableExtra::group_rows("Fay and Wu's H",5 ,8) %>%
  kableExtra::group_rows("Fu and Li's D", 9, 12) %>%
  kableExtra::group_rows("Fu and Li's F", 13, 16) %>%
  kableExtra::group_rows("Zeng's E", 17, 20)
```
<!--

This section is about reporting how a population went, maybe add in a general comparison of the 1000 genomes populations, eg average number of genes specific to a population

The general format that I want for this section is to get:
- the genes that were reported in multiple stats as significant
- genes that were only reported in a particular population
- runs of depressed regions (eg 7/10 windows) that were significant in a row
- pathway/ontology terms


-->



### Polynesian super population



```{r superDiseaseGenesTab}
bind_rows(
lower_sig_stats %>% select(SYMBOL, pop) %>% left_join(., panel %>% select(pop, super_pop) %>% distinct(), by = 'pop') %>% 
  mutate(
    urate = ifelse(SYMBOL %in% urate_goutGR$SYMBOL, TRUE, FALSE),
    obesity = ifelse(SYMBOL %in% obesity_GR$SYMBOL, TRUE, FALSE),
    t2d = ifelse(SYMBOL %in% t2d_GR$SYMBOL, TRUE, FALSE),
    kd = ifelse(SYMBOL %in% kd_GR$SYMBOL, TRUE, FALSE),
    metsyn = ifelse(SYMBOL %in% metsyn_GR$SYMBOL, TRUE, FALSE)
    ) %>% 
  select(SYMBOL, pop, super_pop, urate, obesity, t2d, kd, metsyn) %>% 
  distinct() %>% 
  group_by(pop, super_pop) %>% 
  summarise(urate = sum(urate), obesity = sum(obesity), t2d = sum(t2d), kd = sum(kd), metsyn = sum(metsyn)) %>% 
  mutate(type = "SFS") %>% arrange(super_pop),


bind_rows(sig_ihs_val, sig_nsl_val) %>% 
  select(pop, 'SYMBOL' = genename) %>% 
   left_join(., panel %>% select(pop, super_pop) %>% distinct(), by = 'pop') %>% 
  mutate(
    urate = ifelse(SYMBOL %in% urate_goutGR$SYMBOL, TRUE, FALSE),
    obesity = ifelse(SYMBOL %in% obesity_GR$SYMBOL, TRUE, FALSE),
    t2d = ifelse(SYMBOL %in% t2d_GR$SYMBOL, TRUE, FALSE),
    kd = ifelse(SYMBOL %in% kd_GR$SYMBOL, TRUE, FALSE),
    metsyn = ifelse(SYMBOL %in% metsyn_GR$SYMBOL, TRUE, FALSE)
    ) %>% 
  select(SYMBOL, pop, super_pop, urate, obesity, t2d, kd, metsyn) %>% 
  distinct() %>% 
  group_by(pop, super_pop) %>% 
  summarise(urate = sum(urate), obesity = sum(obesity), t2d = sum(t2d), kd = sum(kd), metsyn = sum(metsyn)) %>% 
 mutate(type = 'Haplotype') %>% arrange(super_pop),

bind_rows(
  bind_rows(sig_ihs_val, sig_nsl_val) %>% 
    select(pop, 'SYMBOL' = genename),
          lower_sig_stats %>% select(SYMBOL, pop)
  ) %>% 
   left_join(., panel %>% select(pop, super_pop) %>% distinct(), by = 'pop') %>%
  mutate(
    urate = ifelse(SYMBOL %in% urate_goutGR$SYMBOL, TRUE, FALSE),
    obesity = ifelse(SYMBOL %in% obesity_GR$SYMBOL, TRUE, FALSE),
    t2d = ifelse(SYMBOL %in% t2d_GR$SYMBOL, TRUE, FALSE),
    kd = ifelse(SYMBOL %in% kd_GR$SYMBOL, TRUE, FALSE),
    metsyn = ifelse(SYMBOL %in% metsyn_GR$SYMBOL, TRUE, FALSE)
    ) %>% 
  select(SYMBOL, pop, super_pop, urate, obesity, t2d, kd, metsyn) %>% 
  distinct() %>% 
  group_by(pop, super_pop) %>%
  summarise(urate = sum(urate), obesity = sum(obesity), t2d = sum(t2d), kd = sum(kd), metsyn = sum(metsyn)) %>% 
   mutate(type = 'combined') %>% arrange(super_pop)
) %>% 
  kable(., col.names = c('Population',"Super Pop",'Urate / Gout','Obesity','T2D','Kidney Disease','Metabolic Syndrome','Type'), caption = '(\\#tab:superDiseaseGenesTab) Number of Genes with evidence of possible positive selection in Super populations')
```


(ref:polyUpset) Upset plot showing the number of genes that have evidence from \gls{ihs} or \gls{nsl}, and any of \gls{td}, \gls{fwh}, \gls{fld}, \gls{flf}, or \gls{ze}, in the four Polynesian populations.

```{r polyUpset, fig.cap = '(ref:polyUpset)'}

# show the intersection of the genes that were deemed significant between the different polynesian populations
UpSetR::upset(fromList(list(CIM = extract_sig("CIM"), 
                            NZM = extract_sig("NZM"), 
                            SAM = extract_sig("SAM"), 
                            TON = extract_sig("TON"))),
                             order.by = "freq", nset = 4, number.angles = 90, set_size.angles = 90, hjust = -0.25, vjust = 0.5, nintersects = NA)
```

There were 218 genes that were in common for the Eastern Polynesian populations that had at least a one marker that was significant in the haplotypic tests, and one window that intersected the same gene that was in the 1st percentile for a frequency based statistic. There was about 50% more genes for the same criteria in the Western Polynesians, with 302 genes. The \gls{nzm} population had the most genes (395) that were not shared with any of the other Polynesian populations.
```{r svgcimupset, eval =FALSE}
svg('images/03_selectionpipeline/poly_upset.svg')
UpSetR::upset(fromList(list(CIM = extract_sig("CIM"), 
                            NZM = extract_sig("NZM"), 
                            SAM = extract_sig("SAM"), 
                            TON = extract_sig("TON"))),
                             order.by = "freq", nset = 4, number.angles = 90, set_size.angles = 90, hjust = -0.25, vjust = 0.5, nintersects = NA)
dev.off()
```

```{r, eval = FALSE}
tmp_cim <- extract_sig("CIM")
tmp_nzm <- extract_sig("NZM")
tmp_sam <- extract_sig("SAM")
tmp_ton <- extract_sig("TON")

tmp_cim[!tmp_cim %in% c(tmp_nzm, tmp_sam, tmp_ton)] %>% cat()
tmp_nzm[!tmp_nzm %in% c(tmp_cim, tmp_sam, tmp_ton)] %>% cat()
tmp_sam[!tmp_sam %in% c(tmp_cim, tmp_nzm, tmp_ton)] %>% cat()
tmp_ton[!tmp_ton %in% c(tmp_cim, tmp_nzm, tmp_sam)] %>% cat()


rm(tmp_cim, tmp_nzm, tmp_sam, tmp_ton)
```




(ref:superUpset) Upset plot showing the intersections of genes that had both haplotypic and SFS evidence at the individaul population leveland were then pooled into their super population group.

```{r superUpset, fig.cap = '(ref:superUpset)', eval = TRUE}

p <- panel %>% mutate(pop2 = paste0(super_pop,'_',pop)) %>% select(pop,pop2) %>% filter(!pop %in% c('WPN','EPN','POL','NAD')) %>% distinct()
pop_genes <- lapply(p$pop, extract_sig)
names(pop_genes) <- p$pop2

superpop_genes <- lapply(c('AFR','AMR','EUR','EAS', 'SAS','POL'), extract_sig_super)
names(superpop_genes) <- c('AFR','AMR','EUR','EAS', 'SAS','POL')

# create an upset plot for the intersection of genes between the super populations
# the key is that each individual population first is 'filtered' for genes that have both haplotypic and sfs significance
UpSetR::upset(fromList(superpop_genes),
                             order.by = c("freq","degree"), nset = length(names(superpop_genes)), number.angles = 90, set_size.angles = 90, hjust = -0.25, vjust = 0.5, nintersects = NA)
```


Figure \@ref(fig:superUpset) shows the intersections of the genes that had significant markers for either \gls{ihs} or \gls{nsl}

Markers that had a significant \gls{ihs} 


Polynesian specific are summarised by 

There were `r sig_ihs_val %>% anti_join(., sig_ihs_val %>% filter(!pop %in% poly_pop), by = c('genename')) %>% select(pop, genename) %>% distinct() %>% group_by(genename) %>% tally() %>% filter( n == 4) %>% NROW()` genes that all Polynesian populations and no other super populations had a \gls{snp} with a significant \gls{ihs} marker, and there were `r sig_ihs_val %>% anti_join(., sig_ihs_val %>% filter(!pop %in% poly_pop), by = c('genename')) %>% select(pop, genename) %>% distinct() %>% group_by(genename) %>% tally() %>% filter(n > 0) %>% NROW()` genes that at least one Polynesian population and no other super populations had. The Eastern Polynesian populations had `r sig_ihs_val %>% anti_join(., sig_ihs_val %>% filter(!(pop %in% c("SAM","TON"))), by = c('genename')) %>% select(pop, genename) %>% distinct() %>% group_by(genename) %>% tally() %>% filter(n > 0) %>% NROW()` genes, whereas the Western Polynesian populations had `r sig_ihs_val %>% anti_join(., sig_ihs_val %>% filter( !(pop %in% c("CIM","NZM"))), by = c('genename')) %>% select(pop, genename) %>% distinct() %>% group_by(genename) %>% tally() %>% filter(n > 0) %>% NROW()` genes.

This compared to the significant \gls{nsl} markers where there were `r sig_nsl_val %>% anti_join(., sig_nsl_val %>% filter(!pop %in% poly_pop), by = c('genename'))  %>% select(pop, genename) %>% distinct() %>% group_by(genename) %>% tally() %>% filter( n == 4) %>% NROW()` genes, that had at least one marker, and were found in only all four Polynesian populations. Within the only Eastern Polynesian populations there were `r sig_nsl_val %>% anti_join(., sig_nsl_val %>% filter(!pop %in% c("SAM","TON")), by = c('genename')) %>% select(pop, genename) %>% distinct() %>% group_by(genename) %>% tally() %>% filter(n > 0) %>% NROW()` genes that had a significant marker. Within the Western Polynesian populations there were `r sig_nsl_val %>% anti_join(., sig_nsl_val %>% filter(!pop %in% c("CIM","NZM")), by = c('genename')) %>% select(pop, genename) %>% distinct() %>% group_by(genename) %>% tally() %>% filter(n > 0) %>% NROW()` genes only significant in that group. When the intersection of Polynesian specific genes for both \gls{ihs} and \gls{nsl} was compared, the number of genes in common was `r inner_join(sig_ihs_val %>% anti_join(., sig_ihs_val %>% filter(!pop %in% poly_pop), by = c('genename')) %>% filter(pop %in% poly_pop) %>% select(pop, genename) %>% distinct() %>% group_by(genename) %>% tally() %>% filter( n == 4), sig_nsl_val %>% anti_join(., sig_nsl_val %>% filter(!pop %in% poly_pop), by = c('genename')) %>% filter(pop %in% poly_pop) %>% select(pop, genename) %>% distinct() %>% group_by(genename) %>% tally() %>% filter( n == 4), by = 'genename') %>% NROW()` that had at least 1 \gls{snp} in only the four Polynesian populations. These were _GRIP1_ and _LINC00661_. 

```{r, eva = FALSE}
# find the genes that were in all 4 poly pops and in both ihs and nsl
inner_join(sig_ihs_val %>% anti_join(., sig_ihs_val %>% filter(!pop %in% poly_pop), by = c('genename')) %>% filter(pop %in% poly_pop) %>% select(pop, genename) %>% distinct() %>% group_by(genename) %>% tally() %>% filter( n == 4), sig_nsl_val %>% anti_join(., sig_nsl_val %>% filter(!pop %in% poly_pop), by = c('genename')) %>% filter(pop %in% poly_pop) %>% select(pop, genename) %>% distinct() %>% group_by(genename) %>% tally() %>% filter( n == 4), by = 'genename')
```

```{r}
# use anti join to remove all gene names that are found in non-polynesisan populations and then find all genes that had all poly
# ie returns genes that were ONLY significant in all 4 Polynesian populations 
sig_ihs_val %>% anti_join(., sig_ihs_val %>% filter(!pop %in% poly_pop), by = c('genename')) %>% filter(pop %in% poly_pop) %>% select(pop, genename) %>% distinct() %>% group_by(genename) %>% tally() %>% filter( n == 4)
```

The number of genes that were shared between all of the Polynesian populations was `r length(intersect(intersect(extract_sig("CIM"), extract_sig("NZM")), intersect(extract_sig("SAM"), extract_sig("TON"))))` (Figure \@ref(polyUpset)). The number of genes in common between the Eastern Polynesian populations was `r length(intersect(extract_sig("CIM"), extract_sig("NZM")))` and between the Western Polynesian populations was `r length(intersect(extract_sig("SAM"), extract_sig("TON")))`.

As shown in table \@ref(tab:sfsWindowsTable), there were similar numbers of genes that intersected the windows that met the significance threshold in either tail between the Polynesian populations of \gls{cim}, \gls{nzm}, \gls{sam}, and \gls{ton}. 







```{r, eval = FALSE}
# sfs sig genes for cim with count of all sfs stats
lower_sig_stats %>% filter(pop == 'CIM') %>% select(-contains("chrom"), -pop) %>% distinct() %>% spread(statname, statname) %>% left_join(., lower_sig_stats %>% filter(pop == 'CIM') %>% select(-contains("chrom"), -pop) %>% distinct() %>% group_by(SYMBOL) %>% tally(), by = "SYMBOL")
```

```{r, eval = FALSE}
# sfs lower significant plus ihs/nsl
lower_sig_stats %>% filter(pop == 'CIM') %>% select(-contains("chrom"), -pop) %>% bind_rows(., sig_ihs_val %>% filter(pop == "CIM") %>% select(SYMBOL = genename)%>% mutate(statname = 'ihs'), sig_nsl_val %>% filter(pop == "CIM") %>% select(SYMBOL = genename)%>% mutate(statname = 'nsl')) %>% distinct() %>% spread(statname, statname) %>% left_join(., lower_sig_stats %>% filter(pop == 'CIM') %>% select(-contains("chrom"), -pop) %>% bind_rows(., sig_ihs_val %>% filter(pop == "CIM") %>% select(SYMBOL = genename)%>% mutate(statname = 'ihs'), sig_nsl_val %>% filter(pop == "CIM") %>% select(SYMBOL = genename)%>% mutate(statname = 'nsl')) %>% distinct() %>% group_by(SYMBOL) %>% tally(), by = "SYMBOL")
```




The previously reported PPARGC1A and CREBRF

No significant snps in poly for PPARGC1A for \gls{ihs} or \gls{nsl}


in the 1st percentile there were no windows for PPARGC1A

the 99th percentile had

fld windows for NZM, SAM, and TON

CREBRF

no polynesian populations had any windows, 1st or 99th and there were not significant snps for \gls{ihs} or \gls{nsl}


```{r markdown3, echo = FALSE, eval = FALSE}
_Tajima's D_

With the populations that made up the \gls{pol} super population there were `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower tail significance threshold, these windows formed `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions totaling `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. These regions contained `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM','NZM','SAM','TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Tajima.D', !pop %in% c('CIM','NZM','SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant with the \gls{pol} populations with \gls{td}. In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that reached significance and made `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions covering a total of `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. There were `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes with `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM','NZM','SAM','TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Tajima.D', !pop %in% c('CIM','NZM','SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` that were only found in the \gls{pol} populations for \gls{td} within these regions.

_Fay and Wu's H_

Within all of the Polynesian populations used, there were  `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM','NZM','SAM','TON')) %>% NROW()` windows that met the significance threshold in the lower tail. These windows formed `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions, and were at total of `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. There were `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM','NZM','SAM','TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes that intersected these regions and `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM','NZM','SAM','TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fay.Wu.H', !pop %in% c('CIM','NZM','SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes were only significant in the \gls{pol} populations for \gls{fwh}.

For the upper tail there were  `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM','NZM','SAM','TON')) %>% NROW()` windows that were reduced to `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions totaling `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp that met the threshold for significance. In these regions there were `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM','NZM','SAM','TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes and `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM','NZM','SAM','TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fay.Wu.H', !pop %in% c('CIM','NZM','SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes were only significant in the \gls{pol} populations in \gls{fwh}.

_Fu and Li's D_

With the populations that made up the \gls{pol} super population there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower tail significance threshold, these windows formed `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions totaling `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. These regions contained `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM','NZM','SAM','TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.D', !pop %in% c('CIM','NZM','SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant with the \gls{pol} populations with \gls{fld}. In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that reached significance and made `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions covering a total of `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. There were `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes with `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM','NZM','SAM','TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.D', !pop %in% c('CIM','NZM','SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` that were only found in the \gls{pol} populations for \gls{fld} within these regions.

_Fu and Li's F_

With the populations that made up the \gls{pol} super population there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower tail significance threshold, these windows formed `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions totaling `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. These regions contained `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM','NZM','SAM','TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM','NZM','SAM','TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.F', !pop %in% c('CIM','NZM','SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant with the \gls{pol} populations with \gls{flf}. In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that reached significance and made `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions covering a total of `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. There were `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM','NZM','SAM','TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes with `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM','NZM','SAM','TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.F', !pop %in% c('CIM','NZM','SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` that were only found in the \gls{pol} populations for \gls{flf} within these regions.

_Zeng's E_

With the populations that made up the \gls{pol} super population there were `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower tail significance threshold, these windows formed `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions totaling `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. These regions contained `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM','NZM','SAM','TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM','NZM','SAM','TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Zeng.E', !pop %in% c('CIM','NZM','SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant with the \gls{pol} populations with \gls{ze}. In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that reached significance and made `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions covering a total of `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. There were `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM','NZM','SAM','TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes with `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM','NZM','SAM','TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Zeng.E', !pop %in% c('CIM','NZM','SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` that were only found in the \gls{pol} populations for \gls{ze} within these regions.
```


A complete list of all of the significant genes for the Polynesian populations from the lower tail can be found in tables \@ref(tab:lowersigtdPol), \@ref(tab:lowersigfwhPol), \@ref(tab:lowersigfldPol), \@ref(tab:lowersigflfPol), and \@ref(tab:lowersigzePol)


```{r createGenetable}
create_gene_table <- function(popname, genelist){
  list(
  td = lower_sig_stats %>% filter(pop == popname, statname == "Tajima.D", SYMBOL %in% genelist) %>% select(chrom, SYMBOL, statname, pop) %>% group_by(SYMBOL, statname) %>% tally(),
  fwh = lower_sig_stats %>% filter(pop == popname, statname == "Fay.Wu.H", SYMBOL %in% genelist)%>% select(chrom, SYMBOL, statname, pop) %>% group_by(SYMBOL, statname) %>% tally(),
  fld = lower_sig_stats %>% filter(pop == popname, statname == "Fu.Li.D", SYMBOL %in% genelist)%>% select(chrom, SYMBOL, statname, pop) %>% group_by(SYMBOL, statname) %>% tally(),
  flf = lower_sig_stats %>% filter(pop == popname, statname == "Fu.Li.F", SYMBOL %in% genelist)%>% select(chrom, SYMBOL, statname, pop) %>% group_by(SYMBOL, statname) %>% tally(),
  ze = lower_sig_stats %>% filter(pop == popname, statname == "Zeng.E", SYMBOL %in% genelist)%>% select(chrom, SYMBOL, statname, pop) %>% group_by(SYMBOL, statname) %>% tally(),
  ihs = sig_ihs_val %>% filter(pop == popname, genename %in% genelist)%>% select(chrom, 'SYMBOL' = genename)  %>% mutate(pop = popname, statname = 'ihs') %>% group_by(SYMBOL, statname) %>% tally(), 
  nsl = sig_nsl_val %>% filter(pop == popname, genename %in% genelist)%>% select(chrom, 'SYMBOL' = genename)  %>% mutate(pop = popname, statname = 'nsl') %>% group_by(SYMBOL,statname) %>% tally(),
  xpehh = c(),#xpehh %>% filter(pop1 == popname, genename %in% genelist, xpehh_value > 0) %>% select(chrom, 'SYMBOL' = genename) %>% mutate(pop = popname, statname = 'xpehh') %>% group_by(SYMBOL, statname) %>% tally(),
  fst = c()
)
}
```



```{r polySigMetGenes}
#table to record all regions that have evidence of postitive selection in urate for CIM
cim_urate_sig <- create_gene_table("CIM", urate_goutGR$SYMBOL)
cim_t2d_sig <- create_gene_table("CIM", t2d_GR$SYMBOL)
cim_obesity_sig <- create_gene_table("CIM", obesity_GR$SYMBOL)
cim_kd_sig <- create_gene_table("CIM", kd_GR$SYMBOL)
cim_metsyn_sig <- create_gene_table("CIM", metsyn_GR$SYMBOL)

nzm_urate_sig <- create_gene_table("NZM", urate_goutGR$SYMBOL)
nzm_t2d_sig <- create_gene_table("NZM", t2d_GR$SYMBOL)
nzm_obesity_sig <- create_gene_table("NZM", obesity_GR$SYMBOL)
nzm_kd_sig <- create_gene_table("NZM", kd_GR$SYMBOL)
nzm_metsyn_sig <- create_gene_table("NZM", metsyn_GR$SYMBOL)

sam_urate_sig <- create_gene_table("SAM", urate_goutGR$SYMBOL)
sam_t2d_sig <- create_gene_table("SAM", t2d_GR$SYMBOL)
sam_obesity_sig <- create_gene_table("SAM", obesity_GR$SYMBOL)
sam_kd_sig <- create_gene_table("SAM", kd_GR$SYMBOL)
sam_metsyn_sig <- create_gene_table("SAM", metsyn_GR$SYMBOL)


ton_urate_sig <- create_gene_table("TON", urate_goutGR$SYMBOL)
ton_t2d_sig <- create_gene_table("TON", t2d_GR$SYMBOL)
ton_obesity_sig <- create_gene_table("TON", obesity_GR$SYMBOL)
ton_kd_sig <- create_gene_table("TON", kd_GR$SYMBOL)
ton_metsyn_sig <- create_gene_table("TON", metsyn_GR$SYMBOL)
```

(ref:intraSelGenes) Genes associated with urate, gout, obesity, \gls{t2d}, kidney disease, and metabolic syndrome having both significance in intra-population SFS and haplotypic based methods in at least one Polynesian population. Numbers inside boxes indicate the number of significant windows (SFS) or SNPs (haplotypic methods) for the population.

```{r intraSelGenes, fig.cap = '(ref:intraSelGenes)', fig.height = 7}
# genes that have at least one poly population with significance in both haplotypic and sfs stats
rbind(
bind_rows(cim_obesity_sig, cim_urate_sig, cim_t2d_sig, cim_metsyn_sig, cim_kd_sig) %>% mutate(pop = 'CIM') %>% distinct(),
bind_rows(nzm_obesity_sig, nzm_urate_sig, nzm_t2d_sig, nzm_metsyn_sig, nzm_kd_sig) %>% mutate(pop = 'NZM') %>% distinct(),
bind_rows(sam_obesity_sig, sam_urate_sig, sam_t2d_sig, sam_metsyn_sig, sam_kd_sig) %>% mutate(pop = "SAM") %>% distinct(),
bind_rows(ton_obesity_sig, ton_urate_sig, ton_t2d_sig, ton_metsyn_sig, ton_kd_sig) %>% mutate(pop = 'TON') %>% distinct()
)  %>% spread(statname, n) %>% filter(SYMBOL %in% (mutate(., haplo = ifelse(!is.na(ihs) | !is.na(nsl), 1, 0), sfs = ifelse(!is.na(Fay.Wu.H) | !is.na(Fu.Li.D) | !is.na(Fu.Li.F) | !is.na(Tajima.D) | !is.na(Zeng.E), 1, 0)) %>% 
  filter(haplo & sfs) %>% .[["SYMBOL"]] %>% unique() %>% as.vector())) %>% 
  gather(key = statname, value=n, 3:NCOL(.)) %>% 
  mutate(sn = statname) %>% 
  mutate(statname = case_when(statname == "Fay.Wu.H" ~ "Fay and Wu's H", 
                              statname == "Tajima.D" ~ "Tajima's D", 
                              statname == 'Fu.Li.D'~ "Fu and Li's D", 
                              statname == 'Fu.Li.F' ~ "Fu and Li's F",
                              statname == 'Zeng.E' ~ "Zeng's E", 
                              statname == 'ihs' ~ 'iHS', 
                              statname == 'nsl'~"nSL")) %>% 
  mutate(statname = factor(statname, levels = c("Fay and Wu's H", "Tajima's D", "Fu and Li's D", "Fu and Li's F", "Zeng's E", "iHS","nSL"), ordered = TRUE)) %>%  
  filter(!is.na(n)) %>% ggplot(., aes(x = pop, y = SYMBOL))  + geom_tile(aes(colour = pop, fill = pop)) + theme_bw() + facet_grid(~statname) + geom_text(inherit.aes = TRUE, aes(label = n)) + theme(panel.grid = element_blank(), axis.text.x = element_text(angle = 90, vjust = 0.5), legend.position = 'bottom') + xlab('Population') + scale_fill_discrete(name = "Population") + guides(colour = FALSE) + scale_y_discrete(expand = c(0,0)) + scale_x_discrete(expand = c(0,0)) 
```


```{r, results = 'asis'}
kegg_sig %>% filter(Adjusted.P.value < 0.05) %>% select(Term, pop ) %>% mutate(pop = toupper(pop) ,yes = "Yes") %>% distinct() %>% spread(pop, yes, "") %>% mutate(term2 = sapply(Term, function(x){strsplit(x, "_")[[1]][1]})) %>% select(-Term) %>% select( term2, everything()) %>% kable(., col.names = c("Pathway Term","CIM","NZM","SAM", "TON"), format = 'latex', booktabs = TRUE, caption = '(\\#tab:keggPath) Pathway terms that were significant in Polynesian populations')
```



\FloatBarrier

### Cook Island Maori

```{r cim_specific, warning = FALSE}
cim_0.1_genes_lower <- lower_sig_stats %>% left_join(., lower_sig_stats_binned, by = c('chrom','chrom_start','chrom_end','statname', 'pop')) %>% filter(quant == 0.1, pop == 'CIM') %>% select(statname, SYMBOL) %>% distinct() %>% group_by(statname) %>%  tally()

cim_0.1_stats_lower <- lower_sig_stats %>% left_join(., lower_sig_stats_binned, by = c('chrom','chrom_start','chrom_end','statname', 'pop')) %>% filter(quant == 0.1, pop == 'CIM') %>% group_by(statname) %>%  tally()

cim_99.9_genes_upper <- upper_sig_stats %>% left_join(., upper_sig_stats_binned, by = c('chrom','chrom_start','chrom_end','statname', 'pop')) %>% filter(quant == 99.9, pop == 'CIM') %>% select(statname, SYMBOL) %>% distinct() %>% group_by(statname) %>%  tally()

cim_99.9_stats_upper <- upper_sig_stats %>% left_join(., upper_sig_stats_binned, by = c('chrom','chrom_start','chrom_end','statname', 'pop')) %>% filter(quant == 99.9, pop == 'CIM') %>% group_by(statname) %>%  tally()
```
> intro statment(s)

#### Pathway Analysis

Functional classification of genes with markers that were significant for \gls{ihs} or \gls{nsl}, or genes that fell within the windows of the 1st percentile and had a value less than 0 for the frequency spectrum based statistics was done using PantherDB. Only \gls{td} had the main biological process as "Binding", and the second as "Catalytic activity". All other statistics each had "Cellular process" and "Metabolic process" as their first and second largest categories. There were no biological processes that were over represented from the gene list of all genes with possible positive selection.

Pathway gene-set enrichment analysis on gene lists for each statistic that met the significance thresholds was done using Enrichr. There was a total of 10 pathways that after multiple testing correction had significant p-values. Nine pathways were significant for the gene list from \gls{nsl} and had a heart and lung focus based on calcium ion movement in the form of genes for calcium transporters and voltage gate calcium channels. There were five genes related to calcium that were in at least five of the pathway terms, they were: CACNA1D, ADCY9, SLC8A1, CACNA2D2, CACNA2D3. There was a single pathway that was significant from the genes from the 1st percentile of \gls{td} which was "bacterial invasion of epithelial cells".




#### Consecutive regions

<!-- There were `r consec_lower_regions %>% filter(stat == 'Tajima.D', pop == 'CIM') %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce(ignore.strand = TRUE) %>% length()` regions of consecutive score depression for \gls{td}, `r consec_lower_regions %>% filter(stat == 'Fay.Wu.H', pop == 'CIM') %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce(ignore.strand = TRUE) %>% length()` for \gls{fwh}, `r consec_lower_regions %>% filter(stat == 'Fu.Li.D', pop == 'CIM') %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce(ignore.strand = TRUE) %>% length()` for \gls{fld}, `r consec_lower_regions %>% filter(stat == 'Fu.Li.F', pop == 'CIM') %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce(ignore.strand = TRUE) %>% length()` for \gls{flf}, and `r consec_lower_regions %>% filter(stat == 'Zeng.E', pop == 'CIM') %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce(ignore.strand = TRUE) %>% length()` for \gls{ze}. -->

Looking at regions of consecutively depressed score regions that also had either a significant \gls{ihs} or \gls{nsl} marker in them, there was a total of five regions that met this criteria. For \gls{td},  there was two regions, the first, a 290 kb region at chr2:24215002-24505001 covered _FAM228B_ and had a single significant marker for both \gls{ihs} and \gls{nsl}. The second,  was 470 kb at chr11:68105002-68575001, and included _LRP5_ and had a single significant marker for \gls{ihs}. \gls{flf} also had a region that covered LRP5 but at chr11:68085002-68355001 and was 270 kb long. For both \gls{flf} and \gls{fld}, there was a 250 kb region at chr16:3545002-3795001 that covered _DNASE1_ with a single marker for \gls{ihs} and \gls{nsl}, and also _TRAP1_ with two markers each for \gls{ihs} and \gls{nsl}. \gls{fwh} had a single 250 kb region that covered MYT1L at chr2:2015002-2265001 and had four significant markers for \gls{ihs} and three for \gls{nsl}. 

There were three regions of consecutively depressed score that intersected obesity associated genes. There was a 240 kb region at chr3:52375002-52615001 that intersected _NT5DC2_ and _PBRM1_ for \gls{fld}. The 250 kb region of chr16:3545002-3795001 intersected  _NLRC3_ for both \gls{fld} and \gls{flf}. And the 310 kb region of chr12:124195002-124505001 intersected _DNAH10_, _ZNF664_, and _CCDC92._ for \gls{flf}. There was a single 280 kb region that intersected the metabolic syndrome associated gene of EDC4 at chr16:67835002-68115001 for \gls{td}. None of the regions that were had consecutively depressed score that intersected genes associated with urate, \gls{t2d}, or kidney disease.

haplotypic results intersecting regions
```{r, eval = FALSE}
# find ihs/nsl intersecting the regions of consecutively depressed score
consec_lower_regions %>% filter( pop == 'CIM') %>% inner_join(., bind_rows(sig_ihs_val, sig_nsl_val) %>% select(SYMBOL = genename, everything()), by = c('pop','SYMBOL')) %>% select(-contains('gene'))
```

<!-- - td -->
<!--     - FAM228B - ihs/nsl chr2:24215002-24505001 w=290000 -->
<!--     - LRP5 - ihs chr11:68105002-68575001 w=470000  -->
<!-- - fwh -->
<!--     - MYT1L 4x ihs, 3x nsl chr2:2015002-2265001 w=250000 -->
<!-- - flf -->
<!--     - LRP5 chr11:68085002-68355001 w=270000 -->

<!-- - chr16:3545002-3795001 w=250000 -->
<!--   - both fld and flf -->
<!--   - DNASE1 ihs/nsl -->
<!--   - TRAP1 2x ihs, 2x nsl -->

#### Haplotypic


Within the most extreme 100 \gls{ihs} values, there were 37 genes. For \gls{nsl} there was 45 genes. Eighteen of these genes were in common, with C11orf65, CNTN4, ADAM29, and LINC00276 having at least 5 significant snps each. The gene with the most significant snps out of any population was _CNTN4_ in \gls{cim} with 27 markers with a significant \gls{ihs} score, the only other population with more than two \gls{snp}s was \gls{nzm} with 14. There were four genes represented in the top 100 markers that were associated with obesity or \gls{t2d}, these were: _ADCY9_, _ARL15_, _PSMD6_, and _ZFAND6_. In total there were  `r sig_ihs_val %>% filter(pop == "CIM") %>% select(genename) %>% anti_join(., sig_ihs_val %>% filter(pop != "CIM") %>% select(genename), by = 'genename') %>% distinct() %>% NROW()` genes that were only significant in \gls{cim} for \gls{ihs}, and `r sig_nsl_val %>% filter(pop == "CIM") %>% select(genename) %>% anti_join(., sig_nsl_val %>% filter(pop !="CIM") %>% select(genename), by = 'genename') %>% distinct() %>% NROW()` for \gls{nsl}.


<!-- ihs C11orf65 (6), ATM (5) CNTN4 (5),  -->
<!-- ADCY9 (obesity), ARL15 (obesity, t2d), PSMD6 (t2d), ZFAND6 (t2d) -->


```{r, eval=FALSE}
p <- "CIM"

(cim_nsl_100 <- sig_nsl_val %>% filter(pop == p) %>% arrange(desc(abs(statvalue))) %>% head(n=100) %>% group_by(genename) %>% tally() %>% arrange(desc(n)))

(cim_ihs_100 <- sig_ihs_val %>% filter(pop == p) %>% arrange(desc(abs(statvalue))) %>% head(n=100) %>% group_by(genename) %>% tally() %>% arrange(desc(n)))

inner_join(cim_nsl_100, cim_ihs_100, by = 'genename' ) %>% mutate(total = n.x + n.y) %>% arrange(desc(total))
rm(p)
```

#### XPEHH

#### FST

<!-- #### Notes -->

<!-- xpehh total number of genes that were significant  CIM versus: -->

<!-- - NZM `r xpehh %>% select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% filter(pop1 == 'CIM', pop2 == "NZM", !is.na(genename)) %>%  group_by(pop1,pop2,genename) %>% tally()  %>% NROW()` -->
<!-- - SAM `r xpehh %>% select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% filter(pop1 == 'CIM', pop2 == "SAM",  !is.na(genename)) %>%  group_by(pop1,pop2,genename) %>% tally() %>%  NROW()` -->
<!-- - TON `r xpehh %>% select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% filter(pop1 == 'CIM', pop2 == "TON",  !is.na(genename)) %>%  group_by(pop1,pop2,genename) %>% tally()  %>% NROW()` -->


<!-- xpehh number of genes that were significant in only CIM versus: -->

<!-- - NZM `r xpehh %>% select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% filter(pop1 == 'CIM', pop2 == "NZM", !is.na(genename)) %>%  group_by(pop1,pop2,genename) %>% tally() %>% anti_join(., xpehh %>% select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% filter(pop1 == 'CIM', pop2 != "NZM")%>%  group_by(pop1,pop2,genename) %>% tally(), by = c('genename')) %>% NROW()` -->
<!-- - SAM `r xpehh %>% select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% filter(pop1 == 'CIM', pop2 == "SAM",  !is.na(genename)) %>%  group_by(pop1,pop2,genename) %>% tally() %>% anti_join(., xpehh %>% select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% filter(pop1 == 'CIM', pop2 != "SAM")%>%  group_by(pop1,pop2,genename) %>% tally(), by = c('genename')) %>% NROW()` -->
<!-- - TON `r xpehh %>% select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% filter(pop1 == 'CIM', pop2 == "TON",  !is.na(genename)) %>%  group_by(pop1,pop2,genename) %>% tally() %>% anti_join(., xpehh %>% select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% filter(pop1 == 'CIM', pop2 != "TON")%>%  group_by(pop1,pop2,genename) %>% tally(), by = c('genename')) %>% NROW()` -->
<!-- - -->

<!-- > BLK was previously identified in Jha2015 and had sig TD and ihs in CIM. The east asians have both td and ihs, and sam/ton just have td. BLK is associated with Systemic lupus erythematosus (SLE) in Asian populations Delgado-Vega2012 -->


<!-- (ref:CIMupset) Upset plot showing the number of \gls{cim} genes that had a marker or window that met the significance threshold for the statistic. -->

```{r CIMupset, fig.cap = '(ref:CIMupset)', eval = FALSE}
UpSetR::upset(fromList(list(`Tajima's D` = lower_sig_stats %>% filter(pop == 'CIM', statname == 'Tajima.D', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(), 
                            `Fay and Wu's H` = lower_sig_stats %>% filter(pop == 'CIM', statname == 'Fay.Wu.H', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(), 
                            `Fu and Li's D` = lower_sig_stats %>% filter(pop == 'CIM', statname == 'Fu.Li.D', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(), 
                            `Fu and Li's F` = lower_sig_stats %>% filter(pop == 'CIM', statname == 'Fu.Li.F', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(),
                            `Zeng's E` = lower_sig_stats %>% filter(pop == 'CIM', statname == 'Zeng.E', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(),
                            iHS = sig_ihs_val %>% filter(pop == 'CIM', !is.na(genename) & genename != "") %>% .[['genename']] %>% unique(), 
                            nSL =sig_nsl_val %>% filter(pop == 'CIM', !is.na(genename) & genename != "") %>% .[['genename']] %>% unique())), 
              order.by = c("freq",'degree'), nset = 7, number.angles = 90, set_size.angles = 90, hjust = -0.25, vjust = 0.5, nintersects = NA)
```

```{r upsetcimsvg, include = FALSE, eval = FALSE}
svg(filename = 'images/03_selectionpipeline/cim_upset.svg')
  UpSetR::upset(fromList(list(`Tajima's D` = lower_sig_stats %>% filter(pop == 'CIM', statname == 'Tajima.D', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(), 
                            `Fay and Wu's H` = lower_sig_stats %>% filter(pop == 'CIM', statname == 'Fay.Wu.H', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(), 
                            `Fu and Li's D` = lower_sig_stats %>% filter(pop == 'CIM', statname == 'Fu.Li.D', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(), 
                            `Fu and Li's F` = lower_sig_stats %>% filter(pop == 'CIM', statname == 'Fu.Li.F', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(),
                            `Zeng's E` = lower_sig_stats %>% filter(pop == 'CIM', statname == 'Zeng.E', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(),
                            iHS = sig_ihs_val %>% filter(pop == 'CIM', !is.na(genename) & genename != "") %>% .[['genename']] %>% unique(), 
                            nSL =sig_nsl_val %>% filter(pop == 'CIM', !is.na(genename) & genename != "") %>% .[['genename']] %>% unique())), 
              order.by = c("freq",'degree'), nset = 7, number.angles = 90, set_size.angles = 90, hjust = -0.25, vjust = 0.5, nintersects = NA)
  dev.off()
```

```{r cim_urate_results, results = 'asis', eval = FALSE}
# look at all significant stats from cim for urate genes
bind_rows(cim_urate_sig)%>% spread(statname,n, fill = "" ) %>% arrange() %>% kable(., caption = '(\\#tab:cimUrate) CIM genes with evidence of selection in urate')
consec_lower_regions %>% filter(pop == 'CIM', SYMBOL %in% urate_goutGR$SYMBOL)

# obesity genes
bind_rows(cim_obesity_sig)%>% spread(statname,n, fill = "" ) %>% arrange() %>% kable(., caption = '(\\#tab:cimObesity) CIM genes with evidence of selection in obesity')
consec_lower_regions %>% filter(pop =='CIM', SYMBOL %in% obesity_GR$SYMBOL) %>% mutate(chrom = substring(seqnames, 4, nchar(seqnames)))%>% arrange(stat, chrom, start)

# T2D
bind_rows(cim_t2d_sig)%>% spread(statname,n, fill = "" ) %>% arrange() %>% kable(., caption = '(\\#tab:cimT2d) CIM genes with evidence of selection in T2D')
consec_lower_regions %>% filter(pop =='CIM', SYMBOL %in% t2d_GR$SYMBOL) %>% mutate(chrom = substring(seqnames, 4, nchar(seqnames)))%>% arrange(stat, chrom, start)

# KD
bind_rows(cim_kd_sig)%>% spread(statname,n, fill = "" ) %>% arrange() %>% kable(., caption = '(\\#tab:cimKd) CIM genes with evidence of selection in KD')
consec_lower_regions %>% filter(pop =='CIM', SYMBOL %in% kd_GR$SYMBOL) %>% mutate(chrom = substring(seqnames, 4, nchar(seqnames)))%>% arrange(stat, chrom, start)

# Metsyn
bind_rows(cim_metsyn_sig)%>% spread(statname,n, fill = "" ) %>% arrange() %>% kable(., caption = '(\\#tab:cimMetsyn) CIM genes with evidence of selection in MetSyn')
consec_lower_regions %>% filter(pop =='CIM', SYMBOL %in% metsyn_GR$SYMBOL) %>% mutate(chrom = substring(seqnames, 4, nchar(seqnames)))%>% arrange(stat, chrom, start)
```

<!-- obesity -->

<!-- ihs/nsl only: ADAMTS9, ADCY9, CCNJL, GDF5, MSRA, NAV1 -->


<!-- T2d: -->

<!-- ADAMTS9, LPP, PSMD6, SSR1, ZFAND6 had both ihs and nsl -->

<!-- ADAM30, NOTCH2 had only fld and flf -->

<!-- THADA had fwh, fld, flf, ze -->

<!-- ARL15, PTPRD had ihs, nsl and td -->

<!-- consecutive regions: -->
<!-- THADA 140kb fwh -->
<!-- ADAM30/NOTCH2 180kb for fld and flf -->


<!-- The \gls{cim} population had similar numbers of windows that were significant in both tails of the distribution compared to the other Polynesian populations but slightly more genes intersected these windows, and were also only significant in \gls{cim} for the lower tail. The upper tail had slightly fewer genes compared to the other Polynesian populations. -->

<!-- In the lower tail there were  `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes for \gls{td},  `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` for \gls{fwh},  `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` for \gls{fld},  `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` for \gls{flf}, and  `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` for \gls{ze}, that intersected windows that met the significance threshold. There were only two genes - _FANCM_ and _RGS3_ - that had windows that met the significance threshold for each of \gls{td}, \gls{fwh}, \gls{fld}, \gls{flf}, and \gls{ze}. There were `r lower_sig_stats %>% filter(pop == 'CIM') %>% select(-contains("chrom"), -pop) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% arrange(desc(n)) %>% filter(n == 4) %>% NROW()` genes that met the threshold in four of the selection statistics, `r lower_sig_stats %>% filter(pop == 'CIM') %>% select(-contains("chrom"), -pop) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% arrange(desc(n)) %>% filter(n == 3) %>% NROW()`  in three selection statistics, `r lower_sig_stats %>% filter(pop == 'CIM') %>% select(-contains("chrom"), -pop) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% arrange(desc(n)) %>% filter(n == 2) %>% NROW()` in two selection statistics, and `r lower_sig_stats %>% filter(pop == 'CIM') %>% select(-contains("chrom"), -pop) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% arrange(desc(n)) %>% filter(n == 1) %>% NROW()` with only one selection statistic meeting the threshold for the lower tail. -->

<!-- \Gls{cim} also had the most independent consecutive regions for \gls{td} and \gls{fwh} out of the Polynesian populations. -->

<!-- the gene with the most number of significant markers in any population for \gls{ihs} was CNTN4, with 27 significant markers in \gls{cim}. Of the 13 other populations that had a significant marker with \gls{ihs}, all but \gls{nzm} which had 14, had no more than 2 markers that were significant. Out of the 27 markers for \gls{cim} 13 were in common exclusively with \gls{nzm}, only 2 markers,  rs2728076 - shared with \gls{cdx} and \gls{chs}, and rs955340 - shared with \gls{clm}, -->

<!-- The largest \gls{ihs} values came from C11orf65, LINC00276, CPQ, SNAP25-AS1 -->

<!-- The largest \gls{nsl} values came from ADAM29, SLC35F2, PGR, and MIR146A -->

<!-- The smallest values for \gls{td} came from an eight window consecutive region at `r lower_sig_stats %>% left_join(., lower_sig_stats_binned) %>% filter(pop == 'CIM',statname == 'Tajima.D') %>% arrange(value) %>% head(n=20) %>% filter(chrom == 'chr22') %>% GRanges() %>% GenomicRanges::reduce() %>% data.frame() %>% mutate(window = paste0(seqnames,':',start,'-',end)) %>% .[['window']]` which covered the following genes TTC38, GTSE1-AS1, GTSE1, CDPF1, PKDREJ, and PPARA.     -->


<!-- The largest region of Fay and Wu's H of windows in the 0.1 percentile was at `r lower_sig_stats %>% left_join(., lower_sig_stats_binned) %>% filter(pop == 'CIM',statname == 'Fay.Wu.H', quant == 0.1) %>% arrange(value) %>% GRanges() %>% GenomicRanges::reduce() %>% data.frame() %>% mutate(window = paste0(seqnames,':',start,'-',end)) %>% arrange(desc(width)) %>% .[['window']] %>% .[1]` and was 80kb wide. In this region was ADH4, ADH5, PCNAP1, and LOC100507053. -->

<!-- The smallest five \gls{fld} values came from windows that covered TCTN1, HVCN1, DOCK5, DNAH10, CALML4, CLN6, ATP1A1, ATP1A1-AS1, ZNF704, and EFCAB5. -->

<!-- \gls{flf} with the smallest values came from windows that covered TCTN1, HVCN1, DOCK5, ATP1A1-AS1, ATP1A1, and ZNF704 -->

<!-- \gls{ze} had ZNF704, LYST, SH3RF1, UQCRC1 from the smallest valued windows. From the 0.1 percentile there was a significant result for the KEGG pathway analysis -->


<!-- There were only significant results for 3 loci that were associated with urate and gout. _ATXN2_ had one window that was significant with \gls{ze}. _RREB1_ had 3 windows that were significant for \gls{td}. _BCAS3_ had significant windows for \gls{fld} (5), \gls{flf} (4), \gls{td} (2), and \gls{fwh} (8). The \gls{fwh} windows also met the criteria to be classed as a consecutively depressed score region spanning 110 Kb. There were no intra-population haplotype based statistics that had a significant value. For \gls{xpehh} there were several loci that were significant, _NRXN2_ with the \gls{eur} populations. _TRIM46_ with the Han Chinese (\gls{chb} and \gls{chs}) and the Japanese (\gls{jpt}). _RREB1_ was significant between \gls{cim} and nearly all populations part of the \gls{afr}, \gls{amr}, \gls{eas}, and \gls{eur} super populations, and only a single population (\gls{pjl}) from the \gls{sas} super population. -->

```{r cimSigRegions, eval = FALSE}
# copied into Panther, then exported the biological process
# also copied into Enrichr and exported the KEGG2016 pathways table

#td
lower_sig_stats %>% filter(pop == 'CIM', statname == 'Tajima.D', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique() %>% cat()
#fwh
lower_sig_stats %>% filter(pop == 'CIM', statname == 'Fay.Wu.H', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique() %>% cat()
#fld
lower_sig_stats %>% filter(pop == 'CIM', statname == 'Fu.Li.D', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique() %>% cat()
#flf
lower_sig_stats %>% filter(pop == 'CIM', statname == 'Fu.Li.F', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique() %>% cat()
#ze
lower_sig_stats %>% filter(pop == 'CIM', statname == 'Zeng.E', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique() %>% cat()
#ihs
sig_ihs_val %>% filter(pop == 'CIM', !is.na(genename)) %>% .[['genename']] %>% unique() %>% cat()
#nsl
sig_nsl_val %>% filter(pop == 'CIM', !is.na(genename)) %>% .[['genename']] %>% unique() %>% cat()

# all with at least 1 sfs and 1 haplotypic
bind_rows(bind_rows(sig_ihs_val, sig_nsl_val) %>% filter(pop == 'CIM')%>% select(pop, "SYMBOL" = genename ) %>% distinct(), lower_sig_stats %>% filter(pop == 'CIM')%>% select(pop, SYMBOL) %>% distinct()) %>% filter(!is.na(SYMBOL))  %>% group_by(SYMBOL) %>% tally() %>% filter(n == 2) %>% .[['SYMBOL']] %>% cat()
```



<!-- - TD main processes: "binding", "catalytic activity" -->
<!-- - FWH main processes: "Cellular process", "metabolic process" -->
<!-- - FLD main processes: "Cellular process", "metabolic process" -->
<!-- - FLF main processes: "Cellular process", "metabolic process" -->
<!-- - ZE main processes: "Cellular process", "metabolic process" -->
<!-- - ihs main processes: "Cellular process", "metabolic process" -->
<!-- - nsl main processes: "Cellular process", "metabolic process" -->
<!-- - all main processes: "Cellular process", "metabolic process" -->
<!-- - all overrepresentation test: nothing -->



<!-- > Genes in the 0.1 percentile for the sfs based stats -->

<!-- The 0.1 percentile had a total of `r cim_0.1_stats_lower %>% filter(statname == 'Tajima.D') %>% .[['n']]` regions and `r cim_0.1_genes_lower %>% filter(statname == 'Tajima.D') %>% .[['n']]` genes for \gls{td}.  -->


```{r, eval = FALSE}
# look at the lower 0.1% of windows
lower_sig_stats_binned %>% filter(pop == 'CIM',quant == 0.1) %>% left_join(.,lower_sig_stats, by = c('statname','chrom','chrom_start','chrom_end','pop')) %>% select(statname, SYMBOL) %>% group_by(statname, SYMBOL) %>% distinct()%>% tally() %>% arrange(statname, SYMBOL)  %>% spread(statname, n) %>% mutate(rs = select(., 2:NCOL(.)) %>% rowSums(., na.rm=TRUE)) %>% filter(rs > 3)

lower_sig_stats_binned %>% filter(pop == 'CIM',quant == 0.1) %>% left_join(.,lower_sig_stats, by = c('statname','chrom','chrom_start','chrom_end','pop')) %>% filter(!is.na(SYMBOL))%>% group_by(statname, SYMBOL) %>% tally()%>% arrange(statname, SYMBOL)  %>% spread(statname, n) %>% filter(!is.na(Zeng.E) )
```

<!-- In the 0.1st percentile there were 84 genes for \gls{td}, 56 for \gls{fwh}, 99 for \gls{ze}, and 59 in common between \gls{fld} and \gls{flf}. There were three genes DNAH1, RGS3, and ZNF704 that were in the 0.1st percentile for each of \gls{fld}, \gls{flf}, \gls{td}, and \gls{ze}  -->



```{r, fig.cap='(ref:cimFANCM)', eval= FALSE, message = FALSE, fig.height = 4, fig.width = 4}
#plot_gene("CIM","FANCM")
#ggsave(last_plot(), file = 'images/03_selectionpipeline/cimFANCM.png')
knitr::include_graphics('images/03_selectionpipeline/cimFANCM.png')
```


```{r, fig.cap='(ref:cimRGS3)', eval= FALSE, message = FALSE, fig.height = 4, fig.width = 4}
#plot_gene("CIM","RGS3")
#ggsave(last_plot(), file = 'images/03_selectionpipeline/cimRGS3.png')
knitr::include_graphics('images/03_selectionpipeline/cimRGS3.png')
```



```{r, eval = FALSE}
tmp_td <- lower_sig_stats %>% filter(pop == "CIM", statname == "Tajima.D") %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce(., ignore.strand = TRUE)
tmp_td$stat <- "td"

tmp_fwh <- lower_sig_stats %>% filter(pop == "CIM", statname == "Fay.Wu.H") %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce(., ignore.strand = TRUE)
tmp_fwh$stat <- "fwh"

tmp <- IRanges::slice(GenomicRanges::coverage(c(tmp_td, tmp_fwh))[[4]], 2)
tmp@ranges
```

<!-- There were fewer genes that intersected the windows that met the threshold for significance in the upper tail for each of \gls{td} (`r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()`), \gls{fwh} (`r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()`), \gls{fld} (`r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()`), \gls{flf} (`r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()`), and \gls{ze} (`r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()`) compared with the lower tail. In the upper tail there was only a single gene - _CSMD1_ - that had windows that met the significance threshold in \gls{td}, \gls{fwh}, \gls{fld}, \gls{flf}, and \gls{ze}. There were  `r upper_sig_stats %>% filter(pop == 'CIM') %>% select(-contains("chrom"), -pop) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% arrange(desc(n)) %>% filter(n == 4) %>% NROW()` genes that met the threshold with four of the selection statistics,  `r upper_sig_stats %>% filter(pop == 'CIM') %>% select(-contains("chrom"), -pop) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% arrange(desc(n)) %>% filter(n == 3) %>% NROW()` genes that met the threshold in three selection statistics,  `r upper_sig_stats %>% filter(pop == 'CIM') %>% select(-contains("chrom"), -pop) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% arrange(desc(n)) %>% filter(n == 2) %>% NROW()` genes with two selection statistics, and  `r upper_sig_stats %>% filter(pop == 'CIM') %>% select(-contains("chrom"), -pop) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% arrange(desc(n)) %>% filter(n == 1) %>% NROW()` genes with only one selection statistic that met the upper tail significance threshold. -->

<!-- \gls{cim} had the most independent regions for \gls{flf} and \gls{ze}, and the most genes that were population specific, out of the the Polynesian populations. -->

<!-- - fwh most pop specific -->
<!-- - flf most regions -->
<!-- - ze most regions -->

<!-- Looking specifically at the genes that were associated with gout and other metabolic diseases, of the \gls{pol} populations, the \gls{cim} population had the fewest genes that were associated with gout and urate that showed possible evidence of positive selection. This was true also for obesity, \gls{t2d}, kidney disease, and metabolic syndrome associated genes (Table \@ref(tab:polDiseaseGenesTab). -->


```{r, eval = FALSE}
bind_rows(sig_ihs_val %>% mutate(statname = 'ihs'), sig_nsl_val %>% mutate(statname = 'nsl')) %>% filter(pop == 'CIM', genename %in% urate_goutGR$SYMBOL) %>% group_by(genename, statname) %>% tally() %>% arrange(desc(n))

bind_rows(sig_ihs_val %>% mutate(statname = 'ihs'), sig_nsl_val %>% mutate(statname = 'nsl')) %>% filter(pop == 'CIM', genename %in% t2d_GR$SYMBOL) %>% group_by(genename, statname) %>% tally() %>% arrange(desc(n))

bind_rows(sig_ihs_val %>% mutate(statname = 'ihs'), sig_nsl_val %>% mutate(statname = 'nsl')) %>% filter(pop == 'CIM', genename %in% obesity_GR$SYMBOL) %>% group_by(genename, statname) %>% tally() %>% arrange(desc(n))

bind_rows(sig_ihs_val %>% mutate(statname = 'ihs'), sig_nsl_val %>% mutate(statname = 'nsl')) %>% filter(pop == 'CIM', genename %in% metsyn_GR$SYMBOL) %>% group_by(genename, statname) %>% tally() %>% arrange(desc(n))
```





```{r, echo = FALSE, eval = FALSE}
_Tajima's D_

The \gls{cim} had a total of `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the significance threshold in the lower tail. This represented `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` non-continuous regions representing `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp of th genome. Within these windows were `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which only `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Tajima.D', !pop %in% c('CIM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes were distinct for \gls{cim} out of the \gls{td} results. In the upper tail there were  `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the upper significance threshold, these windows formed `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM')) %>% select(contains('chrom'))  %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` non-continuous regions covering `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. These regions contained `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Tajima.D', !pop %in% c('CIM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only from \gls{cim} for \gls{td}.

_Fay and Wu's H_

There were `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the significance threshold in the lower tail. These windows made up `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions with a total of  `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. Within these regions were `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fay.Wu.H', !pop %in% c('CIM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant in the \gls{cim} population for \gls{fwh}

In the upper tail there were  `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that me the significance threshold, consisting of `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions that totaled `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. There were  `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes within these regions and `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fay.Wu.H', !pop %in% c('CIM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes were only significant in \gls{cim} for \gls{fwh}.

_Fu and Li's D_

For the \gls{cim} population there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower tail threshold for significance and made `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions covering `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. In these regions there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which  `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.D', !pop %in% c('CIM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant in \gls{cim} for \gls{fld}. In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the significance threshold. These windows made `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions with a total of `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp containing `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, with  `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.D', !pop %in% c('CIM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only meeting significance in \gls{cim} for \gls{fld}.

_Fu and Li's F_

For the \gls{cim} population there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower tail threshold for significance and made `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions covering `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. In these regions there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which  `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.F', !pop %in% c('CIM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant in \gls{cim} for \gls{flf}. In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the significance threshold. These windows made `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions with a total of `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp containing `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, with  `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.F', !pop %in% c('CIM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only meeting significance in \gls{cim} for \gls{flf}.

_Zeng's E_

For the \gls{cim} population there were `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower tail threshold for significance and made `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions covering `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. In these regions there were `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which  `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Zeng.E', !pop %in% c('CIM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant in \gls{cim} for \gls{ze}. In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the significance threshold. These windows made `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions with a total of `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp containing `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, with  `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Zeng.E', !pop %in% c('CIM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only meeting significance in \gls{cim} for \gls{ze}.
```


<!-- - The upper tail had `r consec_upper_regions %>% filter(stat == 'Tajima.D', pop == 'CIM') %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce(ignore.strand = TRUE) %>% length()` regions of extended significant windows for \gls{td}, `r consec_upper_regions %>% filter(stat == 'Fay.Wu.H', pop == 'CIM')%>% GenomicRanges::GRanges() %>% GenomicRanges::reduce(ignore.strand = TRUE) %>% length()` for \gls{fwh}, `r consec_upper_regions %>% filter(stat == 'Fu.Li.D', pop == 'CIM') %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce(ignore.strand = TRUE) %>% length()` for \gls{fld}, `r consec_upper_regions %>% filter(stat == 'Fu.Li.F', pop == 'CIM')%>% GenomicRanges::GRanges() %>% GenomicRanges::reduce(ignore.strand = TRUE) %>% length()` for \gls{flf}, and `r consec_upper_regions %>% filter(stat == 'Zeng.E', pop == 'CIM') %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce(ignore.strand = TRUE) %>% length()` for \gls{ze}. -->

<!-- From the haplotypic tests for selection (\gls{ihs} and \gls{nsl}) there were `r sig_ihs_val %>% filter(pop == "CIM") %>% select(genename) %>% anti_join(., sig_ihs_val %>% filter(pop !="CIM") %>% select(genename), by = 'genename') %>% distinct() %>% NROW()` genes that were only significant in \gls{cim} for \gls{ihs}, and `r sig_nsl_val %>% filter(pop == "CIM") %>% select(genename) %>% anti_join(., sig_nsl_val %>% filter(pop !="CIM") %>% select(genename), by = 'genename') %>% distinct() %>% NROW()` for \gls{nsl}. -->



were there any genes that were significant from the gene lists and related to the thrifty gene hypothesis?


### New Zealand Maori

The \gls{nzm} population had the largest number of genes intersecting windows from the 1st percentile that were only reported in a single population for \gls{fwh}, \gls{fld}, \gls{flf}, and \gls{ze} out of the Polynesian populations.


<!-- - TD main processes "cellular process", "metabolic process" -->
<!-- - FWH main processes "cellular process", "metabolic process" -->
<!-- - FLD main processes "cellular process", "metabolic process" -->
<!-- - FLF main processes "cellular process", "metabolic process" -->
<!-- - ZE main processes "cellular process", "metabolic process" -->
<!-- - ihs main processes "cellular process", "metabolic process" -->
<!-- - nsl main processes "cellular process", "metabolic process" -->
<!-- - all main processes "cellular process", "metabolic process" -->
<!-- - all overrepresentation test: cellular component movement FE=4.47, P = 4.86e-2 -->
<!--     - anatomical structure morphogenesis FE=3.94 P = 2.88e-2 -->

#### Pathway Enrichment

The gene set over representation analysis using PantherDB had 'cellular component movement' with a fold-enrichment of 4.47 (P = 0.0486) and 'anatomical structure morphogenesis' with fold enrichment of 3.94 (P = 0.0288), as the only significant annotations, when all genes that met a threshold were used. 

The pathway gene-set enrichment analysis from Enrichr on the KEGG 2016 pathways had 31 pathways that were significant after multiple testing correction, this was the largest number of pathways out of the Polynesian populations. Thirty were from the significant genes for \gls{nsl}, with many in common with \gls{cim} with calcium transporters and voltage gated calcium channels featuring in many pathways. There were more hormone based pathways that were significant, such renin secretion, oxytocin signalling pathway, adrenergic signalling in cardiomyocytes, thyroid hormone synthesis, aldosterone synthesis and secretion, insulin secretion, and GnRH signalling pathway. Further more, there were clear diabetes related pathways that were significant such as type 2 diabetes mellitus, insulin secretion, and pancreatic secretion pathways. The genes that were significant made up about 10% of the genes in the pathways. The genes that were shared in the most pathways included two adenlyate cylases (_ADCY8_ and _ADCY9_), two phospholipase C genes (_PLCB1_ and _PLCB4_), and calcium voltage-gated channel subunits (_CACNA1C_ and _CACNA1D_). Similar to \gls{cim}, there was a single pathway that was significant from the \gls{td} list and that too was for bacterial invasion of epithelial cells.

#### Haplotypic

From the haplotypic tests for selection (\gls{ihs} and \gls{nsl}) there were `r sig_ihs_val %>% filter(pop == "NZM") %>% select(genename) %>% anti_join(., sig_ihs_val %>% filter(pop != "NZM") %>% select(genename), by = 'genename') %>% distinct() %>% NROW()` genes that were only significant in \gls{nzm} for \gls{ihs}, and `r sig_nsl_val %>% filter(pop == "NZM") %>% select(genename) %>% anti_join(., sig_nsl_val %>% filter(pop != "NZM") %>% select(genename), by = 'genename') %>% distinct() %>% NROW()` for \gls{nsl}. Looking at the genes represented by the most extreme 100 \gls{ihs} and \gls{nsl} markers there were 30 genes for \gls{ihs}, and 40 genes for \gls{nsl}. There were 12 genes in common between the two sets. _CTNNA3_ had five markers in the top 100 for \gls{ihs}, and was also the gene for \gls{nzm} that had the most overall significant markers for both \gls{ihs} and \gls{nsl}. 

#### Consecutive regions

The regions of consecutive depressed score from the 1st percentile of windows  th at also contatined markers with a significant \gls{ihs} or \gls{nsl} value covered three regions of the genome for \gls{nzm}. Two of the regions were for \gls{td}, the first was 270kb at chr2:24215002-24485001, and similar to the region in \gls{cim}. It had a single significant marker for \gls{ihs} in each of _MFSD2B_, _FKBP1B_, and _FAM228B_. The second \gls{td} region was 260 kb at chr4:106555002-106815001 and had a single significant marker for \gls{nsl} in _INTS12_, and a single marker for both \gls{ihs} and \gls{nsl} in _SH3RF1._ There was an overlapping region between \gls{fld} and \gls{flf} that had five significant markers in _TTN-AS1_ for both \gls{ihs} and \gls{nsl}. The 340 kb region for \gls{fld} at chr2:179325002-179665001 contained the 200 kb region for \gls{flf} at chr2:179395002-179595001.

There were no consecutive regions that intersected with genes associated with urate, kidney disease, or \gls{t2d}. There were 2 regions that intersected obesity associated loci; for \gls{td} a 320 kb region at chr16:67145002-67465001 contatined _KCTD19_, and for \gls{flf} 220 kb at chr16:3555002-3775001 contained _NLRC3_. _EDC4_ which is associated with metabolic syndrome had slightly differing regions that covered it. \gls{td} was 330 kb from chr16:67775002-68105001, \gls{fld} was 260 kb from chr16:67785002-68045001, and \gls{flf} was 320 kb from chr16:67775002-68095001.


There were four genes that had at least one significant marker for \gls{ihs} or \gls{nsl}, and also had a window in the 1st percentile from four or more statistics from \gls{td}, \gls{fwh}, \gls{fld}, \gls{flf}, and \gls{ze}. These genes were _CTNNA3_, _RAD51AP2_, _SUPT3H_, and _VSNL1_.


#### XPEHH

#### FST

```{r, eval = FALSE}
p <- "NZM"

(nzm_nsl_100 <- sig_nsl_val %>% filter(pop == p) %>% arrange(desc(abs(statvalue))) %>% head(n=100) %>% group_by(genename) %>% tally() %>% arrange(desc(n)))

(nzm_ihs_100 <- sig_ihs_val %>% filter(pop == p) %>% arrange(desc(abs(statvalue))) %>% head(n=100) %>% group_by(genename) %>% tally() %>% arrange(desc(n)))

inner_join(nzm_nsl_100, nzm_ihs_100, by = 'genename' ) %>% mutate(total = n.x + n.y) %>% arrange(desc(total))
rm(p)
```


<!-- From all of the intra population -->

<!-- - There were `r bind_rows(nzm_urate_sig) %>% select(SYMBOL) %>% distinct() %>% NROW()` of `r length(unique(urate_goutGR$SYMBOL))` urate genes -->
<!-- - There were `r bind_rows(nzm_obesity_sig) %>% select(SYMBOL) %>% distinct() %>% NROW()` of `r length(unique(obesity_GR$SYMBOL))` obesity genes -->
<!-- - There were `r bind_rows(nzm_t2d_sig) %>% select(SYMBOL) %>% distinct() %>% NROW()` of `r length(unique(t2d_GR$SYMBOL))` \gls{t2d} genes -->
<!-- - There were `r bind_rows(nzm_kd_sig) %>% select(SYMBOL) %>% distinct() %>% NROW()` of `r length(unique(kd_GR$SYMBOL))` kidney disease genes -->
<!-- - There were `r bind_rows(nzm_metsyn_sig) %>% select(SYMBOL) %>% distinct() %>% NROW()` of `r length(unique(metsyn_GR$SYMBOL))` metsyn genes -->


```{r, eval = FALSE}
# find ihs/nsl intersecting the regions of consecutively depressed score
consec_lower_regions %>% filter( pop == 'NZM') %>% inner_join(., bind_rows(sig_ihs_val, sig_nsl_val) %>% select(SYMBOL = genename, everything()), by = c('pop','SYMBOL')) %>% select(-contains('gene'))
```

#### Notes

```{r nzmSigRegions, eval = FALSE}
# copied into Panther adn then exported the biological process

#td
lower_sig_stats %>% filter(pop == 'NZM', statname == 'Tajima.D', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique() %>% cat()
#fwh
lower_sig_stats %>% filter(pop == 'NZM', statname == 'Fay.Wu.H', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique() %>% cat()
#fld
lower_sig_stats %>% filter(pop == 'NZM', statname == 'Fu.Li.D', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique() %>% cat()
#flf
lower_sig_stats %>% filter(pop == 'NZM', statname == 'Fu.Li.F', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique() %>% cat()
#ze
lower_sig_stats %>% filter(pop == 'NZM', statname == 'Zeng.E', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique() %>% cat()
#ihs
sig_ihs_val %>% filter(pop == 'NZM', !is.na(genename)) %>% .[['genename']] %>% unique() %>% cat()
#nsl
sig_nsl_val %>% filter(pop == 'NZM', !is.na(genename)) %>% .[['genename']] %>% unique() %>% cat()

# all with at least 1 sfs and 1 haplotypic
bind_rows(bind_rows(sig_ihs_val, sig_nsl_val) %>% filter(pop == 'NZM')%>% select(pop, "SYMBOL" = genename ) %>% distinct(), lower_sig_stats %>% filter(pop == 'NZM')%>% select(pop, SYMBOL) %>% distinct()) %>% filter(!is.na(SYMBOL))  %>% group_by(SYMBOL) %>% tally() %>% filter(n == 2) %>% .[['SYMBOL']] %>% cat()
```

<!-- - TD main processes "cellular process", "metabolic process" -->
<!-- - FWH main processes "cellular process", "metabolic process" -->
<!-- - FLD main processes "cellular process", "metabolic process" -->
<!-- - FLF main processes "cellular process", "metabolic process" -->
<!-- - ZE main processes "cellular process", "metabolic process" -->
<!-- - ihs main processes "cellular process", "metabolic process" -->
<!-- - nsl main processes "cellular process", "metabolic process" -->
<!-- - all main processes "cellular process", "metabolic process" -->
<!-- - all overrepresentation test: cellular component movement FE=4.47, P = 4.86e-2 -->
<!--     - anatomical structure morphogenesis FE=3.94 P = 2.88e-2 -->

<!-- - td -->
<!--     - chr2:24215002-24485001 w=270000 -->
<!--         - MFSD2B ihs -->
<!--         - FKBP1B ihs -->
<!--         - FAM228B ihs -->
<!--     - chr4:106555002-106815001 w=260000 -->
<!--         - INTS12 nsl -->
<!--         - SH3RF1 ihs/nsl -->
<!-- -fld -->
<!--     - chr2:179325002-179665001 w=340000  -->
<!--         - TTN-AS1 5x ihs -->
<!--         - TTN 5x ihs -->
<!-- - flf -->
<!--     - chr2:179395002-179595001 w=200000 -->
<!--         - TTN-AS1 5x ihs -->
<!--         - TTN 5x ihs -->

<!-- no urate, t2d,  -->
<!-- obesity:  -->
<!-- td chr16:67145002-67465001 w=320000 KCTD19 -->
<!-- flf chr16:3555002-3775001 w=220000 NLRC3  -->

<!-- metsyn -->
<!-- td chr16:67775002-68105001 w=330000 EDC4 -->
<!-- fld chr16:67785002-68045001 w=260000 EDC4 -->
<!-- flf chr16:67775002-68095001 w=320000 EDC4 -->


<!-- (ref:upsetNZM) Upset plot for \gls{nzm} showing the numbers of genes that were unique or in common between the different intra-population statistics that met the significance threshold. -->


```{r upsetNzm, fig.cap = '(ref:upsetNZM', eval = FALSE}
UpSetR::upset(fromList(list(`Tajima's D` = lower_sig_stats %>% filter(pop == 'NZM', statname == 'Tajima.D', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(), 
                            `Fay and Wu's H` = lower_sig_stats %>% filter(pop == 'NZM', statname == 'Fay.Wu.H', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(), 
                            `Fu and Li's D` = lower_sig_stats %>% filter(pop == 'NZM', statname == 'Fu.Li.D', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(), 
                            `Fu and Li's F` = lower_sig_stats %>% filter(pop == 'NZM', statname == 'Fu.Li.F', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(),
                            `Zeng's E` = lower_sig_stats %>% filter(pop == 'NZM', statname == 'Zeng.E', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(),
                            iHS = sig_ihs_val %>% filter(pop == 'NZM', !is.na(genename) & genename != "") %>% .[['genename']] %>% unique(), 
                            nSL =sig_nsl_val %>% filter(pop == 'NZM', !is.na(genename) & genename != "") %>% .[['genename']] %>% unique())), 
              order.by = c("freq", "degree"), nsets = 7, number.angles = 90, set_size.angles = 90, hjust = -0.5, vjust = 0.5, nintersects = NA )
```

```{r svgNZMupset, include = FALSE, eval=FALSE}
svg(filename = 'images/03_selectionpipeline/nzm_upset.svg')
  UpSetR::upset(fromList(list(`Tajima's D` = lower_sig_stats %>% filter(pop == 'NZM', statname == 'Tajima.D', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(), 
                            `Fay and Wu's H` = lower_sig_stats %>% filter(pop == 'NZM', statname == 'Fay.Wu.H', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(), 
                            `Fu and Li's D` = lower_sig_stats %>% filter(pop == 'NZM', statname == 'Fu.Li.D', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(), 
                            `Fu and Li's F` = lower_sig_stats %>% filter(pop == 'NZM', statname == 'Fu.Li.F', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(),
                            `Zeng's E` = lower_sig_stats %>% filter(pop == 'NZM', statname == 'Zeng.E', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(),
                            iHS = sig_ihs_val %>% filter(pop == 'NZM', !is.na(genename) & genename != "") %>% .[['genename']] %>% unique(), 
                            nSL =sig_nsl_val %>% filter(pop == 'NZM', !is.na(genename) & genename != "") %>% .[['genename']] %>% unique())), 
              order.by = c("freq", "degree"), nsets = 7, number.angles = 90, set_size.angles = 90, hjust = -0.5, vjust = 0.5, nintersects = NA )
  dev.off()
```




<!-- - 5 stats: `r lower_sig_stats %>% filter(pop %in% c('NZM')) %>% select(SYMBOL, statname) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==5, !is.na(SYMBOL)) %>% .[['SYMBOL']]` -->
<!-- - 4 stats: `r lower_sig_stats %>% filter(pop %in% c('NZM')) %>% select(SYMBOL, statname) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==4, !is.na(SYMBOL)) %>% NROW()` -->
<!-- - 3 stats: `r lower_sig_stats %>% filter(pop %in% c('NZM')) %>% select(SYMBOL, statname) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==3, !is.na(SYMBOL)) %>% NROW()` -->
<!-- - 2 stats: `r lower_sig_stats %>% filter(pop %in% c('NZM')) %>% select(SYMBOL, statname) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==2, !is.na(SYMBOL)) %>% NROW()` -->
<!-- - 1 stats: `r lower_sig_stats %>% filter(pop %in% c('NZM')) %>% select(SYMBOL, statname) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==1, !is.na(SYMBOL)) %>% NROW()` -->





```{r nzmDNAH6, fig.cap='(ref:nzmDNAH6)', eval= FALSE, message = FALSE, fig.height = 4, fig.width = 4}
#plot_gene("NZM","DNAH6")
#ggsave(last_plot(), file = 'images/03_selectionpipeline/nzmDNAH6.png')
knitr::include_graphics('images/03_selectionpipeline/nzmDNAH6.png')
```

```{r nzmDNAH6consec, fig.cap='(ref:nzmDNAH6consec)', eval= FALSE, message = FALSE, fig.height = 4, fig.width = 4}
# g = "DNAH6"
# #retrieve information of the gene of interest
# 
# wh <- genesymbol[g]
# wh <- range(wh, ignore.strand = TRUE)
# 
# stat_plot <- bind_rows(sig_ihs_val, sig_nsl_val) %>% filter(pop == 'NZM', genename == g)%>% mutate(pop = 'NZM', statname = case_when(statid == 20 ~ "nSL", statid == 26 ~ "iHS"), chrom = paste0('chr',chrom)) %>%  select(SYMBOL = genename, chrom, chrom_start, chrom_end, statname) %>%  bind_rows(., consec_lower_regions %>% filter(pop == 'NZM', SYMBOL == g) %>% select('chrom'= seqnames, 'chrom_start' = start, 'chrom_end' = end, 'statname' = stat, SYMBOL) ) %>% GenomicRanges::GRanges() %>% ggbio::autoplot(., aes(fill = statname, colour = statname)) + theme(legend.position = 'top')
# # hasAxis(stat_plot) <- TRUE
# gene_plot <- ggbio::autoplot(Homo.sapiens::Homo.sapiens, which = wh, gap.geom = "chevron")
# hasAxis(gene_plot) <- TRUE
# ggbio::tracks(gene_plot, stat_plot) + ggtitle(paste(g, "consecutive low score"))

# ggsave(last_plot(), file = 'images/03_selectionpipeline/nzmDNAH6consec.png')
knitr::include_graphics('images/03_selectionpipeline/nzmDNAH6consec.png')
# rm(gene_plot, stat_plot, wh, g)
```


```{r nzmRGS3, fig.cap='(ref:nzmRGS3)', eval= TRUE, message = FALSE, fig.height = 4, fig.width = 4}
# plot_gene("NZM","RGS3")
# ggsave(last_plot(), file = 'images/03_selectionpipeline/nzmRGS3.png')
knitr::include_graphics('images/03_selectionpipeline/nzmRGS3.png')
```

```{r nzmCTNNA3, eval= TRUE, message = FALSE, fig.height = 4, fig.width = 4}
# plot_gene("NZM","CTNNA3")
# ggsave(last_plot(), file = 'images/03_selectionpipeline/nzmCTNNA3.png')
knitr::include_graphics('images/03_selectionpipeline/nzmCTNNA3.png')
```

```{r nzmRAD51AP2, eval= TRUE, message = FALSE, fig.height = 4, fig.width = 4}
# plot_gene("NZM","RAD51AP2")
# ggsave(last_plot(), file = 'images/03_selectionpipeline/nzmRAD51AP2.png')
knitr::include_graphics('images/03_selectionpipeline/nzmRAD51AP2.png')
```


```{r nzmVSNL1, eval= TRUE, message = FALSE, fig.height = 4, fig.width = 4}
# plot_gene("NZM","VSNL1")
# ggsave(last_plot(), file = 'images/03_selectionpipeline/nzmVSNL1.png')
knitr::include_graphics('images/03_selectionpipeline/nzmVSNL1.png')
```


```{r nzmSUMPT3H, fig.cap='(ref:nzmSUPT3H)', eval= FALSE, message = FALSE, fig.height = 4, fig.width = 4}
#plot_gene("NZM","SUPT3H")
#ggsave(last_plot(), file = 'images/03_selectionpipeline/nzmSUPT3H.png')
knitr::include_graphics('images/03_selectionpipeline/nzmSUPT3H.png')
```



```{r, eval = FALSE}
# number of genes associated with urate, obesity, t2d, kd, and met syn
bind_rows(sig_ihs_val %>% mutate(statname = 'ihs'), sig_nsl_val %>% mutate(statname = 'nsl')) %>% filter(pop == 'NZM', genename %in% urate_goutGR$SYMBOL) %>% group_by(genename, statname) %>% tally() %>% arrange(desc(n))

bind_rows(sig_ihs_val %>% mutate(statname = 'ihs'), sig_nsl_val %>% mutate(statname = 'nsl')) %>% filter(pop == 'NZM', genename %in% t2d_GR$SYMBOL) %>% group_by(genename, statname) %>% tally() %>% arrange(desc(n))

bind_rows(sig_ihs_val %>% mutate(statname = 'ihs'), sig_nsl_val %>% mutate(statname = 'nsl')) %>% filter(pop == 'NZM', genename %in% obesity_GR$SYMBOL) %>% group_by(genename, statname) %>% tally() %>% arrange(desc(n))

bind_rows(sig_ihs_val %>% mutate(statname = 'ihs'), sig_nsl_val %>% mutate(statname = 'nsl')) %>% filter(pop == 'NZM', genename %in% metsyn_GR$SYMBOL) %>% group_by(genename, statname) %>% tally() %>% arrange(desc(n))
```


```{r, echo = FALSE, eval = FALSE}
<!--
_Tajimas D_

The \gls{nzm} had `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower significance threshold. These windows made `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions covering `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. Theses regions contained `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('NZM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Tajima.D', !pop %in% c('NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes were only from \gls{nzm}. The upper tail had `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that formed `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions covering `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. There were `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('NZM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Tajima.D', !pop %in% c('NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were unique to \gls{nzm} for \gls{td}.

_Fay and Wu's H_

There were `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows forming  `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions that met the significance threshold for the lower tail. These regions were a total of `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp and had `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('NZM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fay.Wu.H', !pop %in% c('NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` of which were only significant in the \gls{nzm} population for \gls{fwh}.

In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that formed  `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` continuous regions that met the significance threshold. These regions covered `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp and had `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('NZM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fay.Wu.H', !pop %in% c('NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant in the \gls{nzm} population for \gls{fwh}


_Fu and Li's D_

For the \gls{nzm} population there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower tail threshold for significance and made `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions covering `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. In these regions there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which  `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('NZM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.D', !pop %in% c('NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant in \gls{nzm} for \gls{fld}. In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the significance threshold. These windows made `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions with a total of `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp containing `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, with  `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('NZM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.D', !pop %in% c('NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only meeting significance in \gls{nzm} for \gls{fld}.

_Fu and Li's F_

For the \gls{nzm} population there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower tail threshold for significance and made `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions covering `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. In these regions there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which  `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('NZM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.F', !pop %in% c('NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant in \gls{nzm} for \gls{flf}. In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the significance threshold. These windows made `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions with a total of `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp containing `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, with  `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('NZM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.F', !pop %in% c('NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only meeting significance in \gls{nzm} for \gls{flf}.

_Zeng's E_

For the \gls{nzm} population there were `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower tail threshold for significance and made `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions covering `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. In these regions there were `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which  `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('NZM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Zeng.E', !pop %in% c('NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant in \gls{nzm} for \gls{ze}. In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the significance threshold. These windows made `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions with a total of `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp containing `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, with  `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('NZM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Zeng.E', !pop %in% c('NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only meeting significance in \gls{nzm} for \gls{ze}.
-->
```


### Samoan


#### Pathway Analysis

<!-- - TD main processes: "cellular process", "metabolic process" -->
<!-- - FWH main processes: "cellular process", "metabolic process" -->
<!-- - FLD main processes: "cellular process", "metabolic process" -->
<!-- - FLF main processes: "cellular process", "metabolic process" -->
<!-- - ZE main processes: "cellular process", "metabolic process" -->
<!-- - ihs main processes: "cellular process", "metabolic process" -->
<!-- - nsl main processes: "cellular process", "metabolic process" -->
<!-- - all main processes: "cellular process", "metabolic process" -->
<!-- - all overrepresentation test: nothing -->

There were no significant results after adjusting for multiple testing when performing the over representation analysis using PantherDB on all genes that had met a significance threshold for any of the intra population selection and neutrality statistics. For all of the intra-population statistics, the main biological functional classification was "cellular process".

There were only two pathways that were significant for the gene-set pathway enrichment analysis using Enrichr. The pathway term "ABC transporters"" was significant for gene lists from both \gls{fld} (adjusted P = 0.0190) with 8 of 44 genes and \gls{ihs} (adjusted P = 0.0128) with 9 of 44 genes. The second pathway that was significant was "long-term potentiation" and that was from the \gls{nsl} gene list (adjusted P = 0.0213) with 9 of 66 genes.


#### Haplotypic

```{r, eval=FALSE}
p <- "SAM"

(sam_nsl_100 <- sig_nsl_val %>% filter(pop == p) %>% arrange(desc(abs(statvalue))) %>% head(n=100) %>% group_by(genename) %>% tally() %>% arrange(desc(n)))

(sam_ihs_100 <- sig_ihs_val %>% filter(pop == p) %>% arrange(desc(abs(statvalue))) %>% head(n=100) %>% group_by(genename) %>% tally() %>% arrange(desc(n)))

inner_join(sam_nsl_100, sam_ihs_100, by = 'genename' ) %>% mutate(total = n.x + n.y) %>% arrange(desc(total))
rm(p)
```

From the haplotypic tests for selection (\gls{ihs} and \gls{nsl}) there were `r sig_ihs_val %>% filter(pop == "SAM") %>% select(genename) %>% anti_join(., sig_ihs_val %>% filter(pop != "SAM") %>% select(genename), by = 'genename') %>% distinct() %>% NROW()` genes that were only significant in \gls{sam} for \gls{ihs}, and `r sig_nsl_val %>% filter(pop == "SAM") %>% select(genename) %>% anti_join(., sig_nsl_val %>% filter(pop != "SAM") %>% select(genename), by = 'genename') %>% distinct() %>% NROW()` for \gls{nsl}.

The genes with the most significant markers for \gls{ihs} were CDH23 and CNTN5 with 12 significant markers each. The gene with the most significant markers for \gls{nsl} was CDH23 with 16 markers. Looking at the most extreme 100 markers for \gls{ihs} there were 35 genes represented, SNX29 had the most with three markers. For \gls{nsl} the most markers in the most extreme 100 was C11orf65. There was a total of 39 genes represented. Between the top markers for both \gls{ihs} and \gls{nsl} there were 10 genes that were in common.

There were six genes that had at windows from at least four of the frequency based statistics and also had at least a single significant marker for either \gls{ihs} or \gls{nsl}. The genes were DISP (two \gls{ihs} markers), PARD3B (one \gls{ihs} and \gls{nsl} marker each), NEK1 (one \gls{ihs} marker), CTNNA3 (one \gls{ihs} marker and five \gls{nsl} markers), CEP112 (one \gls{ihs} and \gls{nsl} marker each), and SUPT3H (one \gls{nsl} marker). 


<!-- PARD3B had 4 different statistics overlap under exon XXX. -->


Nearly all of the windows and markers that intersected CNTNAP5 were overalapping and fell between chr2:125545002-125675001, with the most overlap occuring in an intronic region (Figure \@ref(fig:samCNTNAP5)).

#### Consecutive Regions

There were no consecutively depressed score regoins that intersected with genes associated with urate, obesity, \gls{t2d}, kidney disease, or metabolic syndrome. Regions of consecutive depressed score that also had significant makres for \gls{ihs} or \gls{nsl} in \gls{sam} covered three regions of the genome across two statistics. \gls{td} had two regions, the first was 400 kb at chr4:170025002-170425001 and had a single significant marker for \gls{ihs} in _NEK1_. The second region was 300 kb in length and was found at chr10:76685002-76985001. It had two markers that were significant  for \gls{ihs} and three for \gls{nsl} in _KAT6B_, two markers each for \gls{ihs} and \gls{nsl} at _SAMD8_, and two markers each fro \gls{ihs} and \gls{nsl} for DUSP13. \gls{flf} had a single 200 kb region at chr1:26465002-26665001 that had a single significant marker for \gls{nsl} in _CNKSR1_, and two markers for \gls{ihs} and three for \gls{nsl} in _AIM1L._


#### XPEHH

#### FST






```{r SAMSigRegions, eval = FALSE}
# copied into Panther adn then exported the biological process

#td
lower_sig_stats %>% filter(pop == 'SAM', statname == 'Tajima.D', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique() %>% cat()
#fwh
lower_sig_stats %>% filter(pop == 'SAM', statname == 'Fay.Wu.H', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique() %>% cat()
#fld
lower_sig_stats %>% filter(pop == 'SAM', statname == 'Fu.Li.D', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique() %>% cat()
#flf
lower_sig_stats %>% filter(pop == 'SAM', statname == 'Fu.Li.F', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique() %>% cat()
#ze
lower_sig_stats %>% filter(pop == 'SAM', statname == 'Zeng.E', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique() %>% cat()
#ihs
sig_ihs_val %>% filter(pop == 'SAM', !is.na(genename)) %>% .[['genename']] %>% unique() %>% cat()
#nsl
sig_nsl_val %>% filter(pop == 'SAM', !is.na(genename)) %>% .[['genename']] %>% unique() %>% cat()

# all with at least 1 sfs and 1 haplotypic
bind_rows(bind_rows(sig_ihs_val, sig_nsl_val) %>% filter(pop == 'SAM')%>% select(pop, "SYMBOL" = genename ) %>% distinct(), lower_sig_stats %>% filter(pop == 'SAM')%>% select(pop, SYMBOL) %>% distinct()) %>% filter(!is.na(SYMBOL))  %>% group_by(SYMBOL) %>% tally() %>% filter(n == 2) %>% .[['SYMBOL']] %>% cat()
```




There were `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes that intersected the windows that met the significance threshold in the lower tail for \gls{td}. For \gls{fwh}, there were `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, \gls{fld} had `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()`, \gls{flf} had `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()`, and \gls{ze} had `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()`.









<!-- no urate, obesity, t2d, kd, or metsyn -->

<!-- haplotypic results intersecting regions -->
```{r, eval = FALSE}
# find ihs/nsl intersecting the regions of consecutively depressed score
consec_lower_regions %>% filter( pop == 'SAM') %>% inner_join(., bind_rows(sig_ihs_val, sig_nsl_val) %>% select(SYMBOL = genename, everything()), by = c('pop','SYMBOL')) %>% select(-contains('gene'))
```

<!-- - td -->
<!--     - chr4:170025002-170425001 w=400000 -->
<!--         - NEK1 ihs -->
<!--     - chr10:76685002-76985001 w=300000 -->
<!--         - KAT6B 2x ihs, 3x nsl -->
<!--         - SAMD8 2x ihs, 2x nsl -->
<!--         - DUSP13 2x ihs, 2x nsl -->
<!-- - flf -->
<!--     - chr1:26465002-26665001 w=200000 -->
<!--         - CNKSR1 nsl -->
<!--         - AIM1L 2 x ihs, 3x nsl -->



<!-- (ref:upsetSAM) Upset plot of number of genes that were significant between  statistics for \gls{sam} -->

```{r upsetSam, fig.cap = '(ref:upsetSAM)', eval = FALSE}
UpSetR::upset(fromList(list(`Tajima's D` = lower_sig_stats %>% filter(pop == 'SAM', statname == 'Tajima.D', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(), 
                            `Fay and Wu's H` = lower_sig_stats %>% filter(pop == 'SAM', statname == 'Fay.Wu.H', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(), 
                            `Fu and Li's D` = lower_sig_stats %>% filter(pop == 'SAM', statname == 'Fu.Li.D', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(), 
                            `Fu and Li's F` = lower_sig_stats %>% filter(pop == 'SAM', statname == 'Fu.Li.F', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(),
                            `Zeng's E` = lower_sig_stats %>% filter(pop == 'SAM', statname == 'Zeng.E', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(),
                            iHS = sig_ihs_val %>% filter(pop == 'SAM', !is.na(genename) & genename != "") %>% .[['genename']] %>% unique(), 
                            nSL =sig_nsl_val %>% filter(pop == 'SAM', !is.na(genename) & genename != "") %>% .[['genename']] %>% unique())), 
              order.by = c("freq","degree"), nsets = 7, number.angles = 90, set_size.angles = 90, hjust = -0.25, vjust = 0.5 , nintersects = NA)
```

```{r svgSAMupset, include = FALSE, eval = FALSE}
svg(filename = 'images/03_selectionpipeline/sam_upset.svg')
  UpSetR::upset(fromList(list(`Tajima's D` = lower_sig_stats %>% filter(pop == 'SAM', statname == 'Tajima.D', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(), 
                            `Fay and Wu's H` = lower_sig_stats %>% filter(pop == 'SAM', statname == 'Fay.Wu.H', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(), 
                            `Fu and Li's D` = lower_sig_stats %>% filter(pop == 'SAM', statname == 'Fu.Li.D', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(), 
                            `Fu and Li's F` = lower_sig_stats %>% filter(pop == 'SAM', statname == 'Fu.Li.F', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(),
                            `Zeng's E` = lower_sig_stats %>% filter(pop == 'SAM', statname == 'Zeng.E', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(),
                            iHS = sig_ihs_val %>% filter(pop == 'SAM', !is.na(genename) & genename != "") %>% .[['genename']] %>% unique(), 
                            nSL =sig_nsl_val %>% filter(pop == 'SAM', !is.na(genename) & genename != "") %>% .[['genename']] %>% unique())), 
              order.by = c("freq","degree"), nsets = 7, number.angles = 90, set_size.angles = 90, hjust = -0.25, vjust = 0.5 , nintersects = NA)
  dev.off()
```

(ref:samCNTNAP5) Plot showing markers, genes and significant results for \gls{sam} at _CNTNAP5_

```{r samCNTNAP5, fig.cap='(ref:samCNTNAP5)', eval= TRUE, message = FALSE, fig.height = 4, fig.width = 4}
#retrieve information of the gene of interest
# plot_gene("SAM","CNTNAP5")
# ggsave(last_plot(), file = 'images/03_selectionpipeline/samCNTNAP5.png')
knitr::include_graphics('images/03_selectionpipeline/samCNTNAP5.png')
```


```{r samSPAG16, fig.cap='(ref:samSPAG16)', eval= FALSE, message = FALSE, fig.height = 4, fig.width = 4}
#plot_gene("SAM","SPAG16")
#ggsave(last_plot(), file = 'images/03_selectionpipeline/samSPAG16.png')
knitr::include_graphics('images/03_selectionpipeline/samSPAG16.png')
```

```{r samSPECC1, fig.cap='(ref:samSPECC1)', eval= FALSE, message = FALSE, fig.height = 4, fig.width = 4}
#plot_gene("SAM","SPECC1")
#ggsave(last_plot(), file = 'images/03_selectionpipeline/samSPECC1.png')
knitr::include_graphics('images/03_selectionpipeline/samSPECC1.png')
```

```{r samSUPT3H, fig.cap='(ref:samSUPT3H)', eval= TRUE, message = FALSE, fig.height = 4, fig.width = 4}
# plot_gene("SAM","SUPT3H")
# ggsave(last_plot(), file = 'images/03_selectionpipeline/samSUPT3H.png')
knitr::include_graphics('images/03_selectionpipeline/samSUPT3H.png')
```

```{r samCTNNA3, fig.cap='(ref:samCTNNA3)', eval= TRUE, message = FALSE, fig.height = 4, fig.width = 4}
# plot_gene("SAM","CTNNA3")
# ggsave(last_plot(), file = 'images/03_selectionpipeline/samCTNNA3.png')
knitr::include_graphics('images/03_selectionpipeline/samCTNNA3.png')
```

```{r samCEP112, fig.cap='(ref:samCEP112)', eval= FALSE, message = FALSE, fig.height = 4, fig.width = 4}
# plot_gene("SAM","CEP112")
# ggsave(last_plot(), file = 'images/03_selectionpipeline/samCEP112.png')
knitr::include_graphics('images/03_selectionpipeline/samCEP112.png')
```

```{r samPARD3B, fig.cap='(ref:samPARD3B)', eval= TRUE, message = FALSE, fig.height = 4, fig.width = 4}
# plot_gene("SAM","PARD3B")
# ggsave(last_plot(), file = 'images/03_selectionpipeline/samPARD3B.png')
knitr::include_graphics('images/03_selectionpipeline/samPARD3B.png')
```


<!-- - td most specific -->
<!-- - fld most regions -->

<!-- In the upper tail, there were `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes for \gls{td} that intersected windows that met the significance threshold. \gls{fwh} had `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()`, \gls{fld} had `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()`, \gls{flf} had `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()`, and \gls{ze} had `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()`. -->

<!-- upper -->

<!-- - 5 stats: `r upper_sig_stats %>% filter(pop %in% c('SAM')) %>% select(SYMBOL, statname) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==5, !is.na(SYMBOL)) %>% .[['SYMBOL']]` -->
<!-- - 4 stats: `r upper_sig_stats %>% filter(pop %in% c('SAM')) %>% select(SYMBOL, statname) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==4, !is.na(SYMBOL)) %>% NROW()` -->
<!-- - 3 stats: `r upper_sig_stats %>% filter(pop %in% c('SAM')) %>% select(SYMBOL, statname) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==3, !is.na(SYMBOL)) %>% NROW()` -->
<!-- - 2 stats: `r upper_sig_stats %>% filter(pop %in% c('SAM')) %>% select(SYMBOL, statname) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==2, !is.na(SYMBOL)) %>% NROW()` -->
<!-- - 1 stats: `r upper_sig_stats %>% filter(pop %in% c('SAM')) %>% select(SYMBOL, statname) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==1, !is.na(SYMBOL)) %>% NROW()` -->


\gls{sam} had the most genes that were specific to \gls{sam} for \gls{td} out of the Polynesian populations for both significance thresholds.



```{r, echo = FALSE, eval = FALSE}
_Tajima's D_

In the \gls{sam} population there were `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that made up `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` continuous regions covering `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. In these regions there were`r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Tajima.D', !pop %in% c('SAM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were unique to \gls{sam}. For the upper tail there were  `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that formed  `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions with a total of `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. There were `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes that were covered by these regions, `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Tajima.D', !pop %in% c('SAM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` that were only found in \gls{sam} for \gls{td}.

_Fay and Wu's H_

There were  `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that formed  `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions, totaling  `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp that met the significance threshold for the lower tail. These regions had `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('SAM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fay.Wu.H', !pop %in% c('SAM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant in the \gls{sam} population for \gls{fwh}.

In the upper tail there were  `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the threshold for significance. These windows formed `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions with a total of  `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. There were  `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('SAM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fay.Wu.H', !pop %in% c('SAM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only from the \gls{sam} population for \gls{fwh}. 

_Fu and Li's D_

For the \gls{sam} population there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower tail threshold for significance and made `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions covering `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. In these regions there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which  `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('SAM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.D', !pop %in% c('SAM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant in \gls{sam} for \gls{fld}. In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the significance threshold. These windows made `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions with a total of `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp containing `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, with  `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('SAM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.D', !pop %in% c('SAM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only meeting significance in \gls{sam} for \gls{fld}.

_Fu and Li's F_

For the \gls{sam} population there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower tail threshold for significance and made `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions covering `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. In these regions there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which  `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('SAM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.F', !pop %in% c('SAM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant in \gls{sam} for \gls{flf}. In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the significance threshold. These windows made `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions with a total of `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp containing `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, with  `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('SAM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.F', !pop %in% c('SAM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only meeting significance in \gls{sam} for \gls{flf}.

_Zeng's E_

For the \gls{sam} population there were `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower tail threshold for significance and made `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions covering `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. In these regions there were `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which  `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('SAM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Zeng.E', !pop %in% c('SAM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant in \gls{sam} for \gls{ze}. In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the significance threshold. These windows made `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions with a total of `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp containing `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, with  `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('SAM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Zeng.E', !pop %in% c('SAM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only meeting significance in \gls{sam} for \gls{ze}.
```


> from the multiplot it looks like CREB1 could be interesting

CREB1 Plot
```{r samCREB1, fig.cap='(ref:samCREB1)', eval= FALSE, message = FALSE, fig.height = 4, fig.width = 4, cache = TRUE}
#plot_gene("SAM","CREB1")
#ggsave(last_plot(), file = 'images/03_selectionpipeline/samCREB1.png')
knitr::include_graphics('images/03_selectionpipeline/samCREB1.png')
```


<!-- Enrichr using the 0.1 percentile -->

<!-- TD has 3 significant pathway hits -->

<!-- FLD had 39 pathways -->
<!-- FLF had 48 pathways -->

<!-- ihs > 3.89, significant pathway for ABC transporters -->

### Tongan

```{r TonSigRegions, eval = FALSE}
# copied into Panther adn then exported the biological process

#td
lower_sig_stats %>% filter(pop == 'TON', statname == 'Tajima.D', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique() %>% cat()
#fwh
lower_sig_stats %>% filter(pop == 'TON', statname == 'Fay.Wu.H', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique() %>% cat()
#fld
lower_sig_stats %>% filter(pop == 'TON', statname == 'Fu.Li.D', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique() %>% cat()
#flf
lower_sig_stats %>% filter(pop == 'TON', statname == 'Fu.Li.F', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique() %>% cat()
#ze
lower_sig_stats %>% filter(pop == 'TON', statname == 'Zeng.E', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique() %>% cat()
#ihs
sig_ihs_val %>% filter(pop == 'TON', !is.na(genename)) %>% .[['genename']] %>% unique() %>% cat()
#nsl
sig_nsl_val %>% filter(pop == 'TON', !is.na(genename)) %>% .[['genename']] %>% unique() %>% cat()

# all with at least 1 sfs and 1 haplotypic
bind_rows(bind_rows(sig_ihs_val, sig_nsl_val) %>% filter(pop == 'TON')%>% select(pop, "SYMBOL" = genename ) %>% distinct(), lower_sig_stats %>% filter(pop == 'TON')%>% select(pop, SYMBOL) %>% distinct()) %>% filter(!is.na(SYMBOL))  %>% group_by(SYMBOL) %>% tally() %>% filter(n == 2) %>% .[['SYMBOL']] %>% cat()
```

#### Pathway Analysis

Functional classification using PantherDB for the significant windows from the 1st percentile for the frequency spectrum selection and neutrality tests, and the significant \gls{ihs} and \gls{nsl} each individually, had the main biological process classification as "cellular process" and the second main classification as 'metabolic process'. Combining all the significant genes from any

#### Haplotypic

```{r, eval=FALSE}
p <- "TON"

(ton_nsl_100 <- sig_nsl_val %>% filter(pop == p) %>% arrange(desc(abs(statvalue))) %>% head(n=100) %>% group_by(genename) %>% tally() %>% arrange(desc(n)))

(ton_ihs_100 <- sig_ihs_val %>% filter(pop == p) %>% arrange(desc(abs(statvalue))) %>% head(n=100) %>% group_by(genename) %>% tally() %>% arrange(desc(n)))

inner_join(ton_nsl_100, ton_ihs_100, by = 'genename' ) %>% mutate(total = n.x + n.y) %>% arrange(desc(total))
rm(p)
```



In the most extreme 100 \gls{ihs} marker scores, there were 32 genes represented with _SNX29_ having three markers, and _CDH23_ having two from the 100. The most extreme 100 markers for \gls{nsl} had 30 genes represented with SNX29 having five markers, _PCDH15_ having three, and _DNAH11_ having two of the 100 markers. There was an overlap between the two genelists for the most extreme 100 of 6 genes, with _SNX29_, and _DNAH11_ having the most number of extreme markers for both \gls{ihs}, and \gls{nsl}. The genes with the most number of significant markers overall for \gls{ihs} and \gls{nsl} were _CPNE4_ (16 markers for both \gls{ihs} and \gls{nsl}), _SNX29_ (15 markers for \gls{ihs} and 16 for \gls{nsl}), _CDH23_ (11 markers for \gls{ihs} and 12 for \gls{nsl}), _PCDH15_ (12 markers for \gls{nsl}), and _BANK1_ (9 markers for \gls{ihs} and 10 for \gls{nsl}).

In total there were  `r sig_ihs_val %>% filter(pop == "TON") %>% select(genename) %>% anti_join(., sig_ihs_val %>% filter(pop != "TON") %>% select(genename), by = 'genename') %>% distinct() %>% NROW()` genes that were only significant in \gls{ton} for \gls{ihs}, and `r sig_nsl_val %>% filter(pop == "TON") %>% select(genename) %>% anti_join(., sig_nsl_val %>% filter(pop !="TON") %>% select(genename), by = 'genename') %>% distinct() %>% NROW()` for \gls{nsl}. 

Combining the genes with at least a single significant marker for either \gls{ihs} or \gls{nsl} with the windows in the 1st percentile for at least 4 frequency based intra-population selection and neutrality statistics had three genes that met this criteria. They were _SUPT3H_, _LPO_, and _NRXN3._

#### Consecutive Regions

There were no consecutively depressed score regions that instersected genes asscociated with urate, \gls{t2d}, kidney disease, or metabolic syndrome. There was one region that intersected the obesity associated gene _CCR3_ for \gls{td} and was 260 kb in length at chr3:46035002-46295001. There were three regions with consecutively depressed score that also had significant markers for \gls{ihs} or \gls{nsl} in \gls{ton}. \gls{td} had two regions. The first similar to \gls{sam}, was at chr10:76685002-76985001 and spanned 300 kb and had three significant markers for \gls{ihs} and four for \gls{nsl} in _KAT6B_, three markers for both \gls{ihs} and \gls{nsl} in _SAMD8_, and two markers each for \gls{ihs} and \gls{nsl} in _DUSP13._ The second region at chr17:56125002-56415001 was 290 kb and had a single significant marker for \gls{ihs} in both EPX and LPO. \gls{flf} had a slightly shifted region that also covered _EPX_ and _LPO_, but also included a _RNF43_, all there genes had a single significant marker each for \gls{ihs}. The region was at chr17:56205002-56455001 and was 250 kb in length.



#### XPEHH

#### FST


#### Notes

<!-- - TD main processes: "cellular process", "metabolic process" -->
<!-- - FWD main processes: "cellular process", "metabolic process" -->
<!-- - FLD main processes: "cellular process", "metabolic process" -->
<!-- - FLF main processes: "cellular process", "metabolic process" -->
<!-- - ZE main processes: "cellular process", "metabolic process" -->
<!-- - ihs main processes: "cellular process", "metabolic process" -->
<!-- - nsl main processes: "cellular process", "metabolic process" -->
<!-- - all main processes: "cellular process", "metabolic process" -->
<!-- - all overrepresentation test: nothing -->




<!-- From all of the intra population -->

<!-- - There were `r bind_rows(ton_urate_sig) %>% select(SYMBOL) %>% distinct() %>% NROW()` of `r length(unique(urate_goutGR$SYMBOL))` urate genes -->
<!-- - There were `r bind_rows(ton_obesity_sig) %>% select(SYMBOL) %>% distinct() %>% NROW()` of `r length(unique(obesity_GR$SYMBOL))` obesity genes -->
<!-- - There were `r bind_rows(ton_t2d_sig) %>% select(SYMBOL) %>% distinct() %>% NROW()` of `r length(unique(t2d_GR$SYMBOL))` \gls{t2d} genes -->
<!-- - There were `r bind_rows(ton_kd_sig) %>% select(SYMBOL) %>% distinct() %>% NROW()` of `r length(unique(kd_GR$SYMBOL))` kidney disease genes -->
<!-- - There were `r bind_rows(ton_metsyn_sig) %>% select(SYMBOL) %>% distinct() %>% NROW()` of `r length(unique(metsyn_GR$SYMBOL))` metsyn genes -->

<!-- lower -->

<!-- - 5 stats: `r lower_sig_stats %>% filter(pop %in% c('TON')) %>% select(SYMBOL, statname) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==5, !is.na(SYMBOL)) %>% .[['SYMBOL']]` -->
<!-- - 4 stats: `r lower_sig_stats %>% filter(pop %in% c('TON')) %>% select(SYMBOL, statname) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==4, !is.na(SYMBOL)) %>% NROW()` -->
<!-- - 3 stats: `r lower_sig_stats %>% filter(pop %in% c('TON')) %>% select(SYMBOL, statname) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==3, !is.na(SYMBOL)) %>% NROW()` -->
<!-- - 2 stats: `r lower_sig_stats %>% filter(pop %in% c('TON')) %>% select(SYMBOL, statname) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==2, !is.na(SYMBOL)) %>% NROW()` -->
<!-- - 1 stats: `r lower_sig_stats %>% filter(pop %in% c('TON')) %>% select(SYMBOL, statname) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==1, !is.na(SYMBOL)) %>% NROW()` -->





<!-- no urate, t2d, kd, or metsyn -->

<!-- obesity -->
<!-- td chr3:46035002-46295001 w=260000 CCR3  -->

<!-- haplotypic results intersecting regions -->
```{r, eval = FALSE}
# find ihs/nsl intersecting the regions of consecutively depressed score
consec_lower_regions %>% filter( pop == 'TON') %>% inner_join(., bind_rows(sig_ihs_val, sig_nsl_val) %>% select(SYMBOL = genename, everything()), by = c('pop','SYMBOL')) %>% select(-contains('gene'))
```

<!-- -td -->
<!--     - chr10:76685002-76985001 w=300000 -->
<!--         - KAT6B 3x ihs, 4x nsl -->
<!--         - SAMD8 3x ihs, 3x nsl -->
<!--         - DUSP13 2x ihs, 2x nsl -->
<!--     - chr17:56125002-56415001 w=290000 -->
<!--         - EPX ihs -->
<!--         - LPO ihs -->
<!-- - flf -->
<!--     - chr17 56205002 56455001 250000 -->
<!--         - EPX ihs -->
<!--         - LPO ihs -->
<!--         - RNF43 ihs -->



(ref:upsetTON) TON Upset Plot

```{r upsetTon, fig.cap = '(ref:upsetTON)', eval = FALSE}
UpSetR::upset(fromList(list(`Tajima's D` = lower_sig_stats %>% filter(pop == 'TON', statname == 'Tajima.D', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(), 
                            `Fay and Wu's H` = lower_sig_stats %>% filter(pop == 'TON', statname == 'Fay.Wu.H', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(), 
                            `Fu and Li's D` = lower_sig_stats %>% filter(pop == 'TON', statname == 'Fu.Li.D', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(), 
                            `Fu and Li's F` = lower_sig_stats %>% filter(pop == 'TON', statname == 'Fu.Li.F', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(),
                            `Zeng's E` = lower_sig_stats %>% filter(pop == 'TON', statname == 'Zeng.E', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(),
                            iHS = sig_ihs_val %>% filter(pop == 'TON', !is.na(genename) & genename != "") %>% .[['genename']] %>% unique(), 
                            nSL =sig_nsl_val %>% filter(pop == 'TON', !is.na(genename) & genename != "") %>% .[['genename']] %>% unique())), 
              order.by = c("freq", "degree"), nsets = 7, number.angles = 90, set_size.angles = 90, hjust = -0.25, vjust = 0.5, nintersects = NA)
```


```{r svgTONupset, include = FALSE, eval = FALSE}
svg(filename = 'images/03_selectionpipeline/ton_upset.svg')
UpSetR::upset(fromList(list(`Tajima's D` = lower_sig_stats %>% filter(pop == 'TON', statname == 'Tajima.D', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(), 
                            `Fay and Wu's H` = lower_sig_stats %>% filter(pop == 'TON', statname == 'Fay.Wu.H', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(), 
                            `Fu and Li's D` = lower_sig_stats %>% filter(pop == 'TON', statname == 'Fu.Li.D', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(), 
                            `Fu and Li's F` = lower_sig_stats %>% filter(pop == 'TON', statname == 'Fu.Li.F', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(),
                            `Zeng's E` = lower_sig_stats %>% filter(pop == 'TON', statname == 'Zeng.E', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(),
                            iHS = sig_ihs_val %>% filter(pop == 'TON', !is.na(genename) & genename != "") %>% .[['genename']] %>% unique(), 
                            nSL =sig_nsl_val %>% filter(pop == 'TON', !is.na(genename) & genename != "") %>% .[['genename']] %>% unique())), 
              order.by = c("freq", "degree"), nsets = 7, number.angles = 90, set_size.angles = 90, hjust = -0.25, vjust = 0.5, nintersects = NA)
dev.off()
```

```{r tonLPO,  fig.cap='(ref:tonLPO)', eval= TRUE, message = FALSE, fig.height = 4, fig.width = 4}
# plot_gene("TON","LPO")
# ggsave(last_plot(), file = 'images/03_selectionpipeline/tonLPO.png')
knitr::include_graphics('images/03_selectionpipeline/tonLPO.png')
```



```{r tonNRXN3, fig.cap='(ref:tonNRXN3)', eval= TRUE, message = FALSE, fig.height = 4, fig.width = 4}
# plot_gene("TON","NRXN3")
# ggsave(last_plot(), file = 'images/03_selectionpipeline/tonNRXN3.png')
knitr::include_graphics('images/03_selectionpipeline/tonNRXN3.png')
```

```{r tonCENPP, fig.cap='(ref:tonCENPP)', eval= TRUE, message = FALSE, fig.height = 4, fig.width = 4}
#plot_gene("TON","CENPP")
#ggsave(last_plot(), file = 'images/03_selectionpipeline/tonCENPP.png')
knitr::include_graphics('images/03_selectionpipeline/tonCENPP.png')
```


```{r tonSH3RF1, fig.cap='(ref:tonSH3RF1)', eval= FALSE, message = FALSE, fig.height = 4, fig.width = 4}
#plot_gene("TON","SH3RF1")
#ggsave(last_plot(), file = 'images/03_selectionpipeline/tonSH3RF1.png')
knitr::include_graphics('images/03_selectionpipeline/tonSH3RF1.png')
```

```{r tonSUPT3H, fig.cap='(ref:tonSUPT3H)', eval= TRUE, message = FALSE, fig.height = 4, fig.width = 4}
# plot_gene("TON","SUPT3H")
# ggsave(last_plot(), file = 'images/03_selectionpipeline/tonSUPT3H.png')
knitr::include_graphics('images/03_selectionpipeline/tonSUPT3H.png')
```




<!-- - From the lower tail there were `r consec_lower_regions %>% filter(stat == 'Tajima.D', pop == 'TON') %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce(ignore.strand = TRUE) %>% length()` regions of extended significant windows for \gls{td}, `r consec_lower_regions %>% filter(stat == 'Fay.Wu.H', pop == 'TON') %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce(ignore.strand = TRUE) %>% length()` for \gls{fwh}, `r consec_lower_regions %>% filter(stat == 'Fu.Li.D', pop == 'TON') %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce(ignore.strand = TRUE) %>% length()` for \gls{fld}, `r consec_lower_regions %>% filter(stat == 'Fu.Li.F', pop == 'TON') %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce(ignore.strand = TRUE) %>% length()` for \gls{flf}, and `r consec_lower_regions %>% filter(stat == 'Zeng.E', pop == 'TON') %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce(ignore.strand = TRUE) %>% length()` for \gls{ze}. -->




> from the multi stat plot it looks like NRXN3 and THADA could be interesting



THADA Plot
```{r tonTHADA, fig.cap='(ref:tonTHADA)', eval= TRUE, message = FALSE, fig.height = 4, fig.width = 4}
#plot_gene("TON","THADA")
#ggsave(last_plot(), file = 'images/03_selectionpipeline/tonTHADA.png')
knitr::include_graphics('images/03_selectionpipeline/tonTHADA.png')
```


Enrichr using 0.1 percentile

FLD 1 pathway significant - olfactory transduction
FLF 1 pathway significant - olfactory transduction

ihs > 3.89, type 1 diabetes melitus was top hit but P = 0.079
nsl > 3.89 insulin resistance and axon guidance were both significant

```{r, echo = FALSE, eval = FALSE}
_Tajima's D_

For the \gls{ton} population there were `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower tail threshold for significance and made `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions covering `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. In these regions there were `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which  `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Tajima.D', !pop %in% c('TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant in \gls{ton} for \gls{td}. In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the significance threshold. These windows made `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions with a total of `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp containing `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, with  `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Tajima.D', !pop %in% c('TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only meeting significance in \gls{ton} for \gls{td}.

_Fay and Wu's H_

There were  `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that reached the significance threshold in the lower tail. These windows made `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()`independent continuous regions totaling `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6`. Within these regions there were  `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fay.Wu.H', !pop %in% c('TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes that were only significant in \gls{ton} for \gls{fwh}, from `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes that were significant.

In the upper tail there were  `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the significance threshold making up  `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions that were independent and continuous with a total length of `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. Within these regions there were `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which  `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fay.Wu.H', !pop %in% c('TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were significant in only \gls{ton} for \gls{fwh}.

_Fu and Li's D_

For the \gls{ton} population there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower tail threshold for significance and made `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions covering `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. In these regions there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which  `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.D', !pop %in% c('TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant in \gls{ton} for \gls{fld}. In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the significance threshold. These windows made `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions with a total of `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp containing `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, with  `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.D', !pop %in% c('TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only meeting significance in \gls{ton} for \gls{fld}.

_Fu and Li's F_

For the \gls{ton} population there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower tail threshold for significance and made `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions covering `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. In these regions there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which  `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.F', !pop %in% c('TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant in \gls{ton} for \gls{flf}. In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the significance threshold. These windows made `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions with a total of `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp containing `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, with  `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.F', !pop %in% c('TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only meeting significance in \gls{ton} for \gls{flf}.

_Zeng's E_

For the \gls{ton} population there were `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower tail threshold for significance and made `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions covering `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. In these regions there were `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which  `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Zeng.E', !pop %in% c('TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant in \gls{ton} for \gls{ze}. In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the significance threshold. These windows made `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions with a total of `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp containing `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, with  `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Zeng.E', !pop %in% c('TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only meeting significance in \gls{ton} for \gls{ze}.
```


### Disease associated genes




<!-- Genes talked about are cherry picked based on  -->

<!-- - number of snps that were significant -->
<!-- - SFS and haplotypic -->
<!-- - multiple polynesian populations -->
<!-- - only significant (or highly enriched) within the polynesians -->
<!-- - on multiple gene lists -->
<!-- - prioritised haplotypic results -->


The \gls{cim} had the least number of genes of the Polynesian populations that had a significant result out of the genes that were associated with urate. They also had the least number of genes with significance for obesity, \gls{t2d}, kidney disease, or metabolic syndrome when haplotypic and frequency spectrum methods were combined.

The Western Polynesian populations had the twice the number of genes associated with kidney disease, when looking at both haplotypic and frequency spectrum methods.


The \gls{nzm} population had the highest number of genes associated with obesity and \gls{t2d} that had a significant value in any of the intra-population haplotypic and frequency spectrum methods.

```{r polDiseaseGenesTab}
bind_rows(
lower_sig_stats %>% select(SYMBOL, pop) %>% filter(pop %in% poly_pop) %>% 
  mutate(
    urate = ifelse(SYMBOL %in% urate_goutGR$SYMBOL, TRUE, FALSE),
    obesity = ifelse(SYMBOL %in% obesity_GR$SYMBOL, TRUE, FALSE),
    t2d = ifelse(SYMBOL %in% t2d_GR$SYMBOL, TRUE, FALSE),
    kd = ifelse(SYMBOL %in% kd_GR$SYMBOL, TRUE, FALSE),
    metsyn = ifelse(SYMBOL %in% metsyn_GR$SYMBOL, TRUE, FALSE)
    ) %>% 
  select(SYMBOL, pop, urate, obesity, t2d, kd, metsyn) %>% 
  distinct() %>% 
  group_by(pop) %>% 
  summarise(urate = sum(urate), obesity = sum(obesity), t2d = sum(t2d), kd = sum(kd), metsyn = sum(metsyn)) %>% 
  mutate(type = 'SFS'),
bind_rows(sig_ihs_val, sig_nsl_val) %>% 
  select(pop, 'SYMBOL' = genename) %>% 
  filter(pop %in% poly_pop) %>% 
  mutate(
    urate = ifelse(SYMBOL %in% urate_goutGR$SYMBOL, TRUE, FALSE),
    obesity = ifelse(SYMBOL %in% obesity_GR$SYMBOL, TRUE, FALSE),
    t2d = ifelse(SYMBOL %in% t2d_GR$SYMBOL, TRUE, FALSE),
    kd = ifelse(SYMBOL %in% kd_GR$SYMBOL, TRUE, FALSE),
    metsyn = ifelse(SYMBOL %in% metsyn_GR$SYMBOL, TRUE, FALSE)
    ) %>% 
  select(SYMBOL, pop, urate, obesity, t2d, kd, metsyn) %>% 
  distinct() %>% 
  group_by(pop) %>% 
  summarise(urate = sum(urate), obesity = sum(obesity), t2d = sum(t2d), kd = sum(kd), metsyn = sum(metsyn)) %>% 
  mutate(type = 'Haplotype'),

bind_rows(
  bind_rows(sig_ihs_val, sig_nsl_val) %>% 
    select(pop, 'SYMBOL' = genename),
          lower_sig_stats %>% select(SYMBOL, pop)
)%>% 
  filter(pop %in% poly_pop) %>% 
  mutate(
    urate = ifelse(SYMBOL %in% urate_goutGR$SYMBOL, TRUE, FALSE),
    obesity = ifelse(SYMBOL %in% obesity_GR$SYMBOL, TRUE, FALSE),
    t2d = ifelse(SYMBOL %in% t2d_GR$SYMBOL, TRUE, FALSE),
    kd = ifelse(SYMBOL %in% kd_GR$SYMBOL, TRUE, FALSE),
    metsyn = ifelse(SYMBOL %in% metsyn_GR$SYMBOL, TRUE, FALSE)
    ) %>% 
  select(SYMBOL, pop, urate, obesity, t2d, kd, metsyn) %>% 
  distinct() %>% 
  group_by(pop) %>%
  summarise(urate = sum(urate), obesity = sum(obesity), t2d = sum(t2d), kd = sum(kd), metsyn = sum(metsyn)) %>% 
  mutate(type = 'combined')
) %>% 
  kable(., col.names = c('Population','Urate / Gout','Obesity','T2D','Kidney Disease','Metabolic Syndrome','Type'), caption = '(\\#tab:polDiseaseGenesTab) Number of Genes with evidence of possible positive selection in Polynesians', booktabs = TRUE)
```


#### Urate


```{r, eval = FALSE}
lower_sig_stats %>% filter(pop %in% poly_pop, SYMBOL %in% urate_goutGR$SYMBOL) %>% select(pop, SYMBOL, statname) %>% distinct()%>% spread(statname, statname, "") %>% arrange(SYMBOL)
```

Table \@ref(tab:polDiseaseGenesTab) showed that the Western Polynesian populations of \gls{sam} and \gls{ton} had a higher number of genes associated with urate than the Eastern Polynesian populations of \gls{cim} and \gls{nzm}. The Eastern Polynesian populations had no loci that met the significance threshold from the haplotypic methods of \gls{ihs} and \gls{nsl}. The two main effect loci for serum urate, _SLC2A9_ and _ABCG2_ both had limited evidence of selection. _SLC2A9_ did not have any values for \gls{ihs} or \gls{nsl} because of an extended region of no recombination leading to a large gap in the reference genome which caused the \gls{ehh} calculation to be terminated. There was evidence with \gls{fld} and \gls{flf} which both had windows from _SLC2A9_ in the 1st percentile from \gls{ton}. This was also observed with the \gls{eas} populations of \gls{chb} and \gls{jpt}. There was a 120kb region of consecutively low score for both \gls{fld} with \gls{ton}, \gls{chb}, and \gls{jpt}. This same region was also found in \gls{flf} for \gls{ton}.

For Polynesian populations, _ABCG2_ only had  significant results from \gls{ihs} and \gls{nsl} at the same marker (rs2622626) for \gls{ton}. With \gls{ihs} the \gls{sas} populations of \gls{gih} and \gls{itu} also had significant values for rs2622626. For \gls{nsl}, along with \gls{gih}, and \gls{itu}, the other non-Polynesian populations that had a significant value was the \gls{eas} population \gls{jpt}, and the \gls{afr} population of \gls{yri}. There were no Polynesian populations with values in the 1st percentile for the frequency based statistics, and only \gls{esn} had windows that met the threshold for \gls{fwh}. 

The only other transporter that was associated with urate that had a window meeting the significance threshold in Polynesian populations was _SLC16A9_ in \gls{ton} for \gls{fld}.

The majority of loci associated with urate that showed possible evidence of selection were not urate transporters but instead involved in multiple biological processes. There were four loci that had consecutively low scores in at least one statistic in at least one Polynesian population. These were _ATXN2_, _BAZ1B_, _BCAS3_, and _INHBC_. There were four other loci, _IGFR_, _NFAT5_, _PRKAG2_, and _RREB1_, that had met the threshold but not over a large consecutive region in at least a single Polynesian population. _ATXN2_ had a consecutively low score for \gls{fld} and \gls{flf} in \gls{sam} and also \gls{chb}. Windows in the 1st percentile for \gls{ze} came from \gls{cim}, \gls{nzm}, and \gls{sam}. _BAZ1B_ had consecutively low scores for \gls{td} for \gls{ton} covering 130kb, this same region was seen in the \gls{clm}, \gls{itu}, \gls{pel}, and \gls{stu} populations. Windows for the 1st percentile of \gls{td} were also from \gls{nzm}, \gls{sam}, and \gls{ton}.

_BCAS3_ had 17 populations across multiple statistics that had consecutively low score regions. For \gls{fwh} there was \gls{cim} whereas \gls{flf} had \gls{nzm}. \gls{td} also had \gls{nzm}, along with all of the \gls{amr}, \gls{eas}, and \gls{sas} populations. Each of the Polynesian populations had windows in the 1st percentile for \gls{fwh} and \gls{flf}. For \gls{fld} and \gls{td} the \gls{cim} and \gls{nzm} populations both had windows in the 1st percentile whereas of the Western Polynesian populations, \gls{ton} had windows for \gls{fld}, and \gls{sam} had windows for \gls{td}. Only \gls{nzm} from the Polynesian populations had a region of consecutively low score for \gls{ze} with _INHBC_.

The remaining four loci that didn't have consecutively low score regions. For a urate associated locus, _IGFR_ had the most markers that were significant for a haplotypic test for the Polynesian populations. There were five markers for \gls{nsl} in the \gls{ton} population that were significant. \gls{sam} had a single marker and it was different to the significant markers in \gls{ton}. _NFAT5_ had windows for \gls{nzm} from both \gls{td} and \gls{flf}. _PRKAG2_ had windows from \gls{ton} for both \gls{fld} and \gls{flf}, and \gls{sam} for \gls{fld} and \gls{nsl}. _RREB1_ had windows from \gls{cim}, \gls{nzm}, and \gls{ton} for \gls{td}, and only \gls{nzm} and \gls{sam} for \gls{flf}.






```{r, eval = FALSE}
lower_sig_stats %>% filter(SYMBOL %in% urate_goutGR$SYMBOL)
sig_ihs_val %>% filter( genename %in% urate_goutGR$SYMBOL)
sig_nsl_val %>% filter( genename %in% urate_goutGR$SYMBOL)
```


```{r urate_fst, include = FALSE}
fst_map <- findOverlaps(pol_fstGR, urate_goutGR)

# NZM fst
pol_fst %>% slice(fst_map@from) %>% bind_cols(., data.frame(urate_goutGR[fst_map@to], stringsAsFactors = FALSE) ) %>% select(contains('chrom'), SYMBOL,contains('NZM') )%>% gather("pop","fst", 5:NCOL(.))%>% mutate(pop = sapply(pop, function(x){popswitch(x, "NZM")})) %>% mutate(pop1 = substring(pop,1,3), pop2 = substring(pop, 4, 7)) %>% left_join(., panel %>% select('pop1' = pop, super_pop) %>% distinct(), by = 'pop1') %>% distinct() %>% ggplot(., aes(x = chrom_start, y =fst, by = pop1, colour = super_pop)) + geom_line() + facet_wrap(~SYMBOL, scales = 'free') + ggtitle('NZM FST')


```

```{r, eval = FALSE}
# CIM fst
pol_fst %>% slice(fst_map@from) %>% bind_cols(., data.frame(urate_goutGR[fst_map@to], stringsAsFactors = FALSE) ) %>% select(contains('chrom'), SYMBOL,contains('CIM') )%>% gather("pop","fst", 5:NCOL(.))%>% mutate(pop = sapply(pop, function(x){popswitch(x, "CIM")})) %>% mutate(pop1 = substring(pop,1,3), pop2 = substring(pop, 4, 7)) %>% left_join(., panel %>% select('pop1' = pop, super_pop) %>% distinct(), by = 'pop1') %>% distinct() %>% ggplot(., aes(x = chrom_start, y =fst, by = pop1, colour = super_pop)) + geom_line() + facet_wrap(~SYMBOL, scales = 'free') + ggtitle('CIM FST')
```

```{r, eval = FALSE}
# SAM fst
pol_fst %>% slice(fst_map@from) %>% bind_cols(., data.frame(urate_goutGR[fst_map@to], stringsAsFactors = FALSE) ) %>% select(contains('chrom'), SYMBOL,contains('SAM') )%>% gather("pop","fst", 5:NCOL(.))%>% mutate(pop = sapply(pop, function(x){popswitch(x, "SAM")})) %>% mutate(pop1 = substring(pop,1,3), pop2 = substring(pop, 4, 7)) %>% left_join(., panel %>% select('pop1' = pop, super_pop) %>% distinct(), by = 'pop1') %>% distinct() %>% ggplot(., aes(x = chrom_start, y =fst, by = pop1, colour = super_pop)) + geom_line() + facet_wrap(~SYMBOL, scales = 'free') + ggtitle('SAM FST')
```

```{r, eval = FALSE}
# TON fst
pol_fst %>% slice(fst_map@from) %>% bind_cols(., data.frame(urate_goutGR[fst_map@to], stringsAsFactors = FALSE) ) %>% select(contains('chrom'), SYMBOL,contains('TON') )%>% gather("pop","fst", 5:NCOL(.))%>% mutate(pop = sapply(pop, function(x){popswitch(x, "TON")})) %>% mutate(pop1 = substring(pop,1,3), pop2 = substring(pop, 4, 7)) %>% left_join(., panel %>% select('pop1' = pop, super_pop) %>% distinct(), by = 'pop1') %>% distinct() %>% ggplot(., aes(x = chrom_start, y =fst, by = pop1, colour = super_pop)) + geom_line() + facet_wrap(~SYMBOL, scales = 'free') + ggtitle('TON FST')
```




```{r, eval = FALSE}
bind_rows(sig_ihs_val, sig_nsl_val) %>% filter(genename %in% urate_goutGR$SYMBOL) %>% left_join(., markers) %>% select(-ensgeneid) %>% arrange(genename, pop, marker) %>% View()
```



#### Obesity

```{r}
haplo_obesity <- bind_rows(sig_ihs_val, sig_nsl_val) %>% select(SYMBOL = genename, pop) %>% filter(SYMBOL %in% obesity_GR$SYMBOL, pop %in% poly_pop) %>% distinct()
```

<!--consec regions -->
<!-- - td -->
<!--     - chr16  67145002  67465001 320000      * KCTD19        - NZM Tajima.D -->
<!--     - chr3  46035002  46295001 260000      *   CCR3        + TON Tajima.D -->
<!-- - fld -->
<!--     - chr3  52375002  52615001 240000      * NT5DC2        - CIM  Fu.Li.D -->
<!--     - chr3  52375002  52615001 240000      *  PBRM1        - CIM  Fu.Li.D -->
<!--     - chr16   3545002   3795001 250000      *  NLRC3        - CIM  Fu.Li.D -->
<!-- - flf -->
<!--     - chr12 124195002 124505001 310000      * DNAH10        + CIM  Fu.Li.F -->
<!--     - chr12 124195002 124505001 310000      * ZNF664        + CIM  Fu.Li.F -->
<!--     - chr12 124195002 124505001 310000      * CCDC92        - CIM  Fu.Li.F -->
<!--     - chr16   3545002   3795001 250000      *  NLRC3        - CIM  Fu.Li.F -->



<!-- ihs ABO 2, ACAN 1, ADAMTS9 1, ADCY9 7, CCNJL 3, ERBB4 1, EYA2 1, FCER1A 3, GDF5 1, LY86 1, MSRA 1, MTIF3 1, NAV1 1, STXBP6 1 -->
<!-- nsl ADAMTS9 3, ADCY9 6, ARL15 2, CCNJL 3, GDF5 2, KREMEN1 1, MSRA 2, NAV1 1 -->

<!-- NZM  -->
<!-- td CCR3 and KCTD19 -->
<!-- fld ITIH4, LIN28B, NLRC3, GDF5 -->
<!-- flf LIN28B, NLRC3, KCTC19, GDF5 -->
<!-- ze AGBL4, SRPK2 -->
<!-- ihs ABO, ADAMTS9, ADCY9 2, APOA5, ARL15 2, BTNL2 5, CCNJL, CDKAL1, ERBB4,  FHIT 4, FOXO3, FTO, LY86 2, MAP2K5, MTIF3 3, NFE2L3, RABEP1 -->
<!-- nsl ACAN, ADMATS9 3, ADCY9, AGBL4, CCNJL, FHIT, FOXO3, KCTD15, PCSK5, RABEP1 -->

<!-- SAM -->
<!-- td ERBB4, CCR3, DNAH10 -->
<!-- fwh FER, MC4R -->
<!-- fld LIN28B, TLR4, APOA5, KCTD19 -->
<!-- flf LIN28B, TLR4, KCTD19 -->
<!-- ze AGBL4, ITGB6 -->
<!-- ihs ABO, BDNF, BTNL2, ERBB4, EYA2 2, FCER1A, GPRC5B, HSD17B12 2, IQCK, LRP1B 7, PARK2, SLC39A8, TRIP11 -->
<!-- nsl BDNF, CADM2 2, EFEMP1, ERBB4, FOXO3, GBE1, GRPC5B, GRID1, KREMEN1, LRP1B 7, PARK2, PCSK1, SLC39A8 4 -->

<!-- TON -->
<!-- td ERBB4, CCR3, DNAH10 -->
<!-- fwh ERBB4, ZBTB38, MC4R -->
<!-- fld CCR3, HMGCR, APOA5 -->
<!-- flf TMEM18, ERBB4, CCR3 -->
<!-- ihs ABO, APOA5, BDNF, FCER1A, FHIT, FPGT-TNNI3K, GP2, GPRC5B, ITGB6, LEPR, LINGO2, LRP1B 4, LY86, MSRA, PARK2, STXBP6, TNNI3K, TRIP11 3 -->
<!-- nsl ADAMTS17, BDNF, FHIT, FPGT-TNNI3K, GRID1 2, ITGB6, JAZF1, LRP1B 4, MAP2K5, NRXN3, PGPEP1, SLC39A8 2, TNNI3K -->

<!-- Lower sig -->


<!-- CIM -->
<!-- DNAH10 all but ze -->

<!-- NZM -->
<!-- DNAH10 all 5 stats -->

```{r, eval = FALSE}

c("PTPRD", "PRKAG2","NRXN3","MAP2K5","HSD17B12","GRID1","FHIT","ERBB4","BDNF","ARL15","APOA5","AGBL4") %in% urate_goutGR$SYMBOL

c("PTPRD", "PRKAG2","NRXN3","MAP2K5","HSD17B12","GRID1","FHIT","ERBB4","BDNF","ARL15","APOA5","AGBL4") %in% obesity_GR$SYMBOL

c("PTPRD", "PRKAG2","NRXN3","MAP2K5","HSD17B12","GRID1","FHIT","ERBB4","BDNF","ARL15","APOA5","AGBL4") %in% t2d_GR$SYMBOL
```


There were `r length(unique(obesity_GR$SYMBOL))` obesity associated loci, of which, the Polynesian populations had about 20% of the genes showing some evidence of possible selection (Table \@ref(tab:polDiseaseGenesTab)). _ABO_ was the only gene that had significant markers in all four Polynesian populations, and this was only for \gls{ihs}. \gls{cim} had two markers that were significant, the first, rs1053878, was also significant in \gls{ton}. The second marker, rs55764262, was also significant in \gls{nzm} and \gls{sam}, and was in high \gls{ld} (R^2^ = 0.967) with rs549446. Rs549446 was significant in both \gls{chs} and \gls{jpt}. There were no windows from the frequency based statistics that met the significance threshold. _ABO_ has also been associated with marlaria (see section \@ref(malariaRes)).

There were four genes that had markers signifcant in three of the four Polynesian populations, and only for \gls{ihs}, these were: _BTNL2_, _ERBB4_, _FCER1A_, and _LY86_. There were no genes associated with obesity, with significant markers for \gls{nsl}, found in more than two Polynesian populations. There were two genes, _ADAMTS9_ and _ADCY9_, that had significant markers for both \gls{ihs} and \gls{nsl} in the Eastern Polynesian populations of \gls{cim} and \gls{nzm}. _ADAMTS9_ had a single marker \gls{ihs} and \gls{nsl}, three markers for \gls{nsl}. _ADCY9_ had seven significant markers for \gls{ihs} with \gls{cim} and two for \gls{nzm}. \gls{nsl} had 6 markers for \gls{cim} and 1 for \gls{nzm}. _ARL15_ and _MTIF3_ only had markers that were significant with \gls{ihs}, and _CCNJL_ only had markers that were significant with \gls{nsl}. _ADAMTS9_ and _ARL15_ are also associated with \gls{t2d}. The Western Polynesian populations of \gls{sam} and \gls{ton} had the genes _BDNF_ and _LRP1B_ that had significant markers in both \gls{ihs} and \gls{nsl}. _GPRC5B_, _PARK2_, and _TRIP11_ only had markers that were significant for \gls{ihs}. _GRID1_ and _SLC39A8_ only had markers that were significant with \gls{nsl}.

From the obesity associated genes,  _ERBB4_ had the most number of statistics, across the most Polynesian populations, that met the significance threshold. _ERRB4_ had evidence in all Polynesian populations (Figure \@ref(fig:intraSelGenes)) with three of four having evidence from both haplotypic and frequency spectrum methods. Only \gls{ton} did not have evidence from the haplotypic based methods. The Eastern Polynesian populations had the same \gls{snp} (rs1025753) that was significant, whereas \gls{sam} shared the \gls{snp} (rs6707285) with \gls{khv}. All Polynesian populations had windows that met significance for \gls{td} and \gls{fwh}. The Western Polynesian populations had consecutively depressed score regions for \gls{td}. \gls{cim} and \gls{ton} had consecutively depressed regions for \gls{fwh} and only \gls{ton} for \gls{flf}. The only other populations that showed multiple windows with multiple statistics were the \gls{eas} populations, with the exception of \gls{chs}.

_FHIT_ and _GRID_ both had multiple windows for \gls{fwh} that were in the 1st percentile. _FHIT_ also had \gls{snp}s in both \gls{nzm} and \gls{ton} that were significant for \gls{ihs} and \gls{nsl}. \gls{ton} and \gls{nzm} both had five windows in the 1st percentile for \gls{fwh} that intersected _FHIT_. _GRID_ had markers that were significant in only \gls{nsl} and only in the Western Polynesian populations. Both \gls{sam} and\gls{ton} had windows in the 1st percentile for \gls{fwh} that intersected with _GRID1_.

For all Polynesian populations there were multiple windows that intersected _DNAH10_. There were regions of consecutive depressed \gls{td} score for \gls{sam}, \gls{ton}, and \gls{cim} that intersected _DNAH10_. \gls{cim} also had consecutively depressed score regions for \gls{fld} and \gls{flf} at this locus. The main region was actually the upstream region ranging from _TCTN2_ to _DNAH10_. _ATP6V0A2_ was in the middle and where this signal was focussed. For \gls{nzm} _DNAH10_ had significant windows for each of \gls{td}, \gls{fwh}, \gls{fld}, \gls{flf}, and \gls{ze}. \gls{cim} was the same but without \gls{ze}. The Western Polynesian populations did not have the multiple statistic signal. 



- Only CIM had consecutively depressed \gls{td} score that intersected _CCR3_, a which encodes a chemokine receptor.

- The \gls{sam} population had windows from \gls{td}, \gls{fld}, \gls{flf}, and \gls{ze} that were significant and intersected _CREB1_. The other Polynesian populations only had windows in the 1st perecentile for \gls{td} and \gls{ze}.

- In \gls{ton}, there were significant windows that intersected with _NRXN3_ for \gls{td}, \gls{fwh}, \gls{fld}, and \gls{flf}. 

AGBL4 had 17 windows for \gls{ze} that were in the 1st percentile in \gls{nzm} and had a significant marker for \gls{nsl}

<!-- consec regions for td in sam and ton -->
<!-- consec regions for fwh in cim and ton -->
<!-- consec regoins in flf in ton -->

<!-- cim - td and fwh -->
<!-- nzm - td and fwh -->
<!-- sam - td and fwh -->
<!-- ton - td, flf, and td -->

<!-- ihs in cim, nzm, and sam -->
<!-- nsl in sam -->


<!-- eas, except chs had multiple windows for fwh, td, flf -->

The Eastern Polynesian populations of \gls{cim} and \gls{nzm} had significant results for _ADCY9_. \gls{cim} had 7 markers that were significant for \gls{ihs} and 6 markers for \gls{nsl}. \gls{nzm} had 2 markers for \gls{ihs} and a single marker for \gls{nsl}. The Western Polynesian populations on the other hand had 

```{r, eval = FALSE}
lower_sig_stats %>% filter(pop %in% poly_pop, SYMBOL == "ERBB4") %>% select(SYMBOL, pop) %>% group_by(SYMBOL, pop) %>% tally() %>% spread(pop, n)
lower_sig_stats_binned %>% filter(pop %in% poly_pop, chrom_end >= 124125002 & chrom_start <=124265001,  chrom == 'chr12') %>% ggplot(., aes(x = chrom_start, y = value)) + facet_wrap(statname~pop) + geom_line() + geom_point() + theme_bw()
consec_lower_regions %>% filter(pop == 'NZM', SYMBOL %in% obesity_GR$SYMBOL) 

sig_nsl_val %>% select(SYMBOL = genename, everything(), pop) %>% filter(SYMBOL %in% obesity_GR$SYMBOL, pop == "TON") %>% group_by(SYMBOL) %>% tally()
```


```{r, eval = FALSE}
g <- c("ABO", "BTNL2", "ERBB4", "FCER1A", "LY86","BDNF", "GPRC5B", "LRP1B", "PARK2", "TRIP11", "BDNF", "GRID1", "LRP1B", "SLC39A8", "ADAMTS9", "ADCY9", "ARL15", "MTIF3", "ADAMTS9", "ADCY9", "CCNJL" )
unique(g[g %in% unique(obesity_GR$SYMBOL[obesity_GR$SYMBOL %in% t2d_GR$SYMBOL])])
```



<!-- All -->
<!-- ihs: ABO -->
<!-- nsl:  -->

<!-- 3/4 -->
<!-- ihs: BTNL2, ERBB4, FCER1A, LY86,   -->

<!-- Western Polynesians -->
<!-- ihs: BDNF, GPRC5B, LRP1B, PARK2, TRIP11 -->
<!-- nsl: BDNF, GRID1, LRP1B, SLC39A8 -->

<!-- Eastern Polynesians -->
<!-- ihs: ADAMTS9, ADCY9, ARL15, MTIF3,  -->
<!-- nsl: ADAMTS9, ADCY9, CCNJL,  -->

<!-- TD:  -->
<!-- all pops ARL15, CCR3, CREB1, DNAH10, ERBB4,  -->

<!-- FWH all pops: ERBB4,FHIT, GRID1, LTBP1, ZBTB38 -->
<!-- ep: DNAH10 -->

<!-- FLD: no all pops -->
<!-- FLF: no all pops -->

```{r fst_obesity, eval = FALSE}
fst_map_obesity <- findOverlaps(pol_fstGR, obesity_GR)

# NZM fst
pol_fst %>% slice(fst_map_obesity@from) %>% bind_cols(., data.frame(obesity_GR[fst_map_obesity@to], stringsAsFactors = FALSE) ) %>% filter(SYMBOL == 'ADCY9')%>% select(contains('chrom'), SYMBOL,contains('NZM') )%>% gather("pop","fst", 5:NCOL(.))%>% mutate(pop = popswitch(pop, "NZM")) %>% mutate(pop1 = substring(pop,1,3), pop2 = substring(pop, 4, 7)) %>% left_join(., panel %>% select('pop1' = pop, super_pop) %>% distinct(), by = 'pop1') %>% distinct() %>% ggplot(., aes(x = chrom_start, y =fst, by = pop1, colour = super_pop)) + geom_line() + facet_wrap(~SYMBOL, scales = 'free') + ggtitle('NZM FST')

# CIM fst
pol_fst %>% 
  slice(fst_map_obesity@from) %>% 
  bind_cols(., data.frame(obesity_GR[fst_map_obesity@to], stringsAsFactors = FALSE) ) %>% 
  filter(SYMBOL == 'ADCY9') %>% 
  select(contains('chrom'), SYMBOL,contains('CIM') )%>% 
  gather("pop","fst", 5:NCOL(.))%>% 
  mutate(pop = sapply(pop, function(p){popswitch(p, "CIM")})) %>% 
  mutate(pop1 = substring(pop,1,3), pop2 = substring(pop, 4, 7)) %>% 
  left_join(., panel %>% select('pop1' = pop, super_pop) %>% 
              distinct(), by = 'pop1') %>% 
  distinct() %>% 
  ggplot(., aes(x = chrom_start, y =fst, by = pop1, colour = super_pop)) + geom_line() + facet_wrap(~SYMBOL, scales = 'free') + ggtitle('CIM FST')
```

#### T2D

```{r}
haplo_t2d <- bind_rows(sig_ihs_val, sig_ihs_val) %>% select(SYMBOL = genename, pop) %>% filter(SYMBOL %in% t2d_GR$SYMBOL, pop %in% poly_pop) %>% distinct()
```


\Gls{t2d} is a co-morbidity of hyperuricemia. There is one locus identified from \gls{gwas} for \gls{t2d} that is common with urate associated loci, and 16 in common with obesity. For the Polynesian populations there were `r lower_sig_stats %>% filter(SYMBOL %in% t2d_GR$SYMBOL, pop  %in% poly_pop) %>% select(SYMBOL) %>% distinct() %>% NROW()` loci that were intersected by windows in the 1st percentile of a frequency based statistic. There were `r bind_rows(sig_ihs_val, sig_nsl_val) %>% select(SYMBOL = genename, pop) %>% filter(SYMBOL %in% t2d_GR$SYMBOL, pop  %in% poly_pop) %>% select(SYMBOL) %>% distinct() %>% NROW()` genes that had a marker with a significant value for \gls{ihs} or \gls{nsl}.


The Western Polynesian populations had the fewest significant markers in loci associated with \gls{t2d}. \gls{sam} had the minimum, out of all populations, with only 4 \gls{snp}s (Table \@ref(tab:intraSelPol)). There were no genes associated with \gls{t2d} that had haplotypic evidence in both of the Western Polynesian populations.

In the Eastern Polynesian populations, as mentioned there was _ADAMTS9_ and _ARL15_ also associated with obesity, but there was also _LPP_, _PSMD6_, _ZFAND6_ that had markers that were significant for both \gls{ihs} and \gls{nsl}. _LPP_ had three markers for \gls{ihs} that were significant in \gls{cim}, and six that were significant in \gls{nzm}. For \gls{nsl} there was a single marker that was significant in \gls{cim}, \gls{nzm} and also \gls{ton} 

Aside from the previously mentioned genes of XXX taht were associated with both obesity and \gls{t2d}

_PTPRD_ was the only gene associated with \gls{t2d} that had significant windows and \gls{snp}s in both frequency spectrum and haplotypic methods. All four Polynesian populations had significant \gls{snp}s for \gls{ihs} and all except \gls{nzm} had significant markers for \gls{nsl}. For \gls{ihs}, there were 14 markers that were significant in \gls{cim}, five in \gls{ton}, three in \gls{sam}, and a single marker in \gls{nzm}. With \gls{nsl}, \gls{cim} and \gls{ton} had four markers that were significant and there were three in \gls{sam}. \gls{ton} and \gls{cim} had three windows for \gls{td} that were in the 1st percentile. \gls{nzm} had a single window for both \gls{fwh} and \gls{td} that was in the 1st percentile and intersected _PTPRD_.





_RREB1_ is shared with urate and as previously mentioned had significant windows for \gls{td}, \gls{fld}, and \gls{flf}.

Loci that were in common with obesity included _ARL15_ which had significant windows for all Polynesian populations for \gls{td} and both rs1712139 and rs26775 had significant \gls{ihs} values in the Eastern Polynesian populations. The same markers also had significant \gls{nsl} in \gls{cim}. 


_FTO_ had a significant window for \gls{fld} in the \gls{ton} population and \gls{nzm} had rs4396532 that was significant for \gls{ihs}. _HMGA2_ had significant windows with \gls{fwh} for \gls{cim}, \gls{nzm} and \gls{sam}. _MC4R_ had significant windows for \gls{fwh} in the Western Polynesian populations. The \gls{ton} population had significant windows for \gls{fwh} that intersected _TCF7L2_. And \gls{nzm} had significant windows for \gls{ze} that intersected _GRB14_.

Loci that only had significant windows in the frequency based statistics included _THADA_, all Polynesian populations had significant windows for \gls{fwh}. \gls{cim}, \gls{nzm}, and \gls{ton} also had windows for \gls{ze}. Both \gls{cim} and \gls{ton} had significant windows in \gls{fld} and \gls{flf}. \gls{ton} was the only Polynesian population to also have 
There were regions of consecutively depressed score for \gls{fwh} with \gls{cim} and \gls{ton}. \gls{ton} also had consecutively depressed score regions for \gls{td}.


 
- ARL15, SSR1 for cim and nzm with 2 markers each for ihs, cim had2 markersfor ARL15 for nsl, both had 2 for nsl in SSR1

- MACF1 had windows that were in the 1st percentile for all Polynesian populations for \gls{ze}. \gls{nzm}, \gls{sam}, and \gls{ton} also had windows for \gls{fld} and \gls{flf}. \gls{sam} was the only Polynesian population to have windows in the 1st percentile for \gls{td}.


<!-- all -->
<!-- ihs: PTPRD -->
<!-- nsl none -->

<!-- 3 -->
<!-- ihs: none -->
<!-- nsl: PTPRD -->

<!-- west -->
<!-- ihs: none -->
<!-- nsl: none -->

<!-- east -->
<!-- ihs: ADAMTS9, ARL15, LPP, PSMD6, SSR1, ZFAND6 -->
<!-- nsl: ADAMTS9, LPP, PSMD6, ZFAND6  -->

<!-- ARL15 - cim, nzm, sam, ton for td -->
<!-- FTO - ton fld -->
<!-- HMGA2 cim, nzm, sam fwh -->
<!-- MC4R sam ton fwh -->
<!-- TCF712 ton fwh -->
<!-- GRB14 nzm ze -->

<!-- MACF1: -->
<!-- sam, ton, nzm - flf, flf, ze -->
<!-- only td for sam -->
<!-- only ze for cim -->


<!-- THADA - ton all 5 stats -->
<!-- cim fwh, fld, flf, ze -->
<!-- nzm - fwh, ze -->
<!-- sam - fwh -->

<!-- SND1 - nzm - fld, flf, td, ze -->
<!-- cim - td -->
<!-- sam - flf, td -->






```{r, eval = FALSE}
lower_sig_stats %>% filter(SYMBOL %in% t2d_GR$SYMBOL, pop %in% poly_pop) %>% group_by(pop, SYMBOL, statname) %>% tally() %>% spread(statname, n, "") %>% arrange(SYMBOL)
```

The Western Polynesian populations had consecutively depressed regions for \gls{fwh} for _IRS1_ and _MC4R_

```{r, eval = FALSE}
consec_lower_regions %>% filter(SYMBOL %in% t2d_GR$SYMBOL, pop %in% poly_pop) %>% arrange(SYMBOL, pop)
```
<!-- ALR15, FTO, GRB14, HMGA2, MC4R, TCF7L2 are all shared with obesity -->
<!-- ADAMATS9, CDKAL1, ITGB6, JAZF1 too -->
#### Kidney Disease

```{r}
haplo_kd <- bind_rows(sig_ihs_val, sig_ihs_val) %>% select(SYMBOL = genename, pop) %>% filter(SYMBOL %in% kd_GR$SYMBOL, pop %in% poly_pop) %>% distinct()
```


CIM

\gls{fld} and \gls{flf} had SETDB1
\gls{ze} had ATXN2 and DACH1
\gls{fwh} has ALMS1
\gls{fld}, \gls{flf}, \gls{td} and \gls{fwh} all had significant windows that intersected BCAS3. \gls{fwh} also had a 100 kb region of consecutively depressed score that intersected BCAS3. 


No genes that had ihs in multiple polynesian populations for \gls{ihs}. The Western Polynesian populations had two markers that were significant for \gls{nsl} in WDR72. \gls{ton} also had a significant marker for \gls{ihs} at this locus.


\gls{nzm} had three markers that were significant for \gls{ihs} at PHTF2 and a single significant marker for \gls{nsl}.

#### Metabolic Syndrome

#### Malaria {#malariaRes}
```{r, eval = FALSE}
lower_sig_stats %>% filter(SYMBOL %in% malaria_GR$SYMBOL)
```



```{r, eval = FALSE}
sig_ihs_val %>% filter(genename %in% malaria_GR$SYMBOL) %>% select(-gene_chrom, -statid) %>% left_join(., markers, by = c('chrom','chrom_start')) %>% arrange(genename, marker, pop)
```

Uric acid has a central role in malarial infection and triggering an immune response [@Gallego-Delgado2014]. Given that immune response is a strong candidate to be under selective pressure, malaria associated genes were looked at to see if there were signals of selection. From the GWAS catalog there were three genes that had been associated with malaria that also had evidence of selection with the Polynesian populations. 

The main evidence was found in \gls{nzm} with seven markers 

There were three genes that were associated with malaria and also had significant ihs scores, these were only in 

In ABO, rs8176741 had been associated with severe malaria (PMID:22895189), this was a significant ihs in FIN . The other snps that showed significance for ihs were in no in LD with the rs55764262, which CIM NZM and SAM all had a significant result for (EAS R^2^ 0.002).

Both sam and nzm had snps that were significant for ihs in ATP2B4, sam had rs10494845, and nzm had rs142206068.

```{r, eval = FALSE}
sig_nsl_val %>% filter(genename %in% malaria_GR$SYMBOL) %>% select(-gene_chrom, -statid) %>% left_join(., markers, by = c('chrom','chrom_start'))
```

The \gls{nzm} has multiple markers within the malaria associated gene, _DDC_, for \gls{nsl}. The markers were rs2060762, rs3807558, rs2329371, and rs1966839

rs1966839 (ref = T, alt/anc = C)is in high LD (0.998) with the malaria associated rs1451375 (ref/anc = C, alt = A) where T=C and A=C. both are intron variants . 

(ref:ldDDCnzm) Linkage disequilibrium of \gls{ihs} significant SNPs (rs2060762, rs11238131, rs3807558, rs2329371, and rs1966839) in \gls{nzm} at the _DDC_ locus with malaria (rs1451375), BMI, (rs10499694), and blood metabolite ratios (rs7809234).

```{r '(ref:ldDDCnzm)'}
knitr::include_graphics('images/03_selectionpipeline/LDplot-EAS_DDC_nzm_ihs.png')
```


```{r, eval = FALSE}
pol_fst %>% filter(chrom == 7, chrom_start > 49e6 & chrom_start < 52e6)


```


#### Neurological

#### Psycological

#### Immune

```{r, eval = FALSE}
lower_sig_stats %>% filter(SYMBOL %in% zhang_immune_GR$SYMBOL, pop %in% poly_pop) %>% group_by(pop, statname, SYMBOL) %>% tally() %>% spread(statname, n) %>% arrange(SYMBOL)
```

```{r, eval = FALSE}
consec_lower_regions %>% filter(pop %in% poly_pop, SYMBOL %in% zhang_immune_GR$SYMBOL) %>% select(seqnames, start, end, width, stat, pop) %>% distinct() %>% spread(key = stat, value = stat, fill = "")
```


## Chapter Discussion
<!--
Important to remember that the modern day populations esp AFR won't necessarily have the same results as AFR pops did 100 kya

Pipeline discussion - how were the results?
  - Are the genes that are picked up similar to what has been discovered?

Is there anything special about the population comparisons?
  - shared genes between similar populations?
  - East/West split seen?
  - NZC vs GBR similarity?
  
  @Rafajlovic2014 has a good talk about the use of the empirical distribution for sfs stats.
  
  
Because fld and flf are based on the number of singletons, how is this going to be affected by using chip data that should have essentially 0. presumable this will dramatically affect the interpretability of the statistic and push everything over to the right but is it acceptable to recentre the distribution and look that the extremes from there? does it have biological plausibility?  

what does it mean to have a significant window of one tail neighboured by the other tail? Or on the broad scale what does it mean to have a gene classed as significant in both tails?

effect of phasing when polynesian populations aren't represented in the 1000 genomes populations






-->

Conclusions:

The results of this study show there is indeed evidence of selection in urate associated loci in the Polynesian populations. This evidence however is not specific to urate. This is shown by the fact that there is little signal of selection appearing at the urate transporters themselves, but at the loci that are part of more general metabolic pathways. The reasons for why there would be selection still remain unclear.

There was also evidence of selection in loci that were associated with malaria and were also associated with \gls{t2d} and obesity.

- there is evidence of selection in urate associated loci in the Polynesian populations
- this evidence is not specific to urate however, this is evidenced by the fact that there is little signal in the urate transporters but it is in the loci that are part of more general metabolic pathways
- the reasons as to why there would be selection are unclear
- there is evidence of selection in malaria related loci, that are also associated with T2D and obesity.
- The pathways that showed enrichment were involved voltage-gated Ca channel. these channels are predominantly expressed in the lung and the heart.
- many of the pathways were involved with cell signalling/communication so are vital for life

#### limitations

limitations of using the chip data versus imputed data
- ascertainment bias of snps on the chip @nielsen2007recent
- number of snps was low compared to other analysis - that were either in seq data or used hapmap
- imputation from the chip might not work well given the polys aren't (unconfirmed) well represented in the reference panel
- phasing of haplotypes doesn't have polynesian haplotypes in the panel
- window size meant that snps that snps not in the window might not even be near the actual window ie -45 and +45 kb away from the edge of the defined window in the worst case. This would lead to some difficulty in definitive location. There was a trade off between snp density and window size.
- only looking at positive selection

One of the methods to still use the selection and neutrality tests with the genome-wide data was to use the empirical distribution [@Teshima2006; @Clarke?]


When it came to identifying the 

#### Selection of Urate Associated Genes

The evidence for selection in urate does not appear to be specific for urate itself, but for loci that are involved in more general metabolic pathways. _BCAS3_ featured 



The main effect loci identified through GWAS, _SLC2A9_ and _ABCG2_, had limited evidence of positive selection. 

ABCG2 had a significant marker (rs2622626) with \gls{ihs} and \gls{nsl} and only in \gls{ton} from the Polynesian populations, the only other populations were \gls{gih} and \gls{itu} from the \gls{sas} super population. 

_SLC2A9_ 


> Have to be careful for conclusions that are drawn for the TON and CIM populations as these both have a gout prevalence of ~50% which is 5 fold higher than would be expected in a random population sample.

The shift in the distributions due to having chip data for the selection tests that rely on frequency spectrum is seen here. Mostly the African populations have shifted so far to the right with \gls{td} that the first percentile is > 0. 

XXX The intermediate frequency variants are over represented

With the adaptation of the selection and neutrality tests into genome-wide usage, specifically the sliding window approach, there was the need to balance the number of segregating sites with the window size [@Pybus2014]. This balance is entirely influenced by \gls{snp} density with whole genome (re)sequence data offering high density XXX snps/kb, but \gls{snp} chip data being less dense XXX snps/kb. This means that window size needs to increase for \gls{snp} chip data in order to have the same number of segregating sites. However, as window size increases it also reduces the resolution of the genome and with it the ability to associate a window with particular genomic features.

- Similar to @Grossman2013, we found that BCAS3 had evidence of selection

#### Thrifty gene hypothesis

As discussed in @Gosling2014, the thrifty gene hypothesis in the context of the Pacific 

- Gosling2014 In the pacific context, the history doesn't fit with the thrifty narrative
- Cadzow2016 increased sample size did not repeat the association nor was there convincing evidence for selection


CREBRF

- CREBRF did not have the specific variant the @Minster2016.

The variant raises BMI but is protective for T2D

CHECK -  there was only a single variant that was Samoan specific, however generally we don't see a single variant with a high score in the main reported loci, it is a cluster, eg LCT has a large number of variants.


PPARGC1A

- @Cadzow2016 did not manage to replicate the findings of Myles2011
- there were no snps in the polynesian populations that met the significance threshold


The use of the gwas catalog - the gwases are largely conducted in European populations so might miss some population specific variants/genes (eg Diabetes and Mexicans?)


Zheng2013
SH2B3-ATXN2 has snps associated with urate, and other diseases (rs3184504, rs4766578, rs653178, and rs11065987) are in close LD. derived allele haplotype was associated with increased risk of autoimmune, bp, and urate. haplotype was common in European but not in afr or asi. possible recent sweep due to bacterial infection @Zhernakova2010. The locus also extends to ALDH2 which @Oota2004 suggests is under selection

#### Malaria

[@Gallego-Delgado2014]

Refer to malaria section of results

Big hit in DDC with NZM, ABO and ATP2B4,
those are gwas associated genes

from the obesity list, there was ADCY9, which had a very strong signal for \gls{ihs} for 7 markers in \gls{cim} and 2 markers in \gls{nzm} and were the only populations with significant ihs values. 
A variant, rs10775349, in ADCY9 has been associated with malaria

rs879620 associated with bmi

The malarial resistance gene _HBB_ (identified in [@Currat2002; @Ohasi2004]) was not identified as having evidence of selection




#### possibly useful papers

- When Does Frequency-Independent Selection Maintain Genetic Variation? - Novak2017
- Toward a new history and geography of human genes informed by ancient DNA - Pickrell2017
- Genetic Drift in an Infinite Population: The Pseudohitchhiking Model - Gillespie2000
- Detecting ancient positive selection in humans using extended lineage sorting - Peyregne2017
- A New Test for Detecting Recent Positive Selection that is Free from the Confounding Impacts of Demography - Li2010
- Understanding rare and common diseases in the context of human evolution - Quintana-Murci2016
- Soft sweeps are the dominant mode of adaptation in the human genome - Schrider2017
- Demography-adjusted tests of neutrality based on genome-wide SNP data - Rafajlovic2014
- Decomposing the Site Frequency Spectrum: The Impact of Tree Topology on Neutrality Tests - Ferretti2017
- Statistical Power Analysis of Neutrality Tests Under Demographic Expansions, Contractions and Bottlenecks With Recombination - Ramirez-Soriano2008
- Genetic loci associated with coronary artery disease harbor evidence of selection and antagonistic pleiotropy - Byars2017
- Demographic history, selection and functional diversity of the canine genome - Ostrander2017
- Admixture, Population Structure, and F-Statistics - Peter2016
- Population History and Natural Selection Shape Patterns of Genetic Variation in 132 Genes - akey2004population
- Determinants of genetic diversity - Ellegren2016
- Reconstituting the Frequency Spectrum of Ascertained Single-Nucleotide Polymorphism Data - Nielsen2004
- Correcting for ascertainment biases when analyzing SNP data: applications to the estimation of linkage disequilibrium - Nielsen2003

<!--
gene lists
metabolic - covered already
immune - innate: crohns, bohcets, coeliac
immune - adaptive: t1d, RA, MS, graves, hushimotosis
cancer
pyschicatric: SZ
neurological: AD, PD
-->


<!--
## SelectionTools pipeline

for the selection pipeline we need to split into populations

```{r, engine = 'bash', eval = FALSE}
# working dir: /Volumes/BiochemXsan/scratch/merrimanlab/murray/working_dir/coreExome_selection/NZ_coreExome_1kgp/data/
# make a combined panel file
cut -f1-3 /Volumes/BiocArchive/archive/merrimanlab/reference_files/VCF/1000Genomes_vcf_files/Phase3_March2017/integrated_call_samples_v3.20130502.ALL.panel |\
cat - NZ_coreExome/*panel > nz_1kgp.panel

mkdir Indiv_pops

parallel '
  bcftools view \
    -s $(grep {1} nz_1kgp.panel |\
      cut -f 1 |\
      tr "\n" "," |\
      sed "s/,$//g") \
    -O z \
    -o Indiv_pops/{1}.chr{2}.phased.vcf.gz NZ_1KGP.chr{2}.phased.sample_updated.vcf.gz
  ' ::: $(cut -f 2 nz_1kgp.panel |grep -v "pop" | sort |uniq) ::: $(seq 1 22)
```

Break into super populations to analyse the Polynesians (NZM,CIM,TON,SAM) - make sure to exclude the NZC from the EUR super pop and also exclude the admixed NZM (NAD), Eastern Polynesians (EPN), and Western Polynesians (WPN).

```{r, engine = 'bash', eval = FALSE}
# working dir: /Volumes/BiochemXsan/scratch/merrimanlab/murray/working_dir/coreExome_selection/NZ_coreExome_1kgp/data/
mkdir Super_pops

parallel '
  bcftools view \
    -s $(grep {1} nz_1kgp.panel |\
      grep -v "NAD\|EPN\|NZC\|WPN" |\
      cut -f 1 |\
      tr "\n" "," |\
      sed "s/,$//g") \
    -O z \
    -o Super_pops/{1}.chr{2}.phased.vcf.gz NZ_1KGP.chr{2}.phased.sample_updated.vcf.gz
  ' ::: $(cut -f 3 nz_1kgp.panel |grep -v "pop" | sort |uniq) ::: $(seq 1 22)
```

Now send to NeSI pan cluster for the selectionTools pipeline to take place

Individual populations are first run through selectionTools with iHS, nSL, and XP-EHH calculated later

```{r, engine = 'bash', eval = FALSE}
# on nesi
mkdir -p /home/murray.cadzow/uoo00008/NZ_1KGP_unimputed/{Indiv_pops,Indiv_pops_results}

# locally
parallel '
  scp {} murray.cadzow@login.uoa.nesi.org.nz:~/uoo00008/NZ_1KGP_unimputed/Indiv_pops/ 
  ' ::: $(ls /media/xsan/scratch/merrimanlab/murray/working_dir/coreExome_selection/NZ_coreExome_1kgp/data/Indiv_pops/*gz)

# on nesi
# working dir: /home/murray.cadzow/uoo00008/NZ_1KGP_unimputed/Indiv_pops

# find all the vcfs and use the first prefix to use as the population code
for p in $(ls * | cut -d'.' -f1 | sort | uniq)
do 
  sbatch ../unimputed_selection_pipeline.sl $p ; sleep 2; 
done
```

Then run run_selscan.sl and run_nsl_selscan on each population results file
```{r, engine = 'bash', eval = FALSE}
# on nesi
# working dir: Indiv_pops_results
for p in $(find ./[A-Z]* -type d | grep -v xpehh | sed 's/\.\///g')
do 
  cd $p 
  sbatch -J ihs_$p ../../run_selscan.sl $p
  sbatch -J nsl_$p ../../run_nsl_selscan.sl $p
  cd ../
  sleep 5
done
```

```{r, engine = 'bash', eval = FALSE}
# on nesi
# working dir: /home/murray.cadzow/uoo00008/NZ_1KGP_unimputed/Indiv_pops_results
# individual pops
for p1 in CIM NZC NZM SAM TON
do
  for p2 in ACB ASW BEB CDX CEU CHB CHS CIM CLM EPN ESN FIN GBR GIH GWD IBS ITU JPT KHV LWK MSL MXL NAD NZC NZM PEL PJL PUR SAM STU TON TSI WPN YRI
  do
    if [[ $p1 != $p2 ]]
    then
      echo $p1 $p2
      sbatch ../run_xpehh.sl $p1 $p2
      sleep 5
    fi
  done
done


for p1 in EPN NAD WPN
do
  for p2 in ACB ASW BEB CDX CEU CHB CHS CIM CLM EPN ESN FIN GBR GIH GWD IBS ITU JPT KHV LWK MSL MXL NAD NZC NZM PEL PJL PUR SAM STU TON TSI WPN YRI
  do
    if [[ $p1 != $p2 ]]
    then
      echo $p1 $p2
      sbatch ../run_xpehh.sl $p1 $p2
      sleep 5
    fi
  done
done




# super pops
for p in EUR EAS SAS AMR AFR
do
  sbatch ../run_xpehh.sl POL $p
  sleep 5
done

# Don't actually want the comparisons between NAD and EPN or WPN, or the ones for EPN/WPN with NZM/CIM/SAM/TON
rm -r NAD_{WPN,EPN}
rm -r {EPN,WPN}_{NZM,CIM,SAM,TON}
```
-->