# SelectionTools Pipeline

## Introduction

<!--
motivation: seemed to be little in the way of calculating stats genome wide and wasn't sure how sensitive these stats were to snp density



This chapter is about trying to first of all establish how sensitive the various statistics were to window/slide variation

from the literature there doesn't seem to be any recommendation as to how to compute these statistics genome wide. Up until recently the various statistics were computed at the 'feature' level, usually the gene.

Fst has a paper in early 2000's from Bruce Weir that touched on what sort of size to use howver it largely just came down to 'this window size seems to have enough of a smoothing effect as well as display some of the variation in the data' type approach and was largely just based on a this looks good enough.

Pybus et al specified their window sizes which was roughly 30 kb window and 3kb slide for most statistics however I don't remember reading how they arrived at these sizes.

In cadzow 2014 I tried had tried a few sizes and it it seemed that there was little difference between them other than the smaller the window the more noisy the data looked.

While I was playing around with this for Tajima's D there does seem to be at least some relation to the density of the snps and the window size, which does intuitively make sense, in that if there are none or only 1 snp per window then the stats are going to be pretty poor.

From the core exome work it looks as though 30 kb was ok and at least when the distribution of TD was binned by number of snps per window it didn't look to bad and the overall total distribution of TD looks not too bad. Increasing from 30 kb didn't really change TD that much. the extremes become more extreme but the means increaseas window size increases.
-->

<!-- ascertainment bias will skew selection statistics that are site frequency spectrum dependant in chip data due to the over-representation of intermediate frequency SNPs, these include Tajima's D (upward) and Fay and Wu's H [@Clark2005]


Carlson2005 did windows of 100kb, slide 10kb and mean snps were ~50/window. ascertainment bias esp in the +ve end. continous region on >20 windows where 75% were in bottom 1% was defined as contiguous regions of Tajimas D reduction.
RafajloviÄ‡2014 largely followed Carlson2005, they also discarded windows with less than 5 snps.

Bigman2010 has a custom Tajima's D that is created to be similar to iHS in terms of calculation
-->

literature
Cadzow2014
Cadzow2016
Pybus2014

### objectives
Establish the influence of key parameters in selectionTools pipeline

## Selection dataset creation

## SelectionTools pipeline

for the selection pipeline we need to split into populations

```
# working dir: /Volumes/BiochemXsan/scratch/merrimanlab/murray/working_dir/coreExome_selection/NZ_coreExome_1kgp/data/
# make a combined panel file
cut -f1-3 /Volumes/BiocArchive/archive/merrimanlab/reference_files/VCF/1000Genomes_vcf_files/Phase3_March2017/integrated_call_samples_v3.20130502.ALL.panel |\
cat - NZ_coreExome/*panel > nz_1kgp.panel

mkdir Indiv_pops

parallel '
  bcftools view \
    -s $(grep {1} nz_1kgp.panel |\
      cut -f 1 |\
      tr "\n" "," |\
      sed "s/,$//g") \
    -O z \
    -o Indiv_pops/{1}.chr{2}.phased.vcf.gz NZ_1KGP.chr{2}.phased.sample_updated.vcf.gz
  ' ::: $(cut -f 2 nz_1kgp.panel |grep -v "pop" | sort |uniq) ::: $(seq 1 22)
```

Break into super populations to analyse the Polynesians (NZM,CIM,TON,SAM) - make sure to exclude the NZC from the EUR super pop and also exclude the admixed NZM (NAD), Eastern Polynesians (EPN), and Western Polynesians (WPN).

```
# working dir: /Volumes/BiochemXsan/scratch/merrimanlab/murray/working_dir/coreExome_selection/NZ_coreExome_1kgp/data/
mkdir Super_pops

parallel '
  bcftools view \
    -s $(grep {1} nz_1kgp.panel |\
      grep -v "NAD\|EPN\|NZC\|WPN" |\
      cut -f 1 |\
      tr "\n" "," |\
      sed "s/,$//g") \
    -O z \
    -o Super_pops/{1}.chr{2}.phased.vcf.gz NZ_1KGP.chr{2}.phased.sample_updated.vcf.gz
  ' ::: $(cut -f 3 nz_1kgp.panel |grep -v "pop" | sort |uniq) ::: $(seq 1 22)
```

Now send to NeSI pan cluster for the selectionTools pipeline to take place

Individual populations are first run through selectionTools with iHS, nSL, and XP-EHH calculated later

```
# on nesi
mkdir -p /home/murray.cadzow/uoo00008/NZ_1KGP_unimputed/{Indiv_pops,Indiv_pops_results}

# locally
parallel '
  scp {} murray.cadzow@login.uoa.nesi.org.nz:~/uoo00008/NZ_1KGP_unimputed/Indiv_pops/ 
  ' ::: $(ls /media/xsan/scratch/merrimanlab/murray/working_dir/coreExome_selection/NZ_coreExome_1kgp/data/Indiv_pops/*gz)

# on nesi
# working dir: /home/murray.cadzow/uoo00008/NZ_1KGP_unimputed/Indiv_pops

# find all the vcfs and use the first prefix to use as the population code
for p in $(ls * | cut -d'.' -f1 | sort | uniq)
do 
  sbatch ../unimputed_selection_pipeline.sl $p ; sleep 2; 
done
```

Then run run_selscan.sl and run_nsl_selscan on each population results file
```
# on nesi
# working dir: Indiv_pops_results
for p in $(find ./[A-Z]* -type d | grep -v xpehh | sed 's/\.\///g')
do 
  cd $p 
  sbatch -J ihs_$p ../../run_selscan.sl $p
  sbatch -J nsl_$p ../../run_nsl_selscan.sl $p
  cd ../
  sleep 5
done
```



```
# on nesi
# working dir: /home/murray.cadzow/uoo00008/NZ_1KGP_unimputed/Indiv_pops_results
# individual pops
for p1 in CIM NZC NZM SAM TON
do
  for p2 in ACB ASW BEB CDX CEU CHB CHS CIM CLM EPN ESN FIN GBR GIH GWD IBS ITU JPT KHV LWK MSL MXL NAD NZC NZM PEL PJL PUR SAM STU TON TSI WPN YRI
  do
    if [[ $p1 != $p2 ]]
    then
      echo $p1 $p2
      sbatch ../run_xpehh.sl $p1 $p2
      sleep 5
    fi
  done
done


for p1 in EPN NAD WPN
do
  for p2 in ACB ASW BEB CDX CEU CHB CHS CIM CLM EPN ESN FIN GBR GIH GWD IBS ITU JPT KHV LWK MSL MXL NAD NZC NZM PEL PJL PUR SAM STU TON TSI WPN YRI
  do
    if [[ $p1 != $p2 ]]
    then
      echo $p1 $p2
      sbatch ../run_xpehh.sl $p1 $p2
      sleep 5
    fi
  done
done




# super pops
for p in EUR EAS SAS AMR AFR
do
  sbatch ../run_xpehh.sl POL $p
  sleep 5
done
```

Don't actually want the comparisons between NAD and EPN or WPN, or the ones for EPN/WPN with NZM/CIM/SAM/TON
```
rm -r NAD_{WPN,EPN}
rm -r {EPN,WPN}_{NZM,CIM,SAM,TON}

```

## PopGenome

### Comparision of window sizes
<!--

This is at least covered for TD in the R notebook

window slide was equal to the window size for each size tested.

The rough results are that because of the density of the SNPs on the chip the windows tested 1,5,10,30,50,100 KB for 1,5,10 there is not a great number of SNPs per window. 

The 1 kb window provides a very large number of windows with no snps present and when plotted looks like there is much fewer windows than the 5kb


the window size of 30kb starts to offer a 


-->

### Comparision of window slide



## Comparison of selectionTools Parameters

### Window size
Tajima's _D_, Fay and Wu's _H_, and F~ST~ calculation window size was tested.

- TD = 1,5, and 30 kbp windows
- FWH = 1,5, and 30 kbp windows
- FST = 1 bp and 1 Mbp

iHS and nSL
tested gap width to try and bring in SLC2A9 for coreExome data

### Slide


## Comparison of Imputation Thresholds

## Simulations

### Bottleneck

### Migration

### Expansion

### Positive Selection

### Balancing Selection

### Negative Selection


## Results
<!--
Look at what windows/genes for each population are interesting
Super populations - 1KGP vs POL
Super Populations - 1KGP vs EPN/WPN
Individual populations - CIM/NZM/SAM/TON/NZC
-->
### Intra

#### Tajima's D

The mean Tajima's D for each chromosome can be seen in Table \@ref(tab:meanTDtable) and is found in the appendix

#### Fay and Wu's H

#### Zeng's E

#### iHS/nSL

### Inter

#### F~ST~

#### $\Delta$ DAF

#### XP-EHH




## Chapter Discussion
<!--
Pipeline discussion - how were the results?
  - Are the genes that are picked up similar to what has been discovered?

Is there anything special about the population comparisons?
  - shared genes between similar populations?
  - East/West split seen?
  - NZC vs GBR similarity?
-->
