# Positive Selection in Polynesian Populations

```{r source, include = FALSE, echo = FALSE}
knitr::opts_chunk$set(echo = FALSE)
library(phdFunctions)
#devtools::install_github('murraycadzow/UpSetR')
library(UpSetR)
library(Homo.sapiens)
library(ggbio)
library(GenomicRanges)
library(knitr)
library(kableExtra)
library(xtable)
library(stringr)
data(genesymbol, package = "biovizBase")
library(tidyverse)

source('scripts/multiplot.R')
source('scripts/03_selectionpipeline.R')
poly_pop <- c('CIM','NZM','TON','SAM')
```


In this chapter I investigate regions of the genome that exhibit signatures of selection in Polynesian populations with a particular focus on loci associated with serum urate levels and metabolic conditions.

<!-- Part of this work was originally published in @Cadzow2016 but I have updated the selection analysis to make use of a new dataset that became available after publication. The selection analyses are in this chapter. -->


## Introduction

<!-- Much of the early work in looking at selection was focused at the gene level [@Wright1951;@Fisher] and the methods that were developed were intended for few loci, or limited sequence from a single region [@ref;@ref;@ref]. This was largely due to the high cost and limited availability of whole genome chip or sequence data. It wasn't until after the completion of the human genome project in 2001 and the advent of 'next generation sequencing' where the cost of full genome data rapidly dropped and there became an abundance of whole genome sequence data [@ref]. -->
At the turn of the millennium the draft reference human genome was released [@Lander2001]. Following from this, there were a few large projects to characterise human variation and diversity. The first project was the HapMap project [@Hapmap2005] and was based on \gls{snp} array data. This was followed by the 1000 Genomes Project [@1KGP2010; @1KGP2012; @1KGP2015snp] which was the first large project to be based on whole-genome (re)sequence data. Alongside these variation projects were a few projects that aimed to focus on the genomic diversity of humans, such as the Human Genome Diversity Project [@Cann2002; @Rosenburg2002], and later the Simons Genome Diversity Project [@Mallick2016]. 

Over the last decade there has been a steady flow of studies on genome-wide selection [@voight2006map; @sabeti2006positive; @Coop2009; @tang2007new; @pickrell2009signals; @Grossman2010]. The main population samples for this have been part of the 1000 Genomes Project, which sampled over 2500 individuals for full genome (re)sequencing from five 'super' population groups [@1KGP2015snp]. The five super populations cover each of the geographical regions of Africa, Europe, East Asia, South Asia, and the Americas. This dataset, while capturing a large representation of the human population variation, does not have representation for the populations of Polynesia. 


### Positive selection

### Selection in Polynesian populations

Projects to capture the genetic diversity of humans have sampled populations worldwide, but outside the populations sampled by the 1000 Genomes Project, the population sample sizes have been small [@Cann2002; @Rosenburg2002; @Mallick2016]. Past studies that have included Oceanic populations have had minimal representation of Polynesian populations, e.g. @Kimura2008 only had 24 Tongan individuals. The latest project for human diversity, the Simons Human Diversity Project, did include the New Zealand M$\bar{a}$ori but they were only represented by a single individual and did not feature in many of the analyses [@Mallick2016].

More recently there have been studies that have looked at selection in Polynesian populations but with a focus on a few candidate loci, not at a genome-wide scale [@Myles2011; @Cadzow2016; @Minster2016]. The main feature of these studies has been the investigation of 'thrifty genes'.

#### Thrifty Gene Hypothesis

The "Thrifty genotype hypothesis" was originally hypothesised by @Neel1962, whereby it was posed that due to the differences between individuals with or diabetes that how could a disease that should have strong selection against it continue to have high prevalence if there was not some genetic advantage [@Neel1962]. In the Polynesian context, this hypothesis has been invoked in an attempt to explain the higher prevalence of metabolic related diseases such as \gls{t2d} and obesity [@Myles2011; @Minster2016]. However, there are several refutations of the thrifty genotype for Polynesian populations as being 'just-so' stories [@Cadzow2016; @Gosling2014; @Gosling2015]. An alternatice to the 'thrifty-gene' hypothesis is that the link between metabolic diseases and the innate immune system, susceptibilty of populations to metabolic disease might be due to selective pressure from infectious disease, such as malaria [@Gosling2015]. Uricase, the enzyme for metabolising urate, was lost through multiple mutations - a reduction in promoter activity followed by two further loss of function mutations and occurred before the divergence of humans from the great apes [@Kratzer2014]. The evolutionary history of uricase however has also been described as "the original thrifty-gene" [@Kratzer2014].

### Selection in urate and gout

The benefits of urate include the its ability to increase blood pressure and it is thought to have helped with early hominins to stand upright [@Watanabe2002] and increase blood flow to the brain. Serum urate is also a potent anti-oxidant which accounts for >50% of the anti-oxidant activity of the blood [@Glantzounis2005; @Parmar2009]. Mono-sodium urate crystals that form in conditions of elevated urate are a potent adjuvant for the immune system and enhance the immune response, providing a protective effect against cancer due to the turnover of cells [@Ames1981]. Uric acid has also been identified as being an important part of the malarial response, contributing to cytokine secretion and response of dendritic and T-cells [@GallegoDelgado2014].

<!-- - Possible reasons for the loss @Alvares-Lario2010 -->
<!--     - urate helps raise bp, this would have been beneficial in low salt environments, helped early hominins stand @Wantannabe? -->
<!--     - increases blood flow to the brain -->
<!--     - potent anti-oxidant which accounts for > 50% of the bloods anti-oxidant activity [refs 26,27 alvares-lario2010] -->
<!--     - crystals enhance local immune response - protective against cancer due to the turnover of cells? -->

<!-- ?? The fructose response for leads to insulin desensitivity @Johnston2007? -->
<!--     - fructose blocks glyconeogeneis -> increased glycogen storage -->

An increased level of urate in the blood is called \gls{hu} and is causal of gout. The prevalence of \gls{hu} in Polynesian populations is higher than many other populations [@Gosling2014], and with it are higher prevalences of the co-morbidities of gout, diabetes, obesity, dyslipidemia, kidney disease, and \gls{cvd} [@Winnard2013]. In New Zealand, gout prevalence is ~3.1%, however, prevalence in M$\bar{a}$ori and Pacific peoples is 6-8% with 40% of individuals with gout also having diabetes or \gls{cvd} [@Winnard2013]. Gout is an immune response to the monosodium urate crystals that are formed from the precipitation of urate from the blood and \gls{hu} is a prerequisite for the formation of these crystals [@Merriman2011a].

<!-- \Gls{hu} in the Pacific is seen in Eastern and Western Polynesia, but the prevalence in the ancestral populations of Near and Remote Oceania is reduced. However, Taiwanese aboriginals also have the higher prevalence of \gls{hu} [@Gosling2014]. -->

\Gls{hu} has two main components: over production of urate, and under excretion. Over production is influenced by diet, such as high purine or fructose content, or under genetic control. Under excretion is the main cause of \gls{hu}, caused by either increased reabsorption or decreased secretion. The kidneys are the organ where the majority of secretion and excretion of urate is performed, with about 90% being reabsorbed [@Kutzing2008]. In the proximal tubule, on the apical membrane, the urate transporters of URAT1, OAT4, and SLC2A9 reabsorb urate, whereas ABCC4, NPT4, and NPT1 are responsible for secretion [@Mandal2015]. On the basolateral membrane, OAT1-3, increase intracellular uric acid, but SLC2A9 moves uric acid from intracellular to extracellular [@Mandal2015]. The ATP powered transporter ABCG2 is responsible for excretion of urate in the gut and there are \gls{snp}s that lead to a reduction of function in ABCG2 in Western Polynesians  [@Phipps-Green2010]. Together the genes _SLC2A9_ and _ABCG2_ account for 3.4% of the genetic variability of urate [@Kottgen2013].

From previous studies there is prior evidence of selection at urate associated loci [@Zhang2013a; @Grossman2013; @Ramos2017]. In the review from @Ramos2017 on natural selection in rheumatic disease, one of the main loci for gout, _SLC2A9_, in the \gls{yri} population had evidence. The region of _BCAS3_ and _NACA2_ also had evidence, originally found by @Grossman2013 in the Asian population from the HapMap project data. @Zhang2013a had evidence for the associated loci of _ATXN2_ (rs653178) and _RREB1_ (rs675209)  having undergone selection when using a F~ST~-based method. The co-morbidities of \gls{hu} also have evidence of selection, with loci associated with \gls{t2d} having been found to have evidence of having undergone selection [@pickrell2009signals; @voight2006map; @Grossman2010, @Grossman2013]. Most recently a Samoan specific variant in _CREBRF_ that is protective for \gls{t2d} but causal for obesity has also been identified as having evidence of selection in the Samoan population [@Minster2016].



<!-- - jha2015 rs589691? -->



<!-- > mention the out of africa migration again and the polynesian expansion. East and west are different genetically -->


<!--
motivation: seemed to be little in the way of calculating stats genome wide and wasn't sure how sensitive these stats were to snp density

This chapter is about trying to first of all establish how sensitive the various statistics were to window/slide variation

from the literature there doesn't seem to be any recommendation as to how to compute these statistics genome wide. Up until recently the various statistics were computed at the 'feature' level, usually the gene.

Fst has a paper in early 2000's from Bruce Weir that touched on what sort of size to use howver it largely just came down to 'this window size seems to have enough of a smoothing effect as well as display some of the variation in the data' type approach and was largely just based on a this looks good enough.

Pybus et al specified their window sizes which was roughly 30 kb window and 3kb slide for most statistics however this was decided on unshown data as a trade off between resolution and density.

In cadzow 2014 I tried had tried a few sizes and it it seemed that there was little difference between them other than the smaller the window the more noisy the data looked.

While I was playing around with this for Tajima's D there does seem to be at least some relation to the density of the snps and the window size, which does intuitively make sense, in that if there are none or only 1 snp per window then the stats are going to be pretty poor.

From the core exome work it looks as though 30 kb was ok and at least when the distribution of TD was binned by number of snps per window it didn't look to bad and the overall total distribution of TD looks not too bad. Increasing from 30 kb didn't really change TD that much. the extremes become more extreme but the means increaseas window size increases.
-->

<!-- ascertainment bias will skew selection statistics that are site frequency spectrum dependant in chip data due to the over-representation of intermediate frequency SNPs, these include Tajima's D (upward) and Fay and Wu's H [@Clark2005]


Carlson2005 did windows of 100kb, slide 10kb and mean snps were ~50/window. ascertainment bias esp in the +ve end. continous region on >20 windows where 75% were in bottom 1% was defined as contiguous regions of Tajimas D reduction.
RafajloviÄ‡2014 largely followed Carlson2005, they also discarded windows with less than 5 snps.

Bigman2010 has a custom Tajima's D that is created to be similar to iHS in terms of calculation
-->



### Objectives

- Investigate regions that are under selection in the Polynesian populations
- Investigate regions previously associated with urate, \gls{t2d}, obesity, kidney disease, and metabolic syndrome for signatures of selection

## Methods

The selection of individuals, and preliminary data preparation for the selection data set is detailed in Section \@ref(selectionDataset). It consisted of combining 500 individuals (100 per population for each of \gls{nzm}, \gls{cim}, \gls{sam}, \gls{ton}, and \gls{nzc}) from the Genetics of Gout in Aotearoa study, who had been genoypted on the Illumina CoreExome v24 \gls{snp} array with 2504 individuals from the 1000 Genomes Project. 

<!-- The selection and neutrality statistics were calculated for individuals populations using a combination of two main software packages, PopGenome in R, and selectionTools v1.1. PopGenome was used to calculate the intra population frequency spectrum based tests for selection and neutrality. Haplotypic stats of \gls{ihs}, \gls{nsl}, and \gls{xpehh} were calculated using the selectionTools pipeline with significance set as being the 99th percentile for the absolute normalised score. Frequency based statistics were also generated such as \gls{daf} and F~ST~. -->

### Calculation of Selection and Neutrality Statistics


<!--
Similar to @Pybus2014, @Rafajlovic2014 and @Carlson2005 a trade off between increasing the number of \gls{snp}s per window and adequate resolution was made and the window size of 100 kb and slide of 10 kb was chosen.
-->
The R package PopGenome v2.2.3 [@Pfeifer2014] was used to calculate the site-frequency spectrum statistics \gls{td}, \gls{fwh}, \gls{fld}, \gls{flf}, \gls{ze}, and F~ST~ over non-overlapping windows of 10 kb (see section \@ref(popgenomeMethods) for further details). Except for F~ST~, the empirical distribution for each population was used to create an upper and  a lower significance threshold. The lower threshold was the 1st percentile and less than zero. The upper significance threshold was the 99th percentile and greater than zero. 
<!-- explain the condition better -->
The greater or less than zero condition was to enable the interpretation of the selection and neutrality statistic results. Contiguous regions of score reduction or elevation were created using a rolling window approach, whereby a condition of 15 or more windows out of a minimum of 20 consecutive 10 kb windows were required to have met a significance threshold in the same tail of the distribution.

The calculation of \gls{ihs}, \gls{nsl}, and \gls{xpehh} were calculated and normalised using selscan v1.1.0b [@Szpiech2014] as part of the selectionTools 1.1 [@Cadzow2014] pipeline as detailed in section \@ref(selectionTools). The default setting was used which was a maximum gap extension of 200 kb. Significance was defined as |value| > 2.6 which was equivalent to the most extreme ~1% of values after normalisation using frequency bins of 0.05.


### Pathway enrichment analysis

Pathway enrichment analysis was performed by taking genes that intersected the 1st percentile windows for the \glsdesc{sfs}-based statistics that also had a value < 0, or genes that had a significant marker for \gls{ihs} or \gls{nsl} and inputting them into Enrichr ^[http://amp.pharm.mssm.edu/Enrichr/ accessed 22 November 2017] [@Chen2013b; @Kuleshov2016] and exporting the KEGG 2016 pathway table. 

(ref:enrichraccess) http://amp.pharm.mssm.edu/Enrichr/, accessed 22 November 2017

## Results

```{r}
fdr_low_tab <- fdr_low %>% 
  filter(lowest_quant == lower_per, super_pop == 'POL') %>% 
  arrange(stat, pop) %>%
  ungroup() %>% 
  select(stat, pop, FDR, lowerCI, upperCI, lowest_quant)

fdr_high_tab <- fdr_high %>% 
  filter(highest_quant == upper_per, super_pop == 'POL') %>% 
  arrange(stat, pop) %>% 
  ungroup() %>% 
  select(stat, pop, FDR , lowerCI, upperCI, highest_quant)
```

<!-- > This paragraph needs to do a brief overview and general talk about how there was an overall pattern for the SFS stats and link out to the stat specific stuff -->

Results will be split into two themes, the first will be regions of the genome that show evidence of selection in the Polynesian populations, the second will be focusing specifically on selection of urate and associated loci in Polynesian populations. For reporting there will be a focus on the 1^st^ percentile/lower tail for the intra-population statistics as this is the distribution end that can correspond with positive selection.

For the frequency spectrum-based intra-population statistics of \gls{td}, \gls{fwh}, \gls{fld}, \gls{flf}, and \gls{ze} there were population-specific distributions. There was similarity however between the populations within a super population. The Polynesian populations in general had a left shifted distribution and had the lowest means, medians, and 1^st^ percentiles compared to the other super population groups (Table \@ref(tab:superGlobalSummary)). Within a super population there were similar numbers of windows that were in the 1^st^ percentile and < 0, but across super populations the number of windows differed (Table \@ref(tab:numSigWindows)). Some examples of this were the \gls{afr} populations for \gls{td}, \gls{fld}, and \gls{flf} in the lower tail, had as little as 30% the number of windows that met the lower threshold than the other populations. The Western Polynesian and \gls{eas} populations had up to 50% less windows for \gls{ze} than the  populations of \gls{afr}, \gls{amr}, and \gls{eur}. The Polynesian populations had fewer significant \gls{snp} for \gls{ihs} and \gls{nsl} than the other populations. These observations were the basis of the investigation that formed Chapter \@ref(clustering). 







```{r superGlobalSummary, echo = FALSE, results = 'asis'}
global_super_summary <- readRDS('~/data/NZ_coreExome_1kgp/100kbWindow_intra/filtered/100kbwindows_summary_superpop_popgenome_filtered_3ns_resized.6-11-2017.RDS')
global_super_summary %>%filter(stat != "n.segregating.sites") %>% mutate(stat = sapply(stat, stat_to_short)) %>% mutate(stat = sapply(stat, short_to_latex)) %>% kable(., caption ='(\\#tab:superGlobalSummary) Summary statistics for the intrapopulation frequency spectrum statistics by super population.', digits = 3, booktabs =TRUE, escape = FALSE, col.names = c("Super Population", "Statistic","Mean","SD","Min","1st Percentile","Median","99th Percentile","Max"), format = 'latex') %>%  kableExtra::kable_styling(latex_options = 'scale_down')
```

```{r numSigWindows}
# add on ihs/nsl
ihs_nsl_sum_tab <- sig_ihs_val %>% 
  group_by(pop) %>% 
  tally() %>% 
  select(pop, n_ihs_snps = n) %>% 
  left_join(., sig_nsl_val %>% 
              group_by(pop) %>% 
              tally() %>% 
              select(pop, n_nsl_snps = n) , by = 'pop') %>% 
  left_join(., sig_ihs_val %>% 
              select(pop, chrom, chrom_start, chrom_end) %>% 
              inner_join(., sig_nsl_val %>% 
                           select(pop, chrom, chrom_start, chrom_end), by = c('pop','chrom','chrom_start','chrom_end') ) %>% 
              group_by(pop) %>% 
              tally(), by = "pop") 

all_sfs_windows_table <- lower_sig_stats %>% 
  mutate(statname = paste0(statname,"_lo")) %>%  
  select(contains('chrom'), pop,statname) %>% 
  distinct() %>% 
  group_by(pop,statname) %>% 
  tally() %>% 
  spread(statname,n) %>% 
  full_join(.,upper_sig_stats %>% 
              mutate(statname = paste0(statname,"_up")) %>% 
              select(contains('chrom'), pop,statname) %>% 
              distinct() %>% 
              group_by(pop,statname) %>%
              tally() %>% 
              spread(statname,n) , by = "pop" ) %>% 
  left_join(., panel %>% 
              select(pop, super_pop) %>% 
              distinct(), by = 'pop') %>% 
  arrange(super_pop) %>%
  select(-super_pop)

left_join(all_sfs_windows_table, ihs_nsl_sum_tab %>% select(-n), by = 'pop') %>%
 knitr::kable(., caption = "(\\#tab:numSigWindows) Number of significant windows/markers by selection or neutrality statistic by population.", booktabs = TRUE, format = 'latex', col.names = c("Population",rep(c("\\gls{fwh}", "\\gls{fld}","\\gls{flf}","\\gls{td}","\\gls{ze}"), 2), 'iHS SNPs', 'nSL SNPs'), escape = FALSE) %>%
  kableExtra::add_header_above(., c("", "Lower Tail" = 5, "Upper Tail" = 5, "", "") )%>% 
  kableExtra::group_rows("AFR", 1, 7) %>% 
  kableExtra::group_rows("AMR", 8, 11) %>% 
  kableExtra::group_rows("EAS", 12, 16) %>% 
  kableExtra::group_rows("EUR", 17, 22) %>% 
  kableExtra::group_rows("POL", 23, 26) %>% 
  kableExtra::group_rows("SAS", 27,31) %>% 
  kableExtra::kable_styling(latex_options = 'scale_down')
```


```{r iHSnSLClusSum}
ihs_nsl_clus_tab <- left_join(ihs_clus %>% filter(statid == 26) %>% select(pop,ClusterID, chrom) %>% distinct() %>% group_by(pop) %>% tally() %>% rename(n_ihs_clus = n), nsl_clus %>% filter(statid == 20) %>% select(pop,ClusterID, chrom) %>% distinct() %>% group_by(pop) %>% tally() %>% rename(n_nsl_clus = n), by = 'pop')
  
ihs_nsl_clus_tab <- ihs_nsl_clus_tab %>% left_join(., ihs_nsl_sum_tab %>% left_join(., panel %>% select(pop, super_pop) %>% distinct(), by = 'pop') %>% mutate(pop = paste0(super_pop,'_',pop), by = 'pop'), by = 'pop')  %>% select(-super_pop) %>% select(pop, n_ihs_snps, n_nsl_snps, n, n_ihs_clus, n_nsl_clus)

# ihs_nsl_clus_tab %>% 
#   mutate(pop = substr(pop, 5,7)) %>% 
#   kable(., booktabs = TRUE,
#         col.names = c("Population", "iHS SNPs (n)", "nSL SNPs (n)", "Both (n)", "iHS Clusters (n)", 'nSL Clusters (n)'), 
#         caption = '(\\#tab:iHSnSLClusSum) Number of significant SNPs and clustered regions for iHS and nSL by population.') %>%
#   kableExtra::group_rows("AFR", 1, 7) %>% 
#   kableExtra::group_rows("AMR", 8, 11) %>% 
#   kableExtra::group_rows("EAS", 12, 16) %>% 
#   kableExtra::group_rows("EUR", 17, 22) %>% 
#   kableExtra::group_rows("POL", 23, 26) %>% 
#   kableExtra::group_rows("SAS", 27,31)

# add on number that were shared
#add on number of single snp clusters?

#sig_ihs %>% select(pop, chrom, chrom_start, chrom_end) %>% inner_join(., sig_nsl %>% select(pop, chrom, chrom_start, chrom_end), by = c('pop','chrom','chrom_start','chrom_end') ) %>% group_by(pop) %>% tally()
```


### False discovery rate of windowed statistics {#fdrresults}

```{r fdrpol}
 

fdr_low_tab %>% 
  left_join(., fdr_high_tab, by = c('stat','pop')) %>%
  mutate_at(vars(contains('.x'), contains('.y')), funs(three_dp)) %>%
  mutate(fdr_low = paste0(FDR.x, " (", lowerCI.x,', ', upperCI.x,')'), fdr_high = paste0(FDR.y, " (", lowerCI.y,', ', upperCI.y,')')) %>%
  select(pop, lowest_quant, fdr_low, highest_quant, fdr_high) %>%
  kable(., col.names = c("Population","Percentile", "FDR (95% CI)", "Percentile", "FDR (95% CI)"), caption = "(\\#tab:fdrpol) False discovery rate for windowed frequency spectrum based intra-population statistics by population.", booktabs = TRUE, format = 'latex') %>%
  kableExtra::add_header_above(., c("","Lower tail" = 2, "Upper tail" = 2) ) %>% 
  kableExtra::group_rows("Fay and Wu's H",1, 4) %>%
  kableExtra::group_rows("Fu and Li's D", 5, 8) %>%
  kableExtra::group_rows("Fu and Li's F", 9, 12) %>%
  kableExtra::group_rows("Tajima's D", 13, 16) %>%
  kableExtra::group_rows("Zeng's E", 17, 20) %>%
  kableExtra::footnote("Percentile is the more conservative FDR quantile bin of the permuted data that was equivalent to either the 1st (lower tail) or 99th (upper tail) percentile in the observed data.", general_title = "", threeparttable = TRUE )
```
In order to control for false positives the \gls{fdr} was calculated for \gls{td}, \gls{fld}, \gls{flf}, \gls{fwh}, and \gls{ze}. To calculate the \gls{fdr}, 5000 permutations were generated, each randomly shuffling the genotypes of markers by marker position (i.e. shuffling rows only). For each permutation, for each statistic, the values for quantiles 'bins' were calculated for both tails of the distribution. The 'bins' ranged from 0.01 to 0.1% at 0.01 increments, 0.1 to 1% at 0.1% increments, 1 to 5% at 1% increments, and 10%. The median and 95% confidence intervals were calculated for the values of the quantile 'bins' across the permutations (i.e. the median, 2.5^th^ and 97.5^th^ percentiles for the values of the 1^st^ percentile bin from all permutations). The enrichment for each quantile bin was then calculated by comparing the number of windows that were observed in the data compared to the number expected from the permutations. \Gls{fdr} was then estimated by using the value for the 1^st^ or 99^th^ percentile and finding the nearest conservative quantile bin using the median, that was either larger for the lower tail, or smaller for the upper tail and taking the inverse of enrichment for the bin (Table \@ref(tab:fdrpol)). This estimate will be conservative compared to the true \gls{fdr}.

The \gls{fdr} for \gls{ze} was 1.0 for all Polynesian populations in the lower end of the distribution and the 1st percentile of the data fell within the 2nd percentile bin from the permutations. Whereas, for \gls{td} and \gls{fwh} the quantile bin was `r fdr_low_tab %>%filter(stat %in% c('Tajima.D','Fay.Wu.H')) %>% summarise(min(lowest_quant))`%, with the maximum upper confidence interval being `r fdr_low_tab %>%filter(stat %in% c('Tajima.D','Fay.Wu.H')) %>% summarise(three_dp(max(upperCI)))`. For the purpose of reporting on individual statistics, a threshold of 0.1 was applied, to reduce the expected false positive rate to be less than 1 in 10 on average. \Gls{fld}, \gls{flf}, and \gls{ze} all had an \gls{fdr} above this threshold for the Polynesian populations and will be reported on only as additional evidence when they overlap with other results or were a region of consecutively depressed score.


### Comparison with prior publications


```{r priorGenes, include = FALSE}
# chr_unfiltered_stats <- readRDS('~/Murray/Thesis_backup/NZ_coreExome_1kgp/100kbWindow_intra/unfiltered/100kbwindows_chr_stats_popgenome_long.31-8-2017.RDS')
# head(chr_unfiltered_stats)
# library(GenomicRanges)
# library(TxDb.Hsapiens.UCSC.hg19.knownGene)
# library(org.Hs.eg.db)
# library(tidyverse)
# ns <- chr_unfiltered_stats %>% filter(stat == 'n.segregating.sites') %>% ungroup() %>% mutate(chrom = paste0('chr',chrom))
# 
# hider_genes <- unique(unlist(list(
#   td = c("EDAR","MITF","ATRN","TRPM1", "TMEM33","BNC2","OPRM1","ADAM17"), 
#   fwh = c("MITF","BNC2","TYRP1","MLPH","PDIA6"), 
#   fld = c("MITF","ATRN","KLHL7","TMEM33","PMEL","SILV","BNC2","LYST","DCT","CTSD"), 
#   flf = c("MITF","ATRN","KLHL7","TMEM33","PMEL","SILV","BNC2","LYST","DCT","CTSD"),
#   ihs = c("EDAR","TH","MC1R") 
# )))
# 
# genes_entrez <- na.omit(AnnotationDbi::select(org.Hs.eg.db, keys = hider_genes, columns = c('SYMBOL','ENTREZID'), keytype="SYMBOL"))
# 
# genes_ucsc <- merge(AnnotationDbi::select(TxDb.Hsapiens.UCSC.hg19.knownGene , columns = c("TXID","TXCHROM","TXSTART","TXEND","TXSTRAND"),keys = genes_entrez$ENTREZID, keytype="GENEID"), genes_entrez, by.x ='GENEID', by.y = "ENTREZID")
# genes_ucscGR <- GRanges(genes_ucsc[!is.na(genes_ucsc$TXID) ,])
# genes_ucscGR <- genes_ucscGR[which(genes_ucscGR@seqnames %in% paste0('chr',1:22))]
# 
# ns_GR <- GRanges(ns)
# ol <- findOverlaps(query = genes_ucscGR, subject = ns_GR)
# saveRDS(chr_unfiltered_stats %>% ungroup() %>% filter(stat == 'n.segregating.sites', value < 4) %>% select(contains('chrom'), stat,value) %>% distinct() %>% mutate(chrom_start = as.numeric(chrom_start) + 45000, chrom_end = as.numeric(chrom_end) - 45000 ), file = '~/Murray/Thesis_backup/NZ_coreExome_1kgp/100kbWindow_intra/filtered/100kbwindows_chr_stats_popgenome_removed_windows_resized.25-9-2017.RDS')


#EAS (CHB, CHS, and JPN) regions paper
hider <- read_csv('data/hider2013_S1.csv', col_names = TRUE) %>% mutate(fld = ifelse(is.na(stringr::str_extract(Test, fixed('D*-'))),FALSE, TRUE), flf = ifelse(is.na(stringr::str_extract(Test, fixed('F*-'))),FALSE, TRUE), fwh = ifelse(is.na(stringr::str_extract(Test, fixed('H-'))),FALSE, TRUE), td = ifelse(is.na(stringr::str_extract(Test, fixed('D-'))),FALSE, TRUE) )


# from hider2013 table 1
hider_genes <- list(
  td = c("EDAR","MITF","ATRN","TRPM1", "TMEM33","BNC2","OPRM1","ADAM17"), 
  fwh = c("MITF","BNC2","TYRP1","MLPH","PDIA6"), 
  fld = c("MITF","ATRN","KLHL7","TMEM33","PMEL","SILV","BNC2","LYST","DCT","CTSD"), 
  flf = c("MITF","ATRN","KLHL7","TMEM33","PMEL","SILV","BNC2","LYST","DCT","CTSD"),
  ihs = c("EDAR","TH","MC1R") 
    )

# SAS (GIH and ITU) from Joannalagadda2017
joannalagadda_genes <- list(
td = c("DST","PDGFC","PDPK1","SLC24A5","MITF"),
fwh = c("BLOC1S5","MITF","MLPH","PTS","MBD4"),
ihs = c("HS2ST1","MLPH","SEMA3C","TYR","ADAM17")
)

lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CHB','CHS','JPN'), SYMBOL %in% hider_genes$td) %>% group_by(pop, SYMBOL) %>% tally()

lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CHB','CHS','JPN'), SYMBOL %in% hider_genes$fwh) %>% group_by(pop, SYMBOL) %>% tally()

lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CHB','CHS','JPN'), SYMBOL %in% hider_genes$fld) %>% group_by(pop, SYMBOL) %>% tally()

lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CHB','CHS','JPN'), SYMBOL %in% hider_genes$flf) %>% group_by(pop, SYMBOL) %>% tally()

lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('GIH','ITU'), SYMBOL %in% joannalagadda_genes$td) %>% group_by(pop, SYMBOL) %>% tally()

lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('GIH','ITU'), SYMBOL %in% joannalagadda_genes$fwh) %>% group_by(pop, SYMBOL) %>% tally()



#compare the regions between the Hiders regions and those that I identified as significant

h_tdGR <- hider %>% mutate(chrom =  paste0('chr',chr)) %>% filter(td == TRUE) %>% select(chrom, start,end) %>% GRanges() 

lo_sig_tdGR <- lower_sig_stats %>% filter(pop %in% c("CHB","CHS","JPT"), statname == 'Tajima.D') %>% select(contains('chrom')) %>% GRanges()

hits <- findOverlaps(query = lo_sig_tdGR, subject = h_tdGR)
h_tdGR[hits@to] %>% data.frame() %>% distinct() %>% NROW()
```

```{r ramos}
ramos <- list()
ramos[["ceu"]]  <- c("ALMS1P", "DGUOK", "PARD3B", "KCNIP4", "HLA", "PRKCH", "HIF1A", "IRF4", "TECRL", "CTNNA3", "WWOX", "RABEP1", "NUP88", "COMMD10", "SEMA6A", "ALDH7A1", "SNRPC", "UHRF1BP1", "OSBPL8", "GRM5", "NTNG2", "SETX", "GRHL2")
ramos[["yri"]] <- c("TNFRSF14", "MMEL1", "MAGI3", "PTPN22", "TLR5", "XCR1", "CCR3", "CCDC66", "ARHGEF3", "SLC2A9", "KCNIP4", "ARHGAP31", "CD80", "IL2", "IL21", "TPR3", "HLA", "CCDC167", "MIR4462", "PRSS54", "PRDM1", "ATG5", "TRAF3IP2", "PAK7PRKCH", "HIF1A", "HIP1")
ramos[["asi"]] <- c("NCF2", "RGL1", "PELI1", "HLA", "BCAS3", "NACA2", "CNTN6", "BTLA", "MRPS22", "COMMD10", "SEMA6A", "GTF2I", "KIAA0391", "FAM171A1", "KDM4C", "XKR6", "BLK", "LSMEM1", "NPM1P14")
ramos[["unknown"]] <- c("IL23R", "FCGR2B", "TNFSF4", "CR1", "ANTRX2", "TNIP1", "PTTG1", "VARS", "LSM2", "IKZF1", "IRF8", "TYK2", "CLEC16A", "CIITA", "ITGAM", "ITGAX", "SH2B3", "NAA25", "CD5", "PTGER4")
```

The selection and neutrality statistics that were in the 1^st^ percentile and < 0 were compared to regions and genes identified in @Hider2013 for \gls{eas} and @Jonnalagadda2017 for \gls{sas} as these papers had used a similar, but not identical methodology to identify regions under selection. @Hider2013 investigated selection in the East Asian populations (\gls{chb}, \gls{chs}, and \gls{jpt}) of the 1000 genomes project in the Phase 1 release. For this they used a non-overlapping window size of 25kb and excluded windows with less than 10 segregating sites and then looked at the top 1% of results. They reported regions that were significant for \gls{td}, \gls{fwh}, \gls{fld}, \gls{flf}, and \gls{ihs}. Comparing the regions from @Hider2013 with regions that met the significance threshold for this project there were 118 of 206 regions for \gls{td}, 15 of 151 for \gls{fwh}, 65 of 261 for \gls{fld}, 83 of 255 for \gls{flf}, and 5 of 85 for \gls{ihs}.

Of the genes reported in @Jonnalagadda2017, there were 2 of 5 for \gls{td}, 0 of 5 for \gls{fwh}, and 2 of 5 for \gls{ihs} that were also significant in this project. @Jonnalagadda2017 used the same window set up as @Hider2013, but used the 1000 Genomes Project Phase 3 release for the \gls{gih} and \gls{itu} populations. 

> define these regions

In conjunction with the studies from @Hider2013 and @Jonnalagadda2017, @Ramos2017 looked at evidence of selection in rheumatic disease, including gout, using the \gls{ihs} results from @voight2006map from the HapMap phase II dataset. There were three populations used which were Asian, European, and African. The equivalent populations of \gls{chb} and \gls{jpt} were used to represent to the Asian population, \gls{ceu} for European, and \gls{yri} for African, and unknown was any of the specified populations, from the 1000 genomes individuals when comparing the loci reported in @Ramos2017 with the \gls{ihs} results. Loci reported for European had  `r (sig_ihs_val %>% filter(pop == 'CEU', genename %in% ramos[['ceu']])%>% mutate(pop = 'CEU', statname = case_when(statid == 20 ~ "nSL", statid == 26 ~ "iHS"), chrom = paste0('chr',chrom)) %>%  select(SYMBOL = genename, chrom, statname) %>% distinct() %>% NROW() / length(ramos[["ceu"]] ) *100) %>% round_dp(., 1)`% overlap with \gls{ceu}. The Asian reported loci matched `r (sig_ihs_val %>% filter(pop == 'JPT' | pop == 'CHB', genename %in% ramos[['asi']])%>% mutate(pop = 'ASI', statname = case_when(statid == 20 ~ "nSL", statid == 26 ~ "iHS"), chrom = paste0('chr',chrom)) %>%  select(SYMBOL = genename, chrom, statname) %>% distinct() %>% NROW() / length(ramos[["asi"]] )*100) %>% round_dp(., 1)`% with \gls{chb} and \gls{jpt}, and for African there was `r (sig_ihs_val %>% filter(pop == 'YRI', genename %in% ramos[['yri']])%>% mutate(pop = 'YRI', statname = case_when(statid == 20 ~ "nSL", statid == 26 ~ "iHS"), chrom = paste0('chr',chrom)) %>%  select(SYMBOL = genename, chrom, statname) %>% distinct() %>% NROW() / length(ramos[["yri"]] ) *100) %>% round_dp(., 1)`% of the reported loci also found in \gls{yri}. For the loci reported with an "unknown"" population, `r (sig_ihs_val %>% filter(pop %in% c('CEU','YRI','JPT','CHB' ), genename %in% ramos[['ceu']])%>% mutate(pop = 'ceu_yri_asi', statname = case_when(statid == 21 ~ "nSL", statid == 27 ~ "iHS"), chrom = paste0('chr',chrom)) %>%  select(SYMBOL = genename, chrom, statname) %>% distinct() %>% NROW() / length(ramos[["unknown"]] )*100) %>% round_dp(.,1)`% were found to match in any of \gls{ceu}, \gls{yri}, \gls{chb}, or \gls{jpt}.

The differences in regions being found as significant can be partially attributed to the differing window sizes, and the difference between sequence and chip based genotyping for @Hider2013 and @Jonnalagadda2017 and difference in markers for @Ramos2017. 

<!--
NEED TO RELOOK AT THE HAPLOTYPIC RESULTS 15/12/2017 - DONE
results are from ~/Git_repos/Thesis/selectionpipeline/prior_pubs_comparison_SFS_results.Rmd

values are for the corrected 

results from looking at all of the hider regions

- td
    * 206 / 243 regions with as least 4 sites in my results
    * found 118/206 (0.573) windows from hider regions
- fwh
    * 151 / 158
    * 15 / 151 (0.0993)
- fld
    * 261 / 283
    * 65 / 261  (0.249)
- flf
    * 255 / 281
    * 83 / 255 (0.325)
- ihs (regions are done slightly differently so not a good comparison)
    * 85 / 100
    * 5 / 85 (0.059)

joannalagga genes

@Joannalagga2017 used a similar approach as @Hider2013, in that they used the same window size 
td 2/5
fwh 0/5
ihs 2/5
-->

```{r, eval = FALSE}
# voight2006
sig_ihs_val %>% filter(genename %in% c("SLC44A5","LCT","SNTG1","SPAG4","SYT1","NCOA1","ADCY3","STY1")) %>% group_by(pop, genename) %>% tally() %>% arrange(pop, genename) %>% filter(pop %in% c('CHB',"JPT","CHS","CEU", "YRI"))

```




A gene list derived from @voight2006map Table 1 was created to include genes that showed evidence of selection across different combinations of the populations \gls{ceu}, \gls{chb}, \gls{chs}, \gls{jpt}, and \gls{yri}. All genes except _SPAG4_ were positive for significant \gls{ihs} markers in the appropriate populations they were originally reported in (Table \@ref(tab:voightTab)). 



<!-- Gene | CEU | CHB | CHS | JPT | YRI -->
<!-- ---|---|---|---|---|--- -->
<!-- LCT | 4 | - | - | - | - -->
<!-- SLC44A5 | - | 2 | 5 | 3 | - -->
<!-- NCOA1 | - | - | - | - | 4 -->
<!-- ADCY3 | - | - | -| - | 1 -->
<!-- SNTG1 | 8 | 5 | 9 | 5 | 6 -->
<!-- SPAG4 | 0 | (1) | - | - | 0 -->
<!-- SYT1 | - | - | - | - | 6 -->

```{r}
read.table(text = 'Gene,CEU,CHB,CHS,JPT,YRI
LCT,4,-,-,-,-
SLC44A5,-,2,5,3,-
NCOA1,-,-,-,-,4
ADCY3,-,-,-,-,1
SNTG1,8,5,9,5,6
SPAG4,0,-(1),-,-,0
SYT1,-,-,-,-,6', header = TRUE, stringsAsFactors = FALSE, sep = ',') %>% 
kable(., caption = "(\\#tab:voightTab) List of genes used as positive controls for detection of significant \\gls{ihs} regions.", booktabs = TRUE, format = 'latex', escape = TRUE) %>% 
  kableExtra::footnote(general = c("Number of SNPs that were significant", 
                                   "for iHS, dashes indicate absence in \\\\cite{voight2006map}.",
                                   "Brackets indicate number of significant SNPs in the ",
                                   "CoreExome data, but was not in the population as reported ",
                                   "in \\\\cite{voight2006map}."), general_title = "", escape= FALSE)
```



```{r readFst}
fst <- readRDS('~/data/NZ_coreExome_1kgp/whole_chr_fst_popgenome.3-7-2017.RDS')
fst <- bind_rows(lapply(fst, function(x){x %>% mutate(pop1 = as.character(pop1), pop2 = as.character(pop2))%>% select(-p1,-p2) %>% filter(!pop1 %in% c('WPN','EPN','NAD'), !pop2 %in% c('WPN','EPN','NAD')) %>% left_join(., panel %>% mutate(super_pop = paste(super_pop, pop, sep = '_')) %>% select("pop1" = pop, super_pop) %>% distinct(), by = "pop1" ) %>% select(-pop1) %>% spread(., "super_pop", fst )})) %>% group_by(pop2) %>% summarise_all(mean) %>% left_join(., panel %>% mutate(super_pop = paste(super_pop, pop, sep = '_')) %>% select("pop2" = pop, super_pop) %>% distinct(), by = "pop2" ) %>% select(Population = super_pop, everything(),-chrom, -pop2) %>% arrange(Population) %>% mutate_if(is.numeric, funs(sprintf('%.4f',.))) %>% data.frame(., row.names = NULL)

intra_pol_max_fst <- pol_fst %>% gather("pop",'fst', -contains('chrom'))%>% filter(!is.na(fst))  %>% separate(pop, c("pop1","pop2"), sep = '_') %>% group_by(pop1,pop2) %>% filter(max(fst) == fst, (pop1 %in% poly_pop) & (pop2 %in% poly_pop))
pol_fstGR <- pol_fst %>% select(contains('chrom')) %>% mutate(chrom = paste0("chr",chrom)) %>% GRanges()
```





```{r Fst, eval = FALSE}
fst %>% mutate_at(., .vars =  vars(-Population), .funs= function(x){three_dp(as.numeric(x))}) %>%  rename_at(., .vars = vars(-Population), .funs = function(x){substring(x, 5, 8)} ) %>% mutate(Population = substring(Population, 5, 8)) %>%  knitr::kable(., booktabs = TRUE, caption  = '(\\#tab:Fst) $F_{ST}$', format = 'latex') %>% kableExtra::kable_styling(latex_options = 'scale_down') %>% 
  #kableExtra::row_spec(0, angle = 90) %>%  
  kableExtra::add_header_above(., c("" , "African" = 7, "American" = 4, "East Asian" = 5, "European" = 6, "Polynesian" = 4, "South Asian" = 5)) %>% 
  kableExtra::group_rows("AFR", 1, 7) %>%
  kableExtra::group_rows("AMR", 8, 11) %>% 
  kableExtra::group_rows("EAS", 12,16) %>% 
  kableExtra::group_rows("EUR", 17,22) %>% 
  kableExtra::group_rows("POL",23, 26)%>% 
  kableExtra::group_rows("SAS", 27, 31) %>% kableExtra::landscape()
```



```{r, eval = FALSE}
# regions of max fst between populations
pol_fst %>% gather("pop",'fst', -contains('chrom'))%>% filter(!is.na(fst)) %>% separate(pop, c("p1", "p2"), "_") %>% left_join(., panel %>% select(p1 = pop, super_pop) %>% distinct(), by = 'p1') %>%  group_by(p1, p2) %>%  filter(max(fst)  == fst) %>% arrange(fst)

# number of regions that have fst = 0 between the pairs of populations
pol_fst %>% gather("pop",'fst', -contains('chrom'))%>% filter(!is.na(fst)) %>% separate(pop, c("p1", "p2"), "_") %>% left_join(., panel %>% select(p1 = pop, super_pop) %>% distinct(), by = 'p1') %>%  group_by(p1, p2) %>%  filter(min(fst)  == fst) %>% tally() %>% arrange(n)
```


<!-- supplemental table 5 @1KGP2015snp -->

<!-- > In this section look at regions that are highly differentiated -->
<!-- what are the regions that differentiate the polynesians from the rest of the populations -->
<!-- what are the regions that differentiate the ep from wp -->
<!-- any regions that are population specific? -->


<!-- regions that differentiated the Eastern Polynesian from the Western Polynesian -->




<!-- #### XP-EHH -->

<!-- XPEHH performs well for high frequency variants, near fixation [@sabeti2007; @Vatsiou2016] -->

<!-- Migration weakens the power of xpehh [@Vatsiou2016] -->

<!-- In this sub section I want to explore just general results, and compare back to the ihs/nsl -->

<!-- - how many sig per pop? -->
<!-- - how many genes? -->
<!-- - what is the overlap with ihs/nsl like? -->

<!--
homozygosity and hetersozygosity
correlation with histone marks/DNAse hypersensitivity
look at selected regions/snps and expression in GTex
-->

```{r xpehhSummary}
# number of significant snps per population
xpehh_summary <- left_join(
  xpehh %>% select(contains('chrom'),xpehh_value, pop1, pop2) %>% distinct() %>% group_by(pop1, pop2) %>% tally() %>% spread(pop1, n) %>% select(-NZC),
  xpehh %>% select(contains('chrom'),xpehh_value, pop1, pop2) %>% inner_join(., sig_ihs_val %>% rename(pop1 = pop), by = c('chrom', 'chrom_start','chrom_end','pop1')) %>% distinct() %>% group_by(pop1, pop2) %>% tally() %>% spread(pop1, n) %>% select(-NZC), by = 'pop2', suffix = c("","_ihs"))  %>% 
  left_join(.,
            xpehh %>% select(contains('chrom'),xpehh_value, pop1, pop2) %>% inner_join(., sig_nsl_val %>% rename(pop1 = pop), by = c('chrom', 'chrom_start','chrom_end','pop1')) %>% distinct() %>% group_by(pop1, pop2) %>% tally() %>% spread(pop1, n) %>% select(-NZC), by = 'pop2', suffix = c("","_nsl")) %>% 
  left_join(., panel %>% select(pop2 = pop, super_pop) %>% distinct(), by = 'pop2') %>% arrange(super_pop)



xpehh_ihs_per <- xpehh_summary %>% mutate(cim_ihs_per = CIM_ihs/CIM *100, nzm_ihs_per = NZM_ihs/NZM*100, sam_ihs_per = SAM_ihs/SAM*100, ton_ihs_per = TON_ihs/TON*100 ) %>% select(pop2, contains('per'))
xpehh_nsl_per <- xpehh_summary %>% mutate(cim_nsl_per = CIM_nsl/CIM *100, nzm_nsl_per = NZM_nsl/NZM*100, sam_nsl_per = SAM_nsl/SAM*100, ton_nsl_per = TON_nsl/TON*100 ) %>% select(pop2, contains('per'))
```

```{r, eval = FALSE}
xpehh_summary %>% mutate(CIM = paste0(CIM, " (",round_dp(CIM_ihs/CIM*100,1),", " ,round_dp(CIM_nsl/CIM*100, 1), ")")) %>% 
  mutate(NZM = paste0(NZM, " (",round_dp(NZM_ihs/NZM*100,1),", " ,round_dp(NZM_nsl/NZM*100, 1), ")")) %>% 
  mutate(SAM = paste0(SAM, " (",round_dp(SAM_ihs/SAM*100,1),", " ,round_dp(SAM_nsl/SAM*100, 1), ")")) %>% 
  mutate(TON = paste0(TON, " (",round_dp(TON_ihs/TON*100,1),", " ,round_dp(TON_nsl/TON*100, 1), ")")) %>% 
  mutate_all(., function(x){ifelse(x == "NA (NA, NA)",  "-", x )}) %>% 
  select(Population = pop2, everything(), -contains('_'),-super_pop) %>% 
kable(., format = 'latex', booktab = TRUE, caption = '(\\#tab:xpehhSummary) Number of significant \\gls{xpehh} markers by population and percentage of markers that were also significant for \\gls{ihs} or \\gls{nsl}.') %>% 
  kableExtra::group_rows("AFR", 1, 7) %>%
  kableExtra::group_rows("AMR", 8, 11) %>% 
  kableExtra::group_rows("EAS", 12,16) %>% 
  kableExtra::group_rows("EUR", 17,22) %>% 
  kableExtra::group_rows("POL",23, 26)%>% 
  kableExtra::group_rows("SAS", 27, 31) %>% 
  kableExtra::footnote('Numbers in parentheses are the percent of markers with significant XP-EHH also significant in iHS and percent of markers significant in nSL for the column population.', threeparttable = TRUE)

```



<!-- max xpehh values -->

<!-- The maximum \gls{xpehh} values were  -->

<!-- - cim v CDX 5.87733 at rs659777 -->
<!-- - nzm v itu 5.90709 at rs7356528 -->
<!-- - sam v itu 5.32534 at rs9684601 -->
<!-- - ton v sam 5.95687 at rs4800666 -->

```{r, eval = FALSE}
# genes with the most significant markers
xpehh %>% group_by(pop1, pop2, genename) %>% tally() %>% arrange(genename) %>% group_by(pop1, genename) %>% summarise(per = sum(n)/(max(n)*NROW(pop1)), n()) %>% arrange(desc(per))

```



<!-- ihs -->

<!-- - CIM max per was with SAM 9.146968 -->
<!-- - CIM min per was with NZM 3.803487 -->
<!-- - NZM max per was with ASW 9.056746 -->
<!-- - NZM min per was with CIM 3.074485 -->
<!-- - SAM max per was with CHB 6.773364 -->
<!-- - SAM min per was with NZM 2.980222 -->
<!-- - TON max per was with ASW 5.969022 -->
<!-- - TON min per was with NZM 2.336054 -->

<!-- nsl -->

<!-- - CIM max per was with CLM 7.596765 -->
<!-- - CIM min per was with NZM 2.662441 -->
<!-- - NZM max per was with MXL 5.876068 -->
<!-- - NZM min per was with CIM 1.965135 -->
<!-- - SAM max per was with ASW 5.048924 -->
<!-- - SAM min per was with CIM 2.209661 -->
<!-- - TON max per was with ASW 5.024556 -->
<!-- - TON min per was with NZM 1.660569 -->


```{r, eval = FALSE}
#look at the number of urate genes that were significant for the polynesian populations
xpehh %>%  select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% filter(pop1 %in% c("CIM","NZM","SAM","TON"), xpehh_value > 0) %>% filter(genename %in% urate_goutGR$SYMBOL) %>% group_by(pop1,pop2, genename) %>%tally() %>% spread(pop1, n, fill = "") %>% left_join(., panel %>% select(pop2 = pop, super_pop) %>% distinct(), by = "pop2" ) %>% select(super_pop, everything()) %>% arrange(genename,super_pop,pop2) %>% kable(., caption = 'urate genes significant in Polynesians', longtable =  TRUE)
```

```{r, eval = FALSE, include = FALSE}
# DOESN'T WORK THE WAY I WANT IT TO

# all genes that have both significanct results in nsl/ihs and xpehh for Pol popululations
# calculate the mean number of sig SNPs for EP and WP by gene then look at the diff
xpehh %>% select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% 
  filter(pop1 %in% c("CIM","NZM","SAM","TON"), xpehh_value > 0) %>% 
  select(pop = pop1,pop2, genename) %>% distinct() %>% filter(genename == 'COL22A1') %>%   group_by(pop, genename)%>% 
  tally() %>% 
  mutate(ep = ifelse(pop %in% c("CIM","NZM"), TRUE, FALSE)) %>% 
  group_by(ep, genename) %>% summarise(m = mean(n, na.rm = TRUE)) %>% 
  mutate(pol = ifelse(ep == TRUE, "EP","WP")) %>%
  ungroup() %>%  
  select(-ep) %>% 
  spread(pol, value = m, fill = 0) %>% 
  mutate(diff = abs(EP - WP)) %>% 
  filter(diff > quantile(diff, 0.75))%>% 
  arrange(desc(diff)) %>% head() %>% 
  kable(., caption = 'Difference between Eastern and Western Polynesian populations XP-EHH - All genes', longtable = TRUE, booktabs = TRUE)
```

```{r, eval = FALSE}
# try to find places where Polynesian populations have lots of other populations that are significant
xpehh %>%  
  select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% 
  filter(pop1 %in% c("CIM","NZM","SAM","TON"), xpehh_value > 0) %>% 
  filter(!(pop1 %in% poly_pop & pop2 %in% poly_pop)) %>% 
  select(pop = pop1,pop2, genename) %>% 
  distinct() %>% 
  group_by(pop, genename)%>% tally() %>% arrange(desc(n)) %>% spread(pop, n, 0) %>% filter(CIM >= 20, NZM >= 20, SAM >= 20, TON >= 20, !is.na(genename)) %>% kable(., caption = 'XPEHH with all four Polynesian populations having a significant XPEHH marker in a gene for at least 20 other populations')
```

```{r, eval = FALSE}
# all genes that have both significanct results in nsl/ihs and xpehh for Pol popululations
# calculate the mean number of sig SNPs for EP and WP by gene then look at the diff
xpehh %>%  
  select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% 
  filter(pop1 %in% c("CIM","NZM","SAM","TON"), xpehh_value > 0) %>% 
  select(pop = pop1,pop2, genename) %>% 
  distinct() %>% 
  group_by(pop, genename)%>% tally() %>% 
  # inner_join(., bind_rows(sig_ihs_val, sig_nsl_val) %>% 
  #              select(pop,genename) %>% 
  #              distinct(), by = c('pop','genename')) %>%  
  spread(pop, n, fill = 0) %>% 
  mutate(EP = (CIM+ NZM)/2, WP = (SAM + TON)/2, diff = abs( (CIM+ NZM)/2 - (SAM + TON)/2  ))%>%
  filter(diff > quantile(diff, 0.75), ((WP > 0 & TON > 0 & SAM > 0) | (EP > 0 & CIM > 0 & NZM > 0 ))) %>% 
  arrange(desc(diff)) %>% filter(EP == 0 | WP == 0) %>% head() %>% 
  kable(., caption = 'Difference between Eastern and Western Polynesian populations with significance in iHS or nSL, and XP-EHH - All genes', longtable = TRUE, booktabs = TRUE)
```

<!-- The gene that had the biggest difference in between the East and West Polynesians in terms of the number of populations  -->

<!-- COL22A1 (https://www.ncbi.nlm.nih.gov/pubmed/20222955), which is has been associated with serum creatinine levels -->


```{r, eval = FALSE}
xpehh %>%  
  select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% 
  filter(pop1 %in% c("CIM","NZM","SAM","TON"), xpehh_value > 0) %>% 
  filter(genename %in% c(urate_goutGR$SYMBOL, t2d_GR$SYMBOL, obesity_GR$SYMBOL, metsyn_GR$SYMBOL)) %>% 
  select(pop = pop1,pop2, genename) %>% 
  distinct() %>% 
  group_by(pop, genename)%>% tally() %>% 
  inner_join(., bind_rows(sig_ihs_val, sig_nsl_val) %>%
               select(pop,genename) %>% 
               distinct(), by = c('pop','genename')) %>%  
  spread(pop, n, fill = 0) %>% 
  mutate(diff = abs( (CIM+ NZM)/2 - (SAM + TON)/2  ))%>% 
  #filter(diff > quantile(diff, 0.25)) %>% 
  kable(., caption = '(\\#ref:xpehhPopEpWp) number of populations with significant xpehh and significant ihs/nsl', longtable = TRUE, booktabs = TRUE)
```

<!-- SAM didn't have PTPRD, LPP -->
<!-- TON didn't have ERRB4 -->

<!-- SAM only had TP53INP -->
<!-- TON only for FHIT, JAZF1, NRXN3 -->

<!-- From table \@ref(tab:xpehhPopEpWp) in western polynesian populations there were  -->

<!-- - IGFR -->

<!-- Eastern Polynesian populations -->

<!-- - ADCY9 -->
<!-- - ARL15 -->
<!-- - SSR1 -->




<!-- > Genes associated with gout/metsyn/t2d/obesity/kd with significant hit in ihs/nsl in only pol pops -->

<!--
XXX CHECK XXX

When using the significant loci of \gls{ihs} and \gls{nsl} as a filter on the significant xpehh loci, there were 2 genes that had a significant ihs or nsl value in only Polynesian populations populations that were also significant for XPEHH with Polynesian populations. These genes were _ATP8B2_ and _IFNGR1_.

The \gls{t2d} associated gene, _ATP8B2_, had 4 markers that were significant: rs6702754, rs11586379, rs2988721, and rs200305485. With rs6702754 being the marker with the most number of significant results for _ATP8B2_. Rs200305485 was only significant between the Western and Eastern Polynesian populations. The \gls{ton} population had a significant \gls{xpehh} with the Eastern Polynesian populations of \gls{cim} and \gls{nzm}, and also with all of the \gls{eas} populations of \gls{cdx}, \gls{chb}, \gls{chs}, \gls{khv}, \gls{jpt}, and the \gls{sas} population of \gls{beb}. The \gls{sam} population only had a significant \gls{xpehh} with a subset of the populations that \gls{ton} did. These were \gls{nzm}, \gls{chs}, and \gls{cdx}.  

_IFNGR1_ was from the obesity associated gene list, and only the \gls{nzm} population had both a significant \gls{ihs} and \gls{nsl}, and significant \gls{xpehh}. There were two markers that were significant for \gls{xpehh} with \gls{nzm} and \gls{asw}, rs1887415 and rs9376267.

XXXXX
-->
```{r, eval = FALSE}
bind_rows(sig_ihs_val, sig_nsl_val) %>% filter(genename %in%  c("ADCY9", "CCNJL","EYA2","LY86","PSMD6","TP53INP1")) %>% select(pop, genename, statid, statvalue) %>% arrange(genename)
```


\clearpage

### Selection in Polynesian Populations
<!--
This section is about reporting how a population went, maybe add in a general comparison of the 1000 genomes populations, eg average number of genes specific to a population

The general format that I want for this section is to get:
- the genes that were reported in multiple stats as significant
- genes that were only reported in a particular population
- runs of depressed regions (eg 75% of >20 windows) that were significant in a row
- pathway/ontology terms
-->

There were `r genelists %>% NROW()` genes that were associated with urate and metabolic diseases from the \gls{gwas} catalog (see Table \@ref(tab:gwascatref) for references), with `r bind_rows(sig_ihs_val, sig_nsl_val) %>% select(pop, SYMBOL = genename) %>% bind_rows(., lower_sig_stats %>% select(pop, SYMBOL)) %>% filter(SYMBOL %in% genelists$genename, pop %in% poly_pop) %>% select(SYMBOL) %>% distinct() %>% NROW()` having at least a single window or \gls{snp} meeting a threshold for a single statistic in the Polynesian populations. Genes reported here were highlighted based on three criteria: a significant haplotypic result in multiple Polynesian populations, or having multiple statistics meeting the significance threshold in at least a single Polynesian population, or the locus only having the main significant result in Polynesian populations, with minimal signal in other populations, i.e. multiple significant markers in Polynesian populations and one or two in other populations.

In this subsection results will be presented first for those that were in common between all Polynesian populations and the sub-groups of East and West Polynesia. Following this the results for individual Polynesian populations will be presented.  A complete table of all genes that had results meeting the various thresholds for the various statistics used can be found in Table \@ref(tab:intrasfsPol).

```{r, eval = FALSE}
# distribution of missense variants
sig_ihs_val %>% left_join(., snpeff %>% select(chrom = V1, chrom_start = V2, everything())) %>% select(pop, chrom, chrom_start, eff)%>% distinct() %>%   group_by(pop, eff) %>% tally() %>% filter(eff == "missense_variant") %>% ggplot(., aes(x = pop, y = n)) + geom_bar(stat = "identity") + theme(axis.text.x = element_text(angle = 90))
```


```{r, eval = FALSE}
lower_sig_stats %>% filter(pop %in% poly_pop) %>% select(pop, SYMBOL) %>% bind_rows(., bind_rows(sig_ihs_val, sig_nsl_val) %>% filter(pop %in% poly_pop) %>% select(pop, "SYMBOL"= genename)) %>% distinct() %>% NROW()

lower_sig_stats %>% select(pop, SYMBOL) %>% bind_rows(., bind_rows(sig_ihs_val, sig_nsl_val) %>% select(pop, "SYMBOL"= genename)) %>% distinct() %>% group_by(pop) %>% tally() %>% ungroup() %>% summarise(mean(n))
```


<!-- Table \@ref(tab:sfsWindowsTable) shows the total number of windows for each of the Polynesian populations and the number of regions that met the significance thresholds in the upper and lower tails of the distribution. The number of genes relates to the number of unique genes that intersected with a window that had met the significance threshold. The number of population genes shows the number of genes that intersected the regions meeting the significance threshold but were only significant for in the respective population. -->
```{r upsetPol, fig.caption = 'POL Upset Plot', eval = FALSE}
UpSetR::upset(fromList(list(`Tajima's D` = lower_sig_stats %>% filter(pop %in% poly_pop, statname == 'Tajima.D', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(), 
                            `Fay and Wu's H` = lower_sig_stats %>% filter(pop %in% poly_pop, statname == 'Fay.Wu.H', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(), 
                            `Fu and Li's D` = lower_sig_stats %>% filter(pop %in% poly_pop, statname == 'Fu.Li.D', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(), 
                            `Fu and Li's F` = lower_sig_stats %>% filter(pop %in% poly_pop, statname == 'Fu.Li.F', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(),
                            `Zeng's E` = lower_sig_stats %>% filter(pop %in% poly_pop, statname == 'Zeng.E', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(),
                            iHS = sig_ihs_val %>% filter( pop %in% poly_pop, !is.na(genename) & genename != "") %>% .[['genename']] %>% unique(), 
                            nSL =sig_nsl_val %>% filter(pop %in% poly_pop, !is.na(genename) & genename != "") %>% .[['genename']] %>% unique())), 
              order.by = c("freq","degree"), nset = 7, number.angles = 90, set_size.angles = 90, hjust = -0.25, vjust = 0.5, nintersects = NA)
```


```{r, warning = FALSE, message = FALSE}
# create the dataframe for number of significant regions and genes in Polynesian populations for each stat
sfs_windows_table <- bind_rows(lapply(c("Tajima.D","Fay.Wu.H","Fu.Li.D","Fu.Li.F","Zeng.E"), function(s){
  full_join(bind_rows(lapply(c("CIM","NZM","SAM","TON"), function(p){
    data.frame(pop = p,stat = s, tail= "Lower",
               window = lower_sig_stats %>% filter(statname == s, pop == p) %>% select(contains('chrom')) %>% distinct() %>% NROW(),
               indep_regions = lower_sig_stats %>% filter(statname == s, pop == p) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length(), 
               sig_genes = lower_sig_stats %>% filter(statname == s, pop == p) %>% select('SYMBOL') %>% distinct() %>% NROW(), 
               pop_only_sig_genes = lower_sig_stats %>% filter(statname == s, pop == p) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == s, pop != p), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW())
  })),
  bind_rows(lapply(c("CIM","NZM","SAM","TON"), function(p){
    data.frame(pop = p,stat = s, tail= "Upper",
               window = upper_sig_stats %>% filter(statname == s, pop == p) %>% select(contains('chrom')) %>% distinct() %>% NROW(),
               indep_regions = upper_sig_stats %>% filter(statname == s, pop == p) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length(), 
               sig_genes = upper_sig_stats %>% filter(statname == s, pop == p) %>% select('SYMBOL') %>% distinct() %>% NROW(), 
               pop_only_sig_genes = upper_sig_stats %>% filter(statname == s, pop == p) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == s, pop != p), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW())
  })), by = "pop")
})
) 

```

Each Polynesian population had a similar number of windows in the 1st percentile, except for the Western Polynesian populations with \gls{ze} where they had up to ~50% less (Table \@ref(tab:sfsWindowsTable)).
The Eastern Polynesian populations tended to have more independent non-consecutive regions and number of genes that were intersected by windows than the Western Polynesian populations. However, this did not translate into a higher number of genes that were significant in only a single population. There were similar numbers of genes that intersected the windows that met the significance threshold in either distribution tail between the Polynesian populations of \gls{cim}, \gls{nzm}, \gls{sam}, and \gls{ton}. 

```{r sfsWindowsTable}
sfs_windows_table %>% 
  select(-tail.x, -tail.y, -stat.x, -stat.y) %>% 
  knitr::kable(., format = 'latex', col.names = c("Pop",rep(c("Windows","Regions (n)","Genes (n)","Pop. Genes (n)"),2)), booktabs = TRUE, caption = "(\\#tab:sfsWindowsTable) Number of significant regions and genes in Polynesian populations for the intra population SFS statistics.") %>% kableExtra::kable_styling(., latex_options = 'scale_down') %>% 
  kableExtra::group_rows("Tajima's D", 1,4) %>%
  kableExtra::group_rows("Fay and Wu's H",5 ,8) %>%
  kableExtra::group_rows("Fu and Li's D", 9, 12) %>%
  kableExtra::group_rows("Fu and Li's F", 13, 16) %>%
  kableExtra::group_rows("Zeng's E", 17, 20) %>% 
  kableExtra::add_header_above(., c("" ,"Lower Tail" = 4,"Upper Tail" = 4)) %>% 
  kableExtra::footnote(general = "Regions is the number of independent non-consecutive regions the significant windows form. Genes is the number of unique genes that the significant windows intersected. Pop. Genes is the number of genes that had significant windows intesect in only that particular population.", threeparttable = TRUE, general_title = "", escape = TRUE)
```

#### Polynesian Super Population



```{r sfsDiseaseGenesData}
# create table for superDiseaseGenesTab
superSFSdisease <- lower_sig_stats %>% select(SYMBOL, pop) %>% left_join(., panel %>% select(pop, super_pop) %>% distinct(), by = 'pop') %>% 
    mutate(
      urate = ifelse(SYMBOL %in% urate_goutGR$SYMBOL, TRUE, FALSE),
      obesity = ifelse(SYMBOL %in% obesity_GR$SYMBOL, TRUE, FALSE),
      t2d = ifelse(SYMBOL %in% t2d_GR$SYMBOL, TRUE, FALSE),
      kd = ifelse(SYMBOL %in% kd_GR$SYMBOL, TRUE, FALSE),
      metsyn = ifelse(SYMBOL %in% metsyn_GR$SYMBOL, TRUE, FALSE)
    ) %>% 
    select(SYMBOL, pop, super_pop, urate, obesity, t2d, kd, metsyn) %>% 
    distinct() %>% filter(urate + obesity + t2d + kd + metsyn > 0)

superHaploDisease <- bind_rows(sig_ihs_val, sig_nsl_val) %>% 
    select(pop, 'SYMBOL' = genename) %>% 
    left_join(., panel %>% select(pop, super_pop) %>% distinct(), by = 'pop') %>% 
    mutate(
      urate = ifelse(SYMBOL %in% urate_goutGR$SYMBOL, TRUE, FALSE),
      obesity = ifelse(SYMBOL %in% obesity_GR$SYMBOL, TRUE, FALSE),
      t2d = ifelse(SYMBOL %in% t2d_GR$SYMBOL, TRUE, FALSE),
      kd = ifelse(SYMBOL %in% kd_GR$SYMBOL, TRUE, FALSE),
      metsyn = ifelse(SYMBOL %in% metsyn_GR$SYMBOL, TRUE, FALSE)
    ) %>% 
    select(SYMBOL, pop, super_pop, urate, obesity, t2d, kd, metsyn) %>% 
    distinct() %>% filter(urate + obesity + t2d + kd + metsyn > 0)

superDiseaseGenesTab <- bind_rows(
   superSFSdisease%>% 
    group_by(pop, super_pop) %>% 
    summarise(urate = sum(urate), obesity = sum(obesity), t2d = sum(t2d), kd = sum(kd), metsyn = sum(metsyn)) %>% 
    mutate(type = "SFS") %>% arrange(super_pop),
  superHaploDisease %>% 
    group_by(pop, super_pop) %>% 
    summarise(urate = sum(urate), obesity = sum(obesity), t2d = sum(t2d), kd = sum(kd), metsyn = sum(metsyn)) %>% 
    mutate(type = 'Haplotype') %>% arrange(super_pop),
  
  inner_join(
    bind_rows(sig_ihs_val, sig_nsl_val) %>% 
      select(pop, 'SYMBOL' = genename),
    lower_sig_stats %>% select(SYMBOL, pop), by = c('SYMBOL','pop')
  ) %>% 
    left_join(., panel %>% select(pop, super_pop) %>% distinct(), by = 'pop') %>%
    mutate(
      urate = ifelse(SYMBOL %in% urate_goutGR$SYMBOL, TRUE, FALSE),
      obesity = ifelse(SYMBOL %in% obesity_GR$SYMBOL, TRUE, FALSE),
      t2d = ifelse(SYMBOL %in% t2d_GR$SYMBOL, TRUE, FALSE),
      kd = ifelse(SYMBOL %in% kd_GR$SYMBOL, TRUE, FALSE),
      metsyn = ifelse(SYMBOL %in% metsyn_GR$SYMBOL, TRUE, FALSE)
    ) %>% 
    select(SYMBOL, pop, super_pop, urate, obesity, t2d, kd, metsyn) %>% 
    distinct() %>% 
    group_by(pop, super_pop) %>%
    summarise(urate = sum(urate), obesity = sum(obesity), t2d = sum(t2d), kd = sum(kd), metsyn = sum(metsyn)) %>% 
    mutate(type = 'Combined') %>% arrange(super_pop)
)


```

```{r sfsDiseaseGenesData2}
x <- superSFSdisease%>% 
  group_by(pop, super_pop) %>% 
  summarise(urate = sum(urate), obesity = sum(obesity), t2d = sum(t2d), kd = sum(kd), metsyn = sum(metsyn)) %>% 
  mutate(type = "SFS") %>% arrange(super_pop)

y <- superHaploDisease %>% 
  group_by(pop, super_pop) %>% 
  summarise(urate = sum(urate), obesity = sum(obesity), t2d = sum(t2d), kd = sum(kd), metsyn = sum(metsyn)) %>% 
  mutate(type = 'Haplotype') %>% arrange(super_pop)

z <- inner_join(bind_rows(sig_ihs_val, sig_nsl_val) %>% 
  select(pop, 'SYMBOL' = genename),
lower_sig_stats %>% select(SYMBOL, pop), by = c('SYMBOL','pop')
) %>% 
  left_join(., panel %>% select(pop, super_pop) %>% distinct(), by = 'pop') %>%
  mutate(
    urate = ifelse(SYMBOL %in% urate_goutGR$SYMBOL, TRUE, FALSE),
    obesity = ifelse(SYMBOL %in% obesity_GR$SYMBOL, TRUE, FALSE),
    t2d = ifelse(SYMBOL %in% t2d_GR$SYMBOL, TRUE, FALSE),
    kd = ifelse(SYMBOL %in% kd_GR$SYMBOL, TRUE, FALSE),
    metsyn = ifelse(SYMBOL %in% metsyn_GR$SYMBOL, TRUE, FALSE)
  ) %>% 
  select(SYMBOL, pop, super_pop, urate, obesity, t2d, kd, metsyn) %>% 
  distinct() %>% 
  group_by(pop, super_pop) %>%
  summarise(urate = sum(urate), obesity = sum(obesity), t2d = sum(t2d), kd = sum(kd), metsyn = sum(metsyn)) %>% 
  mutate(type = 'Combined') %>% arrange(super_pop)  

superDiseaseGeneTab2 <- full_join(x,y, suffix = c("SFS","HAP"), by = c("pop","super_pop")) %>% full_join(., z, by = c("pop","super_pop")) %>% unite(., "urate", contains("urate"), sep = '; '  ) %>%  unite(., "obesity", contains("obesity"), sep = '; '  ) %>% unite(., "t2d", contains("t2d"), sep = '; '  ) %>% unite(., "kd", contains("kd"), sep = '; '  ) %>% unite(., "metsyn", contains("metsyn"), sep = '; '  ) %>% select(-contains('type'))


superDiseaseGeneTab3 <- full_join(x,y, suffix = c("SFS","HAP"), by = c("pop","super_pop")) %>% full_join(., z, by = c("pop","super_pop")) %>% select(pop, super_pop, contains("urate"), contains("obesity"), contains("t2d"), contains("kd"), contains("metsyn"))
  
rm(x,y,z)

```



```{r polySigMetGenes}
#table to record all regions that have evidence of postitive selection in genelist for pop
create_gene_table <- function(popname, genelist){
  list(
    td = lower_sig_stats %>% filter(pop == popname, statname == "Tajima.D", SYMBOL %in% genelist) %>% select(chrom, SYMBOL, statname, pop) %>% group_by(SYMBOL, statname) %>% tally(),
    fwh = lower_sig_stats %>% filter(pop == popname, statname == "Fay.Wu.H", SYMBOL %in% genelist)%>% select(chrom, SYMBOL, statname, pop) %>% group_by(SYMBOL, statname) %>% tally(),
    fld = lower_sig_stats %>% filter(pop == popname, statname == "Fu.Li.D", SYMBOL %in% genelist)%>% select(chrom, SYMBOL, statname, pop) %>% group_by(SYMBOL, statname) %>% tally(),
    flf = lower_sig_stats %>% filter(pop == popname, statname == "Fu.Li.F", SYMBOL %in% genelist)%>% select(chrom, SYMBOL, statname, pop) %>% group_by(SYMBOL, statname) %>% tally(),
    ze = lower_sig_stats %>% filter(pop == popname, statname == "Zeng.E", SYMBOL %in% genelist)%>% select(chrom, SYMBOL, statname, pop) %>% group_by(SYMBOL, statname) %>% tally(),
    ihs = sig_ihs_val %>% filter(pop == popname, genename %in% genelist)%>% select(chrom, 'SYMBOL' = genename)  %>% mutate(pop = popname, statname = 'ihs') %>% group_by(SYMBOL, statname) %>% tally(), 
    nsl = sig_nsl_val %>% filter(pop == popname, genename %in% genelist)%>% select(chrom, 'SYMBOL' = genename)  %>% mutate(pop = popname, statname = 'nsl') %>% group_by(SYMBOL,statname) %>% tally(),
    xpehh = c(),#xpehh %>% filter(pop1 == popname, genename %in% genelist, xpehh_value > 0) %>% select(chrom, 'SYMBOL' = genename) %>% mutate(pop = popname, statname = 'xpehh') %>% group_by(SYMBOL, statname) %>% tally(),
    fst = c()
  )
}

cim_urate_sig <- create_gene_table("CIM", urate_goutGR$SYMBOL)
cim_t2d_sig <- create_gene_table("CIM", t2d_GR$SYMBOL)
cim_obesity_sig <- create_gene_table("CIM", obesity_GR$SYMBOL)
cim_kd_sig <- create_gene_table("CIM", kd_GR$SYMBOL)
cim_metsyn_sig <- create_gene_table("CIM", metsyn_GR$SYMBOL)

nzm_urate_sig <- create_gene_table("NZM", urate_goutGR$SYMBOL)
nzm_t2d_sig <- create_gene_table("NZM", t2d_GR$SYMBOL)
nzm_obesity_sig <- create_gene_table("NZM", obesity_GR$SYMBOL)
nzm_kd_sig <- create_gene_table("NZM", kd_GR$SYMBOL)
nzm_metsyn_sig <- create_gene_table("NZM", metsyn_GR$SYMBOL)

sam_urate_sig <- create_gene_table("SAM", urate_goutGR$SYMBOL)
sam_t2d_sig <- create_gene_table("SAM", t2d_GR$SYMBOL)
sam_obesity_sig <- create_gene_table("SAM", obesity_GR$SYMBOL)
sam_kd_sig <- create_gene_table("SAM", kd_GR$SYMBOL)
sam_metsyn_sig <- create_gene_table("SAM", metsyn_GR$SYMBOL)

ton_urate_sig <- create_gene_table("TON", urate_goutGR$SYMBOL)
ton_t2d_sig <- create_gene_table("TON", t2d_GR$SYMBOL)
ton_obesity_sig <- create_gene_table("TON", obesity_GR$SYMBOL)
ton_kd_sig <- create_gene_table("TON", kd_GR$SYMBOL)
ton_metsyn_sig <- create_gene_table("TON", metsyn_GR$SYMBOL)
```

Out of all the genes that met significance, the Polynesian super population had the fewest genes unique to a super population (Figure \@ref(fig:superUpset)). The upset plot shows number of genes that were in common between super populations that had both intra-populational haplotypic and \gls{sfs} evidence. There were 117 genes that only had evidence of selection in Polynesian populations, and 12 genes that had evidence in all super populations.

The number of genes that had at least one marker that was significant in the haplotypic tests, and one window that intersected the same gene that was in the 1st percentile for a frequency based statistic and were shared between all of the Polynesian populations was `r length(intersect(intersect(extract_sig("CIM"), extract_sig("NZM")), intersect(extract_sig("SAM"), extract_sig("TON"))))` (Figure \@ref(fig:polyUpset)). The number of genes in common between the Eastern Polynesian populations that had both intra-populational haplotypic and \gls{sfs} evidences was `r length(intersect(extract_sig("CIM"), extract_sig("NZM")))`. There was about 30% more genes for the same criteria in the Western Polynesians, with `r length(intersect(extract_sig("SAM"), extract_sig("TON")))` genes. Between the individual Polynesian populations, roughly one third of the genes that were significant in a population were not in common with the other Polynesian populations. The \gls{nzm} population had the most genes (395) that were not shared with any of the other Polynesian populations. 

```{r, eval = FALSE}
bind_rows(sig_ihs_val, sig_nsl_val) %>% filter(chrom == 5, chrom_start > 172483355, chrom_end < 173000000  )
```

##### Prevously reported Polynesian "thrifty-genes"

There was no signal of selection for the previously reported 'thrifty genes' of _PPARGC1A_ and _CREBRF_ [@Myles2011; @Minster2016]. There were no markers or windows that met the significance thresholds for any of the neutrality and selection statistics. This was consistent with the results of @Cadzow2016 for _PPARGC1A_. The two identified \gls{snp}s for \gls{bmi} in Samoans at _CREBRF_, rs12513649 and rs373863828 [@Minster2016] were both absent from the CoreExome chip. While there was no selection signal from \gls{ihs} or \gls{nsl} at _CREBRF_ (chr5:172483355-172566291), downstream at chr5:172,800,000-173,000,000 there were four markers for \gls{sam} and seven for \gls{ton} that were significant for \gls{ihs}. At the same location for \gls{nsl} there were six significant markers for \gls{sam} and seven for \gls{ton}. The populations from the \gls{eas} super population had fewer significant markers for both \gls{ihs} and \gls{nsl}. The significant markers at this location in \gls{sam} were consistent with @Minster2016, were intergenic, and intersected the long intergenic non-coding RNA, _CTB-164N12.1_.

##### Genes with possible selection in Polynesian populations

A complete list of all of the genes that had windows in the 1st percentile and values < 0 for the Polynesian populations can be found in table \@ref(tab:intrasfsPol). A complete list of genes that had significant \gls{ihs} or \gls{nsl} in Polynesian populations can be found in Tables \@ref(tab:ihsPolgenes), \@ref(tab:nslPolgenes), and \@ref(tab:xpehhPolgenes).



(ref:superUpset) Upset plot showing the intersections of genes that had both intra-population haplotypic and frequency spectrum-evidence at the individual population level and were then pooled into their super population group. The left histogram shows the total number of genes  with both haplotypic and frequency spectrum-based evidence for each super population. The top histogram is the number of genes that were in common between the super populations. The dots indicate the particular set of super populations for the intersection.

```{r superUpset, fig.cap = '(ref:superUpset)', eval = TRUE}

p <- panel %>% mutate(pop2 = paste0(super_pop,'_',pop)) %>% select(pop,pop2) %>% filter(!pop %in% c('WPN','EPN','POL','NAD')) %>% distinct()
pop_genes <- lapply(p$pop, extract_sig)
names(pop_genes) <- p$pop2

superpop_genes <- lapply(c('AFR','AMR','EUR','EAS', 'SAS','POL'), extract_sig_super)
names(superpop_genes) <- c('AFR','AMR','EUR','EAS', 'SAS','POL')

# create an upset plot for the intersection of genes between the super populations
# the key is that each individual population first is 'filtered' for genes that have both haplotypic and sfs significance
UpSetR::upset(fromList(superpop_genes),
                             order.by = c("freq","degree"), nsets = length(names(superpop_genes)), number.angles = 90, set_size.angles = 90, hjust = -0.25, vjust = 0.5, nintersects = NA)
rm(p)
```

(ref:polyUpset) Upset plot showing the number of genes that have evidence from \gls{ihs} or \gls{nsl}, and met the lower threshold from any of \gls{td}, \gls{fwh}, \gls{fld}, \gls{flf}, or \gls{ze}, in the four Polynesian populations. The left histogram shows the total number of genes  with both haplotypic and frequency spectrum-based evidence for each Polynesian population. The top histogram is the number of genes that were in common between the Polynesian populations. The dots indicate the particular set of super populations for the intersection.

```{r polyUpset, fig.cap = '(ref:polyUpset)'}
# show the intersection of the genes that were deemed significant between the different polynesian populations
UpSetR::upset(fromList(list(CIM = extract_sig("CIM"), 
                            NZM = extract_sig("NZM"), 
                            SAM = extract_sig("SAM"), 
                            TON = extract_sig("TON"))),
                             order.by = "freq", nsets = 4, number.angles = 90, set_size.angles = 90, hjust = -0.25, vjust = 0.5, nintersects = NA)
```




There were `r sig_ihs_val %>% anti_join(., sig_ihs_val %>% filter(!pop %in% poly_pop), by = c('genename')) %>% select(pop, genename) %>% distinct() %>% group_by(genename) %>% tally() %>% filter( n == 4) %>% NROW()` genes (_ANO1-AS1_, _CCDC180_, _COL6A3_, _ESPNL_, _GRIP1_, _LINC00661_, _LINC01006_, _NPAT_, _NPFFR1_, and _PPA1_) that all Polynesian populations and no other super populations had at least one \gls{snp} with a significant \gls{ihs} marker, and there were `r sig_ihs_val %>% anti_join(., sig_ihs_val %>% filter(!pop %in% poly_pop), by = c('genename')) %>% select(pop, genename) %>% distinct() %>% group_by(genename) %>% tally() %>% filter(n > 0) %>% NROW()` genes that at least one Polynesian population and no other super populations had. The Eastern Polynesian populations had `r sig_ihs_val %>% anti_join(., sig_ihs_val %>% filter(!(pop %in% c("SAM","TON"))), by = c('genename')) %>% select(pop, genename) %>% distinct() %>% group_by(genename) %>% tally() %>% filter(n > 0) %>% NROW()` genes, similarly, the Western Polynesian populations had `r sig_ihs_val %>% anti_join(., sig_ihs_val %>% filter( !(pop %in% c("CIM","NZM"))), by = c('genename')) %>% select(pop, genename) %>% distinct() %>% group_by(genename) %>% tally() %>% filter(n > 0) %>% NROW()` genes. This compared to the significant \gls{nsl} markers where there were `r sig_nsl_val %>% anti_join(., sig_nsl_val %>% filter(!pop %in% poly_pop), by = c('genename'))  %>% select(pop, genename) %>% distinct() %>% group_by(genename) %>% tally() %>% filter( n == 4) %>% NROW()` genes, that had at least one significant marker, and were found in only all four Polynesian populations. Within only Eastern Polynesian populations, there were `r sig_nsl_val %>% anti_join(., sig_nsl_val %>% filter(!pop %in% c("SAM","TON")), by = c('genename')) %>% select(pop, genename) %>% distinct() %>% group_by(genename) %>% tally() %>% filter(n > 0) %>% NROW()` genes that had a significant marker, and within the Western Polynesian populations, there were `r sig_nsl_val %>% anti_join(., sig_nsl_val %>% filter(!pop %in% c("CIM","NZM")), by = c('genename')) %>% select(pop, genename) %>% distinct() %>% group_by(genename) %>% tally() %>% filter(n > 0) %>% NROW()` genes only significant in that group. When the intersection of Polynesian specific genes for both \gls{ihs} and \gls{nsl} was compared, the number of genes in common was `r inner_join(sig_ihs_val %>% anti_join(., sig_ihs_val %>% filter(!pop %in% poly_pop), by = c('genename')) %>% filter(pop %in% poly_pop) %>% select(pop, genename) %>% distinct() %>% group_by(genename) %>% tally() %>% filter( n == 4), sig_nsl_val %>% anti_join(., sig_nsl_val %>% filter(!pop %in% poly_pop), by = c('genename')) %>% filter(pop %in% poly_pop) %>% select(pop, genename) %>% distinct() %>% group_by(genename) %>% tally() %>% filter( n == 4), by = 'genename') %>% NROW()` that had at least one \gls{snp} in only the four Polynesian populations. These were _GRIP1_ and _LINC00661_. 

##### Selection in genes associated with urate and metabolic disease

Figure \@ref(fig:intraSelGenes) shows the genes associated with urate, gout and related diseases that had a significant marker with either \gls{ihs} or \gls{nsl} and also had a significant window from a site-frequency spectrum based statistic in at least one Polynesian population. There are only 12 genes that met this criteria. The \gls{t2d} associated gene, _PTPRD_, was the only gene that met the mentioned criteria that had significant markers from all Polynesian populations in \gls{ihs}. The obesity-associated genes _GRID1_, _FHIT_, and _ERBB4_, all had windows from \gls{fwh} that were significant in all Polynesian populations. _ERBB4_ also had significant windows for all Polynesian populations for \gls{td}.

(ref:intraSelGenes) Genes associated with urate, gout, obesity, \gls{t2d}, kidney disease, and metabolic syndrome having both significance in intra-population SFS and haplotypic based methods in at least one Polynesian population. Numbers inside boxes indicate the number of significant windows (\gls{sfs} methods) or \gls{snp}s (haplotypic methods) for the population. 

```{r intraSelGenes, fig.cap = '(ref:intraSelGenes)', fig.width=7.5}
# genes that have at least one poly population with significance in both haplotypic and sfs stats
rbind(
bind_rows(cim_obesity_sig, cim_urate_sig, cim_t2d_sig, cim_metsyn_sig, cim_kd_sig) %>% mutate(pop = 'CIM') %>% distinct(),
bind_rows(nzm_obesity_sig, nzm_urate_sig, nzm_t2d_sig, nzm_metsyn_sig, nzm_kd_sig) %>% mutate(pop = 'NZM') %>% distinct(),
bind_rows(sam_obesity_sig, sam_urate_sig, sam_t2d_sig, sam_metsyn_sig, sam_kd_sig) %>% mutate(pop = "SAM") %>% distinct(),
bind_rows(ton_obesity_sig, ton_urate_sig, ton_t2d_sig, ton_metsyn_sig, ton_kd_sig) %>% mutate(pop = 'TON') %>% distinct()
)  %>% spread(statname, n) %>% filter(SYMBOL %in% (mutate(., haplo = ifelse(!is.na(ihs) | !is.na(nsl), 1, 0), sfs = ifelse(!is.na(Fay.Wu.H) | !is.na(Fu.Li.D) | !is.na(Fu.Li.F) | !is.na(Tajima.D) | !is.na(Zeng.E), 1, 0)) %>% 
  filter(haplo & sfs) %>% .[["SYMBOL"]] %>% unique() %>% as.vector())) %>% 
  gather(key = statname, value=n, 3:NCOL(.)) %>% 
  mutate(sn = statname) %>% 
  mutate(statname = case_when(statname == "Fay.Wu.H" ~ "Fay and Wu's H", 
                              statname == "Tajima.D" ~ "Tajima's D", 
                              statname == 'Fu.Li.D'~ "Fu and Li's D", 
                              statname == 'Fu.Li.F' ~ "Fu and Li's F",
                              statname == 'Zeng.E' ~ "Zeng's E", 
                              statname == 'ihs' ~ 'iHS', 
                              statname == 'nsl'~"nSL")) %>% 
  mutate(statname = factor(statname, levels = c("Fay and Wu's H", "Tajima's D", "Fu and Li's D", "Fu and Li's F", "Zeng's E", "iHS","nSL"), ordered = TRUE)) %>%  
  filter(!is.na(n)) %>% ggplot(., aes(x = pop, y = SYMBOL))  + geom_tile(aes(colour = pop, fill = pop)) + theme_bw() + facet_grid(~statname) + geom_text(inherit.aes = TRUE, aes(label = n)) + theme(panel.grid = element_blank(), axis.text.x = element_text(angle = 90, vjust = 0.5), legend.position = 'bottom') + xlab('Population') + scale_fill_discrete(name = "Population") + guides(colour = FALSE) + scale_y_discrete(expand = c(0,0)) + scale_x_discrete(expand = c(0,0)) 
```

##### Pathway enrichment of genome-wide selection results

Pathway terms from the pathway enrichment analysis using the Enrichr KEGG 2016 table showed there was minimal overlap in the terms that were significant between the Polynesian populations (Table \@ref(tab:keggPath)). There were more pathways in common between the Eastern Polynesian populations than the Western Polynesian populations.

```{r keggPath, results = 'asis'}
kegg_sig %>% 
  filter(Adjusted.P.value < 0.05) %>% 
  select(Term, pop ) %>% 
  mutate(pop = toupper(pop) ,yes = "Yes") %>% 
  distinct() %>% spread(pop, yes, "") %>% 
  mutate(term2 = sapply(Term, function(x){strsplit(x, "_")[[1]][1]})) %>% select(-Term) %>% 
  select( term2, everything()) %>% 
  kable(., col.names = c("Pathway Term","CIM","NZM","SAM", "TON"), format = 'latex', booktabs = TRUE, caption = '(\\#tab:keggPath) Pathway terms that were significant in Polynesian populations in Enrichr KEGG 2016 pathway enrichment analysis', longtable = FALSE, linesep = "") %>% add_header_above(., c("", "Population" = 4)) %>% kable_styling(., latex_options = 'scale_down', font_size = 10)   
```

##### Regions of haplotypic selection in Polynesian populations

The Polynesian populations had the least number of significant \gls{snp}s for both \gls{ihs} and \gls{nsl} (Table \@ref(tab:numSigWindows)).The mean number of significant \gls{snp}s for \gls{ihs} was `r sig_ihs_val %>% group_by(pop) %>% tally() %>% ungroup() %>% summarise(round_dp(mean(n), 1))` (SD  `r sig_ihs_val %>% group_by(pop) %>% tally() %>% ungroup() %>% summarise(round_dp(sd(n), 1))`) and `r sig_nsl_val %>% group_by(pop) %>% tally() %>% ungroup() %>% summarise(round_dp(mean(n), 1))`  (SD `r sig_nsl_val %>% group_by(pop) %>% tally() %>% ungroup() %>% summarise(round_dp(sd(n), 1))`) for \gls{nsl} (Table \@ref(tab:numSigWindows)). The Polynesian populations had a mean of `r sig_ihs_val %>% group_by(pop,superpop) %>% tally() %>% group_by(superpop) %>% summarise(mn = mean(n)) %>% filter(superpop == 'POL') %>% .[['mn']] %>% sprintf("%.0f",.)` \gls{snp}s with an \gls{ihs} value that was significant, compared to  `r sig_nsl_val %>% group_by(pop,superpop) %>% tally() %>% group_by(superpop) %>% summarise(mn = mean(n)) %>% filter(superpop == 'POL') %>% .[['mn']] %>% sprintf("%.0f",.)` for \gls{nsl}. The minimum number of \gls{snp}s that had a significant \gls{ihs} was `r sig_ihs_val %>% group_by(pop) %>% tally() %>% ungroup() %>% filter(min(n) == n) %>% .[['n']]` and was from the \gls{`r sig_ihs_val %>% group_by(pop) %>% tally() %>% ungroup() %>% filter(min(n) == n) %>% .[['pop']] %>% tolower()`} population. For \gls{nsl} this was also \gls{`r sig_nsl_val %>% group_by(pop) %>% tally() %>% ungroup() %>% filter(min(n) == n)  %>% .[['pop']] %>% tolower`} with `r sig_nsl_val %>% group_by(pop) %>% tally() %>% ungroup() %>% filter(min(n) == n) %>% .[['n']]` \gls{snp}s. The maximum number of \gls{snp}s was for \gls{`r sig_ihs_val %>% group_by(pop) %>% tally() %>% ungroup() %>% filter(max(n) == n) %>% .[['pop']] %>% tolower()`} with `r sig_ihs_val %>% group_by(pop) %>% tally() %>% ungroup() %>% filter(max(n) == n) %>% .[['n']]` for \gls{ihs} and \gls{`r sig_nsl_val %>% group_by(pop) %>% tally() %>% ungroup() %>% filter(max(n) == n) %>% .[['pop']] %>% tolower()`} with `r sig_nsl_val %>% group_by(pop) %>% tally() %>% ungroup() %>% filter(max(n) == n) %>% .[['n']]` \gls{snp}s. Between significant \gls{snp}s for \gls{ihs} and \gls{nsl} there was a mean of `r ihs_nsl_sum_tab %>% left_join(., panel %>% select(pop, super_pop) %>% distinct(), by = 'pop') %>% filter(super_pop == 'POL') %>% summarise(round_dp(mean(n), 1))` \gls{snp}s in common for the \gls{pol} populations, `r ihs_nsl_sum_tab %>% left_join(., panel %>% select(pop, super_pop) %>% distinct(), by = 'pop') %>% filter(super_pop == 'EAS') %>% summarise(round_dp(mean(n), 1))` for \gls{eas}, `r ihs_nsl_sum_tab %>% left_join(., panel %>% select(pop, super_pop) %>% distinct(), by = 'pop') %>% filter(super_pop == 'AFR') %>% summarise(round_dp(mean(n), 1))` \gls{afr}, `r ihs_nsl_sum_tab %>% left_join(., panel %>% select(pop, super_pop) %>% distinct(), by = 'pop') %>% filter(super_pop == 'EUR') %>% summarise(round_dp(mean(n), 1))` for \gls{eur}, `r ihs_nsl_sum_tab %>% left_join(., panel %>% select(pop, super_pop) %>% distinct(), by = 'pop') %>% filter(super_pop == 'AMR') %>% summarise(round_dp(mean(n), 1))` for \gls{amr}, and `r ihs_nsl_sum_tab %>% left_join(., panel %>% select(pop, super_pop) %>% distinct(), by = 'pop') %>% filter(super_pop == 'SAS') %>% summarise(round_dp(mean(n), 1))` from \gls{sas}.

```{r nSharedHaplo, eval = TRUE}
join_ihs_nsl <- function(p1, p2){
  inner_join(bind_rows(sig_ihs_val, sig_nsl_val) %>% filter(pop == p1) %>% select(chrom, chrom_start, chrom_end, statid), bind_rows(sig_ihs_val, sig_nsl_val) %>% filter(pop == p2) %>% select(chrom, chrom_start, chrom_end, statid), by = c("chrom", "chrom_start", "chrom_end", "statid")) %>% group_by(statid) %>% tally() %>% mutate(p1 = p1, p2 = p2, statid = case_when(statid == 20 ~ 'nsl', statid == 26 ~ 'ihs')) %>% spread(statid, n)
}
p <- panel %>% select(pop) %>% distinct() %>% pull(pop)

n_shared_haplo <- map(poly_pop, function(x){map2(x, p, ~ join_ihs_nsl(.x, .y)) %>% bind_rows()}) %>% bind_rows()

inner_join(n_shared_haplo %>% select(p1, p2, ihs) %>% spread(p1, ihs), n_shared_haplo %>% select(p1, p2, nsl) %>% spread(p1, nsl), by = 'p2', suffix = c('_ihs', '_nsl')) %>% left_join(., panel %>% select(p2 = pop, super_pop) %>% distinct(), by = 'p2') %>% arrange(super_pop) %>% select(-super_pop) %>% kable(., col.names = c("Population",rep(c("CIM","NZM","SAM","TON"),2)), format = 'latex', booktabs = TRUE, caption = '(\\#tab:nSharedHaplo) Number of significant SNPs that were in common between populations.') %>% kableExtra::add_header_above(., c("", "iHS" = 4,"nSL" =4)) %>%  kableExtra::group_rows("AFR", 1, 7) %>% 
  kableExtra::group_rows("AMR", 8, 11) %>% 
  kableExtra::group_rows("EAS", 12, 16) %>% 
  kableExtra::group_rows("EUR", 17, 22) %>% 
  kableExtra::group_rows("POL", 23, 26) %>% 
  kableExtra::group_rows("SAS", 27,31) 
rm(p)
```


On average there were `r n_shared_haplo %>% filter(!p2 %in% poly_pop) %>% summarise(mean(ihs)) %>% round(., 1)` \gls{snp}s in common between the Polynesian populations and the other populations for \gls{ihs} and `r n_shared_haplo %>% filter(!p2 %in% poly_pop) %>% summarise(mean(nsl)) %>% round(., 1)` \gls{snp}s for \gls{nsl}. Whereas, there was a higher number of significant \gls{snp}s in common between the Polynesian populations with a mean of `r n_shared_haplo %>% filter(p2 %in% poly_pop, p1 != p2) %>% summarise(mean(ihs)) %>% round(., 1)` for \gls{ihs} and `r n_shared_haplo %>% filter(p2 %in% poly_pop, p1 != p2) %>% summarise(mean(nsl)) %>% round(., 1)` for \gls{nsl} (Table \@ref(tab:nSharedHaplo)). The Eastern/Western Polynesian split was also evident with there being a 2-4 fold difference in the number of \gls{snp}s in common between the Eastern and Western Polynesian populations for both \gls{ihs} and \gls{nsl}. There were 24 genes that had at least 1 marker significant across at least 20 populations for all Polynesian populations for \gls{xpehh}. The Eastern Polynesian populations had 28 genes, and the Western Polynesian populations had 49.

<!-- - 24 genes that had at least 20 populations with significant markers in all polynesian populations -->
<!-- - 28 genes that had at least 20 populations with significant markers in only Eastern Polynesian populations  -->
<!-- - 49 genes that had at least 20 populations with significant markers for \gls{xpehh} in only Western Polynesian populations  -->

```{r, eval = FALSE}
xpehh %>%  select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% filter(pop1 %in% c("CIM","NZM","SAM","TON"), xpehh_value > 0) %>% select(pop1,pop2, genename) %>% distinct() %>% left_join(., panel %>% select(pop2 = pop, super_pop) %>% distinct(), by = 'pop2') %>% group_by(pop1, super_pop, genename) %>% tally() %>% spread(super_pop, n, 0)%>% arrange(genename) %>% mutate(rs = AFR + AMR + EAS + EUR + POL + SAS) %>% filter(rs >=20) %>% kable(., caption = 'Number of populations by super population, with significant xpehh for urate/gout associated genes', booktabs = TRUE)
```


```{r eval = FALSE}
#all poly pops
xpehh %>%  select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% filter(pop1 %in% c("CIM","NZM","SAM","TON"), xpehh_value > 0) %>% select(pop1,pop2, genename) %>% distinct() %>% left_join(., panel %>% select(pop2 = pop, super_pop) %>% distinct(), by = 'pop2') %>% group_by(pop1, super_pop, genename) %>% tally() %>% spread(super_pop, n, 0)%>% arrange(genename) %>% mutate(rs = AFR + AMR + EAS + EUR + POL + SAS) %>% filter(rs >=20) %>% select(pop1, genename, rs) %>% group_by(genename) %>% tally() %>% filter( n == 4, !is.na(genename)) %>% pull(genename)

#ep poly
xpehh %>%  select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% filter(pop1 %in% c("CIM","NZM","SAM","TON"), xpehh_value > 0) %>% select(pop1,pop2, genename) %>% distinct() %>% left_join(., panel %>% select(pop2 = pop, super_pop) %>% distinct(), by = 'pop2') %>% group_by(pop1, super_pop, genename) %>% tally() %>% spread(super_pop, n, 0)%>% arrange(genename) %>% mutate(rs = AFR + AMR + EAS + EUR + POL + SAS) %>% filter(rs >=20) %>% select(pop1, genename, rs) %>% spread(pop1, rs) %>% filter(!is.na(CIM), !is.na(NZM), is.na(SAM), is.na(TON)) %>% pull(genename)

#wp poly
xpehh %>%  select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% filter(pop1 %in% c("CIM","NZM","SAM","TON"), xpehh_value > 0) %>% select(pop1,pop2, genename) %>% distinct() %>% left_join(., panel %>% select(pop2 = pop, super_pop) %>% distinct(), by = 'pop2') %>% group_by(pop1, super_pop, genename) %>% tally() %>% spread(super_pop, n, 0)%>% arrange(genename) %>% mutate(rs = AFR + AMR + EAS + EUR + POL + SAS) %>% filter(rs >=20) %>% select(pop1, genename, rs) %>% spread(pop1, rs) %>% filter(is.na(CIM), is.na(NZM), !is.na(SAM), !is.na(TON)) %>% pull(genename)
```



```{r, eval = FALSE}
xpehh %>%  
  select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% 
  filter(pop1 %in% poly_pop, xpehh_value > 0) %>% 
  filter(genename %in% c(urate_goutGR$SYMBOL, t2d_GR$SYMBOL, obesity_GR$SYMBOL, metsyn_GR$SYMBOL)) %>% 
  select(pop = pop1,pop2, genename) %>% 
  distinct() %>% 
  group_by(pop, genename)%>% tally() %>% 
  inner_join(., bind_rows(sig_ihs_val, sig_nsl_val) %>% 
               select(pop,genename) %>% distinct(), by = c('pop','genename')) %>% 
  spread(pop, n, fill = "") %>% 
  anti_join(., bind_rows(sig_ihs_val, sig_nsl_val) %>% 
              select(pop, genename) %>% 
              filter(!pop %in% poly_pop) %>% 
              distinct() %>% 
              select(genename), by = "genename") %>% 
  kable(., caption = 'number of populations with significant xpehh, and significant ihs/nsl only in Polynesian in the urate and related disease associated genes.', longtable = TRUE, booktabs = TRUE)
```


##### F~ST~

F~ST~ was calculated in sliding windows of 10 kb across the genome, pair-wise between the Polynesian populations and all other populations. Similar to the results in section \@ref(wholeFst) for the F~ST~ calculated using entire chromosomes at once, the mean F~ST~ using windows had the Polynesian populations most differentiated from the \gls{afr} populations, with a range of mean F~ST~ from 0.148 to 0.203. The Polynesian populations were least differentiated from the \gls{eas} populations with a range of F~ST~ means from 0.051 to 0.076. Between the Polynesian populations the mean F~ST~ ranged from 0.004, between the Western Polynesian populations of \gls{sam} and \gls{ton}, to 0.029 between \gls{nzm} and \gls{sam}. The Eastern Polynesian populations had a marginally higher mean F~ST~ than the Western Polynesian populations at 0.006. The largest maximum F~ST~ of 0.888, across four windows, was between \gls{ton} and \gls{msl} at chr17:62455002-62495001. This same region was the maximum F~ST~ for the \gls{cim} and \gls{nzm}, and the second largest for \gls{sam}, all with \gls{msl}. The maximum F~ST~ between the Polynesian populations was between `r intra_pol_max_fst %>% ungroup() %>% filter(max(fst)==fst) %>% .[['pop1']]` and `r intra_pol_max_fst %>% ungroup() %>% filter(max(fst)==fst) %>% .[['pop2']]`, with a value of `r intra_pol_max_fst %>% ungroup() %>% filter(max(fst)==fst) %>% .[['fst']] %>% three_dp()` at `r as.character(intra_pol_max_fst %>% ungroup() %>% filter(max(fst)==fst) %>% .[['chrom']])`:`r as.character(intra_pol_max_fst %>% ungroup() %>% filter(max(fst)==fst) %>% .[['chrom_start']])`-`r as.character(intra_pol_max_fst %>% ungroup() %>% filter(max(fst)==fst) %>% .[['chrom_end']])`. The smallest maximum F~ST~ of the Polynesian populations was between `r intra_pol_max_fst %>% ungroup() %>% filter((min(fst))==fst) %>% .[['pop1']]` and `r intra_pol_max_fst %>% ungroup() %>% filter(min(fst)==fst) %>% .[['pop2']]`, with a value of `r intra_pol_max_fst %>% ungroup() %>% filter(min(fst)==fst) %>% .[['fst']] %>% three_dp()` at `r as.character(intra_pol_max_fst %>% ungroup() %>% filter(min(fst)==fst) %>% .[['chrom']])`:`r as.character(intra_pol_max_fst %>% ungroup() %>% filter(min(fst)==fst) %>% .[['chrom_start']])`-`r as.character(intra_pol_max_fst %>% ungroup() %>% filter(min(fst)==fst) %>% .[['chrom_end']])`.

<!-- XXX maybe go into more detail about the differences esp between EP and WP -->

```{r, eval = FALSE}
pol_fst_99 %>% separate(pop, c('pop1','pop2'), '_' ) %>% filter(pop1 %in% poly_pop, pop2 %in% poly_pop) %>% filter(pop1 == "CIM", pop2 == 'NZM')
```



<!-- ##### NOTES -->

<!-- PPARGC1A -->

<!-- - No significant snps in poly for PPARGC1A for \gls{ihs} or \gls{nsl} -->
<!-- - in the 1st percentile there were no windows for PPARGC1A -->
<!-- - the 99th percentile had fld windows for NZM, SAM, and TON -->

<!-- CREBRF -->

<!-- - no polynesian populations had any windows, 1st or 99th  -->
<!-- - no significant snps for \gls{ihs} or \gls{nsl} -->

```{r, eval = FALSE}
# find the genes that were in all 4 poly pops and in both ihs and nsl
inner_join(sig_ihs_val %>% anti_join(., sig_ihs_val %>% filter(!pop %in% poly_pop), by = c('genename')) %>% filter(pop %in% poly_pop) %>% select(pop, genename) %>% distinct() %>% group_by(genename) %>% tally() %>% filter( n == 4), sig_nsl_val %>% anti_join(., sig_nsl_val %>% filter(!pop %in% poly_pop), by = c('genename')) %>% filter(pop %in% poly_pop) %>% select(pop, genename) %>% distinct() %>% group_by(genename) %>% tally() %>% filter( n == 4), by = 'genename')
```

```{r, eval = FALSE}
# use anti join to remove all gene names that are found in non-polynesisan populations and then find all genes that had all poly
# ie returns genes that were ONLY significant in all 4 Polynesian populations 
sig_ihs_val %>% anti_join(., sig_ihs_val %>% filter(!pop %in% poly_pop), by = c('genename')) %>% filter(pop %in% poly_pop) %>% select(pop, genename) %>% distinct() %>% group_by(genename) %>% tally() %>% filter( n == 4)
```


\FloatBarrier

#### Cook Island Maori

The \gls{cim} population was the population that had the largest number of markers in a gene that was significant for \gls{ihs}, out of all populations with 27 significant markers in _CNTN4_. 

```{r cim_specific, warning = FALSE}
cim_0.1_genes_lower <- lower_sig_stats %>% left_join(., lower_sig_stats_binned, by = c('chrom','chrom_start','chrom_end','statname', 'pop')) %>% filter(quant == 0.1, pop == 'CIM') %>% select(statname, SYMBOL) %>% distinct() %>% group_by(statname) %>%  tally()

cim_0.1_stats_lower <- lower_sig_stats %>% left_join(., lower_sig_stats_binned, by = c('chrom','chrom_start','chrom_end','statname', 'pop')) %>% filter(quant == 0.1, pop == 'CIM') %>% group_by(statname) %>%  tally()

cim_99.9_genes_upper <- upper_sig_stats %>% left_join(., upper_sig_stats_binned, by = c('chrom','chrom_start','chrom_end','statname', 'pop')) %>% filter(quant == 99.9, pop == 'CIM') %>% select(statname, SYMBOL) %>% distinct() %>% group_by(statname) %>%  tally()

cim_99.9_stats_upper <- upper_sig_stats %>% left_join(., upper_sig_stats_binned, by = c('chrom','chrom_start','chrom_end','statname', 'pop')) %>% filter(quant == 99.9, pop == 'CIM') %>% group_by(statname) %>%  tally()
```

##### Pathway enrichment analysis of selected loci in \gls{cim}

<!-- Functional classification of genes with markers that were significant for \gls{ihs} or \gls{nsl}, or genes that fell within the windows of the 1st percentile and had a value less than 0 for the frequency spectrum based statistics was done using PantherDB. Only \gls{td} had the main biological process as "Binding", and the second as "Catalytic activity". All other statistics each had "Cellular process" and "Metabolic process" as their first and second largest categories. There were no biological processes that were over represented from the gene list of all genes with possible positive selection. -->

Pathway gene-set enrichment analysis on gene lists for each statistic that met the significance thresholds was done using Enrichr. There was a total of 10 pathways that after multiple testing correction had significant p-values. Nine pathways were significant for the gene list from \gls{nsl} and had a heart and lung focus based on calcium ion movement in the form of genes for calcium transporters and voltage-gated calcium channels. There were five genes related to calcium that were in at least five of the pathway terms: _CACNA1D_, _ADCY9_, _SLC8A1_, _CACNA2D2_, _CACNA2D3._ There was a single pathway that was significant from the genes from the 1st percentile of \gls{td} which was "bacterial invasion of epithelial cells".


##### Consecutive regions of score depression in \gls{cim}

<!-- There were `r consec_lower_regions %>% filter(stat == 'Tajima.D', pop == 'CIM') %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce(ignore.strand = TRUE) %>% length()` regions of consecutive score depression for \gls{td}, `r consec_lower_regions %>% filter(stat == 'Fay.Wu.H', pop == 'CIM') %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce(ignore.strand = TRUE) %>% length()` for \gls{fwh}, `r consec_lower_regions %>% filter(stat == 'Fu.Li.D', pop == 'CIM') %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce(ignore.strand = TRUE) %>% length()` for \gls{fld}, `r consec_lower_regions %>% filter(stat == 'Fu.Li.F', pop == 'CIM') %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce(ignore.strand = TRUE) %>% length()` for \gls{flf}, and `r consec_lower_regions %>% filter(stat == 'Zeng.E', pop == 'CIM') %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce(ignore.strand = TRUE) %>% length()` for \gls{ze}. -->

Looking at regions of consecutively depressed score that also had either a significant \gls{ihs} or \gls{nsl} marker in them, there were a total of five regions that met this criteria. For \gls{td}, there were two regions. The first, a 290 kb region at chr2:24215002-24505001 covered _FAM228B_ and had a single significant marker for both \gls{ihs} and \gls{nsl}. The second, was 470 kb in length at chr11:68105002-68575001, and included _LRP5_ and had a single significant marker for \gls{ihs}. \gls{flf} also had a region that covered _LRP5_ but at chr11:68085002-68355001 and was 270 kb long. For both \gls{flf} and \gls{fld}, there was a 250 kb region at chr16:3545002-3795001 that covered _DNASE1_ with a single marker for \gls{ihs} and \gls{nsl}, and also _TRAP1_ with two markers each for \gls{ihs} and \gls{nsl}. \gls{fwh} had a single 250 kb region that covered _MYT1L_ at chr2:2015002-2265001 and had four significant markers for \gls{ihs} and three for \gls{nsl}. 

<!-- There were three regions of consecutively depressed score that intersected obesity associated genes. There was a 240 kb region at chr3:52375002-52615001 that intersected _NT5DC2_ and _PBRM1_ for \gls{fld}. The 250 kb region of chr16:3545002-3795001 intersected  _NLRC3_ for both \gls{fld} and \gls{flf}. And the 310 kb region of chr12:124195002-124505001 intersected _DNAH10_, _ZNF664_, and _CCDC92_ for \gls{flf}. There was a single 280 kb region that intersected the metabolic syndrome associated gene of EDC4 at chr16:67835002-68115001 for \gls{td}. None of the regions that were had consecutively depressed score that intersected genes associated with urate, \gls{t2d}, or kidney disease. -->


```{r, eval = FALSE}
# find ihs/nsl intersecting the regions of consecutively depressed score
consec_lower_regions %>% filter( pop == 'CIM') %>% inner_join(., bind_rows(sig_ihs_val, sig_nsl_val) %>% select(SYMBOL = genename, everything()), by = c('pop','SYMBOL')) %>% select(-contains('gene'))
```

<!-- - td -->
<!--     - FAM228B - ihs/nsl chr2:24215002-24505001 w=290000 -->
<!--     - LRP5 - ihs chr11:68105002-68575001 w=470000  -->
<!-- - fwh -->
<!--     - MYT1L 4x ihs, 3x nsl chr2:2015002-2265001 w=250000 -->
<!-- - flf -->
<!--     - LRP5 chr11:68085002-68355001 w=270000 -->

<!-- - chr16:3545002-3795001 w=250000 -->
<!--   - both fld and flf -->
<!--   - DNASE1 ihs/nsl -->
<!--   - TRAP1 2x ihs, 2x nsl -->

##### Selection from haplotypic statistics in \gls{cim}.


Within the most extreme 100 \gls{ihs} values, there were 37 genes. For \gls{nsl} there was 45 genes. Eighteen of these genes were in common, with _C11orf65_, _CNTN4_, _ADAM29_, and _LINC00276_ having at least five significant \gls{snp}s each. The gene with the most significant \gls{snp}s out of any population was _CNTN4_ in \gls{cim} with 27 markers having a significant \gls{ihs} score. The only other population with more than two significant \gls{snp}s was \gls{nzm} with 14. There were four genes represented in the top 100 markers that were associated with obesity or \gls{t2d}, these were: _ADCY9_, _ARL15_, _PSMD6_, and _ZFAND6_. In total there were  `r sig_ihs_val %>% filter(pop == "CIM") %>% select(genename) %>% anti_join(., sig_ihs_val %>% filter(pop != "CIM") %>% select(genename), by = 'genename') %>% distinct() %>% NROW()` genes that were only significant in \gls{cim} for \gls{ihs}, and `r sig_nsl_val %>% filter(pop == "CIM") %>% select(genename) %>% anti_join(., sig_nsl_val %>% filter(pop !="CIM") %>% select(genename), by = 'genename') %>% distinct() %>% NROW()` for \gls{nsl}. There were 57 genes that had at least 20 populations with significant markers for \gls{xpehh} in only \gls{cim} from the Polynesian populations.  _ARL15_ was associated with obesity and \gls{t2d} [@ref] and _RREB1_ was assoaciated with \gls{t2d} and urate [@ref]. 


<!-- ihs C11orf65 (6), ATM (5) CNTN4 (5),  -->
<!-- ADCY9 (obesity), ARL15 (obesity, t2d), PSMD6 (t2d), ZFAND6 (t2d) -->


```{r, eval=FALSE}
p <- "CIM"

(cim_nsl_100 <- sig_nsl_val %>% filter(pop == p) %>% arrange(desc(abs(statvalue))) %>% head(n=100) %>% group_by(genename) %>% tally() %>% arrange(desc(n)))

(cim_ihs_100 <- sig_ihs_val %>% filter(pop == p) %>% arrange(desc(abs(statvalue))) %>% head(n=100) %>% group_by(genename) %>% tally() %>% arrange(desc(n)))

inner_join(cim_nsl_100, cim_ihs_100, by = 'genename' ) %>% mutate(total = n.x + n.y) %>% arrange(desc(total))
rm(p)
```


```{r, eval = FALSE}
q <- xpehh %>%  select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% filter(pop1 %in% c("CIM","NZM","SAM","TON"), xpehh_value > 0) %>% select(pop1,pop2, genename) %>% distinct() %>% left_join(., panel %>% select(pop2 = pop, super_pop) %>% distinct(), by = 'pop2') %>% group_by(pop1, super_pop, genename) %>% tally() %>% spread(super_pop, n, 0)%>% arrange(genename) %>% mutate(rs = AFR + AMR + EAS + EUR + POL + SAS) %>% filter(rs >=20) %>% select(pop1, genename, rs) %>% spread(pop1, rs) %>% filter(!is.na(CIM),is.na(NZM), is.na(SAM), is.na(TON)) %>% pull(genename)
q[q %in% genelists$genename]
```




<!-- ##### FST -->

<!-- #### Notes -->

<!-- xpehh total number of genes that were significant  CIM versus: -->

<!-- - NZM `r xpehh %>% select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% filter(pop1 == 'CIM', pop2 == "NZM", !is.na(genename)) %>%  group_by(pop1,pop2,genename) %>% tally()  %>% NROW()` -->
<!-- - SAM `r xpehh %>% select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% filter(pop1 == 'CIM', pop2 == "SAM",  !is.na(genename)) %>%  group_by(pop1,pop2,genename) %>% tally() %>%  NROW()` -->
<!-- - TON `r xpehh %>% select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% filter(pop1 == 'CIM', pop2 == "TON",  !is.na(genename)) %>%  group_by(pop1,pop2,genename) %>% tally()  %>% NROW()` -->


<!-- xpehh number of genes that were significant in only CIM versus: -->

<!-- - NZM `r xpehh %>% select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% filter(pop1 == 'CIM', pop2 == "NZM", !is.na(genename)) %>%  group_by(pop1,pop2,genename) %>% tally() %>% anti_join(., xpehh %>% select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% filter(pop1 == 'CIM', pop2 != "NZM")%>%  group_by(pop1,pop2,genename) %>% tally(), by = c('genename')) %>% NROW()` -->
<!-- - SAM `r xpehh %>% select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% filter(pop1 == 'CIM', pop2 == "SAM",  !is.na(genename)) %>%  group_by(pop1,pop2,genename) %>% tally() %>% anti_join(., xpehh %>% select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% filter(pop1 == 'CIM', pop2 != "SAM")%>%  group_by(pop1,pop2,genename) %>% tally(), by = c('genename')) %>% NROW()` -->
<!-- - TON `r xpehh %>% select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% filter(pop1 == 'CIM', pop2 == "TON",  !is.na(genename)) %>%  group_by(pop1,pop2,genename) %>% tally() %>% anti_join(., xpehh %>% select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% filter(pop1 == 'CIM', pop2 != "TON")%>%  group_by(pop1,pop2,genename) %>% tally(), by = c('genename')) %>% NROW()` -->
<!-- - -->

<!-- > BLK was previously identified in Jha2015 and had sig TD and ihs in CIM. The east asians have both td and ihs, and sam/ton just have td. BLK is associated with Systemic lupus erythematosus (SLE) in Asian populations Delgado-Vega2012 -->


<!-- (ref:CIMupset) Upset plot showing the number of \gls{cim} genes that had a marker or window that met the significance threshold for the statistic. -->

```{r CIMupset, fig.cap = '(ref:CIMupset)', eval = FALSE}
UpSetR::upset(fromList(list(`Tajima's D` = lower_sig_stats %>% filter(pop == 'CIM', statname == 'Tajima.D', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(), 
                            `Fay and Wu's H` = lower_sig_stats %>% filter(pop == 'CIM', statname == 'Fay.Wu.H', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(), 
                            `Fu and Li's D` = lower_sig_stats %>% filter(pop == 'CIM', statname == 'Fu.Li.D', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(), 
                            `Fu and Li's F` = lower_sig_stats %>% filter(pop == 'CIM', statname == 'Fu.Li.F', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(),
                            `Zeng's E` = lower_sig_stats %>% filter(pop == 'CIM', statname == 'Zeng.E', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(),
                            iHS = sig_ihs_val %>% filter(pop == 'CIM', !is.na(genename) & genename != "") %>% .[['genename']] %>% unique(), 
                            nSL =sig_nsl_val %>% filter(pop == 'CIM', !is.na(genename) & genename != "") %>% .[['genename']] %>% unique())), 
              order.by = c("freq",'degree'), nset = 7, number.angles = 90, set_size.angles = 90, hjust = -0.25, vjust = 0.5, nintersects = NA)
```

```{r upsetcimsvg, include = FALSE, eval = FALSE}
svg(filename = 'images/03_selectionpipeline/cim_upset.svg')
  UpSetR::upset(fromList(list(`Tajima's D` = lower_sig_stats %>% filter(pop == 'CIM', statname == 'Tajima.D', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(), 
                            `Fay and Wu's H` = lower_sig_stats %>% filter(pop == 'CIM', statname == 'Fay.Wu.H', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(), 
                            `Fu and Li's D` = lower_sig_stats %>% filter(pop == 'CIM', statname == 'Fu.Li.D', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(), 
                            `Fu and Li's F` = lower_sig_stats %>% filter(pop == 'CIM', statname == 'Fu.Li.F', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(),
                            `Zeng's E` = lower_sig_stats %>% filter(pop == 'CIM', statname == 'Zeng.E', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(),
                            iHS = sig_ihs_val %>% filter(pop == 'CIM', !is.na(genename) & genename != "") %>% .[['genename']] %>% unique(), 
                            nSL =sig_nsl_val %>% filter(pop == 'CIM', !is.na(genename) & genename != "") %>% .[['genename']] %>% unique())), 
              order.by = c("freq",'degree'), nset = 7, number.angles = 90, set_size.angles = 90, hjust = -0.25, vjust = 0.5, nintersects = NA)
  dev.off()
```

```{r cim_urate_results, results = 'asis', eval = FALSE}
# look at all significant stats from cim for urate genes
bind_rows(cim_urate_sig)%>% spread(statname,n, fill = "" ) %>% arrange() %>% kable(., caption = '(\\#tab:cimUrate) CIM genes with evidence of selection in urate')
consec_lower_regions %>% filter(pop == 'CIM', SYMBOL %in% urate_goutGR$SYMBOL)

# obesity genes
bind_rows(cim_obesity_sig)%>% spread(statname,n, fill = "" ) %>% arrange() %>% kable(., caption = '(\\#tab:cimObesity) CIM genes with evidence of selection in obesity')
consec_lower_regions %>% filter(pop =='CIM', SYMBOL %in% obesity_GR$SYMBOL) %>% mutate(chrom = substring(seqnames, 4, nchar(seqnames)))%>% arrange(stat, chrom, start)

# T2D
bind_rows(cim_t2d_sig)%>% spread(statname,n, fill = "" ) %>% arrange() %>% kable(., caption = '(\\#tab:cimT2d) CIM genes with evidence of selection in T2D')
consec_lower_regions %>% filter(pop =='CIM', SYMBOL %in% t2d_GR$SYMBOL) %>% mutate(chrom = substring(seqnames, 4, nchar(seqnames)))%>% arrange(stat, chrom, start)

# KD
bind_rows(cim_kd_sig)%>% spread(statname,n, fill = "" ) %>% arrange() %>% kable(., caption = '(\\#tab:cimKd) CIM genes with evidence of selection in KD')
consec_lower_regions %>% filter(pop =='CIM', SYMBOL %in% kd_GR$SYMBOL) %>% mutate(chrom = substring(seqnames, 4, nchar(seqnames)))%>% arrange(stat, chrom, start)

# Metsyn
bind_rows(cim_metsyn_sig)%>% spread(statname,n, fill = "" ) %>% arrange() %>% kable(., caption = '(\\#tab:cimMetsyn) CIM genes with evidence of selection in MetSyn')
consec_lower_regions %>% filter(pop =='CIM', SYMBOL %in% metsyn_GR$SYMBOL) %>% mutate(chrom = substring(seqnames, 4, nchar(seqnames)))%>% arrange(stat, chrom, start)
```

<!-- obesity -->

<!-- ihs/nsl only: ADAMTS9, ADCY9, CCNJL, GDF5, MSRA, NAV1 -->


<!-- T2d: -->

<!-- ADAMTS9, LPP, PSMD6, SSR1, ZFAND6 had both ihs and nsl -->

<!-- ADAM30, NOTCH2 had only fld and flf -->

<!-- THADA had fwh, fld, flf, ze -->

<!-- ARL15, PTPRD had ihs, nsl and td -->

<!-- consecutive regions: -->
<!-- THADA 140kb fwh -->
<!-- ADAM30/NOTCH2 180kb for fld and flf -->


<!-- The \gls{cim} population had similar numbers of windows that were significant in both tails of the distribution compared to the other Polynesian populations but slightly more genes intersected these windows, and were also only significant in \gls{cim} for the lower tail. The upper tail had slightly fewer genes compared to the other Polynesian populations. -->

<!-- In the lower tail there were  `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes for \gls{td},  `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` for \gls{fwh},  `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` for \gls{fld},  `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` for \gls{flf}, and  `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` for \gls{ze}, that intersected windows that met the significance threshold. There were only two genes - _FANCM_ and _RGS3_ - that had windows that met the significance threshold for each of \gls{td}, \gls{fwh}, \gls{fld}, \gls{flf}, and \gls{ze}. There were `r lower_sig_stats %>% filter(pop == 'CIM') %>% select(-contains("chrom"), -pop) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% arrange(desc(n)) %>% filter(n == 4) %>% NROW()` genes that met the threshold in four of the selection statistics, `r lower_sig_stats %>% filter(pop == 'CIM') %>% select(-contains("chrom"), -pop) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% arrange(desc(n)) %>% filter(n == 3) %>% NROW()`  in three selection statistics, `r lower_sig_stats %>% filter(pop == 'CIM') %>% select(-contains("chrom"), -pop) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% arrange(desc(n)) %>% filter(n == 2) %>% NROW()` in two selection statistics, and `r lower_sig_stats %>% filter(pop == 'CIM') %>% select(-contains("chrom"), -pop) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% arrange(desc(n)) %>% filter(n == 1) %>% NROW()` with only one selection statistic meeting the threshold for the lower tail. -->

<!-- \Gls{cim} also had the most independent consecutive regions for \gls{td} and \gls{fwh} out of the Polynesian populations. -->

<!-- the gene with the most number of significant markers in any population for \gls{ihs} was CNTN4, with 27 significant markers in \gls{cim}. Of the 13 other populations that had a significant marker with \gls{ihs}, all but \gls{nzm} which had 14, had no more than 2 markers that were significant. Out of the 27 markers for \gls{cim} 13 were in common exclusively with \gls{nzm}, only 2 markers,  rs2728076 - shared with \gls{cdx} and \gls{chs}, and rs955340 - shared with \gls{clm}, -->

<!-- The largest \gls{ihs} values came from C11orf65, LINC00276, CPQ, SNAP25-AS1 -->

<!-- The largest \gls{nsl} values came from ADAM29, SLC35F2, PGR, and MIR146A -->

<!-- The smallest values for \gls{td} came from an eight window consecutive region at `r lower_sig_stats %>% left_join(., lower_sig_stats_binned) %>% filter(pop == 'CIM',statname == 'Tajima.D') %>% arrange(value) %>% head(n=20) %>% filter(chrom == 'chr22') %>% GRanges() %>% GenomicRanges::reduce() %>% data.frame() %>% mutate(window = paste0(seqnames,':',start,'-',end)) %>% .[['window']]` which covered the following genes TTC38, GTSE1-AS1, GTSE1, CDPF1, PKDREJ, and PPARA.     -->


<!-- The largest region of Fay and Wu's H of windows in the 0.1 percentile was at `r lower_sig_stats %>% left_join(., lower_sig_stats_binned) %>% filter(pop == 'CIM',statname == 'Fay.Wu.H', quant == 0.1) %>% arrange(value) %>% GRanges() %>% GenomicRanges::reduce() %>% data.frame() %>% mutate(window = paste0(seqnames,':',start,'-',end)) %>% arrange(desc(width)) %>% .[['window']] %>% .[1]` and was 80kb wide. In this region was ADH4, ADH5, PCNAP1, and LOC100507053. -->

<!-- The smallest five \gls{fld} values came from windows that covered TCTN1, HVCN1, DOCK5, DNAH10, CALML4, CLN6, ATP1A1, ATP1A1-AS1, ZNF704, and EFCAB5. -->

<!-- \gls{flf} with the smallest values came from windows that covered TCTN1, HVCN1, DOCK5, ATP1A1-AS1, ATP1A1, and ZNF704 -->

<!-- \gls{ze} had ZNF704, LYST, SH3RF1, UQCRC1 from the smallest valued windows. From the 0.1 percentile there was a significant result for the KEGG pathway analysis -->


<!-- There were only significant results for 3 loci that were associated with urate and gout. _ATXN2_ had one window that was significant with \gls{ze}. _RREB1_ had 3 windows that were significant for \gls{td}. _BCAS3_ had significant windows for \gls{fld} (5), \gls{flf} (4), \gls{td} (2), and \gls{fwh} (8). The \gls{fwh} windows also met the criteria to be classed as a consecutively depressed score region spanning 110 Kb. There were no intra-population haplotype based statistics that had a significant value. For \gls{xpehh} there were several loci that were significant, _NRXN2_ with the \gls{eur} populations. _TRIM46_ with the Han Chinese (\gls{chb} and \gls{chs}) and the Japanese (\gls{jpt}). _RREB1_ was significant between \gls{cim} and nearly all populations part of the \gls{afr}, \gls{amr}, \gls{eas}, and \gls{eur} super populations, and only a single population (\gls{pjl}) from the \gls{sas} super population. -->

```{r cimSigRegions, eval = FALSE}
# copied into Panther, then exported the biological process
# also copied into Enrichr and exported the KEGG2016 pathways table

#td
lower_sig_stats %>% filter(pop == 'CIM', statname == 'Tajima.D', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique() %>% cat()
#fwh
lower_sig_stats %>% filter(pop == 'CIM', statname == 'Fay.Wu.H', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique() %>% cat()
#fld
lower_sig_stats %>% filter(pop == 'CIM', statname == 'Fu.Li.D', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique() %>% cat()
#flf
lower_sig_stats %>% filter(pop == 'CIM', statname == 'Fu.Li.F', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique() %>% cat()
#ze
lower_sig_stats %>% filter(pop == 'CIM', statname == 'Zeng.E', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique() %>% cat()
#ihs
sig_ihs_val %>% filter(pop == 'CIM', !is.na(genename)) %>% .[['genename']] %>% unique() %>% cat()
#nsl
sig_nsl_val %>% filter(pop == 'CIM', !is.na(genename)) %>% .[['genename']] %>% unique() %>% cat()

# all with at least 1 sfs and 1 haplotypic
bind_rows(bind_rows(sig_ihs_val, sig_nsl_val) %>% filter(pop == 'CIM')%>% select(pop, "SYMBOL" = genename ) %>% distinct(), lower_sig_stats %>% filter(pop == 'CIM')%>% select(pop, SYMBOL) %>% distinct()) %>% filter(!is.na(SYMBOL))  %>% group_by(SYMBOL) %>% tally() %>% filter(n == 2) %>% .[['SYMBOL']] %>% cat()
```



<!-- - TD main processes: "binding", "catalytic activity" -->
<!-- - FWH main processes: "Cellular process", "metabolic process" -->
<!-- - FLD main processes: "Cellular process", "metabolic process" -->
<!-- - FLF main processes: "Cellular process", "metabolic process" -->
<!-- - ZE main processes: "Cellular process", "metabolic process" -->
<!-- - ihs main processes: "Cellular process", "metabolic process" -->
<!-- - nsl main processes: "Cellular process", "metabolic process" -->
<!-- - all main processes: "Cellular process", "metabolic process" -->
<!-- - all overrepresentation test: nothing -->



<!-- > Genes in the 0.1 percentile for the sfs based stats -->

<!-- The 0.1 percentile had a total of `r cim_0.1_stats_lower %>% filter(statname == 'Tajima.D') %>% .[['n']]` regions and `r cim_0.1_genes_lower %>% filter(statname == 'Tajima.D') %>% .[['n']]` genes for \gls{td}.  -->


```{r, eval = FALSE}
# look at the lower 0.1% of windows
lower_sig_stats_binned %>% filter(pop == 'CIM',quant == 0.1) %>% left_join(.,lower_sig_stats, by = c('statname','chrom','chrom_start','chrom_end','pop')) %>% select(statname, SYMBOL) %>% group_by(statname, SYMBOL) %>% distinct()%>% tally() %>% arrange(statname, SYMBOL)  %>% spread(statname, n) %>% mutate(rs = select(., 2:NCOL(.)) %>% rowSums(., na.rm=TRUE)) %>% filter(rs > 3)

lower_sig_stats_binned %>% filter(pop == 'CIM',quant == 0.1) %>% left_join(.,lower_sig_stats, by = c('statname','chrom','chrom_start','chrom_end','pop')) %>% filter(!is.na(SYMBOL))%>% group_by(statname, SYMBOL) %>% tally()%>% arrange(statname, SYMBOL)  %>% spread(statname, n) %>% filter(!is.na(Zeng.E) )
```

<!-- In the 0.1st percentile there were 84 genes for \gls{td}, 56 for \gls{fwh}, 99 for \gls{ze}, and 59 in common between \gls{fld} and \gls{flf}. There were three genes DNAH1, RGS3, and ZNF704 that were in the 0.1st percentile for each of \gls{fld}, \gls{flf}, \gls{td}, and \gls{ze}  -->



```{r, fig.cap='(ref:cimFANCM)', eval= FALSE, message = FALSE, fig.height = 4, fig.width = 4}
#plot_gene("CIM","FANCM")
#ggsave(last_plot(), file = 'images/03_selectionpipeline/cimFANCM.png')
knitr::include_graphics('images/03_selectionpipeline/cimFANCM.png')
```


```{r, fig.cap='(ref:cimRGS3)', eval= FALSE, message = FALSE, fig.height = 4, fig.width = 4}
#plot_gene("CIM","RGS3")
#ggsave(last_plot(), file = 'images/03_selectionpipeline/cimRGS3.png')
knitr::include_graphics('images/03_selectionpipeline/cimRGS3.png')
```



```{r, eval = FALSE}
tmp_td <- lower_sig_stats %>% filter(pop == "CIM", statname == "Tajima.D") %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce(., ignore.strand = TRUE)
tmp_td$stat <- "td"

tmp_fwh <- lower_sig_stats %>% filter(pop == "CIM", statname == "Fay.Wu.H") %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce(., ignore.strand = TRUE)
tmp_fwh$stat <- "fwh"

tmp <- IRanges::slice(GenomicRanges::coverage(c(tmp_td, tmp_fwh))[[4]], 2)
tmp@ranges
```

<!-- There were fewer genes that intersected the windows that met the threshold for significance in the upper tail for each of \gls{td} (`r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()`), \gls{fwh} (`r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()`), \gls{fld} (`r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()`), \gls{flf} (`r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()`), and \gls{ze} (`r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()`) compared with the lower tail. In the upper tail there was only a single gene - _CSMD1_ - that had windows that met the significance threshold in \gls{td}, \gls{fwh}, \gls{fld}, \gls{flf}, and \gls{ze}. There were  `r upper_sig_stats %>% filter(pop == 'CIM') %>% select(-contains("chrom"), -pop) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% arrange(desc(n)) %>% filter(n == 4) %>% NROW()` genes that met the threshold with four of the selection statistics,  `r upper_sig_stats %>% filter(pop == 'CIM') %>% select(-contains("chrom"), -pop) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% arrange(desc(n)) %>% filter(n == 3) %>% NROW()` genes that met the threshold in three selection statistics,  `r upper_sig_stats %>% filter(pop == 'CIM') %>% select(-contains("chrom"), -pop) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% arrange(desc(n)) %>% filter(n == 2) %>% NROW()` genes with two selection statistics, and  `r upper_sig_stats %>% filter(pop == 'CIM') %>% select(-contains("chrom"), -pop) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% arrange(desc(n)) %>% filter(n == 1) %>% NROW()` genes with only one selection statistic that met the upper tail significance threshold. -->

<!-- \gls{cim} had the most independent regions for \gls{flf} and \gls{ze}, and the most genes that were population specific, out of the the Polynesian populations. -->

<!-- - fwh most pop specific -->
<!-- - flf most regions -->
<!-- - ze most regions -->

<!-- Looking specifically at the genes that were associated with gout and other metabolic diseases, of the \gls{pol} populations, the \gls{cim} population had the fewest genes that were associated with gout and urate that showed possible evidence of positive selection. This was true also for obesity, \gls{t2d}, kidney disease, and metabolic syndrome associated genes (Table \@ref(tab:polDiseaseGenesTab). -->


```{r, eval = FALSE}
bind_rows(sig_ihs_val %>% mutate(statname = 'ihs'), sig_nsl_val %>% mutate(statname = 'nsl')) %>% filter(pop == 'CIM', genename %in% urate_goutGR$SYMBOL) %>% group_by(genename, statname) %>% tally() %>% arrange(desc(n))

bind_rows(sig_ihs_val %>% mutate(statname = 'ihs'), sig_nsl_val %>% mutate(statname = 'nsl')) %>% filter(pop == 'CIM', genename %in% t2d_GR$SYMBOL) %>% group_by(genename, statname) %>% tally() %>% arrange(desc(n))

bind_rows(sig_ihs_val %>% mutate(statname = 'ihs'), sig_nsl_val %>% mutate(statname = 'nsl')) %>% filter(pop == 'CIM', genename %in% obesity_GR$SYMBOL) %>% group_by(genename, statname) %>% tally() %>% arrange(desc(n))

bind_rows(sig_ihs_val %>% mutate(statname = 'ihs'), sig_nsl_val %>% mutate(statname = 'nsl')) %>% filter(pop == 'CIM', genename %in% metsyn_GR$SYMBOL) %>% group_by(genename, statname) %>% tally() %>% arrange(desc(n))
```





```{r, echo = FALSE, eval = FALSE}
_Tajima's D_

The \gls{cim} had a total of `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the significance threshold in the lower tail. This represented `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` non-continuous regions representing `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp of th genome. Within these windows were `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which only `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Tajima.D', !pop %in% c('CIM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes were distinct for \gls{cim} out of the \gls{td} results. In the upper tail there were  `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the upper significance threshold, these windows formed `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM')) %>% select(contains('chrom'))  %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` non-continuous regions covering `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. These regions contained `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Tajima.D', !pop %in% c('CIM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only from \gls{cim} for \gls{td}.

_Fay and Wu's H_

There were `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the significance threshold in the lower tail. These windows made up `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions with a total of  `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. Within these regions were `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fay.Wu.H', !pop %in% c('CIM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant in the \gls{cim} population for \gls{fwh}

In the upper tail there were  `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that me the significance threshold, consisting of `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions that totaled `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. There were  `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes within these regions and `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fay.Wu.H', !pop %in% c('CIM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes were only significant in \gls{cim} for \gls{fwh}.

_Fu and Li's D_

For the \gls{cim} population there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower tail threshold for significance and made `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions covering `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. In these regions there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which  `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.D', !pop %in% c('CIM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant in \gls{cim} for \gls{fld}. In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the significance threshold. These windows made `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions with a total of `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp containing `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, with  `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.D', !pop %in% c('CIM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only meeting significance in \gls{cim} for \gls{fld}.

_Fu and Li's F_

For the \gls{cim} population there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower tail threshold for significance and made `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions covering `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. In these regions there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which  `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.F', !pop %in% c('CIM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant in \gls{cim} for \gls{flf}. In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the significance threshold. These windows made `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions with a total of `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp containing `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, with  `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.F', !pop %in% c('CIM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only meeting significance in \gls{cim} for \gls{flf}.

_Zeng's E_

For the \gls{cim} population there were `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower tail threshold for significance and made `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions covering `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. In these regions there were `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which  `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Zeng.E', !pop %in% c('CIM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant in \gls{cim} for \gls{ze}. In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the significance threshold. These windows made `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions with a total of `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp containing `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, with  `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Zeng.E', !pop %in% c('CIM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only meeting significance in \gls{cim} for \gls{ze}.
```


<!-- - The upper tail had `r consec_upper_regions %>% filter(stat == 'Tajima.D', pop == 'CIM') %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce(ignore.strand = TRUE) %>% length()` regions of extended significant windows for \gls{td}, `r consec_upper_regions %>% filter(stat == 'Fay.Wu.H', pop == 'CIM')%>% GenomicRanges::GRanges() %>% GenomicRanges::reduce(ignore.strand = TRUE) %>% length()` for \gls{fwh}, `r consec_upper_regions %>% filter(stat == 'Fu.Li.D', pop == 'CIM') %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce(ignore.strand = TRUE) %>% length()` for \gls{fld}, `r consec_upper_regions %>% filter(stat == 'Fu.Li.F', pop == 'CIM')%>% GenomicRanges::GRanges() %>% GenomicRanges::reduce(ignore.strand = TRUE) %>% length()` for \gls{flf}, and `r consec_upper_regions %>% filter(stat == 'Zeng.E', pop == 'CIM') %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce(ignore.strand = TRUE) %>% length()` for \gls{ze}. -->

<!-- From the haplotypic tests for selection (\gls{ihs} and \gls{nsl}) there were `r sig_ihs_val %>% filter(pop == "CIM") %>% select(genename) %>% anti_join(., sig_ihs_val %>% filter(pop !="CIM") %>% select(genename), by = 'genename') %>% distinct() %>% NROW()` genes that were only significant in \gls{cim} for \gls{ihs}, and `r sig_nsl_val %>% filter(pop == "CIM") %>% select(genename) %>% anti_join(., sig_nsl_val %>% filter(pop !="CIM") %>% select(genename), by = 'genename') %>% distinct() %>% NROW()` for \gls{nsl}. -->



<!-- were there any genes that were significant from the gene lists and related to the thrifty gene hypothesis? -->


#### New Zealand Maori

The \gls{nzm} population had the largest number of genes intersecting windows from the 1st percentile that were only reported in a single population for \gls{fwh}, \gls{fld}, \gls{flf}, and \gls{ze} out of the Polynesian populations.


<!-- - TD main processes "cellular process", "metabolic process" -->
<!-- - FWH main processes "cellular process", "metabolic process" -->
<!-- - FLD main processes "cellular process", "metabolic process" -->
<!-- - FLF main processes "cellular process", "metabolic process" -->
<!-- - ZE main processes "cellular process", "metabolic process" -->
<!-- - ihs main processes "cellular process", "metabolic process" -->
<!-- - nsl main processes "cellular process", "metabolic process" -->
<!-- - all main processes "cellular process", "metabolic process" -->
<!-- - all overrepresentation test: cellular component movement FE=4.47, P = 4.86e-2 -->
<!--     - anatomical structure morphogenesis FE=3.94 P = 2.88e-2 -->

##### Pathway enrichment analysis of selected loci in \gls{nzm}

<!-- The gene set over representation analysis using PantherDB had 'cellular component movement' with a fold-enrichment of 4.47 (P = 0.0486) and 'anatomical structure morphogenesis' with fold enrichment of 3.94 (P = 0.0288), as the only significant annotations, when all genes that met a threshold were used.  -->

The pathway gene-set enrichment analysis from Enrichr on the KEGG 2016 pathways had 31 pathways that were significant after multiple testing correction, this was the largest number of pathways out of the Polynesian populations. On average only 9.3% of a pathway was represented by the genes. Thirty were from the significant genes for \gls{nsl}, with many in common with \gls{cim} with calcium transporters and voltage gated calcium channels featuring in many pathways. Compared to the \gls{cim} pathway results, there were more hormone-based pathways that were significant, such as renin secretion, oxytocin signalling pathway, adrenergic signalling in cardiomyocytes, thyroid hormone synthesis, aldosterone synthesis and secretion, insulin secretion, and GnRH signalling pathway. Furthermore, diabetes-related pathways were significant such as type 2 diabetes mellitus, insulin secretion, and pancreatic secretion pathways. The genes that were significant made up about 10% of the genes in the pathways. The genes that were shared in the most pathways included two adenlyate cylases (_ADCY8_ and _ADCY9_), two phospholipase C genes (_PLCB1_ and _PLCB4_), and calcium voltage-gated channel subunits (_CACNA1C_ and _CACNA1D_). Similar to \gls{cim}, there was a single pathway that was significant from the \gls{td} list and that too was for bacterial invasion of epithelial cells.

##### Selection from haplotypic statistics in \gls{nzm}.

From the haplotypic tests for selection (\gls{ihs} and \gls{nsl}) there were `r sig_ihs_val %>% filter(pop == "NZM") %>% select(genename) %>% anti_join(., sig_ihs_val %>% filter(pop != "NZM") %>% select(genename), by = 'genename') %>% distinct() %>% NROW()` genes that were only significant in \gls{nzm} for \gls{ihs}, and `r sig_nsl_val %>% filter(pop == "NZM") %>% select(genename) %>% anti_join(., sig_nsl_val %>% filter(pop != "NZM") %>% select(genename), by = 'genename') %>% distinct() %>% NROW()` for \gls{nsl}. Looking at the genes represented by the most extreme 100 \gls{ihs} and \gls{nsl} markers there were 30 genes for \gls{ihs}, and 40 genes for \gls{nsl}. There were 12 genes in common between the two sets. _CTNNA3_ had five markers in the top 100 for \gls{ihs}, and was also the gene for \gls{nzm} that had the most overall significant markers for both \gls{ihs} and \gls{nsl}. There were 40 genes that had at least 20 populations with significant markers for \gls{xpehh} in only NZM from the Polynesian populations, only one (_PAX5_) of which was associated with obesity [@ref].

 
```{r, eval = FALSE}
q <- xpehh %>%  select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% filter(pop1 %in% c("CIM","NZM","SAM","TON"), xpehh_value > 0) %>% select(pop1,pop2, genename) %>% distinct() %>% left_join(., panel %>% select(pop2 = pop, super_pop) %>% distinct(), by = 'pop2') %>% group_by(pop1, super_pop, genename) %>% tally() %>% spread(super_pop, n, 0)%>% arrange(genename) %>% mutate(rs = AFR + AMR + EAS + EUR + POL + SAS) %>% filter(rs >=20) %>% select(pop1, genename, rs) %>% spread(pop1, rs) %>% filter(is.na(CIM),!is.na(NZM), is.na(SAM), is.na(TON)) %>% pull(genename)
q[q %in% genelists$genename]
```

##### Consecutive regions of score depression in \gls{nzm}

The regions of consecutive depressed score from the 1st percentile of windows  that also contained markers with a significant \gls{ihs} or \gls{nsl} value covered three regions of the genome for \gls{nzm}. Two of the regions were for \gls{td}, the first was 270kb at chr2:24215002-24485001, and overlapped the region in \gls{cim}. It had a single significant marker for \gls{ihs} in each of _MFSD2B_, _FKBP1B_, and _FAM228B_. The second \gls{td} region was 260 kb at chr4:106555002-106815001 and had a single significant marker for \gls{nsl} in _INTS12_, and a single marker for both \gls{ihs} and \gls{nsl} in _SH3RF1_. Another region, a 340 kb region for \gls{fld} at chr2:179325002-179665001 contained a 200 kb region for \gls{flf} at chr2:179395002-179595001. This region also had five significant markers in _TTN-AS1_ for both \gls{ihs} and \gls{nsl}

No consecutive regions intersected with genes associated with urate, kidney disease, or \gls{t2d}. There were 2 regions that intersected obesity-associated loci; for \gls{td} a 320 kb region at chr16:67145002-67465001 contained _KCTD19_, and for \gls{flf}, 220 kb at chr16:3555002-3775001 contained _NLRC3_. _EDC4_ which is associated with metabolic syndrome had slightly differing regions that covered it. \gls{td} was 330 kb spanning chr16:67775002-68105001, \gls{fld} was 260 kb spanning chr16:67785002-68045001, and \gls{flf} was 320 kb spanning chr16:67775002-68095001.


Four genes, _CTNNA3_, _RAD51AP2_, _SUPT3H_, and _VSNL1_, had at least one significant marker for \gls{ihs} or \gls{nsl}, and also had a window in the 1st percentile from four or more statistics from \gls{td}, \gls{fwh}, \gls{fld}, \gls{flf}, and \gls{ze}.




<!-- ##### FST -->

<!-- ##### Notes -->
```{r, eval = FALSE}
p <- "NZM"

(nzm_nsl_100 <- sig_nsl_val %>% filter(pop == p) %>% arrange(desc(abs(statvalue))) %>% head(n=100) %>% group_by(genename) %>% tally() %>% arrange(desc(n)))

(nzm_ihs_100 <- sig_ihs_val %>% filter(pop == p) %>% arrange(desc(abs(statvalue))) %>% head(n=100) %>% group_by(genename) %>% tally() %>% arrange(desc(n)))

inner_join(nzm_nsl_100, nzm_ihs_100, by = 'genename' ) %>% mutate(total = n.x + n.y) %>% arrange(desc(total))
rm(p)
```


<!-- From all of the intra population -->

<!-- - There were `r bind_rows(nzm_urate_sig) %>% select(SYMBOL) %>% distinct() %>% NROW()` of `r length(unique(urate_goutGR$SYMBOL))` urate genes -->
<!-- - There were `r bind_rows(nzm_obesity_sig) %>% select(SYMBOL) %>% distinct() %>% NROW()` of `r length(unique(obesity_GR$SYMBOL))` obesity genes -->
<!-- - There were `r bind_rows(nzm_t2d_sig) %>% select(SYMBOL) %>% distinct() %>% NROW()` of `r length(unique(t2d_GR$SYMBOL))` \gls{t2d} genes -->
<!-- - There were `r bind_rows(nzm_kd_sig) %>% select(SYMBOL) %>% distinct() %>% NROW()` of `r length(unique(kd_GR$SYMBOL))` kidney disease genes -->
<!-- - There were `r bind_rows(nzm_metsyn_sig) %>% select(SYMBOL) %>% distinct() %>% NROW()` of `r length(unique(metsyn_GR$SYMBOL))` metsyn genes -->


```{r, eval = FALSE}
# find ihs/nsl intersecting the regions of consecutively depressed score
consec_lower_regions %>% filter( pop == 'NZM') %>% inner_join(., bind_rows(sig_ihs_val, sig_nsl_val) %>% select(SYMBOL = genename, everything()), by = c('pop','SYMBOL')) %>% select(-contains('gene'))
```



```{r nzmSigRegions, eval = FALSE}
# copied into Panther adn then exported the biological process

#td
lower_sig_stats %>% filter(pop == 'NZM', statname == 'Tajima.D', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique() %>% cat()
#fwh
lower_sig_stats %>% filter(pop == 'NZM', statname == 'Fay.Wu.H', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique() %>% cat()
#fld
lower_sig_stats %>% filter(pop == 'NZM', statname == 'Fu.Li.D', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique() %>% cat()
#flf
lower_sig_stats %>% filter(pop == 'NZM', statname == 'Fu.Li.F', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique() %>% cat()
#ze
lower_sig_stats %>% filter(pop == 'NZM', statname == 'Zeng.E', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique() %>% cat()
#ihs
sig_ihs_val %>% filter(pop == 'NZM', !is.na(genename)) %>% .[['genename']] %>% unique() %>% cat()
#nsl
sig_nsl_val %>% filter(pop == 'NZM', !is.na(genename)) %>% .[['genename']] %>% unique() %>% cat()

# all with at least 1 sfs and 1 haplotypic
bind_rows(bind_rows(sig_ihs_val, sig_nsl_val) %>% filter(pop == 'NZM')%>% select(pop, "SYMBOL" = genename ) %>% distinct(), lower_sig_stats %>% filter(pop == 'NZM')%>% select(pop, SYMBOL) %>% distinct()) %>% filter(!is.na(SYMBOL))  %>% group_by(SYMBOL) %>% tally() %>% filter(n == 2) %>% .[['SYMBOL']] %>% cat()
```

<!-- - TD main processes "cellular process", "metabolic process" -->
<!-- - FWH main processes "cellular process", "metabolic process" -->
<!-- - FLD main processes "cellular process", "metabolic process" -->
<!-- - FLF main processes "cellular process", "metabolic process" -->
<!-- - ZE main processes "cellular process", "metabolic process" -->
<!-- - ihs main processes "cellular process", "metabolic process" -->
<!-- - nsl main processes "cellular process", "metabolic process" -->
<!-- - all main processes "cellular process", "metabolic process" -->
<!-- - all overrepresentation test: cellular component movement FE=4.47, P = 4.86e-2 -->
<!--     - anatomical structure morphogenesis FE=3.94 P = 2.88e-2 -->

<!-- - td -->
<!--     - chr2:24215002-24485001 w=270000 -->
<!--         - MFSD2B ihs -->
<!--         - FKBP1B ihs -->
<!--         - FAM228B ihs -->
<!--     - chr4:106555002-106815001 w=260000 -->
<!--         - INTS12 nsl -->
<!--         - SH3RF1 ihs/nsl -->
<!-- -fld -->
<!--     - chr2:179325002-179665001 w=340000  -->
<!--         - TTN-AS1 5x ihs -->
<!--         - TTN 5x ihs -->
<!-- - flf -->
<!--     - chr2:179395002-179595001 w=200000 -->
<!--         - TTN-AS1 5x ihs -->
<!--         - TTN 5x ihs -->

<!-- no urate, t2d,  -->
<!-- obesity:  -->
<!-- td chr16:67145002-67465001 w=320000 KCTD19 -->
<!-- flf chr16:3555002-3775001 w=220000 NLRC3  -->

<!-- metsyn -->
<!-- td chr16:67775002-68105001 w=330000 EDC4 -->
<!-- fld chr16:67785002-68045001 w=260000 EDC4 -->
<!-- flf chr16:67775002-68095001 w=320000 EDC4 -->


<!-- (ref:upsetNZM) Upset plot for \gls{nzm} showing the numbers of genes that were unique or in common between the different intra-population statistics that met the significance threshold. -->


```{r upsetNzm, fig.cap = '(ref:upsetNZM', eval = FALSE}
UpSetR::upset(fromList(list(`Tajima's D` = lower_sig_stats %>% filter(pop == 'NZM', statname == 'Tajima.D', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(), 
                            `Fay and Wu's H` = lower_sig_stats %>% filter(pop == 'NZM', statname == 'Fay.Wu.H', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(), 
                            `Fu and Li's D` = lower_sig_stats %>% filter(pop == 'NZM', statname == 'Fu.Li.D', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(), 
                            `Fu and Li's F` = lower_sig_stats %>% filter(pop == 'NZM', statname == 'Fu.Li.F', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(),
                            `Zeng's E` = lower_sig_stats %>% filter(pop == 'NZM', statname == 'Zeng.E', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(),
                            iHS = sig_ihs_val %>% filter(pop == 'NZM', !is.na(genename) & genename != "") %>% .[['genename']] %>% unique(), 
                            nSL =sig_nsl_val %>% filter(pop == 'NZM', !is.na(genename) & genename != "") %>% .[['genename']] %>% unique())), 
              order.by = c("freq", "degree"), nsets = 7, number.angles = 90, set_size.angles = 90, hjust = -0.5, vjust = 0.5, nintersects = NA )
```

<!-- - 5 stats: `r lower_sig_stats %>% filter(pop %in% c('NZM')) %>% select(SYMBOL, statname) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==5, !is.na(SYMBOL)) %>% .[['SYMBOL']]` -->
<!-- - 4 stats: `r lower_sig_stats %>% filter(pop %in% c('NZM')) %>% select(SYMBOL, statname) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==4, !is.na(SYMBOL)) %>% NROW()` -->
<!-- - 3 stats: `r lower_sig_stats %>% filter(pop %in% c('NZM')) %>% select(SYMBOL, statname) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==3, !is.na(SYMBOL)) %>% NROW()` -->
<!-- - 2 stats: `r lower_sig_stats %>% filter(pop %in% c('NZM')) %>% select(SYMBOL, statname) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==2, !is.na(SYMBOL)) %>% NROW()` -->
<!-- - 1 stats: `r lower_sig_stats %>% filter(pop %in% c('NZM')) %>% select(SYMBOL, statname) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==1, !is.na(SYMBOL)) %>% NROW()` -->





```{r nzmDNAH6, fig.cap='(ref:nzmDNAH6)', eval= FALSE, message = FALSE, fig.height = 4, fig.width = 4}
#plot_gene("NZM","DNAH6")
#ggsave(last_plot(), file = 'images/03_selectionpipeline/nzmDNAH6.png')
knitr::include_graphics('images/03_selectionpipeline/nzmDNAH6.png')
```

```{r nzmDNAH6consec, fig.cap='(ref:nzmDNAH6consec)', eval= FALSE, message = FALSE, fig.height = 4, fig.width = 4}
# g = "DNAH6"
# #retrieve information of the gene of interest
# 
# wh <- genesymbol[g]
# wh <- range(wh, ignore.strand = TRUE)
# 
# stat_plot <- bind_rows(sig_ihs_val, sig_nsl_val) %>% filter(pop == 'NZM', genename == g)%>% mutate(pop = 'NZM', statname = case_when(statid == 20 ~ "nSL", statid == 26 ~ "iHS"), chrom = paste0('chr',chrom)) %>%  select(SYMBOL = genename, chrom, chrom_start, chrom_end, statname) %>%  bind_rows(., consec_lower_regions %>% filter(pop == 'NZM', SYMBOL == g) %>% select('chrom'= seqnames, 'chrom_start' = start, 'chrom_end' = end, 'statname' = stat, SYMBOL) ) %>% GenomicRanges::GRanges() %>% ggbio::autoplot(., aes(fill = statname, colour = statname)) + theme(legend.position = 'top')
# # hasAxis(stat_plot) <- TRUE
# gene_plot <- ggbio::autoplot(Homo.sapiens::Homo.sapiens, which = wh, gap.geom = "chevron")
# hasAxis(gene_plot) <- TRUE
# ggbio::tracks(gene_plot, stat_plot) + ggtitle(paste(g, "consecutive low score"))

# ggsave(last_plot(), file = 'images/03_selectionpipeline/nzmDNAH6consec.png')
knitr::include_graphics('images/03_selectionpipeline/nzmDNAH6consec.png')
# rm(gene_plot, stat_plot, wh, g)
```


```{r nzmRGS3, fig.cap='(ref:nzmRGS3)', eval= FALSE, message = FALSE, fig.height = 4, fig.width = 4}
# plot_gene("NZM","RGS3")
# ggsave(last_plot(), file = 'images/03_selectionpipeline/nzmRGS3.png')
knitr::include_graphics('images/03_selectionpipeline/nzmRGS3.png')
```

```{r nzmCTNNA3, eval= FALSE, message = FALSE, fig.height = 4, fig.width = 4}
# plot_gene("NZM","CTNNA3")
# ggsave(last_plot(), file = 'images/03_selectionpipeline/nzmCTNNA3.png')
knitr::include_graphics('images/03_selectionpipeline/nzmCTNNA3.png')
```

```{r nzmRAD51AP2, eval= FALSE, message = FALSE, fig.height = 4, fig.width = 4}
# plot_gene("NZM","RAD51AP2")
# ggsave(last_plot(), file = 'images/03_selectionpipeline/nzmRAD51AP2.png')
knitr::include_graphics('images/03_selectionpipeline/nzmRAD51AP2.png')
```


```{r nzmVSNL1, eval= FALSE, message = FALSE, fig.height = 4, fig.width = 4}
# plot_gene("NZM","VSNL1")
# ggsave(last_plot(), file = 'images/03_selectionpipeline/nzmVSNL1.png')
knitr::include_graphics('images/03_selectionpipeline/nzmVSNL1.png')
```


```{r nzmSUMPT3H, fig.cap='(ref:nzmSUPT3H)', eval= FALSE, message = FALSE, fig.height = 4, fig.width = 4}
#plot_gene("NZM","SUPT3H")
#ggsave(last_plot(), file = 'images/03_selectionpipeline/nzmSUPT3H.png')
knitr::include_graphics('images/03_selectionpipeline/nzmSUPT3H.png')
```



```{r, eval = FALSE}
# number of genes associated with urate, obesity, t2d, kd, and met syn
bind_rows(sig_ihs_val %>% mutate(statname = 'ihs'), sig_nsl_val %>% mutate(statname = 'nsl')) %>% filter(pop == 'NZM', genename %in% urate_goutGR$SYMBOL) %>% group_by(genename, statname) %>% tally() %>% arrange(desc(n))

bind_rows(sig_ihs_val %>% mutate(statname = 'ihs'), sig_nsl_val %>% mutate(statname = 'nsl')) %>% filter(pop == 'NZM', genename %in% t2d_GR$SYMBOL) %>% group_by(genename, statname) %>% tally() %>% arrange(desc(n))

bind_rows(sig_ihs_val %>% mutate(statname = 'ihs'), sig_nsl_val %>% mutate(statname = 'nsl')) %>% filter(pop == 'NZM', genename %in% obesity_GR$SYMBOL) %>% group_by(genename, statname) %>% tally() %>% arrange(desc(n))

bind_rows(sig_ihs_val %>% mutate(statname = 'ihs'), sig_nsl_val %>% mutate(statname = 'nsl')) %>% filter(pop == 'NZM', genename %in% metsyn_GR$SYMBOL) %>% group_by(genename, statname) %>% tally() %>% arrange(desc(n))
```


```{r, echo = FALSE, eval = FALSE}
<!--
_Tajimas D_

The \gls{nzm} had `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower significance threshold. These windows made `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions covering `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. Theses regions contained `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('NZM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Tajima.D', !pop %in% c('NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes were only from \gls{nzm}. The upper tail had `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that formed `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions covering `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. There were `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('NZM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Tajima.D', !pop %in% c('NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were unique to \gls{nzm} for \gls{td}.

_Fay and Wu's H_

There were `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows forming  `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions that met the significance threshold for the lower tail. These regions were a total of `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp and had `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('NZM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fay.Wu.H', !pop %in% c('NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` of which were only significant in the \gls{nzm} population for \gls{fwh}.

In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that formed  `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` continuous regions that met the significance threshold. These regions covered `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp and had `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('NZM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fay.Wu.H', !pop %in% c('NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant in the \gls{nzm} population for \gls{fwh}


_Fu and Li's D_

For the \gls{nzm} population there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower tail threshold for significance and made `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions covering `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. In these regions there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which  `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('NZM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.D', !pop %in% c('NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant in \gls{nzm} for \gls{fld}. In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the significance threshold. These windows made `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions with a total of `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp containing `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, with  `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('NZM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.D', !pop %in% c('NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only meeting significance in \gls{nzm} for \gls{fld}.

_Fu and Li's F_

For the \gls{nzm} population there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower tail threshold for significance and made `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions covering `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. In these regions there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which  `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('NZM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.F', !pop %in% c('NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant in \gls{nzm} for \gls{flf}. In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the significance threshold. These windows made `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions with a total of `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp containing `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, with  `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('NZM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.F', !pop %in% c('NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only meeting significance in \gls{nzm} for \gls{flf}.

_Zeng's E_

For the \gls{nzm} population there were `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower tail threshold for significance and made `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions covering `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. In these regions there were `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which  `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('NZM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Zeng.E', !pop %in% c('NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant in \gls{nzm} for \gls{ze}. In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the significance threshold. These windows made `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions with a total of `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('NZM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp containing `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('NZM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, with  `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('NZM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Zeng.E', !pop %in% c('NZM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only meeting significance in \gls{nzm} for \gls{ze}.
-->
```


#### Samoan


##### Pathway enrichment analysis of selected loci in \gls{sam}

<!-- - TD main processes: "cellular process", "metabolic process" -->
<!-- - FWH main processes: "cellular process", "metabolic process" -->
<!-- - FLD main processes: "cellular process", "metabolic process" -->
<!-- - FLF main processes: "cellular process", "metabolic process" -->
<!-- - ZE main processes: "cellular process", "metabolic process" -->
<!-- - ihs main processes: "cellular process", "metabolic process" -->
<!-- - nsl main processes: "cellular process", "metabolic process" -->
<!-- - all main processes: "cellular process", "metabolic process" -->
<!-- - all overrepresentation test: nothing -->

<!-- There were no significant results after adjusting for multiple testing when performing the over representation analysis using PantherDB on all genes that had met a significance threshold for any of the intra population selection and neutrality statistics. For all of the intra-population statistics, the main biological functional classification was "cellular process". -->

Only two pathways were significant for the gene-set pathway enrichment analysis using Enrichr. The pathway term "ABC transporters" was significant for gene lists from both \gls{fld} (adjusted P = 0.0190) with 8 of 44 genes and \gls{ihs} (adjusted P = 0.0128) with 9 of 44 genes. The second pathway that was significant was "long-term potentiation" and that was from the \gls{nsl} gene list (adjusted P = 0.0213) with 9 of 66 genes.


##### Selection from haplotypic statistics in \gls{sam}.

```{r, eval=FALSE}
p <- "SAM"

(sam_nsl_100 <- sig_nsl_val %>% filter(pop == p) %>% arrange(desc(abs(statvalue))) %>% head(n=100) %>% group_by(genename) %>% tally() %>% arrange(desc(n)))

(sam_ihs_100 <- sig_ihs_val %>% filter(pop == p) %>% arrange(desc(abs(statvalue))) %>% head(n=100) %>% group_by(genename) %>% tally() %>% arrange(desc(n)))

inner_join(sam_nsl_100, sam_ihs_100, by = 'genename' ) %>% mutate(total = n.x + n.y) %>% arrange(desc(total))
rm(p)
```

From the haplotypic tests for selection (\gls{ihs} and \gls{nsl}) there were `r sig_ihs_val %>% filter(pop == "SAM") %>% select(genename) %>% anti_join(., sig_ihs_val %>% filter(pop != "SAM") %>% select(genename), by = 'genename') %>% distinct() %>% NROW()` genes that were only significant in \gls{sam} for \gls{ihs}, and `r sig_nsl_val %>% filter(pop == "SAM") %>% select(genename) %>% anti_join(., sig_nsl_val %>% filter(pop != "SAM") %>% select(genename), by = 'genename') %>% distinct() %>% NROW()` for \gls{nsl}. The genes with the most significant markers for \gls{ihs} were _CDH23_ and _CNTN5_ with 12 significant markers each. The gene with the most significant markers for \gls{nsl} was _CDH23_ with 16 markers. Looking at the most extreme 100 markers for \gls{ihs} there were 35 genes represented, _SNX29_ had the most with three markers. For \gls{nsl} the most markers in the most extreme 100 was _C11orf65._ There was a total of 39 genes represented. Between the top markers for both \gls{ihs} and \gls{nsl} there were 10 genes that were in common. There were six genes that had windows from at least four of the frequency-based statistics and also had at least a single significant marker for either \gls{ihs} or \gls{nsl}. The genes were _DISP_ (two \gls{ihs} markers), _PARD3B_ (one \gls{ihs} and \gls{nsl} marker each), _NEK1_ (one \gls{ihs} marker), _CTNNA3_ (one \gls{ihs} marker and five \gls{nsl} markers), _CEP112_ (one \gls{ihs} and \gls{nsl} marker each), and _SUPT3H_ (one \gls{nsl} marker). \gls{xpehh} had 19 genes that had at least 20 populations with significant markers in only \gls{sam} from the Polynesian populations but none had been associated with urate or related co-morbidities.


<!-- PARD3B had 4 different statistics overlap under exon XXX. -->


<!-- Nearly all of the windows and markers that intersected _CNTNAP5_ were overlapping and fell between chr2:125545002-125675001, with the most overlap occurring in an intronic region (Figure \@ref(fig:samCNTNAP5)). -->




```{r, eval = FALSE}
q <- xpehh %>%  select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% filter(pop1 %in% c("CIM","NZM","SAM","TON"), xpehh_value > 0) %>% select(pop1,pop2, genename) %>% distinct() %>% left_join(., panel %>% select(pop2 = pop, super_pop) %>% distinct(), by = 'pop2') %>% group_by(pop1, super_pop, genename) %>% tally() %>% spread(super_pop, n, 0)%>% arrange(genename) %>% mutate(rs = AFR + AMR + EAS + EUR + POL + SAS) %>% filter(rs >=20) %>% select(pop1, genename, rs) %>% spread(pop1, rs) %>% filter(is.na(CIM),is.na(NZM), !is.na(SAM), is.na(TON)) %>% pull(genename)
q[q %in% genelists$genename]
```


##### Consecutive regions of score depression in \gls{sam}

There were no consecutively depressed score regions that intersected with genes associated with urate, obesity, \gls{t2d}, kidney disease, or metabolic syndrome. Regions of consecutive depressed score that also had significant markers for \gls{ihs} or \gls{nsl} in \gls{sam} covered three regions of the genome across two statistics. \gls{td} had two regions, the first was 400 kb at chr4:170025002-170425001 and had a single significant marker for \gls{ihs} in _NEK1_. The second region was 300 kb in length and was found at chr10:76685002-76985001. It had two markers that were significant  for \gls{ihs} and three for \gls{nsl} in _KAT6B_, two markers each for \gls{ihs} and \gls{nsl} at _SAMD8_, and two markers each fro \gls{ihs} and \gls{nsl} for DUSP13. \gls{flf} had a single 200 kb region at chr1:26465002-26665001 that had a single significant marker for \gls{nsl} in _CNKSR1_, and two markers for \gls{ihs} and three for \gls{nsl} in _AIM1L._




<!-- ##### FST -->

<!-- ##### Notes -->


```{r SAMSigRegions, eval = FALSE}
# copied into Panther adn then exported the biological process

#td
lower_sig_stats %>% filter(pop == 'SAM', statname == 'Tajima.D', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique() %>% cat()
#fwh
lower_sig_stats %>% filter(pop == 'SAM', statname == 'Fay.Wu.H', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique() %>% cat()
#fld
lower_sig_stats %>% filter(pop == 'SAM', statname == 'Fu.Li.D', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique() %>% cat()
#flf
lower_sig_stats %>% filter(pop == 'SAM', statname == 'Fu.Li.F', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique() %>% cat()
#ze
lower_sig_stats %>% filter(pop == 'SAM', statname == 'Zeng.E', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique() %>% cat()
#ihs
sig_ihs_val %>% filter(pop == 'SAM', !is.na(genename)) %>% .[['genename']] %>% unique() %>% cat()
#nsl
sig_nsl_val %>% filter(pop == 'SAM', !is.na(genename)) %>% .[['genename']] %>% unique() %>% cat()

# all with at least 1 sfs and 1 haplotypic
bind_rows(bind_rows(sig_ihs_val, sig_nsl_val) %>% filter(pop == 'SAM')%>% select(pop, "SYMBOL" = genename ) %>% distinct(), lower_sig_stats %>% filter(pop == 'SAM')%>% select(pop, SYMBOL) %>% distinct()) %>% filter(!is.na(SYMBOL))  %>% group_by(SYMBOL) %>% tally() %>% filter(n == 2) %>% .[['SYMBOL']] %>% cat()
```



<!-- no urate, obesity, t2d, kd, or metsyn -->

<!-- haplotypic results intersecting regions -->
```{r, eval = FALSE}
# find ihs/nsl intersecting the regions of consecutively depressed score
consec_lower_regions %>% filter( pop == 'SAM') %>% inner_join(., bind_rows(sig_ihs_val, sig_nsl_val) %>% select(SYMBOL = genename, everything()), by = c('pop','SYMBOL')) %>% select(-contains('gene'))
```

<!-- - td -->
<!--     - chr4:170025002-170425001 w=400000 -->
<!--         - NEK1 ihs -->
<!--     - chr10:76685002-76985001 w=300000 -->
<!--         - KAT6B 2x ihs, 3x nsl -->
<!--         - SAMD8 2x ihs, 2x nsl -->
<!--         - DUSP13 2x ihs, 2x nsl -->
<!-- - flf -->
<!--     - chr1:26465002-26665001 w=200000 -->
<!--         - CNKSR1 nsl -->
<!--         - AIM1L 2 x ihs, 3x nsl -->



<!-- (ref:upsetSAM) Upset plot of number of genes that were significant between  statistics for \gls{sam} -->

```{r upsetSam, fig.cap = '(ref:upsetSAM)', eval = FALSE}
UpSetR::upset(fromList(list(`Tajima's D` = lower_sig_stats %>% filter(pop == 'SAM', statname == 'Tajima.D', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(), 
                            `Fay and Wu's H` = lower_sig_stats %>% filter(pop == 'SAM', statname == 'Fay.Wu.H', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(), 
                            `Fu and Li's D` = lower_sig_stats %>% filter(pop == 'SAM', statname == 'Fu.Li.D', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(), 
                            `Fu and Li's F` = lower_sig_stats %>% filter(pop == 'SAM', statname == 'Fu.Li.F', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(),
                            `Zeng's E` = lower_sig_stats %>% filter(pop == 'SAM', statname == 'Zeng.E', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(),
                            iHS = sig_ihs_val %>% filter(pop == 'SAM', !is.na(genename) & genename != "") %>% .[['genename']] %>% unique(), 
                            nSL =sig_nsl_val %>% filter(pop == 'SAM', !is.na(genename) & genename != "") %>% .[['genename']] %>% unique())), 
              order.by = c("freq","degree"), nsets = 7, number.angles = 90, set_size.angles = 90, hjust = -0.25, vjust = 0.5 , nintersects = NA)
```

```{r svgSAMupset, include = FALSE, eval = FALSE}
svg(filename = 'images/03_selectionpipeline/sam_upset.svg')
  UpSetR::upset(fromList(list(`Tajima's D` = lower_sig_stats %>% filter(pop == 'SAM', statname == 'Tajima.D', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(), 
                            `Fay and Wu's H` = lower_sig_stats %>% filter(pop == 'SAM', statname == 'Fay.Wu.H', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(), 
                            `Fu and Li's D` = lower_sig_stats %>% filter(pop == 'SAM', statname == 'Fu.Li.D', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(), 
                            `Fu and Li's F` = lower_sig_stats %>% filter(pop == 'SAM', statname == 'Fu.Li.F', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(),
                            `Zeng's E` = lower_sig_stats %>% filter(pop == 'SAM', statname == 'Zeng.E', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(),
                            iHS = sig_ihs_val %>% filter(pop == 'SAM', !is.na(genename) & genename != "") %>% .[['genename']] %>% unique(), 
                            nSL =sig_nsl_val %>% filter(pop == 'SAM', !is.na(genename) & genename != "") %>% .[['genename']] %>% unique())), 
              order.by = c("freq","degree"), nsets = 7, number.angles = 90, set_size.angles = 90, hjust = -0.25, vjust = 0.5 , nintersects = NA)
  dev.off()
```

<!-- (ref:samCNTNAP5) Plot showing markers, genes and significant results for \gls{sam} at _CNTNAP5_ -->

```{r samCNTNAP5, fig.cap='(ref:samCNTNAP5)', eval= FALSE, message = FALSE, fig.height = 4, fig.width = 4}
#retrieve information of the gene of interest
# plot_gene("SAM","CNTNAP5")
# ggsave(last_plot(), file = 'images/03_selectionpipeline/samCNTNAP5.png')
knitr::include_graphics('images/03_selectionpipeline/samCNTNAP5.png')
```


```{r samSPAG16, fig.cap='(ref:samSPAG16)', eval= FALSE, message = FALSE, fig.height = 4, fig.width = 4}
#plot_gene("SAM","SPAG16")
#ggsave(last_plot(), file = 'images/03_selectionpipeline/samSPAG16.png')
knitr::include_graphics('images/03_selectionpipeline/samSPAG16.png')
```

```{r samSPECC1, fig.cap='(ref:samSPECC1)', eval= FALSE, message = FALSE, fig.height = 4, fig.width = 4}
#plot_gene("SAM","SPECC1")
#ggsave(last_plot(), file = 'images/03_selectionpipeline/samSPECC1.png')
knitr::include_graphics('images/03_selectionpipeline/samSPECC1.png')
```

```{r samSUPT3H, fig.cap='(ref:samSUPT3H)', eval= FALSE, message = FALSE, fig.height = 4, fig.width = 4}
# plot_gene("SAM","SUPT3H")
# ggsave(last_plot(), file = 'images/03_selectionpipeline/samSUPT3H.png')
knitr::include_graphics('images/03_selectionpipeline/samSUPT3H.png')
```

```{r samCTNNA3, fig.cap='(ref:samCTNNA3)', eval= FALSE, message = FALSE, fig.height = 4, fig.width = 4}
# plot_gene("SAM","CTNNA3")
# ggsave(last_plot(), file = 'images/03_selectionpipeline/samCTNNA3.png')
knitr::include_graphics('images/03_selectionpipeline/samCTNNA3.png')
```

```{r samCEP112, fig.cap='(ref:samCEP112)', eval= FALSE, message = FALSE, fig.height = 4, fig.width = 4}
# plot_gene("SAM","CEP112")
# ggsave(last_plot(), file = 'images/03_selectionpipeline/samCEP112.png')
knitr::include_graphics('images/03_selectionpipeline/samCEP112.png')
```

```{r samPARD3B, fig.cap='(ref:samPARD3B)', eval= FALSE, message = FALSE, fig.height = 4, fig.width = 4}
# plot_gene("SAM","PARD3B")
# ggsave(last_plot(), file = 'images/03_selectionpipeline/samPARD3B.png')
knitr::include_graphics('images/03_selectionpipeline/samPARD3B.png')
```


<!-- - td most specific -->
<!-- - fld most regions -->

<!-- In the upper tail, there were `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes for \gls{td} that intersected windows that met the significance threshold. \gls{fwh} had `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()`, \gls{fld} had `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()`, \gls{flf} had `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()`, and \gls{ze} had `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()`. -->

<!-- upper -->

<!-- - 5 stats: `r upper_sig_stats %>% filter(pop %in% c('SAM')) %>% select(SYMBOL, statname) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==5, !is.na(SYMBOL)) %>% .[['SYMBOL']]` -->
<!-- - 4 stats: `r upper_sig_stats %>% filter(pop %in% c('SAM')) %>% select(SYMBOL, statname) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==4, !is.na(SYMBOL)) %>% NROW()` -->
<!-- - 3 stats: `r upper_sig_stats %>% filter(pop %in% c('SAM')) %>% select(SYMBOL, statname) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==3, !is.na(SYMBOL)) %>% NROW()` -->
<!-- - 2 stats: `r upper_sig_stats %>% filter(pop %in% c('SAM')) %>% select(SYMBOL, statname) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==2, !is.na(SYMBOL)) %>% NROW()` -->
<!-- - 1 stats: `r upper_sig_stats %>% filter(pop %in% c('SAM')) %>% select(SYMBOL, statname) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==1, !is.na(SYMBOL)) %>% NROW()` -->


<!-- \gls{sam} had the most genes that were specific to \gls{sam} for \gls{td} out of the Polynesian populations for both significance thresholds. -->



```{r, echo = FALSE, eval = FALSE}
_Tajima's D_

In the \gls{sam} population there were `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that made up `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` continuous regions covering `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. In these regions there were`r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Tajima.D', !pop %in% c('SAM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were unique to \gls{sam}. For the upper tail there were  `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that formed  `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions with a total of `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. There were `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes that were covered by these regions, `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('SAM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Tajima.D', !pop %in% c('SAM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` that were only found in \gls{sam} for \gls{td}.

_Fay and Wu's H_

There were  `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that formed  `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions, totaling  `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp that met the significance threshold for the lower tail. These regions had `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('SAM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fay.Wu.H', !pop %in% c('SAM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant in the \gls{sam} population for \gls{fwh}.

In the upper tail there were  `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the threshold for significance. These windows formed `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions with a total of  `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. There were  `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('SAM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fay.Wu.H', !pop %in% c('SAM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only from the \gls{sam} population for \gls{fwh}. 

_Fu and Li's D_

For the \gls{sam} population there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower tail threshold for significance and made `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions covering `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. In these regions there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which  `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('SAM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.D', !pop %in% c('SAM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant in \gls{sam} for \gls{fld}. In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the significance threshold. These windows made `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions with a total of `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp containing `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, with  `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('SAM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.D', !pop %in% c('SAM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only meeting significance in \gls{sam} for \gls{fld}.

_Fu and Li's F_

For the \gls{sam} population there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower tail threshold for significance and made `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions covering `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. In these regions there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which  `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('SAM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.F', !pop %in% c('SAM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant in \gls{sam} for \gls{flf}. In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the significance threshold. These windows made `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions with a total of `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp containing `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, with  `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('SAM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.F', !pop %in% c('SAM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only meeting significance in \gls{sam} for \gls{flf}.

_Zeng's E_

For the \gls{sam} population there were `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower tail threshold for significance and made `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions covering `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. In these regions there were `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which  `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('SAM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Zeng.E', !pop %in% c('SAM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant in \gls{sam} for \gls{ze}. In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the significance threshold. These windows made `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions with a total of `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('SAM')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp containing `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('SAM')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, with  `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('SAM')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Zeng.E', !pop %in% c('SAM')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only meeting significance in \gls{sam} for \gls{ze}.
```



```{r samCREB1, fig.cap='(ref:samCREB1)', eval= FALSE, message = FALSE, fig.height = 4, fig.width = 4}
#plot_gene("SAM","CREB1")
#ggsave(last_plot(), file = 'images/03_selectionpipeline/samCREB1.png')
knitr::include_graphics('images/03_selectionpipeline/samCREB1.png')
```


<!-- Enrichr using the 0.1 percentile -->

<!-- TD has 3 significant pathway hits -->

<!-- FLD had 39 pathways -->
<!-- FLF had 48 pathways -->

<!-- ihs > 3.89, significant pathway for ABC transporters -->

#### Tongan

```{r TonSigRegions, eval = FALSE}
# copied into Panther adn then exported the biological process

#td
lower_sig_stats %>% filter(pop == 'TON', statname == 'Tajima.D', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique() %>% cat()
#fwh
lower_sig_stats %>% filter(pop == 'TON', statname == 'Fay.Wu.H', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique() %>% cat()
#fld
lower_sig_stats %>% filter(pop == 'TON', statname == 'Fu.Li.D', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique() %>% cat()
#flf
lower_sig_stats %>% filter(pop == 'TON', statname == 'Fu.Li.F', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique() %>% cat()
#ze
lower_sig_stats %>% filter(pop == 'TON', statname == 'Zeng.E', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique() %>% cat()
#ihs
sig_ihs_val %>% filter(pop == 'TON', !is.na(genename)) %>% .[['genename']] %>% unique() %>% cat()
#nsl
sig_nsl_val %>% filter(pop == 'TON', !is.na(genename)) %>% .[['genename']] %>% unique() %>% cat()

# all with at least 1 sfs and 1 haplotypic
bind_rows(bind_rows(sig_ihs_val, sig_nsl_val) %>% filter(pop == 'TON')%>% select(pop, "SYMBOL" = genename ) %>% distinct(), lower_sig_stats %>% filter(pop == 'TON')%>% select(pop, SYMBOL) %>% distinct()) %>% filter(!is.na(SYMBOL))  %>% group_by(SYMBOL) %>% tally() %>% filter(n == 2) %>% .[['SYMBOL']] %>% cat()
```

##### Pathway enrichment analysis of selected loci in \gls{ton}

<!-- Functional classification using PantherDB for the significant windows from the 1st percentile for the frequency spectrum selection and neutrality tests, and the significant \gls{ihs} and \gls{nsl} each individually, had the main biological process classification as "cellular process" and the second main classification as 'metabolic process'. Combining all the significant genes from any -->

Pathway enrichment analysis from the Enrichr KEGG 2016 pathways had `r kegg_sig %>% filter(pop == 'ton') %>% select(Term) %>% distinct() %>% NROW()` different pathways that were significant after multiple testing correction. Of the \gls{sfs} based methods, only \gls{flf} had a significant pathway and this was "Olfactory transduction". There were two pathway terms that were significant from the \gls{ihs} results, these were "ABC transporters" and "ECM-receptor interaction". These were also significant in the \gls{nsl} results, along with another eight terms (Table \@ref(tab:keggPath)). Five of the significant pathway terms were only significant in \gls{ton} out of the Polynesian populations.


##### Selection from haplotypic statistics in \gls{ton}.

```{r, eval=FALSE}
p <- "TON"

(ton_nsl_100 <- sig_nsl_val %>% filter(pop == p) %>% arrange(desc(abs(statvalue))) %>% head(n=100) %>% group_by(genename) %>% tally() %>% arrange(desc(n)))

(ton_ihs_100 <- sig_ihs_val %>% filter(pop == p) %>% arrange(desc(abs(statvalue))) %>% head(n=100) %>% group_by(genename) %>% tally() %>% arrange(desc(n)))

inner_join(ton_nsl_100, ton_ihs_100, by = 'genename' ) %>% mutate(total = n.x + n.y) %>% arrange(desc(total))
rm(p)
```


In total there were  `r sig_ihs_val %>% filter(pop == "TON") %>% select(genename) %>% anti_join(., sig_ihs_val %>% filter(pop != "TON") %>% select(genename), by = 'genename') %>% distinct() %>% NROW()` genes that were only significant in \gls{ton} for \gls{ihs}, and `r sig_nsl_val %>% filter(pop == "TON") %>% select(genename) %>% anti_join(., sig_nsl_val %>% filter(pop !="TON") %>% select(genename), by = 'genename') %>% distinct() %>% NROW()` for \gls{nsl}. In the most extreme 100 \gls{ihs} marker scores, there were 32 genes represented with _SNX29_ having three markers, and _CDH23_ having two from the 100. The most extreme 100 markers for \gls{nsl} had 30 genes represented with SNX29 having five markers, _PCDH15_ having three, and _DNAH11_ having two of the 100 markers. There was an overlap between the two genelists for the most extreme 100 of six genes, with _SNX29_, and _DNAH11_ having the most number of extreme markers for both \gls{ihs}, and \gls{nsl}. The genes with the most number of significant markers overall for \gls{ihs} and \gls{nsl} were _CPNE4_ (16 markers for both \gls{ihs} and \gls{nsl}), _SNX29_ (15 markers for \gls{ihs} and 16 for \gls{nsl}), _CDH23_ (11 markers for \gls{ihs} and 12 for \gls{nsl}), _PCDH15_ (12 markers for \gls{nsl}), and _BANK1_ (9 markers for \gls{ihs} and 10 for \gls{nsl}). There were 37 genes that had at least 20 populations with significant markers for \gls{xpehh} in only TON from the Polynesian populations, including _IGF1R_, _JAZF1_, and _PEPD_ which had been associated with urate or co-morbidities. Combining the genes with at least a single significant marker for either \gls{ihs} or \gls{nsl} with the windows in the 1st percentile for at least four frequency-based intra-population selection and neutrality statistics had three genes that met this criteria. They were _SUPT3H_, _LPO_, and _NRXN3._


```{r, eval = FALSE}
q <- xpehh %>%  select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% filter(pop1 %in% c("CIM","NZM","SAM","TON"), xpehh_value > 0) %>% select(pop1,pop2, genename) %>% distinct() %>% left_join(., panel %>% select(pop2 = pop, super_pop) %>% distinct(), by = 'pop2') %>% group_by(pop1, super_pop, genename) %>% tally() %>% spread(super_pop, n, 0)%>% arrange(genename) %>% mutate(rs = AFR + AMR + EAS + EUR + POL + SAS) %>% filter(rs >=20) %>% select(pop1, genename, rs) %>% spread(pop1, rs) %>% filter(is.na(CIM),is.na(NZM), is.na(SAM), !is.na(TON)) %>% pull(genename)
q[q %in% genelists$genename]
```

##### Consecutive regions of score depression in \gls{ton}

There were no consecutively depressed score regions that intersected genes associated with urate, \gls{t2d}, kidney disease, or metabolic syndrome. There was one region that intersected the obesity associated gene _CCR3_ for \gls{td} and was 260 kb in length at chr3:46035002-46295001. There were three regions with consecutively depressed score that also had significant markers for \gls{ihs} or \gls{nsl} in \gls{ton}. \gls{td} had two regions. The first similar to \gls{sam}, was at chr10:76685002-76985001 and spanned 300 kb and had three significant markers for \gls{ihs} and four for \gls{nsl} in _KAT6B_, three markers for both \gls{ihs} and \gls{nsl} in _SAMD8_, and two markers each for \gls{ihs} and \gls{nsl} in _DUSP13._ The second region at chr17:56125002-56415001 was 290 kb and had a single significant marker for \gls{ihs} in both _EPX_ and _LPO_. \gls{flf} had a slightly shifted region that also covered _EPX_ and _LPO_, but also included _RNF43_. All three genes had a single significant marker each for \gls{ihs}. The region was at chr17:56205002-56455001 and was 250 kb in length.







<!-- ##### FST -->


<!-- ##### Notes -->

<!-- - TD main processes: "cellular process", "metabolic process" -->
<!-- - FWD main processes: "cellular process", "metabolic process" -->
<!-- - FLD main processes: "cellular process", "metabolic process" -->
<!-- - FLF main processes: "cellular process", "metabolic process" -->
<!-- - ZE main processes: "cellular process", "metabolic process" -->
<!-- - ihs main processes: "cellular process", "metabolic process" -->
<!-- - nsl main processes: "cellular process", "metabolic process" -->
<!-- - all main processes: "cellular process", "metabolic process" -->
<!-- - all overrepresentation test: nothing -->




<!-- From all of the intra population -->

<!-- - There were `r bind_rows(ton_urate_sig) %>% select(SYMBOL) %>% distinct() %>% NROW()` of `r length(unique(urate_goutGR$SYMBOL))` urate genes -->
<!-- - There were `r bind_rows(ton_obesity_sig) %>% select(SYMBOL) %>% distinct() %>% NROW()` of `r length(unique(obesity_GR$SYMBOL))` obesity genes -->
<!-- - There were `r bind_rows(ton_t2d_sig) %>% select(SYMBOL) %>% distinct() %>% NROW()` of `r length(unique(t2d_GR$SYMBOL))` \gls{t2d} genes -->
<!-- - There were `r bind_rows(ton_kd_sig) %>% select(SYMBOL) %>% distinct() %>% NROW()` of `r length(unique(kd_GR$SYMBOL))` kidney disease genes -->
<!-- - There were `r bind_rows(ton_metsyn_sig) %>% select(SYMBOL) %>% distinct() %>% NROW()` of `r length(unique(metsyn_GR$SYMBOL))` metsyn genes -->

<!-- lower -->

<!-- - 5 stats: `r lower_sig_stats %>% filter(pop %in% c('TON')) %>% select(SYMBOL, statname) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==5, !is.na(SYMBOL)) %>% .[['SYMBOL']]` -->
<!-- - 4 stats: `r lower_sig_stats %>% filter(pop %in% c('TON')) %>% select(SYMBOL, statname) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==4, !is.na(SYMBOL)) %>% NROW()` -->
<!-- - 3 stats: `r lower_sig_stats %>% filter(pop %in% c('TON')) %>% select(SYMBOL, statname) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==3, !is.na(SYMBOL)) %>% NROW()` -->
<!-- - 2 stats: `r lower_sig_stats %>% filter(pop %in% c('TON')) %>% select(SYMBOL, statname) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==2, !is.na(SYMBOL)) %>% NROW()` -->
<!-- - 1 stats: `r lower_sig_stats %>% filter(pop %in% c('TON')) %>% select(SYMBOL, statname) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==1, !is.na(SYMBOL)) %>% NROW()` -->





<!-- no urate, t2d, kd, or metsyn -->

<!-- obesity -->
<!-- td chr3:46035002-46295001 w=260000 CCR3  -->

<!-- haplotypic results intersecting regions -->
```{r, eval = FALSE}
# find ihs/nsl intersecting the regions of consecutively depressed score
consec_lower_regions %>% filter( pop == 'TON') %>% inner_join(., bind_rows(sig_ihs_val, sig_nsl_val) %>% select(SYMBOL = genename, everything()), by = c('pop','SYMBOL')) %>% select(-contains('gene'))
```

<!-- -td -->
<!--     - chr10:76685002-76985001 w=300000 -->
<!--         - KAT6B 3x ihs, 4x nsl -->
<!--         - SAMD8 3x ihs, 3x nsl -->
<!--         - DUSP13 2x ihs, 2x nsl -->
<!--     - chr17:56125002-56415001 w=290000 -->
<!--         - EPX ihs -->
<!--         - LPO ihs -->
<!-- - flf -->
<!--     - chr17 56205002 56455001 250000 -->
<!--         - EPX ihs -->
<!--         - LPO ihs -->
<!--         - RNF43 ihs -->



(ref:upsetTON) TON Upset Plot

```{r upsetTon, fig.cap = '(ref:upsetTON)', eval = FALSE}
UpSetR::upset(fromList(list(`Tajima's D` = lower_sig_stats %>% filter(pop == 'TON', statname == 'Tajima.D', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(), 
                            `Fay and Wu's H` = lower_sig_stats %>% filter(pop == 'TON', statname == 'Fay.Wu.H', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(), 
                            `Fu and Li's D` = lower_sig_stats %>% filter(pop == 'TON', statname == 'Fu.Li.D', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(), 
                            `Fu and Li's F` = lower_sig_stats %>% filter(pop == 'TON', statname == 'Fu.Li.F', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(),
                            `Zeng's E` = lower_sig_stats %>% filter(pop == 'TON', statname == 'Zeng.E', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(),
                            iHS = sig_ihs_val %>% filter(pop == 'TON', !is.na(genename) & genename != "") %>% .[['genename']] %>% unique(), 
                            nSL =sig_nsl_val %>% filter(pop == 'TON', !is.na(genename) & genename != "") %>% .[['genename']] %>% unique())), 
              order.by = c("freq", "degree"), nsets = 7, number.angles = 90, set_size.angles = 90, hjust = -0.25, vjust = 0.5, nintersects = NA)
```


```{r svgTONupset, include = FALSE, eval = FALSE}
svg(filename = 'images/03_selectionpipeline/ton_upset.svg')
UpSetR::upset(fromList(list(`Tajima's D` = lower_sig_stats %>% filter(pop == 'TON', statname == 'Tajima.D', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(), 
                            `Fay and Wu's H` = lower_sig_stats %>% filter(pop == 'TON', statname == 'Fay.Wu.H', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(), 
                            `Fu and Li's D` = lower_sig_stats %>% filter(pop == 'TON', statname == 'Fu.Li.D', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(), 
                            `Fu and Li's F` = lower_sig_stats %>% filter(pop == 'TON', statname == 'Fu.Li.F', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(),
                            `Zeng's E` = lower_sig_stats %>% filter(pop == 'TON', statname == 'Zeng.E', !is.na(SYMBOL)) %>% .[['SYMBOL']] %>% unique(),
                            iHS = sig_ihs_val %>% filter(pop == 'TON', !is.na(genename) & genename != "") %>% .[['genename']] %>% unique(), 
                            nSL =sig_nsl_val %>% filter(pop == 'TON', !is.na(genename) & genename != "") %>% .[['genename']] %>% unique())), 
              order.by = c("freq", "degree"), nsets = 7, number.angles = 90, set_size.angles = 90, hjust = -0.25, vjust = 0.5, nintersects = NA)
dev.off()
```

```{r tonLPO,  fig.cap='(ref:tonLPO)', eval=FALSE, message = FALSE, fig.height = 4, fig.width = 4}
# plot_gene("TON","LPO")
# ggsave(last_plot(), file = 'images/03_selectionpipeline/tonLPO.png')
knitr::include_graphics('images/03_selectionpipeline/tonLPO.png')
```



```{r tonNRXN3, fig.cap='(ref:tonNRXN3)', eval= FALSE, message = FALSE, fig.height = 4, fig.width = 4}
# plot_gene("TON","NRXN3")
# ggsave(last_plot(), file = 'images/03_selectionpipeline/tonNRXN3.png')
knitr::include_graphics('images/03_selectionpipeline/tonNRXN3.png')
```

```{r tonCENPP, fig.cap='(ref:tonCENPP)', eval= FALSE, message = FALSE, fig.height = 4, fig.width = 4}
#plot_gene("TON","CENPP")
#ggsave(last_plot(), file = 'images/03_selectionpipeline/tonCENPP.png')
knitr::include_graphics('images/03_selectionpipeline/tonCENPP.png')
```


```{r tonSH3RF1, fig.cap='(ref:tonSH3RF1)', eval= FALSE, message = FALSE, fig.height = 4, fig.width = 4}
#plot_gene("TON","SH3RF1")
#ggsave(last_plot(), file = 'images/03_selectionpipeline/tonSH3RF1.png')
knitr::include_graphics('images/03_selectionpipeline/tonSH3RF1.png')
```

```{r tonSUPT3H, fig.cap='(ref:tonSUPT3H)', eval= FALSE, message = FALSE, fig.height = 4, fig.width = 4}
# plot_gene("TON","SUPT3H")
# ggsave(last_plot(), file = 'images/03_selectionpipeline/tonSUPT3H.png')
knitr::include_graphics('images/03_selectionpipeline/tonSUPT3H.png')
```




<!-- - From the lower tail there were `r consec_lower_regions %>% filter(stat == 'Tajima.D', pop == 'TON') %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce(ignore.strand = TRUE) %>% length()` regions of extended significant windows for \gls{td}, `r consec_lower_regions %>% filter(stat == 'Fay.Wu.H', pop == 'TON') %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce(ignore.strand = TRUE) %>% length()` for \gls{fwh}, `r consec_lower_regions %>% filter(stat == 'Fu.Li.D', pop == 'TON') %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce(ignore.strand = TRUE) %>% length()` for \gls{fld}, `r consec_lower_regions %>% filter(stat == 'Fu.Li.F', pop == 'TON') %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce(ignore.strand = TRUE) %>% length()` for \gls{flf}, and `r consec_lower_regions %>% filter(stat == 'Zeng.E', pop == 'TON') %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce(ignore.strand = TRUE) %>% length()` for \gls{ze}. -->






<!-- THADA Plot -->
```{r tonTHADA, fig.cap='(ref:tonTHADA)', eval= FALSE, message = FALSE, fig.height = 4, fig.width = 4}
#plot_gene("TON","THADA")
#ggsave(last_plot(), file = 'images/03_selectionpipeline/tonTHADA.png')
knitr::include_graphics('images/03_selectionpipeline/tonTHADA.png')
```


<!-- Enrichr using 0.1 percentile -->

<!-- FLD 1 pathway significant - olfactory transduction -->
<!-- FLF 1 pathway significant - olfactory transduction -->

<!-- ihs > 3.89, type 1 diabetes melitus was top hit but P = 0.079 -->
<!-- nsl > 3.89 insulin resistance and axon guidance were both significant -->

```{r, echo = FALSE, eval = FALSE}
_Tajima's D_

For the \gls{ton} population there were `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower tail threshold for significance and made `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions covering `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. In these regions there were `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which  `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Tajima.D', !pop %in% c('TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant in \gls{ton} for \gls{td}. In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the significance threshold. These windows made `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions with a total of `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp containing `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, with  `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Tajima.D', !pop %in% c('TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only meeting significance in \gls{ton} for \gls{td}.

_Fay and Wu's H_

There were  `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that reached the significance threshold in the lower tail. These windows made `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()`independent continuous regions totaling `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6`. Within these regions there were  `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fay.Wu.H', !pop %in% c('TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes that were only significant in \gls{ton} for \gls{fwh}, from `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes that were significant.

In the upper tail there were  `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the significance threshold making up  `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions that were independent and continuous with a total length of `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. Within these regions there were `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which  `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fay.Wu.H', !pop %in% c('TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were significant in only \gls{ton} for \gls{fwh}.

_Fu and Li's D_

For the \gls{ton} population there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower tail threshold for significance and made `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions covering `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. In these regions there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which  `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.D', !pop %in% c('TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant in \gls{ton} for \gls{fld}. In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the significance threshold. These windows made `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions with a total of `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp containing `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, with  `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.D', !pop %in% c('TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only meeting significance in \gls{ton} for \gls{fld}.

_Fu and Li's F_

For the \gls{ton} population there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower tail threshold for significance and made `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions covering `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. In these regions there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which  `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.F', !pop %in% c('TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant in \gls{ton} for \gls{flf}. In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the significance threshold. These windows made `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions with a total of `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp containing `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, with  `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.F', !pop %in% c('TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only meeting significance in \gls{ton} for \gls{flf}.

_Zeng's E_

For the \gls{ton} population there were `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower tail threshold for significance and made `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions covering `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. In these regions there were `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, of which  `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Zeng.E', !pop %in% c('TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant in \gls{ton} for \gls{ze}. In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the significance threshold. These windows made `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions with a total of `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp containing `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, with  `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Zeng.E', !pop %in% c('TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes only meeting significance in \gls{ton} for \gls{ze}.
```


### Disease-associated genes

```{r}
filterGenes <- c("ABHD6","ALPK2","APEH","ASAP1","AUH","BIN1","BRD2","C2","C2orf74","C4orf17","C5orf30","C6orf15","CARD6","CARD9","CASP7","CCDC122","CCNY","CCR9","CD200R1","CD247","CD80","CD86","CDH1","CDH3","CFB","CDHR5","CFLAR","CHP1","CLEC16A","CMC1","CNTN6","COBL","DAP","DAP3","DDAH1","DDR1","DDX6","DGKD","ERG","DIEXF", "FKBPL","GRHL2","GLIS3","GIN1","IL12B","IL12RB2","IR1R1","IR1RL1","IL2RB","IL7","ILDR1","INS","KIAA1841","KCNIP4","LACC1" ,"LMO7", "KIF3B","MTMR9","MSTO1","MIR146A","MAGI3","NAV1","OR5B12","OSMR", "PPT2","PPT2-EGFL8", "PXK","PLCL1","PBX2","PDGFB","PEX13","PGPEP1","PNMALL1","TNFSF15","RAPGEF6","RAPGEF6" ,"RSBN1","RUNX1","RUSC2","SAMD12","SDCCAG3","SBNO3","SEC61G", "SGIP1","SHISA5","SLC11A1", "SLC12A1","SLC15A2","SLC16A10","SLC1A7","SLC44A4","SNAPC4","STK32B","TEF","TET3","TLL1","TNC","TOB2","UBAC2","USP34","ZNF433","ZNF438","ZNF774","ZNF831")


```



<!-- Genes talked about are cherry picked based on  -->

<!-- - number of snps that were significant -->
<!-- - SFS and haplotypic -->
<!-- - multiple polynesian populations -->
<!-- - only significant (or highly enriched) within the polynesians -->
<!-- - on multiple gene lists -->
<!-- - prioritised haplotypic results -->

In order to establish if there was evidence of selection in genes associated with urate and diseases with \gls{hu} as a co-morbidity, loci that associated with these conditions were extracted from the \gls{gwas} catalog (as per Section \@ref(diseaselist), see Table \@ref(tab:gwascatref) for references for each trait) and neutrality and selection statistics that intersected these loci were collated. Table \@ref(tab:superDiseaseGenesTab) shows the break down of the number of different statistics by population that met the significance thresholds for the \gls{sfs} based statistics, the haplotype based statistics, and the overlap between the frequency spectrum and haplotypic methods. The \gls{cim} had the least number of genes of the Polynesian populations that had a significant result out of the genes that were associated with urate. The Polynesian populations had the lowest number of genes with significant \gls{ihs} or \gls{nsl} for obesity and the Western Polynesian populations had the lowest number of genes for \gls{t2d}.

```{r polDiseaseGenesTab, eval = FALSE}
bind_rows(
lower_sig_stats %>% select(SYMBOL, pop) %>% filter(pop %in% poly_pop) %>% 
  mutate(
    urate = ifelse(SYMBOL %in% urate_goutGR$SYMBOL, TRUE, FALSE),
    obesity = ifelse(SYMBOL %in% obesity_GR$SYMBOL, TRUE, FALSE),
    t2d = ifelse(SYMBOL %in% t2d_GR$SYMBOL, TRUE, FALSE),
    kd = ifelse(SYMBOL %in% kd_GR$SYMBOL, TRUE, FALSE),
    metsyn = ifelse(SYMBOL %in% metsyn_GR$SYMBOL, TRUE, FALSE)
    ) %>% 
  select(SYMBOL, pop, urate, obesity, t2d, kd, metsyn) %>% 
  distinct() %>% 
  group_by(pop) %>% 
  summarise(urate = sum(urate), obesity = sum(obesity), t2d = sum(t2d), kd = sum(kd), metsyn = sum(metsyn)) %>% 
  mutate(type = 'SFS'),
bind_rows(sig_ihs_val, sig_nsl_val) %>% 
  select(pop, 'SYMBOL' = genename) %>% 
  filter(pop %in% poly_pop) %>% 
  mutate(
    urate = ifelse(SYMBOL %in% urate_goutGR$SYMBOL, TRUE, FALSE),
    obesity = ifelse(SYMBOL %in% obesity_GR$SYMBOL, TRUE, FALSE),
    t2d = ifelse(SYMBOL %in% t2d_GR$SYMBOL, TRUE, FALSE),
    kd = ifelse(SYMBOL %in% kd_GR$SYMBOL, TRUE, FALSE),
    metsyn = ifelse(SYMBOL %in% metsyn_GR$SYMBOL, TRUE, FALSE)
    ) %>% 
  select(SYMBOL, pop, urate, obesity, t2d, kd, metsyn) %>% 
  distinct() %>% 
  group_by(pop) %>% 
  summarise(urate = sum(urate), obesity = sum(obesity), t2d = sum(t2d), kd = sum(kd), metsyn = sum(metsyn)) %>% 
  mutate(type = 'Haplotype'),

bind_rows(
  bind_rows(sig_ihs_val, sig_nsl_val) %>% 
    select(pop, 'SYMBOL' = genename),
          lower_sig_stats %>% select(SYMBOL, pop)
)%>% 
  filter(pop %in% poly_pop) %>% 
  mutate(
    urate = ifelse(SYMBOL %in% urate_goutGR$SYMBOL, TRUE, FALSE),
    obesity = ifelse(SYMBOL %in% obesity_GR$SYMBOL, TRUE, FALSE),
    t2d = ifelse(SYMBOL %in% t2d_GR$SYMBOL, TRUE, FALSE),
    kd = ifelse(SYMBOL %in% kd_GR$SYMBOL, TRUE, FALSE),
    metsyn = ifelse(SYMBOL %in% metsyn_GR$SYMBOL, TRUE, FALSE)
    ) %>% 
  select(SYMBOL, pop, urate, obesity, t2d, kd, metsyn) %>% 
  distinct() %>% 
  group_by(pop) %>%
  summarise(urate = sum(urate), obesity = sum(obesity), t2d = sum(t2d), kd = sum(kd), metsyn = sum(metsyn)) %>% 
  mutate(type = 'combined')
) %>% 
  kable(., col.names = c('Population','Urate / Gout','Obesity','T2D','Kidney Disease','Metabolic Syndrome','Type'), caption = '(\\#tab:polDiseaseGenesTab) Number of Genes with evidence of possible positive selection in Polynesians', booktabs = TRUE)
```

```{r superDiseaseGenesTab, warning=FALSE, eval = FALSE}
superDiseaseGeneTab2 %>% 
  kable(., col.names = c('Population',"Super Pop",'Urate / Gout','Obesity','T2D','Kidney Disease','Metabolic Syndrome'), caption = '(\\#tab:superDiseaseGenesTab) Number of Genes with evidence of possible positive selection in Super populations from intra-population tests from urate and co-morbidities \\gls{gwas} associated loci.', format ='latex', booktab = TRUE) %>% footnote(c("Each category is reported as the number of genes that were significant in the format 'SFS; Haplotypic; Combined'.", "Combined is the number of genes that had both SFS and haplotypic evidence."))
```

```{r}
superDiseaseGeneTab3 %>% kable(., col.names = c("Population","Super Pop.", rep(c("SFS","Hap.","Com."), 5)) , format = 'latex', booktabs = TRUE, caption = '(\\#tab:superDiseaseGenesTab) Number of genes with evidence of possible positive selection in super populations from intra-population tests from urate and co-morbidities \\gls{gwas} associated loci.') %>% kableExtra::row_spec(., 0, angle = 90) %>% add_header_above(., c("", "", 'Urate / Gout'=3,'Obesity'=3,'T2D'=3,'Kidney Disease'=3,'Metabolic Syndrome'=3)) %>% footnote(.,"Hap = haplotypic. Com = combined. Combined is the number of genes that had both SFS and haplotypic evidence.", general_title = "", threeparttable = TRUE)
```



#### Urate and gout
```{r}
g <- urate_goutGR$SYMBOL
uratePol <- xpehh %>%  select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% filter(pop1 %in% c("CIM","NZM","SAM","TON"), xpehh_value > 0) %>% filter(genename %in% g) %>% select(pop1,pop2, genename) %>% distinct() %>% left_join(., panel %>% select(pop2 = pop, super_pop) %>% distinct(), by = 'pop2') %>% group_by(pop1, super_pop, genename) %>% tally() %>% spread(super_pop, n, "") %>% rename(pop = pop1) %>% full_join(., sig_ihs_val %>% filter(!is.na(genename), genename != "", genename %in% g, pop %in% poly_pop) %>% select(pop, genename) %>% group_by(pop,genename) %>% tally(), by = c('pop','genename')) %>% rename(ihs = n) %>% full_join(., sig_nsl_val %>% filter(!is.na(genename), genename != "", genename %in% g, pop %in% poly_pop) %>% select(pop, genename) %>% group_by(pop,genename) %>% tally(), by = c('pop','genename')) %>% rename(nsl = n) %>%full_join(., lower_sig_stats %>% filter(pop %in% poly_pop, SYMBOL %in% g)%>% select(-contains('chrom')) %>% group_by(pop, SYMBOL, statname) %>% tally() %>% spread(statname, n), by = c('pop', 'genename' = 'SYMBOL') ) %>% arrange(genename)

uratePol %>% mutate_all(., function(x){replace(x, is.na(x), "")}) %>% select(genename, pop, everything()) %>%  kable(., caption = '(\\#tab:uratePol) Number of populations, windows, and markers with significant selection and neutrality statistics in Polynesian populations for all GWAS associated genes for urate and gout', col.names = c('Gene','Population','AFR','AMR','EAS','EUR','POL','SAS','iHS','nSL',"Fay & Wu's H", "Fu & Li's D"," Fu & Li's F", "Tajima's D"," Zeng's E"), booktabs = TRUE, format ='latex') %>% kableExtra::row_spec(0, angle = 90) %>% add_header_above(., c("","","XP-EHH" = 6, rep("",7))) # %>% filter((AFR > 0 | AMR >0 | EAS > 0 | ihs > 0 | nsl > 0) & (!is.na(Fay.Wu.H) | !is.na(Fu.Li.D)| !is.na(Fu.Li.F) | !is.na(Tajima.D) | !is.na(Zeng.E)))
rm(g)
```

```{r, eval = FALSE}
lower_sig_stats %>% filter(pop %in% poly_pop, SYMBOL %in% urate_goutGR$SYMBOL) %>% select(pop, SYMBOL, statname) %>% distinct()%>% spread(statname, statname, "") %>% arrange(SYMBOL)
```


The `r urate_goutGR$SYMBOL %>% unique() %>% length()` loci identified through \gls{gwas} [@Kottgen2013] as being associated with urate and gout only had a small proportion with any evidence of possible selection. In the Polynesian populations, from the intra-population neutrality and selection statistics, there were 10 genes that had some evidence. However, there were only three genes with haplotypic evidence, and only one gene (_PRKAG2_) that had both haplotypic and \gls{sfs}-based evidence. The Western Polynesian populations were the only Polynesian populations to have haplotypic-based evidence. Overall, the Polynesian populations did not have more urate genes with evidence than the other populations (Table \@ref(tab:superDiseaseGenesTab)).

The two main effect loci for serum urate, _SLC2A9_ and _ABCG2_ both had limited evidence of selection. _SLC2A9_ did not have any values for \gls{ihs} or \gls{nsl} because of an extended region of no recombination leading to a large gap in the reference genome which caused the \gls{ehh} calculation to be terminated. There was evidence with \gls{fld} and \gls{flf} which both had windows from _SLC2A9_ in the 1st percentile from \gls{ton}. This was also observed with the \gls{eas} populations of \gls{chb} and \gls{jpt}. For Polynesian populations, _ABCG2_ only had  significant results from \gls{ihs} and \gls{nsl} at the same marker (rs2622626) for \gls{ton}. With \gls{ihs} the \gls{sas} populations of \gls{gih} and \gls{itu} also had significant values for rs2622626. For \gls{nsl}, along with \gls{gih}, and \gls{itu}, the other non-Polynesian populations that had a significant value was the \gls{eas} population \gls{jpt}, and the \gls{afr} population of \gls{yri}. Rs2622626 is associated with both urate and gout [@Kottgen2013]. There were no Polynesian populations with values in the 1st percentile for the frequency-based statistics, and only \gls{esn} had windows that met the threshold for \gls{fwh}. The only other transporter that was associated with urate that had a window meeting the significance threshold in Polynesian populations was _SLC16A9_ in \gls{ton} for \gls{fld}.

Table \@ref(tab:uratePol) shows all urate and gout associated genes that had any evidence of possible selection in the Polynesian populations. Some genes of interest included _IGF1R_ and _RREB1_ which both had significant markers with \gls{xpehh} between nearly all Polynesian populations and many of the other populations excepting those in the \gls{sas} super population. _RREB1_ is also associated with \gls{t2d} [@mahajan2014genome]. _BCAS3_ had multiple windows that met the lower threshold for \gls{fwh}, \gls{fld}, \gls{flf}, and \gls{td} in the Polynesian populations but did not have any significant markers for the haplotypic tests of selection. There were no regions of consecutive score depression in the Polynesian populations that intersected the urate and gout loci.

<!-- rs2622626 is the snp that is significant in ton, but also SAS - GIH and ITU, and JPT. should check to see if it is in LD with the gwas hits -->

```{r}
urate_poly_interst <- c("BCAS3","IGF1R","RREB1","ABCG2")
```


```{r ihsUrate, eval = FALSE}
# Gout specific
# number of ihs snps by gout gene by population
sig_ihs_val %>% 
  filter(genename %in% urate_goutGR$SYMBOL) %>% 
  group_by(pop, genename) %>% tally() %>%
  arrange(genename) %>% 
  spread(genename, n, fill= "") %>% 
  arrange(pop) %>% 
  ungroup() %>%  
  right_join(., panel %>% select(pop, super_pop) %>% distinct() %>% filter(!pop %in% c('WPN','EPN','NAD')), by = 'pop', fill = '') %>% arrange(super_pop) %>% 
   select('Population' = pop, everything(), -super_pop) %>% mutate_all(funs(replace(., is.na(.), ""))) %>% 
  kable(., booktabs = TRUE, caption  = 'Number of SNPs with significant iHS in urate and gout associated genes') %>%
  kableExtra::row_spec(0, angle = 45) %>%
  kableExtra::group_rows("AFR", 1, 7) %>%
  kableExtra::group_rows("AMR", 8, 11) %>%
  kableExtra::group_rows("EAS", 12, 16) %>% kableExtra::group_rows("EUR", 17, 22) %>%
  kableExtra::group_rows("POL", 23, 26) %>%
  kableExtra::group_rows("SAS", 27,31) %>% kable_styling(latex_options = c("scale_down")) %>% footnote(., 'XP-EHH is number of populations with significant markers.', general_title = ""))
```

```{r, eval = FALSE}
# gout specific
# number of nsl snps by gout gene by poplation
sig_nsl_val %>% 
  filter(genename %in% urate_goutGR$SYMBOL) %>% 
  group_by(pop, genename) %>% tally() %>%
  arrange(genename) %>% 
  spread(genename, n, fill= "") %>% 
  arrange(pop) %>% 
  ungroup() %>%  
  right_join(., panel %>% select(pop, super_pop) %>% distinct() %>% filter(!pop %in% c('WPN','EPN','NAD')), by = 'pop', fill = '') %>% arrange(super_pop) %>% 
   select('Population' = pop, everything(), -super_pop) %>% mutate_all(funs(replace(., is.na(.), ""))) %>% 
  kable(., booktabs = TRUE, caption  = 'Number of SNPs with significant nSL in urate and gout associated genes') %>%
  kableExtra::row_spec(0, angle = 45) %>%
  kableExtra::group_rows("AFR", 1, 7) %>%
  kableExtra::group_rows("AMR", 8, 11) %>%
  kableExtra::group_rows("EAS", 12, 16) %>% kableExtra::group_rows("EUR", 17, 22) %>%
  kableExtra::group_rows("POL", 23, 26) %>%
  kableExtra::group_rows("SAS", 27,31) %>% kable_styling(latex_options = c("scale_down"))
```




<!-- xpehh -->

<!-- - IGF1R had evidence in all Pol pops -->
<!-- - NRXN2 only had evidence between SAM and TON -->
<!-- - PRKAG2 was ep vs afr -->
<!-- - RREB1 was cim, nzm, and ton with all super pops except sas -->


```{r urate_fst, include = FALSE}
fst_map <- findOverlaps(pol_fstGR, urate_goutGR)

# NZM fst
pol_fst %>% slice(fst_map@from) %>% bind_cols(., data.frame(urate_goutGR[fst_map@to], stringsAsFactors = FALSE) ) %>% select(contains('chrom'), SYMBOL,contains('NZM') )%>% gather("pop","fst", 5:NCOL(.))%>% mutate(pop = sapply(pop, function(x){popswitch(x, "NZM")})) %>% mutate(pop1 = substring(pop,1,3), pop2 = substring(pop, 4, 7)) %>% left_join(., panel %>% select('pop1' = pop, super_pop) %>% distinct(), by = 'pop1') %>% distinct() %>% ggplot(., aes(x = chrom_start, y =fst, by = pop1, colour = super_pop)) + geom_line() + facet_wrap(~SYMBOL, scales = 'free') + ggtitle('NZM FST')
```

```{r, eval = FALSE}
# CIM fst
pol_fst %>% slice(fst_map@from) %>% bind_cols(., data.frame(urate_goutGR[fst_map@to], stringsAsFactors = FALSE) ) %>% select(contains('chrom'), SYMBOL,contains('CIM') )%>% gather("pop","fst", 5:NCOL(.))%>% mutate(pop = sapply(pop, function(x){popswitch(x, "CIM")})) %>% mutate(pop1 = substring(pop,1,3), pop2 = substring(pop, 4, 7)) %>% left_join(., panel %>% select('pop1' = pop, super_pop) %>% distinct(), by = 'pop1') %>% distinct() %>% ggplot(., aes(x = chrom_start, y =fst, by = pop1, colour = super_pop)) + geom_line() + facet_wrap(~SYMBOL, scales = 'free') + ggtitle('CIM FST')
```

```{r, eval = FALSE}
# SAM fst
pol_fst %>% slice(fst_map@from) %>% bind_cols(., data.frame(urate_goutGR[fst_map@to], stringsAsFactors = FALSE) ) %>% select(contains('chrom'), SYMBOL,contains('SAM') )%>% gather("pop","fst", 5:NCOL(.))%>% mutate(pop = sapply(pop, function(x){popswitch(x, "SAM")})) %>% mutate(pop1 = substring(pop,1,3), pop2 = substring(pop, 4, 7)) %>% left_join(., panel %>% select('pop1' = pop, super_pop) %>% distinct(), by = 'pop1') %>% distinct() %>% ggplot(., aes(x = chrom_start, y =fst, by = pop1, colour = super_pop)) + geom_line() + facet_wrap(~SYMBOL, scales = 'free') + ggtitle('SAM FST')
```

```{r, eval = FALSE}
# TON fst
pol_fst %>% slice(fst_map@from) %>% bind_cols(., data.frame(urate_goutGR[fst_map@to], stringsAsFactors = FALSE) ) %>% select(contains('chrom'), SYMBOL,contains('TON') )%>% gather("pop","fst", 5:NCOL(.))%>% mutate(pop = sapply(pop, function(x){popswitch(x, "TON")})) %>% mutate(pop1 = substring(pop,1,3), pop2 = substring(pop, 4, 7)) %>% left_join(., panel %>% select('pop1' = pop, super_pop) %>% distinct(), by = 'pop1') %>% distinct() %>% ggplot(., aes(x = chrom_start, y =fst, by = pop1, colour = super_pop)) + geom_line() + facet_wrap(~SYMBOL, scales = 'free') + ggtitle('TON FST')
```




```{r, eval = FALSE}
bind_rows(sig_ihs_val, sig_nsl_val) %>% filter(genename %in% urate_goutGR$SYMBOL) %>% left_join(., markers) %>% select(-ensgeneid) %>% arrange(genename, pop, marker) %>% View()
```



#### Obesity

```{r obesityPoldf}
g <- obesity_GR$SYMBOL
obesity_remove <- c("ABCA1","ADAMTS17","AGBL4" ,"APOA5","CADM2", "CCDC92","CDKAL1", "CALCRL","DGKG","DNM3","EFEMP1","EYA2","FCER1A","FNDC3B","FOXO3","FTO","GBE1","GP2","HSD17B12","IQCK","KCNMA1","KCTD15","KREMEN1","LIN28B","LYPLAL1","LINGO2","NAV1","NCAM2","NFE2L3","OR10J1","PARK2","PCSK1","PGPEP1","RASA2","RMST","SMAD6","STXBP6","USP37","ZNF664")

obesityPol <- xpehh %>%  select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% filter(pop1 %in% c("CIM","NZM","SAM","TON"), xpehh_value > 0) %>% filter(genename %in% g) %>% select(pop1,pop2, genename) %>% distinct() %>% left_join(., panel %>% select(pop2 = pop, super_pop) %>% distinct(), by = 'pop2') %>% group_by(pop1, super_pop, genename) %>% tally() %>% spread(super_pop, n, 0) %>% rename(pop = pop1) %>% full_join(., sig_ihs_val %>% filter(!is.na(genename), genename != "", genename %in% g, pop %in% poly_pop) %>% select(pop, genename) %>% group_by(pop,genename) %>% tally(), by = c('pop','genename')) %>% rename(ihs = n) %>% full_join(., sig_nsl_val %>% filter(!is.na(genename), genename != "", genename %in% g, pop %in% poly_pop) %>% select(pop, genename) %>% group_by(pop,genename) %>% tally(), by = c('pop','genename')) %>% rename(nsl = n) %>% left_join(., lower_sig_stats %>% select(-contains('chrom')) %>% group_by(pop, SYMBOL, statname) %>% tally() %>% spread(statname, n), by = c('pop', 'genename' = 'SYMBOL') ) %>% arrange(genename) %>% mutate_all(., function(x){replace(x, is.na(x), 0)}) %>% 
  filter(!genename %in% obesity_remove) 
rm(g)
```

```{r}
haplo_obesity <- bind_rows(sig_ihs_val, sig_nsl_val) %>% select(SYMBOL = genename, pop) %>% filter(SYMBOL %in% obesity_GR$SYMBOL, pop %in% poly_pop) %>% distinct()
```

<!--consec regions -->
<!-- - td -->
<!--     - chr16  67145002  67465001 320000      * KCTD19        - NZM Tajima.D -->
<!--     - chr3  46035002  46295001 260000      *   CCR3        + TON Tajima.D -->
<!-- - fld -->
<!--     - chr3  52375002  52615001 240000      * NT5DC2        - CIM  Fu.Li.D -->
<!--     - chr3  52375002  52615001 240000      *  PBRM1        - CIM  Fu.Li.D -->
<!--     - chr16   3545002   3795001 250000      *  NLRC3        - CIM  Fu.Li.D -->
<!-- - flf -->
<!--     - chr12 124195002 124505001 310000      * DNAH10        + CIM  Fu.Li.F -->
<!--     - chr12 124195002 124505001 310000      * ZNF664        + CIM  Fu.Li.F -->
<!--     - chr12 124195002 124505001 310000      * CCDC92        - CIM  Fu.Li.F -->
<!--     - chr16   3545002   3795001 250000      *  NLRC3        - CIM  Fu.Li.F -->



<!-- ihs ABO 2, ACAN 1, ADAMTS9 1, ADCY9 7, CCNJL 3, ERBB4 1, EYA2 1, FCER1A 3, GDF5 1, LY86 1, MSRA 1, MTIF3 1, NAV1 1, STXBP6 1 -->
<!-- nsl ADAMTS9 3, ADCY9 6, ARL15 2, CCNJL 3, GDF5 2, KREMEN1 1, MSRA 2, NAV1 1 -->

<!-- NZM  -->
<!-- td CCR3 and KCTD19 -->
<!-- fld ITIH4, LIN28B, NLRC3, GDF5 -->
<!-- flf LIN28B, NLRC3, KCTC19, GDF5 -->
<!-- ze AGBL4, SRPK2 -->
<!-- ihs ABO, ADAMTS9, ADCY9 2, APOA5, ARL15 2, BTNL2 5, CCNJL, CDKAL1, ERBB4,  FHIT 4, FOXO3, FTO, LY86 2, MAP2K5, MTIF3 3, NFE2L3, RABEP1 -->
<!-- nsl ACAN, ADMATS9 3, ADCY9, AGBL4, CCNJL, FHIT, FOXO3, KCTD15, PCSK5, RABEP1 -->

<!-- SAM -->
<!-- td ERBB4, CCR3, DNAH10 -->
<!-- fwh FER, MC4R -->
<!-- fld LIN28B, TLR4, APOA5, KCTD19 -->
<!-- flf LIN28B, TLR4, KCTD19 -->
<!-- ze AGBL4, ITGB6 -->
<!-- ihs ABO, BDNF, BTNL2, ERBB4, EYA2 2, FCER1A, GPRC5B, HSD17B12 2, IQCK, LRP1B 7, PARK2, SLC39A8, TRIP11 -->
<!-- nsl BDNF, CADM2 2, EFEMP1, ERBB4, FOXO3, GBE1, GRPC5B, GRID1, KREMEN1, LRP1B 7, PARK2, PCSK1, SLC39A8 4 -->

<!-- TON -->
<!-- td ERBB4, CCR3, DNAH10 -->
<!-- fwh ERBB4, ZBTB38, MC4R -->
<!-- fld CCR3, HMGCR, APOA5 -->
<!-- flf TMEM18, ERBB4, CCR3 -->
<!-- ihs ABO, APOA5, BDNF, FCER1A, FHIT, FPGT-TNNI3K, GP2, GPRC5B, ITGB6, LEPR, LINGO2, LRP1B 4, LY86, MSRA, PARK2, STXBP6, TNNI3K, TRIP11 3 -->
<!-- nsl ADAMTS17, BDNF, FHIT, FPGT-TNNI3K, GRID1 2, ITGB6, JAZF1, LRP1B 4, MAP2K5, NRXN3, PGPEP1, SLC39A8 2, TNNI3K -->

<!-- Lower sig -->


<!-- CIM -->
<!-- DNAH10 all but ze -->

<!-- NZM -->
<!-- DNAH10 all 5 stats -->

```{r, eval = FALSE}

c("PTPRD", "PRKAG2","NRXN3","MAP2K5","HSD17B12","GRID1","FHIT","ERBB4","BDNF","ARL15","APOA5","AGBL4") %in% urate_goutGR$SYMBOL

c("PTPRD", "PRKAG2","NRXN3","MAP2K5","HSD17B12","GRID1","FHIT","ERBB4","BDNF","ARL15","APOA5","AGBL4") %in% obesity_GR$SYMBOL

c("PTPRD", "PRKAG2","NRXN3","MAP2K5","HSD17B12","GRID1","FHIT","ERBB4","BDNF","ARL15","APOA5","AGBL4") %in% t2d_GR$SYMBOL
```


```{r, eval = FALSE}
c("MTIF3", "ADAMTS9", "ADCY9", "CCNJL", "DNAH10", "PCSK5", "ACAN", "BTNL2", "LY86", "ABO", "ARL15", "CCR3", "COL6A1", "ERBB4") #Eastern Polynesian populations: 
c("JAZF1", "PEPD", "RABEP1", "LRP1B", "BDNF", "FAM13A", "FER", "GPRC5B", "GRID1", "SLC39A8", "ABO", "ARL15", "CCR3", "COL6A1", "ERBB4") # Western Polynesian population: 
c("ABO", "ARL15", "CCR3", "COL6A1", "ERBB4") # Poly: 

```




There were `r length(unique(obesity_GR$SYMBOL))` obesity-associated loci with three loci in common with urate and gout from the \gls{gwas} catalog (see Table \@ref(tab:gwascatref) for references). The Polynesian populations had about 10% of the obesity-associated genes showing some evidence of possible selection (Table \@ref(tab:superDiseaseGenesTab)). As a super population, the Polynesian populations had fewer obesity-associated genes that were significant for \gls{ihs} or \gls{nsl} compared to the other super populations. Genes that were associated with obesity and had evidence across multiple populations or many significant results are included in Table \@ref(tab:obesityPol). Five loci had evidence in all four Polynesian populations, from a combination of haplotypic and \gls{sfs} statistics, these were _ABO_, _ARL15_, _CCR3_, _COL6A1_, and _ERBB4_. _ABO_ is discussed further in subsection \@ref(malariaRes) in the context of malaria. There were nine obesity associated loci that of the Polynesian populations, the Eastern Polynesian populations had either the entire or the largest signal, these were _MTIF3_, _ADAMTS9_, _ADCY9_, _CCNJL_, _DNAH10_, _PCSK5_, _ACAN_, _BTNL2_, _LY86_. Conversely, ten loci had their entire or largest signal in the Western Polynesian populations, these were _JAZF1_, _PEPD_, _RABEP1_, _LRP1B_, _BDNF_, _FAM13A_, _FER_, _GPRC5B_, _GRID1_, and _SLC39A8_. Four of the loci, _ADAMTS9_, _ARL15_, _JAZF1_, and _PEPD_ were also associated with \gls{t2d}. _FTO_ did have a single marker (rs4396532) that was significant for \gls{ihs} in \gls{nzm}.

Consecutively depressed score regions were in \gls{cim} at chr3:52375002-52615001 and intersected _NT5DC2_ and _PBRM1_ for \gls{fld} in \gls{cim}. Another region was at chr12:124195002-124505001 for \gls{flf} that covered _DNAH10_, _ZNF664_, and _CCDC92._ Both \gls{cim} and \gls{nzm} had a region that intersected _NLRC3_. The \gls{cim} region was for \gls{flf} at  chr16:3545002-3795001, whereas the \gls{nzm} region was slightly smaller and for \gls{fld} at chr16:3555002-3775001. For \gls{td} there were two regions of consecutive score depression that intersected obesity-associated loci, the first in \gls{nzm} at chr16:67145002-67465001 intersected _KCTD19._ The second was in \gls{ton} covered _CCR3_ at chr3:46035002-46295001.

```{r obesityPolTab}
obesityPol %>% 
      mutate_all(., function(x){replace(x, x == 0, "")}) %>% arrange(genename,pop) %>% 
  kable(., caption = '(\\#tab:obesityPol) Obesity-associated loci with multiple signs of possible selection in Polynesian populations.', longtable = TRUE, format = 'latex', col.names = c('Population','Gene','AFR','AMR','EAS','EUR','POL','SAS','iHS','nSL',"Fay & Wu's H", "Fu & Li's D"," Fu & Li's F", "Tajima's D"," Zeng's E"), booktabs = TRUE) %>% 
  kableExtra::row_spec(0, angle = 90) %>% 
  add_header_above(., c("","","XP-EHH" = 6, rep("",7))) %>% 
  kable_styling(latex_options = "repeat_header") %>% 
  kableExtra::footnote(general = c("XP-EHH is the number of populations from the super population that had at", 
                                   "least one marker sigificant in the gene. Integrated haplotype homozygosity",
                                   "score and nSL are the number of significant markers. Fay and Wu's H, Fu and",
                                   "Li's D, Fu and Li's F, Tajima's D, and Zeng's E are the number of windows",
                                   " intersecting the gene that met the lower threshold."), general_title = "", escape = TRUE)# %>% filter((AFR > 0 | AMR >0 | EAS > 0 | ihs > 0 | nsl > 0) & (!is.na(Fay.Wu.H) | !is.na(Fu.Li.D)| !is.na(Fu.Li.F) | !is.na(Tajima.D) | !is.na(Zeng.E)))
```


<!-- BTNL2 had many markers that were significant in \gls{chb} and \gls{jpt}. -->
<!-- > obesity -->


<!-- There were four genes that had markers significant in three of the four Polynesian populations, and only for \gls{ihs}, these were: _BTNL2_, _ERBB4_, _FCER1A_, and _LY86_. There were two genes, _ADAMTS9_ and _ADCY9_, that had significant markers for both \gls{ihs} and \gls{nsl} in the Eastern Polynesian populations of \gls{cim} and \gls{nzm}. _ADAMTS9_ had a single marker \gls{ihs} and \gls{nsl}, three markers for \gls{nsl}. _ADCY9_ had seven significant markers for \gls{ihs} with \gls{cim} and two for \gls{nzm}. \gls{nsl} had 6 markers for \gls{cim} and 1 for \gls{nzm}. _ARL15_ and _MTIF3_ only had markers that were significant with \gls{ihs}, and _CCNJL_ only had markers that were significant with \gls{nsl}. _ADAMTS9_ and _ARL15_ are also associated with \gls{t2d}. The Western Polynesian populations of \gls{sam} and \gls{ton} had the genes _BDNF_ and _LRP1B_ that had significant markers in both \gls{ihs} and \gls{nsl}. _GPRC5B_, _PARK2_, and _TRIP11_ only had markers that were significant for \gls{ihs}. _GRID1_ and _SLC39A8_ only had markers that were significant with \gls{nsl}. -->

<!-- From the obesity associated genes,  _ERBB4_ had the most number of statistics, across the most Polynesian populations, that met the significance threshold. _ERRB4_ had evidence in all Polynesian populations (Figure \@ref(fig:intraSelGenes)) with three of four having evidence from both haplotypic and frequency spectrum methods. Only \gls{ton} did not have evidence from the haplotypic based methods. The Eastern Polynesian populations had the same \gls{snp} (rs1025753) that was significant, whereas \gls{sam} shared the \gls{snp} (rs6707285) with \gls{khv}. All Polynesian populations had windows that met significance for \gls{td} and \gls{fwh}. The Western Polynesian populations had consecutively depressed score regions for \gls{td}. \gls{cim} and \gls{ton} had consecutively depressed regions for \gls{fwh} and only \gls{ton} for \gls{flf}. The only other populations that showed multiple windows with multiple statistics were the \gls{eas} populations, with the exception of \gls{chs}. -->

<!-- _FHIT_ and _GRID_ both had multiple windows for \gls{fwh} that were in the 1st percentile. _FHIT_ also had \gls{snp}s in both \gls{nzm} and \gls{ton} that were significant for \gls{ihs} and \gls{nsl}. \gls{ton} and \gls{nzm} both had five windows in the 1st percentile for \gls{fwh} that intersected _FHIT_. _GRID_ had markers that were significant in only \gls{nsl} and only in the Western Polynesian populations. Both \gls{sam} and\gls{ton} had windows in the 1st percentile for \gls{fwh} that intersected with _GRID1_. -->

<!-- For all Polynesian populations there were multiple windows that intersected _DNAH10_. There were regions of consecutive depressed \gls{td} score for \gls{sam}, \gls{ton}, and \gls{cim} that intersected _DNAH10_. \gls{cim} also had consecutively depressed score regions for \gls{fld} and \gls{flf} at this locus. The main region was actually the upstream region ranging from _TCTN2_ to _DNAH10_. _ATP6V0A2_ was in the middle and where this signal was focussed. For \gls{nzm} _DNAH10_ had significant windows for each of \gls{td}, \gls{fwh}, \gls{fld}, \gls{flf}, and \gls{ze}. \gls{cim} was the same but without \gls{ze}. \gls{cim} had one marker each that was significant for \gls{xpehh} with \gls{acb} and \gls{lwk}, and 11 markers with \gls{gwd}. \gls{nzm} had three markers that were significant for \gls{xpehh} with \gls{gwd}. The Western Polynesian populations did not have the multiple statistic signal.  -->




<!-- - Only CIM had consecutively depressed \gls{td} score that intersected _CCR3_, a which encodes a chemokine receptor. -->

<!-- - The \gls{sam} population had windows from \gls{td}, \gls{fld}, \gls{flf}, and \gls{ze} that were significant and intersected _CREB1_. The other Polynesian populations only had windows in the 1st percentile for \gls{td} and \gls{ze}. -->

<!-- - In \gls{ton}, there were significant windows that intersected with _NRXN3_ for \gls{td}, \gls{fwh}, \gls{fld}, and \gls{flf}.  -->

<!-- AGBL4 had 17 windows for \gls{ze} that were in the 1st percentile in \gls{nzm} and had a significant marker for \gls{nsl} -->

<!-- consec regions for td in sam and ton -->
<!-- consec regions for fwh in cim and ton -->
<!-- consec regoins in flf in ton -->

<!-- cim - td and fwh -->
<!-- nzm - td and fwh -->
<!-- sam - td and fwh -->
<!-- ton - td, flf, and td -->

<!-- ihs in cim, nzm, and sam -->
<!-- nsl in sam -->


<!-- eas, except chs had multiple windows for fwh, td, flf -->

<!-- The Eastern Polynesian populations of \gls{cim} and \gls{nzm} had significant results for _ADCY9_. \gls{cim} had 7 markers that were significant for \gls{ihs} and 6 markers for \gls{nsl}. \gls{nzm} had 2 markers for \gls{ihs} and a single marker for \gls{nsl}. The Western Polynesian populations on the other hand had  -->

```{r, eval = FALSE}
lower_sig_stats %>% filter(pop %in% poly_pop, SYMBOL == "ERBB4") %>% select(SYMBOL, pop) %>% group_by(SYMBOL, pop) %>% tally() %>% spread(pop, n)
lower_sig_stats_binned %>% filter(pop %in% poly_pop, chrom_end >= 124125002 & chrom_start <=124265001,  chrom == 'chr12') %>% ggplot(., aes(x = chrom_start, y = value)) + facet_wrap(statname~pop) + geom_line() + geom_point() + theme_bw()
consec_lower_regions %>% filter(pop == 'NZM', SYMBOL %in% obesity_GR$SYMBOL) 

sig_nsl_val %>% select(SYMBOL = genename, everything(), pop) %>% filter(SYMBOL %in% obesity_GR$SYMBOL, pop == "TON") %>% group_by(SYMBOL) %>% tally()
```


```{r, eval = FALSE}
g <- c("ABO", "BTNL2", "ERBB4", "FCER1A", "LY86","BDNF", "GPRC5B", "LRP1B", "PARK2", "TRIP11", "BDNF", "GRID1", "LRP1B", "SLC39A8", "ADAMTS9", "ADCY9", "ARL15", "MTIF3", "ADAMTS9", "ADCY9", "CCNJL" )
unique(g[g %in% unique(obesity_GR$SYMBOL[obesity_GR$SYMBOL %in% t2d_GR$SYMBOL])])
```

```{r, eval = FALSE}
xpehh %>%  select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% filter(pop1 %in% c("CIM","NZM","SAM","TON"), xpehh_value > 0) %>% filter(genename %in% obesity_GR$SYMBOL) %>% select(pop1,pop2, genename) %>% distinct() %>% left_join(., panel %>% select(pop2 = pop, super_pop) %>% distinct(), by = 'pop2') %>% group_by(pop1, super_pop, genename) %>% tally() %>% spread(super_pop, n, "")%>% arrange(genename) %>% kable(., caption = 'Number of populations by super population, with significant xpehh for urate/gout associated genes', booktabs = TRUE)
```

<!-- XPEHH -->

<!-- - ADCY9 - ep pops, mostly cim -->
<!-- - ARL15 - all pol and with all super pops except eas was minimal -->
<!-- - CCR3 - all pol with nearly all pops except those from eas -->
<!-- - COL6A1 - all except eas, min sas -->
<!-- - ERBB4 - all pol with all except eas pops -->
<!-- - FAM13A - wp against afr and eur -->
<!-- - JAZF1 - wp against all except afr -->
<!-- - LEKR1 - all against afr -->
<!-- - PCSK5 - ep against all except afr -->
<!-- - PEPD - wp against all except sas -->


<!-- All -->
<!-- ihs: ABO -->
<!-- nsl:  -->

<!-- 3/4 -->
<!-- ihs: BTNL2, ERBB4, FCER1A, LY86,   -->

<!-- Western Polynesians -->
<!-- ihs: BDNF, GPRC5B, LRP1B, PARK2, TRIP11 -->
<!-- nsl: BDNF, GRID1, LRP1B, SLC39A8 -->

<!-- Eastern Polynesians -->
<!-- ihs: ADAMTS9, ADCY9, ARL15, MTIF3,  -->
<!-- nsl: ADAMTS9, ADCY9, CCNJL,  -->

<!-- TD:  -->
<!-- all pops ARL15, CCR3, CREB1, DNAH10, ERBB4,  -->

<!-- FWH all pops: ERBB4,FHIT, GRID1, LTBP1, ZBTB38 -->
<!-- ep: DNAH10 -->

<!-- FLD: no all pops -->
<!-- FLF: no all pops -->

```{r fst_obesity, eval = FALSE}
fst_map_obesity <- findOverlaps(pol_fstGR, obesity_GR)

# NZM fst
pol_fst %>% slice(fst_map_obesity@from) %>% bind_cols(., data.frame(obesity_GR[fst_map_obesity@to], stringsAsFactors = FALSE) ) %>% filter(SYMBOL == 'ADCY9')%>% select(contains('chrom'), SYMBOL,contains('NZM') )%>% gather("pop","fst", 5:NCOL(.))%>% mutate(pop = popswitch(pop, "NZM")) %>% mutate(pop1 = substring(pop,1,3), pop2 = substring(pop, 4, 7)) %>% left_join(., panel %>% select('pop1' = pop, super_pop) %>% distinct(), by = 'pop1') %>% distinct() %>% ggplot(., aes(x = chrom_start, y =fst, by = pop1, colour = super_pop)) + geom_line() + facet_wrap(~SYMBOL, scales = 'free') + ggtitle('NZM FST')

# CIM fst
pol_fst %>% 
  slice(fst_map_obesity@from) %>% 
  bind_cols(., data.frame(obesity_GR[fst_map_obesity@to], stringsAsFactors = FALSE) ) %>% 
  filter(SYMBOL == 'ADCY9') %>% 
  select(contains('chrom'), SYMBOL,contains('CIM') )%>% 
  gather("pop","fst", 5:NCOL(.))%>% 
  mutate(pop = sapply(pop, function(p){popswitch(p, "CIM")})) %>% 
  mutate(pop1 = substring(pop,1,3), pop2 = substring(pop, 4, 7)) %>% 
  left_join(., panel %>% select('pop1' = pop, super_pop) %>% 
              distinct(), by = 'pop1') %>% 
  distinct() %>% 
  ggplot(., aes(x = chrom_start, y =fst, by = pop1, colour = super_pop)) + geom_line() + facet_wrap(~SYMBOL, scales = 'free') + ggtitle('CIM FST')
```

#### Type 2 diabetes

```{r}
g <- t2d_GR$SYMBOL
t2d_remove <- c("ADCY5","AP3S2","ATP8B2","CAMK1D","CDC123","CDKAL1","CDKN2A","CDKN2B","FAM60A","FTO","GLIS3","GRK5","IDE","NS-IGF2","ITGB6","LAMA1","PPP2R2C","RASGRP1","ST6GAL1","TP53INP1","TSPAN8","WFS1","ZFAND6", "CCDC85A", "INS-IGF2")#, "TNKS")
t2dPol <- xpehh %>%  select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% filter(pop1 %in% c("CIM","NZM","SAM","TON"), xpehh_value > 0) %>% filter(genename %in% g) %>% select(pop1,pop2, genename) %>% distinct() %>% left_join(., panel %>% select(pop2 = pop, super_pop) %>% distinct(), by = 'pop2') %>% group_by(pop1, super_pop, genename) %>% tally() %>% spread(super_pop, n, 0) %>% rename(pop = pop1) %>% full_join(., sig_ihs_val %>% filter(!is.na(genename), genename != "", genename %in% g, pop %in% poly_pop) %>% select(pop, genename) %>% group_by(pop,genename) %>% tally(), by = c('pop','genename')) %>% rename(ihs = n) %>% full_join(., sig_nsl_val %>% filter(!is.na(genename), genename != "", genename %in% g, pop %in% poly_pop) %>% select(pop, genename) %>% group_by(pop,genename) %>% tally(), by = c('pop','genename')) %>% rename(nsl = n) %>% left_join(., lower_sig_stats %>% select(-contains('chrom')) %>% group_by(pop, SYMBOL, statname) %>% tally() %>% spread(statname, n), by = c('pop', 'genename' = 'SYMBOL') ) %>% arrange(genename) %>% mutate_all(., function(x){replace(x, is.na(x), 0)}) %>% filter(!genename %in% t2d_remove) 

rm(g)
```

```{r}
haplo_t2d <- bind_rows(sig_ihs_val, sig_nsl_val) %>% select(SYMBOL = genename, pop) %>% filter(SYMBOL %in% t2d_GR$SYMBOL, pop %in% poly_pop) %>% distinct()
```


```{r, eval = FALSE}
c("ARL15", "LPP", "PTPRD", "SND1", "THADA")# poly
c("ADAMTS9", "PSMD6", "SSR1")# eastern
c("BCL11A", "JAZF1", "KCNJ11", "PEPD")# western
```

\Gls{t2d} is a co-morbidity of hyperuricemia, and had `r length(unique(t2d_GR$SYMBOL))` genes associated from \gls{gwas} in the \gls{gwas} catalog (see Table \@ref(tab:gwascatref) for references), with one gene in common with urate-associated loci, and 16 genes in common with obesity. For the Polynesian populations there were `r lower_sig_stats %>% filter(SYMBOL %in% t2d_GR$SYMBOL, pop  %in% poly_pop) %>% select(SYMBOL) %>% distinct() %>% NROW()` loci that were intersected by windows in the 1st percentile of a frequency based statistic. There were `r bind_rows(sig_ihs_val, sig_nsl_val) %>% select(SYMBOL = genename, pop) %>% filter(SYMBOL %in% t2d_GR$SYMBOL, pop  %in% poly_pop) %>% select(SYMBOL) %>% distinct() %>% NROW()` genes that had a marker with a significant value for \gls{ihs} or \gls{nsl}. The genes that showed evidence of possible selection across multiple Polynesian populations and statistics are shown in Table \@ref(tab:t2dPol). There were five genes that showed evidence across all Polynesian populations, these were _ARL15_, _LPP_, _PTPRD_, _SND1_, and _THADA._ Three genes had evidence that was mostly in the Eastern Polynesian populations, these were _ADAMTS9_, _PSMD6_, _SSR1_. And the genes that mostly had evidence in the Western Polynesians were _BCL11A_, _JAZF1_, _KCNJ11_, and _PEPD_. As previously mentioned _ADAMTS9_, _ARL15_, _JAZF1_, and _PEPD_ were also associated with obesity.


There were no regions of consecutive score depression for any of the \gls{sfs} based intra-population statistics for any of the Polynesian populations.

```{r, t2dPolTab}
t2dPol %>%  mutate_all(., function(x){replace(x, x == 0, "")}) %>% 
  kable(., caption = '(\\#tab:t2dPol) T2D-associated loci with multiple signs of possible selection in Polynesian populations.', longtable = TRUE, format = 'latex', booktabs = TRUE, col.names = c('Population','Gene','AFR','AMR','EAS','EUR','POL','SAS','iHS','nSL',"Fay & Wu's H", "Fu & Li's D"," Fu & Li's F", "Tajima's D"," Zeng's E")) %>% 
  kableExtra::row_spec(0, angle = 90) %>%  
  add_header_above(., c("","","XP-EHH" = 6, rep("",7)))%>% 
  kable_styling(latex_options = "repeat_header") %>% 
  add_footnote(., notation = "number", label = c("XP-EHH is the number of populations from the super population that had at least one marker sigificant in the gene. \\glsdesc{ihs} and nSL are the number of significant markers. \\gls{fwh}, \\gls{fld}, \\gls{flf}, \\gls{td}, and \\gls{ze} are the number of windows intersecting the gene that met the lower threshold.")) # %>% filter((AFR > 0 | AMR >0 | EAS > 0 | ihs > 0 | nsl > 0) & (!is.na(Fay.Wu.H) | !is.na(Fu.Li.D)| !is.na(Fu.Li.F) | !is.na(Tajima.D) | !is.na(Zeng.E)))
```



<!-- The Western Polynesian populations had the fewest significant markers in loci associated with \gls{t2d}. \gls{sam} had the minimum, out of all populations, with only 4 \gls{snp}s (Table \@ref(tab:intraSelPol)). There were no genes associated with \gls{t2d} that had haplotypic evidence in both of the Western Polynesian populations. -->

<!-- In the Eastern Polynesian populations, as mentioned there was _ADAMTS9_ and _ARL15_ also associated with obesity, but there was also _LPP_, _PSMD6_, _ZFAND6_ that had markers that were significant for both \gls{ihs} and \gls{nsl}. _LPP_ had three markers for \gls{ihs} that were significant in \gls{cim}, and six that were significant in \gls{nzm}. For \gls{nsl} there was a single marker that was significant in \gls{cim}, \gls{nzm} and also \gls{ton}  -->



<!-- _PTPRD_ was the only gene associated with \gls{t2d} that had significant windows and \gls{snp}s in both frequency spectrum and haplotypic methods. All four Polynesian populations had significant \gls{snp}s for \gls{ihs} and all except \gls{nzm} had significant markers for \gls{nsl}. For \gls{ihs}, there were 14 markers that were significant in \gls{cim}, five in \gls{ton}, three in \gls{sam}, and a single marker in \gls{nzm}. With \gls{nsl}, \gls{cim} and \gls{ton} had four markers that were significant and there were three in \gls{sam}. \gls{ton} and \gls{cim} had three windows for \gls{td} that were in the 1st percentile. \gls{nzm} had a single window for both \gls{fwh} and \gls{td} that was in the 1st percentile and intersected _PTPRD_. -->


<!-- _RREB1_ is shared with urate and as previously mentioned had significant windows for \gls{td}, \gls{fld}, and \gls{flf}. -->

<!-- Loci that were in common with obesity included _ARL15_ which had significant windows for all Polynesian populations for \gls{td} and both rs1712139 and rs26775 had significant \gls{ihs} values in the Eastern Polynesian populations. The same markers also had significant \gls{nsl} in \gls{cim}.  -->


<!-- _FTO_ had a significant window for \gls{fld} in the \gls{ton} population and \gls{nzm} had rs4396532 that was significant for \gls{ihs}. _HMGA2_ had significant windows with \gls{fwh} for \gls{cim}, \gls{nzm} and \gls{sam}. _MC4R_ had significant windows for \gls{fwh} in the Western Polynesian populations. The \gls{ton} population had significant windows for \gls{fwh} that intersected _TCF7L2_. And \gls{nzm} had significant windows for \gls{ze} that intersected _GRB14_. -->

<!-- Loci that only had significant windows in the frequency based statistics included _THADA_, all Polynesian populations had significant windows for \gls{fwh}. \gls{cim}, \gls{nzm}, and \gls{ton} also had windows for \gls{ze}. Both \gls{cim} and \gls{ton} had significant windows in \gls{fld} and \gls{flf}. \gls{ton} was the only Polynesian population to also have  -->
<!-- There were regions of consecutively depressed score for \gls{fwh} with \gls{cim} and \gls{ton}. \gls{ton} also had consecutively depressed score regions for \gls{td}. -->



<!-- - ARL15, SSR1 for cim and nzm with 2 markers each for ihs, cim had 2 markers for ARL15 for nsl, both had 2 for nsl in SSR1 -->

<!-- - MACF1 had windows that were in the 1st percentile for all Polynesian populations for \gls{ze}. \gls{nzm}, \gls{sam}, and \gls{ton} also had windows for \gls{fld} and \gls{flf}. \gls{sam} was the only Polynesian population to have windows in the 1st percentile for \gls{td}. -->


```{r, eval = FALSE}
xpehh %>%  select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% filter(pop1 %in% c("CIM","NZM","SAM","TON"), xpehh_value > 0) %>% filter(genename %in% t2d_GR$SYMBOL) %>% select(pop1,pop2, genename) %>% distinct() %>% left_join(., panel %>% select(pop2 = pop, super_pop) %>% distinct(), by = 'pop2') %>% group_by(pop1, super_pop, genename) %>% tally() %>% spread(super_pop, n, "")%>% arrange(genename) %>% kable(., caption = 'Number of populations by super population, with significant xpehh for urate/gout associated genes', booktabs = TRUE)
```

<!-- XPEHH interesting genes -->

<!-- - ARL15 - as per obesity -->
<!-- - JAZF1 - as per obesity -->
<!-- - LPP - all pol, except eas -->
<!-- - PEPD - as pe obesity -->
<!-- - PTPRD - not sam, against all -->
<!-- - RREB1 - as per urate -->
<!-- - SND1 - wp vs afr, amr, eur, sas. ep vs 1 eur, 1 sas -->
<!-- - SSR - ep vs all except sas -->
<!-- - THADA - all against all except eas -->
<!-- - ZMIZ1 - all against eas and sas -->

<!-- all -->
<!-- ihs: PTPRD -->
<!-- nsl none -->

<!-- 3 -->
<!-- ihs: none -->
<!-- nsl: PTPRD -->

<!-- west -->
<!-- ihs: none -->
<!-- nsl: none -->

<!-- east -->
<!-- ihs: ADAMTS9, ARL15, LPP, PSMD6, SSR1, ZFAND6 -->
<!-- nsl: ADAMTS9, LPP, PSMD6, ZFAND6  -->

<!-- ARL15 - cim, nzm, sam, ton for td -->
<!-- FTO - ton fld -->
<!-- HMGA2 cim, nzm, sam fwh -->
<!-- MC4R sam ton fwh -->
<!-- TCF712 ton fwh -->
<!-- GRB14 nzm ze -->

<!-- MACF1: -->
<!-- sam, ton, nzm - flf, flf, ze -->
<!-- only td for sam -->
<!-- only ze for cim -->


<!-- THADA - ton all 5 stats -->
<!-- cim fwh, fld, flf, ze -->
<!-- nzm - fwh, ze -->
<!-- sam - fwh -->

<!-- SND1 - nzm - fld, flf, td, ze -->
<!-- cim - td -->
<!-- sam - flf, td -->






```{r, eval = FALSE}
lower_sig_stats %>% filter(SYMBOL %in% t2d_GR$SYMBOL, pop %in% poly_pop) %>% group_by(pop, SYMBOL, statname) %>% tally() %>% spread(statname, n, "") %>% arrange(SYMBOL)
```



```{r, eval = FALSE}
consec_lower_regions %>% filter(SYMBOL %in% t2d_GR$SYMBOL, pop %in% poly_pop) %>% arrange(SYMBOL, pop)
```
<!-- ALR15, FTO, GRB14, HMGA2, MC4R, TCF7L2 are all shared with obesity -->
<!-- ADAMATS9, CDKAL1, ITGB6, JAZF1 too -->

#### Kidney disease

```{r, eval = FALSE}
bind_rows(sig_ihs_val, sig_nsl_val) %>% select(SYMBOL = genename, pop, statid) %>% filter(SYMBOL %in% kd_GR$SYMBOL, pop %in% poly_pop) %>% group_by(pop, SYMBOL, statid) %>% tally()

lower_sig_stats %>% select(SYMBOL, pop, statname) %>% filter(SYMBOL %in% kd_GR$SYMBOL, pop %in% poly_pop) %>% group_by(pop, SYMBOL, statname) %>% tally() %>% spread(statname, n) %>% arrange(SYMBOL)

consec_lower_regions %>% filter(SYMBOL %in% kd_GR$SYMBOL, pop %in% poly_pop) %>% arrange(SYMBOL, pop)
```







```{r}
g <- kd_GR$SYMBOL
kd_remove <- c("DDX1","FAM63A","GP2","SLC13A3","WDR37")
kdPol <- xpehh %>%  select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% filter(pop1 %in% c("CIM","NZM","SAM","TON"), xpehh_value > 0) %>% filter(genename %in% g) %>% select(pop1,pop2, genename) %>% distinct() %>% left_join(., panel %>% select(pop2 = pop, super_pop) %>% distinct(), by = 'pop2') %>% group_by(pop1, super_pop, genename) %>% tally() %>% spread(super_pop, n, 0) %>% rename(pop = pop1) %>% full_join(., sig_ihs_val %>% filter(!is.na(genename), genename != "", genename %in% g, pop %in% poly_pop) %>% select(pop, genename) %>% group_by(pop,genename) %>% tally(), by = c('pop','genename')) %>% rename(ihs = n) %>% full_join(., sig_nsl_val %>% filter(!is.na(genename), genename != "", genename %in% g, pop %in% poly_pop) %>% select(pop, genename) %>% group_by(pop,genename) %>% tally(), by = c('pop','genename')) %>% rename(nsl = n) %>% left_join(., lower_sig_stats %>% select(-contains('chrom')) %>% group_by(pop, SYMBOL, statname) %>% tally() %>% spread(statname, n), by = c('pop', 'genename' = 'SYMBOL') ) %>% arrange(genename) %>% mutate_all(., function(x){replace(x, is.na(x), 0)}) #%>% filter(!genename %in% kd_remove)
  #filter(genename %in% psychiatric_keep) %>% 


rm(g)
```

There were `r length(unique(kd_GR$SYMBOL))` genes that were associated with kidney disease from the \gls{gwas} catalog (see Table \@ref(tab:gwascatref) for references). Four genes had markers that were significant for \gls{ihs} in the Polynesian populations. \gls{nzm} had 3 markers in _PHTF2_, \gls{sam} had one marker in _DDX1_, and \gls{ton} had one marker in each of _GP2_ and _WDR37_. None of these loci also had windows from the \gls{sfs}-based statistics that were also significant. \gls{nzm} had a single marker that was significant for \gls{nsl} in _PHTF2_. \gls{sam} had a single significant marker for \gls{nsl} in _PRKAG2_ - a urate locus mentioned previously. _WDR72_ had two markers each in \gls{ton} and \gls{sam} that were significant for \gls{nsl}. There were no regions of consecutive score depression that intersected with kidney disease associated genes.

```{r, eval = FALSE}
kdPol %>%   mutate_all(., function(x){replace(x, x == 0, "")})  %>% 
  kable(., caption = '(\\#tab:kdPol) kidney', longtable = TRUE, format = 'latex', col.names = c('Population','Gene','AFR','AMR','EAS','EUR','POL','SAS','iHS','nSL',"Fay & Wu's H", "Fu & Li's D"," Fu & Li's F", "Tajima's D"," Zeng's E")) %>% kableExtra::row_spec(0, angle = 90) %>% kable_styling(latex_options = "repeat_header")# %>% filter((AFR > 0 | AMR >0 | EAS > 0 | ihs > 0 | nsl > 0) & (!is.na(Fay.Wu.H) | !is.na(Fu.Li.D)| !is.na(Fu.Li.F) | !is.na(Tajima.D) | !is.na(Zeng.E)))
```


```{r, eval = FALSE}
xpehh %>%  select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% filter(pop1 %in% c("CIM","NZM","SAM","TON"), xpehh_value > 0) %>% filter(genename %in% kd_GR$SYMBOL) %>% select(pop1,pop2, genename) %>% distinct() %>% left_join(., panel %>% select(pop2 = pop, super_pop) %>% distinct(), by = 'pop2') %>% group_by(pop1, super_pop, genename) %>% tally() %>% spread(super_pop, n, "")%>% arrange(genename) %>% kable(., caption = 'Number of populations by super population, with significant xpehh for kidney disease associated genes', booktabs = TRUE)
```

<!-- XPEHH -->

<!-- SETDB1 - wp vs afr eas -->
<!-- ANXA9 - wp vs amr, eas, eur -->

#### Metabolic syndrome

```{r, eval = FALSE}
bind_rows(sig_ihs_val, sig_nsl_val) %>% select(SYMBOL = genename, pop, statid) %>% filter(SYMBOL %in% metsyn_GR$SYMBOL, pop %in% poly_pop) %>% group_by(pop, SYMBOL, statid) %>% tally()

lower_sig_stats %>% select(SYMBOL, pop, statname) %>% filter(SYMBOL %in% metsyn_GR$SYMBOL, pop %in% poly_pop) %>% group_by(pop, SYMBOL, statname) %>% tally() %>% spread(statname, n) %>% arrange(SYMBOL)

consec_lower_regions %>% filter(SYMBOL %in% metsyn_GR$SYMBOL, pop %in% poly_pop) %>% arrange(SYMBOL, pop)
```

```{r metsynTab}
g <- metsyn_GR$SYMBOL
metsynPol <- xpehh %>%  select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% filter(pop1 %in% c("CIM","NZM","SAM","TON"), xpehh_value > 0) %>% filter(genename %in% g) %>% select(pop1,pop2, genename) %>% distinct() %>% left_join(., panel %>% select(pop2 = pop, super_pop) %>% distinct(), by = 'pop2') %>% group_by(pop1, super_pop, genename) %>% tally() %>% spread(super_pop, n, 0) %>% rename(pop = pop1) %>% full_join(., sig_ihs_val %>% filter(!is.na(genename), genename != "", genename %in% g, pop %in% poly_pop) %>% select(pop, genename) %>% group_by(pop,genename) %>% tally(), by = c('pop','genename')) %>% rename(ihs = n) %>% full_join(., sig_nsl_val %>% filter(!is.na(genename), genename != "", genename %in% g, pop %in% poly_pop) %>% select(pop, genename) %>% group_by(pop,genename) %>% tally(), by = c('pop','genename')) %>% rename(nsl = n) %>% left_join(., lower_sig_stats %>% select(-contains('chrom')) %>% group_by(pop, SYMBOL, statname) %>% tally() %>% spread(statname, n), by = c('pop', 'genename' = 'SYMBOL') ) %>% arrange(genename) %>% mutate_all(., function(x){replace(x, is.na(x), 0)}) 
  #filter(genename %in% psychiatric_keep) %>% 

rm(g)
```

In the \gls{gwas} catalog, there were `r length(unique(metsyn_GR$SYMBOL))` genes associated with metabolic syndrome (see Table \@ref(tab:gwascatref) for references). _APOA5_, also associated with obesity, had a significant marker in each of \gls{nzm} and \gls{ton} for \gls{ihs}. \gls{sam} has a marker that was significant in both \gls{ihs} and \gls{nsl} in _DGKB_. As previously mentioned, \gls{nzm} had a significant marker for \gls{ihs} in _FTO_, which was also associated with obesity and \gls{t2d}.

In the Eastern Polynesian populations there was a region of consecutive score depression for \gls{td} that intersected _EDC4_ at chr16:67835002-68115001 for \gls{cim}, and chr16:67775002-68105001 for \gls{nzm}. \gls{nzm} also had similar regions for \gls{fld} (chr16:67785002-68045001) and \gls{flf} (chr16:67775002-68095001).



```{r, eval = FALSE}
metsynPol %>% mutate_all(., function(x){replace(x, x == 0, "")}) %>% 
  kable(., caption = '(\\#tab:metsynTab) Metabolic syndrome associated genes with signal of selection', longtable = TRUE, format = 'latex', col.names = c('Population','Gene','AFR','AMR','EAS','iHS','nSL',"Fay & Wu's H", "Fu & Li's D"," Fu & Li's F", "Tajima's D"," Zeng's E")) %>% kableExtra::row_spec(0, angle = 90) %>% kable_styling(latex_options = "repeat_header")# %>% filter((AFR > 0 | AMR >0 | EAS > 0 | ihs > 0 | nsl > 0) & (!is.na(Fay.Wu.H) | !is.na(Fu.Li.D)| !is.na(Fu.Li.F) | !is.na(Tajima.D) | !is.na(Zeng.E)))
```


```{r, eval = FALSE}
xpehh %>%  select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% filter(pop1 %in% c("CIM","NZM","SAM","TON"), xpehh_value > 0) %>% filter(genename %in% metsyn_GR$SYMBOL) %>% select(pop1,pop2, genename) %>% distinct() %>% left_join(., panel %>% select(pop2 = pop, super_pop) %>% distinct(), by = 'pop2') %>% group_by(pop1, super_pop, genename) %>% tally() %>% spread(super_pop, n, "")%>% arrange(genename) %>% kable(., caption = 'Number of populations by super population, with significant xpehh for metabolic syndrome associated genes', booktabs = TRUE)
```

<!-- no xpehh results that stood out -->



#### Neurological disorders

```{r, eval = FALSE}
(res <- bind_rows(sig_ihs_val, sig_nsl_val) %>% select(SYMBOL = genename, pop, statid) %>% filter(SYMBOL %in% neurological_GR$SYMBOL, pop %in% poly_pop) %>% group_by(pop, SYMBOL, statid) %>% tally())

lower_sig_stats %>% select(SYMBOL, pop, statname) %>% filter(SYMBOL %in% res$SYMBOL, pop %in% poly_pop) %>% group_by(pop, SYMBOL, statname) %>% tally() %>% spread(statname, n) %>% arrange(SYMBOL)

lower_sig_stats %>% select(SYMBOL, pop, statname) %>% filter(SYMBOL %in% neurological_GR$SYMBOL, pop %in% poly_pop) %>% group_by(pop, SYMBOL, statname) %>% tally() %>% spread(statname, n) %>% arrange(SYMBOL)

consec_lower_regions %>% filter(SYMBOL %in% neurological_GR$SYMBOL, pop %in% poly_pop) %>% arrange(SYMBOL, pop)
rm(res)
```

<!-- main results: -->

<!-- - CNTNAP2 ihs and xpehh along with sfs for all polynesians -->
<!-- - CERS6 all poly for xpehh nearly all other pops -->
<!-- - VPS13C for ep, ihs and fld/flf/ze -->
<!-- - SLC2A13 xpehh all poly pops vs nearly all other pops and ihs -->
<!-- - no regions of consecutive score depression -->
<!-- - DLG2 cim/sam/ton 1 xpehh pop each and all had 10-12 windows for fwh -->
Urate has associations with neurological disease, such as the potential as a biomarker for the progression of Parkinson's disease [@Wen2017]. Loci that were associated with the neurological diseases of Parkinson's disease and Alzheimer's disease from the \gls{gwas} catalog did not intersect with any regions of consecutive score depression of any \gls{sfs}-based statistic. All four Polynesian populations had significant \gls{xpehh} values with populations from the \gls{afr} and \gls{amr} super populations at the _CNTNAP2_ locus. The Western Polynesian populations also had significant \gls{xpehh} values with \gls{eur} populations. There were significant markers for \gls{ihs} and \gls{nsl} for \gls{nzm} and \gls{sam}. All Polynesians populations also had windows in the 1st percentile for \gls{fwh}, with \gls{nzm} having the most with 12. All Polynesians also had windows in the 1st percentile for \gls{td} that intersected _CNTNAP2_. Another locus, _CERS6_, had each Polynesian population with significant \gls{xpehh} markers with nearly all populations from the \gls{afr}, \gls{amr}, \gls{eur} super population groups and approximately half of the populations from both \gls{eas} and \gls{sas} super populations. A third locus that had significant \gls{xpehh} markers for all Polynesian populations was _SLC2A13_. All Polynesian populations had significant \gls{xpehh} markers with most populations from \gls{amr} and \gls{eas} super populations. \gls{nzm}, \gls{cim}, and \gls{sam} also had significant markers with nearly all populations of the \gls{sas} and \gls{eur}. Only \gls{nzm} had significant markers with the \gls{afr} populations. All Polynesian populations had significant markers for \gls{nsl} at _SLC2A13_, and also for \gls{ihs} except for \gls{cim}. _VPS13_ had both Eastern Polynesian populations with two significant \gls{ihs} markers, and windows from the 1st percentile for \gls{fld} (five windows), \gls{flf} (six windows), and \gls{ze} (nine windows for \gls{nzm} and 11 for \gls{cim}).


```{r, eval = TRUE}
# create table of all xpehh, ihs, nsl for neurological genes and match on any sfs stats
g <- neurological_GR$SYMBOL
neurological_keep <- c("CNTNAP2","CERS6","VPS13C","SLC2A13", "DLG", "LRRK2")
neuroPol <- xpehh %>% select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% filter(pop1 %in% c("CIM","NZM","SAM","TON"), xpehh_value > 0) %>%
  filter(genename %in% g) %>%
  select(pop1,pop2, genename) %>%
  distinct() %>%
  left_join(., panel %>%
              select(pop2 = pop, super_pop) %>%
              distinct(), by = 'pop2') %>%
  group_by(pop1, super_pop, genename) %>%
  tally() %>%
  spread(super_pop, n, 0) %>%
  rename(pop = pop1) %>%
  full_join(., sig_ihs_val %>%
              filter(!is.na(genename), genename != "", genename %in% g, pop %in% poly_pop) %>%
              select(pop, genename) %>%
              group_by(pop,genename) %>%
              tally(), by = c('pop','genename')) %>%
  rename(ihs = n) %>%
  full_join(., sig_nsl_val %>%
              filter(!is.na(genename), genename != "", genename %in% g, pop %in% poly_pop) %>%
              select(pop, genename) %>%
              group_by(pop,genename) %>%
              tally(), by = c('pop','genename')) %>%
  rename(nsl = n) %>%
  left_join(., lower_sig_stats %>%
              select(-contains('chrom')) %>%
              group_by(pop, SYMBOL, statname) %>%
              tally() %>%
              spread(statname, n, 0), by = c('pop', 'genename' = 'SYMBOL') ) %>%
  arrange(genename)

rm(g)
```



```{r, eval = TRUE}
neuroPol %>%  mutate_all(., function(x){replace(x, is.na(x), 0)}) %>%  filter(genename %in% neurological_keep) %>% mutate_all(., function(x){replace(x, x == 0, "")}) %>%
  kable(., caption = '(\\#tab:neurologicalPol1) Neurological disease associated genes with multiple signals of possible selection in Polynesian populations.',booktabs = TRUE, col.names = c('Population','Gene','AFR','AMR','EAS','EUR','POL','SAS','iHS','nSL',"Fay & Wu's H", "Fu & Li's D"," Fu & Li's F", "Tajima's D"," Zeng's E")) %>% kableExtra::row_spec(0, angle = 90) %>% add_header_above(., c("","","XP-EHH" = 6, rep("",7))) %>%  kable_styling(latex_options = "repeat_header")

#%>% kable_styling(latex_options = "repeat_header")# %>% filter((AFR > 0 | AMR >0 | EAS > 0 | ihs > 0 | nsl > 0) & (!is.na(Fay.Wu.H) | !is.na(Fu.Li.D)| !is.na(Fu.Li.F) | !is.na(Tajima.D) | !is.na(Zeng.E)))
```

<!-- xpehh interesting genes -->

<!-- - C12orf40 -->
<!-- - CERS6 -->
<!-- - CNTNAP2 -->
<!-- - LRRK2 -->
<!-- - PKP2 - afr, amr, sas -->
<!-- - SLC2A13 -->




<!-- #### Psychiatric disorders -->

```{r, echo = FALSE, eval = FALSE}
(res <- bind_rows(sig_ihs_val, sig_nsl_val) %>% select(SYMBOL = genename, pop, statid) %>% filter(SYMBOL %in% psychiatric_GR$SYMBOL, pop %in% poly_pop) %>% group_by(pop, SYMBOL, statid) %>% tally())

lower_sig_stats %>% select(SYMBOL, pop, statname) %>% filter(SYMBOL %in% res$SYMBOL, pop %in% poly_pop) %>% group_by(pop, SYMBOL, statname) %>% tally() %>% spread(statname, n) %>% arrange(SYMBOL)

lower_sig_stats %>% select(SYMBOL, pop, statname) %>% filter(SYMBOL %in% psychiatric_GR$SYMBOL, pop %in% poly_pop) %>% group_by(pop, SYMBOL, statname) %>% tally() %>% spread(statname, n) %>% arrange(SYMBOL)

consec_lower_regions %>% filter(SYMBOL %in% psychiatric_GR$SYMBOL, pop %in% poly_pop) %>% arrange(seqnames, start, pop, SYMBOL)
rm(res)
```


<!-- Similar to neurological disease, urate is also a possible biomarker for psychiatric diseases such as bipolar disorder [@Albert2015]. The psychiatric disorders selected and gene lists created from were schizophrenia and bipolar disorder and there were `r length(unique(psychiatric_GR$SYMBOL))` genes associated with these diseases. Table  \@ref(tab:psychiatricPol) contains some of the genes that had multiple statistics, or multiple Polynesian populations with evidence of potential selection. One of the main results for this gene list was _CNTN4_ which was the gene with the highest number of significant \gls{snp}s with 27 significant markers for \gls{ihs} and 18 for \gls{nsl} in the \gls{cim} population. Except for \gls{nzm} had 14 markers that were significant for \gls{ihs} and 10 for \gls{nsl}, there was no other population that had more than two significant markers at _CNTN4_. _AUTS2_ was a gene that had a large number of \gls{fwh} windows in the 1st percentile (18) for \gls{fwh} in \gls{nzm} but the only other populations with more than two windows were \gls{ceu}, \gls{clm}, and \gls{gbr}. However, this didn't reach the threshold for being defined as a consecutive region of score depression. -->

<!-- consec regions -->

<!-- - chr10	44725002	44965001	240000 CXCL12 -->
<!--     - cim fwh -->
<!-- - chr16	67775002	68105001	330000 -->
<!--     - nzm td and flf ( chr16	67775002	68095001	320000), fld (chr16	67785002	68045001	260000) -->
<!--     - cim td chr16	67835002	68115001	280000  -->
<!--     - CENPT CTRL DDX28 DPEP2 DPEP3 EDC4 LCAT NRN1L NUTF2 PSKH1 PSMB10 RANBP10 SLC12A4 THAP11 TSNAXIP1 -->
<!-- - chr3	52375002	52615001	240000 -->
<!--     - cim fld -->
<!--     - BAP1 DNAH1 NISCH NT5DC2 PBRM1 PHF7 SEMA3G SMIM4 SMIM4 STAB1 TNNC1 -->
<!-- - chr4	170015002	170435001	420000 -->
<!--     - ton and sam td -->
<!--     - NEK1 -->
<!-- - chr4	170605002	170855001	250000 -->
<!--     - cim fwh -->
<!--     - CLCN3 -->
<!-- - chr6	26915002	27195001	280000 -->
<!--     - sam td -->
<!--     - HIST1H2AG HIST1H2BJ -->
<!-- - chr6	31595002	31835001	240000 -->
<!--     - cim fld -->
<!--     - BAG6 PRRC2A -->

```{r psychiatricPol, eval = FALSE}
g <- psychiatric_GR$SYMBOL
psychiatric_keep <- c("ANK3","ANKRD44","AUTS2","CLCN3","CNTN4","CNTNAP5","CTNNA2","CXCL12","DPYD","GPM6A","NEKK1","PEPD","SLCO6A1","ST6GALNAC3")

xpehh %>%  select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% filter(pop1 %in% c("CIM","NZM","SAM","TON"), xpehh_value > 0) %>% filter(genename %in% g) %>% select(pop1,pop2, genename) %>% distinct() %>% left_join(., panel %>% select(pop2 = pop, super_pop) %>% distinct(), by = 'pop2') %>% group_by(pop1, super_pop, genename) %>% tally() %>% spread(super_pop, n, 0) %>% rename(pop = pop1) %>% full_join(., sig_ihs_val %>% filter(!is.na(genename), genename != "", genename %in% g, pop %in% poly_pop) %>% select(pop, genename) %>% group_by(pop,genename) %>% tally(), by = c('pop','genename')) %>% rename(ihs = n) %>% full_join(., sig_nsl_val %>% filter(!is.na(genename), genename != "", genename %in% g, pop %in% poly_pop) %>% select(pop, genename) %>% group_by(pop,genename) %>% tally(), by = c('pop','genename')) %>% rename(nsl = n) %>% left_join(., lower_sig_stats %>% select(-contains('chrom')) %>% group_by(pop, SYMBOL, statname) %>% tally() %>% spread(statname, n), by = c('pop', 'genename' = 'SYMBOL') ) %>% arrange(genename) %>% mutate_all(., function(x){replace(x, is.na(x), 0)}) %>% 
  filter(genename %in% psychiatric_keep) %>%
      mutate_all(., function(x){replace(x, x == 0, "")}) %>%
  kable(., caption = '(\\#tab:psychiatricPol) Psychiatric disorders', booktabs = TRUE, longtable = TRUE, format = 'latex', col.names = c('Population','Gene','AFR','AMR','EAS','EUR','POL','SAS','iHS','nSL',"Fay & Wu's H", "Fu & Li's D"," Fu & Li's F", "Tajima's D"," Zeng's E")) %>% kableExtra::row_spec(0, angle = 90) %>% add_header_above(., c("","","XP-EHH" = 6, rep("",7))) %>%  kable_styling(latex_options = "repeat_header")# %>% filter((AFR > 0 | AMR >0 | EAS > 0 | ihs > 0 | nsl > 0) & (!is.na(Fay.Wu.H) | !is.na(Fu.Li.D)| !is.na(Fu.Li.F) | !is.na(Tajima.D) | !is.na(Zeng.E)))
rm(g)
```

<!-- main results: -->

<!-- - ANK3 4p vs all super in xpehh, sam with ihs -->
<!-- - ANKRD44 4p vs afr,amr,eur; wp vs sas. 4p fwh, sam flf,fld,td, cim td -->
<!-- - nzm AUTS2 18 windows fwh, 4 td, 1 nsl, xpehh: 4 afr pops, 2 eur -->
<!-- - CLCN3 4p xpehh: afr, amr, eur, sas, fwh, wp:td -->
<!-- - CNTN4 ep xpehh and ihs -->
<!-- - CNTNAP5 wp xpehh fwh, sam nsl and all sfs -->
<!-- - CTNNA2 xpehh, ihs, nsl, fwh -->
<!-- - CXCL12 xpehh, fwh -->
<!-- - DPYD wp ihs/fwh/td -->
<!-- - GPM6A xpehh (ep) - 4p eas, ihs, nsl -->
<!-- - NEKK1 4p xpehh afr,amr,eas(wp),eur,sas. fwh, td, sam flf/fld -->
<!-- - PEPD wp xpehh afr, amr, eas, eur -->
<!-- - SLCO6A1 xpehh afr, amr, eas, eur, sas. ep ihs/nsl/ze -->
<!-- - ST6GALNAC3 afr,amr,eur,sas. wp ihs/nsl. 4p td -->


```{r, eval = FALSE}
xpehh %>%  select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% filter(pop1 %in% c("CIM","NZM","SAM","TON"), xpehh_value > 0) %>% filter(genename %in% psychiatric_GR$SYMBOL) %>% select(pop1,pop2, genename) %>% distinct() %>% left_join(., panel %>% select(pop2 = pop, super_pop) %>% distinct(), by = 'pop2') %>% group_by(pop1, super_pop, genename) %>% tally() %>% spread(super_pop, n, "")%>% arrange(genename) %>% kable(., caption = 'Number of populations by super population, with significant xpehh for psychological disease associated genes', booktabs = TRUE)
```

<!-- XPEHH interesting genes -->

<!-- - ANK3 -->
<!-- - ANKRD44 -->
<!-- - CLCN3 -->
<!-- - CNTN4 - ep -->
<!-- - CNTNAP5 - wp -->
<!-- - CTNNA2 -->
<!-- - CXCL12 -->
<!-- - GPM6A - ep -->
<!-- - NEK1 -->
<!-- - PCCB -->
<!-- - PEPD - wp -->
<!-- - SLCO6A1 -->






#### Inflammatory and autoimmune disorders

```{r, echo = FALSE, eval = FALSE}
(res <- bind_rows(sig_ihs_val, sig_nsl_val) %>% select(SYMBOL = genename, pop, statid) %>% filter(SYMBOL %in% zhang_immune_GR$SYMBOL, pop %in% poly_pop) %>% group_by(pop, SYMBOL, statid) %>% tally())

lower_sig_stats %>% select(SYMBOL, pop, statname) %>% filter(SYMBOL %in% res$SYMBOL, pop %in% poly_pop) %>% group_by(pop, SYMBOL, statname) %>% tally() %>% spread(statname, n) %>% arrange(SYMBOL)

lower_sig_stats %>% select(SYMBOL, pop, statname) %>% filter(SYMBOL %in% zhang_immune_GR$SYMBOL, pop %in% poly_pop) %>% group_by(pop, SYMBOL, statname) %>% tally() %>% spread(statname, n) %>% arrange(SYMBOL)

lower_sig_stats %>% inner_join(., res %>% select(pop, SYMBOL)) %>% group_by(pop, SYMBOL, statname) %>% tally() %>% spread(statname, n) %>% arrange(SYMBOL)

consec_lower_regions %>% filter(SYMBOL %in% zhang_immune_GR$SYMBOL, pop %in% poly_pop) %>% arrange(seqnames, start, end, SYMBOL, pop) %>% select(-TXSTRAND) %>% distinct()

#consec_lower_regions %>% inner_join(., res %>% select(pop, SYMBOL)) %>% group_by(pop, SYMBOL, statname) %>% tally() %>% spread(statname, n) %>% arrange(SYMBOL)


rm(res)
```



<!-- Ramos2015 table 2 -->

```{r zhangImmuneTab}
g <- zhang_immune_GR$SYMBOL
immune_keep = c("ARAP", "ARHGEF3", "ARL15", "ATM", "BLK", "CCR2", "CCR3", "CDH23", "CNTNAP2", "CSMID1", "DSTYK", "ERAP", "FAM167A", "FCHSD2", "FOXP1", "IKZF1", "JAZF1", "KCNB2", "LPP", "LRRK2", "RAD51B", "RREB1", "SLC2A13", "SLCO6A1", "TERF", "THADA", "XKR6" )
immunePol <- xpehh %>%  select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% filter(pop1 %in% c("CIM","NZM","SAM","TON"), xpehh_value > 0) %>% filter(genename %in% g) %>% select(pop1,pop2, genename) %>% distinct() %>% left_join(., panel %>% select(pop2 = pop, super_pop) %>% distinct(), by = 'pop2') %>% group_by(pop1, super_pop, genename) %>% tally() %>% spread(super_pop, n, 0) %>% rename(pop = pop1) %>% full_join(., sig_ihs_val %>% filter(!is.na(genename), genename != "", genename %in% g, pop %in% poly_pop) %>% select(pop, genename) %>% group_by(pop,genename) %>% tally(), by = c('pop','genename')) %>% rename(ihs = n) %>% full_join(., sig_nsl_val %>% filter(!is.na(genename), genename != "", genename %in% g, pop %in% poly_pop) %>% select(pop, genename) %>% group_by(pop,genename) %>% tally(), by = c('pop','genename')) %>% rename(nsl = n) %>% left_join(., lower_sig_stats %>% select(-contains('chrom')) %>% group_by(pop, SYMBOL, statname) %>% tally() %>% spread(statname, n), by = c('pop', 'genename' = 'SYMBOL') ) %>% arrange(genename) %>% mutate_all(., function(x){replace(x, is.na(x), 0)})  
 
rm(g)
```

The list of inflammatory and autoimmune disorders was from @Zhang2013a Table 2, and included: inflammatory bowel disease, Crohn's disease, ulcerative colitis, celiac disease, systemic lupus erythematosus, rheumatoid arthritis, psoriasis, multiple sclerosis, type 1 diabetes, primary biliary cirrhosis, and vitligo. Autoimmunity is caused by the immune system attacking self rather than foreign antigen and involves both innate and adaptive immune response [@Waldner2009]. The genes involved with the immune response have been suggested as possible candidates of selection [@Grossman2013]. Both gout and these disorders have a common cause from the activation of the immune system.

There were `r length(unique(zhang_immune_GR$SYMBOL))` genes associated with these inflammatory and autoimmune disorders, many of which were also in common with the genes associated with urate, obesity, \gls{t2d}, kidney disease, and metabolic syndrome, which is indicative of the immune response being part of all of these conditions. Some of the loci that had multiple statistics, across multiple Polynesian populations indicating possible selection were: _ARHGEF3_, _ARL15_, _ATM_, _BLK_, _CCR3_, _CNTNAP2_, _DSTYK_, _FAM167A_, _FOXP1_, _KCNB2_, _LPP_, _LRRK2_, _RREB1_, _SLC2A13_, _ALCO6A1_, _THADA_, and _XKR6_. _CCR2_ and _FCHSD2_ had a higher number of significant results in the Eastern Polynesian populations, whereas _CDH23_ and _JAZF1_ had a higher number of significant results in the Western Polynesian populations. _CCR2_ and _CCR3_ were also associated with obesity, _ARL15_ and _JAZF1_ were associated with obesity and \gls{t2d}, _LPP_ and _THADA_ were associated \gls{t2d}, and _RREB1_ was associated with urate and \gls{t2d}. Some of the genes had very high numbers of windows from the 1st percentile for \gls{td} intersect with them, such as _RAD51B_ which was a consecutively depressed score region for \gls{sam}, and _XKR6_, which was a consecutively depressed score region for \gls{cim}, \gls{sam} and \gls{ton} populations. 

```{r}
 immunePol %>% filter(genename %in% immune_keep) %>% 
      mutate_all(., function(x){replace(x, x == 0, "")}) %>%
 kable(., caption = '(\\#tab:zhangImmune) Genes associated with various inflammatory and autoimmune diseases that showed possible selection from multiple statistics in multiple Polynesian populations.', longtable = TRUE, booktabs = TRUE, col.names = c('Population','Gene','AFR','AMR','EAS','EUR','POL','SAS','iHS','nSL',"Fay & Wu's H", "Fu & Li's D"," Fu & Li's F", "Tajima's D"," Zeng's E")) %>% kableExtra::row_spec(0, angle = 90) %>% add_header_above(., c("","","XP-EHH" = 6, rep("",7))) %>%  kable_styling(latex_options = "repeat_header")# %>% filter((AFR > 0 | AMR >0 | EAS > 0 | ihs > 0 | nsl > 0) & (!is.na(Fay.Wu.H) | !is.na(Fu.Li.D)| !is.na(Fu.Li.F) | !is.na(Tajima.D) | !is.na(Zeng.E)))
```


```{r, eval = FALSE}
c("ARHGEF3", "ARL15", "ATM", "BLK", "CCR3", "CNTNAP2","DSTYK","FAM167A",  "FOXP1", "KCNB2", "LPP", "LRRK2", "RREB1","SLC2A13","ALCO6A1","THADA","XKR6") #poly
c("CCR2", "FCHSD2") #ep
c("CDH23", "JAZF1") #wp

genelists %>% filter(genename %in% c(c("ARHGEF3", "ARL15", "ATM", "BLK", "CCR3", "CNTNAP2","DSTYK","FAM167A",  "FOXP1", "KCNB2", "LPP", "LRRK2", "RREB1","SLC2A13","ALCO6A1","THADA","XKR6"), c("CCR2", "FCHSD2"),c("CDH23", "JAZF1") ))
```



<!-- main results: -->

<!-- - ARAP ep -->
<!-- - ARHGEF3 -->
<!-- - ARL15 -->
<!-- - ATM -->
<!-- - BLK -->
<!-- - CCR2 ep -->
<!-- - CCR3 -->
<!-- - CDH23 ihs/nsl wp -->
<!-- - CNTNAP2 -->
<!-- - CSMID1 -->
<!-- - DSTYK -->
<!-- - ERAP -->
<!-- - FAM167A -->
<!-- - FCHSD2 -->
<!-- - FOXP1 -->
<!-- - IKZF1 nzm -->
<!-- - JAZF1 ton -->
<!-- - KCNB2 nzm -->
<!-- - LPP -->
<!-- - LRRK2 -->
<!-- - RAD51B sam nzm -->
<!-- - RREB1 cim/nzm/ton -->
<!-- - SLC2A13 -->
<!-- - SLCO6A1 -->
<!-- - TERF -->
<!-- - THADA -->
<!-- - XKR6 -->

```{r, eval = FALSE}
xpehh %>%  select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% filter(pop1 %in% c("CIM","NZM","SAM","TON"), xpehh_value > 0) %>% filter(genename %in% zhang_immune_GR$SYMBOL) %>% select(pop1,pop2, genename) %>% distinct() %>% left_join(., panel %>% select(pop2 = pop, super_pop) %>% distinct(), by = 'pop2') %>% group_by(pop1, super_pop, genename) %>% tally() %>% spread(super_pop, n, "")%>% arrange(genename) %>% kable(., caption = 'Number of populations by super population, with significant xpehh for psychological disease associated genes', booktabs = TRUE)
```

<!-- XPEHH interesting genes -->

<!-- - ARAP1 - ep -->
<!-- - ARL15 -->
<!-- - ATG16L2 - ep -->
<!-- - BLK -->
<!-- - CCR2 - ep -->
<!-- - CCR3 -->
<!-- - CNTNAP2 -->
<!-- - DSTYK -->
<!-- - ERAP1 -->
<!-- - FAM167A -->
<!-- - FCHSD2 - ep -->
<!-- - FOXP1 -->
<!-- - GIN1 -->
<!-- - HLA-DOB/HLA-DQA1/HLA-DQA2/HLA-DQB1/HLA-DQB2 - sam -->
<!-- - IKZF1 - nzm -->
<!-- - JAZF1 - wp -->
<!-- - KCNB2 - nzm -->
<!-- - LPP -->
<!-- - LRRK2 -->
<!-- - MANBA -->
<!-- - RDH10 - all but mostly nzm -->
<!-- - RERE - wp -->
<!-- - RPL7 -->
<!-- - RREB1 - as per urate -->
<!-- - SLC2A13 -->
<!-- - SLCO6A1 -->
<!-- - SMAD7 - nzm -->
<!-- - THADA - as per t2d? -->
<!-- - ZMIZ1 -->



```{r, eval = FALSE}
lower_sig_stats %>% filter(SYMBOL %in% zhang_immune_GR$SYMBOL, pop %in% poly_pop) %>% group_by(pop, statname, SYMBOL) %>% tally() %>% spread(statname, n) %>% arrange(SYMBOL)
```

```{r, eval = FALSE}
consec_lower_regions %>% filter(pop %in% poly_pop, SYMBOL %in% zhang_immune_GR$SYMBOL) %>% select(seqnames, start, end, width, stat, pop) %>% distinct() %>% spread(key = stat, value = stat, fill = "")
```

#### Malaria {#malariaRes}
```{r malariaTab}
g <- malaria_GR$SYMBOL
malariaPol <- xpehh %>%  select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% filter(pop1 %in% c("CIM","NZM","SAM","TON"), xpehh_value > 0) %>% filter(genename %in% g) %>% select(pop1,pop2, genename) %>% distinct() %>% left_join(., panel %>% select(pop2 = pop, super_pop) %>% distinct(), by = 'pop2') %>% group_by(pop1, super_pop, genename) %>% tally() %>% spread(super_pop, n, "") %>% rename(pop = pop1) %>% full_join(., sig_ihs_val %>% filter(!is.na(genename), genename != "", genename %in% g, pop %in% poly_pop) %>% select(pop, genename) %>% group_by(pop,genename) %>% tally(), by = c('pop','genename')) %>% rename(ihs = n) %>% full_join(., sig_nsl_val %>% filter(!is.na(genename), genename != "", genename %in% g, pop %in% poly_pop) %>% select(pop, genename) %>% group_by(pop,genename) %>% tally(), by = c('pop','genename')) %>% rename(nsl = n) %>% left_join(., lower_sig_stats %>% select(-contains('chrom')) %>% group_by(pop, SYMBOL, statname) %>% tally() %>% spread(statname, n), by = c('pop', 'genename' = 'SYMBOL') ) %>% arrange(genename) 
  
#malariaPol %>% mutate_all(., function(x){replace(x, is.na(x), "")}) %>%  kable(., caption = 'Number of significant markers and windows for loci associated with malaria', col.names = c('Population','Gene','AFR','AMR','EAS','EUR','SAS','iHS','nSL',"Fay and Wu's H","Fu and Li's D","Fu and Li's F","Tajima's D","Zeng's E")) %>% kableExtra::row_spec(0, angle = 90) %>% kableExtra::add_header_above(., c("","", "XP-EHH" = 5))# %>% filter((AFR > 0 | AMR >0 | EAS > 0 | ihs > 0 | nsl > 0) & (!is.na(Fay.Wu.H) | !is.na(Fu.Li.D)| !is.na(Fu.Li.F) | !is.na(Tajima.D) | !is.na(Zeng.E)))
rm(g)
```

```{r, eval = FALSE}
xpehh %>%  select(-popid1, -popid2, -posid, -experimentid, -statvalue) %>% filter(pop1 %in% c("CIM","NZM","SAM","TON"), xpehh_value > 0) %>% filter(genename %in% malaria_GR$SYMBOL) %>% select(pop1,pop2, genename) %>% distinct() %>% left_join(., panel %>% select(pop2 = pop, super_pop) %>% distinct(), by = 'pop2') %>% group_by(pop1, super_pop, genename) %>% tally() %>% spread(super_pop, n, "")%>% arrange(genename) %>% kable(., caption = 'Number of populations by super population, with significant xpehh for malaria associated genes', booktabs = TRUE)
```




```{r, eval = FALSE}
sig_ihs_val %>% filter(genename %in% malaria_GR$SYMBOL) %>% select(-gene_chrom, -statid) %>% left_join(., markers, by = c('chrom','chrom_start')) %>% arrange(genename, marker, pop)
```

Urate has a central role in malarial infection and triggering an immune response [@GallegoDelgado2014]. Given that immune response is a strong candidate to be under selective pressure, malaria associated genes were looked at to see if there were signals of selection. From the GWAS catalog there were three genes that had been associated with malaria (of six) that also had evidence of selection with the Polynesian populations. The three genes that also had significant \gls{ihs} scores with Polynesian populations were _ABO_, _ATP2B4_, and _DDC_.

_ABO_ was the only gene that had significant markers in all four Polynesian populations of the malaria associated genes, and this was only for \gls{ihs}. \gls{cim} had two markers that were significant, the first, rs1053878, was also significant in \gls{ton}. The second marker, rs55764262, was also significant in \gls{nzm} and \gls{sam}, and was in high \gls{ld} (\gls{eas} R^2^ = 0.967) with rs549446. Rs549446 was significant in both \gls{chs} and \gls{jpt}. There were no windows from the frequency-based statistics that met the significance threshold. _ABO_ has also been associated with obesity in \gls{gwas} [@Comuzzie2012]. The marker rs8176741 had been associated with severe malaria [@Timmann2012], however, this marker had a significant \gls{ihs} in \gls{fin} but no other populations. The other \gls{snp}s that showed significance for \gls{ihs} were not in \gls{ld} (\gls{eas} R^2^ = 0.002) with rs55764262, for which \gls{cim}, \gls{nzm} and \gls{sam} all had a significant result. Both \gls{cim} and \gls{ton} had a different marker (rs1053878) that was significant for \gls{ihs} in _ABO_.


Both \gls{sam} and \gls{nzm} had \gls{snp}s that were significant for \gls{ihs} in _ATP2B4_, \gls{sam} had rs10494845, and \gls{nzm} had rs142206068. The malaria-associated variant was rs10900585 [@Timmann2012]. All three variants are in \gls{ld} (\gls{eas} R^2^ = 1.0).

_DDC_ encodes dopa decarboxylase and has been associated with malaria [@Jallow2009] and also \gls{bmi} [@Locke2015]. \gls{nzm} had significant markers for \gls{xpehh} for nearly all other populations, \gls{cim} had fewer populations, but across all super population other than \gls{sas}. The \gls{nzm} had multiple markers within the malaria-associated gene, _DDC_, for \gls{nsl}. The markers were rs2060762, rs3807558, rs2329371, and rs1966839. The same markers in addition with rs11238131 and rs4580999 were significant for \gls{ihs}. A \gls{ld} matrix with marker positions is shown in Figure \@ref(fig:ldDDCnzm). The marker rs1966839 (reference/derived allele = T, alternative/ancestral allele = C) was in high LD (0.998) with the malaria-associated rs1451375 (reference/ancestral allele = C, alternative/derived = A) where the reference and alternative alleles correspond between the \gls{snp}s and both are intronic variants. Under a dominant model the A allele of rs1451375 is protective of severe malaria (OR = 0.75, CI = 0.66-0.85, P = 6 x 10-6) [@Jallow2009].




<!-- rs7803788, rs880028, rs10249420 are from Manjurano2015 -->




```{r, eval = FALSE}
sig_nsl_val %>% filter(genename %in% malaria_GR$SYMBOL) %>% select(-gene_chrom, -statid) %>% left_join(., markers, by = c('chrom','chrom_start'))
```


```{r}
# snps from sig_ihs_val, Jallow2009, and Manjurano2015 for DDC locus
ddc_eas_r2 <- read.table(text = "RS_number	rs2060762	rs11238131	rs4580999	rs880028	rs3807558	rs6264	rs2329371	rs1451375	rs10249420	rs7803788	rs1966839
rs2060762	1.0	0.6	0.449	0.07	0.688	NA	0.723	0.492	0.671	0.671	0.49
rs11238131	0.6	1.0	0.585	0.01	0.68	NA	0.531	0.319	0.484	0.484	0.316
rs4580999	0.449	0.585	1.0	0.089	0.674	NA	0.51	0.611	0.46	0.46	0.607
rs880028	0.07	0.01	0.089	1.0	0.088	NA	0.061	0.016	0.064	0.064	0.016
rs3807558	0.688	0.68	0.674	0.088	1.0	NA	0.744	0.497	0.69	0.69	0.495
rs6264	NA	NA	NA	NA	NA	NA	NA	NA	NA	NA	NA
rs2329371	0.723	0.531	0.51	0.061	0.744	NA	1.0	0.707	0.923	0.923	0.698
rs1451375	0.492	0.319	0.611	0.016	0.497	NA	0.707	1.0	0.769	0.769	0.988
rs10249420	0.671	0.484	0.46	0.064	0.69	NA	0.923	0.769	1.0	1.0	0.76
rs7803788	0.671	0.484	0.46	0.064	0.69	NA	0.923	0.769	1.0	1.0	0.76
rs1966839	0.49	0.316	0.607	0.016	0.495	NA	0.698	0.988	0.76	0.76	1.0", header = TRUE   )
```


(ref:ldDDCnzm) Linkage disequilibrium of \gls{ihs} significant SNPs 

<!-- (rs2060762, rs11238131, rs3807558, rs2329371, and rs1966839) in \gls{nzm} at the _DDC_ locus with malaria (rs1451375), BMI, (rs10499694), and blood metabolite ratios (rs7809234). -->

```{r ldDDCnzm, fig.caption= '(ref:ldDDCnzm)', eval = TRUE}
knitr::include_graphics('images/03_selectionpipeline/LDplot_EAS_DDC_nzm_ihs_extras.png')
```


```{r, eval = FALSE}
pol_fst %>% filter(chrom == 7, chrom_start > 49e6 & chrom_start < 52e6)
```



```{r, eval = FALSE}
UpSetR::upset(fromList(list(urate = uratePol$genename, obesity = obesityPol$genename, t2d = t2dPol$genename, kd = kdPol$genename, metsyn = metsynPol$genename, neuro = neurological_keep, psychiatric = psychiatric_keep, immune = immune_keep)), nintersects = NA, nsets = 8)
```


## Chapter Discussion
<!--
Important to remember that the modern day populations esp AFR won't necessarily have the same results as AFR pops did 100 kya

Pipeline discussion - how were the results?
  - Are the genes that are picked up similar to what has been discovered?

Is there anything special about the population comparisons?
  - shared genes between similar populations?
  - East/West split seen?
  - NZC vs GBR similarity?
  
  @Rafajlovic2014 has a good talk about the use of the empirical distribution for sfs stats.
  
  
Because fld and flf are based on the number of singletons, how is this going to be affected by using chip data that should have essentially 0. presumable this will dramatically affect the interpretability of the statistic and push everything over to the right but is it acceptable to recentre the distribution and look that the extremes from there? does it have biological plausibility?  

what does it mean to have a significant window of one tail neighboured by the other tail? Or on the broad scale what does it mean to have a gene classed as significant in both tails?

effect of phasing when polynesian populations aren't represented in the 1000 genomes populations

-->

### Identifying regions under selection in Polynesian populations

One of the primary objectives of this chapter were to investigate regions that were under selection in Polynesian populations. From the intra-population statistics used, there were between 866 and 974 genes identified in the Polynesian populations that had both intra-population haplotypic and \gls{sfs} statistics that were in the 1st percentile. The regions with possible evidence of selection showed that comparative to the other super populations, the Polynesian populations had the fewest genes that only had evidence within the Polynesian super population, whereas the African populations had the largest number of genes that only had evidence in the African super population (Figure \@ref(fig:superUpset)). The Polynesian populations had similar distributions of the \gls{sfs}-based statistics (explored more in Chapter 4).

Pathway analysis on the regions with possible selection in the Polynesian populations revealed that many of the genes that had evidence were involved with various forms of cell signalling, with many related to calcium channels. The gene _CNTN4_ had the highest number of significant markers in \gls{cim} and 3^rd^ most in \gls{nzm} for \gls{ihs}. This particular locus hasn't been associated with urate at a genome-wide significance level, however there is evidence of a potential association [@Chittoor2016]. 

> What are the genes that were coming up? what are the pathways that they were in?

> There were lots of voltage gated calcium channels, where are they expressed?


When looking at genes that have been associated with urate, gout and related co-morbidities, there is no apparent enrichment of genes for particular conditions in the Polynesian populations, when compared to the other populations (Table \@ref(tab:superDiseaseGenesTab)). This may indicate that there hasn't been any selection for the particular traits, but that individually genes may have been selected for, for other roles they play. There was also evidence of selection in loci that were associated with malaria and were also associated with \gls{t2d} and obesity.


<!-- - there is evidence of selection in urate associated loci in the Polynesian populations -->
<!-- - this evidence is not specific to urate however, this is evidenced by the fact that there is little signal in the urate transporters but it is in the loci that are part of more general metabolic pathways -->
<!-- - the reasons as to why there would be selection are unclear -->

<!-- - The pathways that showed enrichment were involved voltage-gated Ca channel. these channels are predominantly expressed in the lung and the heart. -->
<!-- - many of the pathways were involved with cell signalling/communication so are vital for life -->




### Selection of Urate Associated Genes

The second objective was to investigate regions associated with urate, gout and related co-morbidities for evidence of selection in the Polynesian populations. There was evidence for selection in urate associated loci. However, the evidence did not appear to be specific for urate itself, but for loci that are involved in more general metabolic pathways. The main effect loci identified through \gls{gwas}, _SLC2A9_ and _ABCG2_, had limited evidence of positive selection. _ABCG2_ had a significant marker (rs2622626) with \gls{ihs} and \gls{nsl} and only in \gls{ton} from the Polynesian populations, the only other populations were \gls{gih} and \gls{itu} from the \gls{sas} super population. This marker showing significance however might be due to the gout prevalence of the \gls{ton} sample population which was 54%, the same result is not seen in closest population with the most similar ancestral background, \gls{sam}, where the prevalence of gout in the sample population was much lower at 12%.

Only _RREB1_ and _IGFR1_ had haplotypic evidence with \gls{xpehh} across multiple other populations. The other gene of note that didn't have haplotypic evidence, but did have multiple windows in the 1st percentile across multiple statistics and all Polynesian populations was _BCAS3._ _BCAS3_ had been previously identified as selected in @Grossman2013. _RREB1_ was also associated with obesity and \gls{t2d} traits. @Zhang2013a had reported a variant nearby to _RREB1_ (rs675209) as having evidence of selection but this particular marker was absent from the CoreExome chip.

There was a similar situation with obesity, \gls{t2d}, kidney disease, and metabolic syndrome where it was found that there were multiple loci that had evidence of possible selection. The number of genes that had possible evidence of selection was similar in the Polynesian populations to that of the other populations. An interesting point however, was the Western Polynesian populations had the fewest genes associated with \gls{t2d} identified with \gls{ihs} or \gls{nsl} than all the other populations, despite Polynesian populations having a high prevalence of \gls{t2d} [@Winnard2013]. 

Other complex diseases such as the neurological, and, inflammatory and autoimmune also showed multiple genes that showed evidence of selection. One of the 'themes' that appeared was there was genes were often involved in multiple diseases or gene lists, and often involved with the immune system. 





<!-- - Similar to @Grossman2013, we found that BCAS3 had evidence of selection -->

<!-- Zheng2013 -->
<!-- SH2B3-ATXN2 has snps associated with urate, and other diseases (rs3184504, rs4766578, rs653178, and rs11065987) are in close LD. derived allele haplotype was associated with increased risk of autoimmune, bp, and urate. haplotype was common in European but not in afr or asi. possible recent sweep due to bacterial infection @Zhernakova2010. The locus also extends to ALDH2 which @Oota2004 suggests is under selection -->


### Thrifty gene hypothesis

There was no evidence of selection in either of the previously identified loci of _PPARGC1A_ and _CREBRF_ in Polynesian populations. The CoreExome chip did not have rs8192678, but it did have rs1873532 which was in high \gls{ld} (\gls{eas} R^2^ = 0.976). The lack of evidence was consistent with the results of @Cadzow2016 for _PPARGC1A_ where there was no indication of selection from frequency- or haplotype-based methods. 

No selection signal was observed at _CREBRF_ in any of the Polynesian populations that were tested. One reason for this is that the CoreExome chip did not have the specific variants reported in @Minster2016. Downstream from _CREBRF_ there was a consistent signal with what had been reported. This was only found in the Western Polynesian populations and the populations of the \gls{eas} super population so suggests that had the dataset contained the variants that it would have been possible to detect. However, without the two markers reported in @Minster2016 it is unable to be determined if _CREBRF_ had evidence of selection from the CoreExome data.

The pathway analysis of genes that were in the extremes for the different selection and neutrality statistics showed that pathways involved with cell signalling were the main results for the Polynesian populations, rather than metabolic pathways. The genes that indicated selection, even for obesity and \gls{t2d}, which are traditionally thought of as being the thrifty phenotype, were genes that were largely either neurological or involved in the immune system.

\gls{nzm} did have significant results for two diabetes related pathways, "Type II diabetes mellitus_Homo sapiens_hsa04930" and "Insulin secretion_Homo sapiens_hsa04911", however, only up to 10% of the pathways had genes that met the significance thresholds. Of the genes that were part of those pathways, up to 50% were genes that encoded subunits of a voltage-gated calcium channel. 

As discussed in @Gosling2014, the thrifty gene hypothesis in the context of the Pacific does not fit with the anthropological history, and from this analysis it appears to have no to minimal support from the loci that exhibit signals of possible selection. [@Ayub2014c] had a similar finding with respect to \gls{t2d}. While some individual loci show some signs of selection, it appears as if selection at \gls{t2d} loci is not the case.

<!-- ihs = 2.94 -->

<!-- rs373863828 missense variant: The derived allele variant raises BMI but is protective for T2D. lowers Fastign glucose, total cholesterol, hdl and ldl. -->



<!-- FTO - ihs rs4396532 in NZM -->


<!-- LEPR - ihs rs12042877 in TON. only other populations were AFR and EAS -->

### Malaria

Response to infectious challenge is a key mechanism that has been thought of as a selective pressure. Malarial infection provided a possible challenge to which urate may have been subjected to selective pressure as such provided a viable candidate due to the involvement of urate with malarial resistance [@GallegoDelgado2014]. There is also evidence of malaria in the prehistory skeletal remains of the western Pacific islands [@buckley2006predators]. Six loci that were associated with malaria and identified from the \gls{gwas} catalog were investigated for having signals of selection. Three of these genes did have evidence of selection in the Polynesian populations. Two loci that had evidence had also been identified as being associated with obesity and \gls{t2d}. _ABO_ and _ATP2B4_ did not have strong evidence, and it came from only \gls{ihs}. _ATP2B4_ encodes a plasma membrane calcium transporter, another calcium related gene with evidence of selection in \gls{nzm}. _DDC_ on the other hand had comparatively strong haplotypic evidence in the Eastern Polynesian populations. \gls{nzm} had significant \gls{xpehh} results with every other non-Polynesian population and had six significant markers for \gls{ihs}, three of which were in the top 400 most extreme scores. Rs1966839 in _DDC_ was in near perfect \gls{ld} with the malaria associated intronic variant rs1451375 [@Jallow2009].

From the obesity \gls{gwas} associated gene list, there was _ADCY9_, which had a very strong signal for \gls{ihs} for 7 markers in \gls{cim} and 2 markers in \gls{nzm}, that were the only populations with significant \gls{ihs} values. A variant, rs10775349, in _ADCY9_ has been associated with malaria in a candidate gene approach but not in \gls{gwas} [@Maiga2013].

Three of the four loci had possible evidence of selection and were associated with obesity traits. The fourth, _ATP2B4_, being a calcium pump showed similarity with the pathways that showed significant enrichment in Polynesian populations by being related to calcium.   

<!-- rs879620 associated with bmi -->

<!-- The malarial resistance gene _HBB_ (identified in [@Currat2002; @Ohasi2004]) was not identified as having evidence of selection -->

### Study Limitations {#chap3limit}

There were some technical limitations of the study that influenced the results obtained. The use of a \gls{snp} chip meant that there was an ascertainment bias built into the markers observed [@nielsen2007recent]. The quality control process for the CoreExome \gls{snp} chip also has a bias against singletons and low minor allele frequency \gls{snp}s [@Guo2014]. The effect of this was seen in the right shifting of the distributions of the intra-population frequency based statistics of \gls{td}, \gls{fld}, \gls{flf}, and \gls{ze}, all of which make use of low frequency or singleton variants in their calculations (Table \@ref(tab:superGlobalSummary)). This shift was not observed in \gls{fwh} which compared the ratio of high frequency derived and ancestral allele variants with the intermediate frequency variants. The shift was seen in \gls{td}, however with the African populations being shifted to the right, such that the first percentile for many was above zero. The intermediate and high frequency variants being over represented compared to the actual situation and will also have had an impact on the haplotypic statitics due to haplotypes that have low-frequency variants being altered.

Another technical limitation that affected the haplotypes was phasing. The phasing of the haplotypes for the Polynesian populations made use of the 1000 Genomes Project reference haplotype panel for probabilistic phasing and could not make use of trios. This panel does not contain Polynesian-specific haplotypes and so the phasing would be influenced by this. While the majority of common haplotypes will be represented in this panel, haplotypes that are Polynesian-specific are not and this introduces incorrect genotypes. The phasing uncertainty is also not taken into account for any of the methods, this will influence the haplotypic methods of selection. 

With the adaptation of the selection and neutrality tests into genome-wide usage, specifically the sliding window approach, there was the need to balance the number of segregating sites with the window size [@Pybus2014]. This balance is entirely influenced by \gls{snp} density with whole genome (re)sequence data offering 78 million bi-allelic \gls{snp}s in the 1000 Genomes Project Phase 3. However, the CoreExome \gls{snp} chip data had lower density at `r lapply(1:22, function(x){diff(markers %>% filter(chrom ==x) %>% arrange(chrom_start) %>% pull(chrom_start))}) %>% unlist() %>%  mean() %>% round_dp(., 0)` \gls{snp}s/kbp. This means that window size needs to increase for \gls{snp} chip data in order to have the same number of segregating sites. However, as window size increases it also reduces the resolution of the genome and with it the ability to associate a window with particular genomic features.

There was also a higher likelihood of a longer gene being reported as significant due to more windows overlapping the genic region. Not only that, but the window size meant that \gls{snp}s not in the window might not even be near the actual window i.e. -45 and +45 kb away from the edge of the defined window in the worst case. This would lead to some difficulty in definitive location. The flip side to this was that it allows for upstream and downstream effects of a \gls{snp} to be incorporated. There was a trade off between \gls{snp} density and window size. There were `r NROW(markers)` bi-allelic \gls{snp} markers used in this analysis from the CoreExome chip. This was less than the other projects such as the HapMap project, and considerably less than the sequence data that the 1000 Genomes Project used. The marker density may have been able to be improved up though the imputation of the missing markers using a haplotype reference panel. Ideally this analysis would have had a higher marker density, however it was decided that it was best to remain unimputed due to both the haplotype uncertainty at the phasing step and the Polynesian populations not being represented in the reference panel. An example of why having Polynesian haplotypes represented is the finding of @Minster2016, where the variant had no appreciable allele frequency outside of Polynesia. Implications from the marker density used are the number of \gls{snp}s per window was lower than in other studies and specifically when looking at the degradation of haplotypes there is the possibility for early truncation due to lack of data.

The method itself of using the empirical distribution and selecting the extremes for a statistic to determine the region under selection does have precedence [@voight2006map; @Hider2013; @Jonnalagadda2017], however, this does assume that selection had occurred. The underlying null distribution of the genome is also unknown. It does need to be acknowledged that using the extremes of the distribution is likely to lead to many false positives [@Teshima2006]. This was somewhat mitigated in this study by using the permutations of the actual data to calculate the expected \gls{fdr}.

Allele frequency methods, and also the haplotypic methods may get confounded by the gout prevalences of the Polynesian sample sets, especially the \gls{cim} and \gls{ton} sets, where the prevalence is up to five-fold higher than the general population prevalence for those populations. Loci that showed evidence in multiple populations that did not have this enrichment for the disease trait would be expected to have more credibility. This is important to consider to for the co-morbidities of gout too, given the commonalities in associated genes.

One of the limitations of using the \gls{gwas} catalog to create gene lists for conditions is that there is a heavy over representation of genes that are associated to disease in Europeans due to the overall bias to \gls{gwas} research being conducted in European populations [@Haga2010; @Popejoy2016]. This can mean that population-specific associations are missed [@Weissglas-Volkov2013; @Minster2016]. Genes that are associated with a condition can also be missed if they do not reach genome-wide significance in a \gls{gwas} but in a candidate gene approach are significant for instance _ADCY9_ being nominally significant for \gls{bmi} at a genome-wide level but only significant for malaria in a candidate gene approach. 

<!-- One of the methods to still use the selection and neutrality tests with the genome-wide data was to use the empirical distribution [@Teshima2006; @Clarke?] -->


<!-- - imputation from the chip might not work well given the polys aren't (unconfirmed) well represented in the reference panel -->

<!-- - larger genes had a higher likelihood of being 'interesting' due to having the chance of a window intersecting it -->



### Conclusion

The results of this study show there is indeed evidence indicative of selection in urate-associated loci in the Polynesian populations. This evidence however is not specific to urate. This is shown by the fact that there is little signal of selection appearing at the urate transporters themselves, but at the loci that are part of more general pathways. The reasons for why there would be selection still remain unclear but response to pathogenic challenge such as malaria is a potential mechanism.


<!-- #### possibly useful papers -->

<!-- - interpreting pop diff in terms of drift and selection - Cabe1994 -->
<!-- - When Does Frequency-Independent Selection Maintain Genetic Variation? - Novak2017 -->
<!-- - Toward a new history and geography of human genes informed by ancient DNA - Pickrell2017 -->
<!-- - Genetic Drift in an Infinite Population: The Pseudohitchhiking Model - Gillespie2000 -->
<!-- - Detecting ancient positive selection in humans using extended lineage sorting - Peyregne2017 -->
<!-- - A New Test for Detecting Recent Positive Selection that is Free from the Confounding Impacts of Demography - Li2010 -->
<!-- - Understanding rare and common diseases in the context of human evolution - Quintana-Murci2016 -->
<!-- - Soft sweeps are the dominant mode of adaptation in the human genome - Schrider2017 -->
<!-- - Demography-adjusted tests of neutrality based on genome-wide SNP data - Rafajlovic2014 -->
<!-- - Decomposing the Site Frequency Spectrum: The Impact of Tree Topology on Neutrality Tests - Ferretti2017 -->
<!-- - Statistical Power Analysis of Neutrality Tests Under Demographic Expansions, Contractions and Bottlenecks With Recombination - Ramirez-Soriano2008 -->
<!-- - Genetic loci associated with coronary artery disease harbor evidence of selection and antagonistic pleiotropy - Byars2017 -->
<!-- - Demographic history, selection and functional diversity of the canine genome - Ostrander2017 -->
<!-- - Admixture, Population Structure, and F-Statistics - Peter2016 -->
<!-- - Population History and Natural Selection Shape Patterns of Genetic Variation in 132 Genes - akey2004population -->
<!-- - Determinants of genetic diversity - Ellegren2016 -->
<!-- - Reconstituting the Frequency Spectrum of Ascertained Single-Nucleotide Polymorphism Data - Nielsen2004 -->
<!-- - Correcting for ascertainment biases when analyzing SNP data: applications to the estimation of linkage disequilibrium - Nielsen2003 -->

<!--
gene lists
metabolic - covered already
immune - innate: crohns, bohcets, coeliac
immune - adaptive: t1d, RA, MS, graves, hushimotosis
cancer
pyschicatric: SZ
neurological: AD, PD
-->


<!--
## SelectionTools pipeline

for the selection pipeline we need to split into populations

```{r, engine = 'bash', eval = FALSE}
# working dir: /Volumes/BiochemXsan/scratch/merrimanlab/murray/working_dir/coreExome_selection/NZ_coreExome_1kgp/data/
# make a combined panel file
cut -f1-3 /Volumes/BiocArchive/archive/merrimanlab/reference_files/VCF/1000Genomes_vcf_files/Phase3_March2017/integrated_call_samples_v3.20130502.ALL.panel |\
cat - NZ_coreExome/*panel > nz_1kgp.panel

mkdir Indiv_pops

parallel '
  bcftools view \
    -s $(grep {1} nz_1kgp.panel |\
      cut -f 1 |\
      tr "\n" "," |\
      sed "s/,$//g") \
    -O z \
    -o Indiv_pops/{1}.chr{2}.phased.vcf.gz NZ_1KGP.chr{2}.phased.sample_updated.vcf.gz
  ' ::: $(cut -f 2 nz_1kgp.panel |grep -v "pop" | sort |uniq) ::: $(seq 1 22)
```

Break into super populations to analyse the Polynesians (NZM,CIM,TON,SAM) - make sure to exclude the NZC from the EUR super pop and also exclude the admixed NZM (NAD), Eastern Polynesians (EPN), and Western Polynesians (WPN).

```{r, engine = 'bash', eval = FALSE}
# working dir: /Volumes/BiochemXsan/scratch/merrimanlab/murray/working_dir/coreExome_selection/NZ_coreExome_1kgp/data/
mkdir Super_pops

parallel '
  bcftools view \
    -s $(grep {1} nz_1kgp.panel |\
      grep -v "NAD\|EPN\|NZC\|WPN" |\
      cut -f 1 |\
      tr "\n" "," |\
      sed "s/,$//g") \
    -O z \
    -o Super_pops/{1}.chr{2}.phased.vcf.gz NZ_1KGP.chr{2}.phased.sample_updated.vcf.gz
  ' ::: $(cut -f 3 nz_1kgp.panel |grep -v "pop" | sort |uniq) ::: $(seq 1 22)
```

Now send to NeSI pan cluster for the selectionTools pipeline to take place

Individual populations are first run through selectionTools with iHS, nSL, and XP-EHH calculated later

```{r, engine = 'bash', eval = FALSE}
# on nesi
mkdir -p /home/murray.cadzow/uoo00008/NZ_1KGP_unimputed/{Indiv_pops,Indiv_pops_results}

# locally
parallel '
  scp {} murray.cadzow@login.uoa.nesi.org.nz:~/uoo00008/NZ_1KGP_unimputed/Indiv_pops/ 
  ' ::: $(ls /media/xsan/scratch/merrimanlab/murray/working_dir/coreExome_selection/NZ_coreExome_1kgp/data/Indiv_pops/*gz)

# on nesi
# working dir: /home/murray.cadzow/uoo00008/NZ_1KGP_unimputed/Indiv_pops

# find all the vcfs and use the first prefix to use as the population code
for p in $(ls * | cut -d'.' -f1 | sort | uniq)
do 
  sbatch ../unimputed_selection_pipeline.sl $p ; sleep 2; 
done
```

Then run run_selscan.sl and run_nsl_selscan on each population results file
```{r, engine = 'bash', eval = FALSE}
# on nesi
# working dir: Indiv_pops_results
for p in $(find ./[A-Z]* -type d | grep -v xpehh | sed 's/\.\///g')
do 
  cd $p 
  sbatch -J ihs_$p ../../run_selscan.sl $p
  sbatch -J nsl_$p ../../run_nsl_selscan.sl $p
  cd ../
  sleep 5
done
```

```{r, engine = 'bash', eval = FALSE}
# on nesi
# working dir: /home/murray.cadzow/uoo00008/NZ_1KGP_unimputed/Indiv_pops_results
# individual pops
for p1 in CIM NZC NZM SAM TON
do
  for p2 in ACB ASW BEB CDX CEU CHB CHS CIM CLM EPN ESN FIN GBR GIH GWD IBS ITU JPT KHV LWK MSL MXL NAD NZC NZM PEL PJL PUR SAM STU TON TSI WPN YRI
  do
    if [[ $p1 != $p2 ]]
    then
      echo $p1 $p2
      sbatch ../run_xpehh.sl $p1 $p2
      sleep 5
    fi
  done
done


for p1 in EPN NAD WPN
do
  for p2 in ACB ASW BEB CDX CEU CHB CHS CIM CLM EPN ESN FIN GBR GIH GWD IBS ITU JPT KHV LWK MSL MXL NAD NZC NZM PEL PJL PUR SAM STU TON TSI WPN YRI
  do
    if [[ $p1 != $p2 ]]
    then
      echo $p1 $p2
      sbatch ../run_xpehh.sl $p1 $p2
      sleep 5
    fi
  done
done




# super pops
for p in EUR EAS SAS AMR AFR
do
  sbatch ../run_xpehh.sl POL $p
  sleep 5
done

# Don't actually want the comparisons between NAD and EPN or WPN, or the ones for EPN/WPN with NZM/CIM/SAM/TON
rm -r NAD_{WPN,EPN}
rm -r {EPN,WPN}_{NZM,CIM,SAM,TON}
```
-->

<!-- ## UNUSED CODE -->



```{r, include = FALSE, eval = FALSE}
# calculate mean number of snps, total number of clusters, and mean width of clusters by
# population after filtering out single base clusters
ihs_clus %>%select(pop, chrom, ClusterID, chrom_start, chrom_end) %>% distinct() %>% group_by(pop, chrom, ClusterID) %>% summarise(start = min(chrom_start), end = max(chrom_end), snps = n()) %>% distinct() %>% mutate(width = (end - start)/1e6, snp_per_mb = snps/width) %>% filter(!(end - start == 1 & snps == 1)) %>% group_by(pop) %>% summarise(ms = mean(snps), n= n(), mw = mean(width)) %>% arrange(desc(ms, mw))


nsl_clus %>%select(pop, chrom, ClusterID, chrom_start, chrom_end) %>% distinct() %>% group_by(pop, chrom, ClusterID) %>% summarise(start = min(chrom_start), end = max(chrom_end), snps = n()) %>% distinct() %>% mutate(width = (end - start)/1e6, snp_per_mb = snps/width) %>% filter(!(end - start == 1 & snps == 1)) %>% group_by(pop) %>% summarise(ms = mean(snps), n= n(), mw = mean(width)) %>% arrange(desc( ms))
```




```{r, eval = FALSE}
# number of snps that were in the urate and related diseases that were significant in Polynesian populations
bind_rows(sig_ihs_val %>% 
            mutate(statname = 'ihs'), sig_nsl_val %>% 
            mutate(statname = 'nsl')) %>% 
  filter(pop %in% poly_pop, genename %in% c(urate_goutGR$SYMBOL, t2d_GR$SYMBOL, obesity_GR$SYMBOL, metsyn_GR$SYMBOL)) %>% 
  mutate(pop = paste(statname,pop)) %>% 
  group_by(genename, pop) %>% 
  tally() %>% 
  arrange(desc(n)) %>% 
  spread(pop, n, fill = "") %>% 
  kable(., longtable = TRUE, caption = '(\\#ref:iHSnSLPoltable) The number of significant SNPs across associated genes for Polynsians populations for iHS and nSL')
```

```{r n_ihs_gout_snps, include = FALSE}
# calculate the number of sig snps for ihs/nsl for all metabolic related genes for Polynesians exclusively
bind_rows(sig_ihs_val %>% 
            mutate(statname = 'ihs'), sig_nsl_val %>% 
            mutate(statname = 'nsl')) %>%
  filter(pop %in% poly_pop, genename %in% c(urate_goutGR$SYMBOL, t2d_GR$SYMBOL, obesity_GR$SYMBOL, metsyn_GR$SYMBOL)) %>% 
  mutate(pop = paste(statname,pop)) %>% 
  group_by(genename, pop) %>% tally() %>% 
  arrange(desc(n)) %>% 
  anti_join(., bind_rows(sig_ihs_val %>% # remove genes that are found in non-Pol populations
                           mutate(statname = 'ihs'), sig_nsl_val %>% 
                           mutate(statname = 'nsl')) %>% 
              filter(!pop %in% poly_pop) %>% 
              select(pop, genename), by = c('pop', 'genename') ) %>% 
  spread(pop, n, fill = '') %>% 
  kable(., caption = 'Disease genes with significant hit in only Polynesian populations from metabolic lists', longtable = TRUE)
```

```{r markdown3, echo = FALSE, eval = FALSE}
# All poly populations

_Tajima's D_

With the populations that made up the \gls{pol} super population there were `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower tail significance threshold, these windows formed `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions totaling `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. These regions contained `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, `r lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM','NZM','SAM','TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Tajima.D', !pop %in% c('CIM','NZM','SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant with the \gls{pol} populations with \gls{td}. In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that reached significance and made `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions covering a total of `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. There were `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes with `r upper_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM','NZM','SAM','TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Tajima.D', !pop %in% c('CIM','NZM','SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` that were only found in the \gls{pol} populations for \gls{td} within these regions.

_Fay and Wu's H_

Within all of the Polynesian populations used, there were  `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM','NZM','SAM','TON')) %>% NROW()` windows that met the significance threshold in the lower tail. These windows formed `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions, and were at total of `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. There were `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM','NZM','SAM','TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes that intersected these regions and `r lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM','NZM','SAM','TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fay.Wu.H', !pop %in% c('CIM','NZM','SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes were only significant in the \gls{pol} populations for \gls{fwh}.

For the upper tail there were  `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM','NZM','SAM','TON')) %>% NROW()` windows that were reduced to `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions totaling `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp that met the threshold for significance. In these regions there were `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM','NZM','SAM','TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes and `r upper_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM','NZM','SAM','TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fay.Wu.H', !pop %in% c('CIM','NZM','SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` genes were only significant in the \gls{pol} populations in \gls{fwh}.

_Fu and Li's D_

With the populations that made up the \gls{pol} super population there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower tail significance threshold, these windows formed `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions totaling `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. These regions contained `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, `r lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM','NZM','SAM','TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.D', !pop %in% c('CIM','NZM','SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant with the \gls{pol} populations with \gls{fld}. In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that reached significance and made `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions covering a total of `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. There were `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM','NZM','SAM','TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes with `r upper_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM','NZM','SAM','TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.D', !pop %in% c('CIM','NZM','SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` that were only found in the \gls{pol} populations for \gls{fld} within these regions.

_Fu and Li's F_

With the populations that made up the \gls{pol} super population there were `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower tail significance threshold, these windows formed `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions totaling `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. These regions contained `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM','NZM','SAM','TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, `r lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM','NZM','SAM','TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.F', !pop %in% c('CIM','NZM','SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant with the \gls{pol} populations with \gls{flf}. In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that reached significance and made `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions covering a total of `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. There were `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM','NZM','SAM','TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes with `r upper_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM','NZM','SAM','TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Fu.Li.F', !pop %in% c('CIM','NZM','SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` that were only found in the \gls{pol} populations for \gls{flf} within these regions.

_Zeng's E_

With the populations that made up the \gls{pol} super population there were `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that met the lower tail significance threshold, these windows formed `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` independent continuous regions totaling `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. These regions contained `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM','NZM','SAM','TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes, `r lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM','NZM','SAM','TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Zeng.E', !pop %in% c('CIM','NZM','SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` were only significant with the \gls{pol} populations with \gls{ze}. In the upper tail there were `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% NROW()` windows that reached significance and made `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% length()` regions covering a total of `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM','NZM','SAM','TON')) %>% select(contains('chrom')) %>% distinct() %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% GenomicRanges::width() %>% sum()/1e6` Mbp. There were `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM','NZM','SAM','TON')) %>% select('SYMBOL') %>% distinct() %>% NROW()` genes with `r upper_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM','NZM','SAM','TON')) %>% anti_join(., bind_rows(lower_sig_stats, upper_sig_stats) %>% filter(statname == 'Zeng.E', !pop %in% c('CIM','NZM','SAM','TON')), by = "SYMBOL") %>% select(SYMBOL) %>% distinct() %>% NROW()` that were only found in the \gls{pol} populations for \gls{ze} within these regions.
```