# Clustering of Selection Statistics {#clustering}

\glsresetall

<!--
the upper_sig_stats and  lower_sig_stats have a window size of 100 kb so therefore the genes that it says are in those windows might not apply to the 10 kb windows used for the clustering

-->

```{r, include = FALSE, echo = FALSE}
knitr::opts_chunk$set(echo = FALSE)
knitr::opts_chunk$set(cache = FALSE)
library(knitr)
library(kableExtra)
library(tidyverse)
library(ggdendro)
library(ggpubr)
library(phdFunctions)
source('scripts/multiplot.R')
source('scripts/Annotate_genes.R')
source('scripts/04_clustering.R')

```

<!--
Include interpretation in each of the sub-sections

key tasks to complete:
- update introduction WRT embedded comments
- df with summary stats about number of windows for each population
- df with distribution of treecuts proportions
- genes in the tails
-->


This chapter investigates the idea that populations with shared ancestry will display similar signals of selection. To test this idea I will be clustering the various selection statistics and then comparing the clusters to current population migratory history.

## Introduction {#chap4Intro}

<!--
* introduce the out of africa history
* cover the pacific migrations
* Near versus Remote Oceania
* migration to NZ from pacific islands
-->

“Signatures of selection” in a population can be identified in regions of the genome that exhibit a reduction in genetic variability [@Smith1974;@Kaplan1989;@McVean2007]. This reduction in genetic variation can arise when the phenotype of a neutral benefit allele experiences a favourable change in environmental conditions, or a new allele arises conveying a selective advantage [@Hermisson2005]. This results in an increased frequency of both the allele, and linked sites, within a population [@Smith1974]. Genome-wide scans for signatures of selection have shown that  geographically similar populations share similar genetic signatures [@Coop2009; @pickrell2009signals]. Geography can inhibit migration, leading to local adaptations affecting the allele frequency of local populations, leaving the non-local population allele frequency unaffected [@Coop2009]. New Zealand Polynesians are relatively geographically isolated, with a recent settlement history, this means that any signatures of selection found, and not shared with other populations are likely to be a recent and localised adaptation. 

<!-- need to update with Lisa's comments which were
- remote oceania is beyond solomon islands
- need a sentence to explain why we have Polynesians living in NZ -> ref the chapter about 1960's pacific immigrants to NZ
- explain NZ popln of Maori and Polynesian populations
- mention CIM settlement
- explain east/west split
-->

The history of Polynesian migration starts with migration out of Africa (50-100 kya, @Nielsen2017), tracking up to the Levant and Arabian peninsula (45-55 kya) and splitting into South Asia, Indonesia, and Australia (50kya, @Kivisild1999; @Quintana-Murci1999). Concurrently the migration  to Europe was occurring (45 kya). From South Asia the migration continued with the peopling of East Asia (20kya, @Groucutt2015). <!--Lisa: The Pacific – or if you really are talking about Remote Oceania, you are talking about Lapita, which first appears in Near Oceania about 3300BP where they interact with the peoples who have been there for at least 40,000 years and then by 3,000 they are the first to establish settlement on the islands of New Caledonia, Vanuatu, Fiji, Samoa and Tonga. THEN around 1-1.2 kya we have the settlement of the rest of the Polynesian triangle, including the Cook Is avout 1,000 BP and New Zealand about 750 - 800 BP. --> The Pacific was settled in two events, the first was the settlement of Near Oceania (40 kya), this was followed by the settlement of Remote Oceania, reaching Western Polynesia (Samoa and Tonga) during the Lapita expansion (3 kya, @Matisoo-Smith2015; @Skoglund2016). Eastern Polynesia (including the Cook Islands) was then settled (1–1.2 kya, @Wilmshurst2011). The final region of the Polynesian triangle to be settled was that of \gls{nz} by \gls{nz} M\tex{\={a}}ori, occurring 800 ya [@Duggan2014; @Matisoo-Smith2015]. It is from this migration history that modern Polynesian populations come to have a shared common ancestry with modern East Asian populations.

<!--
need something about what consitutes east and west polynesia and why that matters - doone in above and following paras
-->

For largely economic reasons, during the 1950's and 60's there was a rural to urban migration of people in the Pacific. As part of this, many individuals and families settled in New Zealand from Samoa, Tonga, and the Cook Islands in search of work, and now the populations in New Zealand largely outnumber the populations of their islands of origin [@MatisooSmith2012highway]. Within \gls{nz}, the Polynesian populations are mostly focused within the Auckland region [@Barcham2009].

<!--ref for pacific in NZ in 1960's [@MatisooSmith2012highway] 
The modern population of \gls{nz} Maori - done

include paragraph about maori and pacific health in general in new zealand - done

@Hawley2015 @moh

-->

New Zealand M\tex{\={a}}ori and Pacific Islanders in New Zealand over the past decade have been exhibiting a decrease in overall health [@MinistryofHealth2016]. Fruit and vegetable intake, and physical activity has also been reported as decreasing, all the while there has been an increase in obesity, which has an adjusted rate ratio of 1.7 for  M\tex{\={a}}ori versus non- M\tex{\={a}}ori, and 2.4 for Pacific versus non-Pacific [@MinistryofHealth2016]. The rate of increase in \gls{bmi} for Pacific Islanders was five times that of the rest of the world, and fasting plasma glucose was three to four times higher over the 1980-2008 period [@Hawley2015]. The health disparities are captured in the New Zealand statistics where M\tex{\={a}}ori have a mortality rate of 1.8  that of non-M\tex{\={a}}ori, with the rates of obesity and diabetes of particular note, with the M\tex{\={a}}ori standardised rate of 40.6 per 100,000, five times higher than non-M\tex{\={a}}ori of 8.1 per 100,000 [@MinistryofHealth2012].

New Zealand Polynesians have inherent elevated serum urate levels and higher prevalence of gout [@Winnard2012; @Winnard2013] and there are genetic variants in Polynesian populations associated with increased risk to gout and elevated serum urate [@Phipps-Green2010; @Hollis-Moffatt2012]. Serum urate has been associated with metabolic disorders [@Choi2007;@Choi2007b] and genes involved with complex diseases such as \gls{t2d}, gout, and other metabolic related disorders, have shown evidence for selection [@Zhang2013a; @pickrell2009signals;@Hancock2008]. 

It is hypothesised that elevated serum urate may have undergone positive selection in Polynesian populations due to some of the beneficial properties [@Gosling2014], such as its role as a powerful anti-oxidant [@Ames1981], or as an adjuvant for the innate immune system [@Opitz2009]. Serum urate has also been suggested to have a protective neurological effect against dementia, and provides improved cognitive function with age [@Euser2009]. Characterisation of selection within Polynesian populations has been largely limited to population differentiation due to population sample sizes [@Kimura2008; @Mallick2016].

<!-- tie in urate/gout/obesity/t2d so that it makes sense to use those gene lists -->

Here I investigate genetic selection of serum urate and related co-morbidities in Polynesian populations and attempt to determine whether there is a selection signature shared among Polynesian populations.

### Objectives

 - Determine if neutrality and selection tests can be used to group populations.
 - Investigate genetic loci associated with gout, urate, \gls{t2d}, obesity, kidney disease, and metabolic syndrome and their potential shared selective histories.

## Methods


<!--
Important distinction is that this used the 1% threshold NOT the 1% and >/< 0 to select the windows to be used
-->

### Data

The dataset used in this chapter is the selection dataset consisting of 31 population groups, split into six super populations, created from marker matching the Polynesian populations genotyped on the CoreExome \gls{snp} array, with the \gls{1kgp} Phase 3 dataset (section \@ref(selectionDataset)).

### Principal components analysis

\Glsdesc{pca} was performed on the independent genetic markers from the selection dataset (for more detail refer section \@ref(pca)). The first 10 components were calculated and plotted.   

### Admixture analysis

Admixture analysis was performed on the independent genetic markers for the selection dataset. Admixture proportions were calculated using all of the populations in the selection dataset and cross-validation was used to identify the K value with the lowest error for values of K from 1 to 15. Proportions were also calculated by first removing the \gls{amr} and \gls{pol} populations, calculating the K value that provided the lowest error, through cross-validation, and then projecting these ancestral populations onto the \gls{amr} and \gls{pol} populations, again for values of K from 1 to 15. For futher information refer to section \@ref(admixture).


### Frequency spectrum

\Gls{td}, \gls{fwh}, \gls{flf}, \gls{ze}, and F~ST~ were all calculated using PopGenome v2.2.3 [@Pfeifer2014] for each chromosome. Aside from F~ST~, these statistics were also calculated as per section \@ref(popgenomeMethods) to generate non-overlapping windows of 10 kb. For each selection test statistic, windows were ordered by test statistic value, and the 1^st^ and 99^th^ percentiles of the distribution were calculated. For the windowed  approach, hierarchical clustering was then performed on a binary matrix derived from the presence or absence of the windows in the 1^st^ or 99^th^ percentiles, across all populations. Hierarchical clustering using the value calculated for each whole chromosome (i.e., treating the start and end of the chromosome as the window boundary) used the statistic value itself to create the distance matrix. The distance measure used was Euclidean distance and the linkage criteria was "complete linkage" for both sets of clustering. Accuracy of the clustering was assessed by descending the dendrogram until six groups, one for each super population, could be made, and then calculating the maximum proportion of the individual populations of the same super population assigned to the same cluster. Exclusivity of each cluster was calculated using the proportion of populations from a given super population compared to the total number of individual populations assigned that cluster. For example, if a cluster consisted of eight populations, two from super population A and six from super population B, then the exclusivity of the cluster would be 0.25 for super population A, and 0.75 for super population B. F~ST~ was calculated pair-wise for all populations for each chromosome.

### Extended haplotype homozygosity {#haplocluster}

For all \glspl{snp}, \gls{ihs} and \gls{nsl} were calculated and normalised using selscan v1.1.0b [@Szpiech2014] as part of the selectionTools 1.1 [@Cadzow2014] pipeline, as described in Section \@ref(selectionTools). Markers with an absolute \gls{ihs} or \gls{nsl} value greater than 2.6 met the significance threshold, and were clustered into genomic regions using the DBSCAN package v1.1.1 [@dbscanref] in R, where nearby \glspl{snp} were assigned the same group identifier. The search radius used was 200 kb and the minimum number of points for a cluster was one. Cluster regions were created by taking both the minimum and maximum position for each group identifier, population and chromosome. The distance matrix was the result of calculating (A $\cap$ B)/ A and (A $\cap$ B) / B, where A $\cap$ B were the sum of the intersection of clustered regions between population A and population B, and A and B was the sum of the clustered regions for population A and population B respectively. Hierarchical clustering was then performed, using complete linkage for the linkage criteria to create two sets of clusters (for an example see Figure \@ref(fig:ihsclust)).

### Pathway enrichment analysis

Pathway enrichment analysis was performed by taking genes that intersected the clustered \gls{ihs} and \gls{nsl} regions, inputting them into Enrichr^[http://amp.pharm.mssm.edu/Enrichr/ accessed 26 February 2018] [@Chen2013b; @Kuleshov2016] and exporting the KEGG 2016 pathway table.



### Disease-associated genes

Disease associated markers and their annotated genes were downloaded from the \gls{gwas} catalog^[https://www.ebi.ac.uk/gwas/ accessed 19 June 2017] [@MacArthur2017] and filtered for association (chi-squared P < 5x10^-8^). Gene lists were then created for each of the following diseases or traits; urate and gout, \gls{t2d}, obesity, kidney disease, and metabolic syndrome by filtering the study disease trait on keywords (see section \@ref(diseaselist) for details). Table \@ref(tab:gwascatref) shows source disease trait and reference used to create categories of gene lists for gout/urate, obesity, \gls{t2d}, kidney disease, and metabolic syndrome. Selection test statistics values for each population were centred using the population median. The gene lists from these categories were used in Section \@ref(genelistcluster) for hierarchical clustering using the selection test statistic values for windows that intersected these gene regions. Complete linkage and Euclidean distance were used for the clustering.

(ref:gwascataccess) https://www.ebi.ac.uk/gwas/, accessed 19 June 2017


## Results

### Frequency spectrum


#### PCA analysis

<!-- > talk about the nice groupings from the pca -->

<!-- > need to explain why K = 6 for the rest of the analysis -->

The genetic variation captured by \gls{pca} was plotted to visualise the relationship of individuals to one another (Figure \@ref(fig:allpopPCA)). Starting from \gls{pc} 1, each subsequent \gls{pc} captured less of the variation than the previous. \Glsentrydescplural{pc} for the merged \gls{1kgp} data and the New Zealand samples showed in the first two components (which capture the most variation) that the \gls{afr}, \gls{eur}, \gls{sas} super populations all formed separate groupings. \Gls{pc} 1 was responsible for separating the \gls{afr} populations from the others, and \gls{pc} 2 was largely responsible for separating the \gls{eur} and \gls{sas} from each other, and from the \gls{eas} and \gls{pol} populations. The \gls{eas} and \gls{pol} populations were extremely similar in \gls{pc} 1 but the \gls{pol} populations were spread with \gls{pc} 2, with the majority of the spread being accounted for by the Eastern Polynesian populations whereas the Western Polynesian populations had a tight grouping at -0.025 on \gls{pc} 2, slightly below the \gls{eas} group. This can be observed when \gls{pc} 6 is used to separate the Polynesian populations (Figure \@ref(fig:polPCA)). \Gls{pc} 3 separated the \gls{pol} populations from the \gls{eas} populations (Figure \@ref(fig:allpopPCA) B) and \gls{pc} 4 separated the \gls{amr} and \gls{sas} populations (Figure \@ref(fig:allpopPCA) C). The admixture of the \gls{amr} populations was evident in \gls{pc} 1 with a large spread centered with the \gls{eur} populations. The first four \gls{pc} indicated that the six super populations could form super population clusters based on the genetic variation.

(ref:allpopPCA) A) Principal components 1 and 2 for all populations. B) Principal components 1 and 3 for all populations. C) Principal components 1 and 4 for all populationsColoured by super population grouping.

```{r allpopPCA, fig.cap = '(ref:allpopPCA)', echo = FALSE, fig.height = 7, fig.width=5, fig.align='center'}
eigen <- read.table('~/data/NZ_coreExome_1kgp/PCA/indiv_pops/NZ_1KGP_allchr_indep_eigen.pca.evec', stringsAsFactors = FALSE)

eigen <- eigen %>% select("sample" = V1, everything()) %>% left_join(., panel, by = 'sample') %>% filter(!pop %in% c('WPN','EPN','NAD'))
names(eigen)[2:11] <- paste0("PC",1:10)
pc12 <- eigen %>% ggplot(., aes(x = PC1, y = PC2, colour = super_pop)) + geom_point() + theme_bw() + guides(colour = guide_legend(title = "Super Population", ncol = 6, title.position = 'top')) + theme(legend.position = 'bottom', axis.text.x = element_blank(), axis.title.x = element_blank())
pc13 <- eigen %>% ggplot(., aes(x = PC1, y = PC3, colour = super_pop)) + geom_point() + theme_bw() + theme(legend.position = 'none', axis.text.x = element_blank(), axis.title.x = element_blank())
pc14 <- eigen %>% ggplot(., aes(x = PC1, y = PC4, colour = super_pop)) + geom_point() + theme_bw()+ theme(legend.position = 'none') + scale_y_continuous(breaks = c(-0.06, -0.04,-0.02, 0, 0.02, 0.04, 0.06))
leg <- cowplot::get_legend(pc12)
top_row <- cowplot::plot_grid(pc12 + theme(legend.position = 'none'), pc13, pc14 , nrow = 3, labels = c("A","B", "C"), align = 'v', rel_heights = c(1,1,1.2))
cowplot::plot_grid(top_row, leg, nrow=2, ncol = 1, rel_heights = c(1,.1))
```

(ref:eurPCA) Principal components 1 and 2 for populations of the European Super Population.

```{r eurPCA, fig.cap = '(ref:eurPCA)', echo = FALSE, fig.width = 5, fig.height = 3}
#eigen %>% filter(super_pop == 'EUR') %>% ggplot(., aes(x = PC1, y = PC2, colour = pop)) + geom_point() + theme_bw() + guides(colour = guide_legend(title = "Population"))

eigen %>% filter(super_pop == 'EUR') %>% ggplot(., aes(x = PC1, y = PC2, shape = pop, colour = super_pop)) + geom_point(size = 2) + theme_bw() + scale_color_manual(values = c(EUR = '#00BFC4')) + guides(shape = guide_legend(title = "Population"), colour = guide_legend(title = "Super Population")) + scale_shape_manual(values = c(CEU = 0, NZC = 1, GBR = 2, IBS = 3, FIN = 4, TSI = 5))
```

(ref:polPCA) Principal components 2 and 6 for the Polynesian populations.

```{r polPCA, fig.cap = '(ref:polPCA)', echo = FALSE, fig.width = 5, fig.height= 3}
eigen %>% filter(super_pop == 'POL') %>% ggplot(., aes(x = PC6, y = PC2, shape = pop, colour = super_pop)) + geom_point(size = 2) + theme_bw() + scale_color_manual(values = c(POL = '#619CFF')) + scale_shape_manual(values = c("CIM" = 0,"NZM" = 1, "TON" = 2, "SAM" = 3)) + guides(size = FALSE) + guides(shape = guide_legend(title = "Population"), colour = guide_legend(title = "Super Population"))
```

There was no distinction between the \gls{ceu}, \gls{gbr}, and \gls{nzc} populations (Figure \@ref(fig:eurPCA)), this indicated that the use of \gls{snp} array versus sequence data wasn't a major source of variability in the dataset. In the combined dataset the Polynesian populations were split into Eastern and Western Polynesian sub-groups with \gls{pc} 6 (Figure \@ref(fig:polPCA)).


\FloatBarrier

#### Admixture analysis

<!-- > what does the admixture analysis add? -->

<!-- > K = 11  from the use all data in the cross validation -->

<!-- > K = 9 in the remove the admixed populations and use the ancestral proportions to work out the proportions in the admixed pops -->




Admixture analysis is a method that can be used to infer the number of ancestral populations that contribute to current populations by modelling the probability of the observed genotypes using proportions of ancestral populations and the population allele frequencies [@Alexander2009]. Proportions of theoretical ancestral populations for each population were calculated, using all of the populations to establish the number of ancestral populations (K), that resulted in the smallest cross-validation error. Five-fold cross-validation was used to calculate this error for K from 1 to 15. The K value with the lowest cross-validation error was K = 11, with an error of 0.3855 (Table \@ref(tab:admixCV)).  



When the ancestral proportions for K = 11 were plotted for each individual for each population, there were some clear similarities amongst the super populations (Figure \@ref(fig:admix)). Within the \gls{afr} populations there were three main ancestral populations, which correlated with geography, with the East African population of \gls{lwk} being most different to the West African populations (\gls{esn}, \gls{yri}, \gls{gwd}, and \gls{msl}). The populations of \gls{acb} and \gls{asw} showed signs of admixture with similarities to the ancestral populations in the West African populations, with smaller proportions from the ancestral populations of \gls{lwk}, and the European populations. The \gls{eas} populations were represented by two ancestral populations, \gls{cdx} was mostly one, and \gls{jpt} mostly the other, the remaining \gls{eas} populations were mixtures of the two. A single ancestral population represented \gls{sas}, but there were indications of admixture with the ancestral populations of the \gls{eur} and \gls{eas} populations. In the \gls{eur} populations there was again two main ancestral populations, they correlated with North and South Europe. One ancestral population was mostly in \gls{fin}, and the other mostly in \gls{tsi} and \gls{ibs}. \gls{gbr}, \gls{ceu}, and \gls{nzc} were all extremely similar in proportions of the two and ancestral populations. The \gls{amr} populations were mostly differing proportions of the ancestral population that \gls{tsi} had, and an ancestral population that was only in the \gls{amr} populations. There were also small proportions of the ancestral populations that made up the West African populations. The \gls{pol} populations had two ancestral populations, one found in the East Polynesian populations, and the other in the West Polynesian populations. All Polynesian populations had indications of small amounts of admixture with the ancestral populations of the \gls{eur} populations, with the East Polynesian populations having a higher amount than the West Polynesian populations. This analysis was consistent with the results of the \gls{pca}, but also showed that there were sub-groups within the super-populations.




```{r admixData, echo = FALSE}
ref <- read.table('~/data/NZ_coreExome_1kgp/admixture/NZ_1KGP_allchr_admix.11.Q', stringsAsFactors = FALSE)
ref_fam <- read.table('~/data/NZ_coreExome_1kgp/admixture/NZ_1KGP_allchr_admix.fam', header = FALSE, stringsAsFactors = FALSE)
names(ref_fam) <- c('FID','sample','PID','MID','SEX','AFF')
ref_fam <- left_join(ref_fam, panel %>% filter(!pop %in% c('NAD','EPN','WPN','POL')), by = 'sample')

```


(ref:admix) Proportions of ancestral populations as inferred from ADMIXTURE, using K = 11. Ancestral population (AP) represents a particular theoretical ancestral population.

```{r admix, echo = FALSE, fig.height = 9, fig.width = 6, fig.cap = '(ref:admix)'}
# barplot of the reference proportions
cols <- c('#a50026','#d73027','#f46d43','#fdae61','#fee090','#ffffbf','#e0f3f8','#abd9e9','#74add1','#4575b4','#313695')
names(cols) <- paste0("AP", 1:11)
cbind(ref_fam, ref%>% set_colnames(., paste0("AP", 1:NCOL(.))))  %>% gather(., "AP","Q",9:NCOL(.)) %>% mutate(AP = factor(AP, levels = paste0("AP", 1:11), ordered = TRUE)) %>% mutate(super_sample = paste0(super_pop,'_',sample), facet_pop = paste(super_pop, pop)) %>% ggplot(., aes(x = super_sample, y = Q)) + geom_bar(stat = "identity", position = 'stack', aes(fill = AP), width = 1) + facet_wrap(super_pop~pop, scales = "free_x", ncol = 6, shrink = TRUE) + theme_bw() + theme(axis.title.x = element_blank(), axis.text.x = element_blank(), axis.ticks.x = element_blank()) + scale_y_continuous(expand = c(0,0)) + scale_fill_brewer(type='div', palette = "Set3" )
rm(ref,ref_fam)
```








```{r admixDataProjection, echo = FALSE, eval = FALSE}
ref <- read.table('~/data/NZ_coreExome_1kgp/admixture/projected/reference_bootstrap.9.Q', stringsAsFactors = FALSE)
ref_fam <- read.table('~/data/NZ_coreExome_1kgp/admixture/projected/reference.fam', header = FALSE, stringsAsFactors = FALSE)
names(ref_fam) <- c('FID','sample','PID','MID','SEX','AFF')
ref_fam <- left_join(ref_fam, panel %>% filter(!pop %in% c('NAD','EPN','WPN','POL')), by = 'sample')

study <- read.table('~/data/NZ_coreExome_1kgp/admixture/projected/study.9.Q', stringsAsFactors = FALSE)
study_fam <- read.table('~/data/NZ_coreExome_1kgp/admixture/projected/study.fam', header = FALSE, stringsAsFactors = FALSE)
names(study_fam) <- c('FID','sample','PID','MID','SEX','AFF')
study_fam <- left_join(study_fam, panel %>% filter(!pop %in% c('NAD','EPN','WPN','POL')), by = 'sample')
```

```{r, echo = FALSE, fig.height = 9, fig.width = 6, fig.cap = '(ref:admixProj)', eval = FALSE}
#need to modify for K = 9

# barplot of the reference proportions

cbind(ref_fam, ref%>% set_colnames(., paste0("AP", 1:NCOL(.))))  %>% gather(., "AP","Q",9:NCOL(.)) %>% mutate(AP = factor(AP, levels = paste0("AP", 1:9), ordered = TRUE)) %>% mutate(super_sample = paste0(super_pop,'_',sample), facet_pop = paste(super_pop, pop)) %>% ggplot(., aes(x = super_sample, y = Q)) + geom_bar(stat = "identity", position = 'stack', aes(fill = AP), width = 1) + facet_wrap(super_pop~pop, scales = "free_x", ncol = 6, shrink = TRUE) + theme_bw() + theme(axis.title.x = element_blank(), axis.text.x = element_blank(), axis.ticks.x = element_blank()) + scale_y_continuous(expand = c(0,0)) + scale_fill_brewer(type='div', palette = "Set3" )
rm(ref,ref_fam)
```


#### Whole chromosome clustering

Clustering on the full chromosome results for all chromosomes from PopGenome (Figure \@ref(fig:multichrplot)) did not group the populations of each super population together, with the exception of \gls{fwh}, which grouped all super populations, although it split the \gls{amr} populations into two subgroups. \gls{td} grouped the \gls{eur} and most of the \gls{sas} super populations, and had smaller sub-super population groupings of two to three populations. \Gls{cim}, \gls{nzm}, and \gls{sam} were grouped with two \gls{amr} populations, (\gls{pel} and \gls{mxl}), and this group was the least similar to the others. There was no clear pattern to the clustering order of \gls{flf}, although it did have a few small groups of the same super population of no more than three individual populations. The \gls{cim} and \gls{sam} populations were grouped together in both of statistics. \Gls{ze} clustered the \gls{afr} populations together and this cluster was the least similar to the other populations. There was also a grouping of the \gls{eas} populations which had the Western Polynesian populations as part of it.

```{r superGlobalSummary, echo = FALSE, results = 'asis'}
global_super_summary <- readRDS('~/data/NZ_coreExome_1kgp/100kbWindow_intra/filtered/100kbwindows_summary_superpop_popgenome_filtered_3ns_resized.6-11-2017.RDS') %>% filter(stat != 'Fu.Li.D')

```

> need to reference table \@ref(tab:superGlobalSummary) somewhere in this chapter

<!--
- td had eur cluste, nzm-cim, most of a sas, but the other populations are in groups of 2-3, the pol group is quite disimilar to the rest
- fwh has the super populations but has AMR split in half. eas and pol are together and quite disimilar to the rest
- fld looks almost random, nzm-gbr, maybe admixed pops being grouped together
- flf has pockets fo 2-3 small groupings of the super populations, nzm-mxl-pel
- flf and fld have a very similar first split
- ze has afr cluster and is most disimilar. eas cluster has the wp integrated. nzc-clm-pur, eur cluster 
-->

(ref:multichrplot) _Whole Chromosome Clustering._ Hierarchical clustering of the populations using selection test statistics for each entire chromosome. Clustering was performed using Euclidean distance for the distance measure and complete linkage for the linkage criteria. Selection test statistic is in the label for each dendrogram. Coloured by super population.


```{r multichrplot, fig.cap='(ref:multichrplot)', echo = FALSE, fig.height=8}
multiplot(gene_dendros(gene_data = mat_d_list[['td_chr']]) + ggtitle("Tajima's D"),
          gene_dendros(gene_data = mat_d_list[['fwh_chr']]) + ggtitle("Fay and Wu's H"),
          #gene_dendros(gene_data = mat_d_list[['fld_chr']]) + ggtitle("Fu and Li's D"),
          gene_dendros(gene_data = mat_d_list[['flf_chr']]) + ggtitle("Fu and Li's F"),
          gene_dendros(gene_data = mat_d_list[['ze_chr']]) + ggtitle("Zeng's E"),
   cols = 2)
```

\FloatBarrier

#### F~ST~ {#wholeFst}



```{r}
fst <- readRDS('~/data/NZ_coreExome_1kgp/whole_chr_fst_popgenome.3-7-2017.RDS')

fst_mat <- bind_rows(lapply(fst, function(x){x %>% mutate(pop1 = as.character(pop1), pop2 = as.character(pop2))%>% select(-p1,-p2) %>% filter(!pop1 %in% c('WPN','EPN','NAD'), !pop2 %in% c('WPN','EPN','NAD')) %>% left_join(., panel %>% mutate(super_pop = paste(super_pop, pop, sep = '_')) %>% select("pop1" = pop, super_pop) %>% distinct(), by = "pop1" ) %>% select(-pop1) %>% spread(., "super_pop", fst )}))

fst_means <- fst_mat %>% filter(pop2 %in% c('CIM','NZM','SAM','TON')) %>% left_join(., panel %>% mutate(pop2 = pop, p2 = paste0(super_pop,"_",pop)) %>% select(pop2,p2) %>% distinct(), by = "pop2" ) %>% mutate(pop2 = p2) %>% select(pop2, 1:(NCOL(.)-1)) %>% gather("pop1","fst", 3:NCOL(.)) %>%   mutate(sup1 = sapply(pop1, function(x){strsplit(x, '_')[[1]][1]}), sup2 = sapply(pop2, function(x){strsplit(x, '_')[[1]][1]}) ) %>% filter(sup1 != sup2) %>%  group_by(sup2, sup1) %>% summarise(mean = mean(fst) %>% round(., digits = 3)) %>% ungroup()
```


Hierarchical clustering performed on the pair-wise F~ST~ for the whole chromosomes between each individual population grouped all of the super populations, except \gls{amr} into their correct groups (Figure \@ref(fig:wholechrfst)). The \gls{afr} populations were most similar to each other but least similar to the rest of the populations. The Polynesian populations were grouped together and were sub grouped into East and West Polynesian. The largest differentiation was between \gls{cim} and \gls{esn} (mean F~ST~ = 0.203), whereas the smallest differentiation was between \gls{ceu} and \gls{nzc} (mean F~ST~ = 3.42x10^-4^). The mean F~ST~ for \gls{pol} and \gls{afr} was `r fst_means %>% filter(sup2 == "POL", sup1 == "AFR") %>% select(mean) %>% sprintf('%.3f',.)`; \gls{pol} and \gls{amr} was `r fst_means %>% filter(sup2 == "POL", sup1 == "AMR") %>% select(mean) %>% sprintf('%.3f',.)`; \gls{pol} and \gls{eas} was `r fst_means %>% filter(sup2 == "POL", sup1 == "EAS") %>% select(mean) %>% sprintf('%.3f',.)`; \gls{pol} and \gls{eur} was `r fst_means %>% filter(sup2 == "POL", sup1 == "EUR") %>% select(mean) %>% sprintf('%.3f',.)`; \gls{pol} and \gls{sas} was `r fst_means %>% filter(sup2 == "POL", sup1 == "SAS") %>% select(mean) %>% sprintf('%.3f',.)`. This shows that the \gls{pol} populations were least differentiated from \gls{eas} populations and most from the \gls{afr} populations, which is consistent with the migration history.  

(ref:wholechrfst) _F~ST~_ Hierarchical clustering on chromosomal F~ST~ using Euclidean distance and complete linkage. Coloured by super population.

```{r wholechrfst, fig.cap="(ref:wholechrfst)"}
create_fst_dendro() #+ ggtitle(expression("Whole Chromosome F"["ST"]))
#ggsave(last_plot(), file = 'images/04_clustering/fst_dendro.png')
```


<!--
some stats about the largest and smallest fsts and focus on the Polynesians
-->


### Hierarchical clustering of distribution extremes

```{r propData, message = FALSE}
prop_list <- readRDS('~/data/NZ_coreExome_1kgp/100kbWindow_intra/100kbwindows_filtered_3ns_prop_list-31-8-2017.RDS')
prop_list["fld_neg"] <- NULL
prop_list["fld_pos"] <- NULL
prop_list["fld_chr"]<-NULL
load('~/data/gwas_catalog/diseaseGR-12-3-2018.RData')# brings in objects called {gc_urate_gout,urate_gout,kd,metsyn,obesity,t2d}_GR, 
load('~/data/NZ_coreExome_1kgp/100kbWindow_intra/gene_clustering_filtered_3ns-31-8-2017.RData')# brings in objects called gout, obesity, t2d, metsyn, kd
names(gout) <- c('mat_d','prop')
gout$mat_d["Fu.Li.D"] <- NULL
gout$prop["Fu.Li.D"] <- NULL

names(t2d) <-  c('mat_d','prop')
t2d$mat_d["Fu.Li.D"] <- NULL
t2d$prop["Fu.Li.D"] <- NULL

names(obesity) <-  c('mat_d','prop')
obesity$mat_d["Fu.Li.D"] <- NULL
obesity$prop["Fu.Li.D"] <- NULL

names(kd) <-  c('mat_d','prop')
kd$mat_d["Fu.Li.D"] <- NULL
kd$prop["Fu.Li.D"] <- NULL

names(metsyn) <-  c('mat_d','prop')
metsyn$mat_d["Fu.Li.D"] <- NULL
metsyn$prop["Fu.Li.D"] <- NULL
```

Hierarchical clustering of the distribution extremes for each population was done to investigate if there was a similarity between populations and the genetic regions that had the most extreme scores. For the frequency spectrum methods, the 1^st^ and 99^th^ percentiles of windows (i.e., distribution tails) by statistic value were calculated, and a binary matrix of window sharing of significant windows between populations was used for the clustering. For the haplotypic methods, the proportion of regions that overlapped between populations was used to create the distance matrix for the hierarchical clustering. The groups created by the clustering were then assessed for the number of populations from a super population that were assigned the same cluster. Having all populations correctly grouped into their super population was only completely achieved by clustering using the lower tail of \gls{fwh}. If an allowance is made for the two \gls{amr} populations to be grouped with the \gls{eur} populations, as in the chromosomal F~ST~ clustering, then both haplotypic methods and six of the clustered tails of the frequency spectrum based methods clustered populations into their respective super populations.

```{r proportion, results ='asis'}
# prop_list[['td_neg']]$exclusivity %>% data.frame%>% mutate(pop = rownames(.)) %>% gather("cluster", "prop", 1:6) %>% filter(prop > 0) %>% group_by(pop) %>% summarise(mean(prop))
# 
# prop_list[['td_neg']]$super_prop %>% data.frame() %>% spread(super, max_prop) 


bind_rows(list(
  bind_rows(lapply(names(prop_list), function(x){prop_list[[x]]$super_prop %>%  spread(super, max_prop) %>% mutate(stat = sapply(x, function(y){strsplit(y, '_')[[1]][1]}), Type = sapply(x, function(y){strsplit(y, '_')[[1]][2]}) ) %>% select(NCOL(.), 1:(NCOL(.)-1)) })), 
  
  bind_rows(lapply(names(gout[[2]]), function(x){gout[[2]][[x]]$super_prop %>% spread(super, max_prop)%>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1)) })) %>% mutate(Type = 'Urate/Gout'),
  
  bind_rows(lapply(names(t2d[[2]]), function(x){t2d[[2]][[x]]$super_prop %>% spread(super, max_prop)%>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1)) })) %>% mutate(Type = '\\Gls{t2d}'),
  
  bind_rows(lapply(names(obesity[[2]]), function(x){obesity[[2]][[x]]$super_prop %>% spread(super, max_prop)%>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1)) })) %>% mutate(Type = 'Obesity'),
  
  bind_rows(lapply(names(kd[[2]]), function(x){kd[[2]][[x]]$super_prop %>% spread(super, max_prop)%>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1)) })) %>% mutate(Type = 'Kidney Disease'),
  
  bind_rows(lapply(names(metsyn[[2]]), function(x){metsyn[[2]][[x]]$super_prop %>% spread(super, max_prop) %>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1)) })) %>% mutate(Type = 'Metabolic Syndrome')
  
)) %>% mutate(stat = sapply(stat, function(x){short_to_latex(stat_to_short(x))}), 
              Type = sapply(Type, function(x){switch(x, chr = "Chromosome", neg = "Lower Tail", pos ="Upper Tail",x)})) %>% 
  mutate_if(., is.numeric, funs(sprintf('%.2f',.)))%>%
  select(stat, Type, everything()) %>% 
  data.frame %>% 
  arrange(stat) %>%  
  data.frame %>% knitr::kable(., caption = "(\\#tab:proportion) Proportion of populations clustered into their corresponding super population.", booktabs = TRUE, escape = FALSE) %>% footnote(general = "Proportions were calculated independently for each super population based on the cluster group with the maximum number of populations from a particular super population.", general_title = "")
  #column_spec(1, width = "15em")
  #pander::pandoc.table(., caption = "(\\#tab:proportion) Proportion")# %>% kableExtra::group_rows("Fu and Li's D", 1, 7) %>% kableExtra::group_rows("Fu and Li's F", 8, 14) %>% kableExtra::group_rows("Fay and Wu's H", 15, 21) %>% kableExtra::group_rows("Tajima's D", 22, 28) %>% kableExtra::group_rows("Zeng's E", 29, 35) %>% kableExtra::kable_styling(latex_options = "scale_down")

```

```{r exclusivity, results = 'asis'}
#exclusivity
bind_rows(list(
  bind_rows(lapply(names(prop_list), function(x){prop_list[[x]]$exclusivity %>% data.frame%>% mutate(pop = row.names(.)) %>% gather("cluster","prop",1:6) %>% filter(prop > 0) %>% group_by(pop) %>% summarise(mean_prop = mean(prop)) %>% spread(pop, mean_prop) %>% mutate(stat = sapply(x, function(y){strsplit(y, '_')[[1]][1]}), Type = sapply(x, function(y){strsplit(y, '_')[[1]][2]}) ) %>% select(NCOL(.), 1:(NCOL(.)-1)) })),
  
  bind_rows(lapply(names(gout[[2]]), function(x){gout[[2]][[x]]$exclusivity %>% data.frame%>% mutate(pop = row.names(.)) %>% gather("cluster","prop",1:6) %>% filter(prop > 0) %>% group_by(pop) %>% summarise(mean_prop = mean(prop)) %>% spread(pop, mean_prop) %>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1))})) %>% mutate(Type = 'Urate/Gout'),
  
  bind_rows(lapply(names(t2d[[2]]), function(x){t2d[[2]][[x]]$exclusivity %>% data.frame%>% mutate(pop = row.names(.)) %>% gather("cluster","prop",1:6) %>% filter(prop > 0) %>% group_by(pop) %>% summarise(mean_prop = mean(prop)) %>% spread(pop, mean_prop) %>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1))})) %>% mutate(Type = '\\Gls{t2d}'),
  
  bind_rows(lapply(names(obesity[[2]]), function(x){obesity[[2]][[x]]$exclusivity %>% data.frame%>% mutate(pop = row.names(.)) %>% gather("cluster","prop",1:6) %>% filter(prop > 0) %>% group_by(pop) %>% summarise(mean_prop = mean(prop)) %>% spread(pop, mean_prop) %>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1))})) %>% mutate(Type = 'Obesity'),
  
  bind_rows(lapply(names(kd[[2]]), function(x){kd[[2]][[x]]$exclusivity %>% data.frame%>% mutate(pop = row.names(.)) %>% gather("cluster","prop",1:6) %>% filter(prop > 0) %>% group_by(pop) %>% summarise(mean_prop = mean(prop)) %>% spread(pop, mean_prop) %>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1))})) %>% mutate(Type = 'Kidney Disease'),
  
  bind_rows(lapply(names(metsyn[[2]]), function(x){metsyn[[2]][[x]]$exclusivity %>% data.frame%>% mutate(pop = row.names(.)) %>% gather("cluster","prop",1:6) %>% filter(prop > 0) %>% group_by(pop) %>% summarise(mean_prop = mean(prop)) %>% spread(pop, mean_prop) %>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1))})) %>% mutate(Type = 'Metabolic Syndrome')    
)) %>% select(stat, Type, everything()) %>% 
  mutate(stat = sapply(stat, function(x){short_to_latex(stat_to_short(x))}), 
         Type = sapply(Type, function(x){switch(x, chr = "Chromosome", neg = "Lower Tail", pos ="Upper Tail",x)})) %>%
  mutate_if(., is.numeric, funs(sprintf("%.2f",.))) %>% 
  arrange(stat) %>% 
  data.frame %>%
  knitr::kable(., caption = "(\\#tab:exclusivity) Mean exclusivity of clusters for a given super population.", booktabs = TRUE, escape = FALSE) %>% footnote(general = "Exclusivity of each cluster was calculated by using the proportions of populations from a given super population assigned compared to the total number of individual populations assigned that cluster", general_title = "") #%>% kableExtra::group_rows("Fu and Li's D", 1, 7) %>% kableExtra::group_rows("Fu and Li's F", 8, 14) %>% kableExtra::group_rows("Fay and Wu's H", 15, 21) %>% kableExtra::group_rows("Tajima's D", 22, 28) %>% kableExtra::group_rows("Zeng's E", 29, 35) 
```


There were differing numbers of windows used for each population, ranging from a minimum of `r create_windowTable() %>% summarise(min(min))` to a maximum of `r create_windowTable() %>% summarise(max(max))`. The mean number of windows per population was `r create_windowTable() %>% summarise(round(mean(mean), digits = 1))` (SD `r create_windowTable() %>% summarise(round(sd(mean), digits = 1))`). Table \@ref(tab:proportion) and Table \@ref(tab:exclusivity) show the proportion and exclusivity of the clusters respectively, for the clustering of populations by cutting the dendrograms to create six groups, based on the number of super populations. Clustering based on the windows of the extreme tails of the selection statistic distributions resulted in a higher number of the individual populations being clustered into their super populations than for the chromosome-wide selection statistics, and the clusters were also more exclusive to the super populations. This shows that the aggregate frequency spectrum based selection and neutrality statistics don't capture the population differentiation, but that the extremes for each population do replicate the grouping that population differentiation had with F~ST~.


```{r tailwindowSummary, echo = FALSE}
# create table dataset that can be referenced in text
tab <- create_windowTable() %>% mutate(statname = sapply(d, function(x){paste0("\\gls{",strsplit(x, "_")[[1]][1], "}") }),  tail = sapply(d, function(x){strsplit(x, "_")[[1]][2]})) %>% select(statname, tail, super, min,mean,max) %>% mutate(tail = ifelse(tail == 'neg', "Lower","Upper"))
colnames(tab) <- c("Statistic Name","Tail", "Super Population", "Min","Mean","Max")
```


```{r totalwindowTable, results = 'asis', echo = FALSE, eval = FALSE}
# actual table for the number of windows 
create_windowSummaryTable() %>% mutate(Tail = ifelse(sapply(stat, function(x){strsplit(x, '_')[[1]][2]}) == 'neg', "Lower", "Upper" ), `Statistic Name` = paste0('\\gls{',sapply(stat, function(x){strsplit(x, '_')[[1]][1]}),"}") )%>% mutate(sd = round_dp(sd, 1), mean = round_dp(mean, 1)) %>% select(`Statistic Name`, Tail, `Total Windows`= total_windows, Min = min, Median = median, Max = max, Mean = mean, `Std. Dev` = sd) %>% knitr::kable(., caption = '(\\#tab:totalwindowTable) Number of windows in the 1\\^{st} and 99\\^{th} percentiles of the distribution and the number of populations that shared a given window from the distribution tail. ', booktabs =TRUE, escape = FALSE, align = c("llrrrrrr")  )
```

```{r, eval = FALSE}
stat_markerLoc <-function(stat){
IRanges::subsetByOverlaps(marker_loc %>% mutate(start = pos, end = pos+1)%>% GenomicRanges::GRanges(),create_sums(paste0(stat,'_neg')) %>% filter(POLsum == ALLsum) %>% select(1:3) %>% GenomicRanges::GRanges()) %>% data.frame() %>% group_by(eff) %>% tally() %>% full_join(., IRanges::subsetByOverlaps(marker_loc %>% mutate(start = pos, end = pos+1)%>% GenomicRanges::GRanges(),create_sums(paste0(stat,'_pos')) %>% filter(POLsum == ALLsum) %>% select(1:3) %>% GenomicRanges::GRanges()) %>% data.frame() %>% group_by(eff) %>% tally(), by = 'eff') %>% mutate_if(is.numeric, funs(replace_na(., replace = 0))) %>% select(Effect = eff, firstper = n.x, ninetyninthper = n.y)
}

stat_markerLoc("td") %>% full_join(., stat_markerLoc("fwh"),by = 'Effect', suffix =  c("_td","_fwh")) %>% 
  full_join(., stat_markerLoc("flf"),by = 'Effect') %>% 
  full_join(., stat_markerLoc("ze"),by = 'Effect', suffix =  c("_flf","_ze")) %>% 
  mutate_if(is.numeric, funs(replace_na(., replace = 0))) %>% 
  kable( caption = '(\\#tab:perMarkerLoc) Number of markers by variant annotation type that were in the 1\\^{st} or 99\\{th} percentiles for Polynesian populations for the intrapopulation frequency spectrum selection and neutrality statistics.', col.names = c("Effect", rep(c("1\\^{st} percentile", "99\\^{th} percentile" ), 4)), format = 'latex', booktabs = TRUE, escape = FALSE) %>% add_header_above(c("","Tajima's D" =2, "Fay and Wu's H" = 2, "Fu and Lu's F" = 2, "Zeng's E" = 2)) %>% kable_styling(latex_options = "scale_down") %>% landscape()
```




<!--
describe number of total number of windows
polynesian specific windows
ep specific windows
wp specific windows
cim/nzm/sam/ton specific windows

proportions
exclusivity
dendrograms

refer to figure
-->

#### Tajima's _D_ 

##### Tajima's _D_ distributions {#tdDist}

The distributions of the genome-wide \gls{td} values showed that there were differences between each population, and super population (Figure \@ref(fig:tdDensity) and Table \@ref(tab:superGlobalSummaryTD)). The \gls{td} means for the Polynesian populations were the smallest of the super populations at `r global_super_summary %>% filter(super_pop == 'POL', stat == 'Tajima.D') %>% .[["mean"]] %>% three_dp()` (SD `r global_super_summary %>% filter(super_pop == 'POL', stat == 'Tajima.D') %>% .[["sd"]] %>% three_dp()`) with `r gs_td_table %>% filter(pop == 'CIM') %>% .[["mean"]] %>% three_dp()` (SD `r gs_td_table %>% filter(pop == 'CIM') %>% .[["sd"]] %>% three_dp()`) for \gls{cim}, `r gs_td_table %>% filter(pop == 'NZM') %>% .[["mean"]] %>% three_dp()` (SD `r gs_td_table %>% filter(pop == 'NZM') %>% .[["sd"]] %>% three_dp()`) for \gls{nzm}, `r gs_td_table %>% filter(pop == 'SAM') %>% .[["mean"]] %>% three_dp()` (SD `r gs_td_table %>% filter(pop == 'SAM') %>% .[["sd"]] %>% three_dp()`) for \gls{sam}, and `r gs_td_table %>% filter(pop == 'TON') %>% .[["mean"]] %>% three_dp()` (SD `r gs_td_table %>% filter(pop == 'TON') %>% .[["sd"]] %>% three_dp()`) for \gls{ton} (Table \@ref(tab:gsTdTable)). The means of the Polynesian populations were lower than all of the other populations except \gls{pel} and \gls{mxl}, indicating that overall the Polynesian populations had a higher proportion of low frequency to intermediate frequency variants. The \gls{sas} populations had the highest mean \gls{td} overall at `r global_super_summary %>% filter(super_pop == 'SAS', stat == 'Tajima.D') %>% .[["mean"]] %>% three_dp()` (SD `r global_super_summary %>% filter(super_pop == 'SAS', stat == 'Tajima.D') %>% .[["sd"]] %>% three_dp()`). The \gls{afr} populations were the only populations with the 1^st^ percentile above zero, but also had the smallest overall maximum value of `r global_super_summary %>% filter(super_pop == 'AFR', stat == 'Tajima.D') %>% .[["max"]] %>% three_dp()`. The \gls{eur} populations had the highest maximum \gls{td} value at `r global_super_summary %>% filter(super_pop == 'EUR', stat == 'Tajima.D') %>% .[["max"]] %>% three_dp()`. The smallest \gls{td} value of `r gs_td_table %>% filter(min(min) == min) %>% .[['min']] %>% three_dp()` came from the \gls{khv} population of the \gls{eas} super population. The \gls{afr} population had the smallest range (`r td_super_pop_means %>% filter(super_pop == 'AFR') %>% .[['range']] %>% three_dp()`), whereas the \gls{eas} super population had the largest (`r td_super_pop_means %>% filter(super_pop == 'EAS') %>% .[['range']] %>% three_dp()`). The \gls{pol} population had the second largest (`r td_super_pop_means %>% filter(super_pop == 'POL') %>% .[['range']] %>% three_dp()`). A right shifted distribution of \gls{td} can be attributed to ascertainment bias from using a \gls{snp} array, where there is a bias against low frequency markers.

```{r}
global_super_summary %>%filter(stat == "Tajima.D") %>% select(-stat) %>% kable(., caption ='(\\#tab:superGlobalSummaryTD) Summary statistics for \\gls{td} by super population.', digits = 3, booktabs =TRUE, escape = FALSE, col.names = c("Super Population", "Mean","SD","Min","1st Percentile","Median","99th Percentile","Max"), format = 'latex', linesep = "") %>% kableExtra::kable_styling(latex_options = 'scale_down')
```


```{r tdDensity, fig.cap = '(ref:tdDensity)', echo = FALSE}
knitr::include_graphics("images/04_clustering/td_densities.png")
```
(ref:tdDensity) Plot of the distribution of \gls{td} by population. Triangles indicate the minimum and maximum values. Dots indicate the 1^st^ and 99^th^ percentiles for each population.

```{r tdTop1000, fig.cap='(ref:tdTop1000)', echo = FALSE, out.width = "6in"}
#ggsave(plot = four_plot("td","Tajima's D"), filename = 'images/04_clustering/tdTop1000.png', device = 'png', height = 7, width = 7)
knitr::include_graphics('images/04_clustering/tdTop1000.png')
```

(ref:tdTop1000) _\gls{td} Clustering._ Hierarchical clustering of \gls{td} using the upper and lower 1% of the distribution. **A** and **B** Dendrogram representing the clusters of \gls{td} lower and upper tails respectfully using hierarchical clustering with Euclidean distance and complete linkage. Coloured by super population. **C** and **D** Plots representing the windows present in each population from the lower and upper tails respectfully. A blue line represents presence of a window, white represents the absence of a window in the 1% of the distribution for a population.

<!--
- neg gave super pop clusters
- pos gave super pop clusters
- both split the AMR into 2
- number of windows shared by POL with others was X
- number of windows shared EP/WP was X
- number of unique windows was X for CIM, NZM, SAM, TON
- genes involved?
-->

The clustering of the individual populations for the 1% most extreme values from each of the lower and upper tails of the \gls{td} distribution both showed the East/West split in the Polynesians (Figure \@ref(fig:tdTop1000)). But the clustering also showed that the Polynesian populations were most similar to each other. Both tails placed the Polynesian cluster closest to the \gls{eas} cluster, consistent with the migration history. The main groupings are of the populations that were designated as the same super population, although the \gls{amr} super population in both tails has been split into two. The mean number of unique windows per population was `r round(create_sums('td_pos') %>% filter(ALLsum == 1) %>% select(-contains('chrom'), -posid, -contains('sum')) %>% summarise_all(funs(sum)) %>% t() %>% mean(), digits = 1)` (SD `r create_sums('td_pos') %>% filter(ALLsum == 1) %>% select(-contains('chrom'), -posid, -contains('sum')) %>% summarise_all(funs(sum)) %>% t() %>% sd() %>% round(., digits = 1)`) in the upper tail, compared with `r round( create_sums('td_neg') %>% filter(ALLsum == 1) %>% select(-contains('chrom'), -posid, -contains('sum')) %>% summarise_all(funs(sum)) %>% t() %>% mean(), digits = 1)` (SD `r create_sums('td_neg') %>% filter(ALLsum == 1) %>% select(-contains('chrom'), -posid, -contains('sum')) %>% summarise_all(funs(sum)) %>% t() %>% sd() %>% round(., digits = 1)`) in the lower tail, indicating a higher degree of window sharing in the 1^st^ percentiles of populations. The groupings for both the upper and lower tails exclusively grouped each of the \gls{afr}, \gls{sas} and \gls{pol} super populations. The \gls{amr} super population was split into an exclusive group of \gls{mxl} and \gls{pel}, and a group of \gls{clm} and \gls{pur} which was shared with the \gls{eur} super population. All clusters, except the \gls{amr} - \gls{eur} cluster, were each exclusive to a single super population.

##### Clustering Tajima's _D_ 1^st^ percentile


```{r, eval = TRUE, warning = FALSE, message=FALSE}
# get all of the windows that had a Pol hit
td_neg_pol <- create_sums('td_neg') %>% filter(POLsum > 0) %>% mutate(chrom = paste0('chr', chrom))

# annotate the windows with gene names
td_neg_pol_genes <- left_join(td_neg_pol, txdb_gene_annotate(GenomicRanges::GRanges(td_neg_pol)), by = c('chrom' = 'seqnames', "chrom_start" = 'start', "chrom_end" = 'end')) %>% select(contains('chrom'), SYMBOL, everything(), -width, -posid, -TXSTRAND, -strand)
write.csv(file ='Appendices/04-td-pol-firstper-genes.csv', x = td_neg_pol_genes, quote = TRUE, row.names = FALSE, col.names = TRUE )
# Look at the genes to see how many other pops had them too
#td_neg_pol_genes %>% filter( SYMBOL %in% genelists$genename) %>% select(contains('chrom'),SYMBOL, ALLsum, AFRsum, EURsum, SASsum, EASsum, POLsum, contains("EAS"), contains("POL")) %>% arrange(desc(POLsum)) %>% filter(POLsum > 1) %>% View()

# find the number of genes from the disease genes that the polynesians had
#td_neg_pol_genes %>% filter( SYMBOL %in% genelists$genename) %>% select(contains('chrom'),SYMBOL, ALLsum, AFRsum, EURsum, SASsum, EASsum, POLsum, contains("EAS"), contains("POL")) %>% arrange(desc(POLsum)) %>% pull(SYMBOL) %>% unique() %>% length()

# get the genes that were in windows that the Pol pops weren't in
td_neg_no_pol_genes <- txdb_gene_annotate(create_sums('td_neg') %>% filter(POLsum == 0) %>% mutate(chrom = paste0('chr', chrom)) %>% GenomicRanges::GRanges()) %>% pull(SYMBOL) %>% unique()
# find how many genes were only found with the Pols + others
#td_neg_pol_genes %>% filter(!SYMBOL %in% td_neg_no_pol_genes, SYMBOL %in% genelists$genename) %>% pull(SYMBOL) %>% unique() %>% length()
```


In the lower tail, the number of windows that were unique to the Polynesian super population was `r create_sums('td_neg') %>% filter(POLsum == ALLsum) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)`, representing `r round(create_sums('td_neg') %>% filter(POLsum == ALLsum) %>% select(contains('chrom'), contains('sum')) %>% NROW(.) / mat_d_list[['td_neg']] %>% NROW(.) *100, digits = 1)`% of all windows in the lower tail that were used. This compared to super population unique window numbers of `r create_sums('td_neg') %>% filter(AFRsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)`, `r create_sums('td_neg') %>% filter(AMRsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)`, `r create_sums('td_neg') %>% filter(EASsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)`, `r create_sums('td_neg') %>% filter(EURsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)`, `r create_sums('td_neg') %>% filter(SASsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` for the \gls{afr}, \gls{amr}, \gls{eas}, \gls{eur}, and \gls{sas} super populations respectively. Of the unique windows for Polynesian populations, `r create_sums('td_neg') %>% filter(POL_CIM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` were specific to \gls{cim}, `r create_sums('td_neg') %>% filter(POL_NZM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` were specific to \gls{nzm}, `r create_sums('td_neg') %>% filter(POL_SAM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` were specific to \gls{sam}, and `r create_sums('td_neg') %>% filter(POL_TON == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` were specific to \gls{ton}. The Eastern Polynesians had `r create_sums('td_neg') %>% filter((POL_CIM + POL_NZM) == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` unique windows with `r create_sums('td_neg') %>% filter((POL_CIM + POL_NZM) == ALLsum & ALLsum == 2) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` of those shared between \gls{cim} and \gls{nzm}, whereas the Western Polynesians had `r create_sums('td_neg') %>% filter((POL_SAM + POL_TON) == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows, with `r create_sums('td_neg') %>% filter((POL_SAM + POL_TON) == ALLsum & ALLsum == 2) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` shared between \gls{sam} and \gls{ton}. There were `r create_sums('td_neg') %>% filter(POLsum < ALLsum & POLsum == 4) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows where all the Polynesian populations shared a window with a non-Polynesian population. And `r create_sums('td_neg') %>% filter(POLsum < ALLsum & POLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows, intersecting `r td_neg_pol_genes  %>% filter(!is.na(SYMBOL)) %>% pull(SYMBOL) %>% unique() %>% length()` genes, where at least one Polynesian population shared a window with a non-Polynesian population (Table \@ref(tab:polTdNegGenes)). There were `r create_sums('td_neg') %>% filter(POLsum == ALLsum & POLsum == 4) %>% mutate(chrom = paste0('chr', chrom)) %>%select(1:3) %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% txdb_gene_annotate() %>% as_tibble() %>% filter(!is.na(SYMBOL)) %>% pull(SYMBOL) %>% unique() %>% length()` genes that had windows that were in the 1^st^ percentile for all four Polynesian populations. Cell signalling as an immune response, such as the Toll-like receptor cascades were the main pathways this list of genes covered, although this was largely due to _PPP2CB_ and _NOD1_. Two other genes in the list were _CCR3_ and _ARL15_, both of which are associated with obesity related traits [@comuzzie2012novel;@shungin2015genetic], and the latter also associated with \gls{t2d} [@mahajan2014genome].





```{r, eval = FALSE}
create_sums('td_neg') %>% filter(POLsum == ALLsum & POLsum == 4) %>% mutate(chrom = paste0('chr', chrom)) %>%select(1:3) %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% txdb_gene_annotate() %>% as_tibble() %>% filter(!is.na(SYMBOL)) %>% pull(SYMBOL) %>% unique()

create_sums('td_neg') %>% filter(POLsum == ALLsum & POLsum > 0) %>% View()
```

```{r, eval = FALSE}
pol_tail_filter('td_neg', pol_min = 2, all_max = 4) %>% filter() %>% pull(SYMBOL)
pol_tail_filter('td_pos', pol_min = 4, all_max = 4) %>% pull(SYMBOL) %>% unique() %>% sort()
```



```{r, eval = FALSE}
# untested/bad assumption that all genes will be in here
create_sums('td_neg') %>% mutate(chrom = paste0('chr',chrom))%>% filter(POLsum == ALLsum & POLsum > 0) %>% left_join(., lower_sig_stats %>% select(contains('chrom'), SYMBOL) %>% distinct(), by = c('chrom', 'chrom_start', 'chrom_end')) %>% group_by(SYMBOL) %>% tally() %>% 

create_sums('td_neg') %>% filter(POL_CIM + POL_NZM == ALLsum & ALLsum == 2) %>% left_join(., lower_sig_stats %>% mutate(chrom = as.numeric(chrom)) %>% select(contains('chrom'), SYMBOL) %>% distinct(), by = c('chrom', 'chrom_start', 'chrom_end')) %>% group_by(SYMBOL) %>% tally() %>% filter(!is.na(SYMBOL)) %>% NROW(.)

create_sums('td_neg') %>% filter(POL_SAM + POL_TON == ALLsum & ALLsum == 2) %>% left_join(., lower_sig_stats %>% mutate(chrom = as.numeric(chrom)) %>% select(contains('chrom'), SYMBOL) %>% distinct(), by = c('chrom', 'chrom_start', 'chrom_end')) %>% group_by(SYMBOL) %>% tally() %>% filter(!is.na(SYMBOL)) %>% NROW(.)
```


```{r, eval = FALSE}
# untested/bad assumption that all genes will be in here
(td_neg_pol_genes <- create_sums('td_neg') %>% filter(POLsum == 4) %>% left_join(., lower_sig_stats %>% mutate(chrom = as.numeric(chrom)) %>% select(contains('chrom'), SYMBOL) %>% distinct(), by = c('chrom', 'chrom_start', 'chrom_end')) %>% group_by(SYMBOL) %>% tally())
```



```{r, eval = FALSE}
IRanges::subsetByOverlaps(marker_loc %>% mutate(start = pos, end = pos+1)%>% GenomicRanges::GRanges(),create_sums('td_pos') %>% filter(POLsum == ALLsum) %>% select(1:3) %>% GenomicRanges::GRanges()) %>% data.frame() %>% group_by(eff) %>% tally()
```

##### Clustering Tajima's _D_ 99^th^ percentile

```{r, eval = TRUE, warning = FALSE, message = FALSE}
# get all of the windows that had a Pol hit
td_pos_pol <- create_sums('td_pos') %>% filter(POLsum > 0) %>% mutate(chrom = paste0('chr', chrom))

# annotate the windows with gene names
td_pos_pol_genes <- left_join(td_pos_pol, txdb_gene_annotate(GenomicRanges::GRanges(td_pos_pol)), by = c('chrom' = 'seqnames', "chrom_start" = 'start', "chrom_end" = 'end')) %>% select(contains('chrom'), SYMBOL, everything(), -width, -posid, -TXSTRAND, -strand)
write.csv(file ='Appendices/04-td-pol-ninetyninthper-genes.csv', x = td_pos_pol_genes, quote = TRUE, row.names = FALSE, col.names = TRUE )
# Look at the genes to see how many other pops had them too
#td_pos_pol_genes %>% filter( SYMBOL %in% genelists$genename) %>% select(contains('chrom'),SYMBOL, ALLsum, AFRsum, EURsum, SASsum, EASsum, POLsum, contains("EAS"), contains("POL")) %>% arrange(desc(POLsum)) %>% View()

# find the number of genes from the disease genes that the polynesians had
#td_pos_pol_genes %>% filter( SYMBOL %in% genelists$genename) %>% select(contains('chrom'),SYMBOL, ALLsum, AFRsum, EURsum, SASsum, EASsum, POLsum, contains("EAS"), contains("POL")) %>% arrange(desc(POLsum)) %>% pull(SYMBOL) %>% unique() %>% length()

# get the genes that were in windows that the Pol pops weren't in
td_pos_no_pol_genes <- txdb_gene_annotate(create_sums('td_pos') %>% filter(POLsum == 0) %>% mutate(chrom = paste0('chr', chrom)) %>% GenomicRanges::GRanges()) %>% pull(SYMBOL) %>% unique()
# find how many genes were only found with the Pols + others
#td_pos_pol_genes %>% filter(!SYMBOL %in% td_pos_no_pol_genes, SYMBOL %in% genelists$genename) %>% pull(SYMBOL) %>% unique() %>% length()
```

The upper tail had a total of `r create_sums('td_pos') %>% filter(POLsum == ALLsum) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows, or `r round(create_sums('td_pos') %>% filter(POLsum == ALLsum) %>% select(contains('chrom'), contains('sum')) %>% NROW(.) / mat_d_list[['td_pos']] %>% NROW(.) *100, digits = 1)`% that were unique to the Polynesian populations. This compares to super population unique window numbers of `r create_sums('td_pos') %>% filter(AFRsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` (\gls{afr}), `r create_sums('td_pos') %>% filter(AMRsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` (\gls{amr}), `r create_sums('td_pos') %>% filter(EASsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` (\gls{eas}), `r create_sums('td_pos') %>% filter(EURsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` (\gls{eur}), `r create_sums('td_pos') %>% filter(SASsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` (\gls{sas}). Eastern Polynesians contributed `r create_sums('td_pos') %>% filter((POL_CIM + POL_NZM & ALLsum > 0) == ALLsum) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` unique windows, with \gls{cim} having `r create_sums('td_pos') %>% filter(POL_CIM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` unique windows and \gls{nzm} having `r create_sums('td_pos') %>% filter(POL_NZM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows. The Western Polynesians had `r create_sums('td_pos') %>% filter((POL_SAM + POL_TON) == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` unique windows. `r create_sums('td_pos') %>% filter(POL_SAM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` were from only \gls{sam} and `r create_sums('td_pos') %>% filter(POL_TON == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` from only \gls{ton}. There were `r create_sums('td_pos') %>% filter(POLsum < ALLsum & POLsum == 4) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows where all the Polynesian populations shared a window with a non-Polynesian population. And `r create_sums('td_pos') %>% filter(POLsum < ALLsum & POLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows, intersecting `r td_pos_pol_genes %>% filter(!is.na(SYMBOL)) %>% pull(SYMBOL) %>% unique() %>% length()` genes, where at least one Polynesian population shared a windows with a non-Polynesian population (Table \@ref(tab:polTdPosGenes)). 


In both tails the clustering is being driven by a small subset of the regions, this can be seen in Figures \@ref(fig:tdTop1000) C and D where most of the regions that the Polynesian populations have, are not shared with the other populations. It can also be seen that the regions that the Polynesian populations do have in common are also shared with the \gls{eas} populations. This is reflected in the dendrograms in Figures \@ref(fig:tdTop1000) A and B.

```{r, eval = FALSE}
# untested/bad assumption that all genes will be in here
create_sums('td_pos') %>% filter(POLsum == ALLsum & POLsum > 0) %>% left_join(., lower_sig_stats %>% mutate(chrom = as.numeric(chrom)) %>% select(contains('chrom'), SYMBOL) %>% distinct(), by = c('chrom', 'chrom_start', 'chrom_end')) %>% group_by(SYMBOL) %>% tally() %>% filter(!is.na(SYMBOL)) %>% NROW(.)

create_sums('td_pos') %>% filter(POL_CIM + POL_NZM == ALLsum & ALLsum == 2) %>% left_join(., lower_sig_stats %>% mutate(chrom = as.numeric(chrom)) %>% select(contains('chrom'), SYMBOL) %>% distinct(), by = c('chrom', 'chrom_start', 'chrom_end')) %>% group_by(SYMBOL) %>% tally() %>% filter(!is.na(SYMBOL)) %>% NROW(.)

create_sums('td_pos') %>% filter(POL_SAM + POL_TON == ALLsum & ALLsum == 2) %>% left_join(., lower_sig_stats %>% mutate(chrom = as.numeric(chrom)) %>% select(contains('chrom'), SYMBOL) %>% distinct(), by = c('chrom', 'chrom_start', 'chrom_end')) %>% group_by(SYMBOL) %>% tally() %>% filter(!is.na(SYMBOL)) %>% NROW(.)
```

```{r, eval = FALSE}
# untested/bad assumption that all genes will be in here
(td_pos_pol_genes <- create_sums('td_pos') %>% filter(POLsum == 4 ) %>% left_join(., lower_sig_stats %>% mutate(chrom = as.numeric(chrom)) %>% select(contains('chrom'), SYMBOL) %>% distinct(), by = c('chrom', 'chrom_start', 'chrom_end')) %>% group_by(SYMBOL) %>% tally())
```

##### Metabolic disease-associated genes

<!-- > not sure how best to represent the genes listed in this section and the equivalent ones for other stats -->
```{r}
td_met <- list()
td_met$neg_pol <- td_neg_pol_genes %>% filter( SYMBOL %in% genelists$genename) %>% select(contains('chrom'),SYMBOL, ALLsum, AFRsum, EURsum, SASsum, EASsum, POLsum, contains("EAS"), contains("POL")) %>% arrange(desc(POLsum)) %>% filter(POLsum > 0) %>% pull(SYMBOL) %>% unique()

td_met$neg_not_only_pol <-  td_neg_pol_genes %>% filter( SYMBOL %in% genelists$genename) %>% select(contains('chrom'),SYMBOL, ALLsum, AFRsum, EURsum, SASsum, EASsum, POLsum, contains("EAS"), contains("POL")) %>% arrange(desc(POLsum)) %>% filter(POLsum < ALLsum, !(SYMBOL %in% td_neg_no_pol_genes )) %>% pull(SYMBOL) %>% unique()


td_met$pos_pol <- td_pos_pol_genes %>% filter( SYMBOL %in% genelists$genename) %>% select(contains('chrom'),SYMBOL, ALLsum, AFRsum, EURsum, SASsum, EASsum, POLsum, contains("EAS"), contains("POL")) %>% arrange(desc(POLsum)) %>% filter(POLsum > 0) %>% pull(SYMBOL) %>% unique()

# genes that were in polynesians but also other populations
td_met$pos_not_only_pol <-  td_pos_pol_genes %>% filter( SYMBOL %in% genelists$genename) %>% select(contains('chrom'),SYMBOL, ALLsum, AFRsum, EURsum, SASsum, EASsum, POLsum, contains("EAS"), contains("POL")) %>% arrange(desc(POLsum)) %>% filter(POLsum < ALLsum, !(SYMBOL %in% td_pos_no_pol_genes )) %>% pull(SYMBOL) %>% unique()
```


Specifically focusing on the `r NROW(genelists)` genes that were associated with urate, gout, obesity, \gls{t2d}, kidney disease, and metabolic syndrome from the \gls{gwas} catalog (Tables \@ref(tab:gwascatref) and \@ref(tab:gwasgenes)), from the clustering of the extremes, there were a total of `r td_met$neg_pol %>% length()`  genes that had at least one of the Polynesian populations having at least one window in the 1^st^ percentile. This dropped to `r td_met$neg_not_only_pol %>% length()`  genes that had windows in the 1^st^ percentile for at least one Polynesian population, and also in the 1^st^ percentile of other populations. Some examples of this were _CCR3_, which had windows in the 1^st^ percentile for all four Polynesian populations, and the same windows were also in the 1^st^ percentile as the \gls{eas} populations of \gls{cdx} and \gls{khv}. _ARL15_ had windows that were only from the four Polynesian populations. And _DNAH10_ had windows that were in the 1^st^ percentile of the \gls{eas} populations except \gls{cdx}, as well as the Eastern Polynesian populations, \gls{cim} and \gls{nzm}. In the 99^th^ percentile there were `r td_met$pos_pol %>% length()` genes associated with the metabolic diseases that had windows in the Polynesian populations, of those, `r td_met$pos_not_only_pol %>% length()` genes had windows in the 99th percentile for Polynesians and the 99^th^ percentile of other populations. Some examples include _ADCY3_ which had multiple windows with \gls{pol}, \gls{eur}, \gls{eas}, and \gls{sas}. And _NEGR1_ which had windows from both the \gls{eas} and \gls{pol} populations.




#### Fay and Wu's _H_


##### Fay and Wu's _H_ distributions {#fwhDist}

Comparing genome-wide \gls{fwh} revealed that the \gls{pol} super population had the lowest mean (`r fwh_super_pop_means %>% filter(super_pop == 'POL') %>% .[['mean']] %>% three_dp()`), 1^st^ percentile (`r fwh_super_pop_means %>% filter(super_pop == 'POL') %>% .[['lower_1']] %>% three_dp()`), median (`r fwh_super_pop_means %>% filter(super_pop == 'POL') %>% .[['median']] %>% three_dp()`), 99^th^ percentile (`r fwh_super_pop_means %>% filter(super_pop == 'POL') %>% .[['upper_99']] %>% three_dp()`), and maximum values (`r fwh_super_pop_means %>% filter(super_pop == 'POL') %>% .[['max']] %>% three_dp()`) (Figure \@ref(fig:fwhDensity) and Table \@ref(tab:superGlobalSummaryFWH)). Within the \gls{pol} group, the minimum \gls{fwh} value for a Polynesian population was from `r gs_fwh_table %>% filter(super_pop == "POL") %>% filter(min(min) == min) %>% .[["pop"]]`, whereas the maximum value was from `r gs_fwh_table %>% filter(super_pop == "POL") %>% filter(max(max) == max) %>% .[["pop"]]`. \gls{eas} was the second lowest for mean (`r fwh_super_pop_means %>% filter(super_pop == 'EAS') %>% .[['mean']] %>% three_dp()`), 1^st^ percentile (`r fwh_super_pop_means %>% filter(super_pop == 'EAS') %>% .[['lower_1']] %>% three_dp()`), median (`r fwh_super_pop_means %>% filter(super_pop == 'EAS') %>% .[['median']] %>% three_dp()`), and 99^th^ percentile (`r fwh_super_pop_means %>% filter(super_pop == 'EAS') %>% .[['upper_99']] %>% three_dp()`), but had the largest maximum (`r fwh_super_pop_means %>% filter(super_pop == 'EAS') %>% .[['max']] %>% three_dp()`). The \gls{eur} super population had both the smallest minimum and largest range (`r fwh_super_pop_means %>% filter(super_pop == 'EUR') %>% .[['range']] %>% three_dp()`). The \gls{afr} super population were the only group to have a positive mean (`r fwh_super_pop_means %>% filter(super_pop == 'AFR') %>% .[['mean']] %>% three_dp()`), they also had the smallest range and variation (SD `r fwh_super_pop_means %>% filter(super_pop == 'AFR') %>% .[['sd']] %>% three_dp()`). This indicates that the \gls{afr} populations have a deficit of moderate-to-high frequency derived alleles, whereas the \gls{pol} and \gls{eas} populations have the most windows with the highest excess of moderate-to-high frequency derived alleles.

```{r fwhDensity, fig.cap = '(ref:fwhDensity)', echo = FALSE}
knitr::include_graphics("images/04_clustering/fwh_densities.png")
```

(ref:fwhDensity) Plot of the distribution of \gls{fwh} by population. Triangles indicate the minimum and maximum values. Dots indicate the 1^st^ and 99^th^ percentiles for each population.

```{r}
global_super_summary %>%filter(stat == "Fay.Wu.H") %>% select(-stat) %>% kable(., caption ='(\\#tab:superGlobalSummaryFWH) Summary statistics for \\gls{fwh} by super population.', digits = 3, booktabs =TRUE, escape = FALSE, col.names = c("Super Population", "Mean","SD","Min","1st Percentile","Median","99th Percentile","Max"), format = 'latex', linesep = "") %>% kableExtra::kable_styling(latex_options = 'scale_down')
```

(ref:fwhTop1000) _\gls{fwh} Clustering._ Hierarchical clustering of \gls{fwh} using the upper and lower tails of the distribution. **A** and **B** Dendrogram representing the clusters of \gls{fwh} lower and upper tails respectfully using hierarchical clustering with Euclidean distance and complete linkage. Coloured by super population. **C** and **D** Plots representing the windows present in each population from the lower and upper tails respectfully. A blue line represents presence of a window, white represents the absence of a window in the 1% of the distribution for a population.

```{r fwhTop1000, fig.cap='(ref:fwhTop1000)', out.width = "6in"}
#ggsave(four_plot("fwh","Fay and Wu's H"), filename = 'images/04_clustering/fwhTop1000.png', device = 'png', height = 7, width = 7)
knitr::include_graphics('images/04_clustering/fwhTop1000.png')
```

<!--
- Fay and Wu's H put all of the super populations together both neg and pos (split AMR)
- there looks to be a lot of unique regions within the POL
- both neg and pos have the Pol cluster nearest to the EAS
-->

<!--
[x]describe number of total number of windows
[x]polynesian specific windows
[x]ep specific windows
[x]wp specific windows
[x]cim/nzm/sam/ton specific windows

[x]proportions
[x]exclusivity
[]dendrograms

[x] refer to figure
-->


The clustering of \gls{fwh} (Figure \@ref(fig:fwhTop1000)) had the main groupings the same as the defined super population groups, with the exception of the upper tail where the \gls{amr} super population was split in half. The Polynesian populations were clustered together in both the lower and upper tails for \gls{fwh}. In both tails the Polynesian cluster also had the East/West split within the group. The Polynesian population group was closest to the \gls{eas} populations with `r create_sums('fwh_pos') %>% filter(POLsum + EASsum <= ALLsum & POLsum > 0 & EASsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows from the upper tail and `r create_sums('fwh_neg') %>% filter(POLsum + EASsum <= ALLsum & POLsum > 0 & EASsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows from the lower tail in common between at least one Polynesian population and one \gls{eas} population. 

##### Clustering Fay and Wu's _H_ 1^st^ percentile

```{r, eval = TRUE, warning = FALSE, message = FALSE}
# get all of the windows that had a Pol hit
fwh_neg_pol <- create_sums('fwh_neg') %>% filter(POLsum > 0) %>% mutate(chrom = paste0('chr', chrom))

# annotate the windows with gene names
fwh_neg_pol_genes <- left_join(fwh_neg_pol, txdb_gene_annotate(GenomicRanges::GRanges(fwh_neg_pol)), by = c('chrom' = 'seqnames', "chrom_start" = 'start', "chrom_end" = 'end')) %>% select(contains('chrom'), SYMBOL, everything(), -width, -posid, -TXSTRAND, -strand)
write.csv(file ='Appendices/04-fwh-pol-firstper-genes.csv', x = fwh_neg_pol_genes, quote = TRUE, row.names = FALSE, col.names = TRUE )
# Look at the genes to see how many other pops had them too
#fwh_neg_pol_genes %>% filter( SYMBOL %in% genelists$genename) %>% select(contains('chrom'),SYMBOL, ALLsum, AFRsum, EURsum, SASsum, EASsum, POLsum, contains("EAS"), contains("POL")) %>% arrange(desc(POLsum)) %>% View()

# find the number of genes from the disease genes that the polynesians had
#fwh_neg_pol_genes %>% filter( SYMBOL %in% genelists$genename) %>% select(contains('chrom'),SYMBOL, ALLsum, AFRsum, EURsum, SASsum, EASsum, POLsum, contains("EAS"), contains("POL")) %>% arrange(desc(POLsum)) %>% pull(SYMBOL) %>% unique() %>% length()

# get the genes that were in windows that the Pol pops weren't in
fwh_neg_no_pol_genes <- txdb_gene_annotate(create_sums('fwh_neg') %>% filter(POLsum == 0) %>% mutate(chrom = paste0('chr', chrom)) %>% GenomicRanges::GRanges()) %>% pull(SYMBOL) %>% unique()
# find how many genes were only found with the Pols + others
#fwh_neg_pol_genes %>% filter(!SYMBOL %in% fwh_no_pol_genes, SYMBOL %in% genelists$genename) %>% pull(SYMBOL) %>% unique() %>% length()
```

The clustering of the lower tail assigned all the individuals into their super populations and each of the groups were exclusive to their super population. There were a total of `r create_windowSummaryTable() %>% filter(stat == 'fwh_neg') %>% .[['total_windows']]` windows in the lower tail, with `r create_sums('fwh_neg') %>% filter(POLsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows specific to Polynesian populations. This compared to `r create_sums('fwh_neg') %>% filter(AFRsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{afr}, `r create_sums('fwh_neg') %>% filter(AMRsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{amr}, `r create_sums('fwh_neg') %>% filter(EURsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{eur}, and `r create_sums('fwh_neg') %>% filter(EASsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{eas}. The Eastern Polynesian populations had `r create_sums('fwh_neg') %>% filter(POL_CIM + POL_NZM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows, of those, `r create_sums('fwh_neg') %>% filter(POL_CIM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` were specific to \gls{cim} and `r create_sums('fwh_neg') %>% filter(POL_NZM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` were specific to \gls{nzm}. The Western Polynesian populations had `r create_sums('fwh_neg') %>% filter(POL_SAM + POL_TON == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows, with `r create_sums('fwh_neg') %>% filter(POL_SAM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` specific to \gls{sam} and `r create_sums('fwh_neg') %>% filter(POL_TON == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` specific to \gls{ton}. The mean number of windows specific to an individual population was `r create_sums('fwh_neg') %>% filter(ALLsum == 1) %>% select(-contains('chrom'), -posid, -contains('sum')) %>% summarise_all(funs(sum)) %>% t() %>% mean() %>% round(., digits = 1)` (SD `r create_sums('fwh_neg') %>% filter(ALLsum == 1) %>% select(-contains('chrom'), -posid, -contains('sum')) %>% summarise_all(funs(sum)) %>% t() %>% sd() %>% round(., digits = 1)`) in the lower tail. There were `r create_sums('fwh_neg') %>% filter(POLsum < ALLsum & POLsum == 4) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows with all Polynesians that also shared with a non-Polynesian population. And `r create_sums('fwh_neg') %>% filter(POLsum < ALLsum & POLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows, intersecting `r fwh_neg_pol_genes %>% filter(!is.na(SYMBOL)) %>%  pull(SYMBOL) %>% unique() %>% length()` genes with at least one Polynesian population that also shared with a non-Polynesian population (Table \@ref(tab:polFwhNegGenes)). There were `r create_sums('fwh_neg') %>% filter(POLsum == ALLsum & POLsum == 4) %>% mutate(chrom = paste0('chr', chrom)) %>%select(1:3) %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% txdb_gene_annotate() %>% as_tibble() %>% filter(!is.na(SYMBOL)) %>% pull(SYMBOL) %>% unique() %>% length()` genes that had windows that were only in the four Polynesian populations and no others.



##### Clustering Fay and Wu's _H_ 99^th^ percentile

```{r, eval = TRUE, message = FALSE, warning = FALSE}
# get all of the windows that had a Pol hit
fwh_pos_pol <- create_sums('fwh_pos') %>% filter(POLsum > 0) %>% mutate(chrom = paste0('chr', chrom))

# annotate the windows with gene names
fwh_pos_pol_genes <- left_join(fwh_pos_pol, txdb_gene_annotate(GenomicRanges::GRanges(fwh_pos_pol)), by = c('chrom' = 'seqnames', "chrom_start" = 'start', "chrom_end" = 'end')) %>% select(contains('chrom'), SYMBOL, everything(), -width, -posid, -TXSTRAND, -strand)
write.csv(file ='Appendices/04-fwh-pol-ninetyninethper-genes.csv', x = fwh_pos_pol_genes, quote = TRUE, row.names = FALSE, col.names = TRUE )
# Look at the genes to see how many other pops had them too
#fwh_pos_pol_genes %>% filter( SYMBOL %in% genelists$genename) %>% select(contains('chrom'),SYMBOL, ALLsum, AFRsum, EURsum, SASsum, EASsum, POLsum, contains("EAS"), contains("POL")) %>% arrange(desc(POLsum)) %>% View()

# find the number of genes from the disease genes that the polynesians had
#fwh_pos_pol_genes %>% filter( SYMBOL %in% genelists$genename) %>% select(contains('chrom'),SYMBOL, ALLsum, AFRsum, EURsum, SASsum, EASsum, POLsum, contains("EAS"), contains("POL")) %>% arrange(desc(POLsum)) %>% pull(SYMBOL) %>% unique() %>% length()

# get the genes that were in windows that the Pol pops weren't in
fwh_pos_no_pol_genes <- txdb_gene_annotate(create_sums('fwh_pos') %>% filter(POLsum == 0) %>% mutate(chrom = paste0('chr', chrom)) %>% GenomicRanges::GRanges()) %>% pull(SYMBOL) %>% unique()
# find how many genes were only found with the Pols + others
#fwh_pos_pol_genes %>% filter(!SYMBOL %in% fwh_pos_no_pol_genes, SYMBOL %in% genelists$genename) %>% pull(SYMBOL) %>% unique() %>% length()
```


In the upper tail of the \gls{fwh} distribution, the clustering assigned all of the individual populations into their super populations with only \gls{amr} being split. The \gls{clm} and \gls{pur} populations were clustered as part of the \gls{eur} group. Exclusivity was also high with all clusters except the \gls{eur} - \gls{amr} being exclusive to a single super population. Of the `r create_windowSummaryTable() %>% filter(stat == 'fwh_pos') %>% .[['total_windows']]` windows, `r create_sums('fwh_pos') %>% filter(POLsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows were from Polynesian populations only. The Eastern Polynesian populations had `r create_sums('fwh_pos') %>% filter(POL_CIM + POL_NZM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` specific windows, of those, `r create_sums('fwh_pos') %>% filter(POL_CIM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` were unique to \gls{cim} and `r create_sums('fwh_pos') %>% filter(POL_NZM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` were unique to \gls{nzm}. The Western Polynesian populations had `r create_sums('fwh_pos') %>% filter(POL_SAM + POL_TON == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` unique windows, of these, `r create_sums('fwh_pos') %>% filter(POL_SAM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` were specific to \gls{sam} and `r create_sums('fwh_pos') %>% filter(POL_TON == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` specific to \gls{ton}. The mean number of unique windows for all populations was `r create_sums('fwh_pos') %>% filter(ALLsum == 1) %>% select(-contains('chrom'), -posid, -contains('sum')) %>% summarise_all(funs(sum)) %>% t() %>% mean() %>% round(., digits = 1)` (SD `r create_sums('fwh_pos') %>% filter(ALLsum == 1) %>% select(-contains('chrom'), -posid, -contains('sum')) %>% summarise_all(funs(sum)) %>% t() %>% sd() %>% round(., digits = 1)`). There were `r create_sums('fwh_pos') %>% filter(POLsum < ALLsum & POLsum == 4) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows with all Polynesians that also shared with a non-Polynesian population. And `r create_sums('fwh_pos') %>% filter(POLsum < ALLsum & POLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows, intersecting `r fwh_pos_pol_genes %>% filter(!is.na(SYMBOL))%>% pull(SYMBOL) %>% unique() %>% length()` genes, with at least one Polynesian population that also shared with a non-Polynesian population (Table \@ref(tab:polFwhPosGenes)).




The 1^st^ percentile has many of the regions in the Polynesian populations in common with the \gls{eas} populations (Figure \@ref(fig:fwhTop1000)). Whereas, in the 99^th^ percentile there were more regions that were only from the Polynesian populations that were not in common with any other populations. This shows that the regions with high frequency derived alleles are in common with the \gls{eas} populations, indicating they possibly originated prior to the Polynesian migration.


##### Metabolic disease-associated genes

```{r}
fwh_met <- list()
fwh_met$neg_pol <- fwh_neg_pol_genes %>% filter( SYMBOL %in% genelists$genename) %>% select(contains('chrom'),SYMBOL, ALLsum, AFRsum, EURsum, SASsum, EASsum, POLsum, contains("EAS"), contains("POL")) %>% arrange(desc(POLsum)) %>% filter(POLsum > 0) %>% pull(SYMBOL) %>% unique()

fwh_met$neg_not_only_pol <-  fwh_neg_pol_genes %>% filter( SYMBOL %in% genelists$genename) %>% select(contains('chrom'),SYMBOL, ALLsum, AFRsum, EURsum, SASsum, EASsum, POLsum, contains("EAS"), contains("POL")) %>% arrange(desc(POLsum)) %>% filter(POLsum < ALLsum, !(SYMBOL %in% fwh_neg_no_pol_genes )) %>% pull(SYMBOL) %>% unique()


fwh_met$pos_pol <- fwh_pos_pol_genes %>% filter( SYMBOL %in% genelists$genename) %>% select(contains('chrom'),SYMBOL, ALLsum, AFRsum, EURsum, SASsum, EASsum, POLsum, contains("EAS"), contains("POL")) %>% arrange(desc(POLsum)) %>% filter(POLsum > 0) %>% pull(SYMBOL) %>% unique()

# genes that were in polynesians but also other populations
fwh_met$pos_not_only_pol <-  fwh_pos_pol_genes %>% filter( SYMBOL %in% genelists$genename) %>% select(contains('chrom'),SYMBOL, ALLsum, AFRsum, EURsum, SASsum, EASsum, POLsum, contains("EAS"), contains("POL")) %>% arrange(desc(POLsum)) %>% filter(POLsum < ALLsum, !(SYMBOL %in% fwh_pos_no_pol_genes )) %>% pull(SYMBOL) %>% unique()
```

Of the genes that were in the windows of the 1^st^ percentile, there were `r fwh_met$neg_pol %>% length()` genes that were in the list of genes of metabolic disease associated genes and also had windows that intersected them from the Polynesian populations. Thirteen of those genes were for windows from Polynesian populations, as well as others. Some examples of these genes were _RASA2_ and _ERBB4_, and both are associated with obesity traits and involved in signalling pathways. _RASA2_ had windows only from \gls{cim} and \gls{nzm}. _ERBB4_ had windows from all four Polynesian populations, as well as the populations of \gls{eas} - except \gls{jpt}. There were `r fwh_met$pos_pol %>% length()` genes that were in the 99^th^ percentile that were also in the list of genes of metabolic disease associated genes. Some examples of genes that had windows in both Polynesian and other populations in the 99^th^ percentile include _SHROOM3_, _RBMS1_, and _UBE2E2_. _SHROOM3_ was an example of a gene that had windows in common from all the other super populations, except \gls{eas}. _RBMS1_ had multiple windows for across all Polynesian populations, but not other super populations. And _UBE2E2_ had multiple windows across the Polynesian populations, as well as the populations of \gls{eas} and \gls{sas}. _SHROOM3_ was an example of a gene associated with kidney disease, with its role being in the development of the kidney [@Khalili2016]. 


```{r, eval = FALSE}
fwh_pos_pol_genes %>% filter(SYMBOL %in% genelists$genename) %>% select(contains('chrom'), SYMBOL, contains("sum"))
```


#### Fu and Li's _F_

##### Fu and Li's distributions {#flfDist}

The distributions of \gls{flf} were all centred above zero, with the lowest mean being that of \gls{pol} at `r flf_super_pop_means %>% filter(super_pop == 'POL') %>% .[['mean']] %>% three_dp()` (Figure \@ref(fig:flfDensity) and Table \@ref(tab:superGlobalSummaryFLF)). The \gls{pol} populations also had the lowest 1^st^ percentile (`r flf_super_pop_means %>% filter(super_pop == 'POL') %>% .[['lower_1']] %>% three_dp()`), median (`r flf_super_pop_means %>% filter(super_pop == 'POL') %>% .[['median']] %>% three_dp()`), and 99^th^ percentile (`r flf_super_pop_means %>% filter(super_pop == 'POL') %>% .[['upper_99']] %>% three_dp()`). Conversely, the \gls{sas} populations had the highest mean (`r flf_super_pop_means %>% filter(super_pop == 'SAS') %>% .[['mean']] %>% three_dp()`), median (`r flf_super_pop_means %>% filter(super_pop == 'SAS') %>% .[['median']] %>% three_dp()`), and were equal with the \gls{afr} populations with the highest minimum (`r flf_super_pop_means %>% filter(super_pop == 'SAS') %>% .[['min']] %>% three_dp()`). The \gls{afr} populations were the only group to have a positive 1^st^ percentile (`r flf_super_pop_means %>% filter(super_pop == 'AFR') %>% .[['lower_1']] %>% three_dp()`). The \gls{eas} populations had the lowest minimum (`r flf_super_pop_means %>% filter(super_pop == 'EAS') %>% .[['min']] %>% three_dp()`), but the highest 99^th^ percentile (`r flf_super_pop_means %>% filter(super_pop == 'EAS') %>% .[['upper_99']] %>% three_dp()`) and the largest range (`r flf_super_pop_means %>% filter(super_pop == 'EAS') %>% .[['range']] %>% three_dp()`). This pattern was similar to what was seen with \gls{td}.

(ref:flfDensity) Plot of the distribution of \gls{flf} by population. Triangles indicate the minimum and maximum values. Dots indicate the 1^st^ and 99^th^ percentiles for each population.

```{r flfDensity, fig.cap = '(ref:flfDensity)',echo = FALSE}
knitr::include_graphics("images/04_clustering/flf_densities.png")
```

```{r}
global_super_summary %>%filter(stat == "Fu.Li.F") %>% select(-stat) %>% kable(., caption ='(\\#tab:superGlobalSummaryFLF) Summary statistics for \\gls{flf} by super population.', digits = 3, booktabs =TRUE, escape = FALSE, col.names = c("Super Population", "Mean","SD","Min","1st Percentile","Median","99th Percentile","Max"), format = 'latex', linesep = "") %>% kableExtra::kable_styling(latex_options = 'scale_down')
```

(ref:flfTop1000) _\gls{flf} Clustering._ Hierarchical clustering of \gls{flf} using the upper and lower tails of the distribution. **A** and **B** Dendrogram representing the clusters of \gls{flf} lower and upper tails respectfully using hierarchical clustering with Euclidean distance and complete linkage. Coloured by super population. **C** and **D** Plots representing the windows present in each population from the lower and upper tails respectfully. A blue line represents presence of a window, white represents the absence of a window in the 1% of the distribution for a population.

```{r flfTop1000, fig.cap = '(ref:flfTop1000)', out.width = "6in"}
#ggsave(four_plot("flf","Fu and Li's F"), filename = 'images/04_clustering/flfTop1000.png', device = 'png', height = 7, width = 7)
knitr::include_graphics('images/04_clustering/flfTop1000.png')
```

<!--
- pos clustered super populations together, split AMR
- pos looks like there is a reasonable number of windows shared between pops (5188)
- neg looks pretty sparse 17044
-->

<!--
[]describe number of total number of windows
[x]polynesian specific windows
[x]ep specific windows
[x]wp specific windows
[x]cim/nzm/sam/ton specific windows

[]proportions
[x]exclusivity
[]dendrograms

[]refer to figure
-->




Clustering on the 1% \gls{flf} from each tail of the distribution (Figure \@ref(fig:flfTop1000)) had `r create_windowSummaryTable() %>% filter(stat == 'flf_pos') %>% .[['total_windows']]` windows in the upper tail (Figure \@ref(fig:flfTop1000) B and D) and `r create_windowSummaryTable() %>% filter(stat == 'flf_neg') %>% .[['total_windows']]` windows in the lower tail (Figure \@ref(fig:flfTop1000) A and C). The mean number of unique windows per population was `r create_sums('flf_pos') %>% filter(ALLsum == 1) %>% select(-contains('chrom'), -posid, -contains('sum')) %>% summarise_all(funs(sum)) %>% t() %>% mean() %>% round(., digits = 1)` (SD `r create_sums('flf_pos') %>% filter(ALLsum == 1) %>% select(-contains('chrom'), -posid, -contains('sum')) %>% summarise_all(funs(sum)) %>% t() %>% sd() %>% round(., digits = 1)`) in the upper tail, compared with `r create_sums('flf_neg') %>% filter(ALLsum == 1) %>% select(-contains('chrom'), -posid, -contains('sum')) %>% summarise_all(funs(sum)) %>% t() %>% mean() %>% round(., digits = 1)` (SD `r create_sums('flf_neg') %>% filter(ALLsum == 1) %>% select(-contains('chrom'), -posid, -contains('sum')) %>% summarise_all(funs(sum)) %>% t() %>% sd() %>% round(., digits = 1)`) in the lower tail. Both tails clustered the \gls{eas} populations into their own exclusive cluster.


##### Clustering Fu and Li's _F_ 1^st^ percentile

```{r, eval = TRUE, warning = FALSE, message = FALSE}
# get all of the windows that had a Pol hit
flf_neg_pol <- create_sums('flf_neg') %>% filter(POLsum > 0) %>% mutate(chrom = paste0('chr', chrom))

# annotate the windows with gene names
flf_neg_pol_genes <- left_join(flf_neg_pol, txdb_gene_annotate(GenomicRanges::GRanges(flf_neg_pol)), by = c('chrom' = 'seqnames', "chrom_start" = 'start', "chrom_end" = 'end')) %>% select(contains('chrom'), SYMBOL, everything(), -width, -posid, -TXSTRAND, -strand)
write.csv(file ='Appendices/04-flf-pol-firstper-genes.csv', x = flf_neg_pol_genes, quote = TRUE, row.names = FALSE, col.names = TRUE )

# Look at the genes to see how many other pops had them too
#flf_neg_pol_genes %>% filter( SYMBOL %in% genelists$genename) %>% select(contains('chrom'),SYMBOL, ALLsum, AFRsum, EURsum, SASsum, EASsum, POLsum, contains("EAS"), contains("POL")) %>% arrange(desc(POLsum)) %>% View()

# find the number of genes from the disease genes that the polynesians had
#flf_neg_pol_genes %>% filter( SYMBOL %in% genelists$genename) %>% select(contains('chrom'),SYMBOL, ALLsum, AFRsum, EURsum, SASsum, EASsum, POLsum, contains("EAS"), contains("POL")) %>% arrange(desc(POLsum)) %>% pull(SYMBOL) %>% unique() %>% length()

# get the genes that were in windows that the Pol pops weren't in
flf_no_pol_genes <- txdb_gene_annotate(create_sums('flf_neg') %>% filter(POLsum == 0) %>% mutate(chrom = paste0('chr', chrom)) %>% GenomicRanges::GRanges()) %>% pull(SYMBOL) %>% unique()
# find how many genes were only found with the Pols + others
#flf_neg_pol_genes %>% filter(!SYMBOL %in% flf_no_pol_genes, SYMBOL %in% genelists$genename) %>% pull(SYMBOL) %>% unique() %>% length()
```

Clustering on the lower tail windows grouped the majority of the individual populations together into their super populations. The Polynesian populations were grouped together as a West Polynesian group and the \gls{cim} population. The \gls{nzm} population was grouped with the \gls{mxl} and \gls{pel} as part of a larger group with the \gls{eur} super population. Exclusivity of clusters was reasonable with four of six being exclusive. There were a total of `r create_windowSummaryTable() %>% filter(stat == 'flf_neg') %>% .[['total_windows']]` windows with `r create_sums('flf_neg') %>% filter(POLsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows specific to Polynesian populations. This compared to `r create_sums('flf_neg') %>% filter(AFRsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{afr}, `r create_sums('flf_neg') %>% filter(AMRsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{amr}, `r create_sums('flf_neg') %>% filter(EURsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{eur}, and `r create_sums('flf_neg') %>% filter(EASsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{eas}. The Eastern Polynesian populations had a total of `r create_sums('flf_neg') %>% filter(POL_CIM + POL_NZM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` unique windows, `r create_sums('flf_neg') %>% filter(POL_CIM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` were unique to \gls{cim} and `r create_sums('flf_neg') %>% filter(POL_NZM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` were unique to \gls{nzm}. The Western Polynesian populations had `r create_sums('flf_neg') %>% filter(POL_SAM + POL_TON == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` unique windows. Of these, `r create_sums('flf_neg') %>% filter(POL_SAM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` were unique to \gls{sam} and `r create_sums('flf_neg') %>% filter(POL_TON == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` to \gls{ton}. There were `r flf_neg_pol_genes %>% filter(POLsum == 4, !is.na(SYMBOL)) %>% pull(SYMBOL) %>% unique() %>% length()` genes that overlapped windows in the 1^st^ percentiles of all four Polynesian populations. Of these genes, three (_OR4C11_, _OR4C15_, and _OR4C16_) were involved with the significantly enriched "Olfactory transduction_Homo sapiens_hsa04740" pathway (adjusted P = 0.044).



##### Clustering Fu and Li's _F_ 99^th^ percentile

```{r, eval = TRUE, warning = FALSE, message = FALSE}
# get all of the windows that had a Pol hit
flf_pos_pol <- create_sums('flf_pos') %>% filter(POLsum > 0) %>% mutate(chrom = paste0('chr', chrom))

# annotate the windows with gene names
flf_pos_pol_genes <- left_join(flf_pos_pol, txdb_gene_annotate(GenomicRanges::GRanges(flf_pos_pol)), by = c('chrom' = 'seqnames', "chrom_start" = 'start', "chrom_end" = 'end')) %>% select(contains('chrom'), SYMBOL, everything(), -width, -posid, -TXSTRAND, -strand)
write.csv(file ='Appendices/04-flf-pol-ninetyninethper-genes.csv', x = flf_pos_pol_genes, quote = TRUE, row.names = FALSE, col.names = TRUE )

# Look at the genes to see how many other pops had them too
#flf_pos_pol_genes %>% filter( SYMBOL %in% genelists$genename) %>% select(contains('chrom'),SYMBOL, ALLsum, AFRsum, EURsum, SASsum, EASsum, POLsum, contains("EAS"), contains("POL")) %>% arrange(desc(POLsum)) %>% View()

# find the number of genes from the disease genes that the polynesians had
#flf_pos_pol_genes %>% filter( SYMBOL %in% genelists$genename) %>% select(contains('chrom'),SYMBOL, ALLsum, AFRsum, EURsum, SASsum, EASsum, POLsum, contains("EAS"), contains("POL")) %>% arrange(desc(POLsum)) %>% pull(SYMBOL) %>% unique() %>% length()

# get the genes that were in windows that the Pol pops weren't in
flf_no_pol_genes <- txdb_gene_annotate(create_sums('flf_pos') %>% filter(POLsum == 0) %>% mutate(chrom = paste0('chr', chrom)) %>% GenomicRanges::GRanges()) %>% pull(SYMBOL) %>% unique()
# find how many genes were only found with the Pols + others
#flf_pos_pol_genes %>% filter(!SYMBOL %in% flf_no_pol_genes, SYMBOL %in% genelists$genename) %>% pull(SYMBOL) %>% unique() %>% length()
```


Clustering on the windows from the upper tail put the super populations into their own groups, but also split the \gls{amr} populations into two separate groups. The Polynesian populations were also clustered into an East/West split of 2 separate exclusive clusters. Exclusivity of clusters was high with five of the six clusters being exclusive to a single super population, however one cluster contained half of the \gls{amr} populations (\gls{clm} and \gls{pur}) and all of the \gls{eur} and \gls{sas} populations. There were a total of `r create_windowSummaryTable() %>% filter(stat == 'flf_pos') %>% .[['total_windows']]` windows with `r create_sums('flf_pos') %>% filter(POLsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows specific to Polynesian populations. This compared to `r create_sums('flf_pos') %>% filter(AFRsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{afr}, `r create_sums('flf_pos') %>% filter(AMRsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{amr}, `r create_sums('flf_pos') %>% filter(EURsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{eur}, and `r create_sums('flf_pos') %>% filter(EASsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{eas}. The Eastern Polynesian populations had a total of `r create_sums('flf_pos') %>% filter(POL_CIM + POL_NZM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` unique windows with `r create_sums('flf_pos') %>% filter(POL_CIM == ALLsum & ALLsum > 0 ) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` only from \gls{cim} and `r create_sums('flf_pos') %>% filter(POL_NZM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` from \gls{nzm}. From the Western Polynesian populations there were `r create_sums('flf_pos') %>% filter(POL_SAM + POL_TON == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` specific windows. There were `r create_sums('flf_pos') %>% filter(POL_SAM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` specific to \gls{sam} and `r create_sums('flf_pos') %>% filter(POL_TON == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` specific to \gls{ton}. There were `r flf_pos_pol_genes %>% filter(POLsum == 4, !is.na(SYMBOL)) %>% pull(SYMBOL) %>% unique() %>% length()` genes that overlapped windows in the 99^th^ percentiles of all four Polynesian populations. 


##### Metabolic disease associated genes

There were `r flf_neg_pol_genes %>% filter(SYMBOL %in% genelists$genename) %>% pull(SYMBOL) %>% unique() %>% length()` genes that were in the list of genes of metabolic disease associated genes and also had windows in the 1^st^ percentile,that intersected them from the Polynesian populations. Some examples of these genes were _MAFC1_ and _DNAH10_. _MAFC1_ had six windows, some of which were in common with the \gls{amr} and \gls{afr} populations. _DNAH10_ had 18 windows, with about half being in common with the populations of \gls{eur}, \gls{eas}, \gls{sas}, and \gls{amr}. There were `r flf_pos_pol_genes %>% filter(SYMBOL %in% genelists$genename) %>% pull(SYMBOL) %>% unique() %>% length()` of genes that were in the 99^th^ percentile for Polynesians. Some examples were _HLA-B_, _PTPRD_, _LIPC_, _SHROOM3_, and _ABO_. _HLA-B_ was in common with all the super populations, although only half of the populations of \gls{afr}. _PTPRD_ was mostly only the four \gls{pol} populations with a few windows that were in common with a population for either \gls{sas} or \gls{amr}. _LIPC_ had windows in common for all Polynesian populations, and also shared with \gls{eas}, \gls{sas}, and \gls{amr}. _SHROOM3_ was in the Eastern Polynesian populations, but also in common with \gls{eas}, and to a lesser degree \gls{eur} and \gls{amr} and \gls{afr}. _ABO_ was in the Eastern Polynesian populations and also in common with \gls{eas}, \gls{sas}, and \gls{afr}.


```{r, eval = FALSE}
flf_neg_pol_genes %>% filter(SYMBOL %in% genelists$genename) %>% select(contains("chrom"),SYMBOL,ALLsum, contains("sum"), contains("POL")) %>% arrange(desc(POLsum)) %>% View()

flf_pos_pol_genes %>% filter(SYMBOL %in% genelists$genename) %>% select(contains("chrom"),SYMBOL,ALLsum, contains("sum"), contains("POL")) %>% arrange(desc(POLsum)) %>% View()
```


#### Zeng's _E_


##### Zeng's _E_ distributions {#zeDist}

Similar to \gls{td}, all of the distributions for \gls{ze} appear to have been shifted to the right (Figure \@ref(fig:zeDensity) and Table \@ref(tab:superGlobalSummaryZE)). The \gls{afr} populations had the lowest overall distribution, with the lowest mean (`r ze_super_pop_means %>% filter(super_pop == 'AFR') %>% .[['mean']] %>% three_dp()`), 1^st^ percentile (`r ze_super_pop_means %>% filter(super_pop == 'AFR') %>% .[['lower_1']] %>% three_dp()`), median (`r ze_super_pop_means %>% filter(super_pop == 'AFR') %>% .[['median']] %>% three_dp()`), 99^th^ percentile (`r ze_super_pop_means %>% filter(super_pop == 'AFR') %>% .[['upper_99']] %>% three_dp()`), maximum (`r ze_super_pop_means %>% filter(super_pop == 'AFR') %>% .[['max']] %>% three_dp()`), and range (`r ze_super_pop_means %>% filter(super_pop == 'AFR') %>% .[['range']] %>% three_dp()`). The \gls{eur} populations that had the lowest minimum (`r ze_super_pop_means %>% filter(super_pop == 'EUR') %>% .[['min']] %>% three_dp()`) and the highest maximum (`r ze_super_pop_means %>% filter(super_pop == 'EUR') %>% .[['max']] %>% three_dp()`), which also meant the largest range (`r ze_super_pop_means %>% filter(super_pop == 'EUR') %>% .[['range']] %>% three_dp()`). The highest mean of `r ze_super_pop_means %>% filter(super_pop == 'EAS') %>% .[['mean']] %>% three_dp()` was from \gls{eas}. The \gls{pol} populations had the second highest mean (`r ze_super_pop_means %>% filter(super_pop == 'POL') %>% .[['mean']] %>% three_dp()`), median (`r ze_super_pop_means %>% filter(super_pop == 'POL') %>% .[['median']] %>% three_dp()`), 99^th^ percentile (`r ze_super_pop_means %>% filter(super_pop == 'POL') %>% .[['upper_99']] %>% three_dp()`), and maximum (`r ze_super_pop_means %>% filter(super_pop == 'POL') %>% .[['max']] %>% three_dp()`). 
 

(ref:zeDensity) Plot of the distribution of \gls{ze} by population. Triangles indicate the minimum and maximum values. Dots indicate the 1^st^ and 99^th^ percentiles for each population.

```{r zeDensity, fig.cap = '(ref:zeDensity)',echo = FALSE}
knitr::include_graphics("images/04_clustering/ze_densities.png")
```

```{r}
global_super_summary %>%filter(stat == "Zeng.E") %>% select(-stat) %>% kable(., caption ='(\\#tab:superGlobalSummaryZE) Summary statistics for \\gls{ze} by super population.', digits = 3, booktabs =TRUE, escape = FALSE, col.names = c("Super Population", "Mean","SD","Min","1st Percentile","Median","99th Percentile","Max"), format = 'latex', linesep = "") %>% kableExtra::kable_styling(latex_options = 'scale_down')
```

(ref:zeTop1000) _\gls{ze} Clustering._ Hierarchical clustering of \gls{ze} using the upper and lower tails of the distribution. **A** and **B** Dendrogram representing the clusters of \gls{ze} lower and upper tails respectfully using hierarchical clustering with Euclidean distance and complete linkage. Coloured by super population. **C** and **D** Plots representing the windows present in each population from the lower and upper tails respectfully. A blue line represents presence of a window, white represents the absence of a window in the 1% of the distribution for a population.

```{r zeTop1000, fig.cap = '(ref:zeTop1000)', out.width = "6in"}
#ggsave(four_plot("ze","Zeng's E"), filename = 'images/04_clustering/zeTop1000.png', device = 'png', height = 7, width = 7)
knitr::include_graphics('images/04_clustering/zeTop1000.png')
```

<!--
Clustering of \gls{ze} on the positive tail (Figure \@ref(fig:zeTop1000) B and D) of the distribution gave clustering that matched the super popualations, with the Polyesian populations grouped together and subgrouped into an East/West split. The Polyensian population group's next closest group was the \gls{eas} super population. The negative tail (Figure \@ref(fig:zeTop1000) A and C) showed a similar grouping but split the \gls{amr} super population.
-->



\Gls{ze} for both extremes of the distribution (Figure \@ref(fig:zeTop1000)), clustered all populations except \gls{amr} into their respective super populations. The \gls{amr} population was split into \gls{clm} and \gls{pur} which was grouped with the \gls{eur} populations, and \gls{mxl} and \gls{pel} which were grouped as a separate exclusive cluster. The groupings of all super populations was the same for both tails, however, in the dendrogram the difference comes from where the \gls{amr} populations were relative to the other super populations with the \gls{mxl} and \gls{pur} populations changing slightly in how close to the \gls{sas} group they were. Exclusivity of clusters was high with only the \gls{amr} -\gls{eur} group not being exclusive to a single super population. 

##### Clustering Zeng's _E_ 1^st^ percentile

There were a total of `r create_windowSummaryTable() %>% filter(stat == 'ze_neg') %>% .[['total_windows']]` windows, with `r create_sums('ze_neg') %>% filter(POLsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows specific to Polynesian populations in the lower tail. This compared to `r create_sums('ze_neg') %>% filter(AFRsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{afr}, `r create_sums('ze_neg') %>% filter(AMRsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{amr}, `r create_sums('ze_neg') %>% filter(EURsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{eur}, and `r create_sums('ze_neg') %>% filter(EASsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{eas}. There were a total of `r create_sums('ze_neg') %>% filter(POL_CIM + POL_NZM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows specific to the Eastern Polynesian populations, with `r create_sums('ze_neg') %>% filter(POL_CIM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` specific to \gls{cim} and `r create_sums('ze_neg') %>% filter(POL_NZM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` specific to \gls{nzm}. The Western Polynesian populations had `r create_sums('ze_neg') %>% filter(POL_SAM + POL_TON == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` unique windows, `r create_sums('ze_neg') %>% filter(POL_SAM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` were unique to \gls{sam} and `r create_sums('ze_neg') %>% filter(POL_TON == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` unique to \gls{ton}. The mean number of unique windows was `r create_sums('ze_neg') %>% filter(ALLsum == 1) %>% select(-contains('chrom'), -posid, -contains('sum')) %>% summarise_all(funs(sum)) %>% t() %>% mean() %>% round(., digits = 1)` (SD `r create_sums('ze_neg') %>% filter(ALLsum == 1) %>% select(-contains('chrom'), -posid, -contains('sum')) %>% summarise_all(funs(sum)) %>% t() %>% sd() %>% round(., digits = 1)`) in the lower tail. There were `r create_sums('ze_neg') %>% filter(POLsum == ALLsum & POLsum == 4) %>% mutate(chrom = paste0('chr', chrom)) %>%select(1:3) %>% GenomicRanges::GRanges() %>% GenomicRanges::reduce() %>% txdb_gene_annotate() %>% as_tibble() %>% filter(!is.na(SYMBOL)) %>% pull(SYMBOL) %>% unique() %>% length()` genes that intersected windows that were only in the four Polynesian populations and no other populations. None of these genes were associated with urate or co-morbidities.

```{r, eval = TRUE, warning = FALSE, message = FALSE}
# get all of the windows that had a Pol hit
ze_neg_pol <- create_sums('ze_neg') %>% filter(POLsum > 0) %>% mutate(chrom = paste0('chr', chrom))

# annotate the windows with gene names
ze_neg_pol_genes <- left_join(ze_neg_pol, txdb_gene_annotate(GenomicRanges::GRanges(ze_neg_pol)), by = c('chrom' = 'seqnames', "chrom_start" = 'start', "chrom_end" = 'end')) %>% select(contains('chrom'), SYMBOL, everything(), -width, -posid, -TXSTRAND, -strand)
write.csv(file ='Appendices/04-ze-pol-firstper-genes.csv', x = ze_neg_pol_genes, quote = TRUE, row.names = FALSE, col.names = TRUE )

# Look at the genes to see how many other pops had them too
#ze_neg_pol_genes %>% filter( SYMBOL %in% genelists$genename) %>% select(contains('chrom'),SYMBOL, ALLsum, AFRsum, EURsum, SASsum, EASsum, POLsum, contains("EAS"), contains("POL")) %>% arrange(desc(POLsum)) %>% View()

# find the number of genes from the disease genes that the polynesians had
#ze_neg_pol_genes %>% filter( SYMBOL %in% genelists$genename) %>% select(contains('chrom'),SYMBOL, ALLsum, AFRsum, EURsum, SASsum, EASsum, POLsum, contains("EAS"), contains("POL")) %>% arrange(desc(POLsum)) %>% pull(SYMBOL) %>% unique() %>% length()

# get the genes that were in windows that the Pol pops weren't in
ze_no_pol_genes <- txdb_gene_annotate(create_sums('ze_neg') %>% filter(POLsum == 0) %>% mutate(chrom = paste0('chr', chrom)) %>% GenomicRanges::GRanges()) %>% pull(SYMBOL) %>% unique()
# find how many genes were only found with the Pols + others
#ze_neg_pol_genes %>% filter(!SYMBOL %in% ze_no_pol_genes, SYMBOL %in% genelists$genename) %>% pull(SYMBOL) %>% unique() %>% length()
```



##### Clustering Zeng's _E_ 99^th^ percentile

In the upper tail there were a total of `r create_windowSummaryTable() %>% filter(stat == 'ze_pos') %>% .[['total_windows']]` windows with `r create_sums('ze_pos') %>% filter(POLsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows specific to Polynesian populations. This compared to `r create_sums('ze_pos') %>% filter(AFRsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{afr}, `r create_sums('ze_pos') %>% filter(AMRsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{amr}, `r create_sums('ze_pos') %>% filter(EURsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{eur}, and `r create_sums('ze_pos') %>% filter(EASsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{eas}. The Eastern Polynesian populations had `r create_sums('ze_pos') %>% filter(POL_CIM + POL_NZM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` unique windows, the \gls{cim} had`r create_sums('ze_pos') %>% filter(POL_CIM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)`unique windows, and \gls{nzm} had `r create_sums('ze_pos') %>% filter(POL_NZM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` unique windows. The Western Polynesian populations had `r create_sums('ze_pos') %>% filter(POL_SAM + POL_TON == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` unique windows. Of those, `r create_sums('ze_pos') %>% filter(POL_SAM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` were unique to \gls{sam} and `r create_sums('ze_pos') %>% filter(POL_TON == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` unique to \gls{ton}. The mean number of unique windows per population was `r create_sums('ze_pos') %>% filter(ALLsum == 1) %>% select(-contains('chrom'), -posid, -contains('sum')) %>% summarise_all(funs(sum)) %>% t() %>% mean() %>% round(., digits = 1)` (SD `r create_sums('ze_pos') %>% filter(ALLsum == 1) %>% select(-contains('chrom'), -posid, -contains('sum')) %>% summarise_all(funs(sum)) %>% t() %>% sd() %>% round(., digits = 1)`)


```{r, eval = TRUE, warning = FALSE, message = FALSE}
# get all of the windows that had a Pol hit
ze_pos_pol <- create_sums('ze_pos') %>% filter(POLsum > 0) %>% mutate(chrom = paste0('chr', chrom))

# annotate the windows with gene names
ze_pos_pol_genes <- left_join(ze_pos_pol, txdb_gene_annotate(GenomicRanges::GRanges(ze_pos_pol)), by = c('chrom' = 'seqnames', "chrom_start" = 'start', "chrom_end" = 'end')) %>% select(contains('chrom'), SYMBOL, everything(), -width, -posid, -TXSTRAND, -strand)
write.csv(file ='Appendices/04-ze-pol-ninetyninethper-genes.csv', x = ze_pos_pol_genes, quote = TRUE, row.names = FALSE, col.names = TRUE )

# Look at the genes to see how many other pops had them too
#ze_pos_pol_genes %>% filter( SYMBOL %in% genelists$genename) %>% select(contains('chrom'),SYMBOL, ALLsum, AFRsum, EURsum, SASsum, EASsum, POLsum, contains("EAS"), contains("POL")) %>% arrange(desc(POLsum)) %>% View()

# find the number of genes from the disease genes that the polynesians had
#ze_pos_pol_genes %>% filter( SYMBOL %in% genelists$genename) %>% select(contains('chrom'),SYMBOL, ALLsum, AFRsum, EURsum, SASsum, EASsum, POLsum, contains("EAS"), contains("POL")) %>% arrange(desc(POLsum)) %>% pull(SYMBOL) %>% unique() %>% length()

# get the genes that were in windows that the Pol pops weren't in
ze_no_pol_genes <- txdb_gene_annotate(create_sums('ze_pos') %>% filter(POLsum == 0) %>% mutate(chrom = paste0('chr', chrom)) %>% GenomicRanges::GRanges()) %>% pull(SYMBOL) %>% unique()
# find how many genes were only found with the Pols + others
#ze_pos_pol_genes %>% filter(!SYMBOL %in% ze_no_pol_genes, SYMBOL %in% genelists$genename) %>% pull(SYMBOL) %>% unique() %>% length()
```


<!-- create df from the clustering results to be able to pull numbers out of for number of windows per population for each statistic, and for the proportion of pops being clustered -->
##### Metabolic associated diseases

```{r}
ze_met <- list()
ze_met$neg_pol <- ze_neg_pol_genes %>% filter( SYMBOL %in% genelists$genename) %>% select(contains('chrom'),SYMBOL, ALLsum, AFRsum, EURsum, SASsum, EASsum, POLsum, contains("EAS"), contains("POL")) %>% arrange(desc(POLsum)) %>% filter(POLsum > 0) %>% pull(SYMBOL) %>% unique()

ze_met$neg_not_only_pol <-  ze_neg_pol_genes %>% filter( SYMBOL %in% genelists$genename) %>% select(contains('chrom'),SYMBOL, ALLsum, AFRsum, EURsum, SASsum, EASsum, POLsum, contains("EAS"), contains("POL")) %>% arrange(desc(POLsum)) %>% filter(POLsum < ALLsum, !(SYMBOL %in% ze_neg_no_pol_genes )) %>% pull(SYMBOL) %>% unique()


ze_met$pos_pol <- ze_pos_pol_genes %>% filter( SYMBOL %in% genelists$genename) %>% select(contains('chrom'),SYMBOL, ALLsum, AFRsum, EURsum, SASsum, EASsum, POLsum, contains("EAS"), contains("POL")) %>% arrange(desc(POLsum)) %>% filter(POLsum > 0) %>% pull(SYMBOL) %>% unique()

# genes that were in polynesians but also other populations
ze_met$pos_not_only_pol <-  ze_pos_pol_genes %>% filter( SYMBOL %in% genelists$genename) %>% select(contains('chrom'),SYMBOL, ALLsum, AFRsum, EURsum, SASsum, EASsum, POLsum, contains("EAS"), contains("POL")) %>% arrange(desc(POLsum)) %>% filter(POLsum < ALLsum, !(SYMBOL %in% ze_pos_no_pol_genes )) %>% pull(SYMBOL) %>% unique()
```

There were `r ze_neg_pol_genes %>% filter(SYMBOL %in% genelists$genename) %>% pull(SYMBOL) %>% unique() %>% length()` genes that were in the list of genes of metabolic disease associated genes and also had windows in the 1^st^ percentile, that intersected them from the Polynesian populations. Some examples of these genes are _MACF1_, _AGBL4_, _DACH1_, and _RBMS1_. _MACF1_ had windows that were in the 1^st^ percentile of all populations. _AGBL4_ and _DACH1_ had windows in nearly every population except for most \gls{afr} populations. _RBMS1_ was only in the 1^st^ percentile for the Western Polynesian populations \gls{sam} and \gls{ton}. Windows of the 99^th^ percentile in the Polynesian populations that intersected the metabolic disease associated genes tended to also be found in the populations of \gls{eas}, and many also were in common with all other populations but not many were in common with the \gls{afr} populations. There were `r ze_pos_pol_genes %>% filter(SYMBOL %in% genelists$genename) %>% pull(SYMBOL) %>% unique() %>% length()` of genes that were in the 99^th^ percentile for Polynesians. Some examples were _CHST8_, _ERBB4_, _FHIT_, _PTPRD_, and _HMGA2_. _CHST8_ had windows that were in the 99^th^ percentile for the four Polynesian populations, as well as the populations of \gls{eas} with a couple of populations of both \gls{sas} and \gls{eur} too. _ERBB4_ had windows that were in both \gls{pol} and \gls{eas} populations. _FHIT_ had windows for the Western Polynesian populations, \gls{sam} and \gls{ton} and most of the populations from \gls{eas} and \gls{sas}. _PTPRD_ had windows from the Eastern Polynesian populations of \gls{cim} and \gls{nzm}, these windows were also in common with populations mostly from \gls{eur}, while some other windows were in common with the Eastern Polynesian populations and \gls{afr}. _HMGA2_ had windows in all of the super populations except \gls{afr}.

```{r}

```


\FloatBarrier
 
### Hierarchical clustering of selected haplotypic regions

```{r load_ihs_clus}
mat_ihs <- readRDS('~/data/NZ_coreExome_1kgp/haplotype/ihs_clustered_dist_mat-15-12-2017.RDS')
clus_ihs <- readRDS('~/data/NZ_coreExome_1kgp/haplotype/ihs_clus_regions-15-12-2017.RDS')
gr_ihs <- GenomicRanges::GRanges(as.data.frame(clus_ihs))
```

```{r, eval = FALSE}
#nzm vs cim
sum(GenomicRanges::width(GenomicRanges::reduce(GenomicRanges::intersect(gr_ihs[gr_ihs$pop == "POL_NZM"], gr_ihs[gr_ihs$pop == "POL_CIM"]))))/sum(GenomicRanges::width(GenomicRanges::reduce(gr_ihs[gr_ihs$pop == "POL_NZM"])))

sum(GenomicRanges::width(GenomicRanges::reduce(GenomicRanges::intersect(gr_ihs[gr_ihs$pop == "POL_CIM"], gr_ihs[gr_ihs$pop == "POL_NZM"]))))/sum(GenomicRanges::width(GenomicRanges::reduce(gr_ihs[gr_ihs$pop == "POL_CIM"])))

sum(GenomicRanges::width(GenomicRanges::reduce(gr_ihs[gr_ihs$pop %in% c("POL_SAM","POL_TON")])))/sum(GenomicRanges::width(GenomicRanges::reduce(gr_ihs[gr_ihs$pop %in% c("POL_NZM", "POL_CIM", "POL_SAM","POL_TON")])))

```

A similar approach as with clustering of the frequency spectrum based statistics was used to investigate similarities in regions that had evidence of possible selection in the intra-population haplotypic based selection methods. To do this, the significant \gls{ihs} or \gls{nsl} results were clustered into regions, and the percentage of bases overlapping between populations was calculated in these regions. The proportion of shared regions, based on region overlap, was used to cluster populations, given the total region that was significant for a single population. This was to investigate if there was a commonality between populations in the haplotypic regions that had evidence for selection, the hypothesis being that populations in similar geographic regions would have had a similar environmental exposure, and therefore have similar regions of the genome under selective pressure.

Clustering on the proportion of shared regions (see section \@ref(haplocluster)) for both \gls{ihs} (Figure \@ref(fig:ihsclust)) and \gls{nsl} (Figure \@ref(fig:nslclust)) grouped the individual populations into their super populations, with the exception of the \gls{amr} populations which were split into two separate groups. The order of populations differed due to each population having a different total for the size of region that was significant. Both \gls{ihs} and \gls{nsl} gave similar groupings and had similar proportions of selected regions shared.


```{r, eval = FALSE}
#find max and min shared pops
mat_ihs %>% data.frame() %>% mutate(rn = row.names(.)) %>% select(rn, everything()) %>% gather("den", "value", 2:NCOL(.)) %>% filter(min(value) == value)
mat_ihs %>% data.frame() %>% mutate(rn = row.names(.)) %>% select(rn, everything()) %>% gather("den", "value", 2:NCOL(.)) %>% filter(rn != den) %>% filter(max(value) == value)

```

#### \gls{ihs}

For the \gls{ihs} significant regions, the populations with the least regions shared between them were \gls{cim} and \gls{clm}, with `r sprintf("%.1f",mat_ihs['POL_CIM','AMR_CLM'] *100)`% shared. The populations that had the largest proportion of region shared was \gls{nzc} and \gls{ceu} with `r sprintf("%.1f",mat_ihs['EAS_CHB','EAS_CHS'] *100)`% shared. Looking specifically at the proportion of regions shared between Polynesian populations, for the Eastern Polynesian populations there was \gls{nzm} with `r (mat_ihs[rownames(mat_ihs) == "POL_CIM", colnames(mat_ihs) == "POL_NZM"] * 100) %>% sprintf('%.1f',.)`% of regions shared with \gls{cim}, and \gls{cim} shared `r (mat_ihs[rownames(mat_ihs) == "POL_NZM", colnames(mat_ihs) == "POL_CIM"] * 100) %>% sprintf('%.1f',.)`% of regions with \gls{nzm}. In the Western Polynesian populations, \gls{sam} shared `r (mat_ihs[rownames(mat_ihs) == "POL_TON", colnames(mat_ihs) == "POL_SAM"] * 100) %>% sprintf('%.1f',.)`% of regions with \gls{ton}, but \gls{ton} shared `r (mat_ihs[rownames(mat_ihs) == "POL_SAM", colnames(mat_ihs) == "POL_TON"] * 100) %>% sprintf('%.1f',.)`% of regions with \gls{sam}. Out of all the significant \gls{ihs} regions for Polynesian populations, only `r (sum(GenomicRanges::width(GenomicRanges::reduce(GenomicRanges::intersect(GenomicRanges::intersect(gr_ihs[gr_ihs$pop == 'POL_CIM'], gr_ihs[gr_ihs$pop == "POL_NZM"]), GenomicRanges::intersect(gr_ihs[gr_ihs$pop == 'POL_SAM'], gr_ihs[gr_ihs$pop == "POL_TON"])))))/sum(GenomicRanges::width(GenomicRanges::reduce(gr_ihs[gr_ihs$pop %in% c("POL_NZM", "POL_CIM", "POL_SAM","POL_TON")]))) * 100) %>% sprintf('%.1f',.)`% were in common across all Polynesian populations. The Eastern Polynesians accounted for `r (sum(GenomicRanges::width(GenomicRanges::reduce(gr_ihs[gr_ihs$pop %in% c("POL_NZM","POL_CIM")])))/sum(GenomicRanges::width(GenomicRanges::reduce(gr_ihs[gr_ihs$pop %in% c("POL_NZM", "POL_CIM", "POL_SAM","POL_TON")]))) *100) %>% sprintf('%.1f',.)`% of the regions from all of the Polynesian populations while the Western Polynesian populations made up `r (sum(GenomicRanges::width(GenomicRanges::reduce(gr_ihs[gr_ihs$pop %in% c("POL_SAM","POL_TON")])))/sum(GenomicRanges::width(GenomicRanges::reduce(gr_ihs[gr_ihs$pop %in% c("POL_NZM", "POL_CIM", "POL_SAM","POL_TON")]))) * 100) %>% sprintf('%.1f',.)`%.



<!--
CHECK HOW IHS PROPS ARE BEING CALCULATED!!!

what was the overlap in regions like for ihs/nsl

ihs
DOUBLE CHECK THE ORDERS 
double checked and colname is the denominator
-->

```{r calc_ihs_region_share}
# mean sharing if ihs regions between all super populations
sp_list <- list()
super_pops <- c("POL","AMR","AFR","EUR","EAS","SAS")
for(sp1 in super_pops){
 for(sp2 in super_pops){
  if(sp1 != sp2){
   sp_list[[paste0(sp1,"_",sp2)]] <- sum(GenomicRanges::width(GenomicRanges::reduce(GenomicRanges::intersect(gr_ihs[gr_ihs$pop %in% unique(gr_ihs$pop[grep(sp1, gr_ihs$pop)])], gr_ihs[gr_ihs$pop %in% unique(gr_ihs$pop[grep(sp2, gr_ihs$pop)]) ]))))/sum(GenomicRanges::width(GenomicRanges::reduce(gr_ihs[gr_ihs$pop %in% unique(gr_ihs$pop[grep(sp1, gr_ihs$pop)])])))
  }
 }
}
mean_super_share_ihs <- mean(unlist(sp_list))
rm(super_pops, sp_list)
```

The proportion of significant \gls{ihs} regions the Polynesian populations shared with AMR was `r (sum(GenomicRanges::width(GenomicRanges::reduce(GenomicRanges::intersect(gr_ihs[gr_ihs$pop %in% unique(gr_ihs$pop[grep('POL', gr_ihs$pop)])], gr_ihs[gr_ihs$pop %in% unique(gr_ihs$pop[grep('AMR', gr_ihs$pop)]) ]))))/sum(GenomicRanges::width(GenomicRanges::reduce(gr_ihs[gr_ihs$pop %in% unique(gr_ihs$pop[grep('POL', gr_ihs$pop)])]))) * 100) %>% sprintf('%.1f',.)`%, AFR was `r (sum(GenomicRanges::width(GenomicRanges::reduce(GenomicRanges::intersect(gr_ihs[gr_ihs$pop %in% unique(gr_ihs$pop[grep('POL', gr_ihs$pop)])], gr_ihs[gr_ihs$pop %in% unique(gr_ihs$pop[grep('AFR', gr_ihs$pop)]) ]))))/sum(GenomicRanges::width(GenomicRanges::reduce(gr_ihs[gr_ihs$pop %in% unique(gr_ihs$pop[grep('POL', gr_ihs$pop)])]))) * 100) %>% sprintf('%.1f',.)`%, EAS was `r (sum(GenomicRanges::width(GenomicRanges::reduce(GenomicRanges::intersect(gr_ihs[gr_ihs$pop %in% unique(gr_ihs$pop[grep('POL', gr_ihs$pop)])], gr_ihs[gr_ihs$pop %in% unique(gr_ihs$pop[grep('EAS', gr_ihs$pop)]) ]))))/sum(GenomicRanges::width(GenomicRanges::reduce(gr_ihs[gr_ihs$pop %in% unique(gr_ihs$pop[grep('POL', gr_ihs$pop)])]))) * 100) %>% sprintf('%.1f',.)`%, EUR was `r (sum(GenomicRanges::width(GenomicRanges::reduce(GenomicRanges::intersect(gr_ihs[gr_ihs$pop %in% unique(gr_ihs$pop[grep('POL', gr_ihs$pop)])], gr_ihs[gr_ihs$pop %in% unique(gr_ihs$pop[grep('EUR', gr_ihs$pop)]) ]))))/sum(GenomicRanges::width(GenomicRanges::reduce(gr_ihs[gr_ihs$pop %in% unique(gr_ihs$pop[grep('POL', gr_ihs$pop)])]))) * 100) %>% sprintf('%.1f',.)`%, and SAS was `r (sum(GenomicRanges::width(GenomicRanges::reduce(GenomicRanges::intersect(gr_ihs[gr_ihs$pop %in% unique(gr_ihs$pop[grep('POL', gr_ihs$pop)])], gr_ihs[gr_ihs$pop %in% unique(gr_ihs$pop[grep('SAS', gr_ihs$pop)]) ]))))/sum(GenomicRanges::width(GenomicRanges::reduce(gr_ihs[gr_ihs$pop %in% unique(gr_ihs$pop[grep('POL', gr_ihs$pop)])]))) * 100) %>% sprintf('%.1f',.)`%. This was lower than the mean (`r sprintf("%.1f",mean_super_share_ihs * 100)`%) for proportion of regions shared between the other super populations. The Polynesian populations were clustered together with a high proportion of shared significant regions. The Eastern Polynesians had on average `r mat_ihs %>% data.frame %>% mutate(pop1 = row.names(.))%>% gather("pop2","ihs", 1:(NCOL(.)-1)) %>% filter(pop1!=pop2, (pop1 %in% c("POL_CIM","POL_NZM") & pop2 %in% c("POL_CIM","POL_NZM"))) %>% summarise(mean(ihs)*100) %>% sprintf('%.1f',.)`% in common, and the Western Polynesian populations had `r mat_ihs %>% data.frame %>% mutate(pop1 = row.names(.))%>% gather("pop2","ihs", 1:(NCOL(.)-1)) %>% filter(pop1!=pop2, (pop1 %in% c("POL_TON","POL_SAM") & pop2 %in% c("POL_TON","POL_SAM"))) %>% summarise(mean(ihs)*100) %>% sprintf('%.1f',.)`%. However, shared regions between the Eastern and Western Polynesian populations was `r mat_ihs %>% data.frame %>% mutate(pop1 = row.names(.))%>% gather("pop2","ihs", 1:(NCOL(.)-1)) %>% filter((pop1 %in% c("POL_CIM","POL_NZM") & pop2 %in% c("POL_TON","POL_SAM")) | (pop2 %in% c("POL_CIM","POL_NZM") & pop1 %in% c("POL_TON","POL_SAM"))) %>% summarise(mean(ihs)*100) %>% sprintf('%.1f',.)`% on average. Of the clustered regions of significant \gls{ihs} that were in common between all of the \gls{pol} populations, 70 genes intersected the regions. These genes were part of 57 pathways in the \gls{kegg} 2016 pathways in Enrichr, but none were significantly enriched. Regions that were in common between \gls{nzm} and \gls{cim} had 392 genes that intersected, and were involved with 174 pathways. No pathways were significantly enriched. For the regions that were in common between \gls{sam} and \gls{ton}, there were 502 genes that intersected the regions. They belonged to 223 pathways, with only a single pathway significantly enriched (Benjamini-Hochberg adjusted P = 0.023) with seven of 44 genes (_ABCA5_, _ABCC8_, _ABCC5_, _TAP2_, _ABCA9_, _ABCB8_, and _ABCA8_). That pathway was "ABC transporters_Homo sapiens_hsa02010".

```{r, eval = FALSE}
GenomicRanges::intersect(GenomicRanges::intersect(gr_ihs[gr_ihs$pop == "POL_NZM"], gr_ihs[gr_ihs$pop == "POL_CIM"], ignore.strand = TRUE ),
                        GenomicRanges::intersect(gr_ihs[gr_ihs$pop == "POL_SAM"], gr_ihs[gr_ihs$pop == "POL_TON"], ignore.strand = TRUE )) %>% data.frame() %>% mutate(seqnames = paste0('chr',seqnames)) %>% GenomicRanges::GRanges() %>% txdb_gene_annotate() %>% filter(!is.na(SYMBOL)) %>% pull(SYMBOL) %>% unique() %>% cat(sep = '\n')

#nzm/cim
GenomicRanges::intersect(gr_ihs[gr_ihs$pop == "POL_NZM"], gr_ihs[gr_ihs$pop == "POL_CIM"], ignore.strand = TRUE ) %>% data.frame() %>% mutate(seqnames = paste0('chr',seqnames)) %>% GenomicRanges::GRanges() %>% txdb_gene_annotate()%>% filter(!is.na(SYMBOL)) %>% pull(SYMBOL) %>% unique() %>% cat(sep= '\n')

#sam/ton
GenomicRanges::intersect(gr_ihs[gr_ihs$pop == "POL_SAM"], gr_ihs[gr_ihs$pop == "POL_TON"], ignore.strand = TRUE ) %>% data.frame() %>% mutate(seqnames = paste0('chr',seqnames)) %>% GenomicRanges::GRanges() %>% txdb_gene_annotate()%>% filter(!is.na(SYMBOL)) %>% pull(SYMBOL) %>% unique() %>% cat(sep = '\n')
```




```{r ihsclust, fig.cap='(ref:ihsclust)', echo = FALSE, fig.asp = 1}
ihs_dendro_plot()
#ggsave(last_plot(), file = 'images/04_clustering/ihs_dendros.png')
```

(ref:ihsclust) _\gls{ihs} Clustering_ A. Dendrogram from hierarchical clustering based on proportion of significant \gls{ihs} regions shared between populations, coloured by super population. B. Heatmap showing the proportion of significant \gls{ihs} regions shared. The population listed on the axis is used as A in the calculation = (A $\cap$ B)/ A. Axes have been ordered according to the hierarchical clustering. 




#### \gls{nsl}


<!-- nSL -->

```{r load_nsl_clus}
mat_nsl <- readRDS('~/data/NZ_coreExome_1kgp/haplotype/nsl_clustered_dist_mat-15-12-2017.RDS')
clus_nsl <- readRDS('~/data/NZ_coreExome_1kgp/haplotype/nsl_clus_regions-15-12-2017.RDS')
gr_nsl <- GenomicRanges::GRanges(as.data.frame(clus_nsl))
```

```{r calc_nsl_region_share}
# mean sharing if ihs regions between all super populations
sp_list <- list()
super_pops <- c("POL","AMR","AFR","EUR","EAS","SAS")
for(sp1 in super_pops){
  for(sp2 in super_pops){
    if(sp1 != sp2){
     sp_list[[paste0(sp1,"_",sp2)]] <-  sum(GenomicRanges::width(GenomicRanges::reduce(GenomicRanges::intersect(gr_nsl[gr_nsl$pop %in% unique(gr_nsl$pop[grep(sp1, gr_nsl$pop)])], gr_nsl[gr_nsl$pop %in% unique(gr_nsl$pop[grep(sp2, gr_nsl$pop)]) ]))))/sum(GenomicRanges::width(GenomicRanges::reduce(gr_nsl[gr_nsl$pop %in% unique(gr_nsl$pop[grep(sp1, gr_nsl$pop)])])))
    }
  }
}
mean_super_share_nsl <- mean(unlist(sp_list))
rm(super_pops, sp_list)
```

The clustering results using regions of significant \gls{nsl} gave the same groupings as \gls{ihs} (Figure \@ref(fig:nslclust)). This similarity was also seen with the differences in individual populations, where \gls{nzc} and \gls{gbr} had the highest percentage of shared regions at `r sprintf("%.1f",mat_nsl['EUR_NZC','EUR_GBR'] *100)`%. \Gls{nzm} and \gls{beb} shared the lowest percentage of regions at `r sprintf("%.1f",mat_nsl['POL_NZM','SAS_BEB'] *100)`%. The percentage of the significant \gls{nsl} regions the Polynesian populations shared with AMR was `r (sum(GenomicRanges::width(GenomicRanges::reduce(GenomicRanges::intersect(gr_nsl[gr_nsl$pop %in% unique(gr_nsl$pop[grep('POL', gr_nsl$pop)])], gr_nsl[gr_nsl$pop %in% unique(gr_nsl$pop[grep('AMR', gr_nsl$pop)]) ]))))/sum(GenomicRanges::width(GenomicRanges::reduce(gr_nsl[gr_nsl$pop %in% unique(gr_nsl$pop[grep('POL', gr_nsl$pop)])]))) * 100) %>% sprintf('%.1f',.)`%, AFR  was `r (sum(GenomicRanges::width(GenomicRanges::reduce(GenomicRanges::intersect(gr_nsl[gr_nsl$pop %in% unique(gr_nsl$pop[grep('POL', gr_nsl$pop)])], gr_nsl[gr_nsl$pop %in% unique(gr_nsl$pop[grep('AFR', gr_nsl$pop)]) ]))))/sum(GenomicRanges::width(GenomicRanges::reduce(gr_nsl[gr_nsl$pop %in% unique(gr_nsl$pop[grep('POL', gr_nsl$pop)])]))) * 100) %>% sprintf('%.1f',.)`%, EAS   was `r (sum(GenomicRanges::width(GenomicRanges::reduce(GenomicRanges::intersect(gr_nsl[gr_nsl$pop %in% unique(gr_nsl$pop[grep('POL', gr_nsl$pop)])], gr_nsl[gr_nsl$pop %in% unique(gr_nsl$pop[grep('EAS', gr_nsl$pop)]) ]))))/sum(GenomicRanges::width(GenomicRanges::reduce(gr_nsl[gr_nsl$pop %in% unique(gr_nsl$pop[grep('POL', gr_nsl$pop)])]))) * 100) %>% sprintf('%.1f',.)`%, EUR was `r (sum(GenomicRanges::width(GenomicRanges::reduce(GenomicRanges::intersect(gr_nsl[gr_nsl$pop %in% unique(gr_nsl$pop[grep('POL', gr_nsl$pop)])], gr_nsl[gr_nsl$pop %in% unique(gr_nsl$pop[grep('EUR', gr_nsl$pop)]) ]))))/sum(GenomicRanges::width(GenomicRanges::reduce(gr_nsl[gr_nsl$pop %in% unique(gr_nsl$pop[grep('POL', gr_nsl$pop)])]))) * 100) %>% sprintf('%.1f',.)`%, and SAS was `r (sum(GenomicRanges::width(GenomicRanges::reduce(GenomicRanges::intersect(gr_nsl[gr_nsl$pop %in% unique(gr_nsl$pop[grep('POL', gr_nsl$pop)])], gr_nsl[gr_nsl$pop %in% unique(gr_nsl$pop[grep('SAS', gr_nsl$pop)]) ]))))/sum(GenomicRanges::width(GenomicRanges::reduce(gr_nsl[gr_nsl$pop %in% unique(gr_nsl$pop[grep('POL', gr_nsl$pop)])]))) * 100) %>% sprintf('%.1f',.)`%. This was higher than the mean (`r sprintf("%.1f",mean_super_share_nsl * 100)`%) for \gls{eas}, but lower for the other super populations for proportion of region sharing between super populations. The Polynesian populations were clustered together with a high proportion of shared significant regions. The Eastern Polynesians had on average `r mat_nsl %>% data.frame %>% mutate(pop1 = row.names(.))%>% gather("pop2","nsl", 1:(NCOL(.)-1)) %>% filter(pop1!=pop2, (pop1 %in% c("POL_CIM","POL_NZM") & pop2 %in% c("POL_CIM","POL_NZM"))) %>% summarise(mean(nsl)*100) %>% sprintf('%.1f',.)`% in common. NZM shared `r (mat_nsl[rownames(mat_nsl) == "POL_CIM", colnames(mat_nsl) == "POL_NZM"] * 100) %>% sprintf('%.1f',.)`% of regions with CIM, whereas CIM shared `r (mat_nsl[rownames(mat_nsl) == "POL_NZM", colnames(mat_nsl) == "POL_CIM"] * 100) %>% sprintf('%.1f',.)`% of regions with NZM. The Western Polynesians had `r mat_nsl %>% data.frame %>% mutate(pop1 = row.names(.))%>% gather("pop2","nsl", 1:(NCOL(.)-1)) %>% filter(pop1!=pop2, (pop1 %in% c("POL_TON","POL_SAM") & pop2 %in% c("POL_TON","POL_SAM"))) %>% summarise(mean(nsl)*100) %>% sprintf('%.1f',.)`% of regions shared on average. SAM shared `r (mat_nsl[rownames(mat_nsl) == "POL_TON", colnames(mat_nsl) == "POL_SAM"] * 100) %>% sprintf('%.1f',.)`% of regions with TON, but TON shared `r (mat_nsl[rownames(mat_nsl) == "POL_SAM", colnames(mat_nsl) == "POL_TON"] * 100) %>% sprintf('%.1f',.)`% of regions with SAM. However, between the Eastern and Western populations was on average `r mat_nsl %>% data.frame %>% mutate(pop1 = row.names(.))%>% gather("pop2","nsl", 1:(NCOL(.)-1)) %>% filter((pop1 %in% c("POL_CIM","POL_NZM") & pop2 %in% c("POL_TON","POL_SAM")) | (pop2 %in% c("POL_CIM","POL_NZM") & pop1 %in% c("POL_TON","POL_SAM"))) %>% summarise(mean(nsl)*100) %>% sprintf('%.1f',.)`%. Of the regions from all of the Polynesian populations, the Western Polynesian populations made up `r (sum(GenomicRanges::width(GenomicRanges::reduce(gr_nsl[gr_nsl$pop %in% c("POL_SAM","POL_TON")])))/sum(GenomicRanges::width(GenomicRanges::reduce(gr_nsl[gr_nsl$pop %in% c("POL_NZM", "POL_CIM", "POL_SAM","POL_TON")]))) * 100) %>% sprintf('%.1f',.)`%. The Eastern Polynesians made up `r (sum(GenomicRanges::width(GenomicRanges::reduce(gr_nsl[gr_nsl$pop %in% c("POL_NZM","POL_CIM")])))/sum(GenomicRanges::width(GenomicRanges::reduce(gr_nsl[gr_nsl$pop %in% c("POL_NZM", "POL_CIM", "POL_SAM","POL_TON")]))) *100) %>% sprintf('%.1f',.)`%. The total percentage of regions common for all Polynesian populations was `r (sum(GenomicRanges::width(GenomicRanges::reduce(GenomicRanges::intersect(GenomicRanges::intersect(gr_nsl[gr_nsl$pop == 'POL_CIM'], gr_nsl[gr_nsl$pop == "POL_NZM"]), GenomicRanges::intersect(gr_nsl[gr_nsl$pop == 'POL_SAM'], gr_nsl[gr_nsl$pop == "POL_TON"])))))/sum(GenomicRanges::width(GenomicRanges::reduce(gr_nsl[gr_nsl$pop %in% c("POL_NZM", "POL_CIM", "POL_SAM","POL_TON")]))) * 100) %>% sprintf('%.1f',.)`%. This shows that within the Eastern or Western Polynesian groups there was a larger proportion of regions that were similar for significant \gls{nsl}, than between the Eastern and Western groups.

The regions that were in common for significant \gls{nsl} between all of the \gls{pol} populations had 27 genes that intersected. These genes were part of 10 pathways, but none were significantly enriched. Regions that were in common between \gls{nzm} and \gls{cim} had 230 genes that intersected, and were involved with 160 pathways. A single pathway, "Glycosaminoglycan degradation_Homo sapiens_hsa00531", was significantly enriched (adjusted P 9.223x10-3) with four of 19 genes (_HPSE2_, _HYAL1_, _HYAL2_, and _HYAL3_). The regions that were in common between \gls{sam} and \gls{ton} had 319 genes that intersected them. There were 197 pathways that these genes were part of but none were significantly enriched.


```{r, eval = FALSE}
#pol
GenomicRanges::intersect(GenomicRanges::intersect(gr_nsl[gr_nsl$pop == "POL_NZM"], gr_nsl[gr_nsl$pop == "POL_CIM"], ignore.strand = TRUE ),
                         GenomicRanges::intersect(gr_nsl[gr_nsl$pop == "POL_SAM"], gr_nsl[gr_nsl$pop == "POL_TON"], ignore.strand = TRUE )) %>% data.frame() %>% mutate(seqnames = paste0('chr',seqnames)) %>% GenomicRanges::GRanges() %>% txdb_gene_annotate() %>% filter(!is.na(SYMBOL)) %>% pull(SYMBOL) %>% unique() %>% cat(sep = '\n')

# nzm/cim
GenomicRanges::intersect(gr_nsl[gr_nsl$pop == "POL_NZM"], gr_nsl[gr_nsl$pop == "POL_CIM"], ignore.strand = TRUE ) %>% data.frame() %>% mutate(seqnames = paste0('chr',seqnames)) %>% GenomicRanges::GRanges() %>% txdb_gene_annotate()%>% filter(!is.na(SYMBOL)) %>% pull(SYMBOL) %>% unique() %>% cat(sep= '\n')

# sam/ton
GenomicRanges::intersect(gr_nsl[gr_nsl$pop == "POL_SAM"], gr_nsl[gr_nsl$pop == "POL_TON"], ignore.strand = TRUE ) %>% data.frame() %>% mutate(seqnames = paste0('chr',seqnames)) %>% GenomicRanges::GRanges() %>% txdb_gene_annotate()%>% filter(!is.na(SYMBOL)) %>% pull(SYMBOL) %>% unique() %>% cat(sep = '\n')
```

```{r nslclust, fig.cap='(ref:nslclust)', echo = FALSE, fig.asp= 1 }
nsl_dendro_plot()
#ggsave(last_plot(), file = 'images/04_clustering/nsl_dendros.png')
```

(ref:nslclust) _\gls{nsl} Clustering_ A. Dendrogram from hierarchical clustering based on proportion of significant \gls{nsl} regions shared between populations, coloured by super population. B. Heatmap showing the proportion of significant \gls{nsl} regions shared. The population listed on the axis is used as A in the calculation = (A $\cap$ B)/ A. Axes have been ordered according to the hierarchical clustering.


```{r, eval = FALSE}
haplo_markerLoc <-function(statGR){
IRanges::subsetByOverlaps(marker_loc %>% mutate(start = pos, end = pos+1)%>% GenomicRanges::GRanges(),
                          statGR %>% data.frame() %>% select(1:3) %>% GenomicRanges::GRanges()
                          ) %>% data.frame() %>% group_by(eff) %>% tally() %>% mutate_if(is.numeric, funs(replace_na(., replace = 0))) %>% select(Effect = eff, n)
}

haplo_markerLoc(gr_ihs[gr_ihs$pop == "POL_NZM"]) %>% full_join(., haplo_markerLoc(gr_ihs[gr_ihs$pop == "POL_CIM"]),by = 'Effect', suffix =  c("_nzm","_cim")) %>% full_join(., haplo_markerLoc(gr_ihs[gr_ihs$pop == "POL_SAM"]),by = 'Effect') %>% full_join(., haplo_markerLoc(gr_ihs[gr_ihs$pop == "POL_TON"]),by = 'Effect', suffix = c("_sam","_ton")) %>% mutate_if(is.numeric, funs(replace_na(., replace = 0))) %>% kable( caption = '(\\#tab:iHSperMarkerLoc) Number of markers by variant annotation type that were in the significant \\gls{ihs} clustered regions for Polynesian populations.', col.names = c("Effect", "NZM","CIM","SAM","TON"), format = 'latex', booktabs = TRUE) %>% kable_styling(latex_options = "scale_down")
```


```{r, eval = FALSE}
haplo_markerLoc(gr_nsl[gr_nsl$pop == "POL_NZM"]) %>% full_join(., haplo_markerLoc(gr_nsl[gr_nsl$pop == "POL_CIM"]),by = 'Effect', suffix = c("_nzm","_cim")) %>% full_join(., haplo_markerLoc(gr_nsl[gr_nsl$pop == "POL_SAM"]),by = 'Effect') %>% full_join(., haplo_markerLoc(gr_nsl[gr_nsl$pop == "POL_TON"]),by = 'Effect', suffix = c("_sam","_ton")) %>% mutate_if(is.numeric, funs(replace_na(., replace = 0))) %>% kable( caption = '(\\#tab:nSLperMarkerLoc) Number of markers by variant annotation type that were in the significant \\gls{nsl} clustered regions for Polynesian populations.', col.names = c("Effect", "NZM","CIM","SAM","TON"), format = 'latex', booktabs = TRUE) %>% kable_styling(latex_options = "scale_down")
```

 
 
#### Metabolic disease associated genes

```{r, eval = TRUE, warning = FALSE, message = FALSE}
sig_ihs_val_GR <- sig_ihs_val %>% select(chrom, chrom_start, chrom_end, pop, superpop, genename, statvalue, statid) %>% mutate(chrom = paste0('chr',chrom)) %>% GenomicRanges::GRanges()
sig_nsl_val_GR <- sig_nsl_val %>% select(chrom, chrom_start, chrom_end, pop, superpop, genename, statvalue, statid) %>% mutate(chrom = paste0('chr',chrom)) %>% GenomicRanges::GRanges()

GenomeInfoDb::seqlevels(gr_ihs) <- paste0('chr', GenomeInfoDb::seqlevels(gr_ihs))
GenomeInfoDb::seqlevels(gr_nsl) <- paste0('chr',GenomeInfoDb::seqlevels(gr_nsl))
gr <- IRanges::subsetByOverlaps(gr_ihs, c(gc_urate_gout_GR, obesity_GR, t2d_GR, kd_GR, metsyn_GR))
gr2 <- IRanges::subsetByOverlaps(gr_nsl, c(gc_urate_gout_GR, obesity_GR, t2d_GR, kd_GR, metsyn_GR))


subset_snps <- function(snp_listGR, regionsGR, diseaseGR){
  suppressWarnings(suppressMessages( purrr::map_df(panel %>% unite(pop, c("super_pop", "pop")) %>% pull(pop) %>% unique(), ~ IRanges::subsetByOverlaps(snp_listGR[snp_listGR$pop == stringr::str_sub(.x,5, 7)], GenomicRanges::reduce(IRanges::subsetByOverlaps(regionsGR[regionsGR$pop== .x], diseaseGR), ignore.strand = TRUE)) %>% as_tibble() %>% mutate(seqnames = as.character(seqnames)))))
}

disease_ihs_snps <- list()
disease_ihs_snps$urate <- subset_snps(sig_ihs_val_GR, gr, gc_urate_gout_GR)
disease_ihs_snps$obesity <- subset_snps(sig_ihs_val_GR, gr, obesity_GR)
disease_ihs_snps$t2d <- subset_snps(sig_ihs_val_GR, gr, t2d_GR)
disease_ihs_snps$kd <- subset_snps(sig_ihs_val_GR, gr, kd_GR)
disease_ihs_snps$metsyn <- subset_snps(sig_ihs_val_GR, gr, metsyn_GR)

disease_nsl_snps <- list()
disease_nsl_snps$urate <- subset_snps(sig_nsl_val_GR, gr2, gc_urate_gout_GR)
disease_nsl_snps$obesity <- subset_snps(sig_nsl_val_GR, gr2, obesity_GR)
disease_nsl_snps$t2d <- subset_snps(sig_nsl_val_GR, gr2, t2d_GR)
disease_nsl_snps$kd <- subset_snps(sig_nsl_val_GR, gr2, kd_GR)
disease_nsl_snps$metsyn <- subset_snps(sig_nsl_val_GR, gr2, metsyn_GR)



# number of significant regions in each super populaton by gene for both ihs and nsl
# in the metabolic genes
ihs_nsl <- left_join(gr %>% as.tibble(),txdb_gene_annotate(gr) %>% as.tibble()) %>%filter( SYMBOL %in% (filter(., pop %in% paste0("POL_",poly_pop)) %>% pull(SYMBOL)), SYMBOL %in% genelists$genename) %>% mutate(super = stringr::str_sub(pop, 1,3)) %>% select(SYMBOL, pop, super) %>% distinct() %>% group_by(SYMBOL, super) %>% tally() %>% full_join(.,                                                                                             left_join(gr2 %>% as.tibble(),txdb_gene_annotate(gr2) %>% as.tibble()) %>%filter( SYMBOL %in% (filter(., pop %in% paste0("POL_",poly_pop)) %>% pull(SYMBOL)), SYMBOL %in% genelists$genename) %>%  mutate(super = stringr::str_sub(pop, 1,3)) %>% select(SYMBOL, pop, super) %>% distinct() %>% group_by(SYMBOL, super) %>% tally(), by = c("SYMBOL","super"), suffix = c("_ihs","_nsl")
)

# ihs snps snps in the genes that were in the significant regions of metabolic disease genes
a_gr <- GenomicRanges::GRanges(sig_ihs_val %>% filter(genename %in%(ihs_nsl %>% filter(!is.na(n_ihs) & !is.na(n_nsl)) %>% pull(SYMBOL)) ) %>% select(chrom, chrom_start, chrom_end, pop, superpop, genename, statvalue) %>% mutate(statname = 'ihs'))

# nsl snps in the genes that were in the significant regions of metabolic disease genes
b_gr <- GenomicRanges::GRanges(sig_nsl_val %>% filter(genename %in%(ihs_nsl %>% filter(!is.na(n_ihs) & !is.na(n_nsl)) %>% pull(SYMBOL)) ) %>% select(chrom, chrom_start, chrom_end, pop, superpop, genename, statvalue) %>% mutate(statname = 'ihs'))

# n regions
n_regions_ihs_nsl <- NROW(GenomicRanges::reduce(IRanges::subsetByOverlaps(c(gr,gr2), c(gc_urate_gout_GR, obesity_GR, t2d_GR, kd_GR, metsyn_GR)), ignore.strand = TRUE))

n_regions_pol_ihs_nsl <- NROW(GenomicRanges::reduce(IRanges::subsetByOverlaps(c(gr[grep(x = gr$pop, pattern = 'POL')],gr2[grep(x = gr2$pop, pattern = 'POL')]), c(gc_urate_gout_GR, obesity_GR, t2d_GR, kd_GR, metsyn_GR)), ignore.strand = TRUE))

n_regions_ihs <-NROW(GenomicRanges::reduce(IRanges::subsetByOverlaps(gr, c(gc_urate_gout_GR, obesity_GR, t2d_GR, kd_GR, metsyn_GR)), ignore.strand = TRUE))
n_regions_nsl <- NROW(GenomicRanges::reduce(IRanges::subsetByOverlaps(gr2, c(gc_urate_gout_GR, obesity_GR, t2d_GR, kd_GR, metsyn_GR)), ignore.strand = TRUE))

n_regions_ihs_nsl_intersect <- NROW(GenomicRanges::intersect(GenomicRanges::reduce(IRanges::subsetByOverlaps(gr, c(gc_urate_gout_GR, obesity_GR, t2d_GR, kd_GR, metsyn_GR)), ignore.strand = TRUE), GenomicRanges::reduce(IRanges::subsetByOverlaps(gr2, c(gc_urate_gout_GR, obesity_GR, t2d_GR, kd_GR, metsyn_GR)), ignore.strand = TRUE)))

# n snps
n_pol_snps_ihs <- bind_rows(disease_ihs_snps) %>% filter(superpop == 'POL') %>% select(seqnames, start, end) %>% distinct() %>% NROW()

n_pol_snps_nsl <- bind_rows(disease_nsl_snps) %>% filter(superpop == 'POL') %>% select(seqnames, start, end) %>% distinct() %>% NROW()

n_pol_snps_ihs_nsl <- inner_join( bind_rows(disease_ihs_snps) %>% filter(superpop == 'POL') %>% select(seqnames, start, end) %>% distinct(),
              bind_rows(disease_nsl_snps) %>% filter(superpop == 'POL') %>% select(seqnames, start, end) %>% distinct()
             ) %>% NROW() 

# n disease regions
n_disease_pol_urate <- NROW(GenomicRanges::reduce(IRanges::subsetByOverlaps(c(gr[grep(x = gr$pop,  pattern = 'POL')],gr2[grep(x = gr2$pop,  pattern = 'POL')]), gc_urate_gout_GR)))
n_disease_pol_obesity <- NROW(GenomicRanges::reduce(IRanges::subsetByOverlaps(c(gr[grep(x = gr$pop,  pattern = 'POL')],gr2[grep(x = gr2$pop, pattern = 'POL')]), obesity_GR)))
n_disease_pol_t2d <- NROW(GenomicRanges::reduce(IRanges::subsetByOverlaps(c(gr[grep(x = gr$pop, pattern = 'POL')],gr2[grep(x = gr2$pop, pattern = 'POL')]), t2d_GR)))
n_disease_pol_kd <- NROW(GenomicRanges::reduce(IRanges::subsetByOverlaps(c(gr[grep(x = gr$pop, pattern = 'POL')],gr2[grep(x = gr2$pop, pattern = 'POL')]), kd_GR)))
n_disease_pol_metsyn <- NROW(GenomicRanges::reduce(IRanges::subsetByOverlaps(c(gr[grep(x = gr$pop, pattern = 'POL')],gr2[grep(x = gr2$pop, pattern = 'POL')]), metsyn_GR)))
```

In the clustered-significant regions of both \gls{ihs} and \gls{nsl} there were `r n_regions_ihs_nsl` independent regions that overlapped with genes associated with urate and metabolic diseases from all populations, for Polynesian populations the number of regions was `r n_regions_pol_ihs_nsl`. XXX \gls{ihs} had `r n_regions_ihs` regions, while \gls{nsl} had `r n_regions_nsl`, and in common between the two statistics were `r n_regions_ihs_nsl_intersect` regions. The regions of the Polynesian populations covered `r n_pol_snps_ihs` \glspl{snp} for \gls{ihs} and `r n_pol_snps_nsl` for \gls{nsl}, with `r n_pol_snps_ihs_nsl ` in common between them. For the genes that had significant regions in Polynesian populations, `r n_disease_pol_urate` regions intersected genes that had been associated with urate and gout, `r n_disease_pol_obesity` with obesity, `r n_disease_pol_t2d` with \gls{t2d}, `r n_disease_pol_kd` with kidney disease, and `r n_disease_pol_metsyn` with metabolic syndrome.

```{r, message=FALSE, warning=FALSE}
shared_ihs_pol_disease_snps <- left_join(bind_rows(disease_ihs_snps) %>% filter(superpop == "POL") %>% select(seqnames, start, end) %>% distinct(), bind_rows(disease_ihs_snps), by =c("seqnames", "start","end")) %>% select(-genename, -width, -strand) %>% distinct() %>% group_by(seqnames, start, end) %>% mutate(n = n()) %>% filter(n > 1) %>% arrange(desc(n))
shared_ihs_pol_disease_snps_genes <- left_join(shared_ihs_pol_disease_snps, txdb_gene_annotate(GenomicRanges::GRanges(shared_ihs_pol_disease_snps)), by = c('seqnames', 'start','end'))


shared_nsl_pol_disease_snps <- left_join(bind_rows(disease_nsl_snps) %>% filter(superpop == "POL") %>% select(seqnames, start, end) %>% distinct(), bind_rows(disease_nsl_snps), by =c("seqnames", "start","end")) %>% select(-genename, -width, -strand) %>% distinct() %>% group_by(seqnames, start, end) %>% mutate(n = n()) %>% filter(n > 1) %>% arrange(desc(n))

shared_nsl_pol_disease_snps_genes <- left_join(shared_nsl_pol_disease_snps, txdb_gene_annotate(GenomicRanges::GRanges(shared_nsl_pol_disease_snps)), by = c('seqnames', 'start','end'))

# combined stats

# look at intersection
shared_ihs_nsl_pol_disease_snps <-full_join(shared_ihs_pol_disease_snps, shared_nsl_pol_disease_snps, by = c("seqnames", "start", "end", "pop", "superpop"), suffix = c("_ihs", "_nsl"))
shared_ihs_nsl_pol_disease_snps_intersect <- shared_ihs_nsl_pol_disease_snps %>% inner_join(., shared_ihs_nsl_pol_disease_snps %>% filter(superpop == 'POL') %>% filter(!is.na(statid_ihs) & !is.na(statid_nsl)) %>% select(seqnames, start, end))

#shared_ihs_nsl_pol_disease_snps_intersect %>% left_join(., txdb_gene_annotate(GRanges(.))) %>% mutate(max_n = max(n_ihs, n_nsl, na.rm = TRUE)) %>% arrange(desc(max_n)) %>% left_join(., markers %>% mutate(chrom = paste0('chr',chrom)), by = c('seqnames' = 'chrom', 'start' = 'chrom_start') ) %>% distinct() %>% arrange(SYMBOL, marker, superpop, pop) %>% View()

#shared_ihs_nsl_pol_disease_snps_intersect %>% left_join(., txdb_gene_annotate(GRanges(.))) %>% mutate(max_n = max(n_ihs, n_nsl, na.rm = TRUE)) %>% arrange(desc(max_n)) %>% left_join(., markers %>% mutate(chrom = paste0('chr',chrom)), by = c('seqnames' = 'chrom', 'start' = 'chrom_start') ) %>% select(seqnames, start, end, pop, superpop, statvalue_ihs,n_ihs, statvalue_nsl, n_nsl, max_n, marker, SYMBOL)%>% distinct() %>% group_by(marker, SYMBOL,superpop) %>% tally() %>% spread(superpop, n) %>% arrange(SYMBOL, marker, superpop, pop) #%>% View()


# combine the snps from ihs and nsl and annotate with genes
#full_join(shared_ihs_pol_disease_snps, shared_nsl_pol_disease_snps, by = c("seqnames", "start", "end", "pop", "superpop"), suffix = c("_ihs", "_nsl")) %>% left_join(., txdb_gene_annotate(GRanges(.)))

```


Using the significant \glspl{snp} for these \gls{ihs} and \gls{nsl} (as in Chapter 3) there were a total of `r shared_ihs_pol_disease_snps %>% select(seqnames, start, end) %>% distinct() %>% NROW()` \glspl{snp} that were significant in a Polynesian population and also significant in at least one other population for \gls{ihs}, and `r shared_nsl_pol_disease_snps %>% select(seqnames, start, end) %>% distinct() %>% NROW()` for \gls{nsl}. The number of \glspl{snp} that were significant in both \gls{ihs} and \gls{nsl} in the Polynesian population was `r shared_ihs_nsl_pol_disease_snps_intersect %>% select(seqnames, start, end) %>% distinct() %>% NROW()`.

The urate associated gene _ABCG2_ had a single \gls{snp} (rs2622626) that was significant in \gls{ton} that was in common for significance for \gls{ihs} in two other populations, \gls{gih} and \gls{itu}, both were from the \gls{sas} super population. The situation was the same for \gls{nsl}, but with the addition of \gls{jpt}.

The \gls{sam} and \gls{ton} populations had multiple \glspl{snp} in common that were significant for both \gls{ihs} and \gls{nsl}. An example is in _LRP1B_, where there were three \glspl{snp}: rs10173806, rs4591293, and rs2890615. Rs2890615 was also significant in \gls{esn} and \gls{chs}.



_ERBB4_ had a single significant \gls{snp}, rs6707285, in \gls{sam}, and was also significant in \gls{pur}, \gls{khv}, \gls{ibs}, \gls{tsi}, and \gls{gbr}. _FHIT_ also had a single significant \gls{snp}, but this was in the \gls{nzm}, \gls{chb}, \gls{chs}, \gls{cdx}, \gls{khv}, and \gls{pel} populations. Of the metabolic disease genes, _ERBB4_ and _FHIT_ were also found by @pickrell2009signals.

<!-- pickrell ERBB4 in eas, sas, and mideas, oceania. FHIT in bantu -->


One of the main regions, chr20:33470694-34556005, had 11 significant \glspl{snp} in Polynesian populations as well as up to 16 other populations, contained the obesity-associated gene _GDF5_ and the neighbouring genes of _CEP250_ and _UQCC1._ The majority of the significant \glspl{snp} were for the ancestral allele. Both rs1570841 and rs4911502 were significant in the populations of \gls{amr} (\gls{clm}, \gls{mxl} and \gls{pur}), \gls{eas} (\gls{cdx}, \gls{chb}, \gls{chs}, \gls{jpt}, and \gls{khv}), \gls{eur} (\gls{ceu}, \gls{fin}, \gls{gbr}, \gls{ibs}), \gls{pol} (\gls{cim}, \gls{nzm}, and \gls{ton}), and \gls{sas} (\gls{beb} and \gls{itu}. Both \glspl{snp} were for the ancestral allele. Four \glspl{snp}, rs4911178, rs4911494, rs6087704, and rs6087704, intersected both _GDF5_ and _UQCC1_, and were significant for the derived alleles in the \gls{cim} and \gls{jpt} populations.

<!-- CEP250 -->
<!--   - rs1570841 3 AMR, 5 EAS, 4 EUR, 3 POL, 2 SAS -ve -->
<!--   - rs2236164 2 AMR, 5 EAS, 1 EUR, 2 POL +ve -->
<!--   - rs224360 2 EAS, 2 POL +ve -->
<!--   - rs4911502 3 AMR, 5 EAS, 4 EUR, 3 POL, 2 SAS -ve -->
<!--   - rs4911507 3 EAS, 1 EUR, 1 POL -ve -->
<!-- GDF5 -->
<!--   - rs4911178 1 EAS, 1 POL +ve cim/jpt -->
<!--   - rs4911494 1 EAS, 1 POL +ve cim/jpt -->
<!--   - rs6060373 3 AMR, 5 EAS, 4 EUR, 3 POL -ve -->
<!--   - rs6087704 1 EAS, 1 POL +ve cim/jpt -->
<!--   - rs6087705 2 AMR, 5 EAS, 3 EUR, 3 POL, 2 SAS -ve -->
<!--   - rs6088813 1 EAS, 1 POL +ve cim/jpt -->

<!-- UQCC1 (all overlap with GDF5) -->
<!--   - rs4911178 1 EAS, 1 POL +ve -->
<!--   - rs4911494 1 EAS, 1 POL +ve -->
<!--   - rs6060373 3 AMR, 5 EAS, 4 EUR, 3 POL -ve -->
<!--   - rs6088813 1 EAS, 1 POL +ve -->


<!-- ERBB4 -->
<!--   - rs6707285 1 AMR, 1 EAS, 3 EUR, 1 POL -ve sam -->

<!-- FHIT -->
<!--   - rs6806415 1 AMR, 4 EAS, 1 POL -ve nzm -->

<!-- HLA-B -->
<!--   - rs3130696 1 AFR, 4 EAS, 1 EUR, 1 POL, 2 SAS -ve cim -->



```{r, eval = FALSE}
# ihs regions for the disease genes
left_join(gr %>% as.tibble(),txdb_gene_annotate(gr) %>% as.tibble()) %>%filter( SYMBOL %in% (filter(., pop %in% paste0("POL_",poly_pop)) %>% pull(SYMBOL)), SYMBOL %in% genelists$genename) %>%  View()


# need to load pickrell2009 from 03_selectionpipeline.R
ihs_nsl %>% filter(!is.na(n_ihs) & !is.na(n_nsl)) %>% filter(SYMBOL %in% unlist(lapply(pickrell2009, unlist)))


a <- left_join(gr %>% as.tibble(),txdb_gene_annotate(gr) %>% as.tibble()) %>%filter( SYMBOL %in% (filter(., pop %in% paste0("POL_",poly_pop)) %>% pull(SYMBOL)), SYMBOL %in% c("FHIT"))


subsetByOverlaps(a_gr,a_gr[grep(x = a_gr$superpop, 'POL')]) %>% as.tibble() %>% arrange(seqnames, start) %>% View()

subsetByOverlaps(a_gr,a_gr[grep(x = a_gr$superpop, 'POL')]) %>% as.tibble() %>% arrange(seqnames, start) %>% group_by(seqnames, start) %>% mutate(n = n()) %>% filter(n > 1) %>%ungroup() %>%  mutate(seqnames = as.integer(as.character(seqnames))) %>% left_join(., markers,by = c('seqnames' = 'chrom', 'start' = 'chrom_start')) %>% View()

# ihs
subsetByOverlaps(a_gr,a_gr[grep(x = a_gr$superpop, 'POL')]) %>% as.tibble() %>% arrange(seqnames, start) %>% group_by(seqnames, start) %>% mutate(n = n()) %>% filter(n > 1) %>%ungroup() %>%  mutate(seqnames = as.integer(as.character(seqnames))) %>% left_join(., markers,by = c('seqnames' = 'chrom', 'start' = 'chrom_start')) %>% View()

test <- subsetByOverlaps(a_gr,a_gr[grep(x = a_gr$superpop, 'POL')]) %>% as.tibble() %>% arrange(seqnames, start) %>% group_by(seqnames, start) %>% mutate(n = n()) %>% filter(n > 1) %>%ungroup() %>% mutate(seqnames = as.integer(as.character(seqnames))) %>% left_join(., markers,by = c('seqnames' = 'chrom', 'start' = 'chrom_start')) %>% head() %>% select(marker, n, genename, pop, statvalue) 


# nsl

subsetByOverlaps(b_gr,b_gr[grep(x = b_gr$superpop, 'POL')]) %>% as.tibble() %>% arrange(seqnames, start) %>% View()

subsetByOverlaps(b_gr,b_gr[grep(x = b_gr$superpop, 'POL')]) %>% as.tibble() %>% arrange(seqnames, start) %>% group_by(seqnames, start) %>% mutate(n = n()) %>% filter(n > 1) %>%ungroup() %>%  mutate(seqnames = as.integer(as.character(seqnames))) %>% left_join(., markers,by = c('seqnames' = 'chrom', 'start' = 'chrom_start')) %>% View()

```


<!-- - Tongans had a region chr4:88777743-89066715 (289 kbp), overlapped chr4:89066715-89066715 in SAS_GIH and SAS_ITU -->

<!-- - ERBB4, FHIT, and ITGB6 from pickrell2009 -->

<!-- Looking at the ihs results for all metabolic disease genes, for shared snps -->

<!-- - FPGT-TNNI3K at rs520806 -->
<!--     - TON, EAS(CDX,CHS,JPT,KHV) -->

<!-- - LRP1B numerous SAM/TON pairs -->
<!--     - rs10173806 SAM/TON -->
<!--     - rs4591293 SAM/TON -->
<!--     - rs2890615 SAM/TON/ESN/CHS -->

<!-- - ITGB6 -->
<!--     - rs10497213 TON/KHV/CDX -->

<!-- - ERBB4* -->
<!--     - rs1025753 CIM/NZM -->
<!--     - rs6707285 SAM/KHV -->

<!-- - FHIT*  -->
<!--     - rs3772497 TON/BEB/GIH/TSI -->
<!--     - rs6806415 NZM/CDX/CHB/CHS/PEL -->

<!-- - PSMD6 -->
<!--     - rs2280162 CIM/NZM -->

<!-- - ADAMTS9 -->
<!--     - rs6787633 CIM/NZM -->

<!-- -LPP -->
<!--     - rs12495383 CIM/NZM -->
<!--     - rs3856921 CIM/NZM -->
<!--     - rs6444332 CIM/NZM -->

<!-- - ABCG2 -->
<!--     - rs2622626 TON/GIH/ITU -->

<!-- - ARL15 -->
<!--     - rs17213 CIM/NZM -->
<!--     - rs26775 CIM/NZM -->

<!-- - CCNJL -->
<!--     - rs2421778 CIM/NZM -->

<!-- - SSR1 -->
<!--     - rs9505109 CIM/NZM -->
<!--     - rs2714354 CIM/CHS/GWD/NZM -->

<!-- - BTNL2 -->
<!--     - rs41521946 NZM/JPT/FIN/CHB -->
<!--     - rs28362677 NZM/CHB/FIN/JPT -->
<!--     - rs28362678 NZM/CHB/FIN/JPT -->
<!--     - rs2076530 CIM/JPT -->
<!--     - rs28362679 CIM/LWK/NZM -->
<!--     - rs2076529 CIM/JPT -->
<!--     - rs3129954 ASW/PJL/SAM/STU/TSI -->
<!--     - rs3817964 CEU/CHS/CIM/CLM/GBR/ITU/JPT/MXL/PEL/PUR/STU -->
<!--     - rs3817963 CIM/GWD/JPT -->
<!--     - rs2076522 CIM/BEB/CHB/GWD/JPT/STU -->
<!--     - rs3793127 ACB/BEB/CHB/CHS/CIM/CLM/ITU/JPT/MXL/STU -->
<!--     - rs3763308 NZM/ITU/MXL -->

<!-- - PARK2 -->
<!--     - rs9347666 SAM/TON -->

<!-- - MSRA -->
<!--     - rs6983566 CEU/CIM/GBR/IBS/JPT/PUR/TSI -->

<!-- - PTPRD -->
<!--     - rs10977210 CLM/GBR/GIH/PUR/TON -->
<!--     - rs2890854 SAM/TON -->
<!--     - rs16929395 NZM/CIM -->
<!--     - rs4628308 CIM/JPT -->
<!--     - rs10809063 CIM/SAM/TON -->
<!--     - rs587791 SAM/TON -->

<!-- - BDNF -->
<!--     - rs2030323 SAM/TON -->

<!-- - ZFAND6 -->
<!--     - rs12592217 CIM/NZM -->


<!-- - ADCY9 -->
<!--     - rs879620 CIM/NZM -->
<!--     - rs2238443 CIM/NZM -->

<!-- - GPRC5B -->
<!--     - rs183452 SAM/TON -->

<!-- - RABEP1 -->
<!--     - rs2309381 CHB/ESN/ITU/NZM/STU -->

<!-- - GDF5 -->
<!--     - rs224331 CIM/JPT -->


<!-- **nsl** -->

<!-- - FPGT-TNNI3K -->
<!--     - rs520806 CDX/CHS/JPT/KHV/TON -->

<!-- - LRP1B -->
<!--     - rs10173806 SAM/TON -->
<!--     - rs4591293 SAM/TON -->
<!--     - rs2890615 CHS/ESN/SAM/TON -->

<!-- - ERBB4 -->
<!--     - rs6707285 GBR/IBS/KHV/PUR/SAM/TSI -->

<!-- - FHIT -->
<!--     - rs6806415 CDX/CHB/CHS/KHV/NZM -->

<!-- - PSMD6 -->
<!--     - rs2280162 CIM/NZM -->

<!-- - ADAMTS9 -->
<!--     - rs6787633 CIM/NZM -->
<!--     - rs17726803 CIM/NZM -->

<!-- - LPP -->
<!--     - rs9877898 NZM/TON -->

<!-- - ABCG2 -->
<!--     - rs2622626 GIH/ITU/JPT/TON -->

<!-- - SLC39A8 -->
<!--     - rs2165265 SAM/TON -->
<!--     - rs964504 SAM/TON -->

<!-- - CCNJL -->
<!--     - rs2421778 CIM/NZM -->

<!-- - SSR1 -->
<!--     - rs9505109 CIM/NZM -->
<!--     - rs2714354 CIM/NZM -->


<!-- - FOXO3 -->
<!--     - rs9372190 NZM/SAM -->

<!-- - PARK2 -->
<!--     - rs9347666 KHV/SAM -->

<!-- - MSRA -->
<!--     - rs6983566 CEU/CIM/GBR/IBS/JPT/NZC/PUR/TSI -->

<!-- - PTPRD -->
<!--     - rs2890854 SAM/TON -->
<!--     - rs4628308 CIM/JPT -->
<!--     - rs9299106 CIM/PEL -->
<!--     - rs10809063 SAM/TON -->

<!-- - BDNF -->
<!--     - rs2030323 SAM/TON -->

<!-- - MAP2K5 -->
<!--     - rs884202 CDX/CHB/CHS/JPT/MXL/TON -->

<!-- - ZFAND6 -->
<!--     - rs12592217 CIM/NZM -->

<!-- - ADCY9 -->
<!--     - rs879620 CIM/NZM -->

<!-- - RABEP1 -->
<!--     - rs2309381 ITU/NZM/STU -->

<!-- - GDF5 -->
<!--     - rs143384 CIM/JPT -->
<!--     - rs143383 CIM/JPT -->


 <!-- \FloatBarrier -->

### Disease-associated gene clustering {#genelistcluster}
<!-- might be a good idea to use a grid to put multiple plots into one -->
```{r load_gene_clus, include = FALSE}
# brings in objects called {gc_urate_gout,urate_gout,kd,metsyn,obesity,t2d}_GR, 
load('~/data/NZ_coreExome_1kgp/100kbWindow_intra/gene_clustering_filtered_3ns-16-4-2018.RData')# brings in objects called gout, obesity, t2d, metsyn
names(gout) <- c('mat_d','prop')
names(t2d) <-  c('mat_d','prop')
names(obesity) <- c('mat_d','prop')
names(metsyn) <- c('mat_d','prop')
names(kd) <- c('mat_d','prop')


```

To assess if there was specificity of selection for the metabolic disease gene lists or if the population groupings could be obtained from any list of genes, clustering was performed using the actual statistic values for the genes. These values were median-centred to account for the population specific shifting of distributions which would have accounted for the majority of difference in the distance calculation. Gene lists were chosen for metabolic-related disease and compared to lists of random genes (Section \@ref(randomgenes)) to assess clustering specificity to metabolic disease.

Overall, gene lists derived from \gls{gwas} associations clustered the individual populations into their super population groupings. Windows that intersected genes that had been associated with a disease were centred by population median and then grouped using hierarchical clustering. There were varying levels of super population proportions and exclusivity (the proportion of super populations in a single cluster) of clusters (Table \@ref(tab:proportion) and \@ref(tab:exclusivity)). The Polynesian populations clustered together when using each of \gls{fwh}, \gls{td}, and \gls{ze}, and in the majority of clustering (17/20) they were exclusive. This pattern was also similar with the \gls{eas} populations (17/20). For the clusters that contained the entirety of a super population, half the groupings for the entire Polynesian super population were exclusive. Three quarters of the groupings for each of the AFR and EAS super populations were exclusive and grouped the super population in their entirety. This suggests that singletons or rare variants are insufficient in these genes to be able to differentiate between populations compared to the statistics that use the more common variants. 

Comparing the gene lists by statistic, for all selection statistics and gene lists the populations of the \gls{sas} super population were all clustered together, although often with poor exclusivity. The \gls{eur} super population was clustered entirely together but with a ranging degree of exclusivity across the different selection test statistics. \Gls{td}, \gls{fwh}, and \gls{ze} clustered all the super populations in their entirety although the \gls{amr} populations were split. The clustering of super populations was not specific to the gene lists associated with urate, gout, obesity, \gls{t2d}, kidney disease, or, metabolic syndrome, as the use of random lists of genes would cluster the super populations (excluding \gls{amr}) up to 75% of the time (Table \@ref(tab:randgenes)). This indicates that there is likely nothing specific about the urate and gout, \gls{t2d}, obesity, kidney disease, or metabolic syndrome gene lists in terms of "signatures of selection", but they contain similar discriminatory information as other lists of genes, which is likely to be capturing allele frequency differences between populations. This is seen with the results of the clustering using random genes (Section \@ref(randomgenes)), where the number of genes in a gene list has appears to impact on exclusivity and proportion of populations clustered into their super populations.


<!--
- Tajima's D clustered all super populations together (AMR was slightly ungrouped, often PEL out) and get the EP/WP split for all diseases

- Fay and Wu's H clustered all super populations together (AMR was slightly ungrouped, often PEL out) and get the EP/WP split for all diseases

- Fu and Li's D did not have groups that resembled the super populations and the exclusivity of the groups that did have a high proportion of populations from the same super population were not exclusive.

- Fu and Li's F clustered both EUR and SAS for the gene lists.

- Except for AMR, Zeng's E clustered all populations into their super populations. 
-->

<!--
Clustering on other diseases such as pd, macular degeneration, alzhimers and colon cancer revealed similar patters 
-->
#### Urate and gout

There were `r length(unique(urate_goutGR$SYMBOL))` genes that had an association with urate or gout at a genome-wide significance level. This mapped to `r nrow(gout$mat_d$Tajima.D)` windows for each of the frequency based selection test statistics. \gls{ze} and \gls{fwh} clustered each super population into their own group and each cluster was exclusive to a super population. Both \gls{td} and \gls{flf} grouped most of the super populations correctly and nearly all clusters had good exclusivity for each single super population. \Gls{td}, \gls{fwh}, and \gls{ze} also had an Eastern/Western Polynesian population sub group within the Polynesian group (Figure \@ref(fig:goutdendros)). Unlike the F~ST~ clustering, the \gls{amr} populations were grouped together for \gls{td}, \gls{fwh}, and \gls{ze}. The \gls{eas} and \gls{pol} populations were closest for \gls{flf}, \gls{fwh}, and \gls{ze}, but not for \gls{td}, where the \gls{pol} populations were furthest from the other populations.

There were 2 genes; _BCAS3_ and _RREB1_, that had at least one statistic in all of the Polynesian populations that met the threshold of significance in the lower tail. _BCAS3_ met the significance threshold for \gls{fwh} and \gls{flf} in all Polynesian populations and for \gls{td} in all Polynesian populations except \gls{ton}. _INHBC_ and _ATXN2_ both had windows that met the lower tail significance threshold for \gls{ze} in all Polynesians populations, except \gls{sam} for _INHBC_. There were `r lower_sig_stats %>% filter(SYMBOL %in% unique(urate_goutGR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(SYMBOL) %>% distinct() %>% NROW() %>% num_to_word()` of the urate- and gout-associated genes that had evidence from at least one selection statistic of possible selection in any of the Polynesian populations from the lower tail of the distributions. In the upper tail, _VEGFA_ had windows of \gls{ze} values that met the upper tail significance threshold for all Polynesians populations. _BAZ1B_ also met the threshold in the Eastern Polynesian populations. There were no loci that were significant in all Polynesian populations in the upper tail. None of the genes that met the significance threshold are urate transporters which have the strongest genetic effects on urate and gout [@Kottgen2013].


```{markdown, eval = FALSE}
any stat with all populations
 - lower `r lower_sig_stats %>% filter(SYMBOL %in% unique(urate_goutGR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(pop, SYMBOL) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==4) %>% NROW()`
 - upper `r upper_sig_stats %>% filter(SYMBOL %in% unique(urate_goutGR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(pop, SYMBOL) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==4) %>% NROW()`

has all stats
 - lower `r lower_sig_stats %>% filter(SYMBOL %in% unique(urate_goutGR$SYMBOL)) %>% group_by(pop, SYMBOL, statname) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(-n, -statname) %>% group_by(pop, SYMBOL) %>% tally() %>% filter(n == 4) %>% NROW()`
 - upper `r upper_sig_stats %>% filter(SYMBOL %in% unique(urate_goutGR$SYMBOL)) %>% group_by(pop, SYMBOL, statname) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(-n, -statname) %>% group_by(pop, SYMBOL) %>% tally() %>% filter(n == 4) %>% NROW()`



<!--
#gives list of genes that were sig in all pops (POL) but from any stats
`r lower_sig_stats %>% filter(SYMBOL %in% unique(urate_goutGR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(pop, SYMBOL) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==4) %>% .[['SYMBOL']]` 

#gives the number of stats for a gene for a poplation (POL) that were sig
lower_sig_stats %>% left_join(., panel %>% select(pop, super_pop) %>% distinct(), by = 'pop')%>% filter(SYMBOL %in% unique(urate_goutGR$SYMBOL)) %>% group_by(super_pop,pop, statname, SYMBOL) %>%tally() %>% ungroup() %>% select(-n)%>% spread(statname, statname) %>% arrange(SYMBOL, super_pop, pop)%>% filter(pop %in% c("CIM", "NZM","SAM","TON")) %>% mutate(s = rowSums(cbind(!is.na(Fay.Wu.H), !is.na(Fu.Li.D), !is.na(Fu.Li.F), !is.na(Tajima.D), !is.na(Zeng.E)))) %>% View()


upper_sig_stats %>% left_join(., panel %>% select(pop, super_pop) %>% distinct(), by = 'pop')%>% filter(SYMBOL %in% unique(urate_goutGR$SYMBOL)) %>% group_by(super_pop,pop, statname, SYMBOL) %>%tally() %>% ungroup() %>% select(-n)%>% spread(statname, statname) %>% arrange(SYMBOL, super_pop, pop)%>% View()


-->

<!--
- Td gave the super populations, good exclusivity of clusters,ep/wp split
- fwh gave perfect super populations and perfect exclusivity. ep/wp split
- fld not great, sam most disimilar, nzm-gbr, ton-cim-chb-jpt
- flf got most of the super populations. not bad for exclusivity, cim-nzm-sam + ton-chb-jpt as a big cluster. clm/pur in the middle of the eur cluster - spanish sub cluster?
- ze perfect super populations, perfect exclusivity

dendrograms:
all except fld manage to cluster the super populations with the exception of AMR although ze does. fld is somewhat random. except fld all give east/west split
td and flf hace pol as outlier

exclusivity:
zeng e - perfect
td and fay wu's h do a good job but because of the AMR split end up with 7 actual clusters
fld - poor because there are lots of smaller clusters
flf - not great

props:
ze perfect
td and fwh only get amr not complete
flf has prop of >.75, except eur
fld poly .25, rest looking at majority of super pop in same cluster

-->
```


```{r, eval = FALSE}
fst_urate_nzm <- subsetByOverlaps(pol_fstGR, urate_goutGR) %>% as.data.frame() %>% select(contains('NZM')) %>% as.matrix()

cluster_window_stat <- function(mat_d){
  #row.names(mat_d) <- d[,1] # row names are now posid column
  hr_d <- hclust(dist(mat_d,  method="euclidean"), method = "complete")
  hc_d <- hclust(dist(t(mat_d), method="euclidean"), method = "complete")
  return(list(mat_d = mat_d,hr_d = hr_d, hc_d = hc_d)) 
}

a <- cluster_window_stat(fst_urate_nzm)
plot(a$hc_d)
cutree(a$hc_d, 6)

IRanges::subsetByOverlaps((pol_fst_99 %>% mutate(chrom = paste0('chr',chrom)) %>% GenomicRanges::GRanges()), urate_goutGR) %>% data.frame() %>% left_join(., pol_fst_99genes) %>% group_by(pop, SYMBOL) %>% tally()%>% spread(SYMBOL, n, "")
```


(ref:goutdendros) _Urate and Gout Associated Gene Clustering_ Dendrograms created from hierarchical clustering applied to windows from the `r length(unique(urate_goutGR$SYMBOL))` loci associated with urate and gout. Coloured by super population.

```{r goutdendros, fig.cap='(ref:goutdendros)', fig.height=7, message=FALSE}
multiplot(gene_dendros(gene_data = gout$mat_d$Tajima.D[,-1:-3]) + ggtitle("Tajima's D"),
          gene_dendros(gene_data = gout$mat_d$Fay.Wu.H[,-1:-3]) + ggtitle("Fay and Wu's H"),
          #gene_dendros(gene_data = gout$mat_d$Fu.Li.D[,-1:-3]) + ggtitle("Fu and Li's D"),
          gene_dendros(gene_data = gout$mat_d$Fu.Li.F[,-1:-3]) + ggtitle("Fu and Li's F"),
          gene_dendros(gene_data = gout$mat_d$Zeng.E[,-1:-3]) + ggtitle("Zeng's E"),
  cols = 2)
# ggsave(multiplot(gene_dendros(gene_data = gout$mat_d$Tajima.D[,-1:-3]) + ggtitle("Tajima's D"),
#           gene_dendros(gene_data = gout$mat_d$Fay.Wu.H[,-1:-3]) + ggtitle("Fay and Wu's H"),
#           gene_dendros(gene_data = gout$mat_d$Fu.Li.D[,-1:-3]) + ggtitle("Fu and Li's D"),
#           gene_dendros(gene_data = gout$mat_d$Fu.Li.F[,-1:-3]) + ggtitle("Fu and Li's F"),
#           gene_dendros(gene_data = gout$mat_d$Zeng.E[,-1:-3]) + ggtitle("Zeng's E"),
#   cols = 2), file = 'images/04_clustering/urate_dendros.png')
```


#### Obesity


The \gls{gwas} catalog had `r length(unique(obesity_GR$SYMBOL))` genes associated with obesity, this was represented by `r nrow(obesity$mat_d$Tajima.D)` windows and the clustering resulted in the groupings of super populations (Figure \@ref(fig:obesitydendros)). Exclusivity of a single super population to a cluster was also high with the notable exception being the \gls{amr} populations of \gls{clm} and \gls{pur} often being part of the \gls{eur} cluster. The Polynesian super population cluster was also closest to \gls{eas} in all selection statistics and also had an Eastern/Western sup-grouping within the \gls{pol} group.

There were `r lower_sig_stats %>% filter(SYMBOL %in% unique(obesity_GR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(SYMBOL) %>% distinct() %>% NROW()` of the obesity-associated genes that had evidence from at least one selection statistic of possible selection in any of the Polynesian populations from the lower tail of the distributions. In the lower tail there were `r lower_sig_stats %>% filter(SYMBOL %in% unique(obesity_GR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(pop, SYMBOL) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==4) %>% NROW()` loci with at least one window that was significant in the Polynesian populations. _ADAMTSL3_, _AGBL4_, _ARL15_, _CCR3_, _CREB1_, _ERBB4_, _FER_, _FHIT_, _GRID1_, _LTBP1_, _MAP2K5_, and _ZBTB38_ each had in all Polynesian populations, at least one statistic that met the significance threshold in the lower tail. Only _DNAH10_ had a significant window for each of \gls{td}, \gls{fwh}, \gls{flf}, and \gls{ze}, and this was only in the \gls{nzm} population. The upper tail had a total of `r upper_sig_stats %>% filter(SYMBOL %in% unique(obesity_GR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(pop, SYMBOL) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% NROW()` loci that had at least a window that was significant in the Polynesian populations. _ADCY3_, _CHST8_, _CMIP_, _ERBB4_, _FHIT_, _FTO_, _GRID1_, _HMGA2_, _LTBP1_, _NEGR1_, _NRXN3_, _NT5C2_, _PARK2_, _PRKCH_, _RARB_, and  _TCF7L2_ each had in all Polynesian populations, at least one statistic that met the significance threshold in the upper tail.

> _How could this be summarised?_

```{markdown, eval = FALSE}
any stat
 - lower `r lower_sig_stats %>% filter(SYMBOL %in% unique(obesity_GR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(pop, SYMBOL) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==4) %>% NROW()`
 - upper `r upper_sig_stats %>% filter(SYMBOL %in% unique(obesity_GR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(pop, SYMBOL) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==4) %>% NROW()`

has all stats
 - lower `r lower_sig_stats %>% filter(SYMBOL %in% unique(obesity_GR$SYMBOL)) %>% group_by(pop, SYMBOL, statname) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON'))  %>% ungroup()%>% select(-n, -statname) %>% group_by(pop, SYMBOL) %>% tally() %>% filter(n == 4) %>% NROW()`
 - upper `r upper_sig_stats %>% filter(SYMBOL %in% unique(obesity_GR$SYMBOL)) %>% group_by(pop, SYMBOL, statname) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(-n, -statname) %>% group_by(pop, SYMBOL) %>% tally() %>% filter(n == 4) %>% NROW()`

<!--
- td excellent groupings, mxl/pel on their own and clm/pur with eur. exclusivity was also pretty much perfect. ep/wp split
- fwh near perfect, get 3/4 amr being clustered with the eur group. ep/wp split
- fld nzm-mxl. eas, afr grouped. eur, amr and sas inter mixed somewhat. cim-sam-ton
- flf nice super populations, eas-pol, eur-2amr, mxl/pel
- ze good super populations, amr-eur and nearly sas, pol-eas with ep/wp split. pretty good exclusivity.

dendrograms
all except fld had nice clusters although amr was split
flf has pol as outlier

exclusivity
td and fwh - only eur/amr shared
flf afr/pol/eas were exclusive
fld pol was in 4 different clusters
flf and fld had smaller groupings so more sharing

prop
td, fwh and ze all except amr were 1.0
fld did ok except for Poly
flf did ok
-->
```
  
```{r obesitydendros, fig.cap='(ref:obesitydendros)', fig.height=7, message = FALSE}
multiplot(gene_dendros(gene_data = obesity$mat_d$Tajima.D[,-1:-3]) + ggtitle("Tajima's D"),
          gene_dendros(gene_data = obesity$mat_d$Fay.Wu.H[,-1:-3]) + ggtitle("Fay and Wu's H"),
          #gene_dendros(gene_data = obesity$mat_d$Fu.Li.D[,-1:-3]) + ggtitle("Fu and Li's D"),
          gene_dendros(gene_data = obesity$mat_d$Fu.Li.F[,-1:-3]) + ggtitle("Fu and Li's F"),
          gene_dendros(gene_data = obesity$mat_d$Zeng.E[,-1:-3]) + ggtitle("Zeng's E"),
   cols = 2)
# ggsave(multiplot(gene_dendros(gene_data = obesity$mat_d$Tajima.D[,-1:-3]) + ggtitle("Tajima's D"),
#           gene_dendros(gene_data = obesity$mat_d$Fay.Wu.H[,-1:-3]) + ggtitle("Fay and Wu's H"),
#           gene_dendros(gene_data = obesity$mat_d$Fu.Li.D[,-1:-3]) + ggtitle("Fu and Li's D"),
#           gene_dendros(gene_data = obesity$mat_d$Fu.Li.F[,-1:-3]) + ggtitle("Fu and Li's F"),
#           gene_dendros(gene_data = obesity$mat_d$Zeng.E[,-1:-3]) + ggtitle("Zeng's E"),
#    cols = 2), file = 'images/04_clustering/obesity_dendros.png')
```

(ref:obesitydendros) _Obesity Associated Gene Clustering_ Dendrograms created from hierarchical clustering applied to windows from the `r length(unique(obesity_GR$SYMBOL))` loci associated with obesity. Coloured by super population.

#### Type 2 diabetes

The \gls{gwas} catalog had `r length(unique(t2d_GR$SYMBOL))` genes associated with \gls{t2d}. These were represented by `r nrow(t2d$mat_d$Tajima.D)` windows and clustering of the \gls{t2d} associated genes grouped the individual populations into their super population groups. For all clustering the \gls{amr} populations were split, with \gls{clm} and \gls{pur} grouped with the \gls{eur} populations for \gls{td}, \gls{fwh}, and \gls{ze} (Figure \@ref(fig:t2ddendros)). The \gls{pol} populations were assigned two exclusive groups through the cutree method that created an Eastern/Western Polynesian split for \gls{td}, \gls{fwh}, \gls{flf}, and \gls{ze}. \gls{fwh} and \gls{ze} had a similar pattern to the groupings of the whole chromosome F~ST~ clustering.

There were `r lower_sig_stats %>% filter(SYMBOL %in% unique(t2d_GR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(SYMBOL) %>% distinct() %>% NROW()` \gls{t2d} associated genes that had evidence from at least one selection statistic of possible selection in any of the Polynesian populations from the lower tail of the distributions. There were a total of `r upper_sig_stats %>% filter(SYMBOL %in% unique(t2d_GR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(SYMBOL) %>% distinct() %>% NROW()` genes that had a significant window in any Polynesian population. In the lower tail there were `r lower_sig_stats %>% filter(SYMBOL %in% unique(t2d_GR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(pop, SYMBOL) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==4) %>% NROW()` genes with at least a window that met the significance threshold in the Polynesian populations. Only _THADA_ had windows from \gls{td}, \gls{fwh}, \gls{flf}, and \gls{ze} in the \gls{ton} population from the Polynesian populations. This was also true for the \gls{cdx} and \gls{chs} populations. _ARL15_, _MACF1_, _RREB1_, and _THADA_ each had in all Polynesian populations, at least one statistic that was significant in the lower tail. In the upper tail, _HLA-B_, _HMGA2_, _HNF1B_, _PTPRD_, _RBMS1_, _SGCG_, _TCF7L2_, and _THADA_ each had, in all Polynesian populations, at least one statistic that was significant.

> _How could this be summarised?_

```{markdown, eval = FALSE}
any stat
 - lower `r lower_sig_stats %>% filter(SYMBOL %in% unique(t2d_GR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(pop, SYMBOL) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==4) %>% NROW()`
 - upper `r upper_sig_stats %>% filter(SYMBOL %in% unique(t2d_GR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(pop, SYMBOL) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==4) %>% NROW()`

has all stats
 - lower `r lower_sig_stats %>% filter(SYMBOL %in% unique(t2d_GR$SYMBOL)) %>% group_by(pop, SYMBOL, statname) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON'))  %>% ungroup()%>% select(-n, -statname) %>% group_by(pop, SYMBOL) %>% tally() %>% filter(n == 4) %>% NROW()`
 - upper `r upper_sig_stats %>% filter(SYMBOL %in% unique(t2d_GR$SYMBOL)) %>% group_by(pop, SYMBOL, statname) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON'))  %>% ungroup()%>% select(-n, -statname) %>% group_by(pop, SYMBOL) %>% tally() %>% filter(n == 4) %>% NROW()`



<!--

- td ep/wp split, pol as outlier. pretty good clustering and exclusivity
- fwh ep/wp split, pol-eas as outlier, not bad for exclusivity, clustering was good
- fld nzm-pel. 3/4 pol as outlier group. not too bad for exclusivity
- flf pol-eas, ep/wp. pretty good exclusivity and clusters that are pretty close to the super populations
- ze good exclusivity and good clustering. pol-eas, ep/wp split. amr(clm/pur)-eur

dendrograms
td, fwh, and ze super pops, split amr
flf super pops but split amr and eur together
fld only eas and sas together
all have a wp/ep split but only wp for fld

exclusivity
td ok - but has .5amr, eur and sas together
fwh and flf pretty much same as td
this is because pol is 2 groups
fld poor - amr, eur, sas and afr are all mostly in single cluster
ze good - only has .5 amr and eur together


prop
ze all except amr
td, fwh, and flf have 1.0 for eur,afr and sas. pol is 0.5
fld not bad for all except POL
-->
```

(ref:t2ddendros) _Type 2 Diabetes Associated Gene Clustering_ Dendrograms created from hierarchical clustering applied to windows from the `r length(unique(t2d_GR$SYMBOL))` loci associated with \gls{t2d}. Coloured by super population.

```{r t2ddendros, fig.cap='(ref:t2ddendros)', fig.height=7, message = FALSE}
multiplot(gene_dendros(gene_data = t2d$mat_d$Tajima.D[,-1:-3]) + ggtitle("Tajima's D"),
          gene_dendros(gene_data = t2d$mat_d$Fay.Wu.H[,-1:-3]) + ggtitle("Fay and Wu's H"),
          #gene_dendros(gene_data = t2d$mat_d$Fu.Li.D[,-1:-3]) + ggtitle("Fu and Li's D"),
          gene_dendros(gene_data = t2d$mat_d$Fu.Li.F[,-1:-3]) + ggtitle("Fu and Li's F"),
          gene_dendros(gene_data = t2d$mat_d$Zeng.E[,-1:-3]) + ggtitle("Zeng's E"),
   cols = 2)
# ggsave(multiplot(gene_dendros(gene_data = t2d$mat_d$Tajima.D[,-1:-3]) + ggtitle("Tajima's D"),
#           gene_dendros(gene_data = t2d$mat_d$Fay.Wu.H[,-1:-3]) + ggtitle("Fay and Wu's H"),
#           gene_dendros(gene_data = t2d$mat_d$Fu.Li.D[,-1:-3]) + ggtitle("Fu and Li's D"),
#           gene_dendros(gene_data = t2d$mat_d$Fu.Li.F[,-1:-3]) + ggtitle("Fu and Li's F"),
#           gene_dendros(gene_data = t2d$mat_d$Zeng.E[,-1:-3]) + ggtitle("Zeng's E"),
#    cols = 2), file = 'images/04_clustering/t2d_dendros.png')
```


#### Kidney disease

(ref:kidneydendros) _Kidney Disease Associated Gene Clustering_ Dendrograms created from hierarchical clustering applied to windows from the `r length(unique(kd_GR$SYMBOL))` loci associated with kidney disease. Coloured by super population.

```{r kidneydendros, fig.cap='(ref:kidneydendros)', fig.height=7, message = FALSE}
multiplot(gene_dendros(gene_data = kd$mat_d$Tajima.D[,-1:-3]) + ggtitle("Tajima's D"),
          gene_dendros(gene_data = kd$mat_d$Fay.Wu.H[,-1:-3]) + ggtitle("Fay and Wu's H"),
          #gene_dendros(gene_data = kd$mat_d$Fu.Li.D[,-1:-3]) + ggtitle("Fu and Li's D"),
          gene_dendros(gene_data = kd$mat_d$Fu.Li.F[,-1:-3]) + ggtitle("Fu and Li's F"),
          gene_dendros(gene_data = kd$mat_d$Zeng.E[,-1:-3]) + ggtitle("Zeng's E"),
   cols = 2)
# ggsave(multiplot(gene_dendros(gene_data = kd$mat_d$Tajima.D[,-1:-3]) + ggtitle("Tajima's D"),
#           gene_dendros(gene_data = kd$mat_d$Fay.Wu.H[,-1:-3]) + ggtitle("Fay and Wu's H"),
#           gene_dendros(gene_data = kd$mat_d$Fu.Li.D[,-1:-3]) + ggtitle("Fu and Li's D"),
#           gene_dendros(gene_data = kd$mat_d$Fu.Li.F[,-1:-3]) + ggtitle("Fu and Li's F"),
#           gene_dendros(gene_data = kd$mat_d$Zeng.E[,-1:-3]) + ggtitle("Zeng's E"),
#    cols = 2), file = 'images/04_clustering/kd_dendros.png')
```

The GWAS catalog had `r length(unique(kd_GR$SYMBOL))` genes associated with kidney disease, this was represented by `r nrow(kd$mat_d$Tajima.D)` windows. The clustering of the kidney disease associated genes grouped the individual populations into their super population groups for \gls{afr} and \gls{sas} using \gls{td}, \gls{fwh}, \gls{flf}, and \gls{ze}. Populations from \gls{pol} and \gls{eas} were clustered into separate clusters with \gls{td} and \gls{flf} but were clustered together for \gls{fwh} and \gls{ze} (Figure \@ref(fig:kidneydendros)). \gls{td} clustered populations from \gls{afr}, \gls{eas}, \gls{pol}, and \gls{sas} into super population specific and exclusive clusters.




<!--
dendrograms
td and flf have eas/pol/afr and sas clustered
fwh and ze have a shared eas/pol
fld mixed
flf eas,pol,afr, and sas separated nicely
ze afr, eur, amr, sas separated. eas and pol mixed

exclusivity
td: all exclusive except 2amr in with eur
fwh: sas and eur share, pol and eas share
fld: very poor exclusivity
flf: eas and pol share
ze: eas and pol share

prop
td: all except amr
fwh: pol is split
fld no population completely clustered
flf all except EUR
ze only pol and eas not perfect
-->

There were `r lower_sig_stats %>% filter(SYMBOL %in% unique(kd_GR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(SYMBOL) %>% distinct() %>% NROW()` kidney disease associated genes that had evidence from at least one selection statistic of possible selection in any of the Polynesian populations from the lower tail of the distributions. In the lower tail there was `r lower_sig_stats %>% filter(SYMBOL %in% unique(kd_GR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(pop, SYMBOL) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==4) %>% NROW()` gene, _BRAS3_, with at least one window that was significant in the Polynesian populations. All Polynesian populations had at least one statistic that was significant in the upper tail for a window intersecting _PHTF2_. There was a total of `r upper_sig_stats %>% filter(SYMBOL %in% unique(kd_GR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(pop, SYMBOL) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% NROW()` genes that had at least one window that was significant in the upper tail.

> _How could this be summarised?_

```{markdown, eval = FALSE}
any stat
 - lower `r lower_sig_stats %>% filter(SYMBOL %in% unique(kd_GR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(pop, SYMBOL) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==4) %>% NROW()`
 - upper `r upper_sig_stats %>% filter(SYMBOL %in% unique(kd_GR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(pop, SYMBOL) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==4) %>% NROW()`

has all stats
 - lower `r lower_sig_stats %>% filter(SYMBOL %in% unique(kd_GR$SYMBOL)) %>% group_by(pop, SYMBOL, statname) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON'))  %>% ungroup()%>% select(-n, -statname) %>% group_by(pop, SYMBOL) %>% tally() %>% filter(n == 4) %>% NROW()`
 - upper `r upper_sig_stats %>% filter(SYMBOL %in% unique(kd_GR$SYMBOL)) %>% group_by(pop, SYMBOL, statname) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON'))  %>% ungroup()%>% select(-n, -statname) %>% group_by(pop, SYMBOL) %>% tally() %>% filter(n == 4) %>% NROW()`
```



#### Metabolic syndrome

(ref:metsyndendros) _Metabolic Syndrome Associated Gene Clustering_ Dendrograms created from hierarchical clustering applied to windows from the `r length(unique(metsyn_GR$SYMBOL))` loci associated with metabolic syndrome. Coloured by super population.

```{r metsyndendros, fig.cap='(ref:metsyndendros)', fig.height=7, message = FALSE}
multiplot(gene_dendros(gene_data = metsyn$mat_d$Tajima.D[,-1:-3]) + ggtitle("Tajima's D"),
          gene_dendros(gene_data = metsyn$mat_d$Fay.Wu.H[,-1:-3]) + ggtitle("Fay and Wu's H"),
          #gene_dendros(gene_data = metsyn$mat_d$Fu.Li.D[,-1:-3]) + ggtitle("Fu and Li's D"),
          gene_dendros(gene_data = metsyn$mat_d$Fu.Li.F[,-1:-3]) + ggtitle("Fu and Li's F"),
          gene_dendros(gene_data = metsyn$mat_d$Zeng.E[,-1:-3]) + ggtitle("Zeng's E"),
   cols = 2)
# ggsave(multiplot(gene_dendros(gene_data = metsyn$mat_d$Tajima.D[,-1:-3]) + ggtitle("Tajima's D"),
#           gene_dendros(gene_data = metsyn$mat_d$Fay.Wu.H[,-1:-3]) + ggtitle("Fay and Wu's H"),
#           gene_dendros(gene_data = metsyn$mat_d$Fu.Li.D[,-1:-3]) + ggtitle("Fu and Li's D"),
#           gene_dendros(gene_data = metsyn$mat_d$Fu.Li.F[,-1:-3]) + ggtitle("Fu and Li's F"),
#           gene_dendros(gene_data = metsyn$mat_d$Zeng.E[,-1:-3]) + ggtitle("Zeng's E"),
#    cols = 2), file = 'images/04_clustering/metsyn_dendros.png')
```

The GWAS catalog had `r length(unique(metsyn_GR$SYMBOL))` genes associated with metabolic syndrome, this was represented by `r nrow(metsyn$mat_d$Tajima.D)` windows. The clustering of the metabolic syndrome associated genes grouped individual populations into their super population groups for \gls{td} and \gls{fwh} (Figure \@ref(fig:metsyndendros)). For \gls{flf}, the Polynesian populations were assigned two exclusive clusters that followed the Eastern/Western Polynesian split. \Gls{ze} clustered the individual populations, except for \gls{pel}, into their super populations. The Polynesian populations were grouped into entirely exclusive clusters for all statistics.

There were `r lower_sig_stats %>% filter(SYMBOL %in% unique(metsyn_GR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(SYMBOL) %>% distinct() %>% NROW() %>% num_to_word()` metabolic syndrome associated genes that had evidence from at least one selection statistic of possible selection in any of the Polynesian populations from the lower tail of the distributions. In the lower tail there were no genes with at least one window that was significant in the Polynesian populations. _DGKB_, and _LIPC_ each had, in all Polynesian populations, at least one statistic that was significant in the upper tail. There was a total of `r upper_sig_stats %>% filter(SYMBOL %in% unique(metsyn_GR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(pop, SYMBOL) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% NROW() %>% num_to_word()` genes that had at least a window that was significant in the upper tail.

> _How could this be summarised?_

```{markdown, eval = FALSE}
any stat
 - lower `r lower_sig_stats %>% filter(SYMBOL %in% unique(metsyn_GR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(pop, SYMBOL) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==4) %>% NROW()`
 - upper `r upper_sig_stats %>% filter(SYMBOL %in% unique(metsyn_GR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(pop, SYMBOL) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==4) %>% NROW()`

has all stats
 - lower `r lower_sig_stats %>% filter(SYMBOL %in% unique(metsyn_GR$SYMBOL)) %>% group_by(pop, SYMBOL, statname) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON'))  %>% ungroup()%>% select(-n, -statname) %>% group_by(pop, SYMBOL) %>% tally() %>% filter(n == 5) %>% NROW()`
 - upper `r upper_sig_stats %>% filter(SYMBOL %in% unique(metsyn_GR$SYMBOL)) %>% group_by(pop, SYMBOL, statname) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON'))  %>% ungroup()%>% select(-n, -statname) %>% group_by(pop, SYMBOL) %>% tally() %>% filter(n == 5) %>% NROW()`

<!---

dendrograms

exclusivity

prop
td, fwh, and ze all 1.0 except amr (.75)


- td
- fwh
- fld
- flf
- ze

number of significant
-->

<!--
results from other diseases that are late onset and see if they produce the same patterns
-->
```



```{r, results ='asis', eval = FALSE}
# prop_list[['td_neg']]$exclusivity %>% data.frame%>% mutate(pop = rownames(.)) %>% gather("cluster", "prop", 1:6) %>% filter(prop > 0) %>% group_by(pop) %>% summarise(mean(prop))
# 
# prop_list[['td_neg']]$super_prop %>% data.frame() %>% spread(super, max_prop) 

#prop
stat_to_short <- function(x){
  switch(x,
         Tajima.D = "td",
         Fay.Wu.H = "fwh",
         Fu.Li.D = "fld",
         Fu.Li.F = "flf",
         Zeng.E = "ze",
         x
         )
}

short_to_latex <- function(x){switch(x, 
       td = "\\gls{td}",
       fwh = "\\gls{fwh}",
       fld = "\\gls{fld}",
       flf = "\\gls{flf}",
       ze = "\\gls{ze}",
       x
       )
}

bind_rows(list(
  bind_rows(lapply(names(prop_list), function(x){prop_list[[x]]$super_prop %>%  spread(super, max_prop) %>% mutate(stat = sapply(x, function(y){strsplit(y, '_')[[1]][1]}), Type = sapply(x, function(y){strsplit(y, '_')[[1]][2]}) ) %>% select(NCOL(.), 1:(NCOL(.)-1)) })), #%>% mutate_if(., is.numeric, funs(sprintf("%.3f",.))),
  
  bind_rows(lapply(names(gout[[2]]), function(x){gout[[2]][[x]]$super_prop %>% spread(super, max_prop)%>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1)) })) %>% mutate(Type = 'Urate/Gout'),
  
  bind_rows(lapply(names(t2d[[2]]), function(x){t2d[[2]][[x]]$super_prop %>% spread(super, max_prop)%>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1)) })) %>% mutate(Type = '\\gls{t2d}'),
  
  bind_rows(lapply(names(obesity[[2]]), function(x){obesity[[2]][[x]]$super_prop %>% spread(super, max_prop)%>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1)) })) %>% mutate(Type = 'Obesity'),
  
  bind_rows(lapply(names(metsyn[[2]]), function(x){metsyn[[2]][[x]]$super_prop %>% spread(super, max_prop) %>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1)) })) %>% mutate(Type = 'Metabolic Syndrome') 
  
)) %>% mutate(stat = sapply(stat, function(x){short_to_latex(stat_to_short(x))}), Type = sapply(Type, function(x){switch(x, chr = "Chromosome", neg = "Lower Tail", pos ="Upper Tail",x)})) %>% mutate_if(., is.numeric, funs(sprintf('%.2f',.))) %>% select( Type, stat, everything()) %>% arrange(stat) %>% data.frame %>% kable(., caption = "Proportion", escape = FALSE)#knitr::kable(., caption = "Proportion", booktabs = TRUE) %>% kableExtra::group_rows("Fu and Li's D", 1, 7) %>% kableExtra::group_rows("Fu and Li's F", 8, 14) %>% kableExtra::group_rows("Fay and Wu's H", 15, 21) %>% kableExtra::group_rows("Tajima's D", 22, 28) %>% kableExtra::group_rows("Zeng's E", 29, 35) %>% kableExtra::kable_styling(latex_options = "scale_down")
```




```{r, results = 'asis', eval = FALSE}
#exclusivity
bind_rows(list(
  bind_rows(lapply(names(prop_list), function(x){prop_list[[x]]$exclusivity %>% data.frame%>% mutate(pop = row.names(.)) %>% gather("cluster","prop",1:6) %>% filter(prop > 0) %>% group_by(pop) %>% summarise(mean_prop = mean(prop)) %>% spread(pop, mean_prop) %>% mutate(stat = sapply(x, function(y){strsplit(y, '_')[[1]][1]}), Type = sapply(x, function(y){strsplit(y, '_')[[1]][2]}) ) %>% select(NCOL(.), 1:(NCOL(.)-1)) })),
  
  bind_rows(lapply(names(gout[[2]]), function(x){gout[[2]][[x]]$exclusivity %>% data.frame%>% mutate(pop = row.names(.)) %>% gather("cluster","prop",1:6) %>% filter(prop > 0) %>% group_by(pop) %>% summarise(mean_prop = mean(prop)) %>% spread(pop, mean_prop) %>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1))})) %>% mutate(Type = 'Urate/Gout'),
  
  bind_rows(lapply(names(t2d[[2]]), function(x){t2d[[2]][[x]]$exclusivity %>% data.frame%>% mutate(pop = row.names(.)) %>% gather("cluster","prop",1:6) %>% filter(prop > 0) %>% group_by(pop) %>% summarise(mean_prop = mean(prop)) %>% spread(pop, mean_prop) %>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1))})) %>% mutate(Type = '\\gls{t2d}'),
  
  bind_rows(lapply(names(obesity[[2]]), function(x){obesity[[2]][[x]]$exclusivity %>% data.frame%>% mutate(pop = row.names(.)) %>% gather("cluster","prop",1:6) %>% filter(prop > 0) %>% group_by(pop) %>% summarise(mean_prop = mean(prop)) %>% spread(pop, mean_prop) %>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1))})) %>% mutate(Type = 'Obesity'),
  
  bind_rows(lapply(names(metsyn[[2]]), function(x){metsyn[[2]][[x]]$exclusivity %>% data.frame%>% mutate(pop = row.names(.)) %>% gather("cluster","prop",1:6) %>% filter(prop > 0) %>% group_by(pop) %>% summarise(mean_prop = mean(prop)) %>% spread(pop, mean_prop) %>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1))})) %>% mutate(Type = 'Metabolic Syndrome')    
)) %>% select(stat, Type, everything())  %>% mutate(stat = sapply(stat, function(x){short_to_latex(stat_to_short(x))}), Type = sapply(Type, function(x){switch(x, chr = "Chromosome", neg = "Lower Tail", pos ="Upper Tail",x)})) %>% mutate_if(., is.numeric, funs(sprintf("%.3f",.))) %>% arrange(Type) %>%  data.frame %>% pander::pandoc.table(., caption = "Exclusivity")#knitr::kable(., caption = "Exclusivity", booktabs = TRUE) %>% kableExtra::group_rows("Fu and Li's D", 1, 7) %>% kableExtra::group_rows("Fu and Li's F", 8, 14) %>% kableExtra::group_rows("Fay and Wu's H", 15, 21) %>% kableExtra::group_rows("Tajima's D", 22, 28) %>% kableExtra::group_rows("Zeng's E", 29, 35) 
```
<!-- \FloatBarrier -->

### Random draws

```{r load_rand, results='asis'}
# only need to do for one stat since it's a true/false matrix
create_random_tab <- function(rand){
   bind_rows(lapply(seq_along(rand), function(x){rand[[x]]$super_prop})) %>% group_by(super) %>% group_by(super) %>% summarise(min = round(min(max_prop),3), mean = round(mean(max_prop), 3), sd = round(sd(max_prop), 3), max = round(max(max_prop),3), prop50 = round(sum(if_else(max_prop >=0.5, 1, 0))/NROW(.) * 100, 1) , total_complete = sum(if_else(max_prop == 1, 1, 0))) %>% data.frame()
}

# k = 6
td_random <- readRDS('~/data/NZ_coreExome_1kgp/100kbWindow_intra/td_random_filtered_3ns-29-1-2018.RDS')
rand_tab <- create_random_tab(td_random)
#k =  9
td_random9 <- readRDS('~/data/NZ_coreExome_1kgp/100kbWindow_intra/td_random_filtered_3ns_k_9-29-1-2018.RDS')
rand_tab9 <- create_random_tab(td_random9)
# k = 11
td_random11 <- readRDS('~/data/NZ_coreExome_1kgp/100kbWindow_intra/td_random_filtered_3ns_k_11-29-1-2018.RDS')
rand_tab11 <- create_random_tab(td_random11)
```




```{r, eval = FALSE}
#count the number of times the Polynesians were all grouped into an exclusive cluster
a <- c()
b<- c()
for(i in seq_along(td_random)){
  if(td_random[[i]]$tab %>% data.frame %>% spread(Var2, Freq) %>% select(-1) %>% filter(POL == 4 & rowSums(.)== 4) %>% NROW() > 0){a <- c(a,i)
  }
  if(td_random[[i]]$super_prop %>% filter(super == "POL" & max_prop == 1) %>% NROW() > 0){b <- c(b,i)
  }
}
```


```{r randexcl}
#exclusivity
create_exclusivity_tab <- function(rand){
 rand %>% map("exclusivity") %>%
  map(., ~ t(.) %>% data.frame() %>%  mutate_all(., funs(replace(., . == 0, NA)))) %>% # replace 0 because it needs to be excluded in the calulation to not drag the mean down
  map_dfr(~ summarise_all(., funs(mean(., na.rm=TRUE), min(., na.rm = TRUE), max(., na.rm = TRUE)))) %>%
  summarise_all(funs(min,mean,max)) %>% t() %>%
  data.frame() %>%   select(value = 1)  %>%
  mutate(pop = sapply(row.names(.), function(x){strsplit(x, '_')[[1]][1]}),
         stat1 = sapply(row.names(.), function(x){strsplit(x, '_')[[1]][2]}),
         stat2 = sapply(row.names(.), function(x){strsplit(x, '_')[[1]][3]}) ) %>%
  filter(stat1 == stat2) %>% select(-stat2) %>% spread(stat1, value)
}
# td_random %>% head()%>% map("exclusivity") %>%
#   map(., ~ t(.) %>% data.frame() %>%  mutate_all(., funs(replace(., . == 0, NA)))) %>%
#   map_dfr(~ summarise_all(., funs(mean(., na.rm=TRUE), min(., na.rm = TRUE), max(., na.rm = TRUE)))) %>%
#   summarise_all(funs(min,mean,max)) %>% t() %>%
#   data.frame() %>%   select(value = 1)  %>%
#   mutate(pop = sapply(row.names(.), function(x){strsplit(x, '_')[[1]][1]}),
#          stat1 = sapply(row.names(.), function(x){strsplit(x, '_')[[1]][2]}),
#          stat2 = sapply(row.names(.), function(x){strsplit(x, '_')[[1]][3]}) ) %>%
#   filter(stat1 == stat2) %>% select(-stat2) %>% spread(stat1, value)

# excl_tab <- create_exclusivity_tab(td_random)
# excl9_tab <- create_exclusivity_tab(td_random9)
# excl11_tab <- create_exclusivity_tab(td_random11)
# saveRDS(excl_tab, 'data/04-excl_tab.RDS')
# saveRDS(excl9_tab, 'data/04-excl9_tab.RDS')
# saveRDS(excl11_tab, 'data/04-excl11_tab.RDS')

excl_tab<-readRDS('data/04-excl_tab.RDS')
excl9_tab<-readRDS('data/04-excl9_tab.RDS')
excl11_tab<-readRDS('data/04-excl11_tab.RDS')
#excl_tab

# bind_rows(lapply(seq_along(td_random), function(x){td_random[[x]]$exclusivity %>% t() %>% data.frame %>% summarise_all(., funs(max)) %>% filter(POL == 1)})) 
# 
# bind_rows(lapply(seq_along(td_random), function(x){td_random[[x]]$exclusivity %>% data.frame()%>% mutate(pop = row.names(.))})) %>% t()%>% select(-clust) %>% gather("pop","prop") %>% filter(prop > 0) %>% group_by(pop) %>% summarise(min(prop), quantile(prop, 0.25), mean(prop),sd(prop), median(prop),quantile(prop, 0.75) , max(prop))

#count how many iterations a super population was in an exclusive cluster
#
create_rand_tally_tab <- function(rand){
  rand  %>% map("exclusivity") %>% 
    imap_dfr(~ data.frame(.x) %>% 
               mutate(super = row.names(.)) %>% 
               gather(cluster,prop, 1:(NCOL(.)-1)) %>%
               filter(prop == 1) %>%  
               mutate(iter = .y)) %>% 
    group_by(super) %>% 
    tally()
}

# excl_tally <- create_rand_tally_tab(td_random)
# excl9_tally <- create_rand_tally_tab(td_random9)
# excl11_tally <- create_rand_tally_tab(td_random11)
# saveRDS(excl_tally, 'data/04-excl_tally.RDS')
# saveRDS(excl9_tally, 'data/04-excl9_tally.RDS')
# saveRDS(excl11_tally, 'data/04-excl11_tally.RDS')
#bind_rows(lapply(seq_along(td_random), function(x){td_random[[x]]$exclusivity  %>% data.frame %>% mutate(super = row.names(.)) %>% gather(cluster,prop, 1:(NCOL(.)-1)) %>%filter(prop == 1) %>%  mutate(iter = x)})) %>% group_by(super) %>% tally()
#saveRDS(excl_tally, 'data/04-excl_tally.RDS')
excl_tally <- readRDS('data/04-excl_tally.RDS')
#excl_tally

#bind_rows(lapply(seq_along(td_random)[1:10], function(x){td_random[[x]]$exclusivity %>% t() %>% data.frame %>% mutate_all(., funs(replace(., . == 0, NA))) %>% summarise_all(., funs(mean(., na.rm=TRUE), min(., na.rm = TRUE), max(., na.rm = TRUE)))})) %>% summarise_all(funs(min,mean,max)) %>% t %>%data.frame() %>% select(value = 1)  %>% mutate(pop = sapply(row.names(.), function(x){strsplit(x, '_')[[1]][1]}), stat1 = sapply(row.names(.), function(x){strsplit(x, '_')[[1]][2]}), stat2 = sapply(row.names(.), function(x){strsplit(x, '_')[[1]][3]}) ) %>% filter(stat1 == stat2) %>% select(-stat2) %>% spread(stat1, value)

```

```{r over50per, cache = TRUE}
# over50per <- bind_rows(lapply(td_random, function(x){ x[['super_prop']] %>% spread('super', 'max_prop')})) %>% filter(POL >= 2/4, EUR >= 3/6, EAS >= 3/5, AFR >= 3/6, SAS >= 3/5) %>% arrange(POL)
# saveRDS(over50per, file = 'data/04-over50per.RDS')
# over50per2 <- bind_rows(lapply(td_random, function(x){ x[['super_prop']] %>% spread('super', 'max_prop')})) %>% filter(POL >= 2/4, EUR >= 3/6, EAS >= 3/5, AFR >= 3/6, SAS >= 3/5, AMR >=2/4 ) %>% arrange(POL)
# saveRDS(over50per2, file = 'data/04-over50per2.RDS')

over50per <- readRDS('data/04-over50per.RDS')
over50per2 <- readRDS('data/04-over50per2.RDS')
```


<!-- why does the Eur never produce a max prop of 1?-->

#### Random windows

It is possible that obtaining the population groupings by clustering the most extreme windows for each statistic, could have occurred by chance. To assess this, 10,000 iterations of random resampling of 2500 windows per population was undertaken, and these windows were then clustered. This was done so that the extreme tail clustering could be compared to the random windows to see if the extreme window clustering performed better in exclusivity and proportion of super population clustering than by drawing windows at random. The maximum proportion of populations being clustered into their designated super population was then calculated when the tree was cut to have six groups (K = 6), one for each super population. The mean proportion of populations grouped into their super population was less than 0.5 for all super populations, with an overall mean of 0.282 (Table \@ref(tab:randtab)). This compared to a mean proportion of 0.916 for the clustering of the extreme windows. Other values of K were also tested for K = 9, and K = 11, based on the results of the admixture analysis. They both had higher mean exclusivities (K = 9: 0.367, K = 11: 0.424) than K = 6, due to more possible individual clusters. The extra values of K were selected based on the lowest cross-validation errors for the calculations using all populations, and the calculations omitting the \gls{amr} and \gls{pol} populations (section \@ref(admixture)). The remainder of this section will report on K = 6.
<!-- explain CV a little more in a sentence -->

Not a single random draw grouped the all individual populations into all of their respective super populations. Only the populations of the \gls{afr} and \gls{eur} super populations did not each get fully grouped into a cluster. Grouping at least 50% of a super population was at best `r round(rand_tab %>% summarise(max = max(prop50)), digits = 0) %>% pull(max)`% of the draws but for the larger super populations (\gls{afr}, \gls{eas}, \gls{eur}, and \gls{sas}) it was on average `r round(rand_tab %>% filter(!super %in% c('POL','AMR'))%>% summarise(mean = mean(prop50)), digits = 0) %>% pull(mean)`% of the draws. There were only `r over50per %>% NROW() %>% num_to_word()` random draws that had all super populations each having half or more of their individual populations assigned the same group. The mean exclusivity of the random draws was low, ranging from `r min(excl_tab$mean) %>% sprintf('%.2f', .)` to `r max(excl_tab$mean) %>% sprintf('%.2f', .)` and there were no random draws in which the Polynesian populations were all grouped together as a Polynesian exclusive cluster.


Compared to the clustering on the extreme tails, the random windows had poorer exclusivity of clusters with a mean exclusivity of 28%. The populations were clustered into their super populations rarely for the random windows and did not have multiple super populations being clustered as the clustering of the extremes returned (Table \@ref(tab:randtab)). This indicates that there is a degree of similarity of regions that are in the extremes of the distributions and the populations who share them. This is largely due a minimum degree of differences in similarity being needed in order for hierarchical clustering to be effective for clustering. When windows are drawn at random the differences, or distances, between populations is likely to be equivalent, so the hierarchy is flatter, meaning small differences have very large changes to the groupings.

<!-- >important result that needs a little more expansion and also comparison with the extremes -->

```{r randtabpander, results = 'asis'}
rand_tab_pander <- rand_tab %>% mutate_at(., vars(-super, -total_complete, -prop50), funs(sprintf("%.3f", .))) %>% mutate(prop50 = sprintf('%.1f', prop50))
#colnames(rand_tab_pander) <- c('Super Population', 'Min','Mean','Std. Dev','Max','Prop50 (%)', 'Total Complete')
rand_tab_pander %>% kable(., caption = '(\\#tab:randtab) Proportion of individual populations assigned to their super population by hierarchical clustering across 10000 iterations of 2500 randomly drawn windows.', booktabs = TRUE, col.names = c('Super Population', 'Min','Mean','Std. Dev','Max','Prop50 (%)', 'Total Complete (n)'), align = "lrrrrrr" ) %>% footnote(general = "Prop50 is the percentage of the iterations that had at least 50\\\\% of each super population assigned the same cluster. Total complete is the number of iterations all populations of a super population were assigned the same cluster", threeparttable = TRUE, general_title = "", escape = FALSE)

#pander::pandoc.table(rand_tab_pander, caption = '(\\#tab:randtab) *Summary of proportion of individual populations assigned to their super population cluster.* Min is the minimum proportion for the cluster from all iterations. Mean is the mean proportion of the maximum cluster proportions for all iterations. Max is the maximum proportion from all iterations. Prop50 is the number of times at least 50% of the super population was clustered together, expressed as a percentage. Total complete is the number of times the super population had all of its individual populations assigned the same cluster.', justify = c("left",rep("right", NCOL(rand_tab_pander)-1)), keep.line.breaks = TRUE)
```


```{r, eval = FALSE}
#calculate the mean proportion for the population clusters across all iterations
# remove 0 because they aren't actually a cluster
bind_rows(lapply(seq_along(td_random), function(x){td_random[[x]]$pop_clust_prop %>% data.frame()%>% mutate(clust = row.names(.))})) %>% select(-clust) %>% gather("pop","prop") %>% filter(prop > 0) %>% group_by(pop) %>% summarise(min(prop), quantile(prop, 0.25), mean(prop),sd(prop), median(prop),quantile(prop, 0.75) , max(prop))

bind_rows(lapply(seq_along(td_random), function(x){td_random[[x]]$pop_clust_prop %>% data.frame()%>% mutate(clust = row.names(.))})) %>% select(-clust) %>% gather("pop","prop") %>% filter(prop > 0) %>% group_by(pop) %>% ggplot(., aes(x =pop, y = prop)) + geom_violin()
```





```{r, eval = FALSE}
bind_rows(lapply(seq_along(td_random), function(x){td_random[[x]]$pop_clust_prop %>% data.frame()%>% mutate(clust = row.names(.), iter = x)})) 
#bind_rows(lapply(td_random, function(x){ x[['super_prop']] %>% spread('super', 'max_prop')})) %>% filter(POL >= 2/4, EUR >= 3/6, EAS >= 3/5, AFR >= 3/6, SAS >= 3/5) %>% arrange(POL)

#bind_rows(lapply(td_random, function(x){ x %>% spread('super', 'max_prop')})) %>% mutate(rn = as.numeric(rownames(.)), exp_prop = rowSums(.)/6) %>% filter(EUR > .5) %>% arrange(exp_prop)
```

#### Random genes {#randomgenes}

In order to assess whether the clustering from gene lists was specific to the disease associated genes used, lists of random genes were created by selecting genes from a list containing all annotated genes, and then clustered. Random gene lists were generated with either 25 or 100 randomly selected genes, 100 times each. Hierarchical clustering was then performed on the the median centred statistic value based on the population, for windows that intersected genes from the random gene lists for \gls{td}, \gls{fwh}, \gls{flf}, and \gls{ze}. The number of times the Polynesian super population was completely clustered ranged from 5 to 78 (Table \@ref(tab:randgenes)). The Polynesian super population was completely clustered more often with the lists of 100 genes for \gls{td}, \gls{fwh}, \gls{flf}, and \gls{ze} compared to the lists of 25 genes. The number of times all populations were clustered completely was less than 10%. However, if the \gls{amr} super population was not included then it became 56 for the 100 genes, and 36.3 for the 25 genes. \gls{flf} did not perform well at clustering the Polynesian populations into their super population, and only achieved this less than 13% of the time. Clustering all populations was not achieved by \gls{flf}. The other frequency spectrum statistics clustered the Polynesian populations into their super population for at least 50% of the time and did better with more genes. Clustering of the other super populations was similar to the Polynesian super population when the \gls{amr} populations were not included. This indicates that clustering from disease associated gene lists is not specific to particular genes. This was evident from that fact that if the \gls{amr} populations are excluded, by chance the super populations were correctly clustered at least 50% of the time, and this increased to ~75% when more genes were used. It also indicates that genes are better at clustering the super populations than randomly drawn windows. 



```{r, cache = TRUE}
load('~/data/NZ_coreExome_1kgp/100kbWindow_intra/random_gene_clustering_filtered_3ns_sig-14-2-2018.RData')



poly_complete <- bind_rows(
  bind_rows(
    bind_rows(lapply(seq_along(random_genes100), function(x){random_genes100[[x]][[2]]$Tajima.D$super_prop %>% spread(super, max_prop) }))  %>% mutate(statname = 'Tajima.D') , 
    bind_rows(lapply(seq_along(random_genes100), function(x){random_genes100[[x]][[2]]$Fay.Wu.H$super_prop  %>% spread(super, max_prop) })) %>% mutate(statname = 'Fay.Wu.H'),
    bind_rows(lapply(seq_along(random_genes100), function(x){random_genes100[[x]][[2]]$Fu.Li.F$super_prop  %>% spread(super, max_prop) })) %>% mutate(statname = 'Fu.Li.F'),
    bind_rows(lapply(seq_along(random_genes100), function(x){random_genes100[[x]][[2]]$Zeng.E$super_prop  %>%spread(super, max_prop) })) %>% mutate(statname = 'Zeng.E')
  ) %>%mutate(total = select_if(., is.numeric) %>% rowSums) %>% filter(POL == 1) %>% group_by(statname) %>% tally() %>% mutate(gene_n = 100),
  bind_rows(
    bind_rows(lapply(seq_along(random_genes25), function(x){random_genes25[[x]][[2]]$Tajima.D$super_prop %>% spread(super, max_prop) }))  %>% mutate(statname = 'Tajima.D') , 
    bind_rows(lapply(seq_along(random_genes25), function(x){random_genes25[[x]][[2]]$Fay.Wu.H$super_prop  %>% spread(super, max_prop) })) %>% mutate(statname = 'Fay.Wu.H'),
    bind_rows(lapply(seq_along(random_genes25), function(x){random_genes25[[x]][[2]]$Fu.Li.F$super_prop  %>% spread(super, max_prop) })) %>% mutate(statname = 'Fu.Li.F'),
    bind_rows(lapply(seq_along(random_genes25), function(x){random_genes25[[x]][[2]]$Zeng.E$super_prop  %>%spread(super, max_prop) })) %>% mutate(statname = 'Zeng.E')
  ) %>% 
    mutate(total = select_if(., is.numeric) %>% rowSums) %>% filter(POL == 1) %>% group_by(statname) %>% tally() %>% mutate(gene_n = 25)
) 


perc_complete <- bind_rows(
  bind_rows(
    bind_rows(lapply(seq_along(random_genes100), function(x){random_genes100[[x]][[2]]$Tajima.D$super_prop %>% spread(super, max_prop) }))  %>% mutate(statname = 'Tajima.D') , 
    bind_rows(lapply(seq_along(random_genes100), function(x){random_genes100[[x]][[2]]$Fay.Wu.H$super_prop  %>% spread(super, max_prop) })) %>% mutate(statname = 'Fay.Wu.H'),
    bind_rows(lapply(seq_along(random_genes100), function(x){random_genes100[[x]][[2]]$Fu.Li.F$super_prop  %>% spread(super, max_prop) })) %>% mutate(statname = 'Fu.Li.F'),
    bind_rows(lapply(seq_along(random_genes100), function(x){random_genes100[[x]][[2]]$Zeng.E$super_prop  %>%spread(super, max_prop) })) %>% mutate(statname = 'Zeng.E')
  ) %>%mutate(total = select_if(., is.numeric) %>% rowSums) %>% filter(POL >= 4/4, EUR >= 6/6, EAS >= 5/5, AFR >= 6/6, SAS >= 5/5, AMR >= 4/4) %>% group_by(statname) %>% tally() %>% mutate(gene_n = 100),
  bind_rows(
    bind_rows(lapply(seq_along(random_genes25), function(x){random_genes25[[x]][[2]]$Tajima.D$super_prop %>% spread(super, max_prop) }))  %>% mutate(statname = 'Tajima.D') , 
    bind_rows(lapply(seq_along(random_genes25), function(x){random_genes25[[x]][[2]]$Fay.Wu.H$super_prop  %>% spread(super, max_prop) })) %>% mutate(statname = 'Fay.Wu.H'),
    bind_rows(lapply(seq_along(random_genes25), function(x){random_genes25[[x]][[2]]$Fu.Li.F$super_prop  %>% spread(super, max_prop) })) %>% mutate(statname = 'Fu.Li.F'),
    bind_rows(lapply(seq_along(random_genes25), function(x){random_genes25[[x]][[2]]$Zeng.E$super_prop  %>%spread(super, max_prop) })) %>% mutate(statname = 'Zeng.E')
  ) %>% 
    mutate(total = select_if(., is.numeric) %>% rowSums) %>%filter(POL >= 4/4, EUR >= 6/6, EAS >= 5/5, AFR >= 6/6, SAS >= 5/5, AMR >= 4/4)%>% group_by(statname) %>% tally() %>% mutate(gene_n = 25)
)

perc_complete_no_amr <- bind_rows(
  bind_rows(
    bind_rows(lapply(seq_along(random_genes100), function(x){random_genes100[[x]][[2]]$Tajima.D$super_prop %>% spread(super, max_prop) }))  %>% mutate(statname = 'Tajima.D') , 
    bind_rows(lapply(seq_along(random_genes100), function(x){random_genes100[[x]][[2]]$Fay.Wu.H$super_prop  %>% spread(super, max_prop) })) %>% mutate(statname = 'Fay.Wu.H'),
    bind_rows(lapply(seq_along(random_genes100), function(x){random_genes100[[x]][[2]]$Fu.Li.F$super_prop  %>% spread(super, max_prop) })) %>% mutate(statname = 'Fu.Li.F'),
    bind_rows(lapply(seq_along(random_genes100), function(x){random_genes100[[x]][[2]]$Zeng.E$super_prop  %>%spread(super, max_prop) })) %>% mutate(statname = 'Zeng.E')
  ) %>%mutate(total = select_if(., is.numeric) %>% rowSums) %>% filter(POL >= 4/4, EUR >= 6/6, EAS >= 5/5, AFR >= 6/6, SAS >= 5/5) %>% group_by(statname) %>% tally() %>% mutate(gene_n = 100),
  bind_rows(
    bind_rows(lapply(seq_along(random_genes25), function(x){random_genes25[[x]][[2]]$Tajima.D$super_prop %>% spread(super, max_prop) }))  %>% mutate(statname = 'Tajima.D') , 
    bind_rows(lapply(seq_along(random_genes25), function(x){random_genes25[[x]][[2]]$Fay.Wu.H$super_prop  %>% spread(super, max_prop) })) %>% mutate(statname = 'Fay.Wu.H'),
    bind_rows(lapply(seq_along(random_genes25), function(x){random_genes25[[x]][[2]]$Fu.Li.F$super_prop  %>% spread(super, max_prop) })) %>% mutate(statname = 'Fu.Li.F'),
    bind_rows(lapply(seq_along(random_genes25), function(x){random_genes25[[x]][[2]]$Zeng.E$super_prop  %>%spread(super, max_prop) })) %>% mutate(statname = 'Zeng.E')
  ) %>% 
    mutate(total = select_if(., is.numeric) %>% rowSums) %>%filter(POL >= 4/4, EUR >= 6/6, EAS >= 5/5, AFR >= 6/6, SAS >= 5/5)%>% group_by(statname) %>% tally() %>% mutate(gene_n = 25)
)
```

<!-- need to get fld in the 100 -->
```{r}
# add in fld 100 because it has no results, therefore wasn't in the original table
poly_complete %>% full_join(., perc_complete %>% select(allcomplete = n, everything()), by = c('statname','gene_n')) %>% full_join(., perc_complete_no_amr %>% select(noAMR = n, everything()), by = c('statname','gene_n'))%>% mutate(allcomplete = ifelse(is.na(allcomplete),0,allcomplete), noAMR = ifelse(is.na(noAMR),0,noAMR)) %>% arrange(gene_n, statname)%>% select("Statistic" = statname, "Genes (n)" = gene_n, 'POL Complete (%)' = n, 'All Complete (%)' = allcomplete, 'All Complete excl. AMR (%)' = noAMR, everything()) %>% kable(., booktabs = TRUE, caption = '(\\#tab:randgenes) Percentage of draws from randomly selected genes for completely clustered super populations.', align = "lrrrr")
```




\clearpage

## Chapter discussion

<!-- ### Clustering by selection and neutrality statistic -->

<!-- Objective 1: can neutrality and selection statistics be used to group populations? -->

<!-- Yes, the super populations can be grouped -->

<!-- - yes - F~ST~ at chromosome wide scale does -->
<!-- - no - sfs does not at chromosome wide -->
<!-- - yes - sfs does at the extremes of the distribution (mostly) -->
<!-- - no - for random windows -->
<!-- - yes - for haplotypes -->
<!-- - yes - for disease genes -->
<!-- - yes - for random genes -->



### Use of selection and neutrality statistics to group populations

Other studies have previously shown the affinities between Polynesians and other Pacific populations, most often with mitochondrial DNA and Y chromosome markers [@Kayser2010; @Duggan2014a]. Autosomes have also been used to explore Polynesian origins and relationships between Pacific populations, through generally with rather small sample sizes [@Kimura2008; @Kayser2010; @Hudjashov2017; @Hudjashov2018]. This study is the first to compare four Polynesian populations, two of East, and two of West Polynesian ancestry, and of similar sample sizes to those of the \glsdesc{1kgp}. The analysis is important because it adds new evidence of the genetic relationship between the Polynesian populations and other populations. This is important, as understanding the ancestry of populations helps to inform and reduce the stigma of genetic diseases [@Sankar2006].

In Chapter 3 it was noted that the populations had population specific distributions for the frequency spectrum based selection and neutrality statistics. The tails of the distributions contain the most extreme values and are therefore most likely to be the loci that have undergone selection - specifically the negative tail is of interest for positive selection. It was thought that perhaps the tails could be used to group the individual populations using an unsupervised method - in this case, hierarchical clustering - and reassemble the super populations. The first test used the entire chromosome and created a summary statistic that was used to create clusters. It was successful in re-assembling the super populations using F~ST~ which is a measure of population differentiation, and had previously been used within the \gls{1kgp} dataset to establish how differentiated the \gls{1kgp} samples were from each other [@1KGP2015snp]. The population groupings were also the same as in the \gls{pca} that was used to select the individuals for the selection analysis (Figure \@ref(fig:allpopPCA)). Both the F~ST~ results and PCA (for the subject selection) confirmed the genetic relatedness and differentiation of the Polynesian populations in a global context, as being most similar to the East Asian populations [@Kayser2008a]. It also confirmed that within Polynesia, the differentiation between Eastern and Western Polynesia, consistent with the out of Africa human migration and subsequent Polynesian expansion [@Nielsen2017; @Matisoo-Smith2015; @Skoglund2016; @Hudjashov2018]. 

#### Clustering of aggregate statistics 

The whole chromosome summary using the intra-population frequency spectrum neutrality and selection statistics had varying levels of success in re-assembling the super populations. \gls{flf} had mini-groups of 2-3 populations that were part of the same super population, but these mini-groups were not closely linked to the other groups of the same super population. These two statistics are based on singletons, and it is likely that these are not well represented in the markers of the CoreExome chip itself, or the quality control that the New Zealand samples underwent. This would lead to an under-representation of the singletons that actually might be present in the populations. Overall, it was found that the clustering of selection and neutrality test statistics for entire chromosomes did not resemble the grouping that clustering on F~ST~ produced. 

#### Clustering of windowed statistics

However, compared to the entire chromosome summaries, using windows from the extremes of the distribution was effective at clustering the individual populations into their super population groups. \gls{fwh} was the only statistic that grouped all individual populations into their corresponding super populations, whereas \gls{td} and \gls{ze} had the same split of the \gls{amr} populations as with the F~ST~ clustering. The clustering of populations into their super populations was not seen when random windows were used for clustering, indicating that there is a degree of commonality in the extreme windows within a super population group.The haplotypic based methods of \gls{ihs} and \gls{nsl} clustered into super populations and had the \gls{amr} population split too. There was not a high amount of region sharing outside of each super population.

#### Clustering on genic regions

Clustering from gene lists for the frequency spectrum based selection and neutrality statistics consistently grouped the individual populations into their super populations although there was inconsistency for the \gls{amr} populations. With the exception of the \gls{amr} populations, the clustering of populations using windows or haplotypes was consistent with previously reported results [@1KGP2015snp; @pickrell2009signals]. The grouping of populations was also seen with random lists of genes. This would indicate that the commonalities with populations for a gene are not specific to a disease, but instead, a general locus by locus similarity.

<!-- > talk about how selection on genes vs non gene regions - random windows are much more likely to be intergenic 
this is not necessarily the case, due to the way things were annotated
-->

Allele sharing between populations is moderate for common \glspl{snp}, and more so for continental groups, but is minimal for rarer frequencies, due to population divergence [@Gravel2011]. This helps to explain both of the patterns observed with \gls{sfs} based statistics for both the extremes and gene clustering. With the extremes, many of the super populations had windows that were not in the extremes of the other populations, these formed the 'blocks' in the heatmap. This was most evident with the \gls{afr} populations, which had the majority of windows for each statistic not in common with the extremes from the other populations.

The extremes of the distributions represent different types of selection or demographic events and are different depending on the selection test statistic being used [@Tajima1989; @fay2000hitchhiking; @Fu1993; @Zeng2006]. The upper end of the distribution is able to be compared as all of the values that it encompassed were positive in value across all populations for all statistics and represent the same types of selection or demography.

The clustering of the most extreme windows represents, other than a similarity in allele frequencies between populations, a pattern that is consistent with the out of Africa model of human migration [@Mallick2016; @Pagani2016].

### Potential shared selective histories for loci associated with urate and metabolic disease

<!-- Extreme windows  -->

<!-- - any metabolic disease related windows in the extreme? -->
<!-- - were those windows in common with other populations? -->

<!-- Clustering on genes -->

<!-- - did the genes separate the populations? -->

<!-- Haplotypic -->
<!-- - Any genes significant from the clusters? -->
<!-- - were any snps/regions in common between Poly and other populations? -->


<!-- intro/set scene -->
<!-- clustering on the gene values - this says that there is similarities between the populations and what groups are closest -->
<!-- genes that were in the extremes - this givesa compare/contrast for the Polynesian populations -->
<!-- haplotypic sharing - shows if there were likely something haplotypically similar between populations -->
<!-- could the genes that were distinctive explain the disease burden differences -->


The "Out of Africa" model of human migration can be useful in explaining differences between modern populations and potential time points when these differences may have arisen. Under a neutral model the differences in allele frequency between populations are due to genetic drift [@kimura1979neutral]. Demographic events such as bottleneck, migration, expansion, and admixture also contribute to the differences in allele frequencies found between populations [@Wright1951]. Selection on the other hand acts in a targeted manner, therefore if the selection occurred post population split then it would be expected to look neutral in one population but not in the other. However, if the selective event happened in an ancestral population of the two then, dependent on time since divergence, it would be expected to be observed in both populations [@Hermisson2017]. It is also possible to have 'parallel' selection, where both populations experience independent selection [@Tennessen2011]. Using a coalescent approach to identify the common intermediate ancestral populations, if for example, a genomic region identified as under selection in the Polynesian populations was also found in the East Asian populations, but not the Europeans, it could be inferred that the selective pressure was likely after the migration from Europe to East Asia. In the context of modern metabolic disease, the question being asked is, is there evidence in the ancestral populations of selection in metabolic disease associated loci that is in common with the Polynesian populations?

The current metabolic disease burden in the Polynesian populations is higher than other populations [@Winnard2012; @Winnard2013]. This higher disease burden could possibly have arisen due to environmental changes and selective pressure on Polynesian populations after the migration into the Pacific. As found in Chapter 3, however, there was limited evidence for selection, especially in the overlap between haplotypic and \gls{sfs}-based methodologies, in the genes that were associated with urate and associated metabolic diseases (Table \@ref(tab:superDiseaseGenesTab)). Three approaches were taken to investigate the potential of a 'shared selective' history of the loci associated with urate and metabolic disease. These included, investigating the loci that were in either the 1^st^ or 99^th^ percentile, the loci that were in the clustered regions for the significant \glspl{snp} of the haplotypic statistics, \gls{ihs} and \gls{nsl}. And finally, hierarchical clustering of the median-centred values for windows that intersected the loci associated with urate and metabolic disease.

For the \gls{sfs}-based statistics, many of the windows in either the 1^st^ or 99^th^ percentile that intersected the genes associated with metabolic disease were not in common with other populations. The windows that were in common, were usually with the \gls{eas} populations. This is consistent with the migratory history of the Pacific and the shared ancestry the Polynesian populations have with East Asia.


#### Clustering on median-centred gene values

The clustering based on the median-centred values of the statistics was able to reassemble the super population groups. Again the similarity of the allele frequency spectrum based on geography appeared through urate and metabolic disease genes. Both exclusivity and proportion of populations clustered correctly was very high for nearly all populations for the four site-frequency statistics, across the metabolic disease gene lists (Tables \@ref(tab:proportion) and \@ref(tab:exclusivity)). This indicated that there was sufficient similarity between populations of a super population, and differences between super populations that the statistic values were discriminatory. However, the groups were also reassembled with the random draws of genes, with an increase in the number of completely grouped super populations when more genes were used. This showed that the particular genes themselves were not specific to the clustering but the frequency distributions in genes were different and the number of genes influenced the discrimination of populations during clustering.

The simplest explanation of why similar clustering occurred with the gene lists is that with the tests of neutrality or selection that rely on the frequency spectrum, populations that are similar geographically and/or ancestrally, also have a similar allele frequency to each other [@Gravel2011]. This would explain why, with \gls{td}, \gls{fwh}, and \gls{ze}, which are based on comparing ratios of low, intermediate and high frequency variants, all were clustered into their super population groups. 

#### Clustering of extreme value windows

The differences in the extremes of the distributions for the site-frequency spectrum statistics showed that they could be used to cluster the populations into their super populations. The differences and commonalities in the genes that had evidence of association with a metabolic disease, in the 1^st^ percentiles of the statistics for the Polynesian populations, were relatively isolated to only the Polynesian populations. The 1^st^ percentile represents the values with the highest likelihood of being under selection. The windows that were in common with other populations were, for the most part, only found in the populations of \gls{eas}. The windows or \glspl{snp} that were significant in the Polynesian populations within the metabolic disease gene lists were usually limited to commonality with the paired ancestrally similar population (\gls{cim} and \gls{nzm}, or \gls{sam} and \gls{ton}). Windows were more likely to be in common between populations for the 99^th^ percentiles where this represented an increase in high frequency variants.

#### Clustering of haplotype-based statistics

For the haplotypic statistics, there were commonalities between some of the significant \glspl{snp}. The \glspl{snp} that were in common were more often in pairs of either \gls{sam} and \gls{ton}, or \gls{cim} and \gls{nzm}. The \glspl{snp} that were significant for a Polynesian population were more likely to have also been significant in an \gls{eas} population than the other super populations. The regions created from the clustering of significant \glspl{snp} also showed that within the super populations there was a higher commonality than between super populations. This would suggest that, similar to @Coop2009, geographically similar populations are subjected to similar selective environments. The time to the most recent common ancestor of two populations also affects the detection of shared regions as the more time that has passed, the more recombination will have an opportunity to break down the haplotype.

```{r, eval = FALSE}
bind_rows(fwh_pos_pol_genes %>% filter(SYMBOL %in% genelists$genename) %>% select(contains('chrom'), SYMBOL, contains("sum"), contains('POL')) %>% mutate(stat = "FWH_pos", dir = 1),
flf_pos_pol_genes %>% filter(SYMBOL %in% genelists$genename) %>% select(contains('chrom'), SYMBOL, contains("sum"), contains('POL')) %>% mutate(stat = "FLF_pos", dir = 1),
td_pos_pol_genes %>% filter(SYMBOL %in% genelists$genename) %>% select(contains('chrom'), SYMBOL, contains("sum"), contains('POL')) %>% mutate(stat = "TD_pos", dir = 1),
ze_pos_pol_genes %>% filter(SYMBOL %in% genelists$genename) %>% select(contains('chrom'), SYMBOL, contains("sum"), contains('POL')) %>% mutate(stat = 'ZE_pos',dir = 1),
fwh_neg_pol_genes %>% filter(SYMBOL %in% genelists$genename) %>% select(contains('chrom'), SYMBOL, contains("sum"), contains('POL')) %>% mutate(stat = "FWH_neg", dir = -1),
flf_neg_pol_genes %>% filter(SYMBOL %in% genelists$genename) %>% select(contains('chrom'), SYMBOL, contains("sum"), contains('POL')) %>% mutate(stat = "FLF_neg", dir = -1),
td_neg_pol_genes %>% filter(SYMBOL %in% genelists$genename) %>% select(contains('chrom'), SYMBOL, contains("sum"), contains('POL')) %>% mutate(stat = "TD_neg", dir = -1),
ze_neg_pol_genes %>% filter(SYMBOL %in% genelists$genename) %>% select(contains('chrom'), SYMBOL, contains("sum"), contains('POL')) %>% mutate(stat = 'ZE_neg', dir = -1)
) %>% group_by(SYMBOL, stat) %>% summarise(mean_pol = mean(POLsum) %>% round(2), mean_all = mean(ALLsum) %>% round(2), mean_dir = mean(dir) %>% round(2), n = n()) %>% View()
```


<!-- TD ~ low:med freq variants -->
<!-- yes 1st percentile did, but sharing was limited to eas -->
<!-- 99th percentile did not have any windows -->

<!-- fwh ~ high derived:med freq -->
<!-- 1st percentile: limited sharing mostly to either only poly pops, or poly plus eas -->
<!-- 99th percentile: shroom3 very shared. several examples of multiple populations being in common, although eas was less likely to be in those genes -->

<!-- flf ~ singletons -->
<!-- 1st percentile: differing degrees of sharing -->
<!-- 99th percentile: see abo, hla-b, and shroom again. quite a bit of sharing across the populations -->

<!-- ze high:low freq -->
<!-- 1st percentile: some examples spanned many populations, some were just the pairs -->
<!-- 99th percentile: -->









<!-- - The introduction sets the framework that geographically similar populations are in similar environments and therefore exposed to similar selective pressures. -->
<!-- - What we see is there is a higher burden of metabolic disease in Polynesians and not in other populations. Could it be that the selective pressure occurred after the migratory events? -->
<!-- - looking at the the current populations might give us an idea of if there is selective history and if so when a particular event might have taken place -->
<!-- i.e. do we see similar frequencies of disease alleles -->
<!-- - From the sfs results we see that within geographically similar regions they have similar genetic results -->
<!-- - we also get similar results regions with haplotypic selection -->
<!-- - somewhat indicated by the use of the clustering using disease genes -->
<!-- - are there shared haplotypes? -->
<!-- - because we know the migration history, and we also have shown that the modern day populations still show these groupings that correspond to temporal and spatial geography it can help us infer when a particular region of the genome may have had a selective event. -->
<!-- - why does it matter when something happened though? -->
<!--     - distigmatisation -->
<!-- - why? by comparing populations it might give an idea of particular regions that might explain differences in prevalence and or severity/likelihood of developing metabolic disease -->
<!-- obesity what was found? -->
<!-- t2d what was found? -->
<!-- kd what was found? -->
<!-- met syn what was found? -->
<!-- Anything that did have a signal, was it shared with other populations? -->

<!-- was there similarity in the polys? -->
<!-- any differences in the polys? -->



### Limitations {#chap4limit}

<!-- > Detection of selection on polygenic traits... -->

The technical limitations of this dataset based on the markers from the CoreExome \gls{snp} array raised in Section \@ref(chap3limit) apply to this analysis too. Particular to the analysis in this chapter is how the clusters were determined. The \texttt{cutree} method implemented in R [@RCoreTeam2017] that was used to slice the dendrograms, creating the groups used to assess the proportions and exclusivity of the clustering was relatively crude and did not always agree with the clusters that were 'visually apparent' in the dendrogram, although 'visually apparent' could be partially attributed to confirmation bias. Hierarchical clustering was chosen as the clustering algorithm over K-means because the branching nature could be thought of as a metaphor for the population migration histories. It is also possible that different linkage and distance criteria for the clustering algorithm may have produced different clusters, however, any differences in clusters would likely be minor (data not shown).

<!-- give this a test and might then might be able to "data not shown" -->

The \texttt{cutree} method also required that K clusters be returned regardless of what the tree actually looked like, and K < n. The use of K = 6 was chosen due to there being six super populations, however, \gls{clm}, \gls{pur}, \gls{mxl}, and \gls{pel} being classed as a single super population was not reflected in the clustering of the chromosome-wide F~ST~. Many of the selection tests split \gls{amr} into 2 groups, \gls{clm}/\gls{pur} and \gls{msl}/\gls{pel}. This pairing is consistent with what had previously been found from similarity of admixture and F~ST~ [@Gravel2013; @1KGP2015snp]. These \gls{amr} populations had also been identified as being admixed and so it was interesting that for the upper tail of \gls{flf} they were clustered with the \gls{nzm} population, which is also known to be admixed, mostly with New Zealand European [@Hollis-Moffatt2012; @Gravel2013]. Because these \gls{amr} populations were still included, and the requirement for six clusters, this affected the exclusivity measurement of clusters. The proportion measurement was also affected because sometimes a super population was spread across multiple but exclusive clusters. This was the case for the Polynesian populations and the 99^th^ percentile of \gls{flf} where the Eastern Polynesian populations were an exclusive cluster, and the Western Polynesian populations were an exclusive cluster. This meant more clusters were assigned to Polynesian populations than should have been (under the K = 6 assumption there would be one cluster per super population), and for six clusters to remain, the \gls{sas} populations ended up combined with the \gls{eur} populations, based on the distance and linkage criteria used for the clustering.

<!-- Were the regions driving similarity of amr and nzm shared with the europeans? -->

The only statistic that consistently did not cluster the Polynesian populations was \gls{flf}. \Gls{flf} is based on singletons, and for the clustering, the 1^st^ percentile windows was a sparse matrix of windows that were in common between populations. The sparsity of the matrix influenced how well the clustering performed due to the lack of windows that were in common between any pair of populations. At the extremes of similarity (e.g., very similar or very dissimilar), the clusters are disproportionately influenced by only a few windows, which means a large amount of information was not being used for the differentiation of populations. As the lower tail represents an excess of singletons this is likely to have been impacted by using \gls{snp} array data as there was a very strong bias against singletons during the quality control protocol [@Guo2014]. In this situation, because the data has reduced singletons, the distribution for \gls{flf} was shifted to the right, with the upper tail representing a deficit of singletons [@Fu1993]. This shift now means there is an under representation of regions in the lower tail.

As reported in section \@ref(fdrresults), only the \gls{fdr} for the Polynesian populations was below 10% for the 1^st^ percentiles of \gls{td} and \gls{fwh} for the frequency spectrum-based statistics. This means that while the extremes would be considered the most likely ends of the distribution that would contain regions of the genome that had been selected, it cannot be claimed with confidence. Therefore the mechanism through which they became the extremes of the distribution is possibly through selection, however, other neutrality mechanisms are still valid. While it is true that the extremes may include selected loci, it is also true that because of the use of a distribution proportion threshold, there will always be windows or \glspl{snp} that would be identified as possibly being under selection [@Teshima2006]. This means that while there were similarities in the extremes of the distributions for the frequency-based statistics, they are not necessarily under selection, and based on the \gls{fdr} calculated from the permutations, they are likely to be false positives.

A limitation was also some of the lack of 'internal replication', for example, the Eastern and Western Polynesian pairs, where a region would only meet the threshold for significance in a single population and not in the paired population. This could also just be a side effect of discretising a continuous distribution, and especially for values sitting close to the threshold it is easy to imagine a situation where in one population it was significant and the other it was not. An alternative approach could have been to rank all of the windows based on statistic value and determine differences and similarities in rank order between populations.

The rightwards shift in the distributions of \gls{td}, \gls{flf}, and \gls{ze} largely is influenced by the low frequency variants and singletons being under-represented. This effect is due to the ascertainment bias on the \gls{snp} arrays, where there is an excess of common alleles, when compared with sequence data [@Ramirez-Soriano2009]. This shift was more noticeable in the \gls{afr} populations with the 1^st^ percentile being above zero for \gls{td} and \gls{flf}. This meant that the windows that were being compared were not always the going to be inferring the same type of selection or population demographic events as those that were from below zero. The rightward shift won't have affected all genes in a uniform manner - the impact will be different for different genes, based on the frequency of the markers in them. This means that it can't be assumed that the genes in the 1^st^ percentile for the \gls{afr} populations are the equivalent in terms of ranking by statistic value, to those in the other populations. Nor can it be assumed that the distribution that was observed using the markers from the CoreExome would be the same that would be observed from using sequencing data.

Grouping based on the clustering of the significant results in \gls{ihs} or \gls{nsl} may falsely link \glspl{snp} into a 'region of significance'. The 200 kb limit was the maximum distance that selscan by default would allow the calculation of the extended haplotype heterozygosity to extend across a gap of information before terminating. The calculation of region sharing also does not take into account the directionality of the selection, for example, one population might have favoured the ancestral allele, whereas a second favoured the derived allele, the region sharing calculation treats these the same and would consider them a shared region.


<!-- - the fact it was a chip -> ascertainment bias -->
<!-- - the snps on the core exome might be less 'afr centric' -->
<!--      again an ascertainment bias but also would lead to the skewwing of the distributions seen for afr populations -->
<!-- - FDR was high for most of the statistics - therefore can't really claim selection -->
<!-- - clustering  -->
<!-- - pop specific shift in distributions -->
<!--     - can't compare the 1st percentile for some stats  -->
<!-- - cuttree method is crude -->
<!-- - only used hierarchical clustering, other methods may give different results -->
<!--     - selection of K might not be appropriate -->
<!-- - sparsity of the data -->
<!-- - metrics for assessing correct grouping might not be the best -->
<!-- - essentially just the capturing sfs -> geographically similar pops have similar allele freq -->
<!-- - FDR for the extremes suggests that the extremes aren't that different from a null -->

<!-- > The recent work with using the full 1kgp biallelic sites suggests that the distributions are just straight out shifted, although there are some values that are quite different - actually minimal overlap in the regions that were represented in the extremes -->

### Conclusions

It can be concluded that the extremes of the distributions for the intra-population selection and neutrality statistics can be used to group populations into their super populations, however, overall this is largely due to the effects of genetic drift (population migration, expansion, bottlenecks, admixture, etc.), than specifically selected regions of the genome. There is similarity in the regions in the extremes of the selection and neutrality statistics for the populations of a super population, but limited commonality between super populations. Random regions of the genome have varying levels of commonality between populations but is lower than the regions with extreme selection and neutrality statistic values.

Genes that are associated with disease perform well at grouping populations into their super population groups with hierarchical clustering. However, this is not specific to metabolic disease associated genes, but applies to random sets of genes, with an increase in number of genes improving the clustering.

In all combinations of clustering of selection and neutrality statistics used, the East Asian populations were the most similar to the Polynesian populations. There was additional evidence of the differences and similarities of the Polynesian populations, between the Eastern and Western Polynesian's, which reflected the migration history. And there was evidence of regions having a shared ancestry between these populations, along with evidence of a common ancestral population with the modern East Asian populations.






<!-- Gautier2013 in the intro talks about clustering -->


<!-- ### NOTES -->

<!-- Might be able to at least approach this as a confirm the ep/wp similarity and also the similarity of east asian to poly
look at skoglund 2016 and the new paper by Hudjashov2018 to work out how to spin it

-->





<!-- \gls{td} and \gls{ze}, both had either, complete or near complete groupings for a few super populations, especially \gls{afr} and \gls{eur}. These two statistics involve the low-frequency variants in their calculation, and low-frequency markers, again have a bias against them, as they tend to be under-represented in the \gls{snp} chips [\@ref]. There is also a bias against them in the algorithm and steps for genotype calling on the CoreExome \gls{snp} chip. This again means that there is likely an under-representation of low-frequency variants and thus skews the statistics that utilise them. -->


<!-- \Gls{fwh} was the only intra-population selection statistic that re-assembled the nearly all of the super populations in their entirety. \gls{fwh} is based on the ratio of high-frequency derived alleles to intermediate frequency derived alleles. The ancestral allele does not always the equate to the reference, and nor does it always equate to the major allele. This means that the allele frequencies for the markers that were on the \gls{snp} chip that were low frequency have a chance of being high frequency when converted to the ancestral/derived form. -->


<!-- Calculating the statistic as a summary for an entire chromosome doesn't inform much about the state of selection in a population, as selection has acts on targeted regions. What it does is give an indication of the overall frequency distributions of the population and if there is a particular abundance in a particular frequency 'bin' -->


<!-- #### Clustering on the extremes -->

<!-- > What I want to get across in this subsection is that the extremes did group the super populations and that there was 'blocks' of regions that within a super population were common but there was limited sharing of windows between the populations - ie each super population had it's own defining set of windows. The random windows didn't group the populations and this indicates that there is an underlying 'structure to the windows that were in the extremes. -->
<!-- I also want to show that there were degrees of overlap, suggesting some windows had a shared history. -->


<!-- In order to allow for the more targeted nature of selective events to come through, the extremes of the intra-population selection and neutrality statistics were clustered to see if there were regions in that were in common between populations of a super population and also between super populations. -->

<!-- > should calculate sfs on 1000 genomes full data and subset down to matching windows -->


<!-- Hudjashov2017 had 10 tongans, 10 samoans, Hudjashov,2018  -->

<!-- Skoglund2016 -->



<!-- mean unique windows -->

<!-- stat | lower | upper -->
<!-- ---|---|--- -->
<!-- td | 202.4 | 274.5 -->
<!-- fwh | 135.6| 226.4 -->
<!-- fld | 633.7 | 29.2 -->
<!-- flf | 467.2 | 120.6 -->
<!-- ze | 119.5 | 113.5 -->


<!-- bring in F~ST~ windows for the disease genes -->


<!-- Bits that I need to add -->

<!-- - look at the disease genes and see if any are present in either extreme for each of the sfs stats -->
<!-- - of the genes that are found in Pol, how many other populations is this seen with? -->
<!-- - the genes that do share, what are they associated with? and what do the prevalences of those populations look like? -->


<!-- <!--windows -->
<!-- correct: -->
<!-- td: both split amr, clustered others -->
<!-- fwh: upper split amr, lowre did not, all pops clustered -->
<!-- fld: lower split amr, upper only clustered afr, eas, and sas entirely -->
<!-- flf: lower split amr, nzm/mxl/pel. upper split amr clustered others -->
<!-- ze: both split amr, clustered others -->

<!-- It is difficult to apply a biological interpretation onto what  -->

<!--  The lower extreme however was a mix of negative and positive values, with the positive values mostly from the \gls{afr} populations. This mix means that the windows from one population do not necessarily represent the same selective or population demographic event component and therefore are not directly comparable.  -->






<!-- <!-- - Tennessen2011 might be relevant to above -->
<!-- - Bierne2013 includes info on fst outliers -->
<!-- - Verdu2014 might be useful for admixture/clustering -->
<!-- - Pickrell2012 has a model that they used to create a tree -->


<!-- What was observed, was the populations being clustered into their super populations. This would indicate that a) there was an ancient selective event or geographically limited, prior to the populations diverging and subsequent population specific recombination [@Coop2009; @Hermisson2017]. Or b) there was no selective event and the change in allele frequencies due to drift after population divergences was enough to distinguish the populations from each other [\@ref]. There were only a few loci from the gene lists that had met the thresholds for significance to indicate possible selection, however, the \gls{fdr} results from the random permutations (section \@ref(fdrresults)) suggest these are still possible under the null distribution. This would indicate that the most likely explanation is that drift or population demography events (migration, bottleneck, etc.) were responsible for the differences in allele frequency. This can be seen in the fact that the populations from each super population had similar distributions of the selection statistics but these distributions where shifted compared to the other super populations. It was for this reason that the gene list clustering using the actual values needed to be centred using the population median. The random selections of genes add evidence of the clustering being non-specific to pathways or disease gene lists. It is likely that it comes down to sufficient data points that the algorithm is able to discriminate between the populations and overall geographically similar populations have a similar frequency spectrum [@Coop2009].  -->



<!-- urate has a relationship with dementia see @Latourte2017 for more info -->
<!-- elevated SUA is neuro protective for PD  [@Shen2013pd] and AD [@Chen2014], although the AD meta found no sig diff but suggestice that lower  -->


<!-- Gautier2013 has a section (figure6) about getting the tree of ceu/jpt/bia/yri and also about inferring the branch lengths -->




<!-- The choice of out-group for \gls{fwh} can also have an effect as it is used to root the tree <!-- is it or is it used to work out what the frequency is? -->
<!-- In this case the out-group was a single sample created that was homozygote for the ancestral allele -->



<!-- The haplotypic clustering results were similar to @pickrell2009signals -->



<!--
- **what does this answer??**
- using the chromosome wide results did not always return the super clusters.
- often see admixed amr pops grouped with nzm/cim
- clustering using the gout/urate/bmi/t2d/metsyn genes...
- are the genes found more than you would expect by chance?
- cutree method is rather crude
- what are the strengths/weaknesses of each stat
- Fu and Li's D doesn't cluster well
- when clustering on actual values how much does it matter a population has shifted overall distribution? - centre on the median to help fix this



Clustering on the extreme values returned groupings that were consistent with the super population groups.

The clustering of the disease related genes gave the groupings that, with the exception of \gls{fld} were consistent with the super populations.

Using the extremes of the distributions and the gene lists for gout and co-morbidities had clustering that was consistent with the super population groups and largely matched. Clustering on chromosome wide F~ST~ did give the super population groupings and also split the \gls{amr} populations. This was not the case for \gls{td}, \gls{fwh}, \gls{fld}, \gls{flf}, and \gls{ze} using entire chromosomes, which did not produce groups that were consistent with the super population groups. \Gls{fwh} and \gls{ze} chromosome wide did the best at grouping the populations. 

Diseases with a late on-set were chosen as these should have little to no selection acting on them due to affecting the individuals post reproduction age. Clustering using the genes associated with these diseases had very similar groupings to the genes associated with gout and 



clustering on actual values because the stats are based on the underlying frequency distribution are likely to reflect this distribution and not the actual state of selection - therefore I would assume that given any list of genes sufficiently long (say >10), that the population clusters would appear. If the selection event occurred post out of Africa, then there would be some expectation that the clustering would. 

effect size of allele vs allele frequency: Park2011
Human Demographic History Impacts Genetic Risk Prediction across Diverse Populations: Martin2017
Understanding rare and common diseases in the context of human evolution: Quintana-Murci2016 
A Genetic Atlas of Human Admixture History - Hellenthal2014
Mutation Rate Variation is a Primary Determinant of the Distribution of Allele Frequencies in Humans - Harpak2016
Demographic History of Oceania Inferred from Genome-wide Data - Wollstein2010

this is exploratory, not predictive or creating a model, but is evidence that the Polynesian populations are similar most to each other
<!-- what ends of the distribution represent the different types (positive, balancing, negative selection) -->




```{r, results = 'asis', eval = FALSE}
bind_rows(list(
  
)) %>% mutate_if(., is.numeric, funs(sprintf('%.2f',.)))%>%  select(stat, "Gene List" = Disease, everything()) %>% data.frame %>%  pander::pandoc.table(., caption = 'Proportion')

```

```{r, eval = FALSE}
bind_rows(list(
  bind_rows(lapply(names(gout[[2]]), function(x){gout[[2]][[x]]$super_prop %>% spread(super, max_prop)%>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1)) })) %>% mutate(Disease = 'Urate/Gout'),

bind_rows(lapply(names(t2d[[2]]), function(x){t2d[[2]][[x]]$super_prop %>% spread(super, max_prop)%>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1)) })) %>% mutate(Disease = '\\gls{t2d}'),

bind_rows(lapply(names(obesity[[2]]), function(x){obesity[[2]][[x]]$super_prop %>% spread(super, max_prop)%>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1)) })) %>% mutate(Disease = 'Obesity'),

bind_rows(lapply(names(metsyn[[2]]), function(x){metsyn[[2]][[x]]$super_prop %>% spread(super, max_prop) %>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1)) })) %>% mutate(Disease = 'Metabolic Syndrome')
)) %>% mutate_if(., is.numeric, funs(sprintf('%.2f',.)))%>%  select(stat, "Gene List" = Disease, everything()) %>% gather("pop",'prop',3:NCOL(.)) %>%  left_join(., bind_rows(list(
  bind_rows(lapply(names(gout[[2]]), function(x){gout[[2]][[x]]$exclusivity %>% data.frame%>% mutate(pop = row.names(.)) %>% gather("cluster","prop",1:6) %>% filter(prop > 0) %>% group_by(pop) %>% summarise(mean_prop = mean(prop)) %>% spread(pop, mean_prop) %>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1))})) %>% mutate(Disease = 'Urate/Gout'),

bind_rows(lapply(names(t2d[[2]]), function(x){t2d[[2]][[x]]$exclusivity %>% data.frame%>% mutate(pop = row.names(.)) %>% gather("cluster","prop",1:6) %>% filter(prop > 0) %>% group_by(pop) %>% summarise(mean_prop = mean(prop)) %>% spread(pop, mean_prop) %>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1))})) %>% mutate(Disease = '\\gls{t2d}'),

bind_rows(lapply(names(obesity[[2]]), function(x){obesity[[2]][[x]]$exclusivity %>% data.frame%>% mutate(pop = row.names(.)) %>% gather("cluster","prop",1:6) %>% filter(prop > 0) %>% group_by(pop) %>% summarise(mean_prop = mean(prop)) %>% spread(pop, mean_prop) %>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1))})) %>% mutate(Disease = 'Obesity'),

bind_rows(lapply(names(metsyn[[2]]), function(x){metsyn[[2]][[x]]$exclusivity %>% data.frame%>% mutate(pop = row.names(.)) %>% gather("cluster","prop",1:6) %>% filter(prop > 0) %>% group_by(pop) %>% summarise(mean_prop = mean(prop)) %>% spread(pop, mean_prop) %>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1))})) %>% mutate(Disease = 'Metabolic Syndrome')
)) %>% mutate_if(., is.numeric, funs(sprintf('%.2f',.)))%>%  select(stat, "Gene List" = Disease, everything()) %>% gather("pop","excl",3:NCOL(.)), by = c('stat',"Gene List","pop")) %>% filter(excl > 0.99, prop > 0.99) %>% group_by(pop) %>% tally()
```

