# Clustering of Selection Statistics

```{r, include = FALSE, echo = FALSE}
library(tidyverse)
library(ggdendro)
source('scripts/multiplot.R')
```

<!--
key tasks to complete:
- update introduction WRT embedded comments
- df with summary stats about number of windows for each population
- df with distribution of treecuts proportions
- genes in the tails
-->

this is a test for the \gls{nz} acronymn feature to see when \gls{nz} becomes and acronymn \gls{t2d} is here as a test


In this chapter I will investigate the idea that populations that share ancestry will display similar signals of selection. To test this Idea I will be clustering the various selection statistics and then comparing the clusters to current population migratory history.

## Introduction

<!--
* introduce the out of africa history
* cover the pacific migrations
* Near versus Remote Oceania
* migration to NZ from pacific islands
-->

“Signatures of selection” in a population can be identified in regions of the genome that exhibit a reduction in genetic variability [@Smith1974;@Kaplan1989;@McVean2007]. This reduction in genetic variation can arise when the phenotype of a neutral benefit allele experiences a favourable change in environmental conditions or a new allele arises conveying a selective advantage [@Hermisson2005].  This results in an increased frequency of both the allele, and linked sites, within a population [@Smith1974]. Genome-wide scans for signatures of selection have shown geographically similar populations share similar signatures [@Coop2009; @pickrell2009signals]. Geography can inhibit migration leading to local adaptations affecting the allele frequency of local populations, leaving the non-local population allele frequency unaffected [@Coop2009]. New Zealand Polynesians are relatively geographically isolated, with a recent settlement history. 
<!-- need to update with Lisa's comments which were
- remote oceania is beyond solomon islands
- need a sentence to explain why we have Polynesians living in NZ -> ref the chapter about 1960's pacific immigrants to NZ
- explain NZ popln of Maori and Polynesian populations
- mention CIM settlement
- explain east/west split
-->

The history starts with migration out of Africa (50-100 kya) [@Nielsen2017], tracking up to the Levant and Arabian peninsula (? kya) <!-- same with this -->and splitting into South Asia, Indonesia, and Australia (50kya) [@Kivisild1999; @Quintana-Murci1999]. Concurrently the migration  to Europe was occurring (45 kya). From South Asia the migration continued with the peopling of East Asia (20kya) [@Groucutt2015]. <!--Lisa: The Pacific – or if you really are talking about Remote Oceania, you are talking about Lapita, which first appears in Near Oceania about 3300BP where they interact with the peoples who have been there for at least 40,000 years and then by 3,000 they are the first to establish settlement on the islands of New Caledonia, Vanuatu, Fiji, Samoa and Tonga. THEN around 1-1.2 kya we have the settlement of the rest of the Polynesian triangle, including the Cook Is avout 1,000 BP and New Zealand about 750 - 800 BP. --> The Pacific was settled in two events, the first was the settlement of Near Oceania (50 kya) followed by the settlement of Remote Oceania reaching Samoa and Tonga during the Lapita expansion (3 kya, [@Matisoo-Smith2015 ;  @Skoglund2016]) then the rest of the Polynesian triangle settled (1 – 1.2 kya, [@Wilmshurst2011]). The settlement of \gls{nz} by \gls{nz} Maori occurred 800 ya [@Duggan2014; @Matisoo-Smith2015].

<!--ref for pacific in NZ in 1960's [@MatisooSmith2012highway] 
The modern population of \gls{nz} Maori 

For largely ecomomnic reasons, during the 1950's and 60's there was a rural to urban migration of people in the Pacific. As part of this, many settled in New Zealand from Samoan, Tongan and the Cook Islands in search of work and now the populations in New Zealand largely outnumber the populations of their islands origin [@MatisooSmith2012highway]. Within \gls{nz}, the Polynesian populations are mostly focussed within the Auckland region [@Barcham2009]
-->

New Zealand Polynesians have inherent elevated serum urate levels and higher prevalence of gout [@Winnard2012; @Winnard2013a]. Polynesian populations share a common genetic ancestry with East Asia however there are Polynesian specific variants associated with increased risk to gout and elevated serum urate [@Phipps-Green2010; @Hollis-Moffatt2012]. Serum urate has been associated with metabolic disorders [@Choi2007b] and genes involved with complex diseases such as \gls{t2d}, gout, and other metabolic related disorders have shown evidence for selection [@Zhang2013a; @pickrell2009signals]. 

It is hypothesised that elevated serum urate may have undergone positive selection in Polynesians populations due to some of the beneficial properties [@gosling2014], such as its role as a powerful anti-oxidant [@Ames1981], or as an adjuvant for the innate immune system [@Opitz2009]. Serum urate has also been suggested to have a protective neurological effect [@Euser2009]. Characterisation of selection within Polynesian populations has been largely limited to population differentiation due to sample sizes [@Kimura2008; @Mallick2016].

Here we investigate genetic selection of serum urate in Polynesian populations and if there is a selection signature shared amongst Polynesian populations.






## Methods
Selection pipeline for ihs/nsl

Tajima's _D_, Fay and Wu's _H_, Fu and Li's _D_, Fu and Li's _F_, and Zeng's _E_ were all calulated using PopGenome v2.2.3 [@popgenome]. These statistics were calculated using a window size of 30 kb and a slide of 3 kb. For each selection test statistic, windows were ordered by test statistic value and the most extreme 1000 values were selected from each tail of the distribution. Hierarchical clustering was then performed on a binary matrix derived from the presence or absence of the windows across all populations. The distance measure used was euclidean distance and complete linkage was the linkage criteria.

\Gls{ihs} and \gls{nsl} were calculated and normalised using selscan v1.1.0b [@szpiech2014] as part of the selectionTools 1.1 [@cadzow2014] pipeline as described in Section \@ref(selectionTools). markers with an \gls{ihs} or \gls{nsl} that were determined to be significant using a significance threshold of 1% were clustered into genomic regions using the DBSCAN v? algorithm. The maximum distance was XXX kb from the cluster. A distance matrix was then created using the proportion of shared genomic regions between each population. Hierarchical clustering was then performed using euclidean distance as the distance measure, and complete linkage for the linkage criteria.

F~ST~


Disease associated genes were downloaded from the GWAS catalogue (https://www.ebi.ac.uk/gwas/, accessed 19 June 2017) and filtered for P < 5x10^-8^

## Results



### Frequency Spectrum




Clustering on the full chromosome results for all chromsomes from PopGenome
<!-- put each of the similar statistics together into a multiplot -->

#### Tajima's D

(ref:chrTDdendro) _Dendrogram for Tajima's D Clustering._ Hierarchical clustering of the populations based Tajima's D values for each entire chromosome. Clustering was performed using euclidean distance for the distance measure and complete linkage for the linkage criteria.

```{r chrTDdendro, fig.cap='(ref:chrTDdendro)'}
# generated in ~/Git_repos/Thesis/rnotebooks/CoreExome/coreExome_1kg_popgenome_clustering.Rmd
td_dendro <-readRDS('images/04_clustering/td_dendro.RDS')
td_dendro + ggtitle(label = "Tajima's D Clustering",subtitle = "Euclidean Distance with Complete Linkage")
rm(td_dendro)
```

- In Figure \@ref(fig:chrTDdendro) it can be seen there is no clear groupings of the super populations.

#### Fay and Wu's H

(ref:chrFWHdendro) _Fay and Wu's H Clustering._ Hierarchical clustering of the populations based Fay and Wu's H values for each entire chromosome. Clustering was performed using euclidean distance for the distance measure and complete linkage for the linkage criteria.

```{r chrFWHdendro, fig.cap='(ref:chrFWHdendro)'}
# generated in ~/Git_repos/Thesis/rnotebooks/CoreExome/coreExome_1kg_popgenome_clustering.Rmd
fwh_dendro <- readRDS('images/04_clustering/fwh_dendro.RDS')
fwh_dendro + ggtitle(label ="Fay and Wu's H Clustering",subtitle = "Euclidean Distance with Complete Linkage")
rm(fwh_dendro)
```

- In Figure \@ref(fig:chrFWHdendro) it can be seen that the there are clustering of the super populations although the American super populations are distributed across the other clusters.

#### Fu and Li's D

(ref:chrFLDdendro) _Fu and Li's D Clustering._ Hierarchical clustering of the populations based Fu and Li's D values for each entire chromosome. Clustering was performed using euclidean distance for the distance measure and complete linkage for the linkage criteria.

```{r chrFLDdendro, fig.cap='(ref:chrFLDdendro)'}
# generated in ~/Git_repos/Thesis/rnotebooks/CoreExome/coreExome_1kg_popgenome_clustering.Rmd
fld_dendro <- readRDS('images/04_clustering/fld_dendro.RDS')
fld_dendro + ggtitle(label ="Fu and Li's D Clustering",subtitle = "Euclidean Distance with Complete Linkage")
rm(fld_dendro)
```

#### Fu and Li's F

(ref:chrFLFdendro) _Fu and Li's F Clustering._ Hierarchical clustering of the populations based Fu and Li's F values for each entire chromosome. Clustering was performed using euclidean distance for the distance measure and complete linkage for the linkage criteria.

```{r chrFLFdendro, fig.cap='(ref:chrFLFdendro)'}
# generated in ~/Git_repos/Thesis/rnotebooks/CoreExome/coreExome_1kg_popgenome_clustering.Rmd
flf_dendro <- readRDS('images/04_clustering/flf_dendro.RDS')
flf_dendro + ggtitle(label ="Fu and Li's F Clustering",subtitle = "Euclidean Distance with Complete Linkage")
rm(flf_dendro)
```

#### Zeng E

(ref:chrZEdendro) _Zeng's E Clustering._ Hierarchical clustering of the populations based Zeng's E values for each entire chromosome. Clustering was performed using euclidean distance for the distance measure and complete linkage for the linkage criteria.

```{r chrZEdendro, fig.cap='(ref:chrZEdendro)'}
# generated in ~/Git_repos/Thesis/rnotebooks/CoreExome/coreExome_1kg_popgenome_clustering.Rmd
ze_dendro <- readRDS('images/04_clustering/ze_dendro.RDS')
ze_dendro + ggtitle(label ="Zeng's E Clustering",subtitle = "Euclidean Distance with Complete Linkage")
rm(ze_dendro)
```

```{r multichrplot, fig.cap='(ref:multichrplot)', echo = FALSE, fig.height=10}
td_dendro <-readRDS('images/04_clustering/td_dendro.RDS')
p1 <- td_dendro + ggtitle(label = "Tajima's D")

fwh_dendro <- readRDS('images/04_clustering/fwh_dendro.RDS')
p2 <- fwh_dendro + ggtitle(label ="Fay and Wu's H")

fld_dendro <- readRDS('images/04_clustering/fld_dendro.RDS')
p3 <- fld_dendro + ggtitle(label ="Fu and Li's D")

flf_dendro <- readRDS('images/04_clustering/flf_dendro.RDS')
p4 <- flf_dendro + ggtitle(label ="Fu and Li's F")

multiplot(p1, p2, p3, p4, cols = 1)
```

(ref:multichrplot) _Whole Chromosome Clustering._ Hierarchical clustering of the populations using selection test statistics for each entire chromosome. Clustering was performed using euclidean distance for the distance measure and complete linkage for the linkage criteria. Selection test statistic is in the label for each dendrogram.

### based on top X results

The most extreme 1% from each side of the empirical distribution was selected to use as part of a hierachical clustering to cluster the individual populations from the \gls{1kgp} the Polynesian populations. Euclidean distance was used as the distance measure for clusters and complete linkage was used as the linkage method to between clusters. For each population, windows were ranked in ascending order on test value, with ties being given the same rank, and the top 1000 ranks were selected for either tail. Each tail was then clustered separately.

The mean number of windows per population was XXX with a minimum of XXX and a maximum of XXX. To assess the statistical significance of the proportion of populations being assigned the same cluster, a series of XXX random draws was done to establish a null model for each selection test statistic.

Table \@ref(tab:windowTable) shows the distribution of number of windows for each super population used for each selection test statistic. Table \@(tab:totalewindowTable) show the total number of windows used for each.


```{r, results = 'asis', echo = FALSE}
mat_d_list <- readRDS('~/data/NZ_coreExome_1kgp/30kbWindow_intra/30kbwindows_mat_d_list-28-6-2017.RDS')
pander::pandoc.table(bind_rows(lapply(names(mat_d_list), function(x){mat_d_list[[x]] %>% select(-contains('chrom'), -posid) %>% gather(., pop, value, 1:NCOL(.)) %>% group_by(pop) %>% summarise(total_windows =  sum(value)) %>% mutate(d = x)})) %>% mutate(super = sapply(pop, function(x){strsplit(x, '_')[[1]][1]})) %>% select(-pop) %>% group_by(d, super) %>% summarise(min = min(total_windows), mean = mean(total_windows), max = max(total_windows)) %>% data.frame, caption = '(\\#tab:windowTable) Summary statistics for number of windows used in each custering analysis')
```

<!-- want to include the min/mean/max number of windows shared-->
```{r, results = 'asis'}
pander::pandoc.table(bind_rows(lapply(names(mat_d_list), function(x){mat_d_list[[x]] %>% select(-posid, -contains('chrom')) %>% summarise(total_windows = NROW(.), min = min(rowSums(.)), mean = mean(rowSums(.)), max = max(rowSums(.)) ) %>% mutate( stat = x)})) %>% select(stat, total_windows, min, mean, max) %>% data.frame(), caption = '(\\#tab:totalwindowTable) Number of windows'  )
```

```{r}
theme_heat <- theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5), legend.position = 'none', axis.title.y = element_blank(), axis.ticks.y = element_blank(), axis.text.y = element_blank(), axis.title.x = element_blank())
four_plot <- function(mat_d_name,statname){
  multiplot(
    ggdendrogram(hclust(dist(t(mat_d_list[[paste0(mat_d_name,'_neg')]][,-1:-4]), method="euclidean"), method = "complete")) + ggtitle("A.", paste(statname,"lower tail")),
    
    mat_d_list[[paste0(mat_d_name,'_neg')]] %>% select(-contains('chrom')) %>% gather("pop", "value",2:NCOL(.))%>%  ggplot(., aes(x = pop, y = posid)) + geom_tile(aes(fill = value)) +scale_fill_gradient2()+ theme_bw() + theme_heat + ggtitle("C."),
    
    ggdendrogram(hclust(dist(t(mat_d_list[[paste0(mat_d_name,'_pos')]][,-1:-4]), method="euclidean"), method = "complete")) + ggtitle("B.",paste(statname,"upper tail")),
    
    mat_d_list[[paste0(mat_d_name,'_pos')]] %>% select(-contains('chrom')) %>% gather("pop", "value",2:NCOL(.))%>%  ggplot(., aes(x = pop, y = posid)) + geom_tile(aes(fill = value)) +scale_fill_gradient2()+ theme_bw() + theme_heat + ggtitle("D.")
    , cols = 2) 
}


```


#### Tajima's D 
<!-- create a 4 panel multiplot with each dendrogram and heat map for both tails on the same panel -->

```{r tdTop1000, }
four_plot("td","Tajima's D")
```

- neg gave super pop clusters
- pos gave super pop clusters
- both split the AMR into 2
- number of windows shared by POL with others was X
- number of windows shared EP/WP was X
- number of unique windows was X for CIM, NZM, SAM, TON

#### Fay and Wu's H heatmap

```{r fwhTop1000}
four_plot("fwh","Fay and Wu's H")
```



- Fay and Wu's H put all of the super populations together both neg and pos (split AMR)
- there looks to be a lot of unique regions within the POL

#### Fu and Li's D
```{r fldTop1000}
four_plot("fld","Fu and Li's D")
```
- 10 fold difference in the number of windows
- pos had ~2900, neg had ~22000

#### Fu and Li's F
```{r flfTop1000}
four_plot("flf","Fu and Li's F")
```
- pos clustered super populations together, split AMR
- pos looks like there is a reasonable number of windows shared between pops (5188)
- neg looks pretty sparse 17044

#### Zeng's E Heatmap

```{r zeTop1000}
four_plot("ze","Zeng's E")
```

<!-- create df from the clustering results to be able to pull numbers out of for number of windows per population for each statistic, and for the proportion of pops being clustered -->


<!-- 
comparison about the random draws and the proportion of the same super clusters in each main cluster compared with the actual extremes
-->


### Haplotypic

Markers with a significant \gls{ihs} and \gls{nsl} after normalisation were grouped together by using the DBSCAN algorithm, where nearby \gls{snp} were assigned the same group id <!-- more detail here and include dbscan version -->.

For the haplotypic methods a similarity measure was created by calculating the 

1. iHS

2. nSL

### stats based on gene lists
<!-- might be a good idea to use a grid to put multiple plots into one -->
#### urate/gout

#### BMI

#### T2D

## Discussion and Conclusions

- using the chromosome wide results did not always return the super clusters.
- clustering using the gout/urate/bmi/t2d/metsyn genes...

One of the possible reasons why the genes give a similar cluster is that for the tests of selection that rely on the frequency spectrum, populations that are similar have a similar allele frequency to each other.

