# Clustering of Selection Statistics

<!--
the upper_sig_stats and  lower_sig_stats have a window size of 100 kb so therefore the genes that it says are in those windows might not apply to the 10 kb windows used for the clustering

-->


```{r, include = FALSE, echo = FALSE}
knitr::opts_chunk$set(echo = FALSE)
library(knitr)
library(kableExtra)
library(tidyverse)
library(ggdendro)
library(ggpubr)
library(phdFunctions)
source('scripts/multiplot.R')
source('scripts/04_clustering.R')
```

<!--
Include interpretation in each of the sub-sections

key tasks to complete:
- update introduction WRT embedded comments
- df with summary stats about number of windows for each population
- df with distribution of treecuts proportions
- genes in the tails
-->


This chapter investigates the idea that populations with shared ancestry will display similar signals of selection. To test this idea I will be clustering the various selection statistics and then comparing the clusters to current population migratory history.

## Introduction

<!--
* introduce the out of africa history
* cover the pacific migrations
* Near versus Remote Oceania
* migration to NZ from pacific islands
-->

“Signatures of selection” in a population can be identified in regions of the genome that exhibit a reduction in genetic variability [@Smith1974;@Kaplan1989;@McVean2007]. This reduction in genetic variation can arise when the phenotype of a neutral benefit allele experiences a favourable change in environmental conditions, or a new allele arises conveying a selective advantage [@Hermisson2005]. This results in an increased frequency of both the allele, and linked sites, within a population [@Smith1974]. Genome-wide scans for signatures of selection have shown that  geographically similar populations share similar signatures [@Coop2009; @pickrell2009signals]. Geography can inhibit migration ,leading to local adaptations affecting the allele frequency of local populations, leaving the non-local population allele frequency unaffected [@Coop2009]. New Zealand Polynesians are relatively geographically isolated, with a recent settlement history. 
<!-- need to update with Lisa's comments which were
- remote oceania is beyond solomon islands
- need a sentence to explain why we have Polynesians living in NZ -> ref the chapter about 1960's pacific immigrants to NZ
- explain NZ popln of Maori and Polynesian populations
- mention CIM settlement
- explain east/west split
-->

The history of Polynesian migration starts with migration out of Africa (50-100 kya, @Nielsen2017), tracking up to the Levant and Arabian peninsula (45-55 kya) and splitting into South Asia, Indonesia, and Australia (50kya,  @Kivisild1999; @Quintana-Murci1999). Concurrently the migration  to Europe was occurring (45 kya). From South Asia the migration continued with the peopling of East Asia (20kya, @Groucutt2015). <!--Lisa: The Pacific – or if you really are talking about Remote Oceania, you are talking about Lapita, which first appears in Near Oceania about 3300BP where they interact with the peoples who have been there for at least 40,000 years and then by 3,000 they are the first to establish settlement on the islands of New Caledonia, Vanuatu, Fiji, Samoa and Tonga. THEN around 1-1.2 kya we have the settlement of the rest of the Polynesian triangle, including the Cook Is avout 1,000 BP and New Zealand about 750 - 800 BP. --> The Pacific was settled in two events, the first was the settlement of Near Oceania (40 kya) followed by the settlement of Remote Oceania, reaching Western Polynesia (Samoa and Tonga) during the Lapita expansion (3 kya, @Matisoo-Smith2015; @Skoglund2016). Eastern Polynesia (including the Cook Islands) was then settled (1–1.2 kya, @Wilmshurst2011). The final region of the Polynesian triangle to be settled was that of New Zealand by \gls{nz} Maori, occurring 800 ya [@Duggan2014; @Matisoo-Smith2015].

<!--
need something about what consitutes east and west polynesia and why that matters - doone in above and following paras
-->

For largely economic reasons, during the 1950's and 60's there was a rural to urban migration of people in the Pacific. As part of this, many individuals and families settled in New Zealand from Samoa, Tonga, and the Cook Islands in search of work, and now the populations in New Zealand largely outnumber the populations of their islands of origin [@MatisooSmith2012highway]. Within \gls{nz}, the Polynesian populations are mostly focused within the Auckland region [@Barcham2009].

<!--ref for pacific in NZ in 1960's [@MatisooSmith2012highway] 
The modern population of \gls{nz} Maori - done

include paragraph about maori and pacific health in general in new zealand - done

@Hawley2015 @moh

-->

New Zealand M$\bar{a}$ori and Pacific Islanders in New Zealand over the past decade been reporting a decrease in overall health [@MinistryofHealth2016]. Fruit and vegetable intake, and physical activity has also been reported as decreasing, all the while there has been an increase in obesity where there was an adjusted rate ratio for obesity of 1.7 for  M$\bar{a}$ori vs non- M$\bar{a}$ori, and 2.4 for Pacific vs non-Pacific [@MinistryofHealth2016]. The rate of increase in \gls{bmi} for Pacific Islanders was five times that of the rest of the world, and fasting plasma glucose was three to four times higher over the 1980-2008 period [@Hawley2015]. The health disparities are captured in the New Zealand statistics where M$\bar{a}$ori have a mortality rate of 1.8 that of non-M$\bar{a}$ori with the rates of obesity and diabetes of particular note with the M$\bar{a}$ standarised rate of 40.6, five times higher than non-M$\bar{a}$ori of 8.1 [@MinistryofHealth2012].

New Zealand Polynesians have inherent elevated serum urate levels and higher prevalence of gout [@Winnard2012; @Winnard2013]. Polynesian populations share a common genetic ancestry with East Asia however, there are Polynesian specific variants associated with increased risk to gout and elevated serum urate [@Phipps-Green2010; @Hollis-Moffatt2012]. Serum urate has been associated with metabolic disorders [@Choi2007;@Choi2007b] and genes involved with complex diseases such as \gls{t2d}, gout, and other metabolic related disorders have shown evidence for selection [@Zhang2013a; @pickrell2009signals;@Hancock2008]. 

It is hypothesised that elevated serum urate may have undergone positive selection in Polynesians populations due to some of the beneficial properties [@Gosling2014], such as its role as a powerful anti-oxidant [@Ames1981], or as an adjuvant for the innate immune system [@Opitz2009]. Serum urate has also been suggested to have a protective neurological effect against dementia and provides improved cognitive function with age [@Euser2009]. Characterisation of selection within Polynesian populations has been largely limited to population differentiation due to population sample sizes [@Kimura2008; @Mallick2016].

<!-- tie in urate/gout/obesity/t2d so that it makes sense to use those gene lists -->

Here we investigate genetic selection of serum urate and related co-morbidities in Polynesian populations and attempt to determine whether there is a selection signature shared among Polynesian populations.

### Objectives

 - Determine if neutrality and selection tests can be used to group populations.
 - Investigate genetic loci associated with gout, urate, \gls{t2d}, obesity, kidney disease, and metabolic syndrome and their potential shared selective histories.

## Methods


<!--
Important distinction is that this used the 1% threshold NOT the 1% and >/< 0 to select the windows to be used
-->

### Frequency Spectrum

Tajima's _D_, Fay and Wu's _H_, Fu and Li's _D_, Fu and Li's _F_, Zeng's _E_, and F~ST~ were all calculated using PopGenome v2.2.3 [@Pfeifer2014] for each entire chromosome. Excepting F~ST~, these statistics were also calculated as per section \@ref(popgenomeMethods) to generate non-overlapping windows of 10 kb. For each selection test statistic, windows were ordered by test statistic value, and the most extreme 1% of values were selected from each tail of the distribution. For the windowed  approach, hierarchical clustering was then performed on a binary matrix derived from the presence or absence of the windows across all populations. Hierarchical clustering using the value caluculated for each whole chromosome (i.e. treating the start and end of the chromosome as the window boundary) used the statistic value itself to create the distance matrix. The distance measure used was Euclidean distance and the linkage criteria was "complete linkage" for both sets of clustering. Accuracy of the clustering was assessed by descending the dendrogram until six groups could be made, and then calculating the maximum proportion of the individual populations of the same super population assigned to the same cluster. Exclusivity of each cluster was calculated using the proportion of populations from a given super population compared to the total number of individual populations assigned that cluster. For example, if a cluster consisted of eight populations, two from population A and six from population B, then the exclusivity of the cluster would be 0.25 for population A, and 0.75 for population B. F~ST~ was calculated pairwise for all populations for each chromosome.

### Extended Haplotype Homozygosity {#haplocluster}

\glsdesc{ihs} and \gls{nsl} were calculated and normalised using selscan v1.1.0b [@Szpiech2014] as part of the selectionTools 1.1 [@Cadzow2014] pipeline, as described in Section \@ref(selectionTools). Markers with an \gls{ihs} or \gls{nsl} value that was determined to be significant using a significance threshold of 1% were clustered into genomic regions using the DBSCAN package v1.1.1 [@dbscanref] in R where nearby \gls{snp}s were assigned the same group identifier. The search radius used was 200 kb and the minimum number of points for a cluster was one. Cluster regions were created by taking both the minimum and maximum position for each group identifier, population and chromosome. The distance matrix was the result of calculating (A $\cap$ B)/ A and (A $\cap$ B) / B, where A $\cap$ B was the sum of the intersection of clustered regions between population A and population B, and A and B was the sum of the clustered regions for population A and population B respectively. Hierarchical clustering was then performed using Euclidean distance as the distance measure, and complete linkage for the linkage criteria.


### Disease Associated Genes

Disease associated markers and their annotated genes were downloaded from the \gls{gwas} catalog^[https://www.ebi.ac.uk/gwas/, accessed 19 June 2017] [@MacArthur2017]  and filtered for association (chi-squared P < 5x10^-8^). Gene lists were then created for each of the following diseases or traits; urate and gout, \gls{t2d}, obesity, kidney disease, and metabolic syndrome by filtering the study disease trait on keywords (see section \@ref(diseaselist) for details). Table \@ref(tab:gwascatref) shows source disease trait and reference used to create categories of gene lists for gout/urate, obesity, \gls{t2d}, kidney disease, and metabolic syndrome. Selection test statistics values for each population were centred using the population median. The gene lists from these categories were used in Section \@ref(genelistcluster) for hierarchical clustering using the selection test statistic values for windows that intersected these gene regions. Complete linkage and Euclidean distance were used for the clustering.

(ref:gwascataccess) https://www.ebi.ac.uk/gwas/, accessed 19 June 2017


## Results

### Frequency Spectrum


#### Whole Chromosome Clustering

Clustering on the full chromosome results for all chromosomes from PopGenome (Figure \@ref(fig:multichrplot)) did not group the populations of each super population together, with the exception of \gls{fwh}, which grouped all super populations although split the \gls{amr} into two subgroups. \gls{td} grouped the \gls{eur} and most of the \gls{sas} super populations, and had smaller sub-super population groupings of two to three populations. \Gls{cim}, \gls{nzm}, and \gls{sam} were grouped with two \gls{amr} populations, \gls{pel} and \gls{mxl}, and this group was the least similar to the others. There was no clear pattern to the clustering order for both Fu and Li's _D_ and _F_, although \gls{flf} did have a few small groups of the same super population of no more than three individuals populations. The \gls{cim} and \gls{sam} populations were grouped together in both of statistics. \Gls{ze} clustered the \gls{afr} populations together and this cluster was the least similar to the other populations. There was also a grouping of the \gls{eas} populations which had the Western Polynesian populations part of it.

<!--
- td had eur cluste, nzm-cim, most of a sas, but the other populations are in groups of 2-3, the pol group is quite disimilar to the rest
- fwh has the super populations but has AMR split in half. eas and pol are together and quite disimilar to the rest
- fld looks almost random, nzm-gbr, maybe admixed pops being grouped together
- flf has pockets fo 2-3 small groupings of the super populations, nzm-mxl-pel
- flf and fld have a very similar first split
- ze has afr cluster and is most disimilar. eas cluster has the wp integrated. nzc-clm-pur, eur cluster 
-->

(ref:multichrplot) _Whole Chromosome Clustering._ Hierarchical clustering of the populations using selection test statistics for each entire chromosome. Clustering was performed using Euclidean distance for the distance measure and complete linkage for the linkage criteria. Selection test statistic is in the label for each dendrogram. Coloured by super population.


```{r multichrplot, fig.cap='(ref:multichrplot)', echo = FALSE, fig.height=8}
multiplot(gene_dendros(gene_data = mat_d_list[['td_chr']]) + ggtitle("Tajima's D"),
          gene_dendros(gene_data = mat_d_list[['fwh_chr']]) + ggtitle("Fay and Wu's H"),
          gene_dendros(gene_data = mat_d_list[['fld_chr']]) + ggtitle("Fu and Li's D"),
          gene_dendros(gene_data = mat_d_list[['flf_chr']]) + ggtitle("Fu and Li's F"),
          gene_dendros(gene_data = mat_d_list[['ze_chr']]) + ggtitle("Zeng's E"),
   cols = 2)
```

\FloatBarrier

#### F~ST~

(ref:wholechrfst) _F~ST~_ Hierarchical clustering on chromosomal F~ST~ using Euclidean distance and complete linkage. Coloured by super population.

```{r wholechrfst, fig.cap="(ref:wholechrfst)"}
create_fst_dendro() + ggtitle(expression("Whole Chromosome F"["ST"]))
```


```{r}
fst <- readRDS('~/data/NZ_coreExome_1kgp/whole_chr_fst_popgenome.3-7-2017.RDS')

fst_mat <- bind_rows(lapply(fst, function(x){x %>% mutate(pop1 = as.character(pop1), pop2 = as.character(pop2))%>% select(-p1,-p2) %>% filter(!pop1 %in% c('WPN','EPN','NAD'), !pop2 %in% c('WPN','EPN','NAD')) %>% left_join(., panel %>% mutate(super_pop = paste(super_pop, pop, sep = '_')) %>% select("pop1" = pop, super_pop) %>% distinct(), by = "pop1" ) %>% select(-pop1) %>% spread(., "super_pop", fst )}))

fst_means <- fst_mat %>% filter(pop2 %in% c('CIM','NZM','SAM','TON')) %>% left_join(., panel %>% mutate(pop2 = pop, p2 = paste0(super_pop,"_",pop)) %>% select(pop2,p2) %>% distinct(), by = "pop2" ) %>% mutate(pop2 = p2) %>% select(pop2, 1:(NCOL(.)-1)) %>% gather("pop1","fst", 3:NCOL(.)) %>%   mutate(sup1 = sapply(pop1, function(x){strsplit(x, '_')[[1]][1]}), sup2 = sapply(pop2, function(x){strsplit(x, '_')[[1]][1]}) ) %>% filter(sup1 != sup2) %>%  group_by(sup2, sup1) %>% summarise(mean = mean(fst) %>% round(., digits = 3)) %>% ungroup()
```


Hierarchical clustering performed on pairwise F~ST~ for the whole chromosomes between each individual population grouped all of the super populations, except \gls{amr} into their correct groups (Figure \@ref(fig:wholechrfst)). The \gls{afr} populations were most similar to each other but least similar to the rest of the populations. The Polynesian populations were grouped together and were sub grouped into East and West Polynesian. The largest differentiation was between \gls{cim} and \gls{esn} (mean F~ST~ = 0.203), whereas the smallest differentiation was between \gls{ceu} and \gls{nzc} (mean F~ST~ = 3.42x10^-4^). The mean F~ST~ for \gls{pol} and \gls{afr} was `r fst_means %>% filter(sup2 == "POL", sup1 == "AFR") %>% select(mean) %>% sprintf('%.3f',.)`; \gls{pol} and \gls{amr} was `r fst_means %>% filter(sup2 == "POL", sup1 == "AMR") %>% select(mean) %>% sprintf('%.3f',.)`; \gls{pol} and \gls{eas} was `r fst_means %>% filter(sup2 == "POL", sup1 == "EAS") %>% select(mean) %>% sprintf('%.3f',.)`; \gls{pol} and \gls{eur} was `r fst_means %>% filter(sup2 == "POL", sup1 == "EUR") %>% select(mean) %>% sprintf('%.3f',.)`; \gls{pol} and \gls{sas} was `r fst_means %>% filter(sup2 == "POL", sup1 == "SAS") %>% select(mean) %>% sprintf('%.3f',.)`. This shows that the \gls{pol} populations were least differentiated from \gls{eas} populations and most from the \gls{afr} populations and is consistent with the migration history.  

<!--
some stats about the largest and smallest fsts and focus on the Polynesians
-->


### Hierarchical Clustering of Distribution Extremes

Hierachical clustering of the distribution extremes for each population was done to investigate if there was a similarity between populations and the genetic regions that had the most extreme scores. For the frequency spectrum methods, the 1st and 99th percentile of windows (i.e. distribution tails) by statistic value were selected and a binary matrix of window sharing between populations was used for the clustering. For the haplotypic methods the proportion of regions that overlapped between populations was used. The groups created by the clustering were then assessed for the number of populations from a super population that were assigned the same cluster. Having all populations correctly grouped into their super population was only completely achieved by clustering using the lower tail of \gls{fwh}. If an allowance is made for the two \gls{amr} populations to be grouped with the \gls{eur} populations, as in the chromosomal F~ST~ clustering, then both haplotypic methods and six tails of the frequency spectrum based methods clustered populations into their respective super populations.  

There were differing numbers of windows used for each population, ranging from a minimum of `r create_windowTable() %>% summarise(min(min))` to a maximum of `r create_windowTable() %>% summarise(max(max))`. The mean number of windows per population was `r create_windowTable() %>% summarise(round(mean(mean), digits = 1))` (SD = `r create_windowTable() %>% summarise(round(sd(mean), digits = 1))`). Table \@ref(tab:proportion) and Table \@ref(tab:exclusivity) show the proportion and exclusivity of the clusters respectively, for the clustering of populations by cutting the dendrograms to create 6 groups. Clustering from the windows of the extreme tails of the selection statistic distributions had a higher number of the individual populations clustered into their super populations than the chromosome-wide selection statistics and the clusters were also more exclusive to the super populations. This shows that the aggregate frequency spectrum based selection and neutrality statistics don't capture the population differentiation but that using the extremes for each population does replicate the population differentiation shown with F~ST~.


```{r tailwindowSummary, echo = FALSE}
# create table dataset that can be referenced in text
tab <- create_windowTable() %>% mutate(statname = sapply(d, function(x){paste0("\\gls{",strsplit(x, "_")[[1]][1], "}") }),  tail = sapply(d, function(x){strsplit(x, "_")[[1]][2]})) %>% select(statname, tail, super, min,mean,max) %>% mutate(tail = ifelse(tail == 'neg', "Lower","Upper"))
colnames(tab) <- c("Statistic Name","Tail", "Super Population", "Min","Mean","Max")
```


```{r totalwindowTable, results = 'asis', echo = FALSE}
# actual table for the number of windows 
create_windowSummaryTable() %>% mutate(Tail = ifelse(sapply(stat, function(x){strsplit(x, '_')[[1]][2]}) == 'neg', "Lower", "Upper" ), `Statistic Name` = paste0('\\gls{',sapply(stat, function(x){strsplit(x, '_')[[1]][1]}),"}") )%>% select(`Statistic Name`, Tail, `Total Windows`= total_windows, Min = min, Median = median, Max = max, Mean = mean, `Std. Dev` = sd) %>% knitr::kable(., caption = '(\\#tab:totalwindowTable) Number of windows in the distribution tails and the number of populations that shared a given window from the distribution tail. ', booktabs =TRUE, escape = FALSE  )
```

<!--
describe number of total number of windows
polynesian specific windows
ep specific windows
wp specific windows
cim/nzm/sam/ton specific windows

proportions
exclusivity
dendrograms

refer to figure
-->

#### Tajima's _D_ 


```{r tdTop1000, fig.cap='(ref:tdTop1000)',echo = FALSE, fig.height = 7, cache=TRUE}
#ggsave(plot = four_plot("td","Tajima's D"), filename = 'images/04_clustering/tdTop1000.png', device = 'png', height = 7, width = 7)
knitr::include_graphics('images/04_clustering/tdTop1000.png')
```

(ref:tdTop1000) _\gls{td} Clustering._ Hierarchical clustering of \gls{td} using the upper and lower 1% of the distribution. **A** and **B** Dendrogram representing the clusters of \gls{td} lower and upper tails respectfully using hierarchical clustering with Euclidean distance and complete linkage. Coloured by super population. **C** and **D** Plots representing the windows present in each population from the lower and upper tails respectfully. A blue line represents presence of a window, white represents the absence of a window in the 1% of the distribution for a population.

<!--
- neg gave super pop clusters
- pos gave super pop clusters
- both split the AMR into 2
- number of windows shared by POL with others was X
- number of windows shared EP/WP was X
- number of unique windows was X for CIM, NZM, SAM, TON
- genes involved?
-->
The clustering of the individual populations for the 1% most extreme values from each of the lower and upper tails of the \gls{td} distribution both showed the East/West split in the Polynesians (Figure \@ref(fig:tdTop1000)). But also showed that the Polynesian populations were most similar to each other. Both tails placed the Polynesian cluster closest to the \gls{eas} cluster. The main groupings are of the populations that were designated as the same super population. Although the \gls{amr} super population in both tails has been split into two. The mean number of unique windows per population was `r round(create_sums('td_pos') %>% filter(ALLsum == 1) %>% select(-contains('chrom'), -posid, -contains('sum')) %>% summarise_all(funs(sum)) %>% t() %>% mean(), digits = 1)` in the upper tail, compared with `r round( create_sums('td_neg') %>% filter(ALLsum == 1) %>% select(-contains('chrom'), -posid, -contains('sum')) %>% summarise_all(funs(sum)) %>% t() %>% mean(), digits = 1)` in the lower tail. The groupings for both the upper and lower tails exclusively grouped each of the \gls{afr}, \gls{sas} and \gls{pol} super populations. The \gls{amr} super population was split into an exclusive group of \gls{mxl} and \gls{pel}, and a group of \gls{clm} and \gls{pur} which was shared with the \gls{eur} super population. All clusters except the \gls{amr} - \gls{eur} cluster were each exclusive to a single super population.

In the lower tail, the number of windows that were unique to the Polynesian super population was `r create_sums('td_neg') %>% filter(POLsum == ALLsum) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)`, representing `r round(create_sums('td_neg') %>% filter(POLsum == ALLsum) %>% select(contains('chrom'), contains('sum')) %>% NROW(.) / mat_d_list[['td_neg']] %>% NROW(.) *100, digits = 1)`% of all windows in the lower tail that were used. This compared to super population unique window numbers of `r create_sums('td_neg') %>% filter(AFRsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)`, `r create_sums('td_neg') %>% filter(AMRsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)`, `r create_sums('td_neg') %>% filter(EASsum == ALLsum & ALLsum  > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)`, `r create_sums('td_neg') %>% filter(EURsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)`, `r create_sums('td_neg') %>% filter(SASsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` for the \gls{afr}, \gls{amr}, \gls{eas}, \gls{eur}, and \gls{sas} super populations respectively. Of the unique windows for Polynesian populations, `r create_sums('td_neg') %>% filter(POL_CIM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` were specific to \gls{cim}, `r create_sums('td_neg') %>% filter(POL_NZM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` were specific to \gls{nzm}, `r create_sums('td_neg') %>% filter(POL_SAM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` were specific to \gls{sam}, and `r create_sums('td_neg') %>% filter(POL_TON == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` were specific to \gls{ton}. The Eastern Polynesians had `r create_sums('td_neg') %>% filter((POL_CIM + POL_NZM) == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` unique windows with `r create_sums('td_neg') %>% filter((POL_CIM + POL_NZM) == ALLsum & ALLsum == 2) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` of those shared between \gls{cim} and \gls{nzm}, whereas the Western Polynesians had `r create_sums('td_neg') %>% filter((POL_SAM + POL_TON) == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows, with `r create_sums('td_neg') %>% filter((POL_SAM + POL_TON) == ALLsum & ALLsum == 2) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` shared between \gls{sam} and \gls{ton}. There were `r create_sums('td_neg') %>% filter(POLsum < ALLsum & POLsum == 4) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows where all the Polynesian populations shared a window with a non-Polynesian population. And `r create_sums('td_neg') %>% filter(POLsum < ALLsum & POLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows where at least 1 Polynesian population shared a window with a non-Polynesian population.




```{r, eval = FALSE}
# untested/bad assumption that all genes will be in here
create_sums('td_neg') %>% filter(POLsum == ALLsum & POLsum > 0) %>% left_join(., lower_sig_stats %>% mutate(chrom = as.numeric(chrom)) %>% select(contains('chrom'), SYMBOL) %>% distinct(), by = c('chrom', 'chrom_start', 'chrom_end')) %>% group_by(SYMBOL) %>% tally() %>% filter(!is.na(SYMBOL)) %>% NROW(.)

create_sums('td_neg') %>% filter(POL_CIM + POL_NZM == ALLsum & ALLsum == 2) %>% left_join(., lower_sig_stats %>% mutate(chrom = as.numeric(chrom)) %>% select(contains('chrom'), SYMBOL) %>% distinct(), by = c('chrom', 'chrom_start', 'chrom_end')) %>% group_by(SYMBOL) %>% tally() %>% filter(!is.na(SYMBOL)) %>% NROW(.)

create_sums('td_neg') %>% filter(POL_SAM + POL_TON == ALLsum & ALLsum == 2) %>% left_join(., lower_sig_stats %>% mutate(chrom = as.numeric(chrom)) %>% select(contains('chrom'), SYMBOL) %>% distinct(), by = c('chrom', 'chrom_start', 'chrom_end')) %>% group_by(SYMBOL) %>% tally() %>% filter(!is.na(SYMBOL)) %>% NROW(.)
```


```{r, eval = FALSE}
# untested/bad assumption that all genes will be in here
(td_neg_pol_genes <- create_sums('td_neg') %>% filter(POLsum == 4) %>% left_join(., lower_sig_stats %>% mutate(chrom = as.numeric(chrom)) %>% select(contains('chrom'), SYMBOL) %>% distinct(), by = c('chrom', 'chrom_start', 'chrom_end')) %>% group_by(SYMBOL) %>% tally())
```


The upper tail had a total of `r create_sums('td_pos') %>% filter(POLsum == ALLsum) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows, or `r round(create_sums('td_pos') %>% filter(POLsum == ALLsum) %>% select(contains('chrom'), contains('sum')) %>% NROW(.) / mat_d_list[['td_pos']] %>% NROW(.) *100, digits = 1)`% that were unique to the Polynesian populations. This compares to super population unique window numbers of `r create_sums('td_pos') %>% filter(AFRsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` (\gls{afr}), `r create_sums('td_pos') %>% filter(AMRsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` (\gls{amr}), `r create_sums('td_pos') %>% filter(EASsum == ALLsum & ALLsum  > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` (\gls{eas}), `r create_sums('td_pos') %>% filter(EURsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` (\gls{eur}), `r create_sums('td_pos') %>% filter(SASsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` (\gls{sas}). Eastern Polynesians contributed `r create_sums('td_pos') %>% filter((POL_CIM + POL_NZM & ALLsum > 0) == ALLsum) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` unique windows, with \gls{cim} having `r create_sums('td_pos') %>% filter(POL_CIM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` unique windows and \gls{nzm} having `r create_sums('td_pos') %>% filter(POL_NZM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows. The Western Polynesians had `r create_sums('td_pos') %>% filter((POL_SAM + POL_TON) == ALLsum & ALLsum > 0)  %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` unique windows. `r create_sums('td_pos') %>% filter(POL_SAM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` were from only \gls{sam} and `r create_sums('td_pos') %>% filter(POL_TON == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` from only \gls{ton}. There were `r create_sums('td_pos') %>% filter(POLsum < ALLsum & POLsum == 4) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows where all the Polynesian populations shared a window with a non-Polynesian population. And `r create_sums('td_pos') %>% filter(POLsum < ALLsum & POLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows where at least 1 Polynesian population shared a windows with a non-Polynesian population.


```{r, eval = FALSE}
# untested/bad assumption that all genes will be in here
create_sums('td_pos') %>% filter(POLsum == ALLsum & POLsum > 0) %>% left_join(., lower_sig_stats %>% mutate(chrom = as.numeric(chrom)) %>% select(contains('chrom'), SYMBOL) %>% distinct(), by = c('chrom', 'chrom_start', 'chrom_end')) %>% group_by(SYMBOL) %>% tally() %>% filter(!is.na(SYMBOL)) %>% NROW(.)

create_sums('td_pos') %>% filter(POL_CIM + POL_NZM == ALLsum & ALLsum == 2) %>% left_join(., lower_sig_stats %>% mutate(chrom = as.numeric(chrom)) %>% select(contains('chrom'), SYMBOL) %>% distinct(), by = c('chrom', 'chrom_start', 'chrom_end')) %>% group_by(SYMBOL) %>% tally() %>% filter(!is.na(SYMBOL)) %>% NROW(.)

create_sums('td_pos') %>% filter(POL_SAM + POL_TON == ALLsum & ALLsum == 2) %>% left_join(., lower_sig_stats %>% mutate(chrom = as.numeric(chrom)) %>% select(contains('chrom'), SYMBOL) %>% distinct(), by = c('chrom', 'chrom_start', 'chrom_end')) %>% group_by(SYMBOL) %>% tally() %>% filter(!is.na(SYMBOL)) %>% NROW(.)
```

```{r, eval = FALSE}
# untested/bad assumption that all genes will be in here
(td_pos_pol_genes <- create_sums('td_pos') %>% filter(POLsum == 4 ) %>% left_join(., lower_sig_stats %>% mutate(chrom = as.numeric(chrom)) %>% select(contains('chrom'), SYMBOL) %>% distinct(), by = c('chrom', 'chrom_start', 'chrom_end')) %>% group_by(SYMBOL) %>% tally())
```



#### Fay and Wu's _H_

(ref:fwhTop1000) _\gls{fwh} Clustering._ Hierarchical clustering of \gls{fwh} using the upper and lower tails of the distribution. **A** and **B** Dendrogram representing the clusters of \gls{fwh} lower and upper tails respectfully using hierarchical clustering with Euclidean distance and complete linkage. Coloured by super population. **C** and **D** Plots representing the windows present in each population from the lower and upper tails respectfully.

```{r fwhTop1000, fig.cap='(ref:fwhTop1000)', fig.height=7, cache=TRUE}
#ggsave(four_plot("fwh","Fay and Wu's H"), filename = 'images/04_clustering/fwhTop1000.png', device = 'png', height = 7, width = 7)
knitr::include_graphics('images/04_clustering/fwhTop1000.png')
```

<!--
- Fay and Wu's H put all of the super populations together both neg and pos (split AMR)
- there looks to be a lot of unique regions within the POL
- both neg and pos have the Pol cluster nearest to the EAS
-->

<!--
[x]describe number of total number of windows
[x]polynesian specific windows
[x]ep specific windows
[x]wp specific windows
[x]cim/nzm/sam/ton specific windows

[x]proportions
[x]exclusivity
[]dendrograms

[x] refer to figure
-->
The clustering of \gls{fwh} (Figure \@ref(fig:fwhTop1000)) had the main groupings the same as the defined super population groups, with the exception of the upper tail where the \gls{amr} super population was split in half. The Polynesian populations were clustered together in both the lower and upper tails for \gls{fwh}. In both tails the Polynesian cluster also had the East/West split within the group. The Polynesian population group was closest to the \gls{eas} populations with `r create_sums('fwh_pos') %>% filter(POLsum + EASsum <= ALLsum & POLsum > 0 & EASsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows from the upper tail and `r create_sums('fwh_neg') %>% filter(POLsum + EASsum <= ALLsum & POLsum > 0 & EASsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows from the lower tail in common between at least one Polynesian population and one \gls{eas} population. 

The clustering of the lower tail assigned all the individuals into their super populations and each of the groups were exclusive to their super population. There were a total of `r create_windowSummaryTable() %>% filter(stat == 'fwh_neg') %>% .[['total_windows']]` windows with `r create_sums('fwh_neg') %>% filter(POLsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows specific to Polynesian populations. This compared to `r create_sums('fwh_neg') %>% filter(AFRsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{afr}, `r create_sums('fwh_neg') %>% filter(AMRsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{amr}, `r create_sums('fwh_neg') %>% filter(EURsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{eur}, and `r create_sums('fwh_neg') %>% filter(EASsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{eas}. The Eastern Polynesian populations had `r create_sums('fwh_neg') %>% filter(POL_CIM + POL_NZM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows, of those, `r create_sums('fwh_neg') %>% filter(POL_CIM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` were specific to \gls{cim} and `r create_sums('fwh_neg') %>% filter(POL_NZM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` were specific to \gls{nzm}. The Western Polynesian populations had `r create_sums('fwh_neg') %>% filter(POL_SAM + POL_TON == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows, with `r create_sums('fwh_neg') %>% filter(POL_SAM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` specific to \gls{sam} and `r create_sums('fwh_neg') %>% filter(POL_TON == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` specific to \gls{ton}. The mean number of windows specific to an individual population was `r create_sums('fwh_neg') %>% filter(ALLsum == 1) %>% select(-contains('chrom'), -posid, -contains('sum')) %>% summarise_all(funs(sum)) %>% t() %>% mean() %>% round(., digits = 1)` in the lower tail. There were `r create_sums('fwh_neg') %>% filter(POLsum < ALLsum & POLsum == 4) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows with all Polynesians that also shared with a non-Polynesian population. And `r create_sums('fwh_neg') %>% filter(POLsum < ALLsum & POLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows with at least 1 Polynesian population that also shared with a non-Polynesian population.

In the upper tail of the \gls{fwh} distribution, the clustering assigned all of the individual populations into their super populations with only \gls{amr} being split. The \gls{clm} and \gls{pur} populations were clustered as part of the \gls{eur} group. Exclusivity was also high with all clusters except the \gls{eur} - \gls{amr} being exclusive to a single super population. Of the `r create_windowSummaryTable() %>% filter(stat == 'fwh_pos') %>% .[['total_windows']]` windows,  `r create_sums('fwh_pos') %>% filter(POLsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows were from Polynesian populations only. The Eastern Polynesian populations had `r create_sums('fwh_pos') %>% filter(POL_CIM + POL_NZM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` specific windows, of those, `r create_sums('fwh_pos') %>% filter(POL_CIM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` were unique to \gls{cim} and `r create_sums('fwh_pos') %>% filter(POL_NZM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` were unique to \gls{nzm}. The Western Polynesian populations had `r create_sums('fwh_pos') %>% filter(POL_SAM + POL_TON == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` unique windows, of these, `r create_sums('fwh_pos') %>% filter(POL_SAM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` were specific to \gls{sam} and `r create_sums('fwh_pos') %>% filter(POL_TON == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` specific to \gls{ton}. The mean number of unique windows for all populations was `r create_sums('fwh_pos') %>% filter(ALLsum == 1) %>% select(-contains('chrom'), -posid, -contains('sum')) %>% summarise_all(funs(sum)) %>% t() %>% mean() %>% round(., digits = 1)`. There were `r create_sums('fwh_pos') %>% filter(POLsum < ALLsum & POLsum == 4) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows with all Polynesians that also shared with a non-Polynesian population. And `r create_sums('fwh_pos') %>% filter(POLsum < ALLsum & POLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows with at least 1 Polynesian population that also shared with a non-Polynesian population.



#### Fu and Li's _D_

(ref:fldTop1000) _\gls{fld} Clustering._ Hierarchical clustering of \gls{fld} using the upper and lower tails of the distribution. **A** and **B** Dendrogram representing the clusters of \gls{fld} lower and upper tails respectfully using hierarchical clustering with Euclidean distance and complete linkage. Coloured by super population. **C** and **D** Plots representing the windows present in each population from the lower and upper tails respectfully. Colour represents a window being present.

```{r fldTop1000, fig.cap = '(ref:fldTop1000)', fig.height=7, cache=TRUE}
#ggsave(four_plot("fld","Fu and Li's D"), filename = 'images/04_clustering/fldTop1000.png', device = 'png', height = 7, width = 7)
knitr::include_graphics('images/04_clustering/fldTop1000.png')
```

<!--
- 6 fold difference in the number of windows
- pos had ~3300, neg had ~19000
- pol is the least similar to the rest of the 1kgp pops in pos
- sam/nzm and cim/ton split in neg
- only sas and nearly eur are completely grouped together, looks somewhat random
-->

<!--
[x]describe number of total number of windows
[x]polynesian specific windows
[x]ep specific windows
[x]wp specific windows
[x]cim/nzm/sam/ton specific windows

[x]proportions
[x]exclusivity
[]dendrograms

[]refer to figure
-->
There was nearly a six-fold difference in the number of total windows used in the clustering between the upper (`r create_windowSummaryTable() %>% filter(stat == 'fld_pos') %>% .[['total_windows']]`) and lower tails (`r create_windowSummaryTable() %>% filter(stat == 'fld_neg') %>% .[['total_windows']]`). This indicated that there was a large number of regions in common in the upper tail, whereas few windows were shared in the lower tail, with many windows being population specific. In the upper tail there was a mean of `r round(create_windowSummaryTable() %>% filter(stat == 'fld_pos') %>%  .[['mean']], digits = 2)` populations sharing a window, which was `r round(create_windowSummaryTable() %>% filter(stat == 'fld_pos') %>%  .[['mean']] / (create_windowSummaryTable() %>% select(mean) %>% summarise(mean = mean(mean))), digits = 2)` times greater than the mean for all selection test stats used. 

Clustering of the lower tail (Figure \@ref(fig:fldTop1000) A and C), with the exception of the \gls{afr} populations, clustered the individual populations into their super population groups. The exclusivity of these groups however was low, with 17 populations, including all from \gls{amr}, \gls{eur}, and \gls{sas} assigned the same cluster. The \gls{afr} populations were spread over 3 clusters. The \gls{pol} populations were grouped as an exclusive single cluster. There were a total of `r create_windowSummaryTable() %>% filter(stat == 'fld_neg') %>% .[['total_windows']]` windows with `r create_sums('fld_neg') %>% filter(POLsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows specific to Polynesian populations. This compared to `r create_sums('fld_neg') %>% filter(AFRsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{afr}, `r create_sums('fld_neg') %>% filter(AMRsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{amr}, `r create_sums('fld_neg') %>% filter(EURsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{eur}, and `r create_sums('fld_neg') %>% filter(EASsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{eas}. The mean number of population unique windows was `r create_sums('fld_neg') %>% filter(ALLsum == 1) %>% select(-contains('chrom'), -posid, -contains('sum')) %>% summarise_all(funs(sum)) %>% t() %>% mean() %>% round(., digits = 1)` (SD = `r create_sums('fld_neg') %>% filter(ALLsum == 1) %>% select(-contains('chrom'), -posid, -contains('sum')) %>% summarise_all(funs(sum)) %>% t() %>% sd() %>% round(., digits = 1)`) in the lower tail. The Eastern Polynesian populations had a total of `r create_sums('fld_neg') %>% filter(POL_CIM + POL_NZM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` unique windows, `r create_sums('fld_neg') %>% filter(POL_CIM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` were unique to \gls{cim} and `r create_sums('fld_neg') %>% filter(POL_NZM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` were unique to \gls{nzm}. The Western Polynesian populations had `r create_sums('fld_neg') %>% filter(POL_SAM + POL_TON == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` unique windows. Of these, `r create_sums('fld_neg') %>% filter(POL_SAM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` were unique to \gls{sam} and `r create_sums('fld_neg') %>% filter(POL_TON == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` to \gls{ton}. 

The clustering of the upper tail (Figure \@ref(fig:fldTop1000) B and D) grouped most populations into their super populations. The \gls{eas} populations were clustered together. The Western Polynesian populations were clustered together but the Eastern Polynesian populations were not. The \gls{amr} populations \gls{mxl} and \gls{pel} were grouped with the \gls{nzm}. Ten populations were grouped into a single cluster consisting of two from \gls{afr}, two from \gls{amr}, one from \gls{eur} and all five from \gls{sas}. There were a total of `r create_windowSummaryTable() %>% filter(stat == 'fld_pos') %>% .[['total_windows']]` windows with `r create_sums('fld_pos') %>% filter(POLsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows specific to Polynesian populations. This compared to `r create_sums('fld_pos') %>% filter(AFRsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{afr}, `r create_sums('fld_pos') %>% filter(AMRsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{amr}, `r create_sums('fld_pos') %>% filter(EURsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{eur}, and `r create_sums('fld_pos') %>% filter(EASsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{eas}. The mean number of unique windows per population was `r create_sums('fld_pos') %>% filter(ALLsum == 1) %>% select(-contains('chrom'), -posid, -contains('sum')) %>% summarise_all(funs(sum)) %>% t() %>% mean() %>% round(., digits = 1)` (SD = `r create_sums('fld_pos') %>% filter(ALLsum == 1) %>% select(-contains('chrom'), -posid, -contains('sum')) %>% summarise_all(funs(sum)) %>% t() %>% sd() %>% round(., digits = 1)`). The Eastern Polynesian populations had a total of `r create_sums('fld_pos') %>% filter(POL_CIM + POL_NZM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` unique windows with `r create_sums('fld_pos') %>% filter(POL_CIM == ALLsum & ALLsum > 0 ) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` only from \gls{cim} and `r create_sums('fld_pos') %>% filter(POL_NZM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` from \gls{nzm}. From the Western Polynesian populations there were `r create_sums('fld_pos') %>% filter(POL_SAM + POL_TON == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` specific windows. There were `r create_sums('fld_pos') %>% filter(POL_SAM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` specific to \gls{sam} and `r create_sums('fld_pos') %>% filter(POL_TON == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` specific to \gls{ton}. 




#### Fu and Li's _F_

(ref:flfTop1000) _\gls{flf} Clustering._ Hierarchical clustering of \gls{flf} using the upper and lower tails of the distribution. **A** and **B** Dendrogram representing the clusters of \gls{flf} lower and upper tails respectfully using hierarchical clustering with Euclidean distance and complete linkage. Coloured by super population. **C** and **D** Plots representing the windows present in each population from the lower and upper tails respectfully.

```{r flfTop1000, fig.cap = '(ref:flfTop1000)', fig.height=7, cache=TRUE}
#ggsave(four_plot("flf","Fu and Li's F"), filename = 'images/04_clustering/flfTop1000.png', device = 'png', height = 7, width = 7)
knitr::include_graphics('images/04_clustering/flfTop1000.png')
```

<!--
- pos clustered super populations together, split AMR
- pos looks like there is a reasonable number of windows shared between pops (5188)
- neg looks pretty sparse 17044
-->

<!--
[]describe number of total number of windows
[x]polynesian specific windows
[x]ep specific windows
[x]wp specific windows
[x]cim/nzm/sam/ton specific windows

[]proportions
[x]exclusivity
[]dendrograms

[]refer to figure
-->

Clustering on the 1% \gls{flf} from each tail of the distribution (Figure \@ref(fig:flfTop1000)) had `r create_windowSummaryTable() %>% filter(stat == 'flf_pos') %>% .[['total_windows']]` windows in the upper tail (Figure \@ref(fig:flfTop1000) B and D) and `r create_windowSummaryTable() %>% filter(stat == 'flf_neg') %>% .[['total_windows']]` windows in the lower tail (Figure \@ref(fig:flfTop1000) A and C). The mean number of unique windows per population was `r create_sums('flf_pos') %>% filter(ALLsum == 1) %>% select(-contains('chrom'), -posid, -contains('sum')) %>% summarise_all(funs(sum)) %>% t() %>% mean() %>% round(., digits = 1)` (SD = `r create_sums('flf_pos') %>% filter(ALLsum == 1) %>% select(-contains('chrom'), -posid, -contains('sum')) %>% summarise_all(funs(sum)) %>% t() %>% sd() %>% round(., digits = 1)`) in the upper tail, compared with `r create_sums('flf_neg') %>% filter(ALLsum == 1) %>% select(-contains('chrom'), -posid, -contains('sum')) %>% summarise_all(funs(sum)) %>% t() %>% mean() %>% round(., digits = 1)` (SD = `r create_sums('flf_neg') %>% filter(ALLsum == 1) %>% select(-contains('chrom'), -posid, -contains('sum')) %>% summarise_all(funs(sum)) %>% t() %>% sd() %>% round(., digits = 1)`) in the lower tail. Both tails clustered the \gls{eas} populations into their own exclusive cluster.

Clustering on the lower tail windows grouped the majority of the individual populations together into their super populations. The Polynesians populations were grouped together as a West Polynesian group and the \gls{cim} population. The \gls{nzm} population was grouped with the \gls{mxl} and \gls{pel} as part of a larger group with the \gls{eur} super population. Exclusivity of clusters was reasonable with four of six being exclusive. There were a total of `r create_windowSummaryTable() %>% filter(stat == 'flf_neg') %>% .[['total_windows']]` windows with `r create_sums('flf_neg') %>% filter(POLsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows specific to Polynesian populations. This compared to `r create_sums('flf_neg') %>% filter(AFRsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{afr}, `r create_sums('flf_neg') %>% filter(AMRsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{amr}, `r create_sums('flf_neg') %>% filter(EURsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{eur}, and `r create_sums('flf_neg') %>% filter(EASsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{eas}. The Eastern Polynesian populations had a total of `r create_sums('flf_neg') %>% filter(POL_CIM + POL_NZM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` unique windows, `r create_sums('flf_neg') %>% filter(POL_CIM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` were unique to \gls{cim} and `r create_sums('flf_neg') %>% filter(POL_NZM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` were unique to \gls{nzm}. The Western Polynesian populations had `r create_sums('flf_neg') %>% filter(POL_SAM + POL_TON == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` unique windows. Of these, `r create_sums('flf_neg') %>% filter(POL_SAM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` were unique to \gls{sam} and `r create_sums('flf_neg') %>% filter(POL_TON == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` to \gls{ton}.

Clustering on the windows from the upper tail put the super populations into their own groups, but also split the \gls{amr} populations into two separate groups. The Polynesian populations were also clustered into an East/West split of 2 separate exclusive clusters. Exclusivity of clusters was high with five of the six clusters being exclusive to a single super population, however one cluster contained half of the \gls{amr} populations (\gls{clm} and \gls{pur})and all of \gls{eur} and \gls{sas} populations. There were a total of `r create_windowSummaryTable() %>% filter(stat == 'flf_pos') %>% .[['total_windows']]` windows with `r create_sums('flf_pos') %>% filter(POLsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows specific to Polynesian populations. This compared to `r create_sums('flf_pos') %>% filter(AFRsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{afr}, `r create_sums('flf_pos') %>% filter(AMRsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{amr}, `r create_sums('flf_pos') %>% filter(EURsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{eur}, and `r create_sums('flf_pos') %>% filter(EASsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{eas}. The Eastern Polynesian populations had a total of `r create_sums('flf_pos') %>% filter(POL_CIM + POL_NZM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` unique windows with `r create_sums('flf_pos') %>% filter(POL_CIM == ALLsum & ALLsum > 0 ) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` only from \gls{cim} and `r create_sums('flf_pos') %>% filter(POL_NZM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` from \gls{nzm}. From the Western Polynesian populations there were `r create_sums('flf_pos') %>% filter(POL_SAM + POL_TON == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` specific windows. There were `r create_sums('flf_pos') %>% filter(POL_SAM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` specific to \gls{sam} and `r create_sums('flf_pos') %>% filter(POL_TON == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` specific to \gls{ton}.


#### Zeng's _E_

(ref:zeTop1000) _\gls{ze} Clustering._ Hierarchical clustering of \gls{ze} using the upper and lower tails of the distribution. **A** and **B** Dendrogram representing the clusters of \gls{ze} lower and upper tails respectfully using hierarchical clustering with Euclidean distance and complete linkage. Coloured by super population. **C** and **D** Plots representing the windows present in each population from the lower and upper tails respectfully.

```{r zeTop1000, fig.cap = '(ref:zeTop1000)', fig.height=7, cache=TRUE}
#ggsave(four_plot("ze","Zeng's E"), filename = 'images/04_clustering/zeTop1000.png', device = 'png', height = 7, width = 7)
knitr::include_graphics('images/04_clustering/zeTop1000.png')
```

<!--
Clustering of \gls{ze} on the positive tail (Figure \@ref(fig:zeTop1000) B and D) of the distribution gave clustering that matched the super popualations, with the Polyesian populations grouped together and subgrouped into an East/West split. The Polyensian population group's next closest group was the \gls{eas} super population. The negative tail (Figure \@ref(fig:zeTop1000) A and C) showed a similar grouping but split the \gls{amr} super population.
-->


\Gls{ze} for both extremes of the distribution (Figure \@ref(fig:zeTop1000)), clustered all populations except \gls{amr} into their respective super populations. The \gls{amr} population was split into \gls{clm} and \gls{pur} which was grouped with the \gls{eur} populations, and \gls{mxl} and \gls{pel} which were grouped as a separate exclusive cluster. The groupings of all super populations was the same for both tails, however, in the dendrogram the difference comes from where the \gls{amr} populations were relative to the other super populations with the \gls{mxl} and \gls{pur} populations changing slightly in how close to the \gls{sas} group they were. Exclusivity of clusters was high with only the \gls{amr} -\gls{eur} group not being exclusive to a single super population. 

There were a total of `r create_windowSummaryTable() %>% filter(stat == 'ze_neg') %>% .[['total_windows']]` windows, with `r create_sums('ze_neg') %>% filter(POLsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows specific to Polynesian populations in the lower tail. This compared to `r create_sums('ze_neg') %>% filter(AFRsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{afr}, `r create_sums('ze_neg') %>% filter(AMRsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{amr}, `r create_sums('ze_neg') %>% filter(EURsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{eur}, and `r create_sums('ze_neg') %>% filter(EASsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{eas}. There were a total of `r create_sums('ze_neg') %>% filter(POL_CIM + POL_NZM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows specific to the Eastern Polynesian populations, with `r create_sums('ze_neg') %>% filter(POL_CIM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` specific to \gls{cim} and `r create_sums('ze_neg') %>% filter(POL_NZM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` specific to \gls{nzm}. The Western Polynesian populations had `r create_sums('ze_neg') %>% filter(POL_SAM + POL_TON == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` unique windows, `r create_sums('ze_neg') %>% filter(POL_SAM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` were unique to \gls{sam} and `r create_sums('ze_neg') %>% filter(POL_TON == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` unique to \gls{ton}. The mean number of unique windows was `r create_sums('ze_neg') %>% filter(ALLsum == 1) %>% select(-contains('chrom'), -posid, -contains('sum')) %>% summarise_all(funs(sum)) %>% t() %>% mean() %>% round(., digits = 1)` in the lower tail.



In the upper tail there were a total of `r create_windowSummaryTable() %>% filter(stat == 'ze_pos') %>% .[['total_windows']]` windows with `r create_sums('ze_pos') %>% filter(POLsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows specific to Polynesian populations. This compared to `r create_sums('ze_pos') %>% filter(AFRsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{afr}, `r create_sums('ze_pos') %>% filter(AMRsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{amr}, `r create_sums('ze_pos') %>% filter(EURsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{eur}, and `r create_sums('ze_pos') %>% filter(EASsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{eas}. The Eastern Polynesian populations had `r create_sums('ze_pos') %>% filter(POL_CIM + POL_NZM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` unique windows, the \gls{cim} had`r create_sums('ze_pos') %>% filter(POL_CIM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)`unique windows, and \gls{nzm} had `r create_sums('ze_pos') %>% filter(POL_NZM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` unique windows. The Western Polynesian populations had `r create_sums('ze_pos') %>% filter(POL_SAM + POL_TON == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` unique windows. Of those, `r create_sums('ze_pos') %>% filter(POL_SAM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` were unique to \gls{sam} and `r create_sums('ze_pos') %>% filter(POL_TON == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` unique to \gls{ton}. The mean number of unique windows per population was `r create_sums('ze_pos') %>% filter(ALLsum == 1) %>% select(-contains('chrom'), -posid, -contains('sum')) %>% summarise_all(funs(sum)) %>% t() %>% mean() %>% round(., digits = 1)`.


<!-- create df from the clustering results to be able to pull numbers out of for number of windows per population for each statistic, and for the proportion of pops being clustered -->



\FloatBarrier
 
### Hierarchical Clustering of Selected Haplotypic Regions

```{r load_ihs_clus}
mat_ihs <- readRDS('~/data/NZ_coreExome_1kgp/haplotype/ihs_clustered_dist_mat-6-7-2017.RDS')
clus_ihs <- readRDS('~/data/NZ_coreExome_1kgp/haplotype/ihs_clus_regions-14-7-2017.RDS')
gr_ihs <- GenomicRanges::GRanges(as.data.frame(clus_ihs))
```

```{r, eval =  FALSE}
#nzm  vs cim
sum(GenomicRanges::width(GenomicRanges::reduce(GenomicRanges::intersect(gr_ihs[gr_ihs$pop == "POL_NZM"], gr_ihs[gr_ihs$pop == "POL_CIM"]))))/sum(GenomicRanges::width(GenomicRanges::reduce(gr_ihs[gr_ihs$pop == "POL_NZM"])))

sum(GenomicRanges::width(GenomicRanges::reduce(GenomicRanges::intersect(gr_ihs[gr_ihs$pop == "POL_CIM"], gr_ihs[gr_ihs$pop == "POL_NZM"]))))/sum(GenomicRanges::width(GenomicRanges::reduce(gr_ihs[gr_ihs$pop == "POL_CIM"])))

sum(GenomicRanges::width(GenomicRanges::reduce(gr_ihs[gr_ihs$pop %in% c("POL_SAM","POL_TON")])))/sum(GenomicRanges::width(GenomicRanges::reduce(gr_ihs[gr_ihs$pop %in% c("POL_NZM", "POL_CIM", "POL_SAM","POL_TON")])))

```

> A similar approach as with clustering of the frequency spectrum based statistics was used 

The proportion of shared regions was used to cluster populations, where the proportion was defined as the shared genomic regions that were significant between two populations, given the total region that was significant for a single population. Clustering on the proportion of shared regions (see \@ref(haplocluster)) for both \gls{ihs} (Figure \@ref(fig:ihsclust)) and \gls{nsl} (Figure \@ref(fig:nslclust)) grouped the individual populations into their super populations, with the exception of the \gls{amr} populations which were split into two separate groups. The order of populations differed due to each population having a different total for the region that was significant. Both \gls{ihs} and \gls{nsl} gave similar groupings and had similar proportions shared.

For the \gls{ihs} significant regions, the populations with the least regions shared between were \gls{asw} and \gls{cim}, with `r sprintf("%.1f",mat_ihs['POL_CIM','AFR_ASW'] *100)`%  shared. The populations that had the largest proportion of region shared was \gls{nzc} and \gls{ceu} with `r sprintf("%.1f",mat_ihs['EUR_NZC','EUR_CEU'] *100)`% shared. Looking specifically at the proportion of regions shared between Polynesian populations, for the Eastern Polynesians there was \gls{nzm} with `r (mat_ihs[rownames(mat_ihs) == "POL_CIM", colnames(mat_ihs) == "POL_NZM"] * 100) %>% sprintf('%.1f',.)`% of regions shared with \gls{cim}, and \gls{cim} shared `r (mat_ihs[rownames(mat_ihs) == "POL_NZM", colnames(mat_ihs) == "POL_CIM"] * 100) %>% sprintf('%.1f',.)`% of regions with \gls{nzm}. In the Western Polynesian populations, \gls{sam} shared `r (mat_ihs[rownames(mat_ihs) == "POL_TON", colnames(mat_ihs) == "POL_SAM"] * 100) %>% sprintf('%.1f',.)`% of regions with \gls{ton}, but \gls{ton}  shared `r (mat_ihs[rownames(mat_ihs) == "POL_SAM", colnames(mat_ihs) == "POL_TON"] * 100) %>% sprintf('%.1f',.)`% of regions with \gls{sam}. Out of all the significant \gls{ihs} regions for Polynesian populations, only  `r (sum(GenomicRanges::width(GenomicRanges::reduce(GenomicRanges::intersect(GenomicRanges::intersect(gr_ihs[gr_ihs$pop == 'POL_CIM'], gr_ihs[gr_ihs$pop == "POL_NZM"]), GenomicRanges::intersect(gr_ihs[gr_ihs$pop == 'POL_SAM'], gr_ihs[gr_ihs$pop == "POL_TON"])))))/sum(GenomicRanges::width(GenomicRanges::reduce(gr_ihs[gr_ihs$pop %in% c("POL_NZM", "POL_CIM", "POL_SAM","POL_TON")]))) * 100) %>% sprintf('%.1f',.)`% were in common across all Polynesian populations. The Eastern Polynesians accounted for `r (sum(GenomicRanges::width(GenomicRanges::reduce(gr_ihs[gr_ihs$pop %in% c("POL_NZM","POL_CIM")])))/sum(GenomicRanges::width(GenomicRanges::reduce(gr_ihs[gr_ihs$pop %in% c("POL_NZM", "POL_CIM", "POL_SAM","POL_TON")]))) *100) %>% sprintf('%.1f',.)`% of the regions from all of the Polynesian populations while the Western Polynesian populations made up `r (sum(GenomicRanges::width(GenomicRanges::reduce(gr_ihs[gr_ihs$pop %in% c("POL_SAM","POL_TON")])))/sum(GenomicRanges::width(GenomicRanges::reduce(gr_ihs[gr_ihs$pop %in% c("POL_NZM", "POL_CIM", "POL_SAM","POL_TON")]))) * 100) %>% sprintf('%.1f',.)`%.



<!--
CHECK HOW IHS PROPS ARE BEING CALCULATED!!!

what was the overlap in regions like for ihs/nsl

ihs
DOUBLE CHECK THE ORDERS 
double checked and colname is the denominator
-->

```{r calc_ihs_region_share}
# mean sharing if ihs regions between all super populations
sp_list <- list()
super_pops <- c("POL","AMR","AFR","EUR","EAS","SAS")
for(sp1 in super_pops){
  for(sp2 in super_pops){
    if(sp1 != sp2){
     sp_list[[paste0(sp1,"_",sp2)]] <-  sum(GenomicRanges::width(GenomicRanges::reduce(GenomicRanges::intersect(gr_ihs[gr_ihs$pop %in% unique(gr_ihs$pop[grep(sp1, gr_ihs$pop)])], gr_ihs[gr_ihs$pop %in% unique(gr_ihs$pop[grep(sp2, gr_ihs$pop)]) ]))))/sum(GenomicRanges::width(GenomicRanges::reduce(gr_ihs[gr_ihs$pop %in% unique(gr_ihs$pop[grep(sp1, gr_ihs$pop)])])))
    }
  }
}
mean_super_share_ihs <- mean(unlist(sp_list))
rm(super_pops, sp_list)
```

The proportion of significant \gls{ihs} regions the Polynesian populations shared with AMR was `r (sum(GenomicRanges::width(GenomicRanges::reduce(GenomicRanges::intersect(gr_ihs[gr_ihs$pop %in% unique(gr_ihs$pop[grep('POL', gr_ihs$pop)])], gr_ihs[gr_ihs$pop %in% unique(gr_ihs$pop[grep('AMR', gr_ihs$pop)]) ]))))/sum(GenomicRanges::width(GenomicRanges::reduce(gr_ihs[gr_ihs$pop %in% unique(gr_ihs$pop[grep('POL', gr_ihs$pop)])]))) * 100) %>% sprintf('%.1f',.)`%, AFR was `r (sum(GenomicRanges::width(GenomicRanges::reduce(GenomicRanges::intersect(gr_ihs[gr_ihs$pop %in% unique(gr_ihs$pop[grep('POL', gr_ihs$pop)])], gr_ihs[gr_ihs$pop %in% unique(gr_ihs$pop[grep('AFR', gr_ihs$pop)]) ]))))/sum(GenomicRanges::width(GenomicRanges::reduce(gr_ihs[gr_ihs$pop %in% unique(gr_ihs$pop[grep('POL', gr_ihs$pop)])]))) * 100) %>% sprintf('%.1f',.)`%, EAS was `r (sum(GenomicRanges::width(GenomicRanges::reduce(GenomicRanges::intersect(gr_ihs[gr_ihs$pop %in% unique(gr_ihs$pop[grep('POL', gr_ihs$pop)])], gr_ihs[gr_ihs$pop %in% unique(gr_ihs$pop[grep('EAS', gr_ihs$pop)]) ]))))/sum(GenomicRanges::width(GenomicRanges::reduce(gr_ihs[gr_ihs$pop %in% unique(gr_ihs$pop[grep('POL', gr_ihs$pop)])]))) * 100) %>% sprintf('%.1f',.)`%, EUR was `r (sum(GenomicRanges::width(GenomicRanges::reduce(GenomicRanges::intersect(gr_ihs[gr_ihs$pop %in% unique(gr_ihs$pop[grep('POL', gr_ihs$pop)])], gr_ihs[gr_ihs$pop %in% unique(gr_ihs$pop[grep('EUR', gr_ihs$pop)]) ]))))/sum(GenomicRanges::width(GenomicRanges::reduce(gr_ihs[gr_ihs$pop %in% unique(gr_ihs$pop[grep('POL', gr_ihs$pop)])]))) * 100) %>% sprintf('%.1f',.)`%, and SAS was `r (sum(GenomicRanges::width(GenomicRanges::reduce(GenomicRanges::intersect(gr_ihs[gr_ihs$pop %in% unique(gr_ihs$pop[grep('POL', gr_ihs$pop)])], gr_ihs[gr_ihs$pop %in% unique(gr_ihs$pop[grep('SAS', gr_ihs$pop)]) ]))))/sum(GenomicRanges::width(GenomicRanges::reduce(gr_ihs[gr_ihs$pop %in% unique(gr_ihs$pop[grep('POL', gr_ihs$pop)])]))) * 100) %>% sprintf('%.1f',.)`%. This was lower than the mean (`r sprintf("%.1f",mean_super_share_ihs * 100)`%) for proportion of regions sharing between the other super populations. 

The Polynesian populations were clustered together with a high proportion of shared significant regions. The Eastern Polynesians had on average `r mat_ihs %>% data.frame %>% mutate(pop1 = row.names(.))%>% gather("pop2","ihs", 1:(NCOL(.)-1)) %>%  filter(pop1!=pop2, (pop1 %in% c("POL_CIM","POL_NZM") & pop2 %in% c("POL_CIM","POL_NZM"))) %>% summarise(mean(ihs)*100) %>% sprintf('%.1f',.)`% in common, and the Western Polynesians had `r mat_ihs %>% data.frame %>% mutate(pop1 = row.names(.))%>% gather("pop2","ihs", 1:(NCOL(.)-1)) %>%  filter(pop1!=pop2, (pop1 %in% c("POL_TON","POL_SAM") & pop2 %in% c("POL_TON","POL_SAM"))) %>% summarise(mean(ihs)*100) %>% sprintf('%.1f',.)`%. However, shared regions between the Eastern and Western populations was on average `r mat_ihs %>% data.frame %>% mutate(pop1 = row.names(.))%>% gather("pop2","ihs", 1:(NCOL(.)-1)) %>% filter((pop1 %in% c("POL_CIM","POL_NZM") & pop2 %in% c("POL_TON","POL_SAM")) | (pop2 %in% c("POL_CIM","POL_NZM") & pop1 %in% c("POL_TON","POL_SAM")))  %>% summarise(mean(ihs)*100) %>% sprintf('%.1f',.)`%. 

```{r ihsclust, fig.cap='(ref:ihsclust)', echo = FALSE, fig.asp = 1}
ihs_dendro_plot()
```

(ref:ihsclust) _\gls{ihs} Clustering_ A. Dendrogram from hierarchical clustering based on proportion of significant \gls{ihs} regions shared between populations Coloured by super population. B. Heatmap showing the proportion of significant \gls{ihs} regions shared. The population of the axis is A in the calculation = (A $\cap$ B)/ A. Axes have been ordered according to the hierarchical clustering. 

```{r nslclust, fig.cap='(ref:nslclust)', echo = FALSE, fig.asp= 1 }
nsl_dendro_plot()
```

(ref:nslclust) _\gls{nsl} Clustering_ A. Dendrogram from hierarchical clustering based on proportion of significant \gls{nsl} regions shared between populations. Coloured by super population. B. Heatmap showing the proportion of significant \gls{nsl} regions shared. The population of the axis is A the calculation = (A $\cap$ B)/ A. Axes have been ordered according to the hierarchical clustering.

<!-- nSL -->

```{r load_nsl_clus}
mat_nsl <- readRDS('~/data/NZ_coreExome_1kgp/haplotype/nsl_clustered_dist_mat-6-7-2017.RDS')
clus_nsl <- readRDS('~/data/NZ_coreExome_1kgp/haplotype/nsl_clus_regions-14-7-2017.RDS')
gr_nsl <- GenomicRanges::GRanges(as.data.frame(clus_nsl))
```

```{r calc_nsl_region_share}
# mean sharing if ihs regions between all super populations
sp_list <- list()
super_pops <- c("POL","AMR","AFR","EUR","EAS","SAS")
for(sp1 in super_pops){
  for(sp2 in super_pops){
    if(sp1 != sp2){
     sp_list[[paste0(sp1,"_",sp2)]] <-  sum(GenomicRanges::width(GenomicRanges::reduce(GenomicRanges::intersect(gr_nsl[gr_nsl$pop %in% unique(gr_nsl$pop[grep(sp1, gr_nsl$pop)])], gr_nsl[gr_nsl$pop %in% unique(gr_nsl$pop[grep(sp2, gr_nsl$pop)]) ]))))/sum(GenomicRanges::width(GenomicRanges::reduce(gr_nsl[gr_nsl$pop %in% unique(gr_nsl$pop[grep(sp1, gr_nsl$pop)])])))
    }
  }
}
mean_super_share_nsl <- mean(unlist(sp_list))
rm(super_pops, sp_list)
```

The clustering results using regions of significant \gls{nsl} gave the same groupings as \gls{ihs}. This was similarity was also seen with the differences in individual populations, where \gls{nzc} and \gls{ceu} had the highest percentage of shared regions at `r sprintf("%.1f",mat_nsl['EUR_NZC','EUR_CEU'] *100)`%. \Gls{nzm} and \gls{gwd} shared the lowest percentage of regions at `r sprintf("%.1f",mat_nsl['POL_NZM','AFR_GWD'] *100)`%. The percentage of the significant \gls{nsl} regions the Polynesian populations shared with AMR was `r (sum(GenomicRanges::width(GenomicRanges::reduce(GenomicRanges::intersect(gr_nsl[gr_nsl$pop %in% unique(gr_nsl$pop[grep('POL', gr_nsl$pop)])], gr_nsl[gr_nsl$pop %in% unique(gr_nsl$pop[grep('AMR', gr_nsl$pop)]) ]))))/sum(GenomicRanges::width(GenomicRanges::reduce(gr_nsl[gr_nsl$pop %in% unique(gr_nsl$pop[grep('POL', gr_nsl$pop)])]))) * 100) %>% sprintf('%.1f',.)`%, AFR  was `r (sum(GenomicRanges::width(GenomicRanges::reduce(GenomicRanges::intersect(gr_nsl[gr_nsl$pop %in% unique(gr_nsl$pop[grep('POL', gr_nsl$pop)])], gr_nsl[gr_nsl$pop %in% unique(gr_nsl$pop[grep('AFR', gr_nsl$pop)]) ]))))/sum(GenomicRanges::width(GenomicRanges::reduce(gr_nsl[gr_nsl$pop %in% unique(gr_nsl$pop[grep('POL', gr_nsl$pop)])]))) * 100) %>% sprintf('%.1f',.)`%, EAS   was `r (sum(GenomicRanges::width(GenomicRanges::reduce(GenomicRanges::intersect(gr_nsl[gr_nsl$pop %in% unique(gr_nsl$pop[grep('POL', gr_nsl$pop)])], gr_nsl[gr_nsl$pop %in% unique(gr_nsl$pop[grep('EAS', gr_nsl$pop)]) ]))))/sum(GenomicRanges::width(GenomicRanges::reduce(gr_nsl[gr_nsl$pop %in% unique(gr_nsl$pop[grep('POL', gr_nsl$pop)])]))) * 100) %>% sprintf('%.1f',.)`%, EUR was `r (sum(GenomicRanges::width(GenomicRanges::reduce(GenomicRanges::intersect(gr_nsl[gr_nsl$pop %in% unique(gr_nsl$pop[grep('POL', gr_nsl$pop)])], gr_nsl[gr_nsl$pop %in% unique(gr_nsl$pop[grep('EUR', gr_nsl$pop)]) ]))))/sum(GenomicRanges::width(GenomicRanges::reduce(gr_nsl[gr_nsl$pop %in% unique(gr_nsl$pop[grep('POL', gr_nsl$pop)])]))) * 100) %>% sprintf('%.1f',.)`%, and SAS was `r (sum(GenomicRanges::width(GenomicRanges::reduce(GenomicRanges::intersect(gr_nsl[gr_nsl$pop %in% unique(gr_nsl$pop[grep('POL', gr_nsl$pop)])], gr_nsl[gr_nsl$pop %in% unique(gr_nsl$pop[grep('SAS', gr_nsl$pop)]) ]))))/sum(GenomicRanges::width(GenomicRanges::reduce(gr_nsl[gr_nsl$pop %in% unique(gr_nsl$pop[grep('POL', gr_nsl$pop)])]))) * 100) %>% sprintf('%.1f',.)`%. This was higher than the mean (`r sprintf("%.1f",mean_super_share_nsl * 100)`%) for \gls{eas}, but lower for the other super populations for proportion of region sharing between super populations. 


The Polynesian populations were clustered together with a high proportion of shared significant regions. The Eastern Polynesians had on average `r mat_nsl %>% data.frame %>% mutate(pop1 = row.names(.))%>% gather("pop2","nsl", 1:(NCOL(.)-1)) %>%  filter(pop1!=pop2, (pop1 %in% c("POL_CIM","POL_NZM") & pop2 %in% c("POL_CIM","POL_NZM"))) %>% summarise(mean(nsl)*100) %>% sprintf('%.1f',.)`% in common. NZM shared `r (mat_nsl[rownames(mat_nsl) == "POL_CIM", colnames(mat_nsl) == "POL_NZM"] * 100) %>% sprintf('%.1f',.)`% of regions with CIM, whereas CIM shared `r (mat_nsl[rownames(mat_nsl) == "POL_NZM", colnames(mat_nsl) == "POL_CIM"] * 100) %>% sprintf('%.1f',.)`% of regions with NZM. The Western Polynesians had `r mat_nsl %>% data.frame %>% mutate(pop1 = row.names(.))%>% gather("pop2","nsl", 1:(NCOL(.)-1)) %>%  filter(pop1!=pop2, (pop1 %in% c("POL_TON","POL_SAM") & pop2 %in% c("POL_TON","POL_SAM"))) %>% summarise(mean(nsl)*100) %>% sprintf('%.1f',.)`%. SAM shared `r (mat_nsl[rownames(mat_nsl) == "POL_TON", colnames(mat_nsl) == "POL_SAM"] * 100) %>% sprintf('%.1f',.)`% of regions with TON, but TON shared `r (mat_nsl[rownames(mat_nsl) == "POL_SAM", colnames(mat_nsl) == "POL_TON"] * 100) %>% sprintf('%.1f',.)`% of regions with SAM. However between the Eastern and Western populations was on average `r mat_nsl %>% data.frame %>% mutate(pop1 = row.names(.))%>% gather("pop2","nsl", 1:(NCOL(.)-1)) %>% filter((pop1 %in% c("POL_CIM","POL_NZM") & pop2 %in% c("POL_TON","POL_SAM")) | (pop2 %in% c("POL_CIM","POL_NZM") & pop1 %in% c("POL_TON","POL_SAM")))  %>% summarise(mean(nsl)*100) %>% sprintf('%.1f',.)`%. Of the regions from all of the Polynesian populations, the Western Polynesian populations made up `r (sum(GenomicRanges::width(GenomicRanges::reduce(gr_nsl[gr_nsl$pop %in% c("POL_SAM","POL_TON")])))/sum(GenomicRanges::width(GenomicRanges::reduce(gr_nsl[gr_nsl$pop %in% c("POL_NZM", "POL_CIM", "POL_SAM","POL_TON")]))) * 100) %>% sprintf('%.1f',.)`%. The Eastern Polynesians made up `r (sum(GenomicRanges::width(GenomicRanges::reduce(gr_nsl[gr_nsl$pop %in% c("POL_NZM","POL_CIM")])))/sum(GenomicRanges::width(GenomicRanges::reduce(gr_nsl[gr_nsl$pop %in% c("POL_NZM", "POL_CIM", "POL_SAM","POL_TON")]))) *100) %>% sprintf('%.1f',.)`%.  The total percentage of regions common for all Polynesian populations was  `r (sum(GenomicRanges::width(GenomicRanges::reduce(GenomicRanges::intersect(GenomicRanges::intersect(gr_nsl[gr_nsl$pop == 'POL_CIM'], gr_nsl[gr_nsl$pop == "POL_NZM"]), GenomicRanges::intersect(gr_nsl[gr_nsl$pop == 'POL_SAM'], gr_nsl[gr_nsl$pop == "POL_TON"])))))/sum(GenomicRanges::width(GenomicRanges::reduce(gr_nsl[gr_nsl$pop %in% c("POL_NZM", "POL_CIM", "POL_SAM","POL_TON")]))) * 100) %>% sprintf('%.1f',.)`%.
 
 \FloatBarrier

### Disease Associated Gene Clustering {#genelistcluster}
<!-- might be a good idea to use a grid to put multiple plots into one -->
```{r load_gene_clus, include = FALSE}
load('~/data/gwas_catalog/diseaseGR-25-7-2017.RData')# brings in objects called {gc_urate_gout,urate_gout,kd,metsyn,obesity,t2d}_GR, 
load('~/data/NZ_coreExome_1kgp/100kbWindow_intra/gene_clustering_filtered_3ns-31-8-2017.RData')# brings in objects called gout, obesity, t2d, metsyn
names(gout) <- c('mat_d','prop')
names(t2d) <-  c('mat_d','prop')
names(obesity) <-  c('mat_d','prop')
names(metsyn) <-  c('mat_d','prop')
names(kd) <- c('mat_d','prop')
```

> Introductory sentences/paragraphs to remind

To assess if there was something specific about the metabolic disease gene lists, clustering was performed using the actual values for the genes. These values were median centred to account for the population specific shifting of distributions which would have accounted for the majority of difference in the distance calculation. Gene lists were chosen for metabolic related disease, and also for late onset diseases with a genetic component that should have minimal selective pressure.

Overall, gene lists derived from \gls{gwas} associations clustered the individual populations into their super population groupings. Windows that intersected genes that had been associated with a disease were centered by population median and then grouped using hierarchical clustering. There were varying levels of super population proportions and exclusivity (the proportion of super populations in a single cluster) of clusters (Table \@ref(tab:proportion) and \@ref(tab:exclusivity)). The Polynesian populations clustered together when using each of \gls{fwh}, \gls{td}, and \gls{ze}, and in the majority of clustering (18/25) they were exclusive. This pattern was also similar with the \gls{eas} populations (17/25). For the clusters that contained the entirety of a super population, half the groupings for the entire Polynesian super population were exclusive. Three quarters of the groupings for each of the AFR and EAS super populations were exclusive and grouped the super population in their entirety.

Comparing the gene lists by statistic, for all selection statistics and gene lists the populations of the \gls{sas} super population were all clustered together, although often with poor exclusivity. The \gls{eur} super population, with the exception of \gls{fld}, was clustered entirely together but with a ranging degree of exclusivity across the different selection test statistics. \Gls{td}, \gls{fwh}, and \gls{ze} clustered all the super populations in their entirety although the \gls{amr} populations were split. The clustering of super populations was not specific to the gene lists associated with urate, gout, obesity, \gls{t2d}, kidney disease, or, metabolic syndrome but were also seen with lists for genes associated with Alzheimer's disease, Parkinson's disease, age-related macular degeneration, and colerectal cancer (data not shown). This indicates that there is likely nothing specific about the urate and gout, \gls{t2d}, obesity, kidney disease, or metabolic syndrome gene lists, but they contain similar discriminatory information as other lists of genes, which is likely to be capturing allele frequency differences between populations.


<!--
- Tajima's D clustered all super populations together (AMR was slightly ungrouped, often PEL out) and get the EP/WP split for all diseases

- Fay and Wu's H clustered all super populations together (AMR was slightly ungrouped, often PEL out) and get the EP/WP split for all diseases

- Fu and Li's D did not have groups that resembled the super populations and the exclusivity of the groups that did have a high proportion of populations from the same super population were not exclusive.

- Fu and Li's F clustered both EUR and SAS for the gene lists.

- Except for AMR, Zeng's E clustered all populations into their super populations. 
-->

<!--
Clustering on other diseases such as pd, macular degeneration, alzhimers and colon cancer revealed similar patters 
-->
#### Urate and Gout

(ref:goutdendros) _Urate and Gout Associated Gene Clustering_ Dendrograms created from hierarchical clustering applied to windows from the `r length(unique(urate_goutGR$SYMBOL))` loci associated with urate and gout. Coloured by super population.

```{r goutdendros, fig.cap='(ref:goutdendros)', fig.height=7}
multiplot(gene_dendros(gene_data = gout$mat_d$Tajima.D[,-1:-3]) + ggtitle("Tajima's D"),
          gene_dendros(gene_data = gout$mat_d$Fay.Wu.H[,-1:-3]) + ggtitle("Fay and Wu's H"),
          gene_dendros(gene_data = gout$mat_d$Fu.Li.D[,-1:-3]) + ggtitle("Fu and Li's D"),
          gene_dendros(gene_data = gout$mat_d$Fu.Li.F[,-1:-3]) + ggtitle("Fu and Li's F"),
          gene_dendros(gene_data = gout$mat_d$Zeng.E[,-1:-3]) + ggtitle("Zeng's E"),
  cols = 2)
```


There were `r length(unique(urate_goutGR$SYMBOL))` genes that had an association with urate or gout at a genome-wide significance level. This mapped to `r nrow(gout$mat_d$Tajima.D)` windows for each of the frequency based selection test statistics. \gls{ze} and \gls{fwh} clustered each super population into their own group and each cluster was exclusive to a super population. Both \gls{td} and \gls{flf} grouped most of the super populations correctly and nearly all clusters had good exclusivity for each single super population. \Gls{fld} clustered the individual populations into a few groups that resembled their super populations, however, other populations were scattered throughout. \gls{td}, \gls{fwh}, and \gls{ze} also had an Eastern/Western Polynesian population sub group within the Polynesian group (Figure \@ref(fig:goutdendros)). Unlike the F~ST~ clustering, the \gls{amr} populations were grouped together for \gls{td}, \gls{fwh}, and \gls{ze}. The \gls{eas} and \gls{pol} populations were closest for \gls{flf}, \gls{fwh}, and \gls{ze}, but not for \gls{td}, where the \gls{pol} populations were furthest from the other populations.

There were 2 genes; _BCAS3_ and _RREB1_, that had at least one statistic in all of the Polynesian populations that met the threshold of significance in the lower tail. _BCAS3_ met the significance threshold for \gls{fwh} and \gls{flf} in all Polynesian populations, for \gls{fld} all Polynesian populations except \gls{sam}, and for \gls{td} all Polynesian populations except \gls{ton}. _INHBC_ and _ATXN2_ both had windows that met the lower tail significance threshold for \gls{ze} in all Polynesians populations, except \gls{sam} for _INHBC_. There were `r lower_sig_stats %>% filter(SYMBOL %in% unique(urate_goutGR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(SYMBOL) %>% distinct() %>% NROW()` of the urate and gout associated genes that had evidence from at least one selection statistic of possible selection in any of the Polynesian populations from the lower tail of the distributions. In the upper tail, _SLC2A9_ had windows with significant \gls{fld} in the \gls{cim}, \gls{nzm}, and  \gls{sam} populations. _VEGFA_ had windows of \gls{ze} values that met the upper tail significance threshold  for all Polynesians populations. _BAZ1B_ also met the threshold in the Eastern Polynesian populations. There were no loci that were significant in all Polynesian populations in the upper tail.

```{markdown, eval = FALSE}
any stat with all populations
 - lower `r lower_sig_stats %>% filter(SYMBOL %in% unique(urate_goutGR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(pop, SYMBOL) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==4) %>% NROW()`
 - upper `r upper_sig_stats %>% filter(SYMBOL %in% unique(urate_goutGR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(pop, SYMBOL) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==4) %>% NROW()`

has all stats
 - lower `r lower_sig_stats %>% filter(SYMBOL %in% unique(urate_goutGR$SYMBOL)) %>% group_by(pop, SYMBOL, statname) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON'))  %>% ungroup()%>% select(-n, -statname) %>% group_by(pop, SYMBOL) %>% tally() %>% filter(n == 5) %>% NROW()`
 - upper `r upper_sig_stats %>% filter(SYMBOL %in% unique(urate_goutGR$SYMBOL)) %>% group_by(pop, SYMBOL, statname) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON'))  %>% ungroup()%>% select(-n, -statname) %>% group_by(pop, SYMBOL) %>% tally() %>% filter(n == 5) %>% NROW()`



<!--
#gives list of genes that were sig in all pops (POL) but from any stats
`r lower_sig_stats %>% filter(SYMBOL %in% unique(urate_goutGR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(pop, SYMBOL) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==4) %>% .[['SYMBOL']]` 

#gives the number of stats for a gene for a poplation (POL) that were sig
lower_sig_stats %>% left_join(., panel %>% select(pop, super_pop) %>% distinct(), by = 'pop')%>% filter(SYMBOL %in% unique(urate_goutGR$SYMBOL)) %>% group_by(super_pop,pop, statname, SYMBOL) %>%tally()  %>% ungroup() %>% select(-n)%>% spread(statname, statname)  %>% arrange(SYMBOL, super_pop, pop)%>% filter(pop %in% c("CIM", "NZM","SAM","TON")) %>% mutate(s = rowSums(cbind(!is.na(Fay.Wu.H), !is.na(Fu.Li.D), !is.na(Fu.Li.F), !is.na(Tajima.D), !is.na(Zeng.E)))) %>% View()


upper_sig_stats %>% left_join(., panel %>% select(pop, super_pop) %>% distinct(), by = 'pop')%>% filter(SYMBOL %in% unique(urate_goutGR$SYMBOL)) %>% group_by(super_pop,pop, statname, SYMBOL) %>%tally()  %>% ungroup() %>% select(-n)%>% spread(statname, statname)  %>% arrange(SYMBOL, super_pop, pop)%>% View()


-->

<!--
- Td gave the super populations, good exclusivity of clusters,ep/wp split
- fwh gave perfect super populations and perfect exclusivity. ep/wp split
- fld not great, sam most disimilar, nzm-gbr, ton-cim-chb-jpt
- flf got most of the super populations. not bad for exclusivity, cim-nzm-sam + ton-chb-jpt as a big cluster. clm/pur in the middle of the eur cluster - spanish sub cluster?
- ze perfect super populations, perfect exclusivity

dendrograms:
all except fld manage to cluster the super populations with the exception of AMR although ze does. fld is somewhat random. except fld all give east/west split
td and flf hace pol as outlier

exclusivity:
zeng e - perfect
td and fay wu's h do a good job but because of the AMR split end up with 7 actual clusters
fld - poor because there are lots of smaller clusters
flf - not great

props:
ze perfect
td and fwh only get amr not complete
flf has prop of >.75, except eur
fld poly .25, rest looking at majority of super pop in same cluster

-->
```



#### Obesity


The GWAS catalog had `r length(unique(obesity_GR$SYMBOL))` genes associated with obesity, this was represented by `r nrow(obesity$mat_d$Tajima.D)` windows. With the exception of \gls{fld}, clustering resulted in the groupings of super populations (Figure \@ref(fig:obesitydendros)). Exclusivity of a single super population to a cluster was also high with the notable exception being the \gls{amr} populations of \gls{clm} and \gls{pur} often being part of the \gls{eur} cluster. The Polynesian super population cluster was also closest to \gls{eas} in all selection statistics except \gls{fld}. All statistics had an Eastern/Western sup-grouping within the \gls{pol} group, although \gls{fld} did not have the \gls{nzm} as part of this group.

There were `r lower_sig_stats %>% filter(SYMBOL %in% unique(obesity_GR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(SYMBOL) %>% distinct() %>% NROW()` of the obesity associated genes that had evidence from at least one selection statistic of possible selection in any of the Polynesian populations from the lower tail of the distributions. In the lower tail there were `r lower_sig_stats %>% filter(SYMBOL %in% unique(obesity_GR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(pop, SYMBOL) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==4) %>% NROW()` loci with at least one window that was significant in the Polynesian populations. _ADAMTSL3_, _AGBL4_, _ARL15_, _CCR3_, _CREB1_, _ERBB4_, _FER_, _FHIT_, _GRID1_, _LTBP1_, _MAP2K5_, and _ZBTB38_ each had in all Polynesian populations, at least one statistic that met the significance threshold in the lower tail. Only _DNAH10_ had a significant window for each of \gls{td}, \gls{fwh}, \gls{fld}, \gls{flf}, and \gls{ze}, and this was only in the \gls{nzm} population. The upper tail had a total of `r upper_sig_stats %>% filter(SYMBOL %in% unique(obesity_GR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(pop, SYMBOL) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% NROW()` loci that had at least a window that was significant in the Polynesian populations. _ADCY3_, _CHST8_, _CMIP_, _ERBB4_, _FHIT_, _FTO_, _GRID1_, _HMGA2_, _LTBP1_, _NEGR1_, _NRXN3_, _NT5C2_, _PARK2_, _PRKCH_, _RARB_, and  _TCF7L2_ each had in all Polynesian populations, at least one statistic that met the significance threshold in the upper tail.


```{markdown, eval = FALSE}
any stat
 - lower `r lower_sig_stats %>% filter(SYMBOL %in% unique(obesity_GR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(pop, SYMBOL) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==4) %>% NROW()`
 - upper `r upper_sig_stats %>% filter(SYMBOL %in% unique(obesity_GR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(pop, SYMBOL) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==4) %>% NROW()`

has all stats
 - lower `r lower_sig_stats %>% filter(SYMBOL %in% unique(obesity_GR$SYMBOL)) %>% group_by(pop, SYMBOL, statname) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON'))  %>% ungroup()%>% select(-n, -statname) %>% group_by(pop, SYMBOL) %>% tally() %>% filter(n == 5) %>% NROW()`
 - upper `r upper_sig_stats %>% filter(SYMBOL %in% unique(obesity_GR$SYMBOL)) %>% group_by(pop, SYMBOL, statname) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON'))  %>% ungroup()%>% select(-n, -statname) %>% group_by(pop, SYMBOL) %>% tally() %>% filter(n == 5) %>% NROW()`

<!--
- td excellent groupings, mxl/pel on their own and clm/pur with eur. exclusivity was also pretty much perfect. ep/wp split
- fwh near perfect, get 3/4 amr being clustered with the eur group. ep/wp split
- fld nzm-mxl. eas, afr grouped. eur, amr and sas inter mixed somewhat. cim-sam-ton
- flf nice super populations, eas-pol, eur-2amr, mxl/pel
- ze good super populations, amr-eur and nearly sas, pol-eas with ep/wp split. pretty good exclusivity.

dendrograms
all except fld had nice clusters although amr was split
flf has pol as outlier

exclusivity
td and fwh - only eur/amr shared
flf afr/pol/eas were exclusive
fld pol was in 4 different clusters
flf and fld had smaller groupings so more sharing

prop
td, fwh and ze all except amr were 1.0
fld did ok except for Poly
flf did ok
-->
```
  
```{r obesitydendros, fig.cap='(ref:obesitydendros)', fig.height=7}
multiplot(gene_dendros(gene_data = obesity$mat_d$Tajima.D[,-1:-3]) + ggtitle("Tajima's D"),
          gene_dendros(gene_data = obesity$mat_d$Fay.Wu.H[,-1:-3]) + ggtitle("Fay and Wu's H"),
          gene_dendros(gene_data = obesity$mat_d$Fu.Li.D[,-1:-3]) + ggtitle("Fu and Li's D"),
          gene_dendros(gene_data = obesity$mat_d$Fu.Li.F[,-1:-3]) + ggtitle("Fu and Li's F"),
          gene_dendros(gene_data = obesity$mat_d$Zeng.E[,-1:-3]) + ggtitle("Zeng's E"),
   cols = 2)
```

(ref:obesitydendros) _Obesity Associated Gene Clustering_ Dendrograms created from hierarchical clustering applied to windows from the `r length(unique(obesity_GR$SYMBOL))` loci associated with obesity. Coloured by super population.

#### T2D

(ref:t2ddendros) _Type 2 Diabetes Associated Gene Clustering_ Dendrograms created from hierarchical clustering applied to windows from the `r length(unique(t2d_GR$SYMBOL))` loci associated with \gls{t2d}. Coloured by super population.

```{r t2ddendros, fig.cap='(ref:t2ddendros)', fig.height=7}
multiplot(gene_dendros(gene_data = t2d$mat_d$Tajima.D[,-1:-3]) + ggtitle("Tajima's D"),
          gene_dendros(gene_data = t2d$mat_d$Fay.Wu.H[,-1:-3]) + ggtitle("Fay and Wu's H"),
          gene_dendros(gene_data = t2d$mat_d$Fu.Li.D[,-1:-3]) + ggtitle("Fu and Li's D"),
          gene_dendros(gene_data = t2d$mat_d$Fu.Li.F[,-1:-3]) + ggtitle("Fu and Li's F"),
          gene_dendros(gene_data = t2d$mat_d$Zeng.E[,-1:-3]) + ggtitle("Zeng's E"),
   cols = 2)
```

The GWAS catalog had `r length(unique(t2d_GR$SYMBOL))` genes associated with \gls{t2d}, this was represented by `r nrow(t2d$mat_d$Tajima.D)` windows. Except for \gls{fld}, clustering of the \gls{t2d} associated genes grouped the individual populations into their super population groups. For all clustering the \gls{amr} populations were split, with \gls{clm} and \gls{pur} grouped with the \gls{eur} populations for \gls{td}, \gls{fwh}, and \gls{ze} (Figure \@ref(fig:t2ddendros)). The \gls{pol} populations were assigned two exclusive groups through the cutree method that created an Eastern/Western Polynesian split for \gls{td}, \gls{fwh}, \gls{flf}, and \gls{ze}. \gls{fld} had the Eastern Polynesian populations with the \gls{eur} populations. \gls{fwh} and \gls{ze} had a similar pattern to the groupings of the whole chromosome F~ST~ clustering.

There were `r lower_sig_stats %>% filter(SYMBOL %in% unique(t2d_GR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(SYMBOL) %>% distinct() %>% NROW()` \gls{t2d} associated genes that had evidence from at least one selection statistic of possible selection in any of the Polynesian populations from the lower tail of the distributions. There were a total of `r upper_sig_stats %>% filter(SYMBOL %in% unique(t2d_GR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(SYMBOL) %>% distinct() %>% NROW()` genes that had a significant window in any Polynesian population. In the lower tail there were `r lower_sig_stats %>% filter(SYMBOL %in% unique(t2d_GR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(pop, SYMBOL) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==4) %>% NROW()` genes with at least a window that met the significance threshold in the Polynesian populations. Only _THADA_ had windows from \gls{td}, \gls{fwh}, \gls{fld}, \gls{flf}, and \gls{ze} in the \gls{ton} population from the Polynesian populations. This was also true for the \gls{cdx} and \gls{chs} populations. _ARL15_, _MACF1_, _RREB1_, and _THADA_ each had in all Polynesian populations, at least one statistic that was significant in the lower tail. _CAMK1D_, _FTO_, _HLA-B_, _HMGA2_, _HNF1B_, _PPP2R2C_, _PTPRD_, _RBMS1_, _SGCG_, _TCF7L2_, _THADA_, _UBE2E2_, _ZMIZ1_ each had in all Polynesian populations, at least one statistic that was significant in the upper tail.

```{markdown, eval = FALSE}
any stat
 - lower `r lower_sig_stats %>% filter(SYMBOL %in% unique(t2d_GR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(pop, SYMBOL) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==4) %>% NROW()`
 - upper `r upper_sig_stats %>% filter(SYMBOL %in% unique(t2d_GR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(pop, SYMBOL) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==4) %>% NROW()`

has all stats
 - lower `r lower_sig_stats %>% filter(SYMBOL %in% unique(t2d_GR$SYMBOL)) %>% group_by(pop, SYMBOL, statname) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON'))  %>% ungroup()%>% select(-n, -statname) %>% group_by(pop, SYMBOL) %>% tally() %>% filter(n == 5) %>% NROW()`
 - upper `r upper_sig_stats %>% filter(SYMBOL %in% unique(t2d_GR$SYMBOL)) %>% group_by(pop, SYMBOL, statname) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON'))  %>% ungroup()%>% select(-n, -statname) %>% group_by(pop, SYMBOL) %>% tally() %>% filter(n == 5) %>% NROW()`



<!--

- td ep/wp split, pol as outlier. pretty good clustering and exclusivity
- fwh ep/wp split, pol-eas as outlier, not bad for exclusivity, clustering was good
- fld nzm-pel. 3/4 pol as outlier group. not too bad for exclusivity
- flf pol-eas, ep/wp. pretty good exclusivity and clusters that are pretty close to the super populations
- ze good exclusivity and good clustering. pol-eas, ep/wp split. amr(clm/pur)-eur

dendrograms
td, fwh, and ze super pops, split amr
flf super pops but split amr and eur together
fld only eas and sas together
all have a wp/ep split but only wp for fld

exclusivity
td ok - but has .5amr, eur and sas together
fwh and flf pretty much same as td
this is because pol is 2 groups
fld poor - amr, eur, sas and afr are all mostly in single cluster
ze good - only has .5 amr and eur together


prop
ze all except amr
td, fwh, and flf have 1.0 for eur,afr and sas. pol is 0.5
fld not bad for all except POL
-->
```
  
#### Kidney Disease

(ref:kidneydendros) _Kidney Disease Associated Gene Clustering_ Dendrograms created from hierarchical clustering applied to windows from the `r length(unique(kd_GR$SYMBOL))` loci associated with kidney disease. Coloured by super population.

```{r kidneydendros, fig.cap='(ref:kidneydendros)', fig.height=7}
multiplot(gene_dendros(gene_data = kd$mat_d$Tajima.D[,-1:-3]) + ggtitle("Tajima's D"),
          gene_dendros(gene_data = kd$mat_d$Fay.Wu.H[,-1:-3]) + ggtitle("Fay and Wu's H"),
          gene_dendros(gene_data = kd$mat_d$Fu.Li.D[,-1:-3]) + ggtitle("Fu and Li's D"),
          gene_dendros(gene_data = kd$mat_d$Fu.Li.F[,-1:-3]) + ggtitle("Fu and Li's F"),
          gene_dendros(gene_data = kd$mat_d$Zeng.E[,-1:-3]) + ggtitle("Zeng's E"),
   cols = 2)
```

The GWAS catalog had `r length(unique(kd_GR$SYMBOL))` genes associated with kidney disease, this was represented by `r nrow(kd$mat_d$Tajima.D)` windows. The clustering of the kidney disease associated genes grouped the individual populations into their super population groups for \gls{afr} and \gls{sas} using \gls{td}, \gls{fwh}, \gls{flf}, and \gls{ze}. Populations from \gls{pol} and \gls{eas} were clustered into separate clusters with  \gls{td} and \gls{flf} but were clustered together for \gls{fwh} and \gls{ze} (Figure \@ref(fig:kidneydendros)). No populations were entirely grouped into their super populations with \gls{fld}. \gls{td} clustered populations from \gls{afr}, \gls{eas}, \gls{pol}, and \gls{sas} into super population specific and exclusive clusters.




<!--
dendrograms
td and flf have eas/pol/afr and sas clustered
fwh and ze have a shared eas/pol
fld mixed
flf eas,pol,afr, and sas separated nicely
ze afr, eur, amr, sas separated. eas and pol mixed

exclusivity
td: all exclusive except 2amr in with eur
fwh: sas and eur share, pol and eas share
fld: very poor exclusivity
flf: eas and pol share
ze: eas and pol share

prop
td: all except amr
fwh: pol is split
fld no population completely clustered
flf all except EUR
ze only pol and eas not perfect
-->

There were `r lower_sig_stats %>% filter(SYMBOL %in% unique(kd_GR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(SYMBOL) %>% distinct() %>% NROW()` kidney disease associated genes that had evidence from at least one selection statistic of possible selection in any of the Polynesian populations from the lower tail of the distributions. In the lower tail there was `r lower_sig_stats %>% filter(SYMBOL %in% unique(kd_GR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(pop, SYMBOL) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==4) %>% NROW()` gene, _BRAS3_, with at least one window that was significant in the Polynesian populations. All Polynesian populations had at least one statistic that was significant in the upper tail for a window intersecting _PHTF2_. There was a total of `r upper_sig_stats %>% filter(SYMBOL %in% unique(kd_GR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(pop, SYMBOL) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% NROW()` genes that had at least one window that was significant in the upper tail.

```{markdown, eval = FALSE}
any stat
 - lower `r lower_sig_stats %>% filter(SYMBOL %in% unique(kd_GR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(pop, SYMBOL) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==4) %>% NROW()`
 - upper `r upper_sig_stats %>% filter(SYMBOL %in% unique(kd_GR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(pop, SYMBOL) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==4) %>% NROW()`

has all stats
 - lower `r lower_sig_stats %>% filter(SYMBOL %in% unique(kd_GR$SYMBOL)) %>% group_by(pop, SYMBOL, statname) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON'))  %>% ungroup()%>% select(-n, -statname) %>% group_by(pop, SYMBOL) %>% tally() %>% filter(n == 5) %>% NROW()`
 - upper `r upper_sig_stats %>% filter(SYMBOL %in% unique(kd_GR$SYMBOL)) %>% group_by(pop, SYMBOL, statname) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON'))  %>% ungroup()%>% select(-n, -statname) %>% group_by(pop, SYMBOL) %>% tally() %>% filter(n == 5) %>% NROW()`
```



#### Metabolic Syndrome

(ref:metsyndendros) _Metabolic Syndrome Associated Gene Clustering_ Dendrograms created from hierarchical clustering applied to windows from the `r length(unique(metsyn_GR$SYMBOL))` loci associated with metabolic syndrome. Coloured by super population.

```{r metsyndendros, fig.cap='(ref:metsyndendros)', fig.height=7}
multiplot(gene_dendros(gene_data = metsyn$mat_d$Tajima.D[,-1:-3]) + ggtitle("Tajima's D"),
          gene_dendros(gene_data = metsyn$mat_d$Fay.Wu.H[,-1:-3]) + ggtitle("Fay and Wu's H"),
          gene_dendros(gene_data = metsyn$mat_d$Fu.Li.D[,-1:-3]) + ggtitle("Fu and Li's D"),
          gene_dendros(gene_data = metsyn$mat_d$Fu.Li.F[,-1:-3]) + ggtitle("Fu and Li's F"),
          gene_dendros(gene_data = metsyn$mat_d$Zeng.E[,-1:-3]) + ggtitle("Zeng's E"),
   cols = 2)
```

The GWAS catalog had `r length(unique(metsyn_GR$SYMBOL))` genes associated with metabolic syndrome, this was represented by `r nrow(metsyn$mat_d$Tajima.D)` windows. The clustering of the metabolic syndrome associated genes grouped individual populations into their super population groups for \gls{td} and \gls{fwh} (Figure \@ref(fig:metsyndendros)). For \gls{flf}, the Polynesian populations were assigned two exclusive clusters that followed the Eastern/Western Polynesian split. \Gls{ze} clustered the individual populations, except for \gls{pel}, into their super populations. \gls{fld} separated the Polynesian populations into a separate group but did not distinguish the other populations into their super populations. The Polynesian populations were grouped into entirely exclusive clusters for all stats.

There were `r lower_sig_stats %>% filter(SYMBOL %in% unique(metsyn_GR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(SYMBOL) %>% distinct() %>% NROW()` metabolic syndrome associated genes that had evidence from at least one selection statistic of possible selection in any of the Polynesian populations from the lower tail of the distributions. In the lower tail there were `r lower_sig_stats %>% filter(SYMBOL %in% unique(metsyn_GR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(pop, SYMBOL) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==4) %>% NROW()` genes with at least one window that was significant in the Polynesian populations. _DGKB_, _FTO_, _LIPC_, and _MICB_ each had in all Polynesian populations, at least one statistic that was significant in the upper tail. There was a total of `r upper_sig_stats %>% filter(SYMBOL %in% unique(metsyn_GR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(pop, SYMBOL) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% NROW()` genes that had at least a window that was significant in the upper tail.


```{markdown, eval = FALSE}
any stat
 - lower `r lower_sig_stats %>% filter(SYMBOL %in% unique(metsyn_GR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(pop, SYMBOL) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==4) %>% NROW()`
 - upper `r upper_sig_stats %>% filter(SYMBOL %in% unique(metsyn_GR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(pop, SYMBOL) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==4) %>% NROW()`

has all stats
 - lower `r lower_sig_stats %>% filter(SYMBOL %in% unique(metsyn_GR$SYMBOL)) %>% group_by(pop, SYMBOL, statname) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON'))  %>% ungroup()%>% select(-n, -statname) %>% group_by(pop, SYMBOL) %>% tally() %>% filter(n == 5) %>% NROW()`
 - upper `r upper_sig_stats %>% filter(SYMBOL %in% unique(metsyn_GR$SYMBOL)) %>% group_by(pop, SYMBOL, statname) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON'))  %>% ungroup()%>% select(-n, -statname) %>% group_by(pop, SYMBOL) %>% tally() %>% filter(n == 5) %>% NROW()`



<!---

dendrograms

exclusivity


prop
td, fwh, and ze all 1.0 except amr (.75)


- td
- fwh
- fld
- flf
- ze

number of significant
-->





<!--
results from other diseases that are late onset and see if they produce the same patterns
-->
```



```{r, results ='asis', eval = FALSE}
# prop_list[['td_neg']]$exclusivity %>% data.frame%>% mutate(pop = rownames(.)) %>% gather("cluster", "prop", 1:6) %>% filter(prop > 0) %>% group_by(pop) %>% summarise(mean(prop))
# 
# prop_list[['td_neg']]$super_prop %>% data.frame() %>% spread(super, max_prop) 

#prop
stat_to_short <- function(x){
  switch(x,
         Tajima.D = "td",
         Fay.Wu.H = "fwh",
         Fu.Li.D = "fld",
         Fu.Li.F = "flf",
         Zeng.E = "ze",
         x
         )
}

short_to_latex <- function(x){switch(x, 
       td = "\\gls{td}",
       fwh = "\\gls{fwh}",
       fld = "\\gls{fld}",
       flf = "\\gls{flf}",
       ze = "\\gls{ze}",
       x
       )
}

bind_rows(list(
  bind_rows(lapply(names(prop_list), function(x){prop_list[[x]]$super_prop %>%  spread(super, max_prop) %>% mutate(stat = sapply(x, function(y){strsplit(y, '_')[[1]][1]}), Type = sapply(x, function(y){strsplit(y, '_')[[1]][2]}) ) %>% select(NCOL(.), 1:(NCOL(.)-1)) })), #%>% mutate_if(., is.numeric, funs(sprintf("%.3f",.))),
  
  bind_rows(lapply(names(gout[[2]]), function(x){gout[[2]][[x]]$super_prop %>% spread(super, max_prop)%>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1)) })) %>% mutate(Type = 'Urate/Gout'),
  
  bind_rows(lapply(names(t2d[[2]]), function(x){t2d[[2]][[x]]$super_prop %>% spread(super, max_prop)%>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1)) })) %>% mutate(Type = '\\gls{t2d}'),
  
  bind_rows(lapply(names(obesity[[2]]), function(x){obesity[[2]][[x]]$super_prop %>% spread(super, max_prop)%>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1)) })) %>% mutate(Type = 'Obesity'),
  
  bind_rows(lapply(names(metsyn[[2]]), function(x){metsyn[[2]][[x]]$super_prop %>% spread(super, max_prop) %>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1)) })) %>% mutate(Type = 'Metabolic Syndrome') 
  
)) %>% mutate(stat = sapply(stat, function(x){short_to_latex(stat_to_short(x))}), Type = sapply(Type, function(x){switch(x, chr = "Chromosome", neg = "Lower Tail", pos ="Upper Tail",x)})) %>% mutate_if(., is.numeric, funs(sprintf('%.2f',.))) %>%  select( Type, stat, everything()) %>% arrange(stat) %>%  data.frame %>% kable(., caption = "Proportion", escape = FALSE)#knitr::kable(., caption = "Proportion", booktabs = TRUE) %>% kableExtra::group_rows("Fu and Li's D", 1, 7) %>% kableExtra::group_rows("Fu and Li's F", 8, 14) %>% kableExtra::group_rows("Fay and Wu's H", 15, 21) %>% kableExtra::group_rows("Tajima's D", 22, 28) %>% kableExtra::group_rows("Zeng's E", 29, 35) %>% kableExtra::kable_styling(latex_options = "scale_down")
```




```{r, results = 'asis', eval = FALSE}
#exclusivity
bind_rows(list(
  bind_rows(lapply(names(prop_list), function(x){prop_list[[x]]$exclusivity %>% data.frame%>% mutate(pop = row.names(.)) %>% gather("cluster","prop",1:6) %>% filter(prop > 0) %>% group_by(pop) %>% summarise(mean_prop = mean(prop)) %>% spread(pop, mean_prop) %>% mutate(stat = sapply(x, function(y){strsplit(y, '_')[[1]][1]}), Type = sapply(x, function(y){strsplit(y, '_')[[1]][2]}) ) %>% select(NCOL(.), 1:(NCOL(.)-1)) })),
  
  bind_rows(lapply(names(gout[[2]]), function(x){gout[[2]][[x]]$exclusivity %>% data.frame%>% mutate(pop = row.names(.)) %>% gather("cluster","prop",1:6) %>% filter(prop > 0) %>% group_by(pop) %>% summarise(mean_prop = mean(prop)) %>% spread(pop, mean_prop) %>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1))})) %>% mutate(Type = 'Urate/Gout'),
  
  bind_rows(lapply(names(t2d[[2]]), function(x){t2d[[2]][[x]]$exclusivity %>% data.frame%>% mutate(pop = row.names(.)) %>% gather("cluster","prop",1:6) %>% filter(prop > 0) %>% group_by(pop) %>% summarise(mean_prop = mean(prop)) %>% spread(pop, mean_prop) %>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1))})) %>% mutate(Type = '\\gls{t2d}'),
  
  bind_rows(lapply(names(obesity[[2]]), function(x){obesity[[2]][[x]]$exclusivity %>% data.frame%>% mutate(pop = row.names(.)) %>% gather("cluster","prop",1:6) %>% filter(prop > 0) %>% group_by(pop) %>% summarise(mean_prop = mean(prop)) %>% spread(pop, mean_prop) %>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1))})) %>% mutate(Type = 'Obesity'),
  
  bind_rows(lapply(names(metsyn[[2]]), function(x){metsyn[[2]][[x]]$exclusivity %>% data.frame%>% mutate(pop = row.names(.)) %>% gather("cluster","prop",1:6) %>% filter(prop > 0) %>% group_by(pop) %>% summarise(mean_prop = mean(prop)) %>% spread(pop, mean_prop) %>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1))})) %>% mutate(Type = 'Metabolic Syndrome')    
)) %>% select(stat, Type, everything())  %>% mutate(stat = sapply(stat, function(x){short_to_latex(stat_to_short(x))}), Type = sapply(Type, function(x){switch(x, chr = "Chromosome", neg = "Lower Tail", pos ="Upper Tail",x)})) %>% mutate_if(., is.numeric, funs(sprintf("%.3f",.))) %>% arrange(Type) %>%  data.frame %>% pander::pandoc.table(., caption = "Exclusivity")#knitr::kable(., caption = "Exclusivity", booktabs = TRUE) %>% kableExtra::group_rows("Fu and Li's D", 1, 7) %>% kableExtra::group_rows("Fu and Li's F", 8, 14) %>% kableExtra::group_rows("Fay and Wu's H", 15, 21) %>% kableExtra::group_rows("Tajima's D", 22, 28) %>% kableExtra::group_rows("Zeng's E", 29, 35) 
```
\FloatBarrier

### Random draws

```{r load_rand, results='asis'}
td_random <- readRDS('~/data/NZ_coreExome_1kgp/100kbWindow_intra/td_random_filtered_3ns-18-7-2017.RDS')
rand_tab <- bind_rows(lapply(seq_along(td_random), function(x){td_random[[x]]$super_prop})) %>% group_by(super) %>% group_by(super) %>% summarise(min = round(min(max_prop),3), mean = round(mean(max_prop), 3), sd = round(sd(max_prop), 3), max = round(max(max_prop),3), prop50 = round(sum(if_else(max_prop >=0.5, 1, 0))/NROW(.) * 100, 1) , total_complete = sum(if_else(max_prop == 1, 1, 0))) %>% data.frame()
```

```{r, eval = FALSE}
#count the number of times the Polynesians were all grouped into an exclusive cluster
a <- c()
b<- c()
for(i in seq_along(td_random)){
  if(td_random[[i]]$tab %>% data.frame %>% spread(Var2, Freq) %>% select(-1) %>% filter(POL == 4 & rowSums(.)== 4) %>% NROW() > 0){a <- c(a,i)
  }
  if(td_random[[i]]$super_prop %>% filter(super == "POL" & max_prop == 1) %>% NROW() > 0){b <- c(b,i)
  }
}





```


```{r randexcl}
#exclusivity
#excl_tab <- bind_rows(lapply(seq_along(td_random), function(x){td_random[[x]]$exclusivity %>% t() %>% data.frame %>% mutate_all(., funs(replace(., . == 0, NA))) %>% summarise_all(., funs(mean(., na.rm=TRUE), min(., na.rm = TRUE), max(., na.rm = TRUE)))})) %>% summarise_all(funs(min,mean,max)) %>% t %>%data.frame() %>% select(value = 1)  %>% mutate(pop = sapply(row.names(.), function(x){strsplit(x, '_')[[1]][1]}), stat1 = sapply(row.names(.), function(x){strsplit(x, '_')[[1]][2]}), stat2 = sapply(row.names(.), function(x){strsplit(x, '_')[[1]][3]}) ) %>% filter(stat1 == stat2) %>% select(-stat2) %>% spread(stat1, value)
#saveRDS(excl_tab, 'data/04-excl_tab.RDS')
excl_tab<-readRDS('data/04-excl_tab.RDS')
#excl_tab

# bind_rows(lapply(seq_along(td_random), function(x){td_random[[x]]$exclusivity %>% t() %>% data.frame %>% summarise_all(., funs(max)) %>% filter(POL == 1)})) 
# 
# bind_rows(lapply(seq_along(td_random), function(x){td_random[[x]]$exclusivity %>% data.frame()%>% mutate(pop = row.names(.))})) %>% t()%>% select(-clust) %>% gather("pop","prop") %>% filter(prop > 0) %>% group_by(pop) %>% summarise(min(prop), quantile(prop, 0.25), mean(prop),sd(prop), median(prop),quantile(prop, 0.75) , max(prop))

#count how many iterations a super population was in an exclusive cluster
#excl_tally <- bind_rows(lapply(seq_along(td_random), function(x){td_random[[x]]$exclusivity  %>% data.frame %>% mutate(super = row.names(.)) %>% gather(cluster,prop, 1:(NCOL(.)-1)) %>%filter(prop == 1) %>%  mutate(iter = x)})) %>% group_by(super) %>% tally()
#saveRDS(excl_tally, 'data/04-excl_tally.RDS')
excl_tally <- readRDS('data/04-excl_tally.RDS')
#excl_tally

#bind_rows(lapply(seq_along(td_random)[1:10], function(x){td_random[[x]]$exclusivity %>% t() %>% data.frame %>% mutate_all(., funs(replace(., . == 0, NA))) %>% summarise_all(., funs(mean(., na.rm=TRUE), min(., na.rm = TRUE), max(., na.rm = TRUE)))})) %>% summarise_all(funs(min,mean,max)) %>% t %>%data.frame() %>% select(value = 1)  %>% mutate(pop = sapply(row.names(.), function(x){strsplit(x, '_')[[1]][1]}), stat1 = sapply(row.names(.), function(x){strsplit(x, '_')[[1]][2]}), stat2 = sapply(row.names(.), function(x){strsplit(x, '_')[[1]][3]}) ) %>% filter(stat1 == stat2) %>% select(-stat2) %>% spread(stat1, value)

```

```{r over50per, cache = TRUE}
# over50per <- bind_rows(lapply(td_random, function(x){ x[['super_prop']] %>% spread('super', 'max_prop')})) %>% filter(POL >= 2/4, EUR >= 3/6, EAS >= 3/5, AFR >= 3/6, SAS >= 3/5) %>% arrange(POL)
# saveRDS(over50per, file = 'data/04-over50per.RDS')
# over50per2 <- bind_rows(lapply(td_random, function(x){ x[['super_prop']] %>% spread('super', 'max_prop')})) %>% filter(POL >= 2/4, EUR >= 3/6, EAS >= 3/5, AFR >= 3/6, SAS >= 3/5, AMR >=2/4 ) %>% arrange(POL)
# saveRDS(over50per2, file = 'data/04-over50per2.RDS')

over50per <- readRDS('data/04-over50per.RDS')
over50per2 <- readRDS('data/04-over50per2.RDS')
```


<!-- why does the Eur never produce a max prop of 1?-->

> Slightly more intro needed

To assess the likelihood of obtaining the super population clusters by chance, 10,000 iterations of random resampling of 2500 windows per population was undertaken and then clustered. This was done so that the extreme tail clustering could be compared to the random windows in to see if the extreme window clustering performed better in exclusivity and proportion of super population clustering than by drawing windows at random. The maximum proportion of populations being clustered into their designated super population was then calculated when the tree was cut to have 6 groups. The mean proportion of populations grouped into their super population was less than 0.5 for all super populations (Table \@ref(tab:randtab)).

Not a single random draw grouped the all individual populations into all of their respective super populations. Only the populations of the \gls{afr} and \gls{eur} super populations did not each get fully grouped into a cluster. Grouping at least 50% of a super population was at best `r round(rand_tab %>% summarise(max(prop50)), digits = 0)`% of the draws but for the larger super populations (\gls{afr}, \gls{eas}, \gls{eur}, and \gls{sas}) it was on average `r round(rand_tab %>% filter(!super %in% c('POL','AMR'))%>% summarise(mean(prop50)), digits = 0)`% of the draws. There were only `r over50per %>% NROW()` iterations that had all super populations each having half or more of their individual populations assigned the same group. The mean exclusivity of the random draws was low, ranging from `r min(excl_tab$mean) %>% sprintf('%.2f', .)` to `r max(excl_tab$mean) %>% sprintf('%.2f', .)` and there were no iterations in which the Polynesian populations were all grouped together as a Polynesian exclusive cluster.


Compared to the clustering on the extreme tails, the random windows had poorer exclusivity of clusters. The populations were clustered into their super populations rarely for the random windows and did not have multiple super populations being clustered as the clustering of the extremes returned. This indicates that there is a degree of similarity of regions that are in the extremes of the distributions and the populations who share them.

```{r randtabpander, results = 'asis'}
rand_tab_pander <- rand_tab %>% mutate_at(., vars(-super, -total_complete, -prop50), funs(sprintf("%.3f", .))) %>% mutate(prop50 = sprintf('%.1f', prop50))
#colnames(rand_tab_pander) <- c('Super Population', 'Min','Mean','Std. Dev','Max','Prop50 (%)', 'Total Complete')
rand_tab_pander %>% kable(., caption = '(\\#tab:randtab) Summary of proportion of individual populations assigned to their super population cluster.', booktabs = TRUE, col.names =  c('Super Population', 'Min','Mean','Std. Dev','Max','Prop50 (%)', 'Total Complete') )

#pander::pandoc.table(rand_tab_pander, caption = '(\\#tab:randtab) *Summary of proportion of individual populations assigned to their super population cluster.* Min is the minimum proportion for the cluster from all iterations. Mean is the mean proportion of the maximum cluster proportions for all iterations. Max is the maximum proportion from all iterations. Prop50 is the number of times at least 50% of the super population was clustered together, expressed as a percentage. Total complete is the number of times the super population had all of its individual populations assigned the same cluster.', justify = c("left",rep("right", NCOL(rand_tab_pander)-1)), keep.line.breaks = TRUE)
```


```{r, eval = FALSE}
#calculate the mean proportion for the population clusters across all iterations
# remove 0 because they aren't actually a cluster
bind_rows(lapply(seq_along(td_random), function(x){td_random[[x]]$pop_clust_prop %>% data.frame()%>% mutate(clust = row.names(.))})) %>% select(-clust) %>% gather("pop","prop") %>% filter(prop > 0) %>% group_by(pop) %>% summarise(min(prop), quantile(prop, 0.25), mean(prop),sd(prop), median(prop),quantile(prop, 0.75) , max(prop))

bind_rows(lapply(seq_along(td_random), function(x){td_random[[x]]$pop_clust_prop %>% data.frame()%>% mutate(clust = row.names(.))})) %>% select(-clust) %>% gather("pop","prop") %>% filter(prop > 0) %>% group_by(pop) %>% ggplot(., aes(x =pop, y = prop)) + geom_violin()
```





```{r, eval = FALSE}
bind_rows(lapply(seq_along(td_random), function(x){td_random[[x]]$pop_clust_prop %>% data.frame()%>% mutate(clust = row.names(.), iter = x)})) 
#bind_rows(lapply(td_random, function(x){ x[['super_prop']] %>% spread('super', 'max_prop')})) %>% filter(POL >= 2/4, EUR >= 3/6, EAS >= 3/5, AFR >= 3/6, SAS >= 3/5) %>% arrange(POL)

#bind_rows(lapply(td_random, function(x){ x %>% spread('super', 'max_prop')})) %>% mutate(rn = as.numeric(rownames(.)), exp_prop = rowSums(.)/6) %>% filter(EUR > .5) %>% arrange(exp_prop)
```


Random gene lists were generated consisting of either 25 or 100 genes, 100 times each, and median centred based on the population for \gls{td}, \gls{fwh}, \gls{fld}, \gls{flf}, and \gls{ze}. Hierarchical clustering was then performed on the windows that intersected genes from the gene lists. The number of times the Polynesian super population was completely clustered is given in Table \@ref(tab:randgenes). The  Polynesian super population was completely clustered more often with the lists of 100 genes for \gls{td}, \gls{fwh}, \gls{flf}, and \gls{ze} compared to the lists of 25 genes.

The randomly selected genes that were clustered, grouped the individual populations into their super populations more often than the clustering on randomly selected windows.

```{r, cache = TRUE}
load('~/data/NZ_coreExome_1kgp/100kbWindow_intra/random_gene_clustering_filtered_3ns_sig-20-9-2017.RData')



poly_complete <- bind_rows(
  bind_rows(
    bind_rows(lapply(seq_along(random_genes100), function(x){random_genes100[[x]][[2]]$Tajima.D$super_prop %>% spread(super, max_prop) }))  %>% mutate(statname = 'TajimaD') , 
    bind_rows(lapply(seq_along(random_genes100), function(x){random_genes100[[x]][[2]]$Fay.Wu.H$super_prop  %>% spread(super, max_prop) })) %>% mutate(statname = 'Fay.Wu.H'),
    bind_rows(lapply(seq_along(random_genes100), function(x){random_genes100[[x]][[2]]$Fu.Li.D$super_prop %>% spread(super, max_prop) })) %>% mutate(statname = 'Fu.Li.D'),
    bind_rows(lapply(seq_along(random_genes100), function(x){random_genes100[[x]][[2]]$Fu.Li.F$super_prop  %>% spread(super, max_prop) })) %>% mutate(statname = 'Fu.Li.F'),
    bind_rows(lapply(seq_along(random_genes100), function(x){random_genes100[[x]][[2]]$Zeng.E$super_prop  %>%spread(super, max_prop) })) %>% mutate(statname = 'Zeng.E')
  ) %>%mutate(total = select_if(., is.numeric) %>% rowSums) %>% filter(POL == 1) %>% group_by(statname) %>% tally() %>% mutate(gene_n = 100),
  bind_rows(
    bind_rows(lapply(seq_along(random_genes25), function(x){random_genes25[[x]][[2]]$Tajima.D$super_prop %>% spread(super, max_prop) }))  %>% mutate(statname = 'TajimaD') , 
    bind_rows(lapply(seq_along(random_genes25), function(x){random_genes25[[x]][[2]]$Fay.Wu.H$super_prop  %>% spread(super, max_prop) })) %>% mutate(statname = 'Fay.Wu.H'),
    bind_rows(lapply(seq_along(random_genes25), function(x){random_genes25[[x]][[2]]$Fu.Li.D$super_prop %>% spread(super, max_prop) })) %>% mutate(statname = 'Fu.Li.D'),
    bind_rows(lapply(seq_along(random_genes25), function(x){random_genes25[[x]][[2]]$Fu.Li.F$super_prop  %>% spread(super, max_prop) })) %>% mutate(statname = 'Fu.Li.F'),
    bind_rows(lapply(seq_along(random_genes25), function(x){random_genes25[[x]][[2]]$Zeng.E$super_prop  %>%spread(super, max_prop) })) %>% mutate(statname = 'Zeng.E')
  ) %>% 
    mutate(total = select_if(., is.numeric) %>% rowSums) %>% filter(POL == 1) %>% group_by(statname) %>% tally() %>% mutate(gene_n = 25)
) 


perc_complete <- bind_rows(
  bind_rows(
    bind_rows(lapply(seq_along(random_genes100), function(x){random_genes100[[x]][[2]]$Tajima.D$super_prop %>% spread(super, max_prop) }))  %>% mutate(statname = 'TajimaD') , 
    bind_rows(lapply(seq_along(random_genes100), function(x){random_genes100[[x]][[2]]$Fay.Wu.H$super_prop  %>% spread(super, max_prop) })) %>% mutate(statname = 'Fay.Wu.H'),
    bind_rows(lapply(seq_along(random_genes100), function(x){random_genes100[[x]][[2]]$Fu.Li.D$super_prop %>% spread(super, max_prop) })) %>% mutate(statname = 'Fu.Li.D'),
    bind_rows(lapply(seq_along(random_genes100), function(x){random_genes100[[x]][[2]]$Fu.Li.F$super_prop  %>% spread(super, max_prop) })) %>% mutate(statname = 'Fu.Li.F'),
    bind_rows(lapply(seq_along(random_genes100), function(x){random_genes100[[x]][[2]]$Zeng.E$super_prop  %>%spread(super, max_prop) })) %>% mutate(statname = 'Zeng.E')
  ) %>%mutate(total = select_if(., is.numeric) %>% rowSums) %>% filter(POL >= 4/4, EUR >= 6/6, EAS >= 5/5, AFR >= 6/6, SAS >= 5/5, AMR >= 4/4) %>% group_by(statname) %>% tally() %>% mutate(gene_n = 100),
  bind_rows(
    bind_rows(lapply(seq_along(random_genes25), function(x){random_genes25[[x]][[2]]$Tajima.D$super_prop %>% spread(super, max_prop) }))  %>% mutate(statname = 'TajimaD') , 
    bind_rows(lapply(seq_along(random_genes25), function(x){random_genes25[[x]][[2]]$Fay.Wu.H$super_prop  %>% spread(super, max_prop) })) %>% mutate(statname = 'Fay.Wu.H'),
    bind_rows(lapply(seq_along(random_genes25), function(x){random_genes25[[x]][[2]]$Fu.Li.D$super_prop %>% spread(super, max_prop) })) %>% mutate(statname = 'Fu.Li.D'),
    bind_rows(lapply(seq_along(random_genes25), function(x){random_genes25[[x]][[2]]$Fu.Li.F$super_prop  %>% spread(super, max_prop) })) %>% mutate(statname = 'Fu.Li.F'),
    bind_rows(lapply(seq_along(random_genes25), function(x){random_genes25[[x]][[2]]$Zeng.E$super_prop  %>%spread(super, max_prop) })) %>% mutate(statname = 'Zeng.E')
  ) %>% 
    mutate(total = select_if(., is.numeric) %>% rowSums) %>%filter(POL >= 4/4, EUR >= 6/6, EAS >= 5/5, AFR >= 6/6, SAS >= 5/5, AMR >= 4/4)%>% group_by(statname) %>% tally() %>% mutate(gene_n = 25)
)
```

```{r}
poly_complete %>% left_join(., perc_complete %>% select(allcomplete = n, everything()), by = c('statname','gene_n'))%>% mutate(allcomplete = ifelse(is.na(allcomplete),0,allcomplete)) %>% select("Statistic" = statname, "Genes (n)" = gene_n, 'POL Complete (%)' = n, 'All Complete (%)' = allcomplete, everything()) %>% kable(., booktabs = TRUE, caption = '(\\#tab:randgenes) Percentage of draws from randomly selected genes for completely clustered super populations.')
```




\clearpage

## Discussion and Conclusions

_Summary of Results_

Overall, it was found that the clustering of selection test statistics for entire chromosomes did not resemble the grouping that clustering on F~ST~ produced. However, using windows from the extremes of the distribution was effective at clustering the individual populations into their super population groups. \gls{fwh} was the only statistic that grouped all individual populations into their corresponding super populations, whereas \gls{td} and \gls{ze} had the same split of the \gls{amr} populations as with the F~ST~ clustering. The haplotypic based methods of \gls{ihs} and \gls{nsl} clustered into super populations and had the \gls{amr} population split too. There was not a high amount of region sharing outside of each super population. Clustering from gene lists consistently grouped the individual populations into their super populations although there was inconsistency for the \gls{amr} populations. With the exception of the \gls{amr} populations, the clustering of populations using windows or haplotypes was consistent with the results in the @1KGP2015snp (Supplementary table 5). 

<!--windows
correct:
td: both split amr, clustered others
fwh: upper split amr, lowre did not, all pops clustered
fld: lower split amr, upper only clustered afr, eas, and sas entirely
flf: lower split amr, nzm/mxl/pel. upper split amr clustered others
ze: both split amr, clustered others
-->
It is difficult to apply a biological interpretation onto what the clustering of the most extreme windows represents other than a similarity in allele frequencies between populations that have a pattern that is consistent with the out of Africa model of human migration [@Mallick2016; Pagani2016]. The extremes of the distributions represent different types of selection or demographic events and are different depending on the selection test statistic being used [@Tajima1989; @fay2000hitchhiking; @Fu1993; @Zeng2006]. The upper end of the distribution is able to be compared as all of the values that it encompassed were positive in value across all populations for all statistics and represent the same types of selection or demography. The lower extreme however was a mix of negative and positive values, with the positive values mostly from the \gls{afr} populations. This mix means that the windows from one population do not necessarily represent the same selective or demographic event component and therefore are not directly comparable.

The simplest reason that explains the similar clustering that occurred with the gene lists is that for the tests of selection that rely on the frequency spectrum, populations that are similar have a similar allele frequency to each other. This would explain why with \gls{td}, \gls{fwh}, and \gls{ze}, which are based on comparing ratios of low, intermediate and high frequency variants, that they all grouped the individual populations into their super population groups. This differed to the statistics based on singletons (\gls{fld} and \gls{flf}) which did not often group the super populations.

Under a neutral model the differences in allele frequency are due to genetic drift [@kimura1979neutral]. Demographic events such as bottleneck, migration, expansion, and admixture also contribute to the differences in allele frequencies found between populations [@ref; @Wright1951]. Selection on the other hand acts in a targeted manner so therefore if the selection occurred post population split then it should look neutral in one population but not in the other, however if the selective event happened in an ancestral population of the two then dependent on time since divergence, it would be expected to be observed in both populations [@ref]. However it is also possible to have 'parallel' selection, where both populations experience independent selection [@Tennessen2011]. 

- Tennessen2011 might be relevant to above
- Bierne2013 includes info on fst outliers
- Verdu2014 might be useful for admixture/clustering
- Pickrell2012 has a model that they used to create a tree


However, what was observed was the populations being clustered into their super populations. This would indicate that a) this was an ancient selective event prior to the populations diverging and subsequent population specific recombination [@ref]. Or b) there was no selective event and the change in allele frequencies due to drift after population divergences was enough to distinguish the populations from each other [@ref]. There were only a few loci from the gene lists that had met the thresholds for significance to indicate possible selection, this would indicate that the most likely explanation is drift was responsible for the differences in allele frequency. This can be seen in the fact that the populations from each super population had similar distributions of the selection statistics but these distributions where shifted compared to the other super populations. It was for this reason that the gene list clustering using the actual values needed to be centered using the population median. The fact that the other disease gene lists for diseases that are late onset and thus have minimal selective pressure on reproductive success, also exhibited the same group clustering.


 _Limitiations_

- use of chip
- non-imputed

Both \gls{flf} and \gls{fld} had a very sparse lower tail and this would influence how well the clustering performed. At the extremes of similarity (e.g. very similar or very dissimilar), the clusters were determined by only a few differences which means a large amount of information was not being used for the decision. As the lower tail represents and excess of singletons this is likely to have been impacted on by using chip data as there was a very strong bias against them during the quality control protocol [@Guo2014]. In this situation because the data has reduced singletons, the distribution for \gls{fld} and \gls{flf} was shifted to the right, with the upper tail representing a deficit of singletons [@Fu1993]. This shift now means there is an under representation of regions in the lower tail and the clustering was no better than drawing regions at random.

The choice of out-group for \gls{fwh} can also have an effect as it is used to root the tree <!-- is it or is it used to work out what the frequency is? -->. In this case the out-group was a single sample created that was homozygote for the ancestral allele

The cutree method used to assess the proportions and exclusivity of the clustering was relatively crude and did not always match the clusters that were visible in the dendrogram. The use of k = 6 was due to there being 6 super populations, however, \gls{clm}, \gls{pur}, \gls{mxl}, and \gls{pel} being classed as a single super population was not reflected in the clustering. Many of the selection tests split \gls{amr} into 2 groups, CLM/PUR and MXL/PEL. This pairing is consistent with what had previously been found from similarity of admixture and F~ST~ [@Gravel2013; @Auton2015 ]. These \gls{amr} populations had also been identified as being admixed and so it was interesting that for upper tail for both Fu and Li's _D_ and _F_ they were clustered with the \gls{nzm} population which is also know to be admixed, mostly with New Zealand Caucasian [@Hollis-Moffatt2012; @Gravel2013].


urate has a relationship with dementia see @Latourte2017 for more info
elevated SUA is neuro protective for PD  [@Shen2013pd] and AD [@Chen2014], although the AD meta found no sig diff but suggestice that lower 


It can be concluded that the extremes of the distributions for the selection statistics can be used to group populations into their super populations.

Genes that are associated with disease also perform well at grouping populations into their super population groups.

Random regions do not perform well at clustering the super populations. 

Gautier2013 in the intro talks about clustering

<!--
- **what does this answer??**
- using the chromosome wide results did not always return the super clusters.
- often see admixed amr pops grouped with nzm/cim
- clustering using the gout/urate/bmi/t2d/metsyn genes...
- are the genes found more than you would expect by chance?
- cutree method is rather crude
- what are the strengths/weaknesses of each stat
- Fu and Li's D doesn't cluster well
- when clustering on actual values how much does it matter a population has shifted overall distribution? - centre on the median to help fix this



Clustering on the extreme values returned groupings that were consistent with the super population groups.

The clustering of the disease related genes gave the groupings that, with the exception of \gls{fld} were consistent with the super populations.

Using the extremes of the distributions and the gene lists for gout and co-morbidities had clustering that was consistent with the super population groups and largely matched. Clustering on chromosome wide F~ST~ did give the super population groupings and also split the \gls{amr} populations. This was not the case for \gls{td}, \gls{fwh}, \gls{fld}, \gls{flf}, and \gls{ze} using entire chromosomes, which did not produce groups that were consistent with the super population groups. \Gls{fwh} and \gls{ze} chromosome wide did the best at grouping the populations. 

Diseases with a late on-set were chosen as these should have little to no selection acting on them due to affecting the individuals post reproduction age. Clustering using the genes associated with these diseases had very similar groupings to the genes associated with gout and 



clustering on actual values because the stats are based on the underlying frequency distribution are likely to reflect this distribution and not the actual state of selection - therefore I would assume that given any list of genes sufficiently long (say >10), that the population clusters would appear. If the selection event occurred post out of Africa, then there would be some expectation that the clustering would. 

effect size of allele vs allele frequency: Park2011
Human Demographic History Impacts Genetic Risk Prediction across Diverse Populations: Martin2017
Understanding rare and common diseases in the context of human evolution: Quintana-Murci2016 
A Genetic Atlas of Human Admixture History - Hellenthal2014
Mutation Rate Variation is a Primary Determinant of the Distribution of Allele Frequencies in Humans - Harpak2016
Demographic History of Oceania Inferred from Genome-wide Data - Wollstein2010

this is exploratory, not predictive or creating a model, but is evidence that the Polynesian populations are similar most to each other
<!-- what ends of the distribution represent the different types (positive, balancing, negative selection) -->




```{r, results = 'asis', eval = FALSE}
bind_rows(list(
  
)) %>% mutate_if(., is.numeric, funs(sprintf('%.2f',.)))%>%  select(stat, "Gene List" = Disease, everything()) %>% data.frame %>%  pander::pandoc.table(., caption = 'Proportion')

```

```{r, eval = FALSE}
bind_rows(list(
  bind_rows(lapply(names(gout[[2]]), function(x){gout[[2]][[x]]$super_prop %>% spread(super, max_prop)%>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1)) })) %>% mutate(Disease = 'Urate/Gout'),

bind_rows(lapply(names(t2d[[2]]), function(x){t2d[[2]][[x]]$super_prop %>% spread(super, max_prop)%>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1)) })) %>% mutate(Disease = '\\gls{t2d}'),

bind_rows(lapply(names(obesity[[2]]), function(x){obesity[[2]][[x]]$super_prop %>% spread(super, max_prop)%>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1)) })) %>% mutate(Disease = 'Obesity'),

bind_rows(lapply(names(metsyn[[2]]), function(x){metsyn[[2]][[x]]$super_prop %>% spread(super, max_prop) %>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1)) })) %>% mutate(Disease = 'Metabolic Syndrome')
)) %>% mutate_if(., is.numeric, funs(sprintf('%.2f',.)))%>%  select(stat, "Gene List" = Disease, everything()) %>% gather("pop",'prop',3:NCOL(.)) %>%  left_join(., bind_rows(list(
  bind_rows(lapply(names(gout[[2]]), function(x){gout[[2]][[x]]$exclusivity %>% data.frame%>% mutate(pop = row.names(.)) %>% gather("cluster","prop",1:6) %>% filter(prop > 0) %>% group_by(pop) %>% summarise(mean_prop = mean(prop)) %>% spread(pop, mean_prop) %>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1))})) %>% mutate(Disease = 'Urate/Gout'),

bind_rows(lapply(names(t2d[[2]]), function(x){t2d[[2]][[x]]$exclusivity %>% data.frame%>% mutate(pop = row.names(.)) %>% gather("cluster","prop",1:6) %>% filter(prop > 0) %>% group_by(pop) %>% summarise(mean_prop = mean(prop)) %>% spread(pop, mean_prop) %>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1))})) %>% mutate(Disease = '\\gls{t2d}'),

bind_rows(lapply(names(obesity[[2]]), function(x){obesity[[2]][[x]]$exclusivity %>% data.frame%>% mutate(pop = row.names(.)) %>% gather("cluster","prop",1:6) %>% filter(prop > 0) %>% group_by(pop) %>% summarise(mean_prop = mean(prop)) %>% spread(pop, mean_prop) %>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1))})) %>% mutate(Disease = 'Obesity'),

bind_rows(lapply(names(metsyn[[2]]), function(x){metsyn[[2]][[x]]$exclusivity %>% data.frame%>% mutate(pop = row.names(.)) %>% gather("cluster","prop",1:6) %>% filter(prop > 0) %>% group_by(pop) %>% summarise(mean_prop = mean(prop)) %>% spread(pop, mean_prop) %>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1))})) %>% mutate(Disease = 'Metabolic Syndrome')
)) %>% mutate_if(., is.numeric, funs(sprintf('%.2f',.)))%>%  select(stat, "Gene List" = Disease, everything()) %>% gather("pop","excl",3:NCOL(.)), by = c('stat',"Gene List","pop")) %>% filter(excl > 0.99, prop > 0.99) %>% group_by(pop) %>% tally()
```

