# Clustering of Selection Statistics

```{r, include = FALSE, echo = FALSE}
knitr::opts_chunk$set(echo = FALSE)
library(tidyverse)
library(ggdendro)
source('scripts/multiplot.R')
source('scripts/04_clustering.R')
```

<!--
key tasks to complete:
- update introduction WRT embedded comments
- df with summary stats about number of windows for each population
- df with distribution of treecuts proportions
- genes in the tails
-->


In this chapter I will investigate the idea that populations that share ancestry will display similar signals of selection. To test this Idea I will be clustering the various selection statistics and then comparing the clusters to current population migratory history.

## Introduction

<!--
* introduce the out of africa history
* cover the pacific migrations
* Near versus Remote Oceania
* migration to NZ from pacific islands
-->

“Signatures of selection” in a population can be identified in regions of the genome that exhibit a reduction in genetic variability [@Smith1974;@Kaplan1989;@McVean2007]. This reduction in genetic variation can arise when the phenotype of a neutral benefit allele experiences a favourable change in environmental conditions or a new allele arises conveying a selective advantage [@Hermisson2005].  This results in an increased frequency of both the allele, and linked sites, within a population [@Smith1974]. Genome-wide scans for signatures of selection have shown geographically similar populations share similar signatures [@Coop2009; @pickrell2009signals]. Geography can inhibit migration leading to local adaptations affecting the allele frequency of local populations, leaving the non-local population allele frequency unaffected [@Coop2009]. New Zealand Polynesians are relatively geographically isolated, with a recent settlement history. 
<!-- need to update with Lisa's comments which were
- remote oceania is beyond solomon islands
- need a sentence to explain why we have Polynesians living in NZ -> ref the chapter about 1960's pacific immigrants to NZ
- explain NZ popln of Maori and Polynesian populations
- mention CIM settlement
- explain east/west split
-->

The history starts with migration out of Africa (50-100 kya) [@Nielsen2017], tracking up to the Levant and Arabian peninsula (? kya) <!-- same with this -->and splitting into South Asia, Indonesia, and Australia (50kya) [@Kivisild1999; @Quintana-Murci1999]. Concurrently the migration  to Europe was occurring (45 kya). From South Asia the migration continued with the peopling of East Asia (20kya) [@Groucutt2015]. <!--Lisa: The Pacific – or if you really are talking about Remote Oceania, you are talking about Lapita, which first appears in Near Oceania about 3300BP where they interact with the peoples who have been there for at least 40,000 years and then by 3,000 they are the first to establish settlement on the islands of New Caledonia, Vanuatu, Fiji, Samoa and Tonga. THEN around 1-1.2 kya we have the settlement of the rest of the Polynesian triangle, including the Cook Is avout 1,000 BP and New Zealand about 750 - 800 BP. --> The Pacific was settled in two events, the first was the settlement of Near Oceania (50 kya) followed by the settlement of Remote Oceania reaching Samoa and Tonga during the Lapita expansion (3 kya, [@Matisoo-Smith2015 ;  @Skoglund2016]) then the rest of the Polynesian triangle settled (1 – 1.2 kya, [@Wilmshurst2011]). The settlement of New Zealand by \gls{nz} Maori occurred 800 ya [@Duggan2014; @Matisoo-Smith2015].

<!--
need something about what consitutes east and west polynesia and why that matters
-->

For largely economic reasons, during the 1950's and 60's there was a rural to urban migration of people in the Pacific. As part of this, many settled in New Zealand from Samoan, Tongan and the Cook Islands in search of work and now the populations in New Zealand largely outnumber the populations of their islands origin [@MatisooSmith2012highway]. Within \gls{nz}, the Polynesian populations are mostly focused within the Auckland region [@Barcham2009]

<!--ref for pacific in NZ in 1960's [@MatisooSmith2012highway] 
The modern population of \gls{nz} Maori 


-->

New Zealand Polynesians have inherent elevated serum urate levels and higher prevalence of gout [@Winnard2012; @Winnard2013a]. Polynesian populations share a common genetic ancestry with East Asia however there are Polynesian specific variants associated with increased risk to gout and elevated serum urate [@Phipps-Green2010; @Hollis-Moffatt2012]. Serum urate has been associated with metabolic disorders [@Choi2007b] and genes involved with complex diseases such as \gls{t2d}, gout, and other metabolic related disorders have shown evidence for selection [@Zhang2013a; @pickrell2009signals]. 

It is hypothesised that elevated serum urate may have undergone positive selection in Polynesians populations due to some of the beneficial properties [@Gosling2014], such as its role as a powerful anti-oxidant [@Ames1981], or as an adjuvant for the innate immune system [@Opitz2009]. Serum urate has also been suggested to have a protective neurological effect [@Euser2009]. Characterisation of selection within Polynesian populations has been largely limited to population differentiation due to sample sizes [@Kimura2008; @Mallick2016].


<!-- tie in urate/gout/obesity/t2d so that it makes sense to use those gene lists -->

Here we investigate genetic selection of serum urate in Polynesian populations and if there is a selection signature shared amongst Polynesian populations.


## Methods

### Frequency Spectrum

Tajima's _D_, Fay and Wu's _H_, Fu and Li's _D_, Fu and Li's _F_, Zeng's _E_, and F~ST~ were all calculated using PopGenome v2.2.3 [@Pfeifer2014]. Excepting F~ST~, these statistics were calculated using a window size of 100 kb and a slide of 10 kb for each entire chromosome. A filter to remove windows containing fewer than 4 segregating sites was applied. Windows were then updated to have the central 10 kb used as coordinates for the boundary to the window. For each selection test statistic, windows were ordered by test statistic value and the most extreme 1% of values were selected from each tail of the distribution. For the windowed  approach, hierarchical clustering was then performed on a binary matrix derived from the presence or absence of the windows across all populations. Hierarchical clustering using the whole chromosome values used the statistic value itself to create the distance matrix. The distance measure used was euclidean distance and the linkage criteria was complete linkage for both sets of clustering. Accuracy of the clustering was assessed by descending the dendrogram until there 6 groups could be made and then caluclating the maximum proportion of the individual populations of the same super population assigned to the same cluster. Exclusivity of each cluster was calculated using the proportion of populations from a given super population compared to the total number of individual populations assigned that cluster. F~ST~ was calculated pairwise for all populations for each chromosome.

### Haplotypic

\glsdesc{ihs} and \gls{nsl} were calculated and normalised using selscan v1.1.0b [@Szpiech2014] as part of the selectionTools 1.1 [@Cadzow2014] pipeline as described in Section \@ref(selectionTools). Markers with an \gls{ihs} or \gls{nsl} that were determined to be significant using a significance threshold of 1% were clustered into genomic regions using the DBSCAN package v1.1.1 [@dbscanref] in R where nearby \gls{snp}s were assigned the same group id. The search radius used was 200 kb and the minimum number of points for a cluster was 1.  Cluster regions were created by taking the minimum and maximum position for each group id, population and chromosome. The distance matrix was the result of calculating P(A $\cap$ B)/ P(A) and P(A $\cap$ B) / P(B), where P(A $\cap$ B) is the sum of the intersection of clustered regions between population A and population B, and P(A) and P(B) is the sum of the clustered regions for population A and population B respectively. Hierarchical clustering was then performed using euclidean distance as the distance measure, and complete linkage for the linkage criteria.


### Disease Associated Genes

Disease associated markers and their annotated genes were downloaded from the \gls{gwas} catalog^[https://www.ebi.ac.uk/gwas/, accessed 19 June 2017][@MacArthur2017]  and filtered for P < 5x10^-8^. See Table \@ref(tab:gwascataccess)  for a complete list of SNPs and references. Gene lists were then created for each of the following diseases or traits; urate and gout, \gls{t2d}, obesity, and metabolic syndrome by filtering the study disease trait on keywords. Table \@ref(tab:gwascatref) shows source disease trait and reference used to create categories of gene lists for gout/urate, obesity, \gls{t2d}, and metabolic syndrome. The gene lists from these categories were used in Section \@ref(genelistcluster) for clustering using the selection test statistic values for windows that intersected these gene regions.

(ref:gwascataccess) https://www.ebi.ac.uk/gwas/, accessed 19 June 2017




<!-- 
TODO: add column for category
-->

```{r, results = 'asis'}
m <- create_gwas_cat_table() %>% data.frame() %>% mutate("Category" = NA) %>%  arrange(DISEASE.TRAIT, ref) %>% select(-REPORTED.GENE.S., Category, 'Disease_Trait' = DISEASE.TRAIT, 'Reference' = ref) %>% distinct()
#colnames(m) <- c("Disease\nTrait", "Reported\nGenes","Reference")  
pander::pandoc.table(m, split.cells = 30 , caption = "(\\#tab:gwascatref) GWAS catalog studies used. Disease trait as specified in GWAS catalog", keep.line.breaks = TRUE)
```


## Results

### Frequency Spectrum

#### Whole Chromosome Clustering

Clustering on the full chromosome results for all chromosomes from PopGenome (Figure \@ref(fig:multichrplot)) did not group the populations of each super population together, with the exception of Fay and Wu's _H_, which grouped all super populations although split the \gls{amr} into 2 pairs. Tajima's _D_ grouped the \gls{eur} and most of the \gls{sas} super populations, and had smaller sub-super population groupings of 2-3 populations. \Gls{cim}, \gls{nzm}, and \gls{sam} were grouped with 2 \gls{amr} populations, \gls{pel} and \gls{mxl}, and this group was the least similar to the others. There was no clear pattern to the clustering order for both Fu and Li's _D_ and _F_, although Fu and Li's _F_ did have a few small groups of the same super population of no more than 3 individuals populations. The \gls{cim} and \gls{sam} populations were grouped together in both of statistics. Zeng's _E_ clustered the \gls{afr} populations together and this cluster was the least similar to the other populations. There was also a grouping of the \gls{eas} populations which had the Western Polynesian populations part of it.

<!--
- td had eur cluste, nzm-cim, most of a sas, but the other populations are in groups of 2-3, the pol group is quite disimilar to the rest
- fwh has the super populations but has AMR split in half. eas and pol are together and quite disimilar to the rest
- fld looks almost random, nzm-gbr, maybe admixed pops being grouped together
- flf has pockets fo 2-3 small groupings of the super populations, nzm-mxl-pel
- flf and fld have a very similar first split
- ze has afr cluster and is most disimilar. eas cluster has the wp integrated. nzc-clm-pur, eur cluster 
-->

(ref:multichrplot) _Whole Chromosome Clustering._ Hierarchical clustering of the populations using selection test statistics for each entire chromosome. Clustering was performed using euclidean distance for the distance measure and complete linkage for the linkage criteria. Selection test statistic is in the label for each dendrogram. Coloured by super population.

<!--
TODO: make same theme as other dendrograms
-->

```{r multichrplot, fig.cap='(ref:multichrplot)', echo = FALSE, fig.height=8}
multiplot(gene_dendros(gene_data = mat_d_list[['td_chr']]) + ggtitle("Tajima's D"),
          gene_dendros(gene_data = mat_d_list[['fwh_chr']]) + ggtitle("Fay and Wu's H"),
          gene_dendros(gene_data = mat_d_list[['fld_chr']]) + ggtitle("Fu and Li's D"),
          gene_dendros(gene_data = mat_d_list[['flf_chr']]) + ggtitle("Fu and Li's F"),
          gene_dendros(gene_data = mat_d_list[['ze_chr']]) + ggtitle("Zeng's E"),
   cols = 2)
```


#### F~ST~

(ref:wholechrfst) _F~ST~_ Hierachical clustering on chromosomal F~ST~ using euclidean distance and complete linkage. Coloured by super population.

```{r wholechrfst, fig.cap="(ref:wholechrfst)"}
create_fst_dendro() + ggtitle(expression("Whole Chromosome F"["ST"]))
```

```{r}
fst <- readRDS('~/data/NZ_coreExome_1kgp/whole_chr_fst_popgenome.3-7-2017.RDS')

fst_mat <- bind_rows(lapply(fst, function(x){x %>% mutate(pop1 = as.character(pop1), pop2 = as.character(pop2))%>% select(-p1,-p2) %>% filter(!pop1 %in% c('WPN','EPN','NAD'), !pop2 %in% c('WPN','EPN','NAD')) %>% left_join(., panel %>% mutate(super_pop = paste(super_pop, pop, sep = '_')) %>% select("pop1" = pop, super_pop) %>% distinct(), by = "pop1" ) %>% select(-pop1) %>% spread(., "super_pop", fst )}))

fst_means <- fst_mat %>% filter(pop2 %in% c('CIM','NZM','SAM','TON')) %>% left_join(., panel %>% mutate(pop2 = pop, p2 = paste0(super_pop,"_",pop)) %>% select(pop2,p2) %>% distinct(), by = "pop2" ) %>% mutate(pop2 = p2) %>% select(pop2, 1:(NCOL(.)-1)) %>% gather("pop1","fst", 3:NCOL(.)) %>%   mutate(sup1 = sapply(pop1, function(x){strsplit(x, '_')[[1]][1]}), sup2 = sapply(pop2, function(x){strsplit(x, '_')[[1]][1]}) ) %>% filter(sup1 != sup2) %>%  group_by(sup2, sup1) %>% summarise(mean = mean(fst) %>% round(., digits = 3)) %>% ungroup()
```


Heriarchical clustering performed on pairwise F~ST~ for the whole chromosomes between each individual population grouped all of the super populations, except \gls{amr} into their correct groups (Figure \@ref(fig:wholechrfst)). The \gls{afr} populations were most similar to each other but least similar to the rest of the populations. The Polynesians populations were grouped together and were subgrouped into East and West Polynesian. The largest differentiation was between XXX and XXX, where as the smallest differentiation was between XXX and XXX.

The mean F~ST~ for \gls{pol} and \gls{afr} was `r fst_means %>% filter(sup2 == "POL", sup1 == "AFR") %>% select(mean)`; \gls{pol} and \gls{amr} was `r fst_means %>% filter(sup2 == "POL", sup1 == "AMR") %>% select(mean)`; \gls{pol} and \gls{eas} was `r fst_means %>% filter(sup2 == "POL", sup1 == "EAS") %>% select(mean)`; \gls{pol} and \gls{eur} was `r fst_means %>% filter(sup2 == "POL", sup1 == "EUR") %>% select(mean)`; \gls{pol} and \gls{sas} was `r fst_means %>% filter(sup2 == "POL", sup1 == "SAS") %>% select(mean)`.

<!--
some stats about the largest and smallest fsts and focus on the Polynesians
-->

### Hierarchical Clustering of Distribution Extremes

Windows containing the most extreme 1% of values for each selection test statistic from each side of the empirical distribution were selected to use as part of a hierarchical clustering to cluster the individual populations from the \gls{1kgp} and the Polynesian populations. Euclidean distance was used as the distance measure for clusters and complete linkage was used as the linkage method to between clusters. 

The mean number of windows per population was `r create_windowTable() %>% summarise(round(mean(mean), digits = 1))` (sd `r create_windowTable() %>% summarise(round(sd(mean), digits = 1))`) with a minimum of `r create_windowTable() %>% summarise(min(min))` and a maximum of `r create_windowTable() %>% summarise(max(max))`. To assess the statistical significance of the proportion of populations being assigned the same cluster, a series of 1000 random draws was done to establish a null model for each selection test statistic.

Table \@ref(tab:windowTable) shows the distribution of number of windows for each super population used for each selection test statistic. Table \@ref(tab:totalwindowTable) show the total number of windows used for each. 

<!--
TODO: make Tail column nice
-->
```{r, results = 'asis', echo = FALSE}
tab <- create_windowTable() %>% mutate(statname = sapply(d, function(x){paste0("\\gls{",strsplit(x, "_")[[1]][1], "}") }),  tail = sapply(d, function(x){strsplit(x, "_")[[1]][2]})) %>% select(statname, tail, super, min,mean,max) %>% mutate(tail = ifelse(tail == 'neg', "Negative","Positive"))
colnames(tab) <- c("Statistic Name","Tail", "Super Population", "Min","Mean","Max")
pander::pandoc.table(tab, caption = '(\\#tab:windowTable) Summary statistics for number of windows used in each custering analysis. Tail indicates which extreme of the distribution the windows were from. Min and max are the minimum and maximum number of windows contributed from the individual populations for the given super population. Mean is the mean number of windows for the individuals populations for the given super population.' )
```

<!-- 
TODO:
- better column names
- want to include the min/mean/max number of windows shared
-->
```{r, results = 'asis', echo = FALSE}
pander::pandoc.table(create_windowSummaryTable(), caption = '(\\#tab:totalwindowTable) Number of windows. Number of shared windows between populations'  )
```

<!--
describe number of total number of windows
polynesian specific windows
ep specific windows
wp specific windows
cim/nzm/sam/ton specific windows

proportions
exclusivity
dendrograms

refer to figure
-->

#### Tajima's _D_ 


```{r tdTop1000, fig.cap='(ref:tdTop1000)',echo = FALSE, fig.height = 7}
four_plot("td","Tajima's D")
```

(ref:tdTop1000) _\gls{td} Clustering._ Hierarchical clustering of \gls{td} using the positive and negative tails of the distribution. **A** and **B** Dendrogram representing the clusters of \gls{td} lower and upper tails respectfully using hierarchical clustering with euclidean distance and complete linkage. Coloured by super population. **C** and **D** Plots representing the windows present in each population from the lower and upper tails respectfully.

<!--
- neg gave super pop clusters
- pos gave super pop clusters
- both split the AMR into 2
- number of windows shared by POL with others was X
- number of windows shared EP/WP was X
- number of unique windows was X for CIM, NZM, SAM, TON
- genes involved?
-->
The clustering of the individual populations for 1000 most extreme values from each of the negative and positive tails of the \gls{td} distribution both showed the East/West split in the Polynesians (Figure \@ref(fig:tdTop1000)). But also showed that the Polynesian populations were most similar to each other. The negative tail has the Polynesian cluster most closely related to the \gls{afr} super population, whereas, the positive tail has the Polynesian cluster closest to the \gls{eas} cluster. The main groupings are of the populations that were designated as the same super population. Although the \gls{amr} super population in both tails has been split into two. The mean number of unique windows per population was `r round(create_sums('td_pos') %>% filter(ALLsum == 1) %>% select(-contains('chrom'), -posid, -contains('sum')) %>% summarise_all(funs(sum)) %>% t() %>% mean(), digits = 1)` in the positive tail, compared with `r round( create_sums('td_neg') %>% filter(ALLsum == 1) %>% select(-contains('chrom'), -posid, -contains('sum')) %>% summarise_all(funs(sum)) %>% t() %>% mean(), digits = 1)` in the negative tail. The groupings for both the positive an negative tails exclusively grouped the \gls{afr}, \gls{sas} and \gls{pol} super populations. The \gls{amr} super population was split into an exclusive group of \gls{mxl} and \gls{pel}, and a group of \gls{clm} and \gls{pur} which shared was shared with the \gls{eur} super population. 



_Negative values_

The number of windows that were unique to the Polynesian super population was `r create_sums('td_neg') %>% filter(POLsum == ALLsum) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)`, representing `r round(create_sums('td_neg') %>% filter(POLsum == ALLsum) %>% select(contains('chrom'), contains('sum')) %>% NROW(.) / mat_d_list[['td_neg']] %>% NROW(.) *100, digits = 1)`% of the lower most extreme windows. Of those, `r create_sums('td_neg') %>% filter(POL_CIM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` were specific to \gls{cim}, `r create_sums('td_neg') %>% filter(POL_NZM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` were specific to \gls{nzm}, `r create_sums('td_neg') %>% filter(POL_SAM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` were specific to \gls{sam}, and `r create_sums('td_neg') %>% filter(POL_TON == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` were specific to \gls{ton}.

The Eastern Polynesians had `r create_sums('td_neg') %>% filter((POL_CIM + POL_NZM) == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` unique windows with `r create_sums('td_neg') %>% filter((POL_CIM + POL_NZM) == ALLsum & ALLsum == 2) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` of those shared between \gls{cim} and \gls{nzm}. Where as the Western Polynesians had `r create_sums('td_neg') %>% filter((POL_SAM + POL_TON) == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows, with `r create_sums('td_neg') %>% filter((POL_SAM + POL_TON) == ALLsum & ALLsum == 2) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` shared between \gls{cim} and \gls{nzm}.

There were `r create_sums('td_neg') %>% filter(POLsum < ALLsum & ALLsum > 4) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows where all the Polynesian populations shared a window with a non-Polynesian population. This is compared to super population unique window numbers of `r create_sums('td_neg') %>% filter(AFRsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)`, `r create_sums('td_neg') %>% filter(AMRsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)`, `r create_sums('td_neg') %>% filter(EASsum == ALLsum & ALLsum  > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)`, `r create_sums('td_neg') %>% filter(EURsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)`, `r create_sums('td_neg') %>% filter(SASsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` for the \gls{afr}, \gls{amr}, \gls{eas}, \gls{eur}, and \gls{sas} super populations respectively.  


The clustering grouped each of \gls{afr}, \gls{eas}, \gls{eur}, and \gls{sas} in their entirety. The \gls{amr} super population was split into 2, with \gls{clm} and \gls{pur} being grouped with \gls{eur}. All clusters except the \gls{amr} - \gls{eur} cluster were each exclusive to a single super population.

```{r, eval = FALSE}
# untested/bad assumption that all genes will be in here
create_sums('td_neg') %>% filter(POLsum == ALLsum & POLsum > 0) %>% left_join(., lower_sig_stats %>% mutate(chrom = as.numeric(chrom)) %>% select(contains('chrom'), SYMBOL) %>% distinct(), by = c('chrom', 'chrom_start', 'chrom_end')) %>% group_by(SYMBOL) %>% tally() %>% filter(!is.na(SYMBOL)) %>% NROW(.)

create_sums('td_neg') %>% filter(POL_CIM + POL_NZM == ALLsum & ALLsum == 2) %>% left_join(., lower_sig_stats %>% mutate(chrom = as.numeric(chrom)) %>% select(contains('chrom'), SYMBOL) %>% distinct(), by = c('chrom', 'chrom_start', 'chrom_end')) %>% group_by(SYMBOL) %>% tally() %>% filter(!is.na(SYMBOL)) %>% NROW(.)

create_sums('td_neg') %>% filter(POL_SAM + POL_TON == ALLsum & ALLsum == 2) %>% left_join(., lower_sig_stats %>% mutate(chrom = as.numeric(chrom)) %>% select(contains('chrom'), SYMBOL) %>% distinct(), by = c('chrom', 'chrom_start', 'chrom_end')) %>% group_by(SYMBOL) %>% tally() %>% filter(!is.na(SYMBOL)) %>% NROW(.)
```


```{r, eval = FALSE}
# untested/bad assumption that all genes will be in here
(td_neg_pol_genes <- create_sums('td_neg') %>% filter(POLsum == 4) %>% left_join(., lower_sig_stats %>% mutate(chrom = as.numeric(chrom)) %>% select(contains('chrom'), SYMBOL) %>% distinct(), by = c('chrom', 'chrom_start', 'chrom_end')) %>% group_by(SYMBOL) %>% tally())
```

_Positive values_

The positive tail had a total of `r create_sums('td_pos') %>% filter(POLsum == ALLsum) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows, or `r round(create_sums('td_pos') %>% filter(POLsum == ALLsum) %>% select(contains('chrom'), contains('sum')) %>% NROW(.) / mat_d_list[['td_pos']] %>% NROW(.) *100, digits = 1)`% that were unique to the Polynesian populations. Eastern Polynesians made up `r create_sums('td_pos') %>% filter((POL_CIM + POL_NZM & ALLsum > 0) == ALLsum) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` unique windows, with \gls{cim} having `r create_sums('td_pos') %>% filter(POL_CIM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` unique windows and only \gls{nzm} having `r create_sums('td_pos') %>% filter(POL_NZM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows. The Western Polynesians had `r create_sums('td_pos') %>% filter((POL_SAM + POL_TON) == ALLsum & ALLsum > 0)  %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` unique windows. `r create_sums('td_pos') %>% filter(POL_SAM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` were from only \gls{sam} and `r create_sums('td_pos') %>% filter(POL_TON == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` from only \gls{ton}.

There were `r create_sums('td_pos') %>% filter(POLsum < ALLsum & ALLsum > 4) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows where all the Polynesian populations shared a window with a non-Polynesian population. This is compared to super population unique window numbers of `r create_sums('td_pos') %>% filter(AFRsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)`, `r create_sums('td_pos') %>% filter(AMRsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)`, `r create_sums('td_pos') %>% filter(EASsum == ALLsum & ALLsum  > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)`, `r create_sums('td_pos') %>% filter(EURsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)`, `r create_sums('td_pos') %>% filter(SASsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` for the \gls{afr}, \gls{amr}, \gls{eas}, \gls{eur}, and \gls{sas} super populations respectively.  

The positive tail had excellent groupings, with all except the \gls{amr} being grouped into their respective super populations in their entireity. The exclusivity of the clusters was also high, with only \gls{clm} and \gls{pur} being assigned the same group as the \gls{eur} super population.

```{r, eval = FALSE}
# untested/bad assumption that all genes will be in here
create_sums('td_pos') %>% filter(POLsum == ALLsum & POLsum > 0) %>% left_join(., lower_sig_stats %>% mutate(chrom = as.numeric(chrom)) %>% select(contains('chrom'), SYMBOL) %>% distinct(), by = c('chrom', 'chrom_start', 'chrom_end')) %>% group_by(SYMBOL) %>% tally() %>% filter(!is.na(SYMBOL)) %>% NROW(.)

create_sums('td_pos') %>% filter(POL_CIM + POL_NZM == ALLsum & ALLsum == 2) %>% left_join(., lower_sig_stats %>% mutate(chrom = as.numeric(chrom)) %>% select(contains('chrom'), SYMBOL) %>% distinct(), by = c('chrom', 'chrom_start', 'chrom_end')) %>% group_by(SYMBOL) %>% tally() %>% filter(!is.na(SYMBOL)) %>% NROW(.)

create_sums('td_pos') %>% filter(POL_SAM + POL_TON == ALLsum & ALLsum == 2) %>% left_join(., lower_sig_stats %>% mutate(chrom = as.numeric(chrom)) %>% select(contains('chrom'), SYMBOL) %>% distinct(), by = c('chrom', 'chrom_start', 'chrom_end')) %>% group_by(SYMBOL) %>% tally() %>% filter(!is.na(SYMBOL)) %>% NROW(.)
```

```{r, eval = FALSE}
# untested/bad assumption that all genes will be in here
(td_pos_pol_genes <- create_sums('td_pos') %>% filter(POLsum == 4 ) %>% left_join(., lower_sig_stats %>% mutate(chrom = as.numeric(chrom)) %>% select(contains('chrom'), SYMBOL) %>% distinct(), by = c('chrom', 'chrom_start', 'chrom_end')) %>% group_by(SYMBOL) %>% tally())
```



#### Fay and Wu's _H_

(ref:fwhTop1000) _\gls{fwh} Clustering._ Hierarchical clustering of \gls{fwh} using the positive and negative tails of the distribution. **A** and **B** Dendrogram representing the clusters of \gls{fwh} lower and upper tails respectfully using hierarchical clustering with euclidean distance and complete linkage. Coloured by super population. **C** and **D** Plots representing the windows present in each population from the lower and upper tails respectfully.

```{r fwhTop1000, fig.cap='(ref:fwhTop1000)', fig.height=7}
four_plot("fwh","Fay and Wu's H")
```

<!--
- Fay and Wu's H put all of the super populations together both neg and pos (split AMR)
- there looks to be a lot of unique regions within the POL
- both neg and pos have the Pol cluster nearest to the EAS
-->

<!--
[x]describe number of total number of windows
[x]polynesian specific windows
[x]ep specific windows
[x]wp specific windows
[x]cim/nzm/sam/ton specific windows

[x]proportions
[x]exclusivity
[]dendrograms

[] refer to figure
-->
The clustering of Fay and Wu's _H_  (Figure \@ref(fig:fwhTop1000)) had the main groupings that came out the same as the defined super population groups, with the exception of the positive tail where the \gls{amr} super population was split in half. The Polynesian populations were clustered together in both the negative and positive tails for Fay and Wu's _H_. The Polynesian population group was closest to the \gls{eas} populations with `r create_sums('fwh_pos') %>% filter(POLsum + EASsum <= ALLsum & POLsum > 0 & EASsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows from the positive tail and `r create_sums('fwh_neg') %>% filter(POLsum + EASsum <= ALLsum & POLsum > 0 & EASsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows from the negative tail in common between at least one Polynesian population and one \gls{eas} population. In both tails the Polynesian cluster also had the East/West split within the group. 

_negative_

The clustering assigned all the inidividuals into their super populations and each of the groups was exclusive to the super population. There were a total of `r create_windowSummaryTable() %>% filter(stat == 'fwh_neg') %>% .[['total_windows']]` windows with `r create_sums('fwh_neg') %>% filter(POLsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows specific to Polynesian populations. This compared to `r create_sums('fwh_neg') %>% filter(AFRsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{afr}, `r create_sums('fwh_neg') %>% filter(AMRsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{amr}, `r create_sums('fwh_neg') %>% filter(EURsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{eur}, and `r create_sums('fwh_neg') %>% filter(EASsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{eas}. The Eastern Polynesian populations had `r create_sums('fwh_neg') %>% filter(POL_CIM + POL_NZM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows, of those, `r create_sums('fwh_neg') %>% filter(POL_CIM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` were specific to \gls{cim} and `r create_sums('fwh_neg') %>% filter(POL_NZM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` were specific to \gls{nzm}. The Western Polynesian populations had `r create_sums('fwh_neg') %>% filter(POL_SAM + POL_TON == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows, with `r create_sums('fwh_neg') %>% filter(POL_SAM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` specific to \gls{sam} and `r create_sums('fwh_neg') %>% filter(POL_TON == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` specific to \gls{ton}. The mean number of windows specific to an individual population was `r create_sums('fwh_neg') %>% filter(ALLsum == 1) %>% select(-contains('chrom'), -posid, -contains('sum')) %>% summarise_all(funs(sum)) %>% t() %>% mean() %>% round(., digits = 1)` in the negative tail.


_positive_

In the positive tail of the \gls{fwh} distribution, the clustering assigned all of the individual populations into their super populations with only \gls{amr} being split. The \gls{clm} and \gls{pur} populations were clustered as part of the \gls{eur} group. Exclusivity was also high with all clusters except the \gls{eur} - \gls{amr} being exclusive to a single super population. Of the `r create_windowSummaryTable() %>% filter(stat == 'fwh_pos') %>% .[['total_windows']]` windows,  `r create_sums('fwh_pos') %>% filter(POLsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows were from Polynesian populations only. The Eastern Polynesian populations had `r create_sums('fwh_pos') %>% filter(POL_CIM + POL_NZM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` specific windows, of those, `r create_sums('fwh_pos') %>% filter(POL_CIM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` were unique to \gls{cim} and `r create_sums('fwh_pos') %>% filter(POL_NZM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` were unique to \gls{nzm}. The Western Polynesian populations had `r create_sums('fwh_pos') %>% filter(POL_SAM + POL_TON == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` unique windows, of these, `r create_sums('fwh_pos') %>% filter(POL_SAM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` were specific to \gls{sam} and `r create_sums('fwh_pos') %>% filter(POL_TON == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` specific to \gls{ton}. The mean number of unique windows for all populations was `r create_sums('fwh_pos') %>% filter(ALLsum == 1) %>% select(-contains('chrom'), -posid, -contains('sum')) %>% summarise_all(funs(sum)) %>% t() %>% mean() %>% round(., digits = 1)`.



#### Fu and Li's _D_

(ref:fldTop1000) _\gls{fld} Clustering._ Hierarchical clustering of \gls{fld} using the positive and negative tails of the distribution. **A** and **B** Dendrogram representing the clusters of \gls{fld} lower and upper tails respectfully using hierarchical clustering with euclidean distance and complete linkage. Coloured by super population. **C** and **D** Plots representing the windows present in each population from the lower and upper tails respectfully. Colour represents a window being present.

```{r fldTop1000, fig.cap = '(ref:fldTop1000)', fig.height=7}
four_plot("fld","Fu and Li's D")
```

<!--
- 6 fold difference in the number of windows
- pos had ~3300, neg had ~19000
- pol is the least similar to the rest of the 1kgp pops in pos
- sam/nzm and cim/ton split in neg
- only sas and nearly eur are completely grouped together, looks somewhat random
-->

<!--
[x]describe number of total number of windows
[x]polynesian specific windows
[x]ep specific windows
[x]wp specific windows
[x]cim/nzm/sam/ton specific windows

[x]proportions
[x]exclusivity
[]dendrograms

[]refer to figure
-->
There was nearly a six-fold difference in the number of total windows used in the clustering between the positive (`r create_windowSummaryTable() %>% filter(stat == 'fld_pos') %>% .[['total_windows']]`) and negative tails (`r create_windowSummaryTable() %>% filter(stat == 'fld_neg') %>% .[['total_windows']]`). This indicated that there was a large number of regions in common in the positive tail, whereas few windows were shared in the negative tail, with many windows being population specific. In the positive tail there was a mean of `r round(create_windowSummaryTable() %>% filter(stat == 'fld_pos') %>%  .[['mean']], digits = 2)` populations sharing a window, which was `r round(create_windowSummaryTable() %>% filter(stat == 'fld_pos') %>%  .[['mean']] / (create_windowSummaryTable() %>% select(mean) %>% summarise(mean = mean(mean))), digits = 2)` times greater than the mean for all selection test stats used. In the positive tail the Polynesian populations were grouped together and also within the group, paired in their East/West split.


negative

With the exception of the \gls{afr} populations, all super populations had a high proportion being clustered. The exclusivity was low, with 17 populations, including all from \gls{amr}, \gls{eur}, and \gls{sas} assigned the same cluster. The \gls{afr} populations were spread over 3 clusters. The \gls{pol} populations were grouped as an exclusive single cluster. There were a total of `r create_windowSummaryTable() %>% filter(stat == 'fld_neg') %>% .[['total_windows']]` windows with `r create_sums('fld_neg') %>% filter(POLsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows specific to Polynesian populations. This compared to `r create_sums('fld_neg') %>% filter(AFRsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{afr}, `r create_sums('fld_neg') %>% filter(AMRsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{amr}, `r create_sums('fld_neg') %>% filter(EURsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{eur}, and `r create_sums('fld_neg') %>% filter(EASsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{eas}. The Eastern Polynesian populations had a total of `r create_sums('fld_neg') %>% filter(POL_CIM + POL_NZM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` unique windows, `r create_sums('fld_neg') %>% filter(POL_CIM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` were unique to \gls{cim} and `r create_sums('fld_neg') %>% filter(POL_NZM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` were unique to \gls{nzm}. The Western Polynesian populations had `r create_sums('fld_neg') %>% filter(POL_SAM + POL_TON == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` unique windows. Of these `r create_sums('fld_neg') %>% filter(POL_SAM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` were unique to \gls{sam} and `r create_sums('fld_neg') %>% filter(POL_TON == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` to \gls{ton}. The mean number of population unique windows was `r create_sums('fld_neg') %>% filter(ALLsum == 1) %>% select(-contains('chrom'), -posid, -contains('sum')) %>% summarise_all(funs(sum)) %>% t() %>% mean() %>% round(., digits = 1)` (SD = `r create_sums('fld_neg') %>% filter(ALLsum == 1) %>% select(-contains('chrom'), -posid, -contains('sum')) %>% summarise_all(funs(sum)) %>% t() %>% sd() %>% round(., digits = 1)` in the negative tail.


positive

The clustering grouped most populations into their super populations. The \gls{eas} populations were clustered together. The Western Polynesian populations were clustered together but the Eastern Polynesnian populations were not. The \gls{amr} populations \gls{mxl} amd \gls{pel} were grouped with the \gls{nzm}. Ten populations were grouped into a single cluster consisting of 2 from \gls{afr}, 2 from \gls{amr}, 1 from \gls{eur} and all 5 from \gls{sas}. There were a total of `r create_windowSummaryTable() %>% filter(stat == 'fld_pos') %>% .[['total_windows']]` windows with `r create_sums('fld_pos') %>% filter(POLsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows specific to Polynesian populations. This compared to `r create_sums('fld_pos') %>% filter(AFRsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{afr}, `r create_sums('fld_pos') %>% filter(AMRsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{amr}, `r create_sums('fld_pos') %>% filter(EURsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{eur}, and `r create_sums('fld_pos') %>% filter(EASsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{eas}. The Eastern Polynesian populations had a total of `r create_sums('fld_pos') %>% filter(POL_CIM + POL_NZM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` unique windows with `r create_sums('fld_pos') %>% filter(POL_CIM == ALLsum & ALLsum > 0 ) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` only from \gls{cim} and `r create_sums('fld_pos') %>% filter(POL_NZM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` from \gls{nzm}. From the Western Polynesian populations there were `r create_sums('fld_pos') %>% filter(POL_SAM + POL_TON == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` specific windows. There were `r create_sums('fld_pos') %>% filter(POL_SAM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` specific to \gls{sam} and `r create_sums('fld_pos') %>% filter(POL_TON == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` specific to \gls{ton}. The mean number of unique windows per population was `r create_sums('fld_pos') %>% filter(ALLsum == 1) %>% select(-contains('chrom'), -posid, -contains('sum')) %>% summarise_all(funs(sum)) %>% t() %>% mean() %>% round(., digits = 1)` (SD = `r create_sums('fld_pos') %>% filter(ALLsum == 1) %>% select(-contains('chrom'), -posid, -contains('sum')) %>% summarise_all(funs(sum)) %>% t() %>% sd() %>% round(., digits = 1)`).




#### Fu and Li's _F_

(ref:flfTop1000) _\gls{flf} Clustering._ Hierarchical clustering of \gls{flf} using the positive and negative tails of the distribution. *A* and *B* Dendrogram representing the clusters of \gls{flf} lower and upper tails respectfully using hierarchical clustering with euclidean distance and complete linkage. Coloured by super population. *C* and *D* Plots representing the windows present in each population from the lower and upper tails respectfully.

```{r flfTop1000, fig.cap = '(ref:flfTop1000)', fig.height=7}
four_plot("flf","Fu and Li's F")
```

<!--
- pos clustered super populations together, split AMR
- pos looks like there is a reasonable number of windows shared between pops (5188)
- neg looks pretty sparse 17044
-->

<!--
[]describe number of total number of windows
[x]polynesian specific windows
[x]ep specific windows
[x]wp specific windows
[x]cim/nzm/sam/ton specific windows

[]proportions
[x]exclusivity
[]dendrograms

[]refer to figure
-->

Clustering on the \gls{flf} (Figure \@ref(fig:flfTop1000)) had `r create_windowSummaryTable() %>% filter(stat == 'flf_pos') %>% .[['total_windows']]` windows in the positive tail (Figure \@ref(fig:flfTop1000) B and D) and `r create_windowSummaryTable() %>% filter(stat == 'flf_neg') %>% .[['total_windows']]` windows in the negative tail (Figure \@ref(fig:flfTop1000) A and C). The mean number of unique windows per population was `r create_sums('flf_pos') %>% filter(ALLsum == 1) %>% select(-contains('chrom'), -posid, -contains('sum')) %>% summarise_all(funs(sum)) %>% t() %>% mean() %>% round(., digits = 1)` (SD = `r create_sums('flf_pos') %>% filter(ALLsum == 1) %>% select(-contains('chrom'), -posid, -contains('sum')) %>% summarise_all(funs(sum)) %>% t() %>% sd() %>% round(., digits = 1)`) in the positive tail, compared with `r create_sums('flf_neg') %>% filter(ALLsum == 1) %>% select(-contains('chrom'), -posid, -contains('sum')) %>% summarise_all(funs(sum)) %>% t() %>% mean() %>% round(., digits = 1)` (SD = `r create_sums('flf_neg') %>% filter(ALLsum == 1) %>% select(-contains('chrom'), -posid, -contains('sum')) %>% summarise_all(funs(sum)) %>% t() %>% sd() %>% round(., digits = 1)`)in the negative tail.  


negative

Clustering on the negative tail windows grouped the majority of the individual populations together into their super populations. The Polyesians populations were grouped togther as a West Polynesian group and the \gls{cim} population. The \gls{nzm} population was grouped with the \gls{mxl} and \gls{pel} as part of a larger group with the \gls{eur} super population. Exclusivity of clusters was not bad with 4/6 being exclusive. There were a total of `r create_windowSummaryTable() %>% filter(stat == 'flf_neg') %>% .[['total_windows']]` windows with `r create_sums('flf_neg') %>% filter(POLsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows specific to Polynesian populations. This compared to `r create_sums('flf_neg') %>% filter(AFRsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{afr}, `r create_sums('flf_neg') %>% filter(AMRsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{amr}, `r create_sums('flf_neg') %>% filter(EURsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{eur}, and `r create_sums('flf_neg') %>% filter(EASsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{eas}. The Eastern Polynesian populations had a total of `r create_sums('flf_neg') %>% filter(POL_CIM + POL_NZM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` unique windows, `r create_sums('flf_neg') %>% filter(POL_CIM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` were unique to \gls{cim} and `r create_sums('flf_neg') %>% filter(POL_NZM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` were unique to \gls{nzm}. The Western Polynesian populations had `r create_sums('flf_neg') %>% filter(POL_SAM + POL_TON == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` unique windows. Of these `r create_sums('flf_neg') %>% filter(POL_SAM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` were unique to \gls{sam} and `r create_sums('flf_neg') %>% filter(POL_TON == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` to \gls{ton}.



positive

Clustering on the windows from the positive tail put the super populations into their own groups, but also split the \gls{amr} populations into 2 separate groups. The Polynesian populations were also clustered into an East/West split of 2 separate exclusive clusters. Exclusivity of clusters was high with 5/6 clusters being exclusive to a single super population, however one cluster contained half of the \gls{amr} populations (\gls{clm} and \gls{pur})and all of \gls{eur} and \gls{sas} populations. There were a total of `r create_windowSummaryTable() %>% filter(stat == 'flf_pos') %>% .[['total_windows']]` windows with `r create_sums('flf_pos') %>% filter(POLsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows specific to Polynesian populations. This compared to `r create_sums('flf_pos') %>% filter(AFRsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{afr}, `r create_sums('flf_pos') %>% filter(AMRsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{amr}, `r create_sums('flf_pos') %>% filter(EURsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{eur}, and `r create_sums('flf_pos') %>% filter(EASsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{eas}. The Eastern Polynesian populations had a total of `r create_sums('flf_pos') %>% filter(POL_CIM + POL_NZM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` unique windows with `r create_sums('flf_pos') %>% filter(POL_CIM == ALLsum & ALLsum > 0 ) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` only from \gls{cim} and `r create_sums('flf_pos') %>% filter(POL_NZM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` from \gls{nzm}. From the Western Polynesian populations there were `r create_sums('flf_pos') %>% filter(POL_SAM + POL_TON == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` specific windows. There were `r create_sums('flf_pos') %>% filter(POL_SAM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` specific to \gls{sam} and `r create_sums('flf_pos') %>% filter(POL_TON == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` specific to \gls{ton}.


#### Zeng's _E_

(ref:zeTop1000) _\gls{ze} Clustering._ Hierarchical clustering of \gls{ze} using the positive and negative tails of the distribution. *A* and *B* Dendrogram representing the clusters of \gls{ze} lower and upper tails respectfully using hierarchical clustering with euclidean distance and complete linkage. Coloured by super population. *C* and *D* Plots representing the windows present in each population from the lower and upper tails respectfully.

```{r zeTop1000, fig.cap = '(ref:zeTop1000)', fig.height=7}
four_plot("ze","Zeng's E")
```


Clustering on \gls{ze} on the positive tail (Figure \@ref(fig:zeTop1000) B and D) of the distribution gave clustering that matched the super popualations, with the Polyesian populations grouped together and subgrouped into an East/West split. The Polyensian population group's next closest group was the \gls{eas} super population. The negative tail (Figure \@ref(fig:zeTop1000) A and C) showed a similar grouping but split the \gls{amr} super population.


\Gls{ze} for both extremes of the distribution, clustered all populations except \gls{amr} into their respective super populations. The \gls{amr} population was split into \gls{clm} and \gls{pur} which was grouped with the \gls{eur} populations, and \gls{mxl} and \gls{pel} which were grouped as a separate exclusive cluster. Exclusivity of clusters was high with only the \gls{amr} -\gls{eur} group not being exclusive to a single super population.

negative

There were a total of `r create_windowSummaryTable() %>% filter(stat == 'ze_neg') %>% .[['total_windows']]` windows with `r create_sums('ze_neg') %>% filter(POLsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows specific to Polynesian populations. This compared to `r create_sums('ze_neg') %>% filter(AFRsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{afr}, `r create_sums('ze_neg') %>% filter(AMRsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{amr}, `r create_sums('ze_neg') %>% filter(EURsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{eur}, and `r create_sums('ze_neg') %>% filter(EASsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{eas}. There were a total of `r create_sums('ze_neg') %>% filter(POL_CIM + POL_NZM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows specific to the Eastern Polynesian populations, with `r create_sums('ze_neg') %>% filter(POL_CIM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` specific to \gls{cim} and `r create_sums('ze_neg') %>% filter(POL_NZM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` specific to \gls{nzm}. The Western Polynesian populations had `r create_sums('ze_neg') %>% filter(POL_SAM + POL_TON == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` unqiue windows, `r create_sums('ze_neg') %>% filter(POL_SAM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` were unique to \gls{sam} and `r create_sums('ze_neg') %>% filter(POL_TON == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` unique to \gls{ton}. The mean number of unique windows was `r create_sums('ze_neg') %>% filter(ALLsum == 1) %>% select(-contains('chrom'), -posid, -contains('sum')) %>% summarise_all(funs(sum)) %>% t() %>% mean() %>% round(., digits = 1)` in the negative tail.

positive

There were a total of `r create_windowSummaryTable() %>% filter(stat == 'ze_pos') %>% .[['total_windows']]` windows with `r create_sums('ze_pos') %>% filter(POLsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows specific to Polynesian populations. This compared to `r create_sums('ze_pos') %>% filter(AFRsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{afr}, `r create_sums('ze_pos') %>% filter(AMRsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{amr}, `r create_sums('ze_pos') %>% filter(EURsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{eur}, and `r create_sums('ze_pos') %>% filter(EASsum == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` windows for \gls{eas}. The Eastern Polynesian populations had `r create_sums('ze_pos') %>% filter(POL_CIM + POL_NZM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` unique windows, the \gls{cim} had`r create_sums('ze_pos') %>% filter(POL_CIM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)`unique windows, and \gls{nzm} had `r create_sums('ze_pos') %>% filter(POL_NZM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` unique windows. The Western Polynesian popualtions had `r create_sums('ze_pos') %>% filter(POL_SAM + POL_TON == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` unique windows. Of those, `r create_sums('ze_pos') %>% filter(POL_SAM == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` were unique to \gls{sam} and `r create_sums('ze_pos') %>% filter(POL_TON == ALLsum & ALLsum > 0) %>% select(contains('chrom'), contains('sum')) %>% NROW(.)` unique to \gls{ton}. The mean number of unique windows per population was `r create_sums('ze_pos') %>% filter(ALLsum == 1) %>% select(-contains('chrom'), -posid, -contains('sum')) %>% summarise_all(funs(sum)) %>% t() %>% mean() %>% round(., digits = 1)`.


<!-- create df from the clustering results to be able to pull numbers out of for number of windows per population for each statistic, and for the proportion of pops being clustered -->

#### Random draws

```{r, results='asis'}
td_random <- readRDS('~/data/NZ_coreExome_1kgp/100kbWindow_intra/td_random_filtered_3ns-18-7-2017.RDS')
rand_tab <- bind_rows(lapply(seq_along(td_random), function(x){td_random[[x]]$super_prop})) %>% group_by(super) %>% group_by(super) %>% summarise(min = round(min(max_prop),3), mean = round(mean(max_prop), 3), sd = round(sd(max_prop), 3), max = round(max(max_prop),3), prop50 = round(sum(if_else(max_prop >=0.5, 1, 0))/NROW(.) * 100, 1) , total_complete = sum(if_else(max_prop == 1, 1, 0))) %>% data.frame()
rand_tab_pander <- rand_tab
colnames(rand_tab_pander) <- c('Super\nPopulation', 'Min','Mean','Std. Dev','Max','Prop50 (%)', 'Total Complete')
pander::pandoc.table(rand_tab_pander, caption = '(\\#tab:randtab) Summary of proportion of individual populations assigned to their super population cluster. Min is the minimum proportion for the cluster from all iterations. Mean is the mean proportion of the maximum cluster proportions for all iterations. Std. Dev = standard deviation of the mean. Max is the maximum proportion from all iterations. Prop50 is the number of times at least 50% of the super population was clustered together, expressed as a percentage. Total complete is the number of times the super population had all of its individual populations assigned the same cluster.')
```

<!-- why does the Eur never produce a max prop of 1?-->

To assess the likelihood of obtaining the super population clusters by chance, 10000 iterations of random resampling of 1000 windows per population was undertaken and then clustered. The maximum proportion of populations being clustered into their designated super population was then calculated when the tree was cut to have 6 groups. A summary of all iterations is reported in Table \@ref(tab:randtab). 

Not a single random draw grouped the all individual populations into their super population. Only for the Polynesians populations and \gls{amr} fully clustered into the same cluster and this was a rare event. Grouping at least 50% of a super population was at best `r round(rand_tab %>% summarise(max(prop50)) * 100, digits = 0)`% of the draws but for the larger super populations (\gls{afr}, \gls{eas}, \gls{eur}, and \gls{sas}) it was on average `r round(rand_tab %>% filter(!super %in% c('POL','AMR'))%>% summarise(mean(prop50)) * 100, digits = 0)`% of the draws.





```{r, eval = FALSE}
#calculate the mean proportion for the population clusters across all iterations
# remove 0 because they aren't actually a cluster
bind_rows(lapply(seq_along(td_random), function(x){td_random[[x]]$pop_clust_prop %>% data.frame()%>% mutate(clust = row.names(.))})) %>% select(-clust) %>% gather("pop","prop") %>% filter(prop > 0) %>% group_by(pop) %>% summarise(min(prop), quantile(prop, 0.25), mean(prop),sd(prop), median(prop),quantile(prop, 0.75) , max(prop))

bind_rows(lapply(seq_along(td_random), function(x){td_random[[x]]$pop_clust_prop %>% data.frame()%>% mutate(clust = row.names(.))})) %>% select(-clust) %>% gather("pop","prop") %>% filter(prop > 0) %>% group_by(pop) %>% ggplot(., aes(x =pop, y = prop)) + geom_violin()
```




```{r, eval = FALSE}
bind_rows(lapply(seq_along(td_random), function(x){td_random[[x]]$pop_clust_prop %>% data.frame()%>% mutate(clust = row.names(.), iter = x)})) 
#bind_rows(lapply(td_random, function(x){ x %>% spread('super', 'max_prop')})) %>% filter(POL >= 2/4, EUR >= 3/6, EAS >= 3/5, AFR >= 3/6, SAS >= 3/5) %>% arrange(POL)

#bind_rows(lapply(td_random, function(x){ x %>% spread('super', 'max_prop')})) %>% mutate(rn = as.numeric(rownames(.)), exp_prop = rowSums(.)/6) %>% filter(EUR > .5) %>% arrange(exp_prop)
```



### Haplotypic



Clustering on the proportion of shared regions for both \gls{ihs} (Figure \@ref(fig:ihsclust)) and \gls{nsl} (Figure \@ref(fig:nslclust)) grouped the individual populations into their super populations, with the exception of the \gls{amr} populations which were split into 2 separate groups. Both \gls{ihs} and \gls{nsl} gave very similar groupings and had similar proportions shared.

```{r}
mat_ihs <- readRDS('~/data/NZ_coreExome_1kgp/haplotype/ihs_clustered_dist_mat-6-7-2017.RDS')
clus_ihs <- readRDS('~/data/NZ_coreExome_1kgp/haplotype/ihs_clus_regions-14-7-2017.RDS')
gr <- GenomicRanges::GRanges(as.data.frame(clus_ihs))

#nzm  vs cim
sum(GenomicRanges::width(GenomicRanges::reduce(GenomicRanges::intersect(gr[gr$pop == "POL_NZM"], gr[gr$pop == "POL_CIM"]))))/sum(GenomicRanges::width(GenomicRanges::reduce(gr[gr$pop == "POL_NZM"])))

sum(GenomicRanges::width(GenomicRanges::reduce(GenomicRanges::intersect(gr[gr$pop == "POL_CIM"], gr[gr$pop == "POL_NZM"]))))/sum(GenomicRanges::width(GenomicRanges::reduce(gr[gr$pop == "POL_CIM"])))

sum(GenomicRanges::width(GenomicRanges::reduce(gr[gr$pop %in% c("POL_SAM","POL_TON")])))/sum(GenomicRanges::width(GenomicRanges::reduce(gr[gr$pop %in% c("POL_NZM", "POL_CIM", "POL_SAM","POL_TON")])))

```
<!--
CHECK HOW IHS PROPS ARE BEING CALCULATED!!!
-->

what was the overlap in regions like for ihs/nsl

ihs
<!-- DOUBLE CHECK THE ORDERS 
double checked and colname is the denominator
-->
NZM `r (mat_ihs[rownames(mat_ihs) == "POL_CIM", colnames(mat_ihs) == "POL_NZM"] * 100) %>% round(., digits = 2)`% of regions with CIM, whereas CIM shared `r (mat_ihs[rownames(mat_ihs) == "POL_NZM", colnames(mat_ihs) == "POL_CIM"] * 100) %>% round(., digits = 2)`% of regions with NZM


SAM shared `r (mat_ihs[rownames(mat_ihs) == "POL_TON", colnames(mat_ihs) == "POL_SAM"] * 100) %>% round(., digits = 2)`% of regions with TON, but TON shared `r (mat_ihs[rownames(mat_ihs) == "POL_SAM", colnames(mat_ihs) == "POL_TON"] * 100) %>% round(., digits = 2)`% of regions with SAM


The Eastern Polynesians made up `r (sum(GenomicRanges::width(GenomicRanges::reduce(gr[gr$pop %in% c("POL_NZM","POL_CIM")])))/sum(GenomicRanges::width(GenomicRanges::reduce(gr[gr$pop %in% c("POL_NZM", "POL_CIM", "POL_SAM","POL_TON")]))) *100) %>% round(.,2)`% of the regions from all of the Polynesian populations while the Western Polynesian populations made up `r (sum(GenomicRanges::width(GenomicRanges::reduce(gr[gr$pop %in% c("POL_SAM","POL_TON")])))/sum(GenomicRanges::width(GenomicRanges::reduce(gr[gr$pop %in% c("POL_NZM", "POL_CIM", "POL_SAM","POL_TON")]))) * 100) %>% round(., digits = 2)`

Out of all the regions for Polynesian populations only  `r (sum(GenomicRanges::width(GenomicRanges::reduce(GenomicRanges::intersect(GenomicRanges::intersect(gr[gr$pop == 'POL_CIM'], gr[gr$pop == "POL_NZM"]), GenomicRanges::intersect(gr[gr$pop == 'POL_SAM'], gr[gr$pop == "POL_TON"])))))/sum(GenomicRanges::width(GenomicRanges::reduce(gr[gr$pop %in% c("POL_NZM", "POL_CIM", "POL_SAM","POL_TON")]))) * 100) %>% round(., digits = 2)`% was in common.

The amount of region the Polynesian populations shared with AMR was `r (sum(GenomicRanges::width(GenomicRanges::reduce(GenomicRanges::intersect(gr[gr$pop %in% unique(gr$pop[grep('POL', gr$pop)])], gr[gr$pop %in% unique(gr$pop[grep('AMR', gr$pop)]) ]))))/sum(GenomicRanges::width(GenomicRanges::reduce(gr[gr$pop %in% unique(gr$pop[grep('POL', gr$pop)])]))) * 100) %>% round(., digits = 2)`%, AFR  was `r (sum(GenomicRanges::width(GenomicRanges::reduce(GenomicRanges::intersect(gr[gr$pop %in% unique(gr$pop[grep('POL', gr$pop)])], gr[gr$pop %in% unique(gr$pop[grep('AFR', gr$pop)]) ]))))/sum(GenomicRanges::width(GenomicRanges::reduce(gr[gr$pop %in% unique(gr$pop[grep('POL', gr$pop)])]))) * 100) %>% round(., digits = 2)`%, EAS   was `r (sum(GenomicRanges::width(GenomicRanges::reduce(GenomicRanges::intersect(gr[gr$pop %in% unique(gr$pop[grep('POL', gr$pop)])], gr[gr$pop %in% unique(gr$pop[grep('EAS', gr$pop)]) ]))))/sum(GenomicRanges::width(GenomicRanges::reduce(gr[gr$pop %in% unique(gr$pop[grep('POL', gr$pop)])]))) * 100) %>% round(., digits = 2)`%, EUR was `r (sum(GenomicRanges::width(GenomicRanges::reduce(GenomicRanges::intersect(gr[gr$pop %in% unique(gr$pop[grep('POL', gr$pop)])], gr[gr$pop %in% unique(gr$pop[grep('EUR', gr$pop)]) ]))))/sum(GenomicRanges::width(GenomicRanges::reduce(gr[gr$pop %in% unique(gr$pop[grep('POL', gr$pop)])]))) * 100) %>% round(., digits = 2)`%, and SAS was `r (sum(GenomicRanges::width(GenomicRanges::reduce(GenomicRanges::intersect(gr[gr$pop %in% unique(gr$pop[grep('POL', gr$pop)])], gr[gr$pop %in% unique(gr$pop[grep('SAS', gr$pop)]) ]))))/sum(GenomicRanges::width(GenomicRanges::reduce(gr[gr$pop %in% unique(gr$pop[grep('POL', gr$pop)])]))) * 100) %>% round(., digits = 2)`%. this was lower/higher than the mean for the other 



```{r ihsclust, fig.cap='(ref:ihsclust)', echo = FALSE}
ihs_dendro_plot()
```

(ref:ihsclust) _\gls{ihs} Clustering_ A. Dendrogram from hierachical clustering based on proportion of \gls{ihs} regions shared between populations Coloured by super population. B. Heatmap showing the proportion of \gls{ihs} regions shared.

```{r nslclust, fig.cap='(ref:nslclust)', echo = FALSE}
nsl_dendro_plot()
```

(ref:nslclust) _\gls{nsl} Clustering_ A. Dendrogram from hierachical clustering based on proportion of \gls{nsl} regions shared between populations. Coloured by super population. B. Heatmap showing the proportion of \gls{nsl} regions shared.

nsl

```{r}
mat_nsl <- readRDS('~/data/NZ_coreExome_1kgp/haplotype/nsl_clustered_dist_mat-6-7-2017.RDS')
clus_nsl <- readRDS('~/data/NZ_coreExome_1kgp/haplotype/nsl_clus_regions-14-7-2017.RDS')
gr <- GenomicRanges::GRanges(as.data.frame(clus_nsl))
```

NZM `r (mat_ihs[rownames(mat_ihs) == "POL_CIM", colnames(mat_ihs) == "POL_NZM"] * 100) %>% round(., digits = 2)`% of regions with CIM, whereas CIM shared `r (mat_ihs[rownames(mat_ihs) == "POL_NZM", colnames(mat_ihs) == "POL_CIM"] * 100) %>% round(., digits = 2)`% of regions with NZM


SAM shared `r (mat_ihs[rownames(mat_ihs) == "POL_TON", colnames(mat_ihs) == "POL_SAM"] * 100) %>% round(., digits = 2)`% of regions with TON, but TON shared `r (mat_ihs[rownames(mat_ihs) == "POL_SAM", colnames(mat_ihs) == "POL_TON"] * 100) %>% round(., digits = 2)`% of regions with SAM


The Eastern Polynesians made up `r (sum(GenomicRanges::width(GenomicRanges::reduce(gr[gr$pop %in% c("POL_NZM","POL_CIM")])))/sum(GenomicRanges::width(GenomicRanges::reduce(gr[gr$pop %in% c("POL_NZM", "POL_CIM", "POL_SAM","POL_TON")]))) *100) %>% round(.,2)`% of the regions from all of the Polynesian populations while the Western Polynesian populations made up `r (sum(GenomicRanges::width(GenomicRanges::reduce(gr[gr$pop %in% c("POL_SAM","POL_TON")])))/sum(GenomicRanges::width(GenomicRanges::reduce(gr[gr$pop %in% c("POL_NZM", "POL_CIM", "POL_SAM","POL_TON")]))) * 100) %>% round(., digits = 2)`

Out of all the regions for Polynesian populations only  `r (sum(GenomicRanges::width(GenomicRanges::reduce(GenomicRanges::intersect(GenomicRanges::intersect(gr[gr$pop == 'POL_CIM'], gr[gr$pop == "POL_NZM"]), GenomicRanges::intersect(gr[gr$pop == 'POL_SAM'], gr[gr$pop == "POL_TON"])))))/sum(GenomicRanges::width(GenomicRanges::reduce(gr[gr$pop %in% c("POL_NZM", "POL_CIM", "POL_SAM","POL_TON")]))) * 100) %>% round(., digits = 2)`% was in common.

The amount of region the Polynesian populations shared with AMR was `r (sum(GenomicRanges::width(GenomicRanges::reduce(GenomicRanges::intersect(gr[gr$pop %in% unique(gr$pop[grep('POL', gr$pop)])], gr[gr$pop %in% unique(gr$pop[grep('AMR', gr$pop)]) ]))))/sum(GenomicRanges::width(GenomicRanges::reduce(gr[gr$pop %in% unique(gr$pop[grep('POL', gr$pop)])]))) * 100) %>% round(., digits = 2)`%, AFR  was `r (sum(GenomicRanges::width(GenomicRanges::reduce(GenomicRanges::intersect(gr[gr$pop %in% unique(gr$pop[grep('POL', gr$pop)])], gr[gr$pop %in% unique(gr$pop[grep('AFR', gr$pop)]) ]))))/sum(GenomicRanges::width(GenomicRanges::reduce(gr[gr$pop %in% unique(gr$pop[grep('POL', gr$pop)])]))) * 100) %>% round(., digits = 2)`%, EAS   was `r (sum(GenomicRanges::width(GenomicRanges::reduce(GenomicRanges::intersect(gr[gr$pop %in% unique(gr$pop[grep('POL', gr$pop)])], gr[gr$pop %in% unique(gr$pop[grep('EAS', gr$pop)]) ]))))/sum(GenomicRanges::width(GenomicRanges::reduce(gr[gr$pop %in% unique(gr$pop[grep('POL', gr$pop)])]))) * 100) %>% round(., digits = 2)`%, EUR was `r (sum(GenomicRanges::width(GenomicRanges::reduce(GenomicRanges::intersect(gr[gr$pop %in% unique(gr$pop[grep('POL', gr$pop)])], gr[gr$pop %in% unique(gr$pop[grep('EUR', gr$pop)]) ]))))/sum(GenomicRanges::width(GenomicRanges::reduce(gr[gr$pop %in% unique(gr$pop[grep('POL', gr$pop)])]))) * 100) %>% round(., digits = 2)`%, and SAS was `r (sum(GenomicRanges::width(GenomicRanges::reduce(GenomicRanges::intersect(gr[gr$pop %in% unique(gr$pop[grep('POL', gr$pop)])], gr[gr$pop %in% unique(gr$pop[grep('SAS', gr$pop)]) ]))))/sum(GenomicRanges::width(GenomicRanges::reduce(gr[gr$pop %in% unique(gr$pop[grep('POL', gr$pop)])]))) * 100) %>% round(., digits = 2)`%. this was lower/higher than the mean for the other 




### Disease Associated Gene Clustering {#genelistcluster}
<!-- might be a good idea to use a grid to put multiple plots into one -->
```{r, include = FALSE}
load('~/data/gwas_catalog/diseaseGR-26-6-2017.RData')# brings in objects called 
load('~/data/NZ_coreExome_1kgp/100kbWindow_intra/gene_clustering_filtered_3ns-13-7-2017.RData')# brings in objects called gout, obesity, t2d, metsyn
names(gout) <- c('mat_d','prop')
names(t2d) <-  c('mat_d','prop')
names(obesity) <-  c('mat_d','prop')
names(metsyn) <-  c('mat_d','prop')
```


The actual values for the windows that intersected the associated gene windows were used to create the distance matrix

- Tajima's D clustered all super populations together (AMR was slightly ungrouped, often PEL out) and get the EP/WP split for all diseases
- Fay and Wu's H clustered all super populations together (AMR was slightly ungrouped, often PEL out) and get the EP/WP split for all diseases
- Fu and Li's D


#### Urate and Gout

There were 2 genes; _BCAS3_ and _RREB1_, that had at least one statistic in all of the Polynesian populations that met the threshold of significance. _BCAS3_ met significance in the lower tail for \gls{fld} and \gls{flf} with \gls{cim}, \gls{nzm}, and \gls{ton}; \gls{fwh} had \gls{cim}, \gls{nzm}, and \gls{sam}; \gls{td} with \gls{cim} and \gls{nzm}. There were `r lower_sig_stats %>% filter(SYMBOL %in% unique(urate_goutGR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(SYMBOL) %>% distinct() %>% NROW()` of the urate and gout associated genes that had evidence from at least one selection statistic of possible selection in any of the Polynesian populations from the lower tail of the distributions. _SLC2A9_ had a significant \gls{fld} from the positive tail of the distribution and with \gls{cim}, \gls{nzm}, \gls{sam}. There was a total of `r upper_sig_stats %>% filter(SYMBOL %in% unique(urate_goutGR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(SYMBOL) %>% distinct() %>% NROW()` genes that had a significant window in any. 

There were `r length(unique(urate_goutGR$SYMBOL))` genes that had an association with urate or gout at a genome-wide significance. This mapped to `r nrow(gout$mat_d$Tajima.D)` windows for each of the frequency based selection test statistics. \gls{ze} and \gls{fwh} clustered each super population into their own group and each cluster was exclusive to that super population. Both \gls{td} and \gls{flf} grouped most of the super populations correctly and nearly all clusters had good exclusivity for a single super population. \gls{fld} did not cluster the super populations well. With the exception of \gls{fld}, the clustering for the other selection test statistics gave an East/West Polynesian population sub group within the Polyensian group (Figure \@ref(fig:goutdendros)).



<!--`r lower_sig_stats %>% filter(SYMBOL %in% unique(urate_goutGR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(pop, SYMBOL) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==4) %>% .[['SYMBOL']]` -->

<!--
- Td gave the super populations, good exclusivity of clusters,ep/wp split
- fwh gave perfect super populations and perfect exclusivity. ep/wp split
- fld not great, sam most disimilar, nzm-gbr, ton-cim-chb-jpt
- flf got most of the super populations. not bad for exclusivity, cim-nzm-sam + ton-chb-jpt as a big cluster. clm/pur in the middle of the eur cluster - spanish sub cluster?
- ze perfect super populations, perfect exclusivity

dendrograms:
all except fld manage to cluster the super populations with the exception of AMR although ze does. fld is somewhat random. except fld all give east/west split
td and flf hace pol as outlier

exclusivity:
zeng e - perfect
td and fay wu's h do a good job but because of the AMR split end up with 7 actual clusters
fld - poor because there are lots of smaller clusters
flf - not great

props:
ze perfect
td and fwh only get amr not complete
flf has prop of >.75, except eur
fld poly .25, rest looking at majority of super pop in same cluster

-->

```{r, fig.cap='(ref:goutdendros)', fig.height=7}
multiplot(gene_dendros(gene_data = gout$mat_d$Tajima.D[,-1:-3]) + ggtitle("Tajima's D"),
          gene_dendros(gene_data = gout$mat_d$Fay.Wu.H[,-1:-3]) + ggtitle("Fay and Wu's H"),
          gene_dendros(gene_data = gout$mat_d$Fu.Li.D[,-1:-3]) + ggtitle("Fu and Li's D"),
          gene_dendros(gene_data = gout$mat_d$Fu.Li.F[,-1:-3]) + ggtitle("Fu and Li's F"),
          gene_dendros(gene_data = gout$mat_d$Zeng.E[,-1:-3]) + ggtitle("Zeng's E"),
  cols = 2)
```

(ref:goutdendros) _Urate and Gout Associated Gene Clustering_ Dendrograns created from hierarchical clustering applied to windows from the `r length(unique(urate_goutGR$SYMBOL))` loci associated with urate and gout. Coloured by super population.


#### Obesity


The GWAS catalog had `r length(unique(obesity_GR$SYMBOL))` genes associated with obesity, this was represented by `r nrow(obesity$mat_d$Tajima.D)` windows. With the exception of \gls{fld}, clustering resulted in the groupings of super populations. Exclusivity of the a single super population to a cluster was also high with the notible exception being the \gls{amr} populations of \gls{clm} and \gls{pur} often being part of the \gls{eur} cluster. The Polynesan population cluster was also closest to \gls{eas} in all except \gls{fld}.

There were `r lower_sig_stats %>% filter(SYMBOL %in% unique(obesity_GR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(SYMBOL) %>% distinct() %>% NROW()` of the obesity associated genes that had evidence from at least one selection statistic of possible selection in any of the Polynesian populations from the lower tail of the distributions. There was a total of `r upper_sig_stats %>% filter(SYMBOL %in% unique(obesity_GR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(SYMBOL) %>% distinct() %>% NROW()` genes that had a significant window in any. 

In the lower tail there were `r lower_sig_stats %>% filter(SYMBOL %in% unique(obesity_GR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(pop, SYMBOL) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==4) %>% NROW()` loci with at least a window that was significant in the Polynesian populations. _ADAMTSL3_, _AGBL4_, _ARL15_, _CCR3_, _CREB1_, _ERBB4_, _FER_, _FHIT_, _GRID1_, _NRXN3_, _TRIP11_, _ZBTB38_ each had in all Polynesian populations, at least one statistic that was significant in the lower tail. _ADCY3_, _ARL15_, _CHST8_, _CMIP_, _ERBB4_, _FHIT_, _FTO_, _GRID1_, _HMGA2_, _LRP1B_, _LTBP1_, _NEGR1_, _NRXN3_, _NT5C2_, _PARK2_, _RARB_ each had in all Polynesian populations, at least one statistic that was significant in the upper tail. There was also a total of `r upper_sig_stats %>% filter(SYMBOL %in% unique(obesity_GR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(pop, SYMBOL) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==4) %>% NROW()` loci that had at least a window that was significant in the upper tail.

<!--
- td excellent groupings, mxl/pel on their own and clm/pur with eur. exclusivity was also pretty much perfect. ep/wp split
- fwh near perfect, get 3/4 amr being clustered with the eur group. ep/wp split
- fld nzm-mxl. eas, afr grouped. eur, amr and sas inter mixed somewhat. cim-sam-ton
- flf nice super populations, eas-pol, eur-2amr, mxl/pel
- ze good super populations, amr-eur and nearly sas, pol-eas with ep/wp split. pretty good exclusivity.

dendrograms
all except fld had nice clusters although amr was split
flf has pol as outlier

exclusivity
td and fwh - only eur/amr shared
flf afr/pol/eas were exclusive
fld pol was in 4 different clusters
flf and fld had smaller groupings so more sharing

prop
td, fwh and ze all except amr were 1.0
fld did ok except for Poly
flf did ok
-->

```{r, fig.cap='(ref:obesitydendros)', fig.height=7}
multiplot(gene_dendros(gene_data = obesity$mat_d$Tajima.D[,-1:-3]) + ggtitle("Tajima's D"),
          gene_dendros(gene_data = obesity$mat_d$Fay.Wu.H[,-1:-3]) + ggtitle("Fay and Wu's H"),
          gene_dendros(gene_data = obesity$mat_d$Fu.Li.D[,-1:-3]) + ggtitle("Fu and Li's D"),
          gene_dendros(gene_data = obesity$mat_d$Fu.Li.F[,-1:-3]) + ggtitle("Fu and Li's F"),
          gene_dendros(gene_data = obesity$mat_d$Zeng.E[,-1:-3]) + ggtitle("Zeng's E"),
   cols = 2)
```

(ref:obesitydendros) _Obesity Associated Gene Clustering_ Dendrograns created from hierarchical clustering applied to windows from the `r length(unique(obesity_GR$SYMBOL))` loci associated with obesity. Coloured by super population.

#### T2D

The GWAS catalog had `r length(unique(t2d_GR$SYMBOL))` genes associated with \gls{t2d}, this was represented by `r nrow(t2d$mat_d$Tajima.D)` windows.


There were `r lower_sig_stats %>% filter(SYMBOL %in% unique(t2d_GR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(SYMBOL) %>% distinct() %>% NROW()` \gls{t2d} associated genes that had evidence from at least one selection statistic of possible selection in any of the Polynesian populations from the lower tail of the distributions. There was a total of `r upper_sig_stats %>% filter(SYMBOL %in% unique(t2d_GR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(SYMBOL) %>% distinct() %>% NROW()` genes that had a significant window in any. 

In the lower tail there were `r lower_sig_stats %>% filter(SYMBOL %in% unique(t2d_GR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(pop, SYMBOL) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==4) %>% NROW()` loci with at least a window that was significant in the Polynesian populations. _ARL15_, _MACF1_, and _RREB1_ each had in all Polynesian populations, at least one statistic that was significant in the lower tail. _ARL15_, _CAMK1D_, _FTO_, _HLA-B_, _HMGA2_, _HNF1B_, _PPP2R2C_, _PTPRD_, _RBMS1_, _SGCG_, _UBE2E2_, and _ZMIZ1_ each had in all Polynesian populations, at least one statistic that was significant in the upper tail. There was also a total of `r upper_sig_stats %>% filter(SYMBOL %in% unique(t2d_GR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(pop, SYMBOL) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==4) %>% NROW()` loci that had at least a window that was significant in the upper tail.





<!--

- td ep/wp split, pol as outlier. pretty good clustering and exclusivity
- fwh ep/wp split, pol-eas as outlier, not bad for exclusivity, clustering was good
- fld nzm-pel. 3/4 pol as outlier group. not too bad for exclusivity
- flf pol-eas, ep/wp. pretty good exclusivity and clusters that are pretty close to the super populations
- ze good exclusivity and good clustering. pol-eas, ep/wp split. amr(clm/pur)-eur

dendrograms
td, fwh, and ze super pops, split amr
flf super pops but split amr and eur together
fld only eas and sas together
all have a wp/ep split but only wp for fld

exclusivity
td ok - but has .5amr, eur and sas together
fwh and flf pretty much same as td
this is because pol is 2 groups
fld poor - amr, eur, sas and afr are all mostly in single cluster
ze good - only has .5 amr and eur together


prop
ze all except amr
td, fwh, and flf have 1.0 for eur,afr and sas. pol is 0.5
fld not bad for all except POL
-->

```{r, fig.cap='(ref:t2ddendros)', fig.height=7}
multiplot(gene_dendros(gene_data = t2d$mat_d$Tajima.D[,-1:-3]) + ggtitle("Tajima's D"),
          gene_dendros(gene_data = t2d$mat_d$Fay.Wu.H[,-1:-3]) + ggtitle("Fay and Wu's H"),
          gene_dendros(gene_data = t2d$mat_d$Fu.Li.D[,-1:-3]) + ggtitle("Fu and Li's D"),
          gene_dendros(gene_data = t2d$mat_d$Fu.Li.F[,-1:-3]) + ggtitle("Fu and Li's F"),
          gene_dendros(gene_data = t2d$mat_d$Zeng.E[,-1:-3]) + ggtitle("Zeng's E"),
   cols = 2)
```

(ref:t2ddendros) _Type 2 Diabetes Associated Gene Clustering_ Dendrograns created from hierarchical clustering applied to windows from the `r length(unique(t2d_GR$SYMBOL))` loci associated with obesity. Coloured by super population.

#### Metabolic Syndrome

The GWAS catalog had `r length(unique(metsyn_GR$SYMBOL))` genes associated with metabolic syndrome, this was represented by `r nrow(metsyn$mat_d$Tajima.D)` windows.


There were `r lower_sig_stats %>% filter(SYMBOL %in% unique(metsyn_GR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(SYMBOL) %>% distinct() %>% NROW()` metabolic syndrome associated genes that had evidence from at least one selection statistic of possible selection in any of the Polynesian populations from the lower tail of the distributions. There was a total of `r upper_sig_stats %>% filter(SYMBOL %in% unique(metsyn_GR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(SYMBOL) %>% distinct() %>% NROW()` genes that had a significant window in any. 

In the lower tail there were `r lower_sig_stats %>% filter(SYMBOL %in% unique(metsyn_GR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(pop, SYMBOL) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==4) %>% NROW()` loci with at least a window that was significant in the Polynesian populations. _DGKB_, _FTO_, _LIPC_, and _MICB_ each had in all Polynesian populations, at least one statistic that was significant in the upper tail. There was also a total of `r upper_sig_stats %>% filter(SYMBOL %in% unique(metsyn_GR$SYMBOL)) %>% group_by(pop, statname, SYMBOL) %>%tally() %>% filter(pop %in% c('CIM','NZM','SAM','TON')) %>% ungroup()%>% select(pop, SYMBOL) %>% distinct() %>% group_by(SYMBOL) %>% tally() %>% filter(n ==4) %>% NROW()` loci that had at least a window that was significant in the upper tail.

<!---

dendrograms

exclusivity


prop
td, fwh, and ze all 1.0 except amr (.75)


- td
- fwh
- fld
- flf
- ze

number of significant
-->

```{r, fig.cap='(ref:metsyndendros)', fig.height=7}
multiplot(gene_dendros(gene_data = metsyn$mat_d$Tajima.D[,-1:-3]) + ggtitle("Tajima's D"),
          gene_dendros(gene_data = metsyn$mat_d$Fay.Wu.H[,-1:-3]) + ggtitle("Fay and Wu's H"),
          gene_dendros(gene_data = metsyn$mat_d$Fu.Li.D[,-1:-3]) + ggtitle("Fu and Li's D"),
          gene_dendros(gene_data = metsyn$mat_d$Fu.Li.F[,-1:-3]) + ggtitle("Fu and Li's F"),
          gene_dendros(gene_data = metsyn$mat_d$Zeng.E[,-1:-3]) + ggtitle("Zeng's E"),
   cols = 2)
```

(ref:obesitydendros) _Metabolic Syndrome Associated Gene Clustering_ Dendrograns created from hierarchical clustering applied to windows from the `r length(unique(metsyn_GR$SYMBOL))` loci associated with metabolic syndrome. Coloured by super population.

<!--
results from other diseases that are late onset and see if they produce the same patterns
-->

## Discussion and Conclusions

- using the chromosome wide results did not always return the super clusters.
- often see admixed amr pops grouped with nzm/cim
- clustering using the gout/urate/bmi/t2d/metsyn genes...
- are the genes found more than you would expect by chance?
- cutree method is rather crude
- what are the strengths/weaknesses of each stat
- Fu and Li's D doesn't cluster well
- when clustering on actual values how much does it matter a population has shifted overall distribution?

all selection stats had afr, eur, eas, sas, pol clustered as single clusters each but split the amr into clm/pur and pel/mxl, except \gls{fwh} which for the negative tail clustered all populations into their super populations perfectly.

One of the possible reasons why the genes give a similar cluster is that for the tests of selection that rely on the frequency spectrum, populations that are similar have a similar allele frequency to each other.

Clustering on the extreme values returned groupings that were consistent with the super population groups.

The clustering of the disease related genes gave the groupings that, with the exception of \gls{fld} were consistent with the super populations.

Using the extremes of the distributions and the gene lists for gout and co-morbidities had clustering that was consistent with the super population groups and largely matched. Clustering on chromosome wide F~ST~ did give the super population groupings and also split the \gls{amr} populations. This was not the case for \gls{td}, \gls{fwh}, gls{fld}, \gls{flf}, and \gls{ze} using entire chromosomes, which did not produce groups that were consistent with the super population groups. \Gls{fwh} and \gls{ze} chromosome wide did the best at grouping the populations.



The cutree method used to assess the proportions and exclusivity of the clustering was relatively crude and did not always match the clusters that were visable in the dendrogram. The use of k = 6 was due to there being 6 super populations, however, \gls{clm}, \gls{pur}, \gls{mxl}, and \gls{pel} being classed as a single super population was not reflected in the clustering. Many of the selection tests split \gls{amr} into 2 groups, CLM/PUR and MXL/PEL. This is consistent with what had previously been found [@ref]. These \gls{amr} populations had also been identified as being admixed and so it was interesting that for some of the statistics XXX they would be clustered with the \gls{nzm} population which is also know to have some admixture [@ref].


this is exploratory, not predictive or creating a model
<!-- what ends of the distribution represent the different types (positive, balancing, negative selection) -->


```{r}
prop_list[['td_neg']]$exclusivity %>% data.frame%>% mutate(pop = rownames(.)) %>% gather("cluster", "prop", 1:6) %>% filter(prop > 0) %>% group_by(pop) %>% summarise(mean(prop))
```

