# Selection and Association Studies

\glsresetall

```{r assocLoad, include=FALSE, warning = FALSE}
source('scripts/Annotate_genes.R')
library(GenomicRanges)
library(phdFunctions)
library(knitr)
library(kableExtra)
library(tidyverse)
source('scripts/05_ukbb_filteredSnps.R') # load in the filtered results for the age,sex,bmi adjusted gwases  (P < 1e-3, nchar(A1) == 1, TEST =='ADD')
knitr::opts_chunk$set(echo = FALSE)
models <- readRDS('data/Ukbiobank/pheno_models.RDS')
pheno_models <- bind_rows(lapply(models, function(x){cbind(adjuster = x$adjuster,broom::glance(x$model))})) %>% arrange(AIC)

markers <- read.table('~/data/NZ_coreExome_1kgp/nz_1kg_markers.txt', stringsAsFactors = FALSE, header = FALSE)
names(markers) <- c("chrom","chrom_start","marker","ref","alt")

get_daf_files <- function(pop){
  map_dfr(list.files( path = "~/data/NZ_coreExome_1kgp/daf/", recursive = TRUE, pattern = pop), ~ read_delim(paste0("~/data/NZ_coreExome_1kgp/daf/",.x), delim = '\t') %>% mutate(pop = pop, chrom = as.integer(str_split(.x, '/') %>% map(., 1) %>% str_sub(start = 4)) ))
}
daf_alleles <-map_dfr(c("CIM","NZM","SAM","TON"), ~get_daf_files(.x)) %>% select(chrom, pos = Pos, Ref, Alt, Anc) %>% distinct() %>% arrange(chrom, pos)

load('~/data/ukbiobank_gout_bd_refined2016-04-26.RData')
fam <- read.table('~/Git_repos/bookdown_thesis/data/Ukbiobank/chr1impv1.fam', stringsAsFactors = FALSE)

# genelists
load('~/data/gwas_catalog/diseaseGR-12-3-2018.RData')
genelists <- data.frame(pheno = c(rep("obesity", length(obesity_GR$SYMBOL)), rep("urate", length( gc_urate_gout_GR$SYMBOL)), rep('t2d', length(t2d_GR$SYMBOL)), rep("kd", length(kd_GR$SYMBOL)), rep("metsyn", length(metsyn_GR$SYMBOL))),  genename = c(obesity_GR$SYMBOL, gc_urate_gout_GR$SYMBOL, t2d_GR$SYMBOL, kd_GR$SYMBOL, metsyn_GR$SYMBOL), stringsAsFactors = FALSE) %>% distinct() %>%  mutate(present = 1) %>% spread(pheno, present, 0)

# f.31.0.0 = sex, 1 = male
# f.48.0.0 = waist
# f.50.0.0 = height
# f.21001.0.0 = BMI
# f.21000.0.0 = ethnicity
# f.20002.* =  self report illness

ukbio_gout <- bd_refined_gout %>% filter(f.eid %in% fam$V2) %>% select(goutaff, 'age' = f.21003.0.0,"bmi" = f.21001.0.0, 'sex' = f.31.0.0, 'height' = f.50.0.0, 'waist' = f.48.0.0, 'hip' = f.49.0.0, 'eth' = f.21000.0.0, febuxostat, allopurinol, colchicine, sulphinpyrazone, gouthosp, goutwinnard, goutself, gout_winnard_self, gout_self_ult, goutself, goutult ) %>% mutate(waist_height_ratio = waist / height, sex_factor = factor(if_else(sex == 0, "female", "male", missing = NULL)), cc = if_else(goutaff == 1, "case", "control", missing = NULL), waist_hip_ratio = waist / hip) %>% filter(!is.na(goutaff) & eth %in% c(1001,1002,1003)) %>% group_by(cc) 
```

This chapter investigates the performance of different gout case definitions on \gls{gwas}. It also looks at how selection statistics could be used for the prioritisation of variants from \gls{gwas}.

A subset of these results have been published as part of @Cadzow2017; the gout definition testing, heritability of gout, and the replication of the @Kottgen2013 results for gout, using the UK Biobank data. The results and discussion from that paper have been included and expanded on here.



<!-- useful papers: -->

<!-- - @Fay2013 -->
<!-- - @Dudley2012 -->

<!-- prioritisation of variants from gwas -->
<!-- - @Hou2013 -->
<!-- - maybe @Kindt2013 -->

<!-- Improvement gwas through the use of imputation: -->

<!-- - Machini nature papers 2010? -->
<!-- - HRC panel paper -->

<!-- refer to gwas done by ukbio in all phenotypes and say it looks similar -->

<!-- Casto2011 -->

## Genetic associations with Gout

### Performance of gout definitions

Genome-wide association studies provide a genome-wide statistical framework for an association analysis between a phenotype of interest, and the genotypes of a genetic marker, generally using linear regression for a continuous trait, or logistic regression for a dichotomous trait [@Lee2011;@Zuk2012]. The association test, in a \gls{gwas} context, is repeated for all available markers across the genome. In epidemiological studies the use of an accurate case definition is important [@Olijhoek2007]. There is variation in the gout definitions used from the multi-purpose cohort studies that are used for genetic analysis. Multi-purpose cohort studies are cohorts that are collected in a way that enables the use of the data to answer many different research questions, some well known cohorts include the \gls{wtccc} cohort, the \gls{aric} study, and the \gls{fhs} [@Burton2007;@Aric1989; @Splansky2007]. 

The use of a consistent case definition is also important for comparisons between cohorts, this can be done by combining the genetic data and performing the \gls{gwas}. Another common way of combining genetic association studies is through meta-analysis. In both instances, having the same case definitions between studies reduces the variation. The power of genetic case-control studies increases with accurate case definition. This means that there are the maximum number of genuine cases, and the maximum number of disease free controls [@Colhoun2003]. For gout, the gold standard case definition is the presence of mono-sodium urate crystals in the synovial fluid of the joint [@Wallace1977a], not possible for multi-purpose cohorts. A substitute for crystals in the synovial fluid is the \gls{acr} criteria [@Wallace1977a]. The criteria require a minimum of six of twelve conditions to be met. It is uncommon for multi-purpose cohorts to report either the gold-standard, or \gls{acr} criteria, but instead, for a self-report diagnosis of conditions and diseases to be reported.

The UK Biobank is a publicly available data set consisting of 500,000 individuals, mostly of European ancestry, from around the United Kingdom, aged between 40 and 69. It contains genetic, health, and lifestyle information. As part of the UK Biobank collection process, participants filled in detailed surveys on health and lifestyle questions, provided biological samples, and provided access to administrative health records. The UK Biobank includes the self-report diagnoses that were collected by survey at the time of recruitment, but it also has hospital admission information, for both primary and secondary diagnosis. This collection of both self-report information combined with hospital admission information provides a unique opportunity in a large cohort to investigate possible gout definitions and their influence on \gls{gwas}.


### Genetic associations for gout in Polynesian populations

As has been previously mentioned (sections \@ref(goutMetSyn), \@ref(chap4Intro)), New Zealand Polynesians have a higher prevalence of gout and metabolic disease [@Winnard2013]. Of the 28 genetic variants associated with serum urate levels that have been previously reported in European populations, 17 also had significant associations with gout [@Kottgen2013]. In the Polynesian populations of New Zealand, nine have replicated the association with gout, with some Polynesian-specific effects being noted [@Phipps-Green2016]. The inherent higher prevalence of gout in Polynesian populations, combined with a lack of adequate sample sizes to perform \gls{gwas} with sufficient power to detect loci, other than the main effect loci of _ABCG2_ and _SLC2A9_, means that alternative approaches are needed to find additional evidence to reduce the number of potentially false negative results.

> What are some of the approaches that can be used?

### Use of selection statistics to inform GWAS

\gls{gwas} in the last few years have required to have larger and larger cohorts, in a quest to discover the source of 'missing heritability' [@Visscher2012;@Yang2017]. The requirement for larger cohorts comes from the inverse relationship between effect-size and statistical power. The source of missing heritability is thought to be found in the polygenic effects of many small effect genetic variants and unmarked uncommon variants of stronger effect sizes [@Spencer2009]. \gls{gwas} also suffer from the need to control the false positive rate and as such have a stringent association significance threshold [@Fadista2016]. One of the side-effects of this threshold is that variants that have a true effect but due to cohort size do not reach statistical significance are lost in the 'noise' of the \gls{gwas} signal. One of the possibilities to be able to identify such variants is to use methodologies that capture the non-additive genetic variance and pair this with the \gls{gwas}. The use of selection statistics in combination with \gls{gwas} is one of these methodologies that can be applied to prioritise \gls{gwas} results [@Ayodo2007, @Field2016a].

### Objectives

- Test performance of gout definitions on \gls{gwas}
- Investigate the use of selection statistics in conjunction with \gls{gwas}

## Methods

### European Gout GWAS {#ukbbgwas}

<!-- acr report using unfiltered controls -->
This \gls{gwas} was performed using the interim data release from the UK Biobank (approval number 12611) downloaded November 2015. There was genetic information for `r NROW(fam)` individuals. For the \gls{gwas} itself, it was limited to be only people who had self reported an ethnic background of British, Irish, or 'any other white background'.

Inclusion criteria were: European ethnicity, aged between 40 and 69 years, and having genome wide genotypes available. The exclusion criteria used were: mismatch between self-reported sex and genetic sex, genotype quality control failure, relatedness, or either a primary or secondary hospital diagnosis of kidney disease (International Classification of Diseases, Tenth Revision (ICD-10), codes I12, I13, N00-N05, N07, N11, N14, N17â€“N19, Q61, N25.0, Z49, Z94.0, Z99.2), participants aged 70 years and over, and those with kidney disease, because these are risk factors for secondary gout.

Individuals from the UK Biobank cohort were genotyped on an Axiom array with 820,967 markers (Affymetrix, Santa Clara, CA, USA). The genotypes were phased using SHAPEIT3, and then imputed using IMPUTE2 with both the UK10K impute reference panel^[https://www.uk10k.org], and the haplotype reference consortium impute panel [@McCarthy2016], to bring the total number of \glspl{snp} to approximately 73.3 million. Genotyping, phasing, and imputation were performed by the UK Biobank.

#### Gout definitions

Four main classification criteria were used to define gout cases. _Hospital defined gout_ involved either a primary or secondary hospital diagnosis for gout (ICD-10 code M10, including sub-codes). _Self-report of gout_ was defined as reporting having gout at the time of the study interview. _Use of \gls{ult}_ used self-report data for the use of allopurinol, febuxostat, or sulphinpyrazone, and not having a hospital diagnosis of leukaemia or lymphoma (ICD-10 codes C81-C96). _Winnard defined gout_ was based on @Winnard2012 and consisted of a hospital diagnosis of gout, or self report of gout specific medication (\gls{ult} or colchicine). The largest number of people were classified through either self-report or the use of urate lowering therapy (Figure \@ref(fig:ukbbupsetplot)). Additional exclusion criteria were applied to participants not classified as having gout from these definitions. The criteria were cortico-steroidal use, non-steroidal anti-inflammatory drug use or probenicid use. 

Confidence intervals for proportions of each gout definition out of the combination of all gout definitions were calculated using the \texttt{prop.test} function from the _stats_ package in R.


```{r, warning = FALSE, eval = FALSE}

bind_rows(lapply(models, function(x){cbind(adjuster = x$adjuster,broom::glance(x$model))})) %>% arrange(AIC)
```

```{r, results = 'asis', eval = FALSE}

broom::tidy(models[[5]]$model) %>% mutate(p.value = format(p.value, scientific = TRUE, digits = 3))%>% kable(., digits = 3, caption = '(\\#tab:phenomodel) Logistic regression for gout adjusted by age, sex, and BMI')
```


```{r, eval = FALSE}
lapply(c(1,2,4,3,5 ), function(x){broom::tidy(models[[x]]$model)})
```

```{r ukbbNumbers}
ukbb_n <- list()
ukbb_n$n_controls <- ukbio_gout %>% filter(goutaff ==0) %>% NROW()
ukbb_n$n_gout <- ukbio_gout %>% filter(goutaff ==1) %>% NROW()
ukbb_n$n_total <- ukbb_n$n_controls + ukbb_n$n_gout
ukbb_n$n_self <- ukbio_gout %>% filter(goutaff ==1 & goutself ==1) %>% NROW()
ukbb_n$n_selfult <- ukbio_gout %>% filter(goutaff ==1 & (goutself ==1 | goutult == 1)) %>% NROW()
ukbb_n$n_ult <- ukbio_gout %>% filter(goutaff ==1 & goutult ==1) %>% NROW()
ukbb_n$n_winnard <- ukbio_gout %>% filter(goutaff ==1 & goutwinnard == 1) %>% NROW()
ukbb_n$n_hosp <- ukbio_gout %>% filter(goutaff ==1 & gouthosp == 1) %>% NROW()

ukbb_male_n <- list()
ukbb_male_n$n_controls <- ukbio_gout %>% filter(goutaff ==0 & sex == 1) %>% NROW()
ukbb_male_n$n_gout <- ukbio_gout %>% filter(goutaff ==1 & sex == 1) %>% NROW()
ukbb_male_n$n_total <- ukbb_male_n$n_controls + ukbb_male_n$n_gout
ukbb_male_n$n_self <- ukbio_gout %>% filter(goutaff ==1 & goutself ==1 & sex == 1) %>% NROW()
ukbb_male_n$n_selfult <- ukbio_gout %>% filter(goutaff ==1 & (goutself ==1 | goutult == 1) & sex == 1) %>% NROW()
ukbb_male_n$n_ult <- ukbio_gout %>% filter(goutaff ==1 & goutult ==1 & sex == 1) %>% NROW()
ukbb_male_n$n_winnard <- ukbio_gout %>% filter(goutaff ==1 & goutwinnard == 1 & sex == 1) %>% NROW()
ukbb_male_n$n_hosp <- ukbio_gout %>% filter(goutaff ==1 & gouthosp == 1 & sex == 1) %>% NROW()

ukbb_female_n <- list()
ukbb_female_n$n_controls <- ukbio_gout %>% filter(goutaff ==0 & sex == 0) %>% NROW()
ukbb_female_n$n_gout <- ukbio_gout %>% filter(goutaff ==1 & sex == 0) %>% NROW()
ukbb_female_n$n_total <- ukbb_female_n$n_controls + ukbb_female_n$n_gout
ukbb_female_n$n_self <- ukbio_gout %>% filter(goutaff ==1 & goutself ==1 & sex == 0) %>% NROW()
ukbb_female_n$n_selfult <- ukbio_gout %>% filter(goutaff ==1 & (goutself ==1 | goutult == 1) & sex == 0) %>% NROW()
ukbb_female_n$n_ult <- ukbio_gout %>% filter(goutaff ==1 & goutult ==1 & sex == 0) %>% NROW()
ukbb_female_n$n_winnard <- ukbio_gout %>% filter(goutaff ==1 & goutwinnard == 1 & sex == 0) %>% NROW()
ukbb_female_n$n_hosp <- ukbio_gout %>% filter(goutaff ==1 & gouthosp == 1 & sex == 0) %>% NROW()

```



#### Gout association test

A \gls{gwas} was performed using logistic regression in Plink1.9b [@plink2; @Chang2015] for each gout classification, with age, sex, and \gls{bmi} as co-variates. The association results for each \gls{gwas} were then compared with the 30 \glspl{snp} reported in Table 1 of @Kottgen2013 that had previously been associated with urate and gout in a cohort of >140,000 Europeans. 


#### Heritabililty

For each gout classification, the genetic component of heritability explained was calculated. This was done by first randomly selecting 10,000 controls and then combining this same subset with cases from each of the gout classifications. The genetic variance was calculated for each chromosome separately and then combined. The dichotomous case-control phenotype was transformed onto a continuous liability scale using the restricted maximum likelihood analysis in the GCTA software (v1.26.0, @Yang2011part) and a general population prevalence of gout of 2% (section \@ref(gctaHeritability)). The liability scale operates on a liability threshold model whereby an unseen continuous trait has a threshold above which one is a case [@Lee2011; @Zuk2012]. Heritability estimates were compared using the formula h~1~ - h~2~ and the SE calculated using formula \@ref(eq:h2se).


\begin{equation} 
 SE = \sqrt{SE_1^2 + SE_2^2}
 (\#eq:h2se)
\end{equation} 

### Polynesian Gout GWAS

A multi-cohort gout dataset was assembled by combining individuals from the Genetics of Gout, Diabetes, and Kidney Disease in Aotearoa cohorts and the Ngati Porou Hauora Charitable Trust cohort. Genotyping on the Inifinium CoreExome v24 bead-chip was performed at the University of Queensland (Centre for Clinical Genomics) for the Genetics of Gout, Diabetes, and Kidney Disease in Aotearoa cohorts and at AgResearch (Invermay Agricultural Centre) for the Ngati Porou Hauora Charitable Trust cohort. The genotype quality control procedures described in section \@ref(coreExomeQC) were applied per cohort, and then as necessary after combining the cohorts. The genotype quality control, and combining of cohorts were performed by Dr Tanya Major, Merriman Lab. \Glsfirst{pca} was performed on the combined dataset with the first 10 components outputted and used as covariates to adjust for population substructure in the \gls{gwas}.

```{r, message = FALSE, warning = FALSE}
poly_fam <- read_delim('~/data/PolyGwas/QC1_7-FullMerge1.0and1.1_ExDup_MattBrown_Czech-plus_polygoutAff.fam', delim = '\t', col_names = FALSE)
poly_phenos <- read_delim('~/data/PolyGwas/coreEx_phenos021216.txt', delim = '\t') %>% select(SUBJECT, AGECOL, SEX, BMI, HEIGHT, WEIGHT, ETHCLASS, ETH_SUM, POLYETHCLASS)
```


The \gls{gwas} was performed for gout using individuals with Polynesian ancestry. Gout affection was determined by the \gls{acr} criteria [@Wallace1977a], or a doctor diagnosis, or if enrolled in a gout drug trial. All participants were older than 18, and provided informed written consent. Exclusion criteria included unknown gout affection status, non-Polynesian self-reported ancestry, or a mismatch between self-reported ancestry and \gls{pca} clustering. The \gls{gwas} was adjusted for age, sex, and the first ten \glspl{pc} calculated from 2,858 ancestry informative markers (as identified by Illumina). Markers were removed if the minor allele frequency was less than 0.01, or had an \gls{hwe} chi-squared exact test P < 1x10^-6^. The \gls{gwas} was conducted by Dr Tanya Major(Merriman Lab) using Plink v1.9b32.


<!-- polynesian_gout-affection.pheno -->
<!--     - gout either ACR criteria or GP diagnosed / in a gout drug trial (only 119 not ACR) -->
<!--     - excluded unknown gout status, non-polynesian (either self-report or PCA mismatch), and people <18 years old -->



## Results

### UK Biobank

#### Participant and association model characteristics

There were genome-wide genotype data available for 105,421 participants, after applying the inclusion and exclusion criteria. The mean \gls{bmi} for all participants was `r ukbio_gout %>%ungroup() %>%  summarise(bmi = mean(bmi, na.rm = TRUE)) %>% pull(bmi) %>% round(., 2)` kg/m^2^, the mean age was `r ukbio_gout %>%ungroup() %>% summarise(age = mean(age, na.rm = TRUE)) %>% pull(age) %>% round(., 2)` years, and `r ukbio_gout %>%ungroup() %>% summarise(percent_male = (1 - sum(sex)/NROW(.)) *100) %>% pull(percent_male) %>% round(., 2)`% were male. A breakdown by affection status for anthropomorphic traits, drug use, and gout classifications is provided in Table \@ref(tab:ukbbclin).

Prior to performing the \gls{gwas} analyses using the different classifications for gout, the best model (determined by \gls{aic} [@Akaike1974]) for phenotype co-variate adjustment was determined by logistic regression. Known co-variates for gout included age, sex, and \gls{bmi}. Each of these co-variates were tested individually and together. Each covariate had a significant association with gout on their own: age (P = \num{`r broom::tidy(models[[1]]$model) %>% filter(term == 'age') %>% .[['p.value']] %>% format(., scientific = TRUE, digits = 3)`}), sex (P = \num{`r broom::tidy(models[[2]]$model) %>% filter(term == 'sex_factormale') %>% .[['p.value']] %>% format(., scientific = TRUE, digits = 3)`}), and \gls{bmi} (P = \num{`r broom::tidy(models[[4]]$model) %>% filter(term == 'bmi') %>% .[['p.value']] %>% format(., scientific = TRUE, digits = 3)`}). In the combined model, with all three covariates there was an \gls{aic} for the model of 19558.30 compared to single variable models for age (\gls{aic} 22706.05), sex (\gls{aic} 21041.80) and \gls{bmi} (\gls{aic} 22001.03). Based on the \gls{aic} for the combined model, age, sex, and \gls{bmi} were used as the covariates in the \gls{gwas} for gout. 



#### Performance of gout classification criteria

There were a total of `r ukbb_n$n_gout` individuals that had any classification of gout and `r ukbb_n$n_controls` controls. Of those with any classification of gout, there were `r ukbb_n$n_hosp` with a hospital diagnosis, `r ukbb_n$n_ult` with urate lowering therapy, `r ukbb_n$n_winnard` with the Winnard criteria, and `r ukbb_n$n_self` with self-reported gout. The overlaps in classification between each group can be seen in Figure \@ref(fig:ukbbupsetplot). There was considerable overlap between the self-report, urate lowering therapy, and Winnard classifications with 1242 individuals meeting all three classifications. There were 571 (27.6%) of 2066 participants who could only be defined through self-report. And for hospital diagnosis, 126 (33.0%) of 382, did not meet the self-report of gout, or \gls{ult} use definition criteria.

The prevalence of gout from all definitions was `r round(ukbb_n$n_gout/ukbb_n$n_control *100,1)`%. This ranged from the lowest prevalence of gout of 0.36%, from the hospital diagnosis definition, through to the highest of 2.18%, by self-report of gout or \gls{ult} usage in the study population (Table \@ref(tab:ukbbprev)).

```{r, results = 'asis'}
  ukbio_gout %>% summarise(n = n(), `Male (\\%)` = round((1-sum(sex)/NROW(.))*100,2), `Mean Age (SD)` = paste0(round(mean(age, na.rm = TRUE),2), " (",round(sd(age, na.rm = TRUE),2), ")" )  , `Mean BMI (SD)` = paste0(round(mean(bmi, na.rm = TRUE),2), " (", round(sd(bmi, na.rm = TRUE),2),")"),  Febuxostat = sum(febuxostat, na.rm = TRUE), Allopurinol = sum(allopurinol, na.rm = TRUE), Sulphinpyrazone = sum(sulphinpyrazone, na.rm = TRUE), Colchicine = sum(colchicine, na.rm = TRUE), Hospital = sum(gouthosp, na.rm =TRUE), `Self-report` = sum(goutself, na.rm=TRUE), Winnard = sum(goutwinnard, na.rm = TRUE), ULT = sum(goutult, na.rm=TRUE) ) %>% t() %>% data.frame() %>% mutate(var = rownames(.)) %>% dplyr::slice(., -1) %>% select(var, 'Case' = 1, 'Control' = 2) %>% kable(., booktabs = TRUE, caption = "(\\#tab:ukbbclin) Clinical details of participants in the UK Biobank.", format = "latex", escape = FALSE, col.names = c("", "Case", "Control"), align = 'lcc') %>% group_rows(., "Gout medication", 5, 8) %>% group_rows(., "Gout classification", 9, 12) %>% footnote(general = "Mean age is reported in years. BMI is reported in kg/m\\\\textsuperscript{2}", escape = FALSE, general_title = FALSE) %>% kable_styling(latex_options = 'hold_position')
```

(ref:ukbbupsetplot) An "upset plot" showing the number of samples (left) and intersections (upper) for the different classification criteria.

```{r ukbbupsetplot, fig.width=5.5, fig.cap='(ref:ukbbupsetplot)', fig.asp=1/1.5 }

listInput <- readRDS('data/Ukbiobank/upsetPlotList.RDS')
UpSetR::upset(UpSetR::fromList(listInput), order.by = "freq", text.scale = 1.3 , nintersects = NA, set_size.angles = 90)
```


```{r, message = FALSE}
entire <- purrr::map_df(names(ukbb_n)[!names(ukbb_n)  %in% c("n_controls", 'n_total', "n_gout")], ~ c(broom::tidy(prop.test(x = ukbb_n[[.x]], n = ukbb_n$n_total )), def = .x, n =ukbb_n[[.x]] )) %>% select(def, n, estimate, conf.low, conf.high) %>% mutate_at(.vars = c("estimate", "conf.low", "conf.high"), .funs = function(x){round(x * 100, 2)}) %>% mutate(study_entire = paste0(n, ", ", estimate,"\\% (",conf.low, "-", conf.high, ")")) %>% select(def,study_entire)

male <- purrr::map_df(names(ukbb_male_n)[!names(ukbb_male_n)  %in% c("n_controls", 'n_total', "n_gout")], ~ c(broom::tidy(prop.test(x = ukbb_male_n[[.x]], n = ukbb_male_n$n_total )), def = .x, n =ukbb_male_n[[.x]] )) %>% select(def, n, estimate, conf.low, conf.high) %>% mutate_at(.vars = c("estimate", "conf.low", "conf.high"), .funs = function(x){round(x * 100, 2)}) %>% mutate(study_male = paste0(n, ", ", estimate,"\\% (",conf.low, "-", conf.high, ")")) %>% select(def, study_male)

female <- purrr::map_df(names(ukbb_female_n)[!names(ukbb_female_n)  %in% c("n_controls", 'n_total', "n_gout")], ~ c(broom::tidy(prop.test(x = ukbb_female_n[[.x]], n = ukbb_female_n$n_total )), def = .x, n =ukbb_female_n[[.x]] )) %>% select(def, n, estimate, conf.low, conf.high) %>% mutate_at(.vars = c("estimate", "conf.low", "conf.high"), .funs = function(x){round(x * 100, 2)}) %>% mutate(study_female = paste0(n, ", ", estimate,"\\% (",conf.low, "-", conf.high, ")")) %>% select(def, study_female)

all_gout <- purrr::map_df(names(ukbb_n)[!names(ukbb_n)  %in% c("n_controls", 'n_total', "n_gout")], ~ c(broom::tidy(prop.test(x = ukbb_n[[.x]], n = ukbb_n$n_gout )), def = .x, n =ukbb_n[[.x]] )) %>% select(def, n, estimate, conf.low, conf.high) %>% mutate_at(.vars = c("estimate", "conf.low", "conf.high"), .funs = function(x){round(x * 100, 2)}) %>% mutate(study_gout = paste0(estimate,"\\% (",conf.low, "-", conf.high, ")")) %>% select(def, study_gout)

full_join(entire, male) %>% full_join(., female) %>% full_join(., all_gout) %>% 
  mutate(def = case_when(def == "n_self" ~ "Self-report of gout diagnosis",
                         def == "n_ult" ~ "ULT use",
                         def == "n_hosp" ~ "Hospital diagnosis",
                         def == "n_winnard" ~ "Winnard definition",
                         def == "n_selfult" ~ "Self-report of gout or ULT use"
                         )) %>% 
 kable(caption = '(\\#tab:ukbbprev) Number, prevalence (95\\% CI) of participants defined as gout cases', col.names = c("Definition", "No. of subjects, prevalence (95\\% CI) in entire study population (n = 105,421)", "No. of subjects, prevalence (95\\% CI) in male participants (n = 51,844) ","No. of subjects, prevalence (95\\% CI) in female participants (n = 53,577)", "Percentage (95\\% CI) of study population with gout using any definition (n = 2432)"),format = 'latex', booktabs = TRUE, escape = FALSE) %>% column_spec(., column = c(2:5), width = '3.5cm')%>% landscape() 

```

#### Replication of @Kottgen2013 urate associated loci

```{r, message = FALSE}
kot_t1 <- read.csv("~/data/kottgen_table1.csv", header = TRUE, stringsAsFactors = FALSE)
uk_kot_t1 <-read_delim(file = 'data/Ukbiobank/kottgen_t1_age_sex_bmi_results.txt', delim = '\t', col_names = c("file","SNP","POS","A1","TEST","NMISS","OR", "SE", "L95", "U95", "STAT", "P")) %>% 
  separate(file, c("file","CHR"), sep = ':') %>% 
  mutate(file = stringi::stri_replace(file, "selfult", regex ="self_ult" )) %>% 
  mutate(gout = map_chr(stringi::stri_split(file, regex = "_"), 1) %>% 
           stringi::stri_sub(9)) %>% 
  left_join(., kot_t1 %>% select(SNP, grail_gene), by = "SNP")

# create the table of kottgen snps from the data
# age, sex, and bmi adjusted
kot_bim <- read_delim('data/Ukbiobank/kottgen_ukbb_alleles.txt', col_names = c("CHR","SNP","GPOS","POS","A1","A2"), delim = " ")

#calculate number of snps that are significant
uk_kot_psig <- uk_kot_t1 %>% select(-file, -TEST, -SE, -NMISS,-STAT, -POS) %>% mutate(P_bac = P) %>% mutate(P = purrr::map_chr(P, ~ format_p_md(p = .x, format = 'latex'))) %>% arrange(P) %>% select(-grail_gene, -OR, -CHR) %>%group_by(gout) %>% mutate(P_sig = P_bac < (0.05/NROW(.)), P_gw = P_bac < 5e-8 ) %>% summarise( P_gw = sum(P_gw),P_exp = sum(P_sig)) %>% filter(gout != 'all')%>% gather("sig", "value",2:3) %>% spread(gout, value) %>% rename("SNP" = sig) %>% mutate_all(as.character) %>% mutate(A1 = "", REF ="",ALT ="")
```

Analysis of the 30 reported urate-associated \glspl{snp} from @Kottgen2013 revealed similar odds ratios for all gout definitions (Figure \@ref(fig:kotplot), Table \@ref(tab:ukbbkot)). Experiment-wide significance was defined as P < `r as.character(signif(0.05/30,2))`, and genome-wide significance as P < 5x10^-8^ [@Fadista2016]. There were differing numbers of \glspl{snp} that had a significant association with gout, both at genome- and the experiment-wide thresholds between the gout definitions used. Meeting genome-wide significance, there were five \glspl{snp} (one in each of _ABCG2_, _GCKR_, _SLC17A3_, _SLC22A12_, and _SLC2A9_) for both the self-reported gout or \gls{ult} usage definition and the self-reported gout definition, four \glspl{snp} (_ABCG2_, _GCKR_, _SLC22A12_, and _SLC2A9_) for the \gls{ult} usage definition, three \glspl{snp} (_ABCG2_, _GCKR_, and _SLC2A9_) for the Winnard definition, and two \glspl{snp} (_ABCG2_ and _SLC2A9_) for the hospital diagnosis definition. For all definitions, the effect size and strength of association was larger for _ABCG2_ than _SLC2A9_. At the experiment-wide significance threshold, there were 13 \glspl{snp} for the self-reported gout or \gls{ult} use definition, 12 \glspl{snp} for the self-reported gout definition, 11 for \gls{ult} use definition, 10 for the Winnard definition, and three \glspl{snp} for the hospital diagnosis definition.


```{r, nkotsig}
n_kot_sig <- uk_kot_t1 %>% group_by(gout) %>% summarise(gw = sum(P < 5e-8), ew = sum(P < 0.05/30))
```





(ref:kotplot) Plot showing ORs (95% CI) for the 30 \glspl{snp} reported in @Kottgen2013 based on gout definitions. Original Kottgen reported ORs are shown as red circles. The reported Kottgen OR was inverted to maintain consistency of reported effect allele (A1) for the following markers: rs2307394, rs2079742, rs7224610, rs7188445, rs1178977, rs6770152, rs7193778, rs164009, rs653178, rs2078267, rs478607, and rs12498742.

```{r kotplot, fig.height=7, fig.cap = '(ref:kotplot)', warning = FALSE}

flip = c("rs2307394", "rs2079742", "rs7224610", "rs7188445","rs1178977", "rs6770152", "rs7193778", "rs164009", "rs653178","rs2078267", "rs478607", "rs12498742")
uk_kot_t1 %>% 
  #mutate(OR = ifelse(SNP %in% flip, 1/OR, OR), L95 = ifelse(SNP %in% flip, 1/L95, L95), U95 = ifelse(SNP %in% flip, 1/U95, U95))  %>%  
  select(gout, SNP, grail_gene,OR, L95, U95) %>% 
  bind_rows(., kot_t1 %>% mutate(gout = "Kottgen") %>% select(gout, SNP, grail_gene, OR) %>% mutate(OR = ifelse(SNP %in% flip, 1/OR, OR) )) %>% unite("SNP", c("grail_gene", "SNP"), sep = '\n') %>% mutate(highlight = ifelse(gout == "Kottgen", "b","a")) %>% filter(gout != 'all') %>% mutate(gout = factor(gout, ordered = TRUE)) %>% mutate(gout = forcats::fct_relevel(gout, "Kottgen", "self","ult","hosp","winnard","selfult"))%>% mutate(gout = forcats::fct_recode(gout, "Hospital" = "hosp", "Self Report" = 'self', "Self Report or ULT" = 'selfult', "ULT" = "ult", "Winnard" = "winnard" )) %>% 
              # mutate(gout = case_when(
              #   gout == 'hosp' ~ "Hospital", 
              #   gout == "self" ~ "Self Report", 
              #   gout == "selfult" ~ "Self Report or ULT",
              #   gout == "ult" ~ "ULT",
              #   gout == "winnard" ~ "Winnard")) %>% 
              ggplot(data = ., aes(x = gout, y = OR, shape = highlight, colour = highlight)) + 
              geom_point() + facet_wrap(~SNP) + geom_linerange(aes(x = gout, ymin = L95, ymax = U95)) +geom_hline(yintercept = 1) +theme_bw() + theme(axis.text.x = element_text(angle=-90, vjust = 0.5, hjust = 0)) + theme(axis.title = element_blank(), legend.position = 'none') + scale_color_manual(values = c("black","red")) 
```

#### Heritability explained

To see if there was a difference between gout classifications in the genetic component of the variance explained by an additive model in gout, heritability estimates were calculated for each gout classification. To do this, 10,000 controls were randomly selected and were used with each different gout criteria, and heritability estimates for each chromosome were calculated using GCTA 1.26.0 [@Yang2011part]. The heritability estimates were 0.289 (SE 0.034) for the self-report of gout or ULT use definition, 0.283 (SE 0.036) for the self-report of gout definition, 0.282 (SE 0.040) for the Winnard definition, 0.308 (SE 0.044) for the ULT use definition and 0.236 (SE 0.160) for the hospital diagnosis definition. There were no statistically significant differences between the heritability estimates for any of the different classifications. The heritability estimates for gout were similar to the estimates of genetic variance of serum urate previously reported [@Kottgen2013].

```{r, message=FALSE}
# still need to flip snps to be in the same order as kottgen
uk_kot_t1 %>% 
  select(-file, -TEST, -SE, -NMISS,-STAT, -POS) %>%  
  left_join(., kot_bim %>% select(SNP, A1, A2), by = c("SNP", "A1")) %>% 
  #mutate(OR = ifelse(A1 != REF, 1/OR, OR), L95 = ifelse(A1 != REF, 1/U95, L95), U95 = ifelse(A1 != REF, 1/L95, U95), mod = ifelse(A1 != REF, 1, 0)) %>% 
  mutate_at(., vars("OR",contains('95')), funs(sprintf("%.2f",. ))) %>% 
  unite("CI",c("L95","U95"), sep = " -") %>%
  mutate(P = purrr::map_chr(P, ~ format_p_md(p = .x, format = 'latex'))) %>%  
  mutate(Effect = paste0(OR, ' [', CI,'], ', P), grail_gene = map_chr(grail_gene, ~ cell_spec(.x, italic = TRUE, format = 'latex'))) %>% arrange(P) %>%  select(-OR, -CI, -P, -CHR) %>%
  spread(key = gout, value = Effect) %>% 
  select(grail_gene, everything(), -all) %>%
  arrange(grail_gene) %>% 
  #bind_rows(., uk_kot_psig ) %>% 
  kable(., format = 'latex', caption = '(\\#tab:ukbbkot) The odds ratios for the 30 variants reported in \\citet{Kottgen2013} for the different gout classification methods.', col.names = c("Grail gene","Marker", "A1", "A2", "Hospital","Self-report","Self-report or ULT","ULT","Winnard"), booktabs = TRUE, escape = FALSE) %>% #group_rows("Number of significant markers", NROW(.)-1, NROW(.)) %>% 
  kable_styling(latex_options = c('scale_down',"hold_position")) %>% 
  footnote(general = "OR [95\\\\% CI], P. ULT = Urate lowering therapy. The reported effect allele is A1. Effects are adjusted by age, sex, and BMI." , general_title = "", threeparttable = TRUE , escape = FALSE) %>% 
  landscape()

```


<!-- Kottgen refs 10-12 put heritability explained for serum urate at 40-70%. Kottgen reported heritability explained in serum urate loci that were adjusted for age and sex from 0.27 to 0.41. -->

\FloatBarrier

### GWAS results


```{r, message = FALSE, warning = FALSE}
gwas_annotate <- function(df){
  txdb_gene_annotate(df %>% 
                       filter(P < 1e-5, nchar(A1) ==1 ) %>% 
                       mutate(chrom = paste0("chr",CHR), chrom_start = BP, chrom_end = BP +1, marker = SNP) %>%
                       select(contains('chrom'), marker) %>% 
                       GRanges()) %>% 
    left_join(., df %>% mutate(start = BP, seqnames = paste0("chr",CHR)), by = c('seqnames','start')) %>%
    select(CHR, SNP, BP, SYMBOL, A1, OR, SE, L95, U95, P ) %>% 
    arrange(P) %>%
    mutate(SYMBOL = ifelse(is.na(SYMBOL), paste0("NA_",CHR,"_",BP) , SYMBOL)) %>% 
    group_by(CHR, SYMBOL ) %>% filter(min(P) == P) %>% 
    arrange(CHR, BP)
}


top_snps <-list()
top_snps$hosp <- gwas_annotate(hosp_asb)
top_snps$self <- gwas_annotate(self_asb)
top_snps$selfult <- gwas_annotate(selfult_asb)
top_snps$winnard <- gwas_annotate(winnard_asb)
top_snps$ult <- gwas_annotate(ult_asb)

nom_snps <-list()
nom_snps$hosp <- hosp_asb %>% filter(P < 1e-5) %>% pull(SNP)
nom_snps$self <- self_asb %>% filter(P < 1e-5) %>% pull(SNP)
nom_snps$selfult <- selfult_asb %>% filter(P < 1e-5) %>% pull(SNP)
nom_snps$winnard <- winnard_asb %>% filter(P < 1e-5) %>% pull(SNP)
nom_snps$ult <- ult_asb %>% filter(P < 1e-5) %>% pull(SNP)
```

#### UK Biobank gout defintions \gls{gwas}

The total number of genome-wide significant \glspl{snp} was `r nom_snps$hosp %>% length()` for the hospital definition, `r nom_snps$self %>% length()` for the self-reported gout definition, `r nom_snps$ult %>% length()` for the \gls{ult} use definition, `r nom_snps$winnard %>% length()` for the Winnard definition, and `r nom_snps$selfult %>% length()` for the self-reported gout or \gls{ult} use definition. There was considerable overlap between the definitions in the \glspl{snp} that met the nominal significance threshold of P < 1x10^-5^ (Figure \@ref(fig:ukOverlap)). There were five main regions of the genome that had peaks of association with gout. These were at _GCKR_ on chromosome 2, _SLC2A9_ and _ABCG2_ on chromosome 4, the _SLC17A1-SLC17A3_ region on chromosome 6, and the _SLC22A11-SLC22A12_ region on chromosome 11 (Figure \@ref(fig:manhatSelfUltAgeSexBmi)). Only the peaks at _SLC2A9_ and _ABCG2_ reached genome-wide significance in the hospital definition.

There were a total of ten top \glspl{snp} from seven genes that were associated with gout at a genome-wide significance threshold, across the definitions used (Table \@ref(tab:ukgwas)). The top \gls{snp} for each locus differed between definitions for _GCKR_ and _SLC2A9_, and _SLC22A12_. _GCKR_ had the top \gls{snp} (rs780093) for self-report of gout, self-report of gout or \gls{ult} use, or \gls{ult} , whereas the top \gls{snp} for the Winnard definition was rs780094. The top \gls{snp} was rs9994216 in _SLC2A9_ for the self-reported gout definition, and rs4697701 for the \gls{ult} usage, self-report of gout, and the Winnard definitions. _SLC22A12_ had rs11231837 for the top \gls{snp} for the self-reported gout and self-reported gout or \gls{ult} use definitions, whereas, rs7929627 was the top \gls{snp} for the \gls{ult} usage, and Winnard definitions. None of the top \glspl{snp} were the same as reported by @Kottgen2013 in Table 1.

The main reason for the difference in the reported top \glspl{snp} is due to minor differences in genotyping rates between the \glspl{snp}. Another reason that explains the difference, is the @Kottgen2013 \glspl{snp} were reported based on association with urate levels, as opposed to gout, where the top \gls{snp} differed between definitions this again can be attributed to genotyping rates. In the case of _GCKR_ and _SLC22A12_ the variants were in complete \gls{ld} but had slightly different genotyping rates. In the case of _ABCG2_ and _SLC17A3_ the reported variants were in complete \gls{ld} with the reported corresponding @Kottgen2013 \gls{snp}. The effect sizes were all extremely similar and consistent with the lead \glspl{snp} that were previously reported by @Kottgen2013.

> need to back up claim about genotyping rates, also workout if the top snps were imputed


<!-- _Why were they different from Kottgen?_ -->


<!-- A reason why they were not necessarily the reported in kotgen table 1 is that the kottgen snps are based on urate -->


<!-- SLC2A9 -->

<!-- - kot = rs12498742 -->
<!-- - uk = rs9994216 / rs4697701 (R^2^ = 0.9356) rs734553 (0.6285 / 0.6929 ) -->
<!-- - r^2^ = 0.7409 / 0.8121 / 0.8617 -->

<!-- ABCG2 -->

<!-- - kot = rs2231142 -->
<!-- - uk = rs4148155 -->
<!-- - r^2^ = 1.0 GBR -->

<!-- GCKR -->

<!-- - kot = rs1260326 -->
<!-- - uk = rs780093 / rs780094 (r^2^ = 1.0) -->
<!-- - r^2^ = 0.8646 / 0.8646 -->

<!-- SLC17A1 -->

<!-- - kot = rs1165151 -->
<!-- - uk = rs1165195 -->
<!-- - r^2^ = 0.9309 -->

<!-- SLC17A3 -->

<!-- - kot = rs1165151 -->
<!-- - uk =  rs1747550 -->
<!-- - r^2^ = 1.0 -->

<!-- SLC22A12 -->

<!-- - kottgen = rs478607 -->
<!-- - uk = rs11231837 / rs7929627 r^2^ = 1.0 -->
<!-- - r^2^ = 0.8485 / 0.8485 -->

<!-- NRXN2 -->

<!-- - kot = rs478607  -->
<!-- - uk = rs10897521 -->
<!-- - r^2^ = 0.8485 -->

(ref:ukOverlap) Overlap of markers that reached nominal genome-wide significance between the different gout classifications. The upper histogram indicates the size of the intersection and the left histogram indicates the total number of nominally significant \glspl{snp} by gout definition.

```{r ukOverlap, fig.cap = '(ref:ukOverlap)'}
listInput <- list(`Hospital` = hosp_asb %>% filter(P < 1e-5, nchar(A1) ==1 ) %>% pull(SNP),
                  `Self-report` = self_asb %>% filter(P < 1e-5, nchar(A1) ==1 ) %>% pull(SNP),
                  ULT = ult_asb %>% filter(P < 1e-5, nchar(A1) ==1 ) %>% pull(SNP),
                  `Self-report or ULT` = selfult_asb %>% filter(P < 1e-5, nchar(A1) ==1 ) %>% pull(SNP),
                  Winnard = winnard_asb %>% filter(P < 1e-5, nchar(A1) ==1 ) %>% pull(SNP)
                  )

UpSetR::upset(UpSetR::fromList(listInput), order.by = "freq", text.scale = 1.3, nintersects = NA, set_size.angles = 90 )
```



```{r}
alleles <- tibble(SNP = c("rs780093","rs4148155", "rs734553", "rs9994216", "rs4697701", "rs1165195", "rs1747550", "rs10897521", "rs11231837", "rs7929627", "rs780094"), 
                      A1 = c("T", "G", "G","G", "A", "T", "A", "T", "T", "G", "T"), 
                      A2 = c("C", "A", "T", "G","G", "G", "G", "C", "C", "A", "C"))

purrr::map_dfr(names(top_snps), ~ top_snps[[.x]] %>% filter(SYMBOL %in% c(kot_t1$grail_gene, kot_t1$closest_gene)) %>% mutate(group = .x)) %>% arrange(CHR,SYMBOL) %>% select(-SE) %>% filter(P < 5e-8) %>% mutate(P = as.character(P)) %>% select(-BP) %>% ungroup() %>% mutate(group = case_when(group == "hosp" ~ 'Hospital',group == 'self' ~ "Self-report", group == "ult" ~ 'ULT', group == 'selfult'~ "Self-report or ULT",group == 'winnard' ~ "Winnard",  TRUE ~ group), P = purrr::map_chr(as.numeric(P), ~ format_p_md(p = .x, format = 'latex'))) %>% mutate(effect = paste0(OR, ' [',L95,'-',U95,'], ',P), SYMBOL = map_chr(SYMBOL, ~ cell_spec(.x, italic = TRUE, format = 'latex'))) %>% left_join(., alleles, by = c("SNP", "A1")) %>% select(CHR, group, SYMBOL,SNP, A1, A2, effect)%>% 
  
  kable(.,col.names = c("Chromosome","Definition","Gene name", "Marker","A1","A2","OR [95\\% CI], P"), digits = 2, caption = '(\\#tab:ukgwas) Top genome-wide significant SNPs by gout definition in the UK Biobank cohort.', booktabs = TRUE, format='latex', escape = FALSE) %>% footnote(general = "A1 is the effect allele.", general_title = "") %>% kable_styling(latex_options = "scale_down")
```






```{r, message = FALSE, warning = FALSE, eval = FALSE}
dat <- hosp_asb 
txdb_gene_annotate(dat %>% filter(P < 5e-8, nchar(A1) ==1 ) %>% mutate(chrom = paste0("chr",CHR), chrom_start = BP, chrom_end = BP +1, marker = SNP) %>% select(contains('chrom'), marker) %>% GRanges()) %>% left_join(., dat %>% mutate(start = BP, seqnames = paste0("chr",CHR)), by = c('seqnames','start')) %>% select(CHR, SNP, BP, SYMBOL, A1, OR, SE, L95, U95, P ) %>% arrange(P) %>% mutate(SYMBOL = ifelse(is.na(SYMBOL), paste0("NA_",CHR,"_",BP) , SYMBOL)) %>% group_by(CHR, SYMBOL ) %>% filter(min(P) == P) %>% arrange(CHR, BP) %>% mutate(P = format(P, scientific = TRUE)) %>% kable()
```




<!-- manhattan plot for the self ult adjusted for age, sex and bmi figure \@ref(fig:manhatSelfUltAgeSexBmi) -->

(ref:manhatSelfUltAgeSexBmi) Manhattan plot for association with gout, adjusted for age, sex and \gls{bmi} using the self-reported gout or \gls{ult} usage gout definition. The dotted line indicates the genome-wide significance threshold level of 5x10^-8^. The genome-wide significant peaks in order are _GCKR_ (chr2), _SLC2A9_ and _ABGC2_ (chr4), _SLC17A1-SLC17A3_ (chr6), and _NRXN2-SLC22A12_ (chr11).

```{r manhatSelfUltAgeSexBmi, echo = FALSE, fig.cap='(ref:manhatSelfUltAgeSexBmi)', fig.width=7, fig.asp=1, out.width= '95%', fig.align='center'}
knitr::include_graphics('images/05_selection_and_association/self_ult_age_sex_bmi.png')
```

```{r manhatAllAgeSexBmi,echo = FALSE, fig.cap='(ref:manhatAllAgeSexBmi)', fig.width=10, fig.asp=1, out.width = '95%', fig.align='center', out.extra='angle=90', eval = FALSE}
knitr::include_graphics('images/05_selection_and_association/all_age_sex_bmi.png')
```

(ref:manhatAllAgeSexBmi) _Manhattan plot for Gout GWAS adjusted for age, sex and BMI_

```{r, engine='block', eval = FALSE, echo = FALSE, include=FALSE }
For this section I want to compare the results for each of the gwases and see if the same 
snps come out and perhaps in the same order.

Also want to compare the results from the age+sex adjusted versus the results from the age+sex+waist+waist:height
```

<!---
ALL GROUP: the most significant snps are from ABCG2. SLC2A9 also features highly. The majority of the significant snps are from these two genes.
--->
\FloatBarrier

#### Polynesian GWAS

```{r, warning = TRUE, message = FALSE}
polyGout <- read_delim('~/data/PolyGwas/polynesian_gout_CoreExome24v1.0-1_rsIDconverted.assoc.logistic.tab', col_names = TRUE, delim = '\t')%>%filter(TEST == "ADD", !(A1 %in% c("I","D") )) %>% separate(., col = "SNP", into = c("SNP","version"), sep = '\\.' , fill = 'right') %>% filter(!duplicated(SNP))



```

The \gls{gwas} for gout in the Polynesian populations of New Zealand had a total of 2402 individuals, with a mean age of 48.6 (SD 14.9), was 60.5% male, and had a mean \gls{bmi} of 35.0 (SD 8.35). A break down by gout affection is shown in Table \@ref(tab:polyGwasclinc). There were only two markers that met the threshold for genome-wide significance, rs2725215 (risk allele = T, OR = 1.939, 95% CI = 1.567-2.40, P = 1.162 x 10^-9^) and rs2231142 (risk allele = T, OR = 2.306, 95% CI =1.891-2.81, P = 1.570 x 10^-16^), both located in _ABCG2_. Two other markers reached the nominal significance threshold, rs2728108 (_ABCG2_) and rs11034401 (inter-genic). Other genes that had \glspl{snp} that were close to the nominal significance threshold, in the 10^-4^ > P > 10^-5^ range, were _LOC105376819, PPP2R2C, SLC2A9, IBSP, MEPE, PKD2, ABCG2, LINC01331, CHN2, BICC1, TRPC2, LOC107984536, DUSP6, DCN, FARP1, LOC101928635, CA10, SLC39A11, ZNF543, LOC105372532, DEFB118, DEFB119_, and _PTPRT_ (Figure \@ref(fig:polyManhat)). This list of genes included the strongest effect loci for serum urate (_SLC2A9_ and _ABCG2_), but the majority of these loci have not previously been associated with gout or serum urate.

> look up the new loci in kottgen/ukbiobank (is BICC1 in kottgen/okada meta?)

```{r}
inner_join(poly_fam, poly_phenos, by = c("X2" = "SUBJECT")) %>% filter(X6 != -9) %>% mutate(sex = case_when(SEX == 'Male' ~ 1, SEX == "Female"~ 0)) %>% group_by(X6) %>% summarise(n = n(), age = paste0(round(mean(AGECOL, na.rm = TRUE),2)," (", round(sd(AGECOL, na.rm = TRUE), 2),")" ), bmi = paste0(round(mean(BMI, na.rm = TRUE),2), ' (',round(sd(BMI, na.rm = TRUE),2),')'),                                                                                          sex = round(sum(sex, na.rm = TRUE)/n()*100, 1))%>% 
  t() %>% data.frame() %>% rownames_to_column() %>% slice(-1) %>% mutate(rowname = case_when(rowname == 'age' ~ 'Mean Age (SD)', rowname == 'sex' ~ 'Sex (\\% Male)', rowname == 'bmi' ~ 'Mean BMI (SD)', TRUE ~ rowname))%>% kable(col.names = c("", "Controls", "Cases"), escape = FALSE, booktabs = TRUE, caption = '(\\#tab:polyGwasclinc) Clinical details for participants of the Polynesian gout GWAS', align = 'lcc') %>% footnote(general = c("Mean age is reported in years. BMI is reported in kg/m\\\\textsuperscript{2}."), escape = FALSE, general_title = "")
```


(ref:polyManhat) Manhattan plot for gout GWAS, adjusted for age and sex, in Polynesian ancestry individuals showing -log~10~(P) for the association test by genomic position. The red line indicates the genome-wide significance threshold of P = 5 x10^-8^, the blue line indicates the suggestive significance threshold of P = 10^-5^.

```{r polyManhat, fig.cap='(ref:polyManhat)'}
polyGout %>% filter(TEST == "ADD", P < 0.05) %>% qqman::manhattan(., ylim= c(0,10) )
```





<!-- chr 1  rs10492987 DDI2 -->
<!-- rs11807599 no gene -->
<!-- rs12750492 no -->

<!-- chr 4 -->
<!-- rs4689443 PPP2R2C -->
<!-- rs3733591 SLC2A9 -->
<!-- rs3775948 SLC2A9 -->
<!-- rs17013182 ABCG2 -->
<!-- rs7698623 -->
<!-- rs2725220 -->
<!-- rs2725215 -->
<!-- rs2728108 -->
<!-- rs1871744 -->
<!-- rs2231142 -->

<!-- chr 5 -->
<!-- rs10515178 LINC01331 -->

<!-- chr 6 -->
<!-- rs639954 no gene -->

<!-- chr 7 -->
<!-- rs6944596 CPVL/CHN2 -->

<!-- Chr9 -->
<!-- rs7040701 no gene -->
<!-- rs7856710 no gene -->
<!-- rs928031 no gene -->

<!-- chr10 -->
<!-- rs6481407 BICC1 -->

<!-- chr11 -->
<!-- rs7115284 AB231779 -->
<!-- rs2271584 TRPC2 -->
<!-- rs11034401 no gene -->

<!-- chr12 -->
<!-- rs4631964 no -->
<!-- rs7955708 no -->
<!-- rs11116180 no -->
<!-- rs2279574 DUSP6 -->
<!-- rs7305095 no -->
<!-- rs7441 DCN -->

<!-- chr13 -->
<!-- rs9517301 FARP1 -->
<!-- rs1866962 no -->
<!-- rs7179609 ALDH1A2 -->
<!-- rs10518968 no -->

<!-- chr17 -->
<!-- rs9905480 no -->
<!-- rs1526189 CA10 -->
<!-- rs4969131 SLC39A11 -->
<!-- rs7502006 SLC39A11 -->

<!-- chr19 -->
<!-- rs13345625 ZNF543 -->

<!-- chr20 -->
<!-- rs6041522 no -->
<!-- rs34173055 DEFB118 -->
<!-- rs12329612 DEFB119 -->
<!-- rs6120065 DEFB119 -->
<!-- rs4281967 PTPRT -->

```{r, eval = FALSE}
polyGout %>% filter(TEST == "ADD",  P < 1e-5) 
```


### Comparison of haplotypic selection with gout \gls{gwas} 

In order to determine if additional loci (which didn't meet the significance threshold in \gls{gwas}), had additional information from the selection tests that could prioritise the \gls{gwas} results, the gout \gls{gwas} and selection results were combined. The "European Ancestry" combination, combined the \gls{gwas} performed in the UK Biobank using the self-reported gout or \gls{ult} usage with the haplotypic selection results for the \gls{gbr} population. The "Polynesian Ancestry" combination, combined the Polynesian gout \gls{gwas} with the haplotypic selection results for the Polynesian populations of \gls{cim}, \gls{nzm}, \gls{sam}, and \gls{ton}. All of the analyses were restricted to the markers of the CoreExome \gls{snp} array. A new threshold for the combinations was used that required an |\gls{ihs}|  > 2 or |\gls{nsl}| > 2 and a \gls{gwas} P < 2 x 10^-4^. the \gls{ihs} and \gls{nsl} threshold was selected because |\gls{ihs}| or |\gls{nsl}| > 2 is equivalent to P < 0.05, since the statistics are very similar to a Z-score. A "probability score" for the combined statistics was calculated using Score~combined~ = P~GWAS~ x P~selection~, where P~selection~ was given by 1 - P(|Z|), and Z was the \gls{ihs} or \gls{nsl} value. With the Polynesian populations, the maximum |value| was used. The combined threshold was set as Score~combined~ < 5 x 10^-8^, to mirror the \gls{gwas} significance threshold.



#### UK Biobank GWAS and haplotypic selection in \gls{gbr}

```{r}
# ihs and nsl sig markers
sig_ihs_val <- readRDS('~/data/NZ_coreExome_1kgp/haplotype/sig_ihs_values_with_genes_14-11-2017.RDS') %>% filter(abs(statvalue) > 2 )
sig_nsl_val <- readRDS('~/data/NZ_coreExome_1kgp/haplotype/sig_nsl_values_with_genes_14-11-2017.RDS') %>% filter(abs(statvalue) > 2 )



```

```{r, message = FALSE}
f <- list.files(path = '~/data/ukbiobank_gwas/filtered_p0.05', pattern = '.*self_ult.*', full.names = TRUE)
selfult_asb_p0_05 <- purrr::map_dfr(f,~ read_tsv(.x, col_names = TRUE ))

ukbb_p05_ihs2 <- sig_ihs_val %>% filter(pop == 'GBR', abs(statvalue) > 2.0 ) %>% inner_join(., selfult_asb_p0_05, by = c("chrom" = "CHR", "chrom_start" = "BP")) %>% select(-gene_chrom, -STAT, -TEST)
ukbb_p05_nsl2 <- sig_nsl_val %>% filter(pop == 'GBR', abs(statvalue) > 2.0 ) %>% inner_join(., selfult_asb_p0_05, by = c("chrom" = "CHR", "chrom_start" = "BP")) %>% select(-gene_chrom, -STAT, -TEST)
```



When the results from the UK Biobank \gls{gwas} for gout (using the self-reported gout or \gls{ult} usage definition) and the \gls{ihs} and \gls{nsl} results for the \gls{gbr} population were combined, there was only one \gls{snp}, rs12638016, with an |\gls{ihs}| > 2.6 (equivalent to the most extreme 1%) that also had P < 10^-5^ in the \gls{gwas}. If the significance threshold for \gls{ihs} was reduced to |\gls{ihs}| > 2 (approximately the 5% extreme most values), then there were three \glspl{snp}, all located at _SLC22A12_ or within 1 kb, that were genome-wide significant in the \gls{gwas} (rs505802, rs9734313, and rs559946), and four \glspl{snp} that were nominally significant. There were `r ukbb_p05_ihs2 %>% filter(abs(statvalue) > 2.0, P< 0.01) %>% NROW()` \glspl{snp} that had an |\gls{ihs}| > 2 and a gout \gls{gwas} of  P < 0.01. This decreased to `r ukbb_p05_ihs2 %>% filter(abs(statvalue) > 2.0, P< 2e-4) %>% NROW() %>% num_to_word()` \glspl{snp}, when the \gls{gwas} P threshold was reduced to 2 x 10^-4^ (Table \@ref(tab:haplogwas)).

Using a threshold of |\gls{nsl}| > 2 and a nominally significant P < 10^-5^, two \glspl{snp} were identified, both on chromosome 6 (Table \@ref(tab:haplogwas)). The first was rs4712972, located in _SLC17A4._ The second was rs501220, located in _SLC17A3_. If the thresholds were relaxed, there were `r ukbb_p05_nsl2 %>% filter(abs(statvalue) > 2.0, P< 0.01) %>% NROW()` \glspl{snp} that had a |\gls{nsl}| > 2 and a gout \gls{gwas} P < 0.01. This decreased to `r ukbb_p05_nsl2 %>% filter(abs(statvalue) > 2.0, P< 2e-4) %>% NROW() %>% num_to_word()`, when a \gls{gwas} P threshold of 2 x 10^-4^ was used (Table \@ref(tab:haplogwas)).


```{r}
ukhaplogwas <- bind_rows(ukbb_p05_ihs2, ukbb_p05_nsl2) %>% 
  filter(abs(statvalue) > 2.0, P < 2e-4) %>% 
  mutate(p_com = P * pnorm(abs(statvalue), lower.tail = FALSE)) %>% 
  #filter(p_com < 5e-8) %>% 
  select(chrom, chrom_start, SNP, genename, statvalue, A1, OR, L95, U95, P, statid, p_com) %>% 
  mutate_at(.vars = c("OR","L95","U95", "statvalue"), funs(three_dp)) %>% 
  mutate(P = purrr::map_chr(P, ~format_p_md(.x, format = 'latex')), p_com = purrr::map_chr(p_com, ~ if_else(.x < 5e-8,format_p_ke(.x, format = 'latex', bold = TRUE, escape = FALSE), format_p_md(.x, format = 'latex')) )) %>% 
  mutate(effect = paste0(OR, ' [',L95,'-',U95,'], ',P)) %>% 
  select(-(OR:P)) %>% rename("pos" = chrom_start) %>% 
  select(chrom, pos, SNP, genename, A1, effect, 'GBR' = statvalue, p_com) %>% left_join(.,daf_alleles, by = c("chrom", "pos"))

ukhaplogwas %>% left_join(., pull(., SNP) %>% unique() %>% rsnps::ncbi_snp_query() %>% as.tibble() %>% select(Query, Gene, Chromosome) %>% mutate(Chromosome = as.integer(Chromosome), Gene = replace_na(Gene,"")), 
                          by = c("chrom" = "Chromosome","SNP" ="Query")) %>% 
  mutate(Gene = map_chr(Gene, ~ cell_spec(.x, italic = TRUE, format = 'latex')))%>% 
  select(Chromsome = chrom, Position = pos, Marker = SNP, `Gene name` = Gene, Ref, Alt, Anc, GBR,A1, `OR [95\\% CI], P` = effect, `Score combined` = p_com) %>%   
  kable(., format = 'latex', booktabs = TRUE, caption = "(\\#tab:haplogwas) SNPs that had |iHS| or |nSL| > 2 from the GBR, 1000 Genomes Project population and a gout association P > 2 x 10\\textsuperscript{-4} in the UK Biobank self-reported gout or ULT definition.", escape = FALSE, align = 'rrlllllrlll') %>% 
  kableExtra::group_rows(., group_label = "iHS",start_row = 1, end_row = 10) %>% 
  kableExtra::group_rows(., "nSL", 11, 17) %>% add_header_above(c(rep("",7), "Selection","GWAS"= 2, "")) %>% 
  footnote(general = c("GWAS results are age, sex and BMI adjusted. Ref is the reference allele in GRCh37, Alt is the alternative allele, Anc is the ancestral allele, and","A1 is the effect allele. Score combined < 5 x 10\\\\textsuperscript{-8} is in bold."), escape = FALSE, general_title = "") %>% 
  kable_styling(latex_options = 'scale_down')

```




```{r, echo = FALSE, eval = FALSE}
ukbb_p05_nsl2 %>% filter(abs(statvalue) > 2.0, P< 1e-4)
```

```{r, echo = FALSE, eval = FALSE}
ukbb_p05_nsl2 %>% filter(abs(statvalue) > 2.6, P< 0.05) %>% ggplot(., aes(x = -log10(P), y = statvalue)) + geom_point()
```



#### Polynesian \gls{gwas} and haplotypic selection in Polynesian populations


```{r, message = FALSE}
#polyGout %>% filter(TEST == "ADD", P < 1e-4) %>% select(-version) %>% distinct() %>% group_by(SNP) %>% tally() %>% arrange(desc(n))

get_haplo_sel_files <- function(pop, stat){
  p <- paste0("~/data/NZ_coreExome_1kgp/haplotype/poly_ihs_nsl_norm/",stat,'/')
  map_dfr(list.files(path = p , recursive = TRUE, pattern = pop), ~ read_delim(paste0(p,.x), delim = '\t', col_names = FALSE) %>% select(marker = X1, chrom_start = X2, stat_value = X7) %>% mutate(pop = pop,statid = stat, chrom = as.integer(str_split(.x, '\\.') %>% map(., 1) %>% str_sub(start = 4)) ))
}

poly_ihs_sig <- map_dfr(c("CIM","NZM","SAM","TON"), ~ get_haplo_sel_files(.x,"ihs")) %>% 
  spread(pop, stat_value) %>% 
  arrange(chrom, chrom_start) %>% 
  filter(abs(CIM) > 2 | abs(NZM) > 2 | abs(SAM) > 2 | abs(TON) > 2)

poly_ihs_sig_genes <- poly_ihs_sig %>% 
  mutate(chrom = paste0('chr',chrom),chrom_end = chrom_start + 1)%>% 
  GRanges() %>% txdb_gene_annotate() %>% 
  mutate(chrom = as.numeric(str_sub(seqnames, start = 4)))
poly_ihs_sig <- left_join(poly_ihs_sig, poly_ihs_sig_genes, by = c('chrom', "chrom_start" = 'start')) %>% select(-seqnames, -end, -width, -strand)

poly_p05_ihs2 <- poly_ihs_sig %>% inner_join(., polyGout, by = c("chrom" = "CHR", "chrom_start" = "BP")) %>%
  filter(P < 0.05)%>% 
  select( -STAT, -TEST, -marker) %>% 
  left_join(., daf_alleles, by = c('chrom', 'chrom_start' = 'pos'))


poly_nsl_sig <- map_dfr(c("CIM","NZM","SAM","TON"), ~ get_haplo_sel_files(.x,"nsl")) %>% 
  spread(pop, stat_value) %>% 
  arrange(chrom, chrom_start) %>% 
  filter(abs(CIM) > 2 | abs(NZM) > 2 | abs(SAM) > 2 | abs(TON) > 2)

poly_nsl_sig_genes <- poly_nsl_sig %>%
  mutate(chrom = paste0('chr',chrom),chrom_end = chrom_start + 1) %>% 
  GRanges() %>% txdb_gene_annotate() %>% 
  mutate(chrom = as.numeric(str_sub(seqnames, start = 4)))

poly_nsl_sig <- left_join(poly_nsl_sig, poly_nsl_sig_genes, by = c('chrom', "chrom_start" = 'start')) %>% select(-seqnames, -end, -width, -strand)

poly_p05_nsl2 <- poly_nsl_sig %>% inner_join(., polyGout, by = c("chrom" = "CHR", "chrom_start" = "BP")) %>% 
  filter(P < 0.05)%>% 
  select( -STAT, -TEST, -marker) %>% left_join(., daf_alleles, by = c('chrom', 'chrom_start' = 'pos'))

polyhaplogwas <- bind_rows(poly_p05_ihs2, poly_p05_nsl2) %>% 
  filter(P< 2e-4) %>% select(chrom, chrom_start, SNP, SYMBOL, Ref,Alt,Anc, CIM,NZM,SAM,TON, A1, OR, L95, U95, P, statid) %>% 
  mutate(statvalue = pmax(abs(CIM),abs(NZM),abs(SAM), abs(TON), na.rm = TRUE)) %>% 
  mutate(p_com = P * pnorm(abs(statvalue), lower.tail = FALSE)) %>% 
  mutate(P = purrr::map_chr(P, ~format_p_md(.x, format = 'latex')), p_com = purrr::map_chr(p_com, ~ if_else(.x < 5e-8,format_p_ke(.x, format = 'latex', bold = TRUE, escape = FALSE), format_p_md(.x, format = 'latex')) )) %>% 
  mutate_at(c("NZM","CIM","SAM","TON", "OR"), funs(three_dp)) %>% 
  mutate_at(c("NZM","CIM","SAM","TON"), funs(case_when(. == 'NA' ~ "-", TRUE ~ .))) %>% 
  mutate(effect = paste0(OR, ' [',L95,'-',U95,'], ',P)) %>% 
  select(-(OR:P)) %>% 
  arrange(chrom, chrom_start) %>% 
  rename("pos" = chrom_start) %>% 
  arrange(statid,chrom, pos) %>% replace_na(list(SYMBOL = ""))
```


Combining the results of the \gls{gwas} for gout in Polynesians with the results of the \gls{ihs} analysis produced a single marker that met the threshold of an |\gls{ihs}| > 2.6 and a \gls{gwas} P < 5 x 10^-8^; that \gls{snp} was rs2725215, located in _PKD2._ Rs2725215 also had a |\gls{nsl}| > 2.6. Both \gls{ihs} and \gls{nsl} thresholds were only met for \gls{ton}. There were `r poly_p05_ihs2 %>%  NROW()` \glspl{snp} that had a \gls{gwas} P < 0.05 and an |\gls{ihs}| > 2; this reduced to `r polyhaplogwas %>% filter(statid == 'ihs') %>% NROW()` at the combined threshold of |\gls{ihs}| > 2 and P < 2 x 10^-4^ (Table \@ref(tab:polyhaplogwas)). Combining the \gls{nsl} results with the \gls{gwas}, there were `r poly_p05_nsl2 %>% NROW()` \glspl{snp} that had a \gls{gwas} P < 0.05 and a |\gls{nsl}| > 2; this reduced to `r polyhaplogwas %>% filter(statid == 'nsl') %>% NROW()` at the combined threshold of |\gls{nsl}| > 2 and P < 2 x 10^-4^ (Table \@ref(tab:polyhaplogwas)).
 
At the combined threshold of |\gls{ihs}| > 2 or |\gls{nsl}| > 2 and a \gls{gwas} P < 2 x 10^-4^, there were seven \glspl{snp}, located within  _IBSP_, _PKD2_, _ABCG2_ and _PPCDC_, found with both \gls{ihs} and \gls{nsl} (Table \@ref(tab:polyhaplogwas)). _CHN2/CPVL_ and _SLC39A11_ only had \glspl{snp} that met the \gls{nsl} threshold. Out of all of the \glspl{snp} that met the combined threshold, ten \glspl{snp} has evidence from both \gls{ihs} and \gls{nsl}. Rs12908919 (_PPCDC_) was the only \gls{snp} that showed significance in multiple populations (\gls{sam} and \gls{ton}) and had support from both \gls{ihs} and \gls{nsl} (Table \@ref(tab:polyhaplogwas)), whereas, there were six \glspl{snp} that had support from only the \gls{ton} population, for both \gls{ihs} and \gls{nsl}. The loci represented were different to the \glspl{snp} in the European ancestry \gls{gwas} and selection analysis.

The region containing _IBSP_, _PKD2_, and _ABCG2_ had multiple \glspl{snp} that met the suggestive significance threshold. Of those, six \glspl{snp} had evidence of both selection and association (Figure \@ref(fig:locusZoom), Table \@ref(tab:polyhaplogwas)). The selection signal came only from the Western Polynesian populations, with no indication in the Eastern Polynesian populations of selection in this region. The recombination rate in this region had intermittent small spikes, with three of the variants in _PDK2_ having an \gls{ld} R^2^ > 0.4 with rs2231142 (calculated using \gls{eas}). These variants also had an association with gout that met the suggestive significance threshold.

(ref:locusZoom) Locus zoom plot of Polynesian gout \gls{gwas} results for the region chr4:88,600,000-89,200,000, covering _IBSP_, _PKD2_, and _ABCG2_. Points indicate SNP association with gout by genomic position in the Polynesian gout GWAS. Linkage disequilibrium with rs2231142 is indicated by colour. Recombination rate in the East Asian super population of the 1000 Genomes Project is indicated by the blue line. The dashed line indicates the Genome-wide significance threshold P = 5 x 10^-8^.

```{r locusZoom, out.width='80%', fig.cap='(ref:locusZoom)', fig.align='center'}
include_graphics('images/05_selection_and_association/polygwas_locuszoom.ASN.chr4.88.6-89.2mb-1.png')
```

```{r, eval = FALSE}
sig_ihs_val %>% filter(superpop == 'POL') %>% inner_join(., polyGout %>% filter(P < 5e-2, TEST == "ADD"), by = c("chrom" = "CHR", "chrom_start" = "BP")) %>% spread(pop, statvalue) %>% arrange(P) %>% filter((!is.na(CIM) & !is.na(NZM)) | (!is.na(TON) & !is.na(SAM))) %>% filter(abs(NZM) > 2.6 | abs(CIM) > 2.6 | abs(SAM) > 2.6 | abs(TON) > 2.6) %>% mutate( P = purrr::map_chr(as.numeric(P), ~ format_p_md(p = .x, format = 'latex'))) %>% mutate(effect = paste0(OR, ' [',L95,'-',U95,'], ',P)) %>% select(-superpop,-(OR:P), -gene_start, -gene_end, -TEST, -statid, -contains('gene_'), -ensgeneid, -NMISS, -sig, -STAT, -chrom_end) %>% select(-version)%>%  kable(., longtable = TRUE, booktabs = TRUE, format = 'latex', digits = 3) %>% kable_styling(font_size = 10)
```

```{r, eval = FALSE}
sig_ihs_val %>% filter(superpop == 'POL') %>% inner_join(., polyGout %>% filter(P < 0.01, TEST == "ADD"), by = c("chrom" = "CHR", "chrom_start" = "BP")) %>% ggplot(., aes(x = -log10(P), y = statvalue, colour = pop)) + geom_point() + geom_hline(yintercept = c(2,-2))
```


```{r, eval = FALSE}
sig_nsl_val %>% filter(superpop == 'POL') %>% inner_join(., polyGout %>% filter(P < 0.01, TEST == "ADD"), by = c("chrom" = "CHR", "chrom_start" = "BP")) %>% ggplot(., aes(x = -log10(P), y = statvalue, colour = pop)) + geom_point()
```

```{r}
poly_n <- list()
poly_n$tenNeg4 <- polyGout %>% filter(TEST == "ADD", P < 1e-4) %>% select(-version) %>% distinct() %>% group_by(SNP) %>% tally() %>% NROW()
poly_n$ihs <- sig_ihs_val %>% filter(superpop == 'POL') %>% inner_join(., polyGout %>% filter(P < 1e-4, TEST == "ADD"), by = c("chrom" = "CHR", "chrom_start" = "BP")) %>% pull(SNP) %>% unique()
poly_n$nsl <- sig_nsl_val %>% filter(superpop == 'POL') %>% inner_join(., polyGout %>% filter(P < 1e-4, TEST == "ADD"), by = c("chrom" = "CHR", "chrom_start" = "BP")) %>% pull(SNP) %>% unique()
```



```{r, eval = FALSE}
# CNT4N
polyGout %>% filter(between(BP, 2140497, 3099645), CHR == 3, P < 0.05) %>% arrange(P)
```


<!-- CNTN4 has some of the strongest signal of selection from \gls{ihs} and \gls{nsl} (section \@ref(polyihs)). The association with gout in Polynesian had four \glspl{snp} with P < 0.05, rs9849237 (P = 0.03173), rs2320375 (P = 0.03250), rs2728076 (P = 0.03468), and rs6763008 (P = 0.03932) -->

<!-- > PPARGC1A -->

<!-- rs1873532 ld with rs8192678 is 0.976 in EAS, no sign of association at that snp, but there is an 2.6 > ihs > 2 for nzm/sam/ton. association consistent with cadzow2016, selection just outside of the 1% threshold  -->



<!-- - The \glspl{snp} with the most extreme \gls{ihs} and \gls{nsl} values tended to have the the weakest associations with gout. -->
<!-- - there were more snps with nsl than ihs -->


```{r, eval = FALSE}
 sig_ihs_val %>% filter(superpop == 'POL') %>% inner_join(., polyGout %>% filter(P < 1e-4, TEST == "ADD"), by = c("chrom" = "CHR", "chrom_start" = "BP")) %>% select(-gene_chrom, -TEST, -ensgeneid, -STAT, -sig, -statid)  %>% mutate( P = purrr::map_chr(as.numeric(P), ~ format_p_md(p = .x, format = 'latex'))) %>% mutate(effect = paste0(OR, ' [',L95,'-',U95,'], ',P)) %>% select(-superpop,-(OR:P), -gene_start, -gene_end) %>% kable(., format = 'latex', longtable = TRUE, booktabs = TRUE)
```


```{r}
polyhaplogwas %>% 
  left_join(., pull(., SNP) %>% unique() %>% rsnps::ncbi_snp_query() %>% as.tibble() %>% select(Query, Gene, Chromosome) %>% mutate(Chromosome = as.integer(Chromosome), Gene = replace_na(Gene,"")), 
            by = c("chrom" = "Chromosome","SNP" ="Query")) %>% 
  mutate(Gene = map_chr(Gene, ~ cell_spec(.x, italic = TRUE, format = 'latex'))) %>% 
  select(Chromosome = chrom, Position = pos, Marker = SNP, `Gene name` = Gene, Ref, Alt, Anc, CIM, NZM, SAM, TON, A1, `OR [95\\% CI], P` = effect, `Score combined` = p_com) %>% 
  kable(., format = 'latex', booktabs = TRUE, #align = 'rrlllllrrrrlll', # unsure why alignment makes the table not display properly
        caption = "(\\#tab:polyhaplogwas) SNPs that had |iHS| or |nSL| > 2 in the New Zealand Polynesian populations and a gout association P > 2 x 10\\textsuperscript{-4} in the Polynesian GWAS.", escape = FALSE) %>% group_rows(., 'iHS', 1, 11) %>% group_rows(., "nSL", 12, 25) %>% 
  kableExtra::add_header_above(c(rep("",7), "Selection" = 4, "GWAS" = 2)) %>% 
  footnote(general = c("GWAS results are age, sex, and PCA adjusted. Ref is the reference allele in GRCh37, Alt is the alternative allele, Anc is the ancestral allele, and", "A1 is the effect allele. Score combined < 5 x 10\\\\textsuperscript{-8} is in bold."), escape = FALSE, general_title = "") %>%
  kable_styling(latex_options = 'scale_down')
```




## Chapter Discussion

<!-- - Concepts, estimation and interpretation of SNPbased heritability - Yang2016 -->
<!-- - Identifying genetic variants that affect viability in large cohorts - Mostafavi2017 -->

### Performance of gout definition



<!-- > _ukbio paper_: Accurate and consistent definitions of the case and control states is important to have the maximise power for case-control cohort studies. Consistency of definition also provides better replication of association across differing cohorts [\@ref]. The analysis of the UK Biobank data showed that the self-report of gout or \gls{ult} use definition had the highest number of gout cases, and the best precision from the definitions used, for association with gout. These findings are consistent with the recent SUGAR study -->

> explain sugar paper a little

The UK Biobank had a larger number of cases for the gout \gls{gwas} than @Kottgen2013. This increase in cases reflects the increase in strength of association that was seen. In the SUGAR paper [@Dalbeth2016], the definition that provided the highest specificity (82%) and sensitivity (72%) was self-reported gout or on \gls{ult} when tested against the mono-sodium urate crystal identification as the gold standard. Consistent with this, the work presented here found that the self-report of gout or \gls{ult} use definition also provided the highest precision from the definitions used, and supports the use of this definition when better gout classification methods are not available, such as \gls{acr} criteria or mono-sodium urate crystal identification.

The different definitions of gout used in this study may represent the different disease presentations or patient populations. Not all patients were captured by all criteria, and the Winnard definition was not as good as self-report of gout or \gls{ult}, despite a similarity in definitions, and also had lower precision in the genetic association. The hospital diagnosis definition had the lowest prevalence and was restrictive, making it the least likely to capture most people with gout. One third of the people that met this definition did not meet any of the self-report definitions, for gout, and/or \gls{ult} usage. There are a few reasons that might explain this, first the hospitalised population may have a different disease presentation from those in the community identified by self-report or \gls{ult} use. Secondly, a diagnosis of gout made in a hospital by subsequently be revised to a different diagnosis; this is not taken into account with the current methodology. Therefore, when self report information is available it is recommended that the self-report of gout or \gls{ult} use be used as the definition to define cases.

Each of the gout definitions tested had genome-wide significance for \glspl{snp} associated with gout in _ABCG2_ and _SLC2A9_. These two genes encode proteins that transport urate in the gut (ABCG2) and proximal tubule of the kidney (SLC2A9). The large effect sizes in the association with gout, show similarity with their large effect sizes in the control of serum urate levels [@Kottgen2013].

Heritability estimates for the variance attributed to the additive effects of common \glspl{snp} of 0.282-0.308 (excluding the hospital definition) were in line with those reported by @Kottgen2013, with the same approach using the GCTA software of 0.27 to 0.41 for serum urate levels. There was also no statistically significant differences between the heritability estimates for the different definitions. The comparability between the heritability estimates for gout and serum urate by common genetic variants suggests the genetic heritabilities of serum urate levels and gout is similar. However, also contributing to the risk of gout are environmental factors, such as diet and medications. Because the estimates are constructed under an additive model, it does not take into account the non-additive variance, such as gene-gene or gene-environment interactions, or the effect of structural variants such as copy number variation.



<!-- - the gwas is also very similar to that which was done as part of the phenome-wide gwas -->

<!-- - all definitions tested abcg2/slc2a9 were genome-wide significant -->
<!-- - large effect sizes -> effect sizes in GWAS of control of SU levels - consistent with their role in regulation of SU and gout risk -->
<!-- - heritability estimates for variance explained by all common snps 0.282-0.308 (ex hospital def) -->
<!-- - within the range from kottgen (0.27-0.41) and they also calculated using GCTA for SU -->
<!-- - estimates of variance explained in european by common snps are comparable, suggesting that the common genetic variant-mediated heritabilities of SU and gout are similar -->
<!-- - diet, medications and other environmental factors contribute to risk of gout -->
<!-- - heritabilities are estimated under an additive model, so don't include non-additive variance (GxG and GxE), rare variants, CNV -->
<!-- - limited to European -> might not translate to other populations -->
<!-- - limited in that no gold-standard available to determine FPR/FNR -->




### \gls{gwas} and selection

The main peak of association in the Manhattan plots (Figures \@ref(fig:manhatSelfUltAgeSexBmi) and \@ref(fig:polyManhat)) for the \gls{gwas} analyses of both the UK Biobank, and the New Zealand Polynesians was _ABCG2_, this was consistent with previous \gls{gwas} analyses for gout, compared to \gls{gwas} for serum urate which has the strongest association with _SLC2A9_ [@Kottgen2013; @Okada2012; @nakayama2017gwas]. Combining the haplotypic selection results for \gls{ihs} and \gls{nsl} from the \gls{gbr} population with the UK Biobank gout \gls{gwas} showed additional evidence at _SLC22A12_, _PTPN11_, _SLC17A3_, _SLC17A4_ , and _ADAM10_. However, from the \gls{gwas} those genes had already met the suggestive significance threshold of P < 10^-5^. All except _ADAM10_ had previously been reported as being significant at a genome-wide threshold for association with serum urate levels [@Kottgen2013]. _ADAM10_ had not been previously associated at a genome-wide threshold with either gout or serum urate levels, but had previously been reported as a strong candidate for positive selection [@Deschamps2016]. _ADAM10_ is involved in the innate immune system with the Notch signalling pathway, but also becomes down-regulated during interaction with the extracellular proteins PfSEL1/PfSEL2 of _P. falciparum_ [@Singh2009].

The combination of the New Zealand Polynesian gout \gls{gwas} with the haplotypic selection statistics of \gls{ihs} and \gls{nsl} from the Polynesian populations of \gls{cim}, \gls{nzm}, \gls{sam}, and \gls{ton} did not have any corroboration with the equivalent combination of analyses in the European ancestry cohort. The \gls{ton} population was the main source of the selection statistics that met the threshold, this could be due to the high proportion of gout patients (> 50%) that were included in the sample. Between the \gls{ihs} and \gls{nsl} based results, there was a high degree of overlap in the \glspl{snp} that met the thresholds for both, although \gls{nsl} had three \glspl{snp} that didn't meet the threshold in \gls{ihs}.

<!-- > Rs12908919 (_PPCDC_) was the only \gls{snp} that had more than one Polynesian population with both \gls{ihs} and \gls{nsl} in the top 5%. _PPCDC_ encodes phosphopantothenoylcysteine decarboxylase -->

The _IBSP/PKD2/ABCG2_ region has been associated with both gout and urate levels [@Yang2010], but often just the lead \gls{snp} of rs2231142 is reported, as it has the strongest association and is one of the most likely causative variants due to the Q141K amino acid change that leads a reduction in expression and a less efficient transporter [@Woodward2009]. The _IBSP_ locus encodes bone sialoprotein, which is involved with bone development [@Kerr1993]. Loci nearby such as _MEPE_ and _SPP1_, encode proteins with similar structure and functions, suggesting a shared evolutionary history [@Rowe2000]. In the presence of urate crystals, as with gout, _IBSP_ expression is reduced and affects osetoblast differentiation [@Chhana2011]. _PKD2_ is involved with kidney disease [@Mochizuki1996; Hildebrandt2010], and Polynesian populations have a 3.5 fold higher incidence of end-stage renal disease, compared to Europeans [@Collins2017book]. For all the variants that had both selection and \gls{gwas} signal at _PKD2_, the selection signal was for the ancestral allele, which was also the risk allele for gout. The selection signal was absent in the Eastern Polynesian populations, which displayed selection statistics in favour of the derived/protective gout alleles at _PKD2_. As previously discussed (section \@ref(selectionDissussion)), many of the loci that have evidence suggesting selection in Western Polynesian populations are calcium channels. The _IBSP/PKD2/ABCG2_ region, is functionally involved with calcium, bone sialoprotein has a high affinity for calcium [@Kerr1993], and PKD2 has some homology with calcium channels, containing domains that are consistent with calcium binding [@Mochizuki1996], again demonstrating this calcium-related selection pattern.


<!-- Tony: ADAM10 - any nominal evidence for assocation with SU/gout? -->
<!-- Tony: ISBP - interesting -> could be a MSU crystal response locus, including (ABCG2)? -->

<!-- potential reference for PKD2 and urate if needed \cite{Lee2012} -->
<!-- korean association of rs2725220 with urate Sull2014 -->

<!-- difference between the loci from the ukbb/gbr combination versus the poly/poly combination. -->

<!-- - ukbb/gbr combo had differences even between the ihs/nsl, only 3? were consistent (what could ADAM10) do? -->
<!-- - poly had many consistent genes between the ihs and nsl -->
<!-- - mostly driven by TON - is this due to the TON pop being very high in gout? -->
<!-- - poly has the ibsp/pkd2/abcg2 region as being influential -->
<!--     - ibsp = bone density?, pkd2 is kidney related I think, and abcg2 is well recognised for gout -->
<!--     - PPCDC possible malarial link? how could the malaria bit be linked to the gout bit? -->
<!--         - also looks like it might be a western polynesian loci -->

<!-- lbsp has something to do with bones and urate? http://ard.bmj.com.ezproxy.otago.ac.nz/content/70/9/1684.long -->

<!-- The main peak in the polynesian gwas is the same as the ukbiobank, abcg2 - this is different to kottgen, with the serum urate levels association where slc2a9 is the main effect loci -->

<!-- - The ukbiobank had quite different loci that fell at the suggestive significance level compared to the poly gwas -->
<!-- - the poly gwas didn't highly defined peaks in the manhattan plot, looks rather noisy and this can largely be attributed to the sample size -->


### Limitations

In the analysis of the different gout definitions, there was a difference between the the definitions in the number of \glspl{snp} that had a significant association with gout, with the hospital diagnosis definition having the fewest. One of the key differences between the definitions was the number of cases defined for each, ranging from 382 to 2295, this will have impacted on the power for each \gls{gwas} to detect associations. 
Similar to the \gls{gwas} results from the different gout definitions, the differences between heritability estimates of the different definitions may too be impacted by the differing numbers of cases. Due to computational reasons, only a subsample of the controls was used in the calculation of the heritability estimates. The use of all of the controls may have also changed the results of the heritability analysis. Another limitation of the definitions was the lack of 'gold standard' to which the definitions could be compared, and therefore sensitivity and specificity was unable to be calculated for each definition. The ancestry of the UK Biobank samples included for the gout definitions were European, this means that results and conclusions from the definition analysis may not be applicable to other ancestral backgrounds. 

With the combination of the selection statistics with the results of the gout \gls{gwas} analyses, the variants that were able to be compared were limited by the selection analysis, where the selection results were only from the markers that were available on the CoreExome \gls{snp} array. This is despite the UK Biobank \gls{gwas} being conducted on an imputed dataset, where there were results for 9.3 million markers after filtering. This compared to the total number of 236,868 markers for which there were \gls{ihs} or \gls{nsl} results. It is possible that if there were \gls{ihs} and \gls{nsl} results for a greater number of markers, more loci may have been prioritised. The Polynesian \gls{gwas} was limited by sample size, and also in the number of markers for which there were both association and selection results.The marker density of the CoreExome \gls{snp} array could have been improved through imputation, however, for Polynesian populations, the public haplotype reference panels lack representation of Polynesian populations. This means that Polynesian specific variants (such as rs373863828, @Minster2016) and therefore haplotypes, are not incorporated in the imputation, and as a result the imputation accuracy suffers [@Howie2011]. 

The combination of the Eastern and Western Polynesian populations into a single grouped population for the \gls{gwas} may also affect the results, as there are population specific effects, even between similar ancestral backgrounds such as the East and West Polynesian populations. Such differences with respect to gout have been seen with _ABCG2_ [@Phipps-Green2010]. This was similar to what was observed in the selection results at the extended locus including _IBSP, PKD2, and ABCG2_ displaying a difference between Eastern and Western Polynesian populations.

<!-- Tony: will be interesting to repeat you selection on the imputed WP data (after submission) -->


<!-- - poly gwas -->
<!--     - the lack of power -> samplesize -->
<!--     - data set wasn't imputed -->
<!--     - imputation of the Polynesian data is difficult, lack of reference haplotypes -->

<!-- - Poly selection/gwas -->
<!--     - the selection results could be because of higher than general population prevalence of gout in the cohort -->
<!--     - reduced markers used -->

<!-- - EUR selection/gwas -->
<!-- - the greatly reduced number of markers because the selection results were calculated in the coreexome markers -->


### Conclusions

The analyses in this chapter for performance of the case-definition for gout association studies showed that in the absence of the gold-standard of observing urate crystals in the synovial fluid, or the \gls{acr} criteria, it is recommended to use the self-report or \gls{ult} definition, as this gives the best performance, out of the definitions tested in this analysis. It was also shown that incorporation of selection analyses with \gls{gwas} does provide evidence for additional evidence for variants associated with gout, despite these associations already mostly being nominally significant from \gls{gwas}, the selection analyses do provide an avenue for prioritisation of \gls{gwas} results. Further work such as incorporating multiple selection statistics, and the use of whole genome sequenced data or imputed genetic data would be useful to improve the prioritising of \gls{gwas} results.
  

<!-- When performing genetic association studies, and only self-report information is available, i -->
<!-- - when better definitions are not available, self-reported gout or \gls{ult} is recommended -->
<!-- - combining selection results with gwas may suggest additional loci of interest, but \gls{gwas} itself will still pick them up -->

<!-- > Maybe bring up the current discussion about the gcta for the use of heritability versus ldak? -->

<!-- TANYA -->

<!-- Whole-genome principal component analysis vectors were calculated using a subset of 2,858 ancestry informative markers (as identified by Illumina) extracted from the CoreExome whole-genome genotypes. The SmartPCA (EIGENSOFT v6.0.1) [25]) program was used to perform this prinicipal component analysis, with an output of 10 eigenvectors, no outlier removal, and no population size limit. Individuals of non-Polynesian ancestry were included in the analysis group, and the first four vectors plotted against each other to view the clustering of ancestral groupings (Asian, European, Eastern Polynesian, and Western Polynesian). The first four vectors were chosen for inclusion as covariates in the linear regression to account for population stratification and cryptic relatedness, which may cause confounding. -->
<!-- The Illumina Infinium CoreExome v24 bead chip platform was used to genotype participants for ~500,000 variants across the whole-genome. Genotyping was performed at the University of Queensland (Centre for Clinical Genomics) for the Genetics of Gout, Diabetes, and Kidney Disease in Aotearoa cohorts and at AgResearch (Invermay Agricultural Centre) for the Ngati Porou Hauora Charitable Trust cohort. Bead chip genotyping batches were auto-clustered using GenomeStudio v2011.1 software (Illumina, San Diego). The Illumina GenomeStudio best practice guidelines and quality control protocols of Guo et al. (2014) were applied to these auto-clustered genotypes to ensure final genotype calls were of the highest possible quality [23, 24]. After the initial quality control protocols were applied the genotyping batches were merged and relevant quality control steps repeated in the full dataset. -->



<!-- How I did the Polynesian GWAS(s): -->
<!-- Run from: -->
<!-- /Volumes/userdata/staff_users/tanyaflynn/ShortCuts/murray_working_dir/coreExome/Batch_Merge/Merged_Merriman_Brown_Matsuo/PlusStrand_Merge -->
<!-- File Set-Ups -->
<!-- QC1_7-FullMerge1.0and1.1_ExDup_MattBrown_Czech-plus_rsIDconverted -->
<!--     - merged QC batches 1 to 7, matt brown's samples & the czech samples -->
<!--     - excluded duplicate samples -->
<!-- polynesian_gout-affection.pheno -->
<!--     - gout either ACR criteria or GP diagnosed / in a gout drug trial (only 119 not ACR) -->
<!--     - excluded unknown gout status, non-polynesian (either self-report or PCA mismatch), and people <18 years old -->
<!-- QC1_7-FullMerge1.0and1.1_ExDup_MattBrown_Czech-plus_polygout.cov -->
<!--     - sex, age, and PCAs 1-10 (PCAs calculated in full genotype group) -->
<!-- east-poly_list.txt -->
<!--     - anyone self-reporting east polynesian with a PCA labelled east polynesian or general polynesian (not west) -->
<!-- west-poly_list.txt -->
<!--     - anyone self-reporting west polynesian with a PCA labelled west polynesian or general polynesian (not east) -->
<!--     - accidentally used the wrong column for this & excluded Pukapukan & Niuean samples due to their unique clustering -->
<!--         - this list has 772 people, the updated list has 939 (have not updated analysis yet) -->

<!-- Polynesian Gout - actual command run -->
<!-- plink2 \ -->
<!-- --bfile QC1_7-FullMerge1.0and1.1_ExDup_MattBrown_Czech-plus_rsIDconverted \ -->
<!--   --pheno polynesian_gout-affection.pheno \ -->
<!--   --logistic \ -->
<!--   --ci 0.95 \ -->
<!--   --covar QC1_7-FullMerge1.0and1.1_ExDup_MattBrown_Czech-plus_polygout.cov \ -->
<!--   --covar-name Sex,Age,PCA1,PCA2,PCA3,PCA4,PCA5,PCA6,PCA7,PCA8,PCA9,PCA10 \ -->
<!--   --hardy \ -->
<!--   --hwe 1e-6 \ -->
<!--   --freq case-control \ -->
<!--   --maf 0.01 \ -->
<!--   --geno 0.38 \ # to remove the extra czech sample only SNPs -->
<!--   --allow-no-sex \ -->
<!--   --out ../new_polyGWAS/polynesian_gout_CoreExome24v1.0-1_rsIDconverted -->

<!-- East Polynesian Gout - actual command run -->
<!-- plink2 \ -->
<!--   --bfile QC1_7-FullMerge1.0and1.1_ExDup_MattBrown_Czech-plus_rsIDconverted \ -->
<!--   --pheno polynesian_gout-affection.pheno \ -->
<!--   --keep east-poly_list.txt \ -->
<!--   --logistic \ -->
<!--   --ci 0.95 \ -->
<!--   --covar QC1_7-FullMerge1.0and1.1_ExDup_MattBrown_Czech-plus_polygout.cov \ -->
<!--   --covar-name Sex,Age,PCA1,PCA2,PCA3,PCA4,PCA5,PCA6,PCA7,PCA8,PCA9,PCA10 \ -->
<!--   --hardy \ -->
<!--   --hwe 1e-6 \ -->
<!--   --freq case-control \ -->
<!--   --maf 0.01 \ -->
<!--   --geno 0.38 \ # to remove the extra czech sample only SNPs -->
<!--   --allow-no-sex \ -->
<!--   --out ../new_polyGWAS/east-polynesian_gout_CoreExome24v1.0-1_rsIDconverted -->
<!-- West Polynesian Gout - actual command run -->
<!-- plink2 \ -->
<!--   --bfile QC1_7-FullMerge1.0and1.1_ExDup_MattBrown_Czech-plus_rsIDconverted \ -->
<!--   --pheno polynesian_gout-affection.pheno \ -->
<!--   --keep west-poly_list.txt \ -->
<!--   --logistic \ -->
<!--   --ci 0.95 \ -->
<!--   --covar QC1_7-FullMerge1.0and1.1_ExDup_MattBrown_Czech-plus_polygout.cov \ -->
<!--   --covar-name Sex,Age,PCA1,PCA2,PCA3,PCA4,PCA5,PCA6,PCA7,PCA8,PCA9,PCA10 \ -->
<!--   --hardy \ -->
<!--   --hwe 1e-6 \ -->
<!--   --freq case-control \ -->
<!--   --maf 0.01 \ -->
<!--   --geno 0.38 \ # to remove the extra czech sample only SNPs -->
<!--   --allow-no-sex \ -->
<!--   --out ../new_polyGWAS/west-polynesian_gout_CoreExome24v1.0-1_rsIDconverted -->
<!-- ## results files were moved to poly_gwas2017 folder at some point -->
<!-- ###### -->
<!-- Polynesian Urate GWAS: -->
<!-- Run in: -->
<!-- /Volumes/userdata/staff_users/tanyaflynn/ShortCuts/murray_working_dir/coreExome/Batch_Merge/Merged_Merriman_Brown_Matsuo/ -->
<!-- plink2 \ -->
<!--   --bed QC1_7-FullMerge1.0and1.1_ExDup_MattBrown_Czech-plus_correctAff.bed -->
<!--   --bim QC1_7-FullMerge1.0and1.1_ExDup_MattBrown_Czech-plus_correctAff.bim -->
<!--   --fam QC1_7-FullMerge1.0and1.1_ExDup_MattBrown_Czech-plus_polyurateAff.fam -->
<!--   --keep Polynesian_Controls.txt # identical set to controls used in gout regression -->
<!--   --linear -->
<!--   --ci 0.95 -->
<!--   --covar QC1_7-FullMerge1.0and1.1_ExDup_MattBrown_Czech-plus_polygout.cov -->
<!--   --covar-name Sex,Age,PCA1,PCA2,PCA3,PCA4,PCA5,PCA6,PCA7,PCA8,PCA9,PCA10 -->
<!--   --hardy -->
<!--   --hwe 1e-6 -->
<!--   --freq -->
<!--   --maf 0.01 -->
<!--   --geno 0.38 -->
<!--   --allow-no-sex -->
<!--   --out ../../poly_gwas2017/PolyUrate_CoreExome24v1.0-1_NZ-MB-HM_Gout -->