# Selection and Association Studies

```{r assocLoad, include=FALSE, warning = FALSE}
source('scripts/Annotate_genes.R')
library(GenomicRanges)
library(phdFunctions)
library(knitr)
library(kableExtra)
library(tidyverse)
source('scripts/05_ukbb_filteredSnps.R')
knitr::opts_chunk$set(echo = FALSE)
models <- readRDS('data/Ukbiobank/pheno_models.RDS')
pheno_models <- bind_rows(lapply(models, function(x){cbind(adjuster = x$adjuster,broom::glance(x$model))})) %>% arrange(AIC)


load('~/data/ukbiobank_gout_bd_refined2016-04-26.RData')
fam <- read.table('~/Git_repos/bookdown_thesis/data/Ukbiobank/chr1impv1.fam', stringsAsFactors = FALSE)

# f.31.0.0 = sex, 1 = male
# f.48.0.0 = waist
# f.50.0.0 = height
# f.21001.0.0 = BMI
# f.21000.0.0 = ethnicity
# f.20002.* =  self report illness

ukbio_gout <- bd_refined_gout %>% filter(f.eid %in% fam$V2) %>% select(goutaff, 'age' = f.21003.0.0,"bmi" = f.21001.0.0, 'sex' = f.31.0.0, 'height' = f.50.0.0, 'waist' = f.48.0.0, 'hip' = f.49.0.0, 'eth' = f.21000.0.0, febuxostat, allopurinol, colchicine, sulphinpyrazone, gouthosp, goutwinnard, goutself, gout_winnard_self, gout_self_ult, goutself, goutult ) %>% mutate(waist_height_ratio = waist / height, sex_factor = factor(if_else(sex == 0, "female", "male", missing = NULL)), cc = if_else(goutaff == 1, "case", "control", missing = NULL), waist_hip_ratio = waist / hip) %>% filter(!is.na(goutaff) & eth %in% c(1001,1002,1003)) %>% group_by(cc) 
```


This chapter is about using the results of selection analyses to better inform \gls{gwas}

The first section of this chapter has been published as part of @Cadzow2017

The UK Biobank provides a unique opportunity

objectives:
- Investigate the performance of gout definitions on association studies
- Investigate the use of association studies in conjunction with selection analyses



Selection, GWAS, and Imputation of Polynesian Populations


useful papers:

- @Fay2013
- @Dudley2012

prioritisation of variants from gwas
- @Hou2013
- maybe @Kindt2013

Improvement gwas through the use of imputation:

- Machini nature papers 2010?
- HRC panel paper

refer to gwas done by ukbio in all phenotypes and say it looks similar
## Association studies for Gout

### Performance of gout definitions

Part of this work has been published in @Cadzow2017.

This section describes the gout GWAS performed in the UKBiobank data and the comparision with the results from the selection analysis in the 1000 Genomes Project data.



_genome-wide association studies_

Genome-wide association studies provide a statistical framework for an association analysis between a phenotype of interest, and a genotypes of a genetic marker, usually using linear regression for a continuous trait, or logisitic regression for a dichotomous trait. The association test, in a \gls{gwas} context, is repeated for all available markers across the genome.

_Introduction of ukbiobank_

The UK Biobank is a publically available data set consisting of 500,000 individuals, mostly of European ancestry, from around the United Kingdom, aged between 40 and 69. It contains genetic, health, and lifestype information. As part of the UK Biobank collection process, participants filled in detailed surveys on health and lifestyle questions, provided biological samples, and provided access to administrative health records.


_Motivation behind using the different definitions_

In epidemiological studies the use of an accurate case definition is important. There is variation in the gout definitions used from the multi-purpose cohort studies that are used for genetic analysis, some well known cohorts include the Wellcome Trust case control constortium cohort, the Framingham Heart Study and the Artherosis risk in the community study [@Burton2007;@Aric1989; \@fhs]. 

The use of a consistent case definition is also important for comparisons between cohorts, this can be done by combining the genetic data and performing the \gls{gwas}. Another common way of combining genetic association studies is through meta-analysis. In both instances, having the same case definitions between studies reduces the variation. Power of genetic case-control studies increases with accurate case definition. This means that there are the maximum number of cases, and the maximum number of disease free controls [\@ref]. For gout, the gold standard case definition is the mono-soldium urate crystals in the synovial fluid of the joint [\@ref]. A sustitute for crystals in the synovial fluid is the \gls{acr} criteria [@Wallace1977]. The criteria require a minimum of six of twelve conditions to be met. It is uncommon for multipurpose cohorts to report either the gold-standard, or ACR criteria, but instead, for a self-report diagnosis of conditions and diseases to be reported. The UK Biobank includes the self-report diagnoses that were collected by survey at the time of recruitment, but it also has hospital admission information, for both primary and secondary diagnosis.  





### Polynesian associations with gout

Phipps-Green2016

### Use of selection statistics to inform GWAS

## Methods

### European Gout GWAS {#ukbbgwas}

<!-- acr report using unfiltered controls -->
This \gls{gwas} was performed using the interim data release from the UK Biobank (approval number 12611) downloaded November 2015. There was genetic information for `r NROW(fam)` individuals. For the \gls{gwas} itself, it was limited to be only people who had self reported an ethnic background of British, Irish, or 'any other white background'.

Inclusion criteria were European ethnicity, aged between 40 and 69 years and had genome wide genotypes available. The exclusion criteria used was: mismatch between self-reported sex and genetic sex, genotype quality control failure, relatedness, or either a primary or secondary hospital diagnosis of kidney disease (International Classification of Diseases, Tenth Revision (ICD-10), codes I12, I13, N00-N05, N07, N11, N14, N17â€“N19, Q61, N25.0, Z49, Z94.0, Z99.2), participants aged 70 years and over, and those with kidney disease, because these are risk factors for secondary gout.

Inidivduals from the UK Biobank cohort were genotyped on an Axiom array with 820,967 markers (Affymetrix, Santa Clara, CA, USA). The genotypes were phased using SHAPEIT3, and then imputed using IMPUTE2 with both the UK10K impute reference panel, and the haplotype reference consortium impute panel, to bring the total number of \glspl{snp} to approximately 73.3 milion. Genotyping, phasing, and imputation were performed by the UK Biobank.

#### Classifications used

There were four main classification criteria that were used to define gout cases. _Hospital defined gout_ involved either a primary or secondary hospital diagnosis for gout (ICD-10 code M10, including sub-codes). _Self-report of gout_ was defined as reporting having gout at the time of the study interview. _Use of \gls{ult}_ used self-report data for the use of allopurinol, febuxostat, or sulphinpyrazone, and not having a hospital diagnosis of leukemia or lymphoma (ICD-10 codes C81-C96). _Winnard defined gout_ was based on @Winnard2012 and consisted of a hospital diagnosis of gout, or self report of gout specific medication (\gls{ult} or colchicine). The largest number of people were classified through either self-report or the use of urate lowering therapy (Figure \@ref(fig:ukbbupsetplot)). Additional exclusion criteria were applied to participants not classified as having gout from these definitions. The criteria were cortico-steroidal use, non-steroidal anti-inflammatory drug use or probenicid use. 

Confidence intervals for proportions were calculated using the \texttt{prop.test} function from the _stats_ package in R.


```{r, warning = FALSE, eval = FALSE}

bind_rows(lapply(models, function(x){cbind(adjuster = x$adjuster,broom::glance(x$model))})) %>% arrange(AIC)
```

```{r, results = 'asis', eval = FALSE}

broom::tidy(models[[5]]$model) %>% mutate(p.value = format(p.value, scientific = TRUE, digits = 3))%>% kable(., digits = 3, caption = '(\\#tab:phenomodel) Logistic regression for gout adjusted by age, sex, and BMI')
```


```{r, eval = FALSE}
lapply(c(1,2,4,3,5 ), function(x){broom::tidy(models[[x]]$model)})
```

```{r ukbbNumbers}
ukbb_n <- list()
ukbb_n$n_controls <- ukbio_gout %>% filter(goutaff ==0) %>% NROW()
ukbb_n$n_gout <- ukbio_gout %>% filter(goutaff ==1) %>% NROW()
ukbb_n$n_total <- ukbb_n$n_controls + ukbb_n$n_gout
ukbb_n$n_self <- ukbio_gout %>% filter(goutaff ==1 & goutself ==1) %>% NROW()
ukbb_n$n_selfult <- ukbio_gout %>% filter(goutaff ==1 & (goutself ==1 | goutult == 1)) %>% NROW()
ukbb_n$n_ult <- ukbio_gout %>% filter(goutaff ==1 & goutult ==1) %>% NROW()
ukbb_n$n_winnard <- ukbio_gout %>% filter(goutaff ==1 & goutwinnard == 1) %>% NROW()
ukbb_n$n_hosp <- ukbio_gout %>% filter(goutaff ==1 & gouthosp == 1) %>% NROW()

ukbb_male_n <- list()
ukbb_male_n$n_controls <- ukbio_gout %>% filter(goutaff ==0 & sex == 1) %>% NROW()
ukbb_male_n$n_gout <- ukbio_gout %>% filter(goutaff ==1 & sex == 1) %>% NROW()
ukbb_male_n$n_total <- ukbb_male_n$n_controls + ukbb_male_n$n_gout
ukbb_male_n$n_self <- ukbio_gout %>% filter(goutaff ==1 & goutself ==1 & sex == 1) %>% NROW()
ukbb_male_n$n_selfult <- ukbio_gout %>% filter(goutaff ==1 & (goutself ==1 | goutult == 1) & sex == 1) %>% NROW()
ukbb_male_n$n_ult <- ukbio_gout %>% filter(goutaff ==1 & goutult ==1 & sex == 1) %>% NROW()
ukbb_male_n$n_winnard <- ukbio_gout %>% filter(goutaff ==1 & goutwinnard == 1 & sex == 1) %>% NROW()
ukbb_male_n$n_hosp <- ukbio_gout %>% filter(goutaff ==1 & gouthosp == 1 & sex == 1) %>% NROW()

ukbb_female_n <- list()
ukbb_female_n$n_controls <- ukbio_gout %>% filter(goutaff ==0 & sex == 0) %>% NROW()
ukbb_female_n$n_gout <- ukbio_gout %>% filter(goutaff ==1 & sex == 0) %>% NROW()
ukbb_female_n$n_total <- ukbb_female_n$n_controls + ukbb_female_n$n_gout
ukbb_female_n$n_self <- ukbio_gout %>% filter(goutaff ==1 & goutself ==1 & sex == 0) %>% NROW()
ukbb_female_n$n_selfult <- ukbio_gout %>% filter(goutaff ==1 & (goutself ==1 | goutult == 1) & sex == 0) %>% NROW()
ukbb_female_n$n_ult <- ukbio_gout %>% filter(goutaff ==1 & goutult ==1 & sex == 0) %>% NROW()
ukbb_female_n$n_winnard <- ukbio_gout %>% filter(goutaff ==1 & goutwinnard == 1 & sex == 0) %>% NROW()
ukbb_female_n$n_hosp <- ukbio_gout %>% filter(goutaff ==1 & gouthosp == 1 & sex == 0) %>% NROW()

```



#### Gout association test

A \gls{gwas} was performed using logisitic regression in Plink1.9b [@plink2; @Chang2015] for each gout classification, with age, sex, and \gls{bmi} as co-variates. The association results for each \gls{gwas} were then compared with the 30 \glspl{snp} reported in Table 1 of @Kottgen2013 that had previously been associated with urate and gout in a cohort of >140,000 Europeans. 


#### Heriabililty

For each gout classification, the genetic component of heritability explained was calculated. This was done by first randomly selected 10,000 controls and then combining this same subset with cases from each of the gout classifications. The genetic variance was calculated for each chromosome separately and then combined. The dichotomous case-control phenotype was transformed onto the continuous liability scale using the restricted maximum likelihood analysis using GCTA (v1.26.0) and a general population prevalence of gout of 2% (section \@ref(gctaHeritability)). Heritabilty estimates were compared using the formula h~1~ - h~2~ and the SE calculated using formula \@ref(eq:h2se).

> need to explain the continuity scale and the reml approach

\begin{equation} 
 SE = \sqrt{SE_1^2 + SE_2^2}
 (\#eq:h2se)
\end{equation} 

### Polynesian Gout GWAS

#### Polynesian gout gwas dataset

A multi-cohort gout dataset was assembled by combining individuals from the Genetics of Gout, Diabetes, and Kidney Disease in Aotearoa cohorts and the Ngati Porou Hauora Charitable Trust cohort. Genotyping on the Inifinium CoreExome v24 bead-chip was performed at the University of Queensland (Centre for Clinical Genomics) for the Genetics of Gout, Diabetes, and Kidney Disease in Aotearoa cohorts and at AgResearch (Invermay Agricultural Centre) for the Ngati Porou Hauora Charitable Trust cohort. The genotype quality control procedures as described in section \@ref(coreExomeQC) were applied per cohort, and then as necessary after combining the cohorts. The genotype quality control, and combining of cohorts was performed by Dr Tanya Major, Merriman Lab. \Glsdesc{pca} was performed on the combined dataset with the first 10 components outputted and 



#### Polynesian gout GWAS

A \gls{gwas} was also performed for individuals with Polynesian ancestry. Gout affection was determined by the American college of rheumatology criteria [\@ref], or a doctor diagnosis, or enrolled in a gout drug trial. All participants were older than 18, and provided informed written consent. Exclusion criteria included unknown gout affection status, non-Polynesian self-reported ancestry, or a mismatch between self-report ancestry and \gls{pca} clustering.

<!-- polynesian_gout-affection.pheno -->
<!--     - gout either ACR criteria or GP diagnosed / in a gout drug trial (only 119 not ACR) -->
<!--     - excluded unknown gout status, non-polynesian (either self-report or PCA mismatch), and people <18 years old -->


The \gls{gwas} was adjusted for age, sex, and the first ten \glspl{pc} calculated from 2,858 ancestry informative markers (as identified by Illumina). Markers were filtered to remove if the minor allele frequency was less than 0.01, or had an \gls{hwe} chi-squared exact test P > 1x10^-6^. The \gls{gwas} was conducted by Dr Tanya Major(Merriman Lab).




## Results

### UK Biobank

#### Particpant and association model characteristics

There was genome-wide genotype data for 105,421 participants, after applying the inclusion and exclusion criteria. The mean \gls{bmi} for all participants was `r ukbio_gout %>%ungroup() %>%  summarise(bmi = mean(bmi, na.rm = TRUE)) %>% pull(bmi) %>% round(., 2)` kg/m^2^, the mean age was `r ukbio_gout %>%ungroup() %>% summarise(age = mean(age, na.rm = TRUE)) %>% pull(age) %>% round(., 2)` years, and `r ukbio_gout %>%ungroup() %>% summarise(percent_male = (1 - sum(sex)/NROW(.)) *100) %>% pull(percent_male) %>% round(., 2)`% were male. A breakdown of by affection status for anthropomorphic traits, drug use, and gout classifications is in Table \@ref(tab:ukbbclin).

Prior to performing the \glspl{gwas} using the different classifications for gout, the best model (determined by \gls{aic}) for phenotypic co-variate adjustment was determined by logistic regression. Known co-variates for gout included age, sex, and \gls{bmi}. Each of these co-variates were tested individually and together. Each covariate had a significant of association on their own: age (P = \num{`r broom::tidy(models[[1]]$model) %>% filter(term == 'age') %>% .[['p.value']] %>% format(., scientific = TRUE, digits = 3)`}), sex (P = \num{`r broom::tidy(models[[2]]$model) %>% filter(term == 'sex_factormale') %>% .[['p.value']] %>% format(., scientific = TRUE, digits = 3)`}), and \gls{bmi} (P = \num{`r broom::tidy(models[[4]]$model) %>% filter(term == 'bmi') %>% .[['p.value']] %>% format(., scientific = TRUE, digits = 3)`}). In the combined model, with all three covariates there was an \gls{aic} for the model of 19558.30 compared to models with each with age (\gls{aic} 22706.05), sex (\gls{aic} 21041.80) and \gls{bmi} (\gls{aic} 22001.03). Based on the \gls{aic} for the combined model, age, sex, and \gls{bmi} were used as the covariates in the \gls{gwas} for gout. 

```{r, results = 'asis'}
  ukbio_gout %>% summarise(n = n(), `Male (\\%)` = round(1 - sum(sex)/NROW(.),2), `Mean Age (SD)` = paste0(round(mean(age, na.rm = TRUE),2), " (",round(sd(age, na.rm = TRUE),2), ")" )  , `Mean BMI (SD)` = paste0(round(mean(bmi, na.rm = TRUE),2), " (", round(sd(bmi, na.rm = TRUE),2),")"),  Febuxostat = sum(febuxostat, na.rm = TRUE), Allopurinol = sum(allopurinol, na.rm = TRUE), Sulphinpyrazone = sum(sulphinpyrazone, na.rm = TRUE), Colchicine = sum(colchicine, na.rm = TRUE), Hospital = sum(gouthosp, na.rm =TRUE), `Self-report` = sum(goutself, na.rm=TRUE), Winnard = sum(goutwinnard, na.rm = TRUE), ULT = sum(goutult, na.rm=TRUE) ) %>% t() %>% data.frame() %>% mutate(var = rownames(.)) %>% dplyr::slice(., -1) %>% select(var, 'Case' = 1, 'Control' = 2) %>% kable(., booktabs = TRUE, caption = "(\\#tab:ukbbclin) Clinical details of participants.", format = "latex", escape = FALSE, col.names = c("", "Case", "Control")) %>% group_rows(., "Gout medication", 5, 8) %>% group_rows(., "Gout classification", 9, 12)
```

#### Performance of classification

There were a total of `r ukbb_n$n_gout` individuals that had any classification of gout and `r ukbb_n$n_controls` controls. Of those with any classification of gout, there were `r ukbb_n$n_hosp` with a hospital diagnosis, `r ukbb_n$n_ult` with urate lowering therapy, `r ukbb_n$n_winnard` with the Winnard criteria, and `r ukbb_n$n_self` with self-reported gout. The overlaps in classification between each group can be seen in Figure \@ref(fig:ukbbupsetplot). There was considerable overlap between the self-report, urate lowering therapy, and Winnard classifications with 1242 individuals meeting all three classifications. There were 571 (27.6%) of 2066 participants who could only be defined through self-report. And for hospital diagnosis, 126 (33.0%) of 382, did not meet the self-report of gout, or \gls{ult} use definition criteria.


The prevalence of gout from all definitions was `r round(ukbb_n$n_gout/ukbb_n$n_control *100,1)`%, this ranged from the lowest prevalence of gout of 0.36%, from the hospital diagnosis defintion, through to the highest of 2.18%, by self-report of gout or \gls{ult} usage in the study population (Table \@ref(tab:ukbbprev)).

(ref:ukbbupsetplot) An upset plot showing the total sizes (left) and intersections (upper) of the different classification criteria.

```{r ukbbupsetplot, fig.width=6.5, fig.cap='(ref:ukbbupsetplot)', fig.asp=1/1.5 }

listInput <- readRDS('data/Ukbiobank/upsetPlotList.RDS')
UpSetR::upset(UpSetR::fromList(listInput), order.by = "freq", text.scale = 1.3 , nintersects = NA)
```


```{r, message = FALSE}
entire <- purrr::map_df(names(ukbb_n)[!names(ukbb_n)  %in% c("n_controls", 'n_total', "n_gout")], ~ c(broom::tidy(prop.test(x = ukbb_n[[.x]], n = ukbb_n$n_total )), def = .x, n =ukbb_n[[.x]] )) %>% select(def, n, estimate, conf.low, conf.high) %>% mutate_at(.vars = c("estimate", "conf.low", "conf.high"), .funs = function(x){round(x * 100, 2)}) %>% mutate(study_entire = paste0(n, ", ", estimate,"\\% (",conf.low, "-", conf.high, ")")) %>% select(def,study_entire)

male <- purrr::map_df(names(ukbb_n)[!names(ukbb_n)  %in% c("n_controls", 'n_total', "n_gout")], ~ c(broom::tidy(prop.test(x = ukbb_n[[.x]], n = ukbb_n$n_total )), def = .x, n =ukbb_n[[.x]] )) %>% select(def, n, estimate, conf.low, conf.high) %>% mutate_at(.vars = c("estimate", "conf.low", "conf.high"), .funs = function(x){round(x * 100, 2)}) %>% mutate(study_male = paste0(n, ", ", estimate,"\\% (",conf.low, "-", conf.high, ")")) %>% select(def, study_male)

female <- purrr::map_df(names(ukbb_n)[!names(ukbb_n)  %in% c("n_controls", 'n_total', "n_gout")], ~ c(broom::tidy(prop.test(x = ukbb_n[[.x]], n = ukbb_n$n_total )), def = .x, n =ukbb_n[[.x]] )) %>% select(def, n, estimate, conf.low, conf.high) %>% mutate_at(.vars = c("estimate", "conf.low", "conf.high"), .funs = function(x){round(x * 100, 2)}) %>% mutate(study_female = paste0(n, ", ", estimate,"\\% (",conf.low, "-", conf.high, ")")) %>% select(def, study_female)

all_gout <- purrr::map_df(names(ukbb_n)[!names(ukbb_n)  %in% c("n_controls", 'n_total', "n_gout")], ~ c(broom::tidy(prop.test(x = ukbb_n[[.x]], n = ukbb_n$n_gout )), def = .x, n =ukbb_n[[.x]] )) %>% select(def, n, estimate, conf.low, conf.high) %>% mutate_at(.vars = c("estimate", "conf.low", "conf.high"), .funs = function(x){round(x * 100, 2)}) %>% mutate(study_gout = paste0(estimate,"\\% (",conf.low, "-", conf.high, ")")) %>% select(def, study_gout)

full_join(entire, male) %>% full_join(., female) %>% full_join(., all_gout) %>% 
  mutate(def = case_when(def == "n_self" ~ "Self-report of gout diagnosis",
                         def == "n_ult" ~ "ULT use",
                         def == "n_hosp" ~ "Hospital diagnosis",
                         def == "n_winnard" ~ "Winnard definition",
                         def == "n_selfult" ~ "Self-report of gout or ULT use"
                         )) %>% 
 kable(caption = '(\\#tab:ukbbprev) Number, prevalence (95\\% CI) of participants defined as gout cases', col.names = c("Definition", "No. of subjects, prevalence (95\\% CI) in entire study population (n = 105,421)", "No. of subjects, prevalence (95\\% CI) in male participants (n = 51,844) ","No. of subjects, prevalence (95\\% CI) in female participants (n = 53,577)", "Percentage (95\\% CI) of study population with gout using any definition (n = 2432)"),format = 'latex', booktabs = TRUE, escape = FALSE) %>% column_spec(., column = c(2:5), width = '3.5cm')%>% landscape() 

```

#### Replication of @Kottgen2013 urate associated loci

```{r, message = FALSE}
kot_t1 <- read.csv("~/data/kottgen_table1.csv", header = TRUE, stringsAsFactors = FALSE)
uk_kot_t1 <-read_delim(file = 'data/Ukbiobank/kottgen_t1_age_sex_bmi_results.txt', delim = '\t', col_names = c("file","SNP","POS","A1","TEST","NMISS","OR", "SE", "L95", "U95", "STAT", "P")) %>% 
  separate(file, c("file","CHR"), sep = ':') %>% 
  mutate(file = stringi::stri_replace(file, "selfult", regex ="self_ult" )) %>% 
  mutate(gout = map_chr(stringi::stri_split(file, regex = "_"), 1) %>% 
           stringi::stri_sub(9)) %>% 
  left_join(., kot_t1 %>% select(SNP, grail_gene), by = "SNP")

# create the table of kottgen snps from the data
# age, sex, and bmi adjusted
kot_bim <- read_delim('data/Ukbiobank/kottgen_ukbb_alleles.txt', col_names = c("CHR","SNP","GPOS","POS","A1","A2"), delim = " ")

#calculate number of snps that are significant
uk_kot_psig <- uk_kot_t1 %>% select(-file, -TEST, -SE, -NMISS,-STAT, -POS) %>% mutate(P_bac = P) %>% mutate(P = purrr::map_chr(P, ~ format_p_md(p = .x, format = 'latex'))) %>% arrange(P) %>% select(-grail_gene, -OR, -CHR) %>%group_by(gout) %>% mutate(P_sig = P_bac < (0.05/NROW(.)), P_gw = P_bac < 5e-8 ) %>% summarise( P_gw = sum(P_gw),P_exp = sum(P_sig)) %>% filter(gout != 'all')%>% gather("sig", "value",2:3) %>% spread(gout, value) %>% rename("SNP" = sig) %>% mutate_all(as.character) %>% mutate(A1 = "", REF ="",ALT ="")
```

Analysis of the 30 reported urate associated \glspl{snp} from @Kottgen2013 revealed similar odds ratios for all gout definitions (Figure \@ref(fig:kotplot), Table \@ref(tab:ukbbkot)). Experiment wide significance was defined as P < `r as.character(signif(0.005/30,2))`, and genome-wide significance as P < 5x10^-8^. There were differing numbers of \glspl{snp} that had a sigificant association with gout, both at genome and the experiment-wide thresholds between the gout definitions used. Meeting genome-wide significance, there were five \glspl{snp} (_ABCG2_, _GCKR_, _SLC17A3_, _SLC22A12_, and _SLC2A9_) for both the self-reported gout or \gls{ult} usage definition and the self-reported gout defintion, four \glspl{snp} (_ABCG2_, _GCKR_, _SLC22A12_, and _SLC2A9_) for the \gls{ult} usage definition, three \glspl{snp} (_ABCG2_, _GCKR_, and _SLC2A9_) for the Winnard definition, and two \glspl{snp} (_ABCG2_ and _SLC2A9_) for the hospital diagnosis definition. For all definitions, the effect size and strength of association was larger for _ABCG2_ than _SLC2A9_. At the experiment-wide significance threshold, there were 13 \glspl{snp} for the self-reported gout or \gls{ult} use definition, 12 \glspl{snp} for the self-reported gout definition, 11 for \gls{ult} use definition, 10 for the Winnard definition, and three \glspl{snp} for the hospital diagnosis definition.


```{r, nkotsig}
n_kot_sig <- uk_kot_t1 %>% group_by(gout) %>% summarise(gw = sum(P < 5e-8), ew = sum(P < 0.05/30))
```



```{r, message=FALSE}
# still need to flip snps to be in the same order as kottgen
uk_kot_t1 %>% 
  select(-file, -TEST, -SE, -NMISS,-STAT, -POS) %>%  
  left_join(., kot_bim %>% select(SNP, A1, A2), by = c("SNP", "A1")) %>% 
  #mutate(OR = ifelse(A1 != REF, 1/OR, OR), L95 = ifelse(A1 != REF, 1/U95, L95), U95 = ifelse(A1 != REF, 1/L95, U95), mod = ifelse(A1 != REF, 1, 0)) %>% 
  mutate_at(., vars("OR",contains('95')), funs(sprintf("%.2f",. ))) %>% 
  unite("CI",c("L95","U95"), sep = " -") %>%
  mutate(P = purrr::map_chr(P, ~ format_p_md(p = .x, format = 'latex'))) %>%  
  mutate(Effect = paste0(OR, ' [', CI,'], ', P)) %>% arrange(P) %>%  select(-OR, -CI, -P, -CHR) %>%
  spread(key = gout, value = Effect) %>% 
  select(grail_gene, everything(), -all) %>%
  arrange(grail_gene) %>% 
  #bind_rows(., uk_kot_psig ) %>% 
  kable(., format = 'latex', caption = '(\\#tab:ukbbkot) The odds ratios for the 30 variants reported in \\citet{Kottgen2013} for the different gout classification methods.', col.names = c("Grail gene","Marker", "A1", "A2", "Hospital","Self-report","Self-report or ULT","ULT","Winnard"), booktabs = TRUE, escape = FALSE) %>% #group_rows("Number of significant markers", NROW(.)-1, NROW(.)) %>% 
  kable_styling(latex_options = 'scale_down') %>% footnote(general = "OR [95\\\\% CI], P. ULT = Urate lowering therapy. The reported effect allele is A1. Effects are adjusted by age, sex, and BMI." , general_title = "", threeparttable = TRUE , escape = FALSE) %>% landscape()

```

(ref:kotplot) Plot showing ORs (95% CI) for the 30 \glspl{snp} reported in @Kottgen2013 based on gout definitions. Original Kottgen reported ORs are shown as red circles. The reported Kottgen OR was inverted to maintain consistency of reported effect allele (A1) for the following markers: rs2307394, rs2079742, rs7224610, rs7188445, rs1178977, rs6770152, rs7193778, rs164009, rs653178, rs2078267, rs478607, and rs12498742.

```{r kotplot, fig.height=7, fig.cap = '(ref:kotplot)', warning = FALSE}

flip = c("rs2307394", "rs2079742", "rs7224610", "rs7188445","rs1178977", "rs6770152", "rs7193778", "rs164009", "rs653178","rs2078267", "rs478607", "rs12498742")
uk_kot_t1 %>% 
  #mutate(OR = ifelse(SNP %in% flip, 1/OR, OR), L95 = ifelse(SNP %in% flip, 1/L95, L95), U95 = ifelse(SNP %in% flip, 1/U95, U95))  %>%  
  select(gout, SNP, grail_gene,OR, L95, U95) %>% 
  bind_rows(., kot_t1 %>% mutate(gout = "Kottgen") %>% select(gout, SNP, grail_gene, OR) %>% mutate(OR = ifelse(SNP %in% flip, 1/OR, OR) )) %>% unite("SNP", c("grail_gene", "SNP"), sep = '\n') %>% mutate(highlight = ifelse(gout == "Kottgen", "b","a")) %>% filter(gout != 'all') %>% mutate(gout = factor(gout, ordered = TRUE)) %>% mutate(gout = forcats::fct_relevel(gout, "Kottgen", "self","ult","hosp","winnard","selfult"))%>% mutate(gout = forcats::fct_recode(gout, "Hospital" = "hosp", "Self Report" = 'self', "Self Report or ULT" = 'selfult', "ULT" = "ult", "Winnard" = "winnard" )) %>% 
              # mutate(gout = case_when(
              #   gout == 'hosp' ~ "Hospital", 
              #   gout == "self" ~ "Self Report", 
              #   gout == "selfult" ~ "Self Report or ULT",
              #   gout == "ult" ~ "ULT",
              #   gout == "winnard" ~ "Winnard")) %>% 
              ggplot(data = ., aes(x = gout, y = OR, shape = highlight, colour = highlight)) + 
              geom_point() + facet_wrap(~SNP) + geom_linerange(aes(x = gout, ymin = L95, ymax = U95)) +geom_hline(yintercept = 1) +theme_bw() + theme(axis.text.x = element_text(angle=-90, vjust = 0.5, hjust = 0)) + theme(axis.title = element_blank(), legend.position = 'none') + scale_color_manual(values = c("black","red")) 
```





#### GWAS results

The top SNPs from the GWAS were:

number of significant snps

- hosp `r hosp_asb %>% filter(P < 1e-8, nchar(A1) ==1 ) %>% NROW()`
- self `r self_asb %>% filter(P < 1e-8, nchar(A1) ==1 ) %>% NROW()`
- ult `r ult_asb %>% filter(P < 1e-8, nchar(A1) ==1 ) %>% NROW()`
- self ult `r selfult_asb %>% filter(P < 1e-8, nchar(A1) ==1 ) %>% NROW()`
- winnard `r winnard_asb %>% filter(P < 1e-8, nchar(A1) ==1 ) %>% NROW()`

(ref:Overlap) Overlap of markers that reached nominal genome-wide significance between the different gout classifications.

```{r ukOverlap, fig.cap = '(ref:ukOverlap)'}
listInput <- list(hosp = hosp_asb %>% filter(P < 1e-5, nchar(A1) ==1 ) %>% pull(SNP),
                  self = self_asb %>% filter(P < 1e-5, nchar(A1) ==1 ) %>% pull(SNP),
                  ult = ult_asb %>% filter(P < 1e-5, nchar(A1) ==1 ) %>% pull(SNP),
                  selfult = selfult_asb %>% filter(P < 1e-5, nchar(A1) ==1 ) %>% pull(SNP),
                  winnard = winnard_asb %>% filter(P < 1e-5, nchar(A1) ==1 ) %>% pull(SNP)
                  )

UpSetR::upset(UpSetR::fromList(listInput), order.by = "freq", text.scale = 1.3, nintersects = NA )
```









- hosp

```{r}
hosp_asb %>% filter(P < 1e-8, nchar(A1) ==1 ) %>% mutate(P = format(P, scientific = TRUE)) %>% head() %>% kable()
```


```{r, message = FALSE, warning = FALSE}
dat <- hosp_asb 
txdb_gene_annotate(dat %>% filter(P < 1e-8, nchar(A1) ==1 ) %>% mutate(chrom = paste0("chr",CHR), chrom_start = BP, chrom_end = BP +1, marker = SNP) %>% select(contains('chrom'), marker) %>% GRanges()) %>% left_join(., dat %>% mutate(start = BP, seqnames = paste0("chr",CHR)), by = c('seqnames','start')) %>% select(CHR, SNP, BP, SYMBOL, A1, OR, SE, L95, U95, P ) %>% arrange(P) %>% mutate(SYMBOL = ifelse(is.na(SYMBOL), paste0("NA_",CHR,"_",BP) , SYMBOL)) %>% group_by(CHR, SYMBOL ) %>% filter(min(P) == P) %>% arrange(CHR, BP) %>% mutate(P = format(P, scientific = TRUE)) %>% kable()
```


- self

There were ```r self_asb %>% filter(P < 1e-8, nchar(A1) ==1) %>% mutate(P = format(P, scientific = TRUE)) %>% tally()``` \glspl{snp} that met the genome-wide significance threshold. 

```{r}
self_asb %>% filter(P < 1e-8, nchar(A1) ==1) %>% mutate(P = format(P, scientific = TRUE)) %>% dim()# kable()
```

```{r, message = FALSE, warning=FALSE}
dat <- self_asb 
txdb_gene_annotate(dat %>% filter(P < 1e-8, nchar(A1) ==1 ) %>% mutate(chrom = paste0("chr",CHR), chrom_start = BP, chrom_end = BP +1, marker = SNP) %>% select(contains('chrom'), marker) %>% GRanges()) %>% left_join(., dat %>% mutate(start = BP, seqnames = paste0("chr",CHR)), by = c('seqnames','start')) %>% select(CHR, SNP, BP, SYMBOL, A1, OR, SE, L95, U95, P ) %>% arrange(P) %>% mutate(SYMBOL = ifelse(is.na(SYMBOL), paste0("NA_",CHR,"_",BP) , SYMBOL)) %>% group_by(CHR, SYMBOL ) %>% filter(min(P) == P) %>% arrange(CHR, BP) %>% mutate(P = format(P, scientific = TRUE)) %>% head() %>% kable()
```



- self + ULT

```{r, message = FALSE, warning=FALSE}
dat <- selfult_asb 
txdb_gene_annotate(dat %>% filter(P < 1e-8, nchar(A1) ==1 ) %>% mutate(chrom = paste0("chr",CHR), chrom_start = BP, chrom_end = BP +1, marker = SNP) %>% select(contains('chrom'), marker) %>% GRanges()) %>% left_join(., dat %>% mutate(start = BP, seqnames = paste0("chr",CHR)), by = c('seqnames','start')) %>% select(CHR, SNP, BP, SYMBOL, A1, OR, SE, L95, U95, P ) %>% arrange(P) %>% mutate(SYMBOL = ifelse(is.na(SYMBOL), paste0("NA_",CHR,"_",BP) , SYMBOL)) %>% group_by(CHR, SYMBOL ) %>% filter(min(P) == P) %>% arrange(CHR, BP) %>% mutate(P = format(P, scientific = TRUE)) %>% head() %>% kable()
```


- ULT

```{r, message = FALSE, warning=FALSE}
dat <- ult_asb 
txdb_gene_annotate(dat %>% filter(P < 1e-8, nchar(A1) ==1 ) %>% mutate(chrom = paste0("chr",CHR), chrom_start = BP, chrom_end = BP +1, marker = SNP) %>% select(contains('chrom'), marker) %>% GRanges()) %>% left_join(., dat %>% mutate(start = BP, seqnames = paste0("chr",CHR)), by = c('seqnames','start')) %>% select(CHR, SNP, BP, SYMBOL, A1, OR, SE, L95, U95, P ) %>% arrange(P) %>% mutate(SYMBOL = ifelse(is.na(SYMBOL), paste0("NA_",CHR,"_",BP) , SYMBOL)) %>% group_by(CHR, SYMBOL ) %>% filter(min(P) == P) %>% arrange(CHR, BP) %>% mutate(P = format(P, scientific = TRUE)) %>%head() %>% kable()
```


- all


```{r, message = FALSE, warning = FALSE}
dat <- all_asb 
txdb_gene_annotate(dat %>% filter(P < 1e-8, nchar(A1) ==1 ) %>% mutate(chrom = paste0("chr",CHR), chrom_start = BP, chrom_end = BP +1, marker = SNP) %>% select(contains('chrom'), marker) %>% GRanges()) %>% left_join(., dat %>% mutate(start = BP, seqnames = paste0("chr",CHR)), by = c('seqnames','start')) %>% select(CHR, SNP, BP, SYMBOL, A1, OR, SE, L95, U95, P ) %>% arrange(P) %>% mutate(SYMBOL = ifelse(is.na(SYMBOL), paste0("NA_",CHR,"_",BP) , SYMBOL)) %>% group_by(CHR, SYMBOL ) %>% filter(min(P) == P) %>% arrange(CHR, BP) %>% mutate(P = format(P, scientific = TRUE)) %>% head() %>% kable()
```


manhattan plot for the all \@ref(fig:manhatAllAgeSex)

(ref:manhatAllAgeSex) _Manhattan plot for Gout GWAS adjusted for age and sex_

```{r manhatAllAgeSex, echo = FALSE, fig.cap='(ref:manhatAllAgeSex)', fig.width=7, fig.asp=1, out.width= '95%', fig.align='center'}
knitr::include_graphics('images/05_selection_and_association/all_age_sex.png')
```

```{r manhatAllAgeSexBmi,echo = FALSE, fig.cap='(ref:manhatAllAgeSexBmi)', fig.width=10, fig.asp=1, out.width = '95%', fig.align='center', out.extra='angle=90'}
knitr::include_graphics('images/05_selection_and_association/all_age_sex_bmi.png')
```

(ref:manhatAllAgeSexBmi) _Manhattan plot for Gout GWAS adjusted for age, sex and BMI_

```{r, engine='block', eval = FALSE, echo = FALSE, include=FALSE }
For this section I want to compare the results for each of the gwases and see if the same 
snps come out and perhaps in the same order.

Also want to compare the results from the age+sex adjusted versus the results from the age+sex+waist+waist:height
```

<!---
ALL GROUP: the most significant snps are from ABCG2. SLC2A9 also features highly. The majority of the significant snps are from these two genes.
--->

#### Heritability Explained

```{r, engine='block', eval = FALSE, echo = FALSE, include=FALSE }
here I want to include the results from the heritability analysis across the different criteria
```

To see if there was a difference between gout classifications in the genetic component of the variance explained by an additive model in gout, heritability estimates were calculated for each gout classification. To do this, 10000 controls were randomly selected and were used with each different gout criteria and heritability estimates for each chromosome were calculated using GCTA 1.26.0 [@Yang2011; @Yang2011part]. The heritability estimates were 0.289 (SE 0.034) for the self-report of gout or ULT use definition, 0.283 (SE 0.036) for the self-report of gout definition, 0.282 (SE 0.040) for the Winnard definition, 0.308 (SE 0.044) for the ULT use definition and 0.236 (SE 0.160) for the hospital diagnosis definition. There were no statistically significant differences between the heritability estimates of any of the different classifications. The heritability estimates were similar to estimates of genetic variance of serum urate [@Kottgen2013]




<!-- Kottgen refs 10-12 put heritability explained for serum urate at 40-70%. Kottgen reported heritability explained in serum urate loci that were adjusted for age and sex from 0.27 to 0.41. -->



### European selection 

```{r}
# ihs and nsl sig markers
sig_ihs_val <- readRDS('~/data/NZ_coreExome_1kgp/haplotype/sig_ihs_values_with_genes_14-11-2017.RDS') %>% filter(abs(statvalue) > 2 )
sig_nsl_val <- readRDS('~/data/NZ_coreExome_1kgp/haplotype/sig_nsl_values_with_genes_14-11-2017.RDS') %>% filter(abs(statvalue) > 2 )

```


In order to investigate the relationship between the haplotypic selection results and the gout \gls{gwas} using the UK Biobank, the significant results for \gls{ihs} and \gls{nsl} for \gls{gbr} from the \gls{1kgp} were compared to those for the \gls{gwas}. Given that the classification of self-reported gout or on \gls{ult} had the best performance, it was selected as the gout classification method to compare to.

There were no \gls{ihs} \glspl{snp} that also had P < 5x10^-5^ in the \gls{gwas}. If the significance threshold for \gls{ihs} was reduced to |\gls{ihs}| > 2 (approximately the 5% extreme most values), then there were three \glspl{snp}, all located at _SLC22A12_ or within 1kb, that were genome-wide significant in the \gls{gwas} (rs505802, rs9734313, and rs559946), and four \glspl{snp} that were nominally significant.

There was a single marker - rs7185008 - that had both a significant \gls{nsl} score, and had a nominally significant \gls{gwas} P-value (P = 1.8x10^-5^, adjusted for age, sex, and \gls{bmi}). 



```{r}
sig_ihs_val %>% filter(pop == 'GBR') %>% inner_join(., selfult_asb, by = c("chrom" = "CHR", "chrom_start" = "BP"))
```


```{r}
sig_nsl_val %>% filter(pop == 'GBR') %>% inner_join(., all_asb, by = c("chrom" = "CHR", "chrom_start" = "BP"))
```

### Polynesian GWAS and Selection





```{r}
polyGout <- read.table('~/data/PolyGwas/polynesian_gout_CoreExome24v1.0-1_rsIDconverted.assoc.logistic.tab', header = TRUE, stringsAsFactors = FALSE)
```

```{r}
polyGout %>% filter(TEST == "ADD", P < 0.05) %>% qqman::manhattan(., ylim= c(0,10))
```


```{r}
polyGout %>% filter(TEST == "ADD", P < 1e-5)
```

The two markers that met the threshold for genome-wide significance, rs2725215 and rs2231142 located in _ABCG2_. Two other markers reached the nominal significance threshold, rs2728108 (_ABCG2_) and rs11034401 (inter-genic). This is a problem of having insufficient samples to have the power to detect the other gout loci.

In order to try and prioritise the variants of the gwas, combining the results of the logisitic regression with those of the selection analysis was done in order to investigate snps that did not reach the genome-wide threshold for sigificance in gwas but did have a significant ihs or nsl result.


```{r}
 sig_ihs_val %>% filter(superpop == 'POL') %>% inner_join(., polyGout %>% filter(P < 5e-2, TEST == "ADD"), by = c("chrom" = "CHR", "chrom_start" = "BP")) %>% spread(pop, statvalue) %>% arrange(P) %>% filter((!is.na(CIM) & !is.na(NZM)) | (!is.na(TON) & !is.na(SAM))) %>% filter(abs(NZM) > 2.6 | abs(CIM) > 2.6 | abs(SAM) > 2.6 | abs(TON) > 2.6) %>% select(-TEST, -statid, -contains('gene_'), -ensgeneid, -NMISS, -sig, -STAT) %>% kable(., longtable = TRUE, booktabs = TRUE, format = 'latex')
```

```{r}
sig_ihs_val %>% filter(superpop == 'POL') %>% inner_join(., polyGout %>% filter(P < 5e-2, TEST == "ADD"), by = c("chrom" = "CHR", "chrom_start" = "BP")) %>% ggplot(., aes(x = -log10(P), y = statvalue)) + geom_point()

sig_nsl_val %>% filter(superpop == 'POL') %>% inner_join(., polyGout %>% filter(P < 5e-2, TEST == "ADD"), by = c("chrom" = "CHR", "chrom_start" = "BP")) %>% ggplot(., aes(x = -log10(P), y = statvalue)) + geom_point()
```

> Compared to the comparison in GWAS for gout with the ukbiobank, the Polynesian gwas did not have many snps with a p value lower than 10^-4^

> There were XXX snps that had an ihs that was greater than XXX and a P value less than XXX. With nls there were similar/different snps. The snps that were in common between the two statitics were XXX.

> CNT4N




## Chapter Discussion

- Concepts, estimation and interpretation of SNPbased heritability - Yang2016
- Identifying genetic variants that affect viability in large cohorts - Mostafavi2017

### GWAS results

- what do the different classifications mean
- 


----

UKBB paper discussion


- consistent findings with SUGAR paper
- in sugar paper highest spec/sense with self-report/ULT
- support the use of the self-report or ULT definition when better classification isn't available
- not all patients were captured by all criteria
- winnard was not as good as self-report or ULT use
- when self-report is info available recommend self-report/ULT
- all definitions tested abcg2/slc2a9 were genome-wide significant
- large effect sizes -> effect sizes in GWAS of control of SU levels - consistent with their role in regulation of SU and gout risk
- heritability estimates for variance explained by all common snps 0.282-0.308 (ex hospital def)
- within the range from kottgen (0.27-0.41) and they also calculated using GCTA for SU
- estimates of variance explained in european by common snps are comparable, suggesting that the common genetic variant-mediated heritabilities of SU and gout are similar
- diet, medications and other environmental factors contribute to risk of gout
- heritabilities are estimated under an additive model, so don't include non-additive variance (GxG and GxE), rare variants, CNV
- limited to european -> might not translate to other populations
- limited in that no gold-standard available to determine FPR/FNR


----


Poly gwas

ABCG2 is the usual contender, the other variant rs11034401is an intergenic variant on chr11

- talk about rs2231142 a little and why it is the main gout result

the polynesian gwas is largely underpowered

### Heritability explained

The heritability explained under an additive model for gout was similar to that which was reported for serum urate by @Kottgen2013. There were no statistically significant differences between the classifications. This could come down to the number of cases for each classification 

### GWAS and selection

There was limited seection for gout and urate assocaited loci in the \gls{gbr} population


### Limitations

- computationally the grm takes a lot of resources, hence only using 10000 controls, and not the full dataset
- the differences between the gwases for the ukbiobank are likely a numbers issue
- poly gwas
    - the lack of power -> samplesize
    - data set wasn't imputed
    - imputation of the Polynesian data is difficult, lack of reference haplotypes
    
- Poly selection/gwas
    - the selection results could be because of higher than general population prevalenvce of gout in the cohort
    - reduced markers used

- EUR selection/gwas
    - the greatly reduced number of markers because the selection results were calculated in the coreexome markers


> Maybe bring up the current discussion about the gcta for the use of heritability versus ldak?

<!-- TANYA -->

<!-- Whole-genome principal component analysis vectors were calculated using a subset of 2,858 ancestry informative markers (as identified by Illumina) extracted from the CoreExome whole-genome genotypes. The SmartPCA (EIGENSOFT v6.0.1) [25]) program was used to perform this prinicipal component analysis, with an output of 10 eigenvectors, no outlier removal, and no population size limit. Individuals of non-Polynesian ancestry were included in the analysis group, and the first four vectors plotted against each other to view the clustering of ancestral groupings (Asian, European, Eastern Polynesian, and Western Polynesian). The first four vectors were chosen for inclusion as covariates in the linear regression to account for population stratification and cryptic relatedness, which may cause confounding. -->
<!-- The Illumina Infinium CoreExome v24 bead chip platform was used to genotype participants for ~500,000 variants across the whole-genome. Genotyping was performed at the University of Queensland (Centre for Clinical Genomics) for the Genetics of Gout, Diabetes, and Kidney Disease in Aotearoa cohorts and at AgResearch (Invermay Agricultural Centre) for the Ngati Porou Hauora Charitable Trust cohort. Bead chip genotyping batches were auto-clustered using GenomeStudio v2011.1 software (Illumina, San Diego). The Illumina GenomeStudio best practice guidelines and quality control protocols of Guo et al. (2014) were applied to these auto-clustered genotypes to ensure final genotype calls were of the highest possible quality [23, 24]. After the initial quality control protocols were applied the genotyping batches were merged and relevant quality control steps repeated in the full dataset. -->



<!-- How I did the Polynesian GWAS(s): -->
<!-- Run from: -->
<!-- /Volumes/userdata/staff_users/tanyaflynn/ShortCuts/murray_working_dir/coreExome/Batch_Merge/Merged_Merriman_Brown_Matsuo/PlusStrand_Merge -->
<!-- File Set-Ups -->
<!-- QC1_7-FullMerge1.0and1.1_ExDup_MattBrown_Czech-plus_rsIDconverted -->
<!--     - merged QC batches 1 to 7, matt brown's samples & the czech samples -->
<!--     - excluded duplicate samples -->
<!-- polynesian_gout-affection.pheno -->
<!--     - gout either ACR criteria or GP diagnosed / in a gout drug trial (only 119 not ACR) -->
<!--     - excluded unknown gout status, non-polynesian (either self-report or PCA mismatch), and people <18 years old -->
<!-- QC1_7-FullMerge1.0and1.1_ExDup_MattBrown_Czech-plus_polygout.cov -->
<!--     - sex, age, and PCAs 1-10 (PCAs calculated in full genotype group) -->
<!-- east-poly_list.txt -->
<!--     - anyone self-reporting east polynesian with a PCA labelled east polynesian or general polynesian (not west) -->
<!-- west-poly_list.txt -->
<!--     - anyone self-reporting west polynesian with a PCA labelled west polynesian or general polynesian (not east) -->
<!--     - accidentally used the wrong column for this & excluded Pukapukan & Niuean samples due to their unique clustering -->
<!--         - this list has 772 people, the updated list has 939 (have not updated analysis yet) -->

<!-- Polynesian Gout - actual command run -->
<!-- plink2 \ -->
<!--   --bfile QC1_7-FullMerge1.0and1.1_ExDup_MattBrown_Czech-plus_rsIDconverted \ -->
<!--   --pheno polynesian_gout-affection.pheno \ -->
<!--   --logistic \ -->
<!--   --ci 0.95 \ -->
<!--   --covar QC1_7-FullMerge1.0and1.1_ExDup_MattBrown_Czech-plus_polygout.cov \ -->
<!--   --covar-name Sex,Age,PCA1,PCA2,PCA3,PCA4,PCA5,PCA6,PCA7,PCA8,PCA9,PCA10 \ -->
<!--   --hardy \ -->
<!--   --hwe 1e-6 \ -->
<!--   --freq case-control \ -->
<!--   --maf 0.01 \ -->
<!--   --geno 0.38 \ # to remove the extra czech sample only SNPs -->
<!--   --allow-no-sex \ -->
<!--   --out ../new_polyGWAS/polynesian_gout_CoreExome24v1.0-1_rsIDconverted -->

<!-- East Polynesian Gout - actual command run -->
<!-- plink2 \ -->
<!--   --bfile QC1_7-FullMerge1.0and1.1_ExDup_MattBrown_Czech-plus_rsIDconverted \ -->
<!--   --pheno polynesian_gout-affection.pheno \ -->
<!--   --keep east-poly_list.txt \ -->
<!--   --logistic \ -->
<!--   --ci 0.95 \ -->
<!--   --covar QC1_7-FullMerge1.0and1.1_ExDup_MattBrown_Czech-plus_polygout.cov \ -->
<!--   --covar-name Sex,Age,PCA1,PCA2,PCA3,PCA4,PCA5,PCA6,PCA7,PCA8,PCA9,PCA10 \ -->
<!--   --hardy \ -->
<!--   --hwe 1e-6 \ -->
<!--   --freq case-control \ -->
<!--   --maf 0.01 \ -->
<!--   --geno 0.38 \ # to remove the extra czech sample only SNPs -->
<!--   --allow-no-sex \ -->
<!--   --out ../new_polyGWAS/east-polynesian_gout_CoreExome24v1.0-1_rsIDconverted -->
<!-- West Polynesian Gout - actual command run -->
<!-- plink2 \ -->
<!--   --bfile QC1_7-FullMerge1.0and1.1_ExDup_MattBrown_Czech-plus_rsIDconverted \ -->
<!--   --pheno polynesian_gout-affection.pheno \ -->
<!--   --keep west-poly_list.txt \ -->
<!--   --logistic \ -->
<!--   --ci 0.95 \ -->
<!--   --covar QC1_7-FullMerge1.0and1.1_ExDup_MattBrown_Czech-plus_polygout.cov \ -->
<!--   --covar-name Sex,Age,PCA1,PCA2,PCA3,PCA4,PCA5,PCA6,PCA7,PCA8,PCA9,PCA10 \ -->
<!--   --hardy \ -->
<!--   --hwe 1e-6 \ -->
<!--   --freq case-control \ -->
<!--   --maf 0.01 \ -->
<!--   --geno 0.38 \ # to remove the extra czech sample only SNPs -->
<!--   --allow-no-sex \ -->
<!--   --out ../new_polyGWAS/west-polynesian_gout_CoreExome24v1.0-1_rsIDconverted -->
<!-- ## results files were moved to poly_gwas2017 folder at some point -->
<!-- ###### -->
<!-- Polynesian Urate GWAS: -->
<!-- Run in: -->
<!-- /Volumes/userdata/staff_users/tanyaflynn/ShortCuts/murray_working_dir/coreExome/Batch_Merge/Merged_Merriman_Brown_Matsuo/ -->
<!-- plink2 \ -->
<!--   --bed QC1_7-FullMerge1.0and1.1_ExDup_MattBrown_Czech-plus_correctAff.bed -->
<!--   --bim QC1_7-FullMerge1.0and1.1_ExDup_MattBrown_Czech-plus_correctAff.bim -->
<!--   --fam QC1_7-FullMerge1.0and1.1_ExDup_MattBrown_Czech-plus_polyurateAff.fam -->
<!--   --keep Polynesian_Controls.txt # identical set to controls used in gout regression -->
<!--   --linear -->
<!--   --ci 0.95 -->
<!--   --covar QC1_7-FullMerge1.0and1.1_ExDup_MattBrown_Czech-plus_polygout.cov -->
<!--   --covar-name Sex,Age,PCA1,PCA2,PCA3,PCA4,PCA5,PCA6,PCA7,PCA8,PCA9,PCA10 -->
<!--   --hardy -->
<!--   --hwe 1e-6 -->
<!--   --freq -->
<!--   --maf 0.01 -->
<!--   --geno 0.38 -->
<!--   --allow-no-sex -->
<!--   --out ../../poly_gwas2017/PolyUrate_CoreExome24v1.0-1_NZ-MB-HM_Gout -->