# Selection and Association Studies

```{r assocLoad, include=FALSE, warning = FALSE}
source('scripts/Annotate_genes.R')
library(GenomicRanges)
library(knitr)
library(kableExtra)
library(tidyverse)
source('scripts/05_ukbb_filteredSnps.R')
knitr::opts_chunk$set(echo = FALSE)
models <- readRDS('data/Ukbiobank/pheno_models.RDS')
pheno_models <- bind_rows(lapply(models, function(x){cbind(adjuster = x$adjuster,broom::glance(x$model))})) %>% arrange(AIC)


load('~/data/ukbiobank_gout_bd_refined2016-04-26.RData')
fam <- read.table('~/Git_repos/bookdown_thesis/data/Ukbiobank/chr1impv1.fam', stringsAsFactors = FALSE)

# f.31.0.0 = sex, 1 = male
# f.48.0.0 = waist
# f.50.0.0 = height
# f.21001.0.0 = BMI
# f.21000.0.0 = ethnicity
# f.20002.* =  self report illness

ukbio_gout <- bd_refined_gout %>% filter(f.eid %in% fam$V2) %>% select(goutaff, 'age' = f.21003.0.0,"bmi" = f.21001.0.0, 'sex' = f.31.0.0, 'height' = f.50.0.0, 'waist' = f.48.0.0, 'hip' = f.49.0.0, 'eth' = f.21000.0.0, febuxostat, allopurinol, colchicine, sulphinpyrazone, gouthosp, goutwinnard, goutself, gout_winnard_self, gout_self_ult, goutself, goutult ) %>% mutate(waist_height_ratio = waist / height, sex_factor = factor(if_else(sex == 0, "female", "male", missing = NULL)), cc = if_else(goutaff == 1, "case", "control", missing = NULL), waist_hip_ratio = waist / hip) %>% filter(!is.na(goutaff) & eth %in% c(1001,1002,1003)) %>% group_by(cc) 
```


This chapter about using the results of selection analyses to better inform \gls{gwas}

The first section of this chapter has been published as part of @Cadzow2017

The UK Biobank provides a unique opportunity

objectives:
- Investigate the performance of gout definitions on association studies
- Investigate the use of association studies in conjunction with selection analyses



Selection, GWAS, and Imputation of Polynesian Populations


useful papers:

- @Fay2013
- @Dudley2012

prioritisation of variants from gwas
- @Hou2013
- maybe @Kindt2013

Improvement gwas through the use of imputation:

- Machini nature papers 2010?
- HRC panel paper

refer to gwas done by ukbio in all phenotypes and say it looks similar

## Performance of gout definitions

Part of this work has been published in @Cadzow2017.

This section describes the gout GWAS and vGWAS performed in the UKBiobank data and the comparision with the results from the selection analysis in the 1000 Genomes Project data.


UK Biobank approval 12611


_Introduction of ukbiobank_

The UK Biobank is a publically available 

The Biobank is a collection of 500,000 individuals, mostly of European ancestry, from around the United Kingdom, aged between 40 and 69. It consists of genetic,  health, and lifestype information.


_Motivation behind using the different definitions_

In epidemiological studies the use of an accurate case definition is important. 

There is variation in the gout definitions used from the multipurpose cohort studies that are used for genetic analysis. 

Consistent case definition is also important for comparisons between cohorts, this can be done by combining the genetic data and performing the \gls{gwas}. Another common way of combining genetic association studies is through meta-analysis. Having the same case definitions between studies reduces the variations 

For gout, the gold standard case definition is the mono-soldium urate crystals in the synovial fluid of the joint. A sustitute for crystals in the synovial fluid is the American College of Rheumatology (ACR) criteria. The criteria require a minimum of six of twelve conditions to be met. 

It is uncommon for multipurpose cohorts to report either the gold-standard, or ACR criteria, and instead more common for a self-report diagnosis to be reported. The UK Biobank includes the self-report diagnosis that were collected by survey at the time of recruitment, but it also has hospital admission information, for both primary and secondary diagnosis.  



Power of genetic case-control studies increases with accurate case definition. This means that there are the maximum number of cases, and the maximum number of disease free controls.





### European Gout GWAS

<!-- acr report using unfiltered controls -->
This \gls{gwas} was performed using the interim data release from the UK Biobank (approval number 12611) downloaded November 2015. There was genetic information for `r NROW(fam)` individuals. For the \gls{gwas} itself, it was limited to be only people who had self reported an ethnic background of British, Irish, or 'any other white background'.

Inclusion criteria were European ethnicity, aged between 40 and 69 years and had genome wide genotypes available. The exclusion criteria used was: mismatch between self-reported sex and genetic sex, genotype quality control failure, relatedness, or either a primary or secondary hospital diagnosis of kidney disease (International Classification of Diseases, Tenth Revision (ICD-10), codes I12, I13, N00-N05, N07, N11, N14, N17â€“N19, Q61, N25.0, Z49, Z94.0, Z99.2), participants aged 70 years and over, and those with kidney disease, because these are risk factors for secondary gout.


_Classifications used_

There were four main classification criteria that were used to define gout cases. Hospital defined gout involved either a primary or secondary hospital diagnosis for gout (ICD-10 code M10, including sub-codes). Self-report of gout was defined as reporting having gout at the time of the study interview. Use of \gls{ult} used self-report data for the use of allopurinol, febuxostat, or sulphinpyrazone, and not having a hospital diagnosis of leukemia or lymphoma (ICD-10 codes C81-C96). The Winnard definition was based on @Winnard2012 and consisted of a hospital diagnosis of gout, or self report of gout specific medication (\gls{ult} or colchicine). The largest number of people were classified through either self-report or the use of urate lowering therapy (Figure \@ref(fig:ukbbupsetplot)). Additional exclusion criteria were applied to participants not classified as having gout from these definitions. The criteria were cortico-steroidal use, non-steroidal anti-inflammatory drug use or probenicid use. 


Prior to conducting the \gls{gwas} for gout, phenotypic models were assessed using any method of classifying gout. Logistic regression was performed and the Akaike information criterion (AIC) for each model was used for model selection. The covariates used included: age, sex, BMI. Each covariate was significant (P < 0.05) on their own, age (P = \num{`r broom::tidy(models[[1]]$model) %>% filter(term == 'age') %>% .[['p.value']] %>% format(., scientific = TRUE, digits = 3)`}), sex (P = \num{`r broom::tidy(models[[2]]$model) %>% filter(term == 'sex_factormale') %>% .[['p.value']] %>% format(., scientific = TRUE, digits = 3)`}), and \gls{bmi} (P = \num{`r broom::tidy(models[[4]]$model) %>% filter(term == 'bmi') %>% .[['p.value']] %>% format(., scientific = TRUE, digits = 3)`}). In the combined model with all three covariates there was an AIC for the model of 19558.30 compared to models with each with age (AIC 22706.05), sex (AIC 21041.80) and \gls{bmi} (AIC 22001.03). Based on the AIC for the combined model, age, sex, and \gls{bmi} were used as the covariates in the \gls{gwas} for gout. The mean \gls{bmi} for all participants was `r ukbio_gout %>%ungroup() %>%  summarise(bmi = mean(bmi, na.rm = TRUE)) %>% pull(bmi) %>% round(., 2)` kg/m^2^, the mean age was `r ukbio_gout %>%ungroup() %>% summarise(age = mean(age, na.rm = TRUE)) %>% pull(age) %>% round(., 2)` years, and `r ukbio_gout %>%ungroup() %>% summarise(percent_male = (1 - sum(sex)/NROW(.)) *100) %>% pull(percent_male) %>% round(., 2)`% were male. A breakdown by affection status is in Table \@ref(tab:ukbbclin).



```{r, warning = FALSE, eval = FALSE}

bind_rows(lapply(models, function(x){cbind(adjuster = x$adjuster,broom::glance(x$model))})) %>% arrange(AIC)
```

```{r, results = 'asis', eval = FALSE}

broom::tidy(models[[5]]$model) %>% mutate(p.value = format(p.value, scientific = TRUE, digits = 3))%>% kable(., digits = 3, caption = '(\\#tab:phenomodel) Logistic regression for gout adjusted by age, sex, and BMI')
```


```{r, eval = FALSE}
lapply(c(1,2,4,3,5 ), function(x){broom::tidy(models[[x]]$model)})
```


** UKBiobank participant characteristics **

```{r, results = 'asis'}
ukbio_gout %>% summarise(n = n(), percent_male = 1 - sum(sex)/NROW(.), mean_age = mean(age, na.rm = TRUE), mean_bmi = mean(bmi, na.rm = TRUE),  febuxostat = sum(febuxostat, na.rm = TRUE), allopurinol = sum(allopurinol, na.rm = TRUE), sulphinpyrazone = sum(sulphinpyrazone, na.rm = TRUE), colchicine = sum(colchicine, na.rm = TRUE), hosp = sum(gouthosp, na.rm =TRUE), self = sum(goutself, na.rm=TRUE), winnard = sum(goutwinnard, na.rm = TRUE), ult = sum(goutult, na.rm=TRUE) ) %>% t() %>% data.frame() %>% mutate(var = rownames(.)) %>%  dplyr::slice(., -1) %>% select(var, 'Case' = 1, 'Control' = 2) %>% kable(., booktabs = TRUE, caption = "(\\#tab:ukbbclin) Clinical details of participants.")
```



(ref:ukbbupsetplot) An upset plot showing the total sizes (left) and intersections (upper) of the different classification criteria.

```{r ukbbupsetplot, fig.width=6.5, fig.cap='(ref:ukbbupsetplot)', fig.asp=1/1.5 }

listInput <- readRDS('data/Ukbiobank/upsetPlotList.RDS')
UpSetR::upset(UpSetR::fromList(listInput), order.by = "freq", text.scale = 1.3 )
```

There were a total of `r ukbio_gout %>% filter(goutaff ==1) %>% NROW()` individuals that had any classification of gout and `r ukbio_gout %>% filter(goutaff ==0) %>% NROW()` controls. Of those with any classification of gout, there were `r ukbio_gout %>% filter(goutaff ==1 & gouthosp == 1) %>% NROW()` with a hospital diagnosis, `r ukbio_gout %>% filter(goutaff ==1 & goutult ==1) %>% NROW()` with urate lowering therapy, `r ukbio_gout %>% filter(goutaff ==1 & goutwinnard == 1) %>% NROW()` with the Winnard criteria, and `r ukbio_gout %>% filter(goutaff ==1 & goutself ==1) %>% NROW()` with self-reported gout. The overlaps in classification between each group can be seen in Figure \@ref(fig:ukbbupsetplot). There was considerable overlap between the self-report, urate lowering therapy, and Winnard classifications with 1242 individuals meeting all three classifications.



\gls{gwas} were performed using Plink1.9b [@plink] for each gout classification and 

- GWAS results
- Heritability explained

```{r, message = FALSE}
kot_t1 <- read.csv("~/data/kottgen_table1.csv", header = TRUE, stringsAsFactors = FALSE)
uk_kot_t1 <-read_delim(file = 'data/Ukbiobank/kottgen_t1_age_sex_bmi_results.txt', delim = '\t', col_names = c("file","SNP","POS","A1","TEST","NMISS","OR", "SE", "L95", "U95", "STAT", "P")) %>% separate(file, c("file","CHR"), sep = ':') %>% mutate(file = stringi::stri_replace(file, "selfult", regex ="self_ult" ))%>%  mutate(gout = map_chr(stringi::stri_split(file, regex = "_"), 1) %>% stringi::stri_sub(9)) %>% left_join(., kot_t1 %>% select(SNP, grail_gene), by = "SNP")
```

(ref:kotplot) Plot showing ORs (95% CI) for the 30 \gls{snp}s reported in @Kottgen2013 based on gout definitions. Original Kottgen reported ORs are shown as red circles.

```{r kotplot, fig.height=7, fig.cap = '(ref:kotplot)', warning = FALSE}

flip = c("rs2307394", "rs2079742", "rs7224610", "rs7188445","rs1178977", "rs6770152", "rs7193778", "rs164009", "rs653178","rs2078267", "rs478607", "rs12498742")
uk_kot_t1 %>% mutate(OR = ifelse(SNP %in% flip, 1/OR, OR), L95 = ifelse(SNP %in% flip, 1/L95, L95), U95 = ifelse(SNP %in% flip, 1/U95, U95))  %>%  select(gout, SNP, grail_gene,OR, L95, U95) %>% 
bind_rows(., kot_t1 %>% mutate(gout = "Kottgen") %>% select(gout, SNP, grail_gene, OR) ) %>% unite("SNP", c("grail_gene", "SNP"), sep = '\n') %>% mutate(highlight = ifelse(gout == "Kottgen", "b","a")) %>% filter(gout != 'all') %>% mutate(gout = factor(gout, ordered = TRUE)) %>% mutate(gout = forcats::fct_relevel(gout, "Kottgen", "self","ult","hosp","winnard","selfult"))%>% mutate(gout = forcats::fct_recode(gout, "Hospital" = "hosp", "Self Report" = 'self', "Self Report or ULT" = 'selfult', "ULT" = "ult", "Winnard" = "winnard" )) %>% 
  # mutate(gout = case_when(
  #   gout == 'hosp' ~ "Hospital", 
  #   gout == "self" ~ "Self Report", 
  #   gout == "selfult" ~ "Self Report or ULT",
  #   gout == "ult" ~ "ULT",
  #   gout == "winnard" ~ "Winnard")) %>% 
  ggplot(data = ., aes(x = gout, y = OR, shape = highlight, colour = highlight)) + 
  geom_point() + facet_wrap(~SNP) + geom_linerange(aes(x = gout, ymin = L95, ymax = U95)) +geom_hline(yintercept = 1) +theme_bw() + theme(axis.text.x = element_text(angle=-90, vjust = 0.5, hjust = 0)) + theme(axis.title = element_blank(), legend.position = 'none') + scale_color_manual(values = c("black","red")) 
```




#### GWAS results

The top SNPs from the GWAS were:

number of significant snps

- hosp `r hosp_asb %>% filter(P < 1e-8, nchar(A1) ==1 ) %>% NROW()`
- self `r self_asb %>% filter(P < 1e-8, nchar(A1) ==1 ) %>% NROW()`
- ult `r ult_asb %>% filter(P < 1e-8, nchar(A1) ==1 ) %>% NROW()`
- self ult `r selfult_asb %>% filter(P < 1e-8, nchar(A1) ==1 ) %>% NROW()`
- winnard `r winnard_asb %>% filter(P < 1e-8, nchar(A1) ==1 ) %>% NROW()`

(ref:Overlap) Overlap of markers that reached at nominal genome-wide significance between the different gout classifications.

```{r ukOverlap, fig.cap = '(ref:ukOverlap)'}
listInput <- list(hosp = hosp_asb %>% filter(P < 1e-5, nchar(A1) ==1 ) %>% pull(SNP),
                  self = self_asb %>% filter(P < 1e-5, nchar(A1) ==1 ) %>% pull(SNP),
                  ult = ult_asb %>% filter(P < 1e-5, nchar(A1) ==1 ) %>% pull(SNP),
                  selfult = selfult_asb %>% filter(P < 1e-5, nchar(A1) ==1 ) %>% pull(SNP),
                  winnard = winnard_asb %>% filter(P < 1e-5, nchar(A1) ==1 ) %>% pull(SNP)
                  )

UpSetR::upset(UpSetR::fromList(listInput), order.by = "freq", text.scale = 1.3, nintersects = NA )
```


```{r}
# create the table of kottgen snps from the data
# age, sex, and bmi adjusted

# still need to flip snps to be in the same order as kottgen
uk_kot_t1 %>% select(-file, -TEST, -SE, -NMISS,-STAT, -POS) %>% mutate_at(., vars("OR",contains('95')), funs(sprintf("%.2f",. ))) %>% unite("CI",c("L95","U95"), sep = " -") %>% mutate(SNP = paste0(SNP, ' (', grail_gene,')'), Effect = paste0(OR, ' [', CI,'], ', P)) %>% arrange(P) %>%  select(-grail_gene, -OR, -CI, -P, -CHR) %>% spread(key = gout, value = Effect) %>% select(-all)%>% kable(., format = 'latex', caption = '(\\#tab:ukbbkot)', booktabs = TRUE) %>% kable_styling(latex_options = 'scale_down')

```



- hosp

```{r}
hosp_asb %>% filter(P < 1e-8, nchar(A1) ==1 ) %>% mutate(P = format(P, scientific = TRUE)) %>% head() %>%  kable()
```


```{r}
dat <- hosp_asb 
txdb_gene_annotate(dat %>%  filter(P < 1e-8, nchar(A1) ==1 ) %>% mutate(chrom = paste0("chr",CHR), chrom_start = BP, chrom_end = BP +1, marker = SNP) %>% select(contains('chrom'), marker) %>% GRanges()) %>% left_join(., dat %>% mutate(start = BP, seqnames = paste0("chr",CHR)), by = c('seqnames','start')) %>% select(CHR, SNP, BP, SYMBOL, A1, OR, SE, L95, U95, P ) %>% arrange(P) %>% mutate(SYMBOL = ifelse(is.na(SYMBOL), paste0("NA_",CHR,"_",BP) , SYMBOL)) %>% group_by(CHR, SYMBOL ) %>% filter(min(P) == P) %>% arrange(CHR, BP) %>% mutate(P = format(P, scientific = TRUE)) %>% kable()
```


- self

There were ```r self_asb %>% filter(P < 1e-8,  nchar(A1) ==1) %>% mutate(P = format(P, scientific = TRUE)) %>% tally()``` \gls{snp}s that met the genome-wide significance threshold. 

```{r}
self_asb %>% filter(P < 1e-8,  nchar(A1) ==1) %>% mutate(P = format(P, scientific = TRUE)) %>% dim()# kable()
```

```{r}
dat <- self_asb 
txdb_gene_annotate(dat %>%  filter(P < 1e-8, nchar(A1) ==1 ) %>% mutate(chrom = paste0("chr",CHR), chrom_start = BP, chrom_end = BP +1, marker = SNP) %>% select(contains('chrom'), marker) %>% GRanges()) %>% left_join(., dat %>% mutate(start = BP, seqnames = paste0("chr",CHR)), by = c('seqnames','start')) %>% select(CHR, SNP, BP, SYMBOL, A1, OR, SE, L95, U95, P ) %>% arrange(P) %>% mutate(SYMBOL = ifelse(is.na(SYMBOL), paste0("NA_",CHR,"_",BP) , SYMBOL)) %>% group_by(CHR, SYMBOL ) %>% filter(min(P) == P) %>% arrange(CHR, BP) %>% mutate(P = format(P, scientific = TRUE)) %>% kable()
```



- self + ULT

```{r}
dat <- selfult_asb 
txdb_gene_annotate(dat %>%  filter(P < 1e-8, nchar(A1) ==1 ) %>% mutate(chrom = paste0("chr",CHR), chrom_start = BP, chrom_end = BP +1, marker = SNP) %>% select(contains('chrom'), marker) %>% GRanges()) %>% left_join(., dat %>% mutate(start = BP, seqnames = paste0("chr",CHR)), by = c('seqnames','start')) %>% select(CHR, SNP, BP, SYMBOL, A1, OR, SE, L95, U95, P ) %>% arrange(P) %>% mutate(SYMBOL = ifelse(is.na(SYMBOL), paste0("NA_",CHR,"_",BP) , SYMBOL)) %>% group_by(CHR, SYMBOL ) %>% filter(min(P) == P) %>% arrange(CHR, BP) %>% mutate(P = format(P, scientific = TRUE)) %>% kable()
```


- ULT

```{r}
dat <- ult_asb 
txdb_gene_annotate(dat %>%  filter(P < 1e-8, nchar(A1) ==1 ) %>% mutate(chrom = paste0("chr",CHR), chrom_start = BP, chrom_end = BP +1, marker = SNP) %>% select(contains('chrom'), marker) %>% GRanges()) %>% left_join(., dat %>% mutate(start = BP, seqnames = paste0("chr",CHR)), by = c('seqnames','start')) %>% select(CHR, SNP, BP, SYMBOL, A1, OR, SE, L95, U95, P ) %>% arrange(P) %>% mutate(SYMBOL = ifelse(is.na(SYMBOL), paste0("NA_",CHR,"_",BP) , SYMBOL)) %>% group_by(CHR, SYMBOL ) %>% filter(min(P) == P) %>% arrange(CHR, BP) %>% mutate(P = format(P, scientific = TRUE)) %>% kable()
```


- all


```{r}
dat <- all_asb 
txdb_gene_annotate(dat %>%  filter(P < 1e-8, nchar(A1) ==1 ) %>% mutate(chrom = paste0("chr",CHR), chrom_start = BP, chrom_end = BP +1, marker = SNP) %>% select(contains('chrom'), marker) %>% GRanges()) %>% left_join(., dat %>% mutate(start = BP, seqnames = paste0("chr",CHR)), by = c('seqnames','start')) %>% select(CHR, SNP, BP, SYMBOL, A1, OR, SE, L95, U95, P ) %>% arrange(P) %>% mutate(SYMBOL = ifelse(is.na(SYMBOL), paste0("NA_",CHR,"_",BP) , SYMBOL)) %>% group_by(CHR, SYMBOL ) %>% filter(min(P) == P) %>% arrange(CHR, BP) %>% mutate(P = format(P, scientific = TRUE)) %>% kable()
```


manhattan plot for the all \@ref(fig:manhatAllAgeSex)

(ref:manhatAllAgeSex) _Manhattan plot for Gout GWAS adjusted for age and sex_

```{r manhatAllAgeSex, echo = FALSE, fig.cap='(ref:manhatAllAgeSex)', fig.width=7, fig.asp=1, out.width= '95%', fig.align='center'}
knitr::include_graphics('images/05_selection_and_association/all_age_sex.png')
```

```{r manhatAllAgeSexBmi,echo = FALSE, fig.cap='(ref:manhatAllAgeSexBmi)', fig.width=10, fig.asp=1, out.width = '95%', fig.align='center', out.extra='angle=90'}
knitr::include_graphics('images/05_selection_and_association/all_age_sex_bmi.png')
```

(ref:manhatAllAgeSexBmi) _Manhattan plot for Gout GWAS adjusted for age, sex and BMI_

```{r, engine='block', eval = FALSE, echo = FALSE, include=FALSE }
For this section I want to compare the results for each of the gwases and see if the same 
snps come out and perhaps in the same order.

Also want to compare the results from the age+sex adjusted versus the results from the age+sex+waist+waist:height
```

<!---
ALL GROUP: the most significant snps are from ABCG2. SLC2A9 also features highly. The majority of the significant snps are from these two genes.
--->

#### Heritability Explained

```{r, engine='block', eval = FALSE, echo = FALSE, include=FALSE }
here I want to include the results from the heritability analysis across the different criteria
```

To see if there was a difference between gout classifications in the genetic component of the variance explained by an additive model in gout, heritability estimates were calculated for each gout classification. To do this, 10000 controls were randomly selected and were used with each different gout criteria and heritability estimates for each chromosome were calculated using GCTA 1.26.0 [@ref]. The heritability estimates were 0.289 (SE 0.034) for the self-report of gout or ULT use definition, 0.283 (SE 0.036) for the self-report of gout definition, 0.282 (SE 0.040) for the Winnard definition, 0.308 (SE 0.044) for the ULT use definition and 0.236 (SE 0.160) for the hospital diagnosis definition. There were no statistically significant differences between the heritability estimates of any of the different classifications. The heritability estimates were similar to estimates of genetic variance of serum urate [@Kottgen2013]




<!-- Kottgen refs 10-12 put heritability explained for serum urate at 40-70%.  Kottgen reported heritability explained in serum urate loci that were adjusted for age and sex from 0.27 to 0.41. -->



### European selection 
- EUR/GBR/CEU selection results






## Polynesian GWAS and Selection


A \gls{gwas} was also performed for individuals with Polynesian ancestry. Gout affection was determined by the American college of rheumatology criteria [@ref], or a doctor diagnosis, or enrolled in a gout drug trial. All participants were older than 18. 

The \gls{gwas} was adjusted for age, sex, and the first ten principle components calculated from XXX ancestry informative markers.

The \gls{gwas} was conducted by Dr Tanya Major, a member of the Merriman Lab group.


Filtered for hwe < 1e-6 and maf > 0.01


```{r}
polyGout <- read.table('~/data/PolyGwas/polynesian_gout_CoreExome24v1.0-1_rsIDconverted.assoc.logistic.tab', header = TRUE, stringsAsFactors = FALSE)
```

```{r}
polyGout %>% filter(TEST == "ADD", P < 0.05) %>% qqman::manhattan(., ylim= c(0,10))
```


```{r}
polyGout %>% filter(TEST == "ADD", P < 1e-5)
```

There two markers that met the threshold for genome-wide significance, rs2725215 and rs2231142. Two other markers reached the nominal significance threshold, rs2728108 and rs11034401.

## Chapter Discussion

- Concepts, estimation and interpretation of SNPbased heritability - Yang2016
- Identifying genetic variants that affect viability in large cohorts - Mostafavi2017


### Heritability explained

The heritability explained under an additive model for gout was similar to that which was reported for serum urate by @Kottgen2013. There were no statistically significant differences between the classifications. This could come down to the number of cases for each classification 

<!-- TANYA -->
<!-- How I did the Polynesian GWAS(s): -->
<!-- Run from: -->
<!-- /Volumes/userdata/staff_users/tanyaflynn/ShortCuts/murray_working_dir/coreExome/Batch_Merge/Merged_Merriman_Brown_Matsuo/PlusStrand_Merge -->
<!-- File Set-Ups -->
<!-- QC1_7-FullMerge1.0and1.1_ExDup_MattBrown_Czech-plus_rsIDconverted -->
<!--     - merged QC batches 1 to 7, matt brown's samples & the czech samples -->
<!--     - excluded duplicate samples -->
<!-- polynesian_gout-affection.pheno -->
<!--     - gout either ACR criteria or GP diagnosed / in a gout drug trial (only 119 not ACR) -->
<!--     - excluded unknown gout status, non-polynesian (either self-report or PCA mismatch), and people <18 years old -->
<!-- QC1_7-FullMerge1.0and1.1_ExDup_MattBrown_Czech-plus_polygout.cov -->
<!--     - sex, age, and PCAs 1-10 (PCAs calculated in full genotype group) -->
<!-- east-poly_list.txt -->
<!--     - anyone self-reporting east polynesian with a PCA labelled east polynesian or general polynesian (not west) -->
<!-- west-poly_list.txt -->
<!--     - anyone self-reporting west polynesian with a PCA labelled west polynesian or general polynesian (not east) -->
<!--     - accidentally used the wrong column for this & excluded Pukapukan & Niuean samples due to their unique clustering -->
<!--         - this list has 772 people, the updated list has 939 (have not updated analysis yet) -->
<!-- Polynesian Gout - actual command run -->
<!-- plink2 \ -->
<!--   --bfile QC1_7-FullMerge1.0and1.1_ExDup_MattBrown_Czech-plus_rsIDconverted \ -->
<!--   --pheno polynesian_gout-affection.pheno \ -->
<!--   --logistic \ -->
<!--   --ci 0.95 \ -->
<!--   --covar QC1_7-FullMerge1.0and1.1_ExDup_MattBrown_Czech-plus_polygout.cov \ -->
<!--   --covar-name Sex,Age,PCA1,PCA2,PCA3,PCA4,PCA5,PCA6,PCA7,PCA8,PCA9,PCA10 \ -->
<!--   --hardy \ -->
<!--   --hwe 1e-6 \ -->
<!--   --freq case-control \ -->
<!--   --maf 0.01 \ -->
<!--   --geno 0.38 \ # to remove the extra czech sample only SNPs -->
<!--   --allow-no-sex \ -->
<!--   --out ../new_polyGWAS/polynesian_gout_CoreExome24v1.0-1_rsIDconverted -->
<!-- East Polynesian Gout - actual command run -->
<!-- plink2 \ -->
<!--   --bfile QC1_7-FullMerge1.0and1.1_ExDup_MattBrown_Czech-plus_rsIDconverted \ -->
<!--   --pheno polynesian_gout-affection.pheno \ -->
<!--   --keep east-poly_list.txt \ -->
<!--   --logistic \ -->
<!--   --ci 0.95 \ -->
<!--   --covar QC1_7-FullMerge1.0and1.1_ExDup_MattBrown_Czech-plus_polygout.cov \ -->
<!--   --covar-name Sex,Age,PCA1,PCA2,PCA3,PCA4,PCA5,PCA6,PCA7,PCA8,PCA9,PCA10 \ -->
<!--   --hardy \ -->
<!--   --hwe 1e-6 \ -->
<!--   --freq case-control \ -->
<!--   --maf 0.01 \ -->
<!--   --geno 0.38 \ # to remove the extra czech sample only SNPs -->
<!--   --allow-no-sex \ -->
<!--   --out ../new_polyGWAS/east-polynesian_gout_CoreExome24v1.0-1_rsIDconverted -->
<!-- West Polynesian Gout - actual command run -->
<!-- plink2 \ -->
<!--   --bfile QC1_7-FullMerge1.0and1.1_ExDup_MattBrown_Czech-plus_rsIDconverted \ -->
<!--   --pheno polynesian_gout-affection.pheno \ -->
<!--   --keep west-poly_list.txt \ -->
<!--   --logistic \ -->
<!--   --ci 0.95 \ -->
<!--   --covar QC1_7-FullMerge1.0and1.1_ExDup_MattBrown_Czech-plus_polygout.cov \ -->
<!--   --covar-name Sex,Age,PCA1,PCA2,PCA3,PCA4,PCA5,PCA6,PCA7,PCA8,PCA9,PCA10 \ -->
<!--   --hardy \ -->
<!--   --hwe 1e-6 \ -->
<!--   --freq case-control \ -->
<!--   --maf 0.01 \ -->
<!--   --geno 0.38 \ # to remove the extra czech sample only SNPs -->
<!--   --allow-no-sex \ -->
<!--   --out ../new_polyGWAS/west-polynesian_gout_CoreExome24v1.0-1_rsIDconverted -->
<!-- ## results files were moved to poly_gwas2017 folder at some point -->
<!-- ###### -->
<!-- Polynesian Urate GWAS: -->
<!-- Run in: -->
<!-- /Volumes/userdata/staff_users/tanyaflynn/ShortCuts/murray_working_dir/coreExome/Batch_Merge/Merged_Merriman_Brown_Matsuo/ -->
<!-- plink2 \ -->
<!--   --bed QC1_7-FullMerge1.0and1.1_ExDup_MattBrown_Czech-plus_correctAff.bed -->
<!--   --bim QC1_7-FullMerge1.0and1.1_ExDup_MattBrown_Czech-plus_correctAff.bim -->
<!--   --fam QC1_7-FullMerge1.0and1.1_ExDup_MattBrown_Czech-plus_polyurateAff.fam -->
<!--   --keep Polynesian_Controls.txt # identical set to controls used in gout regression -->
<!--   --linear -->
<!--   --ci 0.95 -->
<!--   --covar QC1_7-FullMerge1.0and1.1_ExDup_MattBrown_Czech-plus_polygout.cov -->
<!--   --covar-name Sex,Age,PCA1,PCA2,PCA3,PCA4,PCA5,PCA6,PCA7,PCA8,PCA9,PCA10 -->
<!--   --hardy -->
<!--   --hwe 1e-6 -->
<!--   --freq -->
<!--   --maf 0.01 -->
<!--   --geno 0.38 -->
<!--   --allow-no-sex -->
<!--   --out ../../poly_gwas2017/PolyUrate_CoreExome24v1.0-1_NZ-MB-HM_Gout -->