# Conclusions 

## Summary

This thesis set out to investigate the role of selection in Polynesian populations and the effects on urate and metabolic disease. There were three main objectives that were covered. In chapter 3, the focus was on identifying regions of the genome in Polynesian populations that exhibited 'signals of selection' and if any of these regions had previously been associated with urate and metabolic disease. 

Regions that had evidence of possible selection were included calcium channels, which were involved in cardiac and cell signalling functions, both of which are likely to have selective pressure acting on them.

The loci that were associated with urate that exhibited evidence of possible selection were primarily not urate transporters but loci that were associated with multiple metabolic diseases.

The urate transporters that did show signs of selection, had weak evidence and only in a single sample Polynesian population that was enriched for gout 10-fold higher than the general population it was representing.



- there was limited evidence of selection in metabolic disease
- For gout there was limited evidence of selection, where there was it was not in the main effect loci for gout (slc2a9/abgc2) or the urate transporters, except TON where it is possible it's due to the high gout proportion
- diabetes
- obesity
- metabolic syndrome
- kidney disease



Chapter 4 was an investigation into the genetic similarities of populations. This chapter added to the evidence that the Polynesian populations had similarities to modern day East Asian populations, and from the migration and settlement histories have a shared ancestry. There was also evidence from all of the clustering methodologies that while the Polynesian populations were most similar to each other in a global context, there was in fact genetic differences between the East and West Polynesian populations.

The selection and neutrality statistics showed that populations within a super population were most similar to one another, with the greatest differences being between super populations. There was also evidence of a shared ancestry in both the frequency spectrum of variants, as well as with haplotypes that was consistent with the Out of Africa migration and subsequent population movements. There was not however a specific signal for selection that appeared in the Polynesian populations for urate or any of the metabolic disease associated loci, but instead a commonality between the frequency spectrum of similar geographic populations. 

The extremes of the frequency based selection and neutrality statistics showed that there was similarity in the regions of the genome within a super population, but the regions that were in the extremes differed between super populations.


- The populations had similar statistic values within a super population but super populations were different
- The polynesians were most closely related to EAS.
- The east and west were most similar but also different


Chapter 5 investigated the use of gout definition that was common amungst multi-purpose cohorts and assessed the performance of multiple definitions. It was found that the best definition was that of self-reported gout or self-report of \gls{ult} usage, when the \gls{acr} criteria or observation of urate crystals in the synovial fluid is not available. The use of selection statistics in prioritisation of \gls{gwas} loci revealed that there were several loci that had possible evidence of selection that were "in the noise" of the \gls{gwas} signal, however, these loci were limited to previously identified regions.



The combination of \gls{gwas} results for the haplotypic selection results 


Overall



summary of regions that were selected

was gout/urate and other diseases under selection?


## Study limitations

One of the biggest limiting factors for this research project was that it used \gls{snp} array data. Ideally whole genome sequencing would have been used, however, Polynesian populations are largely under-represented in large sequencing projects. Using \gls{snp} array data impacted on all aspects of the selection statistics analysis. The marker density of the \gls{snp} array meant that the windowed statistics had a small number of markers per window, compared to the HapMap data, and also whole genome sequence data giving the highest density. This meant that the minimum number of markers of four was conveying signal for 100 kb regions of the genome. The data from the CoreExome \gls{snp} array could have been imputed, however, due to the Polynesian haplotypes

- chip data
- imputation
- phasing
    - polynesians aren't in the 1kgp reference haplotypes
- filters
    - QC filters
    - HWE
    - MAF

Another limiting factor for the analyses in chapters 3, 4 and 5, was the focus on only positive selection. 
  
- focus on positive selection
-time-frame of the statistics used
    
use of the gwas catalog, while extremely useful in terms of automatic defining of gene lists, does have a few drawbacks

- there is a potential bias in the genes that are being reported, refer to the large bias in the gwas populations that are used -> heavily favour the europeans
- genes that are involved might not have been genome-wide significant and therefore get missed
- the technicalities of annotating, the reported gene might not be the actual gene, what happens when the snp is intergenic (closest gene?),  

    
One of the limitations of using the gwas catalog is for gene associations
- bias of the populations
- genes that are involved might not be genome wide significant therefore missed

the time frames for the statistics has a wide range, many of the differences that are being looked at are in the < 10kya time frame, so therefore some of these methods might not be entirely appropriate but also the nature of the chip data means that some of the newer methods aren't necessarily applicable. especially because the chip data is missing the very low frequency variants due to the bias of chip data having both an ascertainment bias

if it isn't selection, then what could explain the differences in the disease affections?

- bottlenecking
- migration
- admixture
- "drift"

paper about how demography history needs to be taken into account

also paper about how talking about selection on pleotropic traits is very difficult

talk about the distributions and that with the new push for sequence data, esp in poly that the it should be repeated

## Future directions

Future work in this area would include the generation of a high quality set of Polynesian whole genome sequences that could be used to remove the ascertainment bias that the \gls{snp} arrays have. High quality sequences could also be used to supplement the reference haplotype panels that are currently available to include haplotypes that are specific to Polynesian populations which could then be used improved haplotype phasing, and for imputation of Polynesian specific variants.

Further study could also look beyond positive selection and into other selection types such as background selection which has been XXX [\@ref background selection in humans paper].