# Conclusions and Discussions


Chapter 3 showed:

- there was limited evidence of selection in metabolic disease
- For gout there was limited evidence of selection, where there was it was not in the main effect loci for gout (slc2a9/abgc2) or the urate transporters
- diabetes
- obesity
- metabolic syndrome
- kidney disease

-

Chapter 4 showed

Chapter 5 showed


Overall



summary of regions that were selected

was gout/urate and other diseases under selection?




- chip data
- imputation
- phasing
    - polynesians aren't in the 1kgp reference haplotypes
- filters
    - QC filters
    - HWE
    - MAF
    
use of the gwas catalog, while extremely useful in terms of automatic defining of gene lists, does have a few drawbacks

- there is a potential bias in the genes that are being reported, refer to the large bias in the gwas populations that are used -> heavily favour the europeans
- genes that are involved might not have been genome-wide significant and therefore get missed
- the techinicalities of annotating, the reported gene might not be the actual gene, what happens when the snp is intergenic (closest gene?),  

    
One of the limitations of using the gwas catalog is for gene associations
- bias of the populations
- genes that are involved might not be genome wide significant therefore missed

the time frames for the statistics has a wide range, many of the differences that are being looked at are in the < 10kya time frame, so therefore some of these methods might not be entirely appropriate but also the nature of the chip data means that some of the newer methods aren't necessarily appliable. especially because the chip data is missing the very low frequency variants due to the bias of chip data hvaing both an ascertainment bias

if it isn't selection, then what could explain the differences in the disease affections?

- bottlenecking
- migration
- admixture
- "drift"

talk about the distributions and that with the new push for sequence data, esp in poly that the it should be repeated