# (APPENDIX) Appendix {-}

```{r, messages = FALSE, warning=FALSE, include=FALSE}
#knitr::opts_knit$set(root.dir = '/home/murraycadzow/data/NZ_coreExome_1kgp/' )
library(tidyverse)
library(formattable)
```
# Appendix A
Unimputed CoreExome analysis

## QC of genotypes



## Phasing

The 

### SelectionTools Pipeline
The subject lists were created using selectionAnalysis/coreExome_subject_selection.Rmd
once the subject lists were created they were extracted from 

```{r, eval=FALSE, tidy=TRUE, tidy.opts=list(width.cutoff=60), engine='bash'}
# on biocvisg

# this command 
$DIR =~/Murray/Bioinformatics/working_dir/StrandFixed/CoreExome/QC_Batch2_NZ/vcf/Phased/
parallel 'bcftools view \
-r {2} -S coreExome_{1}_selection.txt \
-o {1}_chr{2}_coreExome.vcf.gz \
-O z {3}/all_chr_phased_strandaligned.filt_refallele_maf0.01_hwe_checked_ExRef.vcf.gz \
' ::: nzcau nzm cim ton sam ::: $(seq 1 22) ::: $DIR
```

To match with the markers from 1000 Genomes Phase 3
```{bash, eval = FALSE, tidy=TRUE, tidy.opts=list(width.cutoff=60)}
# on biocvisg
bcftools view \
-O v \
-m 2 \
-M 2 \
-H \
-v snps all_chr_phased_strandaligned.filt_refallele_maf0.01_hwe_checked_ExRef.vcf.gz \
-o - | cut -f1,2 | grep -v "#" > coreExome_markers.txt 

#1kg marker lists
cd ~/Murray/Bioinformatics/Reference_Files/VCF/1000Genomes_vcf_files/Phase3_Sept2014
parallel 'bcftools view \
-O v \
-m 2 \
-M 2 \
-H \
-v snps ALL.chr{}.phase3_shapeit2_mvncall_integrated_v5.20130502.genotypes.vcf.gz -o - \
| cut -f1,2 | grep -v "#" > chr{}_markers.txt ' ::: $(seq 1 22)
```

the marker list was then read into R and merged with the the marker lists made for each chromosome from the Phase3 VCFs
```{r, eval = FALSE}
# on biocmerrimanlab
ce_markers <- read.table('/media/xsan/scratch/merrimanlab/murray/working_dir/StrandFixed/CoreExome/QC_Batch2_NZ/vcf/Phased/coreExome_markers.txt', header=FALSE)

kg_markers <- data.frame()
for(i in 1:22){
  tmp <- read.table(file= paste0('/media/xsan/archive/merrimanlab/reference_files/VCF/1000Genomes_vcf_files/Phase3_Sept2014/chr',i,'_markers.txt'), header=FALSE)
  tmp <-  tmp[ tmp[,2] %in% ce_markers[,2],]
  kg_markers <- rbind(kg_markers, tmp)
  print(i)
}
ce <- merge(ce_markers, kg_markers, by = c("V1","V2"), all = FALSE)
write.table(ce, file = '/media/xsan/archive/merrimanlab/reference_files/VCF/1000Genomes_vcf_files/Phase3_Sept2014/coreExome_1kgp_markers.txt', quote=FALSE, col.names=FALSE, row.names = FALSE, sep = "\t")
```

once the intersection of the marker lists were made the markers were pull out of both the core Exome vcf and the 1000 Genomes Project vcf
```{bash, eval = FALSE}
# on biocvisg
#wd = /Volumes/BiocArchive/merrimanlab/reference_files/VCF/1000Genomes_vcf_files/Phase3_Sept2014/
parallel 
  'bcftools 
    view 
      --force-samples 
      -s $(grep {1} integrated_call_samples_v3.20130502.ALL.panel | cut -f1 |tr "\n" ",") 
      -R /Volumes/BiocArchive/merrimanlab/reference_files/VCF/1000Genomes_vcf_files/Phase3_Sept2014/coreExome_1kgp_markers.txt 
      -O z 
      -m 2 
      -M 2 
      -v snps 
      -o ~/Murray/Bioinformatics/working_dir/coreExome_selection/NZ/Unimputed_1kg_marker_matched/{1}.chr{2}_biallelic_coreExome_markers.vcf.gz
      ALL.chr{2}.phase3_shapeit2_mvncall_integrated_v5.20130502.genotypes.vcf.gz ' ::: $(cat integrated_call_samples_v3.20130502.ALL.panel | cut -f2 | grep -v 'pop' | sort |uniq) ::: $(seq 1 22)

#wd = ~/Murray/Bioinformatics/working_dir/coreExome_selection/NZ
nzcau,nzm,cim,ton,sam
parallel 'bcftools view -R /Volumes/BiocArchive/merrimanlab/reference_files/VCF/1000Genomes_vcf_files/Phase3_Sept2014/coreExome_1kgp_markers.txt -O z -m 2 -M 2 -v snps -o Unimputed_1kg_marker_matched/{1}.chr{2}_biallelic_coreExome_markers.vcf.gz {1}_chr{2}_coreExome.vcf.gz ' ::: nzcau nzm cim ton sam ::: $(seq 1 22)

```

Once each population vcf file was created they were transferred to the NeSI PAN cluster where the selectionTools1.1 pipeline was run
```{bash selectionToolsScript, eval = FALSE}
#!/bin/bash
#SBATCH -J selection
#SBATCH -A uoo00008         # Project Account
#SBATCH --time=06:00:00     # Walltime
#SBATCH --mem-per-cpu=30240  # memory/cpu (in MB)
#SBATCH --cpus-per-task=1   # 12 OpenMP Threads
#SBATCH --array=1-22
#SBATCH --mail-user=murray.cadzow@otago.ac.nz
#SBATCH --mail-type=ALL

POP=$1
i=$SLURM_ARRAY_TASK_ID
DIR=$SLURM_SUBMIT_DIR
module load Python/3.5.0-intel-2015a
module load R/3.2.1-intel-2015a

srun tar -C $TMP_DIR -xzf $DIR/1kg_nz_coreExome.tar.gz ${POP}.chr${i}_biallelic_coreExome_markers.vcf.gz
cd $TMP_DIR/
srun gzip -dc ${POP}.chr${i}_biallelic_coreExome_markers.vcf.gz |grep -v 'contig' > ${POP}.chr${i}_biallelic_coreExome_markers_nocontig.vcf

srun python ~/.local/bin/selection_pipeline \
-i ${POP}.chr${i}_biallelic_coreExome_markers_nocontig.vcf \
--phased-vcf \
-c $i \
--config-file $DIR/unimputed_defaults_nesi_18-3-16.cfg \
--maf 0.01 \
--hwe 0.001 \
--TajimaD 30 \
--fay-Window-Width 30 \
--fay-Window-Jump 30 \
--ehh-window-size 10 \
--ehh-overlap 2 \
--big-gap 200 \
--small-gap 20 \
--small-gap-penalty 20 \
--population $POP \
--cores 1 \
--no-clean-up \
--no-ihs

module unload Python/3.5.0-intel-2015a 
module load Python/2.7.9-intel-2015a

srun python /home/murray.cadzow/uoo00008/MerrimanSelectionPipeline/selection_pipeline/haps_interpolate.py \
--haps results/${POP}_aachanged.haps --output ${POP}_genetic_dist.haps --genetic-map \
/home/murray.cadzow/uoo00008/MerrimanSelectionPipeline/referencefiles/genetic_maps/genetic_map_chr${i}_combined_b37.txt \
--physical-position-output ${POP}_genetic_dist.pos

srun  python /home/murray.cadzow/uoo00008/MerrimanSelectionPipeline/selection_pipeline/haps_to_selscan.py \
--haps ${POP}_genetic_dist.haps \
--pos ${POP}_genetic_dist.pos \
--output ${POP}_${axom}_${i}_selscan \
--chr ${i}

srun rm $TMP_DIR/${POP}_chr${i}.vcf
srun gzip $TMP_DIR/results/*vcforig
cd $TMP_DIR
srun tar -czf ${POP}_chr${i}.tar.gz *
mkdir $DIR/${POP}
srun cp $TMP_DIR/*.tar.gz $DIR/${POP}

cd $DIR/${POP}
sleep 10
sbatch --array=${i} ../run_selscan.sl ${POP} 
sbatch --array=${i} ../run_nsl_selscan.sl ${POP} 
```

```{bash run_selscan.sl, eval=FALSE}

```

```{bash run_nsl_selscan.sl, eval=FALSE}

```
### Results directory setup

```{r, engine = 'bash', eval = FALSE}
#!/bin/bash
results_dir=/Volumes/BiochemXsan/scratch/merrimanlab/murray/working_dir/coreExome_selection/NZ_coreExome_1kgp/data/dbload
input_dir=/Volumes/BiochemXsan/scratch/merrimanlab/murray/working_dir/coreExome_selection/NZ_coreExome_1kgp/data/nesi_results

# extract all the results files generated from the selection pipeline
for i in $(seq 1 22)
do
    mkdir -p $results_dir/{daf,fawh,fixed_vcf,fst,ihs,kaks,nsl,tajd,xpehh}/chr${i}
    echo chr${i}
    echo ihs
    parallel 'tar -C {3}/ihs/chr{1} -xzf {2} *out *log --wildcards' ::: ${i} ::: $(ls $input_dir/*/*chr${i}.ihs.tar.gz) ::: $results_dir
    echo nsl
    parallel 'tar -C {3}/nsl/chr{1} -xzf {2} *out *log --wildcards' ::: ${i} ::: $(ls $input_dir/*/*chr${i}.nsl.tar.gz) ::: $results_dir
    echo tajd
    parallel 'tar -C {3}/tajd/chr{1} -xzf {2} *taj_d --wildcards' ::: ${i} ::: $(ls $input_dir/*/*chr${i}.tar.gz) ::: $results_dir
    mv $results_dir/tajd/chr${i}/*/results/* $results_dir/tajd/chr${i}/
    rmdir $results_dir/tajd/chr${i}/*/results
    echo fawh
    parallel 'tar -C {3}/fawh/chr{1} -xzf {2} *faw --wildcards' ::: ${i} ::: $(ls $input_dir/*/*chr${i}.tar.gz) ::: $results_dir
    mv $results_dir/fawh/chr${i}/*/results/* $results_dir/fawh/chr${i}/
    rmdir $results_dir/fawh/chr${i}/*/results

    echo daf
    parallel 'tar -C {3}/daf/chr{1} -xzf {2} *aachanged.af --wildcards' ::: ${i} ::: $(ls $input_dir/*/*chr${i}.tar.gz) ::: $results_dir
    mv $results_dir/daf/chr${i}/*/*.af $results_dir/daf/chr${i}/
    rename _aachanged.af ${i}_aachanged.af $results_dir/daf/chr${i}/*
    echo fixed_vcf
    parallel 'tar -C {3}/fixed_vcf/chr{1} -xzf {2} *fixed.vcf --wildcards' ::: ${i} ::: $(ls $input_dir/*/*chr${i}.tar.gz) ::: $results_dir
    mv $results_dir/fixed_vcf/chr${i}/*/*vcf $results_dir/fixed_vcf/chr${i}/
    echo xpehh
    parallel 'tar -C {3}/xpehh/chr{1} -xzf {2} *out *log --wildcards' ::: ${i} ::: $(ls $input_dir/xpehh/*/*chr${i}_xpehh.tar.gz) ::: $results_dir
    
done

# calc the number of files for each population
for pop in $(basename -a $(find $input_dir -type d | grep -v "xpehh" |grep [a-zA-Z] ) |grep -v "results")
do
    echo ------ $pop -------
    echo ihs $(ls $results_dir/ihs/chr*/${pop}*out | wc -l)

    echo nsl $(ls $results_dir/nsl/chr*/${pop}*out | wc -l)

    echo fawh $(ls $results_dir/fawh/chr*/${pop}*faw | wc -l)

    echo tajd $(ls $results_dir/tajd/chr*/${pop}*taj_d | wc -l)

    echo daf $(ls $results_dir/daf/chr*/${pop}*af | wc -l)

    echo fixed_vcf $(ls $results_dir/fixed_vcf/chr*/${pop}*vcf | wc -l)

    echo xpehh $(ls $results_dir/xpehh/chr*/${pop}*out | wc -l)
    echo -------------------
    echo
    echo
done

# clean up some of the extraction
cd $results_dir/fixed_vcf/
for pop in $(basename -a $(find $input_dir -type d | grep -v "xpehh" |grep [a-zA-Z] ) |grep -v "results")
do
    rmdir chr*/$pop
done

# combine vcfs, run snpEff and create panel files
parallel 'bgzip {} && tabix -f -p vcf {}.gz ' ::: $(ls */*vcf)
mkdir combined
parallel -j 6 'bcftools merge $(ls chr{}/*gz| grep "AMR\|AFR\|EUR\|EAS\|SAS\|POL") | bgzip -c > combined/combined_super_chr{}.vcf.gz' ::: $(seq 1 22)
parallel -j 6 'bcftools merge $(ls chr{}/*gz | grep -v "AMR\|AFR\|EUR\|EAS\|SAS\|POL\|EPN\|WPN") | bgzip -c > combined/combined_chr{}.vcf.gz' ::: $(seq 1 22)
parallel -j 16 'java -jar ~/Murray/src/snpEff4.2/snpEff/snpEff.jar -c ~/Murray/src/snpEff4.2/snpEff/snpEff.config -v GRCh37.75 -strict -stats chr{2}/{1}{2}.html chr{2}/{1}{2}_fixed.vcf.gz | bgzip -c > chr{2}/{1}{2}_fixed_ann.vcf.gz ' ::: $(ls chr22/*gz | sed 's/chr22\///g' |cut -d'2' -f1) ::: $(seq 1 22)
for pop in $(basename -a -s 1_fixed.vcf.gz chr1/*fixed.vcf.gz); do zcat chr1/${pop}1_fixed.vcf.gz | head -100 | grep '^#CHR' |cut -f10- | tr '\t' '\n' | awk '{print $1"\t"$1}' > ${pop}.panel ; done

# calculate Fst
parallel -j 10 '
for p1 in POL
do
    for p2 in AFR AMR EUR EAS SAS
    do
        vcftools --gzvcf combined/combined_super_chr{}.vcf.gz --weir-fst-pop ${p1}.panel --weir-fst-pop ${p2}.panel  --fst-window-size 1000000 --fst-window-step 1000 --out ../fst/chr{}/${p1}_${p2}_{}
    done
done
' ::: $(seq 1 22)

parallel -j 10 '
for p1 in CIM NZC NZM NAD TON SAM EPN WPN
do      
    for p2 in $(basename -a -s .panel *.panel | grep -v "AMR\|AFR\|EUR\|EAS\|SAS\|POL")
    do
       if [[ $p1 != $p2 ]]
           then
               vcftools --gzvcf combined/combined_chr{}.vcf.gz --weir-fst-pop ${p1}.panel --weir-fst-pop ${p2}.panel  --fst-window-size 1000000 --fst-window-step 1000 --out ../fst/chr{}/${p1}_${p2}_{}
       fi
    done
done
' ::: $(seq 1 22)

# generate kaks files
parallel 'zgrep -i "^#\|synonymous\|missense" {} > $(echo {}| cut -d'/' -f1)/$(basename -s .vcf.gz {}).txt ; python3.4 /Volumes/BiocArchive/staff_groups/merrimanlab/Merriman_Documents/Murray/SelectionPipeline/selectionTools-dev/extrascripts/kaks.py $(echo {}| cut -d'/' -f1)/$(basename -s .vcf.gz {}).txt > $(echo {} | cut -d'/' -f1)/$(basename -s .vcf.gz {}).kaks ' ::: $(ls chr*/*ann.vcf.gz)

# move and rename the kaks files
cd $results_dir/
for i in $(seq 1 22);do mv fixed_vcf/chr${i}/*.kaks kaks/chr${i}/ ; done
for i in $(seq 1 22);do rename ${i}_fixed_ann ${i} $(ls kaks/chr${i}/*) ; done

# normalise the ihs files
cd $results_dir/ihs
for pop in $(ls chr1/*out | cut -d'/' -f2 | cut -d'_' -f1 | sort -u)
do 
    ~/Murray/src/selscan/bin/linux/norm --ihs --files chr*/${pop}_*.out --crit-percent 0.99 --log ${pop}.log
    ~/Murray/src/selscan/bin/linux/norm --ihs --files chr*/gap1mb/${pop}_*.out --crit-percent 0.99 --log ${pop}_gap1mb.log
done

# rename the ihs files to remove the underscore
for i in $(seq 1 22)
do 
    rename _${i} ${i} chr${i}/*
    rename _${i} ${i} chr${i}/gap1mb/*
    mkdir -p norm/chr${i}/gap1mb
    mv chr${i}/*norm norm/chr${i}/
    mv chr${i}/gap1mb/*norm norm/chr${i}/gap1mb/
done

# normalise the nsl files
cd $results_dir/nsl
for pop in $(ls chr1/*out | cut -d'/' -f2 | cut -d'_' -f1 | sort -u) 
do 
    ~/Murray/src/selscan/bin/linux/norm --ihs --files chr*/${pop}_*.out --crit-percent 0.99 --log ${pop}.log
    ~/Murray/src/selscan/bin/linux/norm --ihs --files chr*/gap1mb/${pop}_*.out --crit-percent 0.99 --log ${pop}_gap1mb.log
done

# rename and move the nsl files
for i in $(seq 1 22)
do 
    rename _${i} ${i} chr${i}/*
    mkdir -p norm/chr${i}
    mv chr${i}/*norm norm/chr${i}/
    mv chr${i}/gap1mb/*norm norm/chr${i}/gap1mb/
done


# normalise xpehh files
cd $results_dir/xpehh
for p1 in POL CIM NZC NZM NAD TON SAM EPN WPN;
do
    for p2 in $(ls chr1/${p1}*.out | cut -d'_' -f2)
    do
        if [[ $p1 != $p2 ]]
        then
            ~/Murray/src/selscan/bin/linux/norm --xpehh --files chr*/${p1}_${p2}*.out --crit-percent 0.99 --log ${p1}_${p2}.log
            ~/Murray/src/selscan/bin/linux/norm --xpehh --files chr*/gap1mb/${p1}_${p2}*.out --crit-percent 0.99 --log ${p1}_${p2}_gap1mb.log
        fi
    done
done

# move normalised xpehh files
for i in $(seq 1 22)
do 
    mkdir -p norm/chr${i}/gap1mb
    mv chr${i}/*norm norm/chr${i}/
    mv chr${i}/gap1mb/*norm norm/chr${i}/gap1mb/
done
```

### Database creation and loading



## Tables

This is a test sentence to find out if table \@ref(tab:meanSegSites) is referenceable and a really long table is \@ref(tab:selStatSummary)
<!--

-->
```{r}
# This is created by coreExome_30kb_windows_intra.Rmd
load("~/data/NZ_coreExome_1kgp/chr_summary_popgenome.9-6-2017.RData")
pops <- list()
panel <- read.delim('/home/murraycadzow/data/NZ_coreExome_1kgp/nz_1kgp.panel', stringsAsFactors = FALSE)
for(p in unique(panel$pop)){
  pops[[p]] <- panel %>% filter(pop == p) %>%  .[["sample"]]
}

create_sel_stat_table <- function(sel_stat, cap){
  knitr::kable(chr_summary %>% filter(stat == sel_stat) %>% ungroup() %>%  select(pop, chrom, mean) %>% spread(.,chrom, mean) %>% left_join(., panel %>% select(pop, super_pop) %>% distinct(), "pop") %>% select(pop, super_pop, 2:23) %>% arrange(super_pop, pop), align = c('l','c',rep('r', 22)), col.names = c("Population", "Super Population", paste0('Chr',1:22)), caption = cap, digits =  3)
}
```

```{r, results = 'asis'}
threedp <- function(num){
  sprintf('%.3f', num)
}
pander::pandoc.table(chr_summary %>% mutate(mean = sapply(mean, threedp), sd = sapply(sd, threedp), lower_1 = sapply(lower_1, threedp), upper_99 = sapply(upper_99, threedp), median = sapply(median, threedp), min = sapply(min,threedp), max = sapply(max, threedp)) %>% arrange(chrom,pop,stat) %>% data.frame(),  caption = '(\\#tab:selStatSummary) Summary of the selection test statistics for all populations and all autosomes', split.tables = 80, style = 'multiline', justify = 'lccrrrrrrr')
```



```{r meanSegSitesTable, results='asis', echo = FALSE}
create_sel_stat_table("n.segregating.sites", "Mean Number of Segregating Sites")
```

Tajimas D Table
```{r meanTDtable, results='asis', echo = FALSE}
create_sel_stat_table("Tajima.D", "Mean Tajima's D")
```

Fay and Wu's H Table
```{r meanFWHtable, results='asis', echo = FALSE}
create_sel_stat_table("Fay.Wu.H", "Mean Fay and Wu's H")
```

Fu and Li D Table
```{r meanFLDtable, results='asis', echo = FALSE}
create_sel_stat_table("Fu.Li.D","Mean Fu and Li D")
```

Fu and Li F Table
```{r, results='asis', echo = FALSE}
create_sel_stat_table("Fu.Li.F","Mean Fu and Li F")
```

Zeng E Table
```{r, results='asis', echo = FALSE}
create_sel_stat_table("Zeng.E", "Mean Zeng E")
```