# Appendix D

SelectionTools Pipeline NeSI Scripts

## unimputed_selection_pipeline.sl

**/home/murray.cadzow/uoo00008/NZ_1KGP_unimputed/unimputed_selection_pipeline.sl**
```
#!/bin/bash
#SBATCH -J selection
#SBATCH -A uoo00008         # Project Account
#SBATCH --time=00:30:00     # Walltime
#SBATCH --mem-per-cpu=2048  # memory/cpu (in MB)
#SBATCH --cpus-per-task=1   # 12 OpenMP Threads
#SBATCH --array=1-22
#SBATCH --mail-user=murray.cadzow@otago.ac.nz
#SBATCH --mail-type=FAIL

POP=$1
i=$SLURM_ARRAY_TASK_ID
DIR=$SLURM_SUBMIT_DIR
module load Python/3.5.0-intel-2015a
module load R/3.2.1-intel-2015a
mkdir $TMP_DIR/${POP}
#srun tar -C $TMP_DIR/${POP}/ -xzf $DIR/${POP}.tar.gz ${POP}.chr${i}_biallelic_coreExome_markers.vcf
srun gzip -dc $DIR/${POP}.chr${i}.phased.vcf.gz > $TMP_DIR/${POP}/${POP}.chr${i}.phased.vcf
cd $TMP_DIR/${POP}/
  #srun cat ${POP}.chr${i}_biallelic_coreExome_markers.vcf | grep -v '^##contig' > ${POP}_chr${i}_coreExome.vcf
srun python ~/.local/bin/selection_pipeline \
-i ${POP}.chr${i}.phased.vcf \
--phased-vcf \
-c $i \
--config-file /home/murray.cadzow/uoo00008/NZ_1KGP_unimputed/unimputed_defaults_nesi_18-3-16.cfg \
--maf 0.01 \
--hwe 0.000001 \
--TajimaD 30 \
--fay-Window-Width 30 \
--fay-Window-Jump 30 \
--ehh-window-size 10 \
--ehh-overlap 2 \
--big-gap 200 \
--small-gap 20 \
--small-gap-penalty 20 \
--population $POP \
--cores 1 \
--no-clean-up \
--no-ihs

module unload Python/3.5.0-intel-2015a 
module load Python/2.7.9-intel-2015a

srun python /home/murray.cadzow/uoo00008/MerrimanSelectionPipeline/selection_pipeline/haps_interpolate.py \
--haps results/${POP}_aachanged.haps \
--output ${POP}_genetic_dist.haps \
--genetic-map /home/murray.cadzow/uoo00008/MerrimanSelectionPipeline/referencefiles/genetic_maps/genetic_map_chr${i}_combined_b37.txt \
--physical-position-output ${POP}_genetic_dist.pos

srun python /home/murray.cadzow/uoo00008/MerrimanSelectionPipeline/selection_pipeline/haps_to_selscan.py \
--haps ${POP}_genetic_dist.haps \
--pos ${POP}_genetic_dist.pos \
--output ${POP}_${i}_selscan \
--chr ${i}

srun gzip $TMP_DIR/${POP}/results/*vcforig
cd $TMP_DIR
srun tar -czf ${POP}.chr${i}.tar.gz *
mkdir -p  /home/murray.cadzow/uoo00008/NZ_1KGP_unimputed/Indiv_pops_results/$POP
srun cp $TMP_DIR/*.tar.gz /home/murray.cadzow/uoo00008/NZ_1KGP_unimputed/Indiv_pops_results/${POP}/
```

### unimputed_defaults_nesi-18-3-16.cfg

**/home/murray.cadzow/uoo00008/NZ_1KGP_unimputed/unimputed_defaults_nesi_18-3-16.cfg**
```
#
# Defaults config file for VCF process
#
# If the executables are on your path 
# just the executable name is required.
#
# ? is the willcard flag for the prefix options



[system]
cores_avaliable = 1
# Library settings do not change, the library folder are appended to the path when running the program#
[environment]
LD_LIBRARY_PATH=/home/murray.cadzow/uoo00008/MerrimanSelectionPipeline/lib
PERL5LIB=/home/murray.cadzow/uoo00008/MerrimanSelectionPipeline/lib/perl5
[selection_pipeline]
selection_pipeline_executable = ~/.local/bin/selection_pipeline
[vcftools]
vcf_tools_executable = /home/murray.cadzow/uoo00008/MerrimanSelectionPipeline/bin/vcftools
vcf_subset_executable = /home/murray.cadzow/uoo00008/MerrimanSelectionPipeline/bin/vcf-subset
vcf_merge_executable = /home/murray.cadzow/uoo00008/MerrimanSelectionPipeline/bin/vcf-merge
vcf_concat_executable = /home/murray.cadzow/uoo00008/MerrimanSelectionPipeline/bin/vcf-concat
extra_args= 
[genetic_map]
genetic_map_dir= /home/murray.cadzow/uoo00008/MerrimanSelectionPipeline/referencefiles/genetic_maps
genetic_map_prefix=genetic_map_chr?_combined_b37.txt
[shapeit]
shapeit_executable= /home/murray.cadzow/uoo00008/MerrimanSelectionPipeline/bin/shapeit
extra_args =
[impute2]
impute_executable = /home/murray.cadzow/uoo00008/MerrimanSelectionPipeline/bin/impute2
impute_map_dir= /home/murray.cadzow/uoo00008/MerrimanSelectionPipeline/referencefiles/impute_ref
impute_reference_dir= /home/murray.cadzow/uoo00008/MerrimanSelectionPipeline/referencefiles/impute_ref
impute_map_prefix=genetic_map_chr?_combined_b37.txt
impute_reference_prefix=ALL.chr?.integrated_phase1_v3.20101123.snps_indels_svs.genotypes.nomono
extra_args = 
[plink]
plink_executable =/home/murray.cadzow/uoo00008/MerrimanSelectionPipeline/bin/plink
extra_args = 
[Rscript]
rscript_executable = Rscript
indel_filter = /home/murray.cadzow/uoo00008/MerrimanSelectionPipeline/corescripts/haps_indel_and_maf_filter.R
generate_rsb = /home/murray.cadzow/uoo00008/MerrimanSelectionPipeline/corescripts/generate_rsb.R
extra_args=
[haps_scripts]
haps_to_hapmap_script= /home/murray.cadzow/.local/bin/haps_to_hapmap
haps_filter_script = /home/murray.cadzow/.local/bin/haps_filters
haps_interpolate_script = /home/murray.cadzow/.local/bin/haps_interpolate
haps_to_selscan_script = /home/murray.cadzow/.local/bin/haps_to_selscan
[ancestral_allele]
split_by_chromosome = True
# not used unless split_by_chromosome is set to False
ancestral_fasta_header_regex = 
# not used unless split_by_chromosome is set to False
ancestral_fasta_file =
ancestral_allele_script= /home/murray.cadzow/.local/bin/ancestral_annotation

ancestral_fasta_dir=/home/murray.cadzow/uoo00008/MerrimanSelectionPipeline/referencefiles/ancestral_ref/homo_sapiens_ancestor_GRCh37_e65
ancestral_prefix=homo_sapiens_ancestor_?.fa
[qctool]
qctool_executable=/home/murray.cadzow/uoo00008/MerrimanSelectionPipeline/bin/qctool
[selscan]
selscan_executable=/home/murray.cadzow/uoo00008/MerrimanSelectionPipeline/bin/selscan
[multicore_ihh]
multicore_ihh = /home/murray.cadzow/uoo00008/MerrimanSelectionPipeline/corescripts/multicore_iHH.R
[variscan]
variscan_executable = /home/murray.cadzow/uoo00008/MerrimanSelectionPipeline/bin/variscan
[java]
java_executable = /usr/bin/java
[beagle]
beagle_jar = /home/murray.cadzow/uoo00008/MerrimanSelectionPipeline/bin/beagle.jar
vm_size = 4g
```


## run_selscan.sl

**/home/murray.cadzow/uoo00008/NZ_1KGP_unimputed/run_selscan.sl**
  ```
#!/bin/bash
#SBATCH -J selscan_array
#SBATCH -A uoo00008         # Project Account
#SBATCH --time=05:59:00     # Walltime
#SBATCH --mem=4096  # memory/node (in MB)
#SBATCH --cpus-per-task=4   # 10 OpenMP Threads
#SBATCH --array=1-22


POP=$1
i=$SLURM_ARRAY_TASK_ID
echo $POP chr $i

DIR=$SLURM_SUBMIT_DIR


srun tar -C $TMP_DIR -xzf ${POP}.chr${i}.tar.gz ${POP}/${POP}_${i}_selscan* 
cd $TMP_DIR/$POP && \
srun ~/uoo00008/selscan/src/selscan \
  --ihs \
  --hap ${POP}_${i}_selscan.selscanhaps \
  --map  ${POP}_${i}_selscan.selscanmap \
  --ihs-detail \
  --threads 4 \
  --out ${TMP_DIR}/${POP}/${POP}_${i}
  
mkdir $TMP_DIR/$POP/gap1mb
srun ~/uoo00008/selscan/src/selscan \
  --ihs \
  --hap ${POP}_${i}_selscan.selscanhaps \
  --map  ${POP}_${i}_selscan.selscanmap \
  --ihs-detail \
  --threads 4 \
  --max-gap 1000000 \
  --out ${TMP_DIR}/${POP}/gap1mb/${POP}_${i}

srun tar -czf ${POP}.chr${i}.ihs.tar.gz *log *.out gap1mb
srun cp ${POP}.chr${i}.ihs.tar.gz $DIR/
```

### run_nsl_selscan.sl

**/home/murray.cadzow/uoo00008/NZ_1KGP_unimputed/run_nsl_selscan.sl**
  
  ```
#!/bin/bash
#SBATCH -J selscan_array
#SBATCH -A uoo00008         # Project Account
#SBATCH --time=05:59:00     # Walltime
#SBATCH --mem=2048  # memory/node (in MB)
#SBATCH --cpus-per-task=4   # 10 OpenMP Threads
#SBATCH --array=1-22


POP=$1
i=$SLURM_ARRAY_TASK_ID
echo $POP chr $i

DIR=$SLURM_SUBMIT_DIR

srun tar -C $TMP_DIR -xzf ${POP}.chr${i}.tar.gz ${POP}/${POP}_${i}_selscan* 
cd $TMP_DIR/$POP && \
srun ~/uoo00008/selscan/src/selscan \
  --nsl \
  --hap ${POP}_${i}_selscan.selscanhaps \
  --map  ${POP}_${i}_selscan.selscanmap \
  --threads 4 \
  --out ${TMP_DIR}/${POP}/${POP}_${i}
  
mkdir $TMP_DIR/$POP/gap1mb
srun ~/uoo00008/selscan/src/selscan \
  --nsl \
  --hap ${POP}_${i}_selscan.selscanhaps \
  --map  ${POP}_${i}_selscan.selscanmap \
  --threads 4 \
  --max-gap 1000000 \
  --out ${TMP_DIR}/${POP}/gap1mb/${POP}_${i}
  
srun tar -czf ${POP}.chr${i}.nsl.tar.gz *log *.out gap1mb
srun cp ${POP}.chr${i}.nsl.tar.gz $DIR
```




### run_xpehh.sl

**/home/murray.cadzow/uoo00008/NZ_1KGP_unimputed/run_xpehh.sl**
```
#!/bin/bash
#SBATCH -J selscan_array
#SBATCH -A uoo00008         # Project Account
#SBATCH --time=12:00:00     # Walltime
#SBATCH --mem=2048  # memory/node (in MB)
#SBATCH --cpus-per-task=4   # 10 OpenMP Threads
#SBATCH --array=1-22
#SBATCH -C sb

POP1=$1
POP2=$2
i=$SLURM_ARRAY_TASK_ID
#echo $POP chr $i

DIR=$SLURM_SUBMIT_DIR

module load Python/2.7.8-goolf-1.5.14


srun tar -C $TMP_DIR -xzf $DIR/${POP1}/${POP1}.chr${i}.tar.gz $POP1/${POP1}_${i}_selscan.selscanhaps $POP1/${POP1}_${i}_selscan.selscanmap

#need selscan file for POP
srun tar -C $TMP_DIR -xzf $DIR/${POP2}/${POP2}.chr${i}.tar.gz $POP2/${POP2}_${i}_selscan.selscanhaps $POP2/${POP2}_${i}_selscan.selscanmap
#merge selscan files
mkdir $TMP_DIR/${POP1}_${POP2}

cd $TMP_DIR/${POP1}_${POP2} && srun python /home/murray.cadzow/uoo00008/MerrimanSelectionPipeline/selection_pipeline/selscan_to_selscan_xpehh.py \
--pop1-prefix $TMP_DIR/$POP1/${POP1}_${i}_selscan \
--pop1-name ${POP1} \
--pop2-prefix $TMP_DIR/$POP2/${POP2}_${i}_selscan \
--pop2-name ${POP2} \
-c ${i} \
--out ./
  
  ls $TMP_DIR/*
  
  #srun tar -C $TMP_DIR -xzf ${POP1}_${POP2}.tar.gz ${POP1}_${POP2}/*${i}.xpehh* ${POP1}_${POP2}/*${i}.*.xp*
  cd $TMP_DIR && srun ~/uoo00008/selscan/src/selscan \
--xpehh \
--hap ${POP1}_${POP2}/${POP1}_${i}.matches_${POP2}.xpehh_selscanhaps \
--map  ${POP1}_${POP2}/${POP1}_${POP2}_${i}.xpehh_selscanmap \
--ref ${POP1}_${POP2}/${POP2}_${i}.matches_${POP1}.xp_ehh_selscanhaps  \
--threads 4 \
--out ${TMP_DIR}/${POP1}_${POP2}_${i}

mkdir $TMP_DIR/gap1mb

srun ~/uoo00008/selscan/src/selscan \
--xpehh \
--hap ${POP1}_${POP2}/${POP1}_${i}.matches_${POP2}.xpehh_selscanhaps \
--map  ${POP1}_${POP2}/${POP1}_${POP2}_${i}.xpehh_selscanmap \
--ref ${POP1}_${POP2}/${POP2}_${i}.matches_${POP1}.xp_ehh_selscanhaps  \
--threads 4 \
--max-gap 1000000 \
--out ${TMP_DIR}/gap1mb/${POP1}_${POP2}_${i}

cd $TMP_DIR
srun tar -czf ${POP1}_${POP2}_chr${i}_xpehh.tar.gz ${POP1}_${POP2}* gap1mb
mkdir -p $DIR/xpehh/${POP1}_${POP2}
srun cp ${POP1}_${POP2}_chr${i}_xpehh.tar.gz $DIR/xpehh/${POP1}_${POP2}/
```