```{r, messages = FALSE, warning=FALSE, include=FALSE}
#knitr::opts_knit$set(root.dir = '/home/murraycadzow/data/NZ_coreExome_1kgp/' )
knitr::opts_chunk$set(echo = FALSE)
library(tidyverse)
library(formattable)
library(knitr)
library(kableExtra)
library(phdFunctions)
#select <- dplyr::select

poly_pop <- c("CIM","NZM","SAM","TON")
```
# Appendix E

## Electronic Supplements


```{r}
sig_ihs_val <- readRDS('~/data/NZ_coreExome_1kgp/haplotype/sig_ihs_values_with_genes_14-11-2017.RDS') %>% filter(abs(statvalue) > 2.6 )
sig_nsl_val <- readRDS('~/data/NZ_coreExome_1kgp/haplotype/sig_nsl_values_with_genes_14-11-2017.RDS') %>% filter(abs(statvalue) > 2.6 )
markers <- read.table('~/data/NZ_coreExome_1kgp/nz_1kg_markers.txt', stringsAsFactors = FALSE, header = FALSE)
names(markers) <- c("chrom","chrom_start","marker","ref","alt") 
```

### Hider Regions

```{r, results='hold'}
hider_regions_found <- readRDS('data/03-hider-regions-found.RDS')
hider_regions_found %>% mutate_at(4:NCOL(.), funs(replace_na(., '-'))) %>% mutate(chr = as.numeric(str_sub(chrom, 4)))%>% arrange(chr, chrom_start) %>% select(-chr) %>% kable(., col.names = c("Chrom","Start", "End", "\\gls{td} Region", "\\gls{td} Present","\\gls{fwh} Region", "\\gls{fwh} Present", "\\gls{flf} Region", "\\gls{flf} Present", '\\gls{ihs} Region', '\\gls{ihs} Present'), format = 'latex', booktabs = TRUE, longtable = TRUE, linesep ='', caption = '(\\#tab:hiderRegionsFound)', escape = FALSE) %>% column_spec(4:11, width = "5em") %>% kable_styling(latex_options = c('repeat_header'),font_size = 8) %>% landscape()
```
```{r ramos}
ramos <- list()
ramos[["ceu"]] <- c("ALMS1P", "DGUOK", "PARD3B", "KCNIP4", "HLA", "PRKCH", "HIF1A", "IRF4", "TECRL", "CTNNA3", "WWOX", "RABEP1", "NUP88", "COMMD10", "SEMA6A", "ALDH7A1", "SNRPC", "UHRF1BP1", "OSBPL8", "GRM5", "NTNG2", "SETX", "GRHL2")
ramos[["yri"]] <- c("TNFRSF14", "MMEL1", "MAGI3", "PTPN22", "TLR5", "XCR1", "CCR3", "CCDC66", "ARHGEF3", "SLC2A9", "KCNIP4", "ARHGAP31", "CD80", "IL2", "IL21", "TPR3", "HLA", "CCDC167", "MIR4462", "PRSS54", "PRDM1", "ATG5", "TRAF3IP2", "PAK7PRKCH", "HIF1A", "HIP1")
ramos[["asi"]] <- c("NCF2", "RGL1", "PELI1", "HLA", "BCAS3", "NACA2", "CNTN6", "BTLA", "MRPS22", "COMMD10", "SEMA6A", "GTF2I", "KIAA0391", "FAM171A1", "KDM4C", "XKR6", "BLK", "LSMEM1", "NPM1P14")
ramos[["unknown"]] <- c("IL23R", "FCGR2B", "TNFSF4", "CR1", "ANTRX2", "TNIP1", "PTTG1", "VARS", "LSM2", "IKZF1", "IRF8", "TYK2", "CLEC16A", "CIITA", "ITGAM", "ITGAX", "SH2B3", "NAA25", "CD5", "PTGER4")
```

```{r, eval = FALSE}
(sig_ihs_val %>% filter(pop == 'CEU', genename %in% ramos[['ceu']])%>% mutate(pop = 'CEU', statname = case_when(statid == 20 ~ "nSL", statid == 26 ~ "iHS"), chrom = paste0('chr',chrom)) %>% select(SYMBOL = genename, chrom, statname) %>% distinct()) 

```



```{r, results = 'asis', eval = FALSE}
# modify so that the full table comes out
pander::pandoc.table(chr_summary %>% mutate(mean = sapply(mean, three_dp), sd = sapply(sd, three_dp), lower_1 = sapply(lower_1, three_dp), upper_99 = sapply(upper_99, three_dp), median = sapply(median, three_dp), min = sapply(min,three_dp), max = sapply(max, three_dp)) %>% arrange(chrom,pop,stat) %>% head() %>% data.frame(),  caption = '(\\#tab:selStatSummary) Summary of the selection test statistics for all populations and all autosomes', split.tables = 80, style = 'multiline', justify = 'lccrrrrrrr')
```

```{r meanSegSitesTable, results='asis', echo = FALSE, eval=FALSE}
create_sel_stat_table("n.segregating.sites", "Mean Number of Segregating Sites")
```

<!-- Tajimas D Table -->
```{r meanTDtable, results='asis', echo = FALSE, eval = FALSE}
create_sel_stat_table("Tajima.D", "Mean Tajima's D")
```

<!-- Fay and Wu's H Table -->
```{r meanFWHtable, results='asis', echo = FALSE, eval = FALSE}
create_sel_stat_table("Fay.Wu.H", "Mean Fay and Wu's H")
```

<!-- Fu and Li F Table -->
```{r, results='asis', echo = FALSE, eval = FALSE}
create_sel_stat_table("Fu.Li.F","Mean Fu and Li F")
```

<!--Zeng E Table -->
```{r, results='asis', echo = FALSE, eval = FALSE}
create_sel_stat_table("Zeng.E", "Mean Zeng E")
```

```{r, echo = FALSE, results='hold'}
# This is created by coreExome_100kb_windows_intra_filtered.Rmd
chr_summary <- readRDS("~/data/NZ_coreExome_1kgp/100kbWindow_intra/filtered/100kbwindows_chr_summary_popgenome_filtered_3ns_resized.31-8-2017.RDS") %>% filter(stat != "Fu.Li.D")
pops <- list()
panel <- read.delim('/home/murraycadzow/data/NZ_coreExome_1kgp/nz_1kgp.panel', stringsAsFactors = FALSE) %>% filter(!pop %in% c("WPN","EPN","POL","NAD"))
for(p in unique(panel$pop)){
  pops[[p]] <- panel %>% filter(pop == p) %>%  .[["sample"]]
}

global_summary <- readRDS("~/data/NZ_coreExome_1kgp/100kbWindow_intra/filtered/100kbwindows_summary_popgenome_filtered_3ns_resized.31-8-2017.RDS") %>% filter(!pop %in% c("WPN","EPN","POL","NAD"), stat != "Fu.Li.D")

```

```{r, echo = FALSE, results='hold'}
global_summary  %>% left_join(., panel %>% select(pop, super_pop) %>% distinct(), by = 'pop') %>% arrange(super_pop) %>% select(stat, super_pop, pop, mean, sd, min, lower_1,median,upper_99, max)  %>% arrange(stat,super_pop, pop) %>%  data.frame()%>% mutate(stat = case_when(stat == "Tajima.D" ~ "Tajima's D", stat == "Fay.Wu.H" ~ "Fay and Wu's H", stat == "Fu.Li.F" ~ "Fu and Li's F", stat == "Zeng.E" ~ "Zeng's E", stat == "n.segregating.sites" ~ 'Num. segregating sites' ))  %>% rename(Statistic = stat, 'Super Population' = super_pop, Population = pop, Mean = mean, SD = sd, Minimum = min, '1st Percentile' = lower_1, Median = median, '99th Percentile' = upper_99, Maximum = max)%T>% {write.csv(., file = '~/Git_repos/bookdown_thesis/Appendices/03-gsTable.csv', row.names = FALSE, quote = FALSE)} %>% kable(., caption = '(\\#tab:gsTable) Global Summary Table', format = 'latex', booktabs = TRUE, longtable = TRUE, linesep = "", digits = 3) %>% kable_styling(font_size = 8, latex_options = c('repeat_header')) %>% footnote(general = "Appendices/03-gsTable.csv")
```

```{r, warning =TRUE}
lower_sig_stats <- readRDS('~/data/NZ_coreExome_1kgp/100kbWindow_intra/filtered/100kbwindows_lower_sig_stat_genes_filtered_ns3_resized-31-8-2017.RDS')
lower_sig_stats <- bind_rows(lapply(names(lower_sig_stats), function(y){
  lower_sig_stats[[y]] <- bind_rows(
    lapply(names(lower_sig_stats[[y]]), function(x){
      lower_sig_stats[[y]][[x]] %>% mutate(pop = x, statname =y, chrom = as.character(chrom)) 
    }
    ))
})) %>% filter(!pop %in% c("NAD","EPN","WPN","POL"), statname != 'Fu.Li.D')

upper_sig_stats <-readRDS('~/data/NZ_coreExome_1kgp/100kbWindow_intra/filtered/100kbwindows_upper_sig_stat_genes_filtered_ns3_resized-31-8-2017.RDS')
upper_sig_stats <- bind_rows(lapply(names(upper_sig_stats), function(y){
  upper_sig_stats[[y]] <- bind_rows(
    lapply(names(upper_sig_stats[[y]]), function(x){
      upper_sig_stats[[y]][[x]] %>% mutate(pop = x, statname =y, chrom = as.character(chrom)) 
    }
    ))
})) %>% filter(!pop %in% c("NAD","EPN","WPN","POL"), statname != 'Fu.Li.D')

```

```{r intrasfsPol, results='hold'}
# create a table of all of the loci that had windows in the 1st percentile and < 0 in value
lower_sig_stats %>% filter( pop %in% c('CIM','NZM','SAM','TON')) %>% select(chrom, statname, SYMBOL, pop) %>% distinct() %>% spread(., key=pop, value = pop) %>% replace_na(list(CIM = "", NZM = "", SAM = "", TON = "")) %>% unite(., col = Pop, c('CIM','NZM','SAM','TON'), sep = ',')  %>% mutate(Pop = stringr::str_replace_all(Pop, "[,]+", ',') %>% stringr::str_replace_all(., pattern = '^[,]+|[,]$', replacement = "") %>% stringr::str_replace_all(., ",", ", ")) %>% mutate(chr = as.numeric(sapply(chrom, function(x){strsplit(x, split = 'r')[[1]][2]}))) %>% arrange(statname, chr) %>% select(statname, everything(), -chr) %>% filter(!is.na(SYMBOL)) %>% mutate(stat = case_when(statname == "Tajima.D" ~ "Tajima's D", statname == "Fay.Wu.H" ~ "Fay and Wu's H", statname == "Fu.Li.F" ~ "Fu and Li's F", statname == "Zeng.E" ~ "Zeng's E" ))  %>% rename('Statistic' = statname, Chrom = chrom, 'Gene name' = SYMBOL, Population = Pop)  %T>% {write.csv(., file = '~/Git_repos/bookdown_thesis/Appendices/03-intrasfsPol.csv', row.names = FALSE, quote = TRUE)} %>% head(3) %>% kable(., caption = '(\\#tab:intrasfsPol) All genes that had a window in the 1st percentile and value < 0 from a Polynesian population', booktabs = TRUE, format = 'latex') %>% footnote(general = "Electronic Supplement: Appendices/03-intrasfsPol.csv", escape = FALSE, general_title = "")
```

```{r, results='hold', eval = FALSE}

sig_ihs_val %>% filter(superpop == 'POL', !is.na(genename) & genename != "") %>% select(Pop = pop, chrom, chrom_start, chrom_end, genename, iHS = statvalue) %>% arrange(chrom, chrom_start, Pop) %T>% {write.csv(., file = "Appendices/03-ihs-Pol.csv", row.names = FALSE, quote = FALSE)}  %>% head(3) %>% kable(., booktabs = TRUE, format = 'latex', caption = '(\\#tab:ihsPolgenes) All genes that had a significant iHS value in Polynesian populations.') %>% kableExtra::footnote("Appendices/03-ihs-Pol.csv", threeparttable = TRUE)

```

```{r, results='hold', eval = FALSE}

sig_nsl_val %>% filter(superpop == 'POL', !is.na(genename) & genename != "") %>% select(Pop = pop, chrom, chrom_start, chrom_end, genename, nSL = statvalue) %>% arrange(chrom, chrom_start, Pop) %T>% {write.csv(., file = "Appendices/03-nsl-Pol.csv", row.names = FALSE, quote = FALSE)}  %>% head(3) %>% kable(., booktabs = TRUE, format = 'latex', caption = '(\\#tab:nslPolgenes) All genes that had a significant nSL value in Polynesian populations.') %>% kableExtra::footnote("Appendices/03-nsl-Pol.csv", threeparttable = TRUE)

```

### Significant \gls{ihs} and \gls{nsl} markers in Polynesian populations

```{r, eval = TRUE, message = FALSE}
# process for creating table:
# load all ihs/nsl data
# filter to be only loci that had a significant result in any ihs/nsl from the poly pops
# apply bolding to the cells based on the populations and if significant (value > 2.6)
# spread out the columns to be pop1_ihs pop2_ihs pop2_nsl pop2_nsl etc


# poly_ihs <-list.files(path = '~/data/NZ_coreExome_1kgp/haplotype/poly_ihs_nsl_norm/ihs', full.names = TRUE) %>% 
# purrr::map_df(., ~read_delim(.x,delim = "\t" , col_names = FALSE) %>% select(marker = X1, chrom_start = X2, af = X4, ihs= X7, sig = X8) %>% mutate(pop = str_sub(str_split(.x, pattern = "/")[[1]][9], 1, 3), chrom = str_split(.x, pattern = "/")[[1]][9] %>% str_split(., pattern=fixed(".")) %>% purrr::map_chr(., 1) %>% str_sub(., 4) %>% as.numeric()))
# 
# poly_nsl <-list.files(path = '~/data/NZ_coreExome_1kgp/haplotype/poly_ihs_nsl_norm/nsl', full.names = TRUE) %>% 
# purrr::map_df(., ~read_delim(.x,delim = "\t" , col_names = FALSE) %>% select(marker = X1, chrom_start = X2, af = X4, nsl= X7, sig = X8) %>% mutate(pop = str_sub(str_split(.x, pattern = "/")[[1]][9], 1, 3), chrom = str_split(.x, pattern = "/")[[1]][9] %>% str_split(., pattern=fixed(".")) %>% purrr::map_chr(., 1) %>% str_sub(., 4) %>% as.numeric()))
# 
# ihsnsl_table <- full_join(
#   left_join(bind_rows(sig_ihs_val,sig_nsl_val) %>% 
#               filter(pop %in% poly_pop) %>% select(chrom, chrom_start) %>% distinct(), poly_ihs, by = c("chrom", "chrom_start")) %>%
#     mutate(ihs = if_else(abs(ihs) > 2.6, cell_spec(three_dp(ihs), bold = TRUE, format = 'latex'), three_dp(ihs))) %>% 
#     select(-af, -sig) %>% filter(!is.na(pop)) %>% spread(pop, ihs),
#   left_join(bind_rows(sig_ihs_val,sig_nsl_val) %>% 
#               filter(pop %in% poly_pop) %>% select(chrom, chrom_start) %>% distinct(), poly_nsl, by = c("chrom", "chrom_start")) %>%
#     mutate(nsl = if_else(abs(nsl) > 2.6, cell_spec(three_dp(nsl), bold = TRUE, format = 'latex'), three_dp(nsl))) %>% 
#     select(-af, -sig)%>% filter(!is.na(pop)) %>% spread(pop, nsl), by = c("chrom", 'chrom_start',"marker"), 
#   suffix = c('_ihs', '_nsl')) %>% 
#   mutate_at(vars(contains('_ihs'),contains('_nsl')), funs(case_when(is.na(.) ~ "-", TRUE ~ .))) %>%
#   select(chrom, position = chrom_start, marker, everything()) %>%
#   rename(Chromosome = chrom, Position = position, Marker = marker, 'CIM iHS' = CIM_ihs, 'NZM iHS' = NZM_ihs, 'SAM iHS' = SAM_ihs, 'TON iHS' = TON_ihs, 'CIM nSL' = CIM_nsl, 'NZM nSL' = NZM_nsl, 'SAM nSL' = SAM_nsl, 'TON nSL' = TON_nsl)
# saveRDS(object = ihsnsl_table, file='Appendices/03-ihsnsl_table.RDS')

ihsnsl_table <- readRDS(file='Appendices/03-ihsnsl_table.RDS')
ihsnsl_table%>%  
  head() %>% 
  kable(., booktabs = TRUE, linesep = "", escape = FALSE,format = 'latex', 
        caption = '(\\#tab:ihsnslPolGenes) Significant markers for iHS and nSL in Polynesian populations.') %>% kable_styling(latex_options = 'scale_down', font_size = 8)
```


```{r, eval = FALSE}
full_join(sig_ihs_val %>% filter(superpop == 'POL', !is.na(genename) & genename != "") %>% select(Pop = pop, chrom, chrom_start, chrom_end, genename, iHS = statvalue) %>% arrange(chrom, chrom_start, Pop)%>% spread(Pop, iHS), sig_nsl_val %>% filter(superpop == 'POL', !is.na(genename) & genename != "") %>% select(Pop = pop, chrom, chrom_start, chrom_end, genename, nSL = statvalue) %>% arrange(chrom, chrom_start, Pop) %>% spread(Pop, nSL), by = c("chrom", "chrom_start", "chrom_end", "genename"), suffix= c("_ihs", "_nsl")) %>% select(-chrom_end) %>% rename(position = chrom_start) %>% mutate_at(vars(contains('_ihs'),contains('_nsl')), funs(three_dp)) %>% mutate_at(vars(contains('_ihs'),contains('_nsl')), funs(case_when(. == "NA" ~ "", TRUE ~ .))) %>% left_join(., markers %>% select(chrom, chrom_start, marker), by = c('chrom', "position" = 'chrom_start')) %>% select(chrom, position, marker, everything()) %>%  rename(Chromosome = chrom, Position = position, Marker = marker, 'Gene Name' = genename, 'CIM iHS' = CIM_ihs, 'NZM iHS' = NZM_ihs, 'SAM iHS' = SAM_ihs, 'TON iHS' = TON_ihs, 'CIM nSL' = CIM_nsl, 'NZM nSL' = NZM_nsl, 'SAM nSL' = SAM_nsl, 'TON nSL' = TON_nsl)%>%  head() %>% kable(., booktabs = TRUE, linesep = "", format = 'latex', caption = '(\\#tab:ihsnslPolGenes) Significant markers for iHS and nSL in Polynesian populations.') %>% kable_styling(latex_options = 'scale_down', font_size = 8)

rm(sig_ihs_val)
rm(sig_nsl_val)        
```

### Significant \gls{xpehh} markers in Polynesian populations

```{r, eval = TRUE, message=FALSE, results='hold'}
xpehh <- read_delim('~/data/NZ_coreExome_1kgp/haplotype/sig_xpehh_29-10-2017.csv', delim = ',') %>% 
  filter(!pop1 %in% c("POL","EPN","WPN", "NAD"), !pop2 %in% c("POL","EPN","WPN", "NAD")) %>% filter(abs(xpehh_value) > 2.6)

xpehh %>% filter(!is.na(genename) & genename != "")%>% mutate(xpehh_value =three_dp(xpehh_value)) %>% 
  select(Pop1 = pop1,Pop2 = pop2, chrom, chrom_start, chrom_end,XPEHH = xpehh_value, Genename = genename)  %>% arrange(chrom, chrom_start, Pop1, Pop2) %T>% 
  {
  #write.csv(., file = "Appendices/03-xpehh-Pol.csv", row.names = FALSE, quote = FALSE)
  } %>% group_by(Pop2, chrom, chrom_start, chrom_end, Genename) %>% mutate(grouped_id = row_number()) %>% ungroup() %>% spread(Pop2, XPEHH) %>% arrange(chrom, chrom_start, Pop1) %>% 
  select(-grouped_id, -chrom_end) %>% 
  dplyr::rename(Population = Pop1, Chromsome = chrom, Position = chrom_start) %>%   head(20) %>% 
  kable(., booktabs = TRUE, format = 'latex',linesep = "", 
        caption = '(\\#tab:xpehhPolgenes) Genes that had a significant XP-EHH values in Polynesian populations.') %>% 
  column_spec(column = 1:3, width = '4em') %>% column_spec(column = 5:36, width = '2em')%>%  
  kableExtra::kable_styling(font_size = 6)%>% 
  kableExtra::footnote("Appendices/03-xpehh-Pol.csv", threeparttable = TRUE) %>% landscape()
rm(xpehh)
```

\FloatBarrier

```{r, eval = TRUE, results='hold'}
immunePol <- readRDS("Appendices/03-immunePol.RDS")
 immunePol %>% #filter(genename %in% immune_keep) %>% 
  mutate_all(., function(x){replace(x, x == 0, "")}) %>%
  mutate(genename = map_chr(genename, ~ cell_spec(.x, italic = TRUE, format = 'latex'))) %>% 
 kable(., caption = '(\\#tab:zhangImmune) Loci associated with various inflammatory and autoimmune diseases that showed signs of possible selection in Polynesian populations.', 
       longtable = TRUE, booktabs = TRUE, escape = FALSE, linesep = '',
       col.names = c('Population','Gene','AFR','AMR','EAS','EUR','POL','SAS','iHS','nSL',"Fay \\& Wu's H", " Fu \\& Li's F", "Tajima's D"," Zeng's E"), format = 'latex') %>% kableExtra::row_spec(0, angle = 90) %>% add_header_above(., c("","","XP-EHH" = 6, rep("",6))) %>%  kable_styling(latex_options = "repeat_header", font_size = 6) %>% 
   kableExtra::footnote(general = c("XP-EHH is the number of populations from the super population that had at least one marker sigificant in the gene. Integrated haplotype homozygosity score and nSL are the number of significant markers. \\\\gls{fwh}, \\\\gls{flf}, \\\\gls{td}, and \\\\gls{ze} are the number of windows intersecting the gene that met the lower threshold."), general_title = "", escape = FALSE, threeparttable = TRUE)
  # %>% filter((AFR > 0 | AMR >0 | EAS > 0 | ihs > 0 | nsl > 0) & (!is.na(Fay.Wu.H) | !is.na(Fu.Li.D)| !is.na(Fu.Li.F) | !is.na(Tajima.D) | !is.na(Zeng.E)))
```



```{r tdPol, results = 'asis', warning = FALSE, eval = FALSE}
# write table to csv file and create just the header of the table in the pdf
lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM','NZM','SAM','TON')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Tajima.D', !pop %in% c('CIM','NZM','SAM','TON')), by = "SYMBOL") %>% select(chrom, SYMBOL, pop) %>% distinct() %>% spread(., key=pop, value = pop) %>% replace_na(list(CIM = "", NZM = "", SAM = "", TON = "")) %>% unite(., col = Pop, c('CIM','NZM','SAM','TON'), sep = ',')  %>% mutate(Pop = stringr::str_replace_all(Pop, "[,]+", ',') %>% stringr::str_replace_all(., pattern = '^[,]+|[,]$', replacement = "") %>% stringr::str_replace_all(., ",", ", ")) %>% mutate(chr = as.numeric(sapply(chrom, function(x){strsplit(x, split = 'r')[[1]][2]}))) %>% arrange(chr) %>% select(-chr) %T>%  {write.csv(., file = '~/Git_repos/bookdown_thesis/Appendices/03_tdPol.csv', row.names = FALSE, quote = TRUE)} %>% head(0) %>%  kable(., booktabs = TRUE, caption = "(\\#tab:lowersigtdPol) Loci that had a Tajima's D in the lower tail that was only found in a Polynesian population", format = 'latex') %>% footnote(general = "Electronic Supplement: Appendices/03\\\\_tdPol.csv", escape = FALSE)
```

```{r fwhPol, results='asis', warning= FALSE, echo = FALSE, eval = FALSE}
lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM','NZM','SAM','TON')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Fay.Wu.H', !pop %in% c('CIM','NZM','SAM','TON')), by = "SYMBOL") %>% select(chrom, SYMBOL, pop) %>% distinct() %>% spread(., key=pop, value = pop) %>% replace_na(list(CIM = "", NZM = "", SAM = "", TON = "")) %>% unite(., col = Pop, c('CIM','NZM','SAM','TON'), sep = ',') %>% mutate(Pop = stringr::str_replace_all(Pop, "[,]+", ',') %>% stringr::str_replace_all(., pattern = '^[,]+|[,]$', replacement = "") %>% stringr::str_replace_all(., ",", ", ")) %>% mutate(chr = as.numeric(sapply(chrom, function(x){strsplit(x, split = 'r')[[1]][2]}))) %>% arrange(chr) %>% select(-chr)%>%  write.csv(., file = '~/Git_repos/bookdown_thesis/Appendices/03_fwhPol.csv', row.names = FALSE, quote = TRUE) # %>%  kable(., booktab = TRUE, longtable = TRUE,caption = "(\\#tab:lowersigfwhPol) Loci that had a Fay and Wu's H in the lower tail that was only found in a Polynesian population")
```

```{r flfPol, results = 'asis', warning = FALSE, echo = FALSE, eval = FALSE}
lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM','NZM','SAM','TON')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Fu.Li.F', !pop %in% c('CIM','NZM','SAM','TON')), by = "SYMBOL") %>% select(chrom, SYMBOL, pop) %>% distinct() %>% spread(., key=pop, value = pop) %>% replace_na(list(CIM = "", NZM = "", SAM = "", TON = "")) %>% unite(., col = Pop, c('CIM','NZM','SAM','TON'), sep = ',') %>% mutate(Pop = stringr::str_replace_all(Pop, "[,]+", ',') %>% stringr::str_replace_all(., pattern = '^[,]+|[,]$', replacement = "") %>% stringr::str_replace_all(., ",", ", "))%>% mutate(chr = as.numeric(sapply(chrom, function(x){strsplit(x, split = 'r')[[1]][2]}))) %>% arrange(chr) %>% select(-chr) %>% head() %>% kable(., booktab = TRUE, longtable = TRUE,caption = "(\\#tab:lowersigflfPol) Loci that had a Fu and Li's F in the lower tail that was only found in a Polynesian population")
```

```{r zePol, results='asis', eval = FALSE}
lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM','NZM','SAM','TON')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Zeng.E', !pop %in% c('CIM','NZM','SAM','TON')), by = "SYMBOL") %>% select(chrom, SYMBOL, pop) %>% distinct() %>% spread(., key=pop, value = pop) %>% replace_na(list(CIM = "", NZM = "", SAM = "", TON = "")) %>% unite(., col = Pop, c('CIM','NZM','SAM','TON'), sep = ',') %>% mutate(Pop = stringr::str_replace_all(Pop, "[,]+", ',') %>% stringr::str_replace_all(., pattern = '^[,]+|[,]$', replacement = "") %>% stringr::str_replace_all(., ",", ", "))%>% mutate(chr = as.numeric(sapply(chrom, function(x){strsplit(x, split = 'r')[[1]][2]}))) %>% arrange(chr) %>% select(-chr) %>%head() %>%  kable(.,booktab = TRUE, longtable = TRUE, caption = "(\\#tab:lowersigzePol) Loci that had a Zeng's E in the lower tail that was only found in a Polynesian population")
```

\FloatBarrier

## Chapter 4 Tables

```{r, echo = FALSE, eval = TRUE}
k_cv <- read.table('~/data/NZ_coreExome_1kgp/admixture/CV_error.txt', header = FALSE) %>% full_join(.,  read.table('~/data/NZ_coreExome_1kgp/admixture/projected/CV_error.txt', header = FALSE) %>% dplyr::rename( cv_proj = V2), by = "V1") %>% arrange(V1) %>% dplyr::rename(K = V1, "CV Error" = V2, "Projection CV Error" = cv_proj)

k_cv %>% select(1:2) %>% mutate(`CV Error` = formatC(`CV Error`, digits = 4, format ='f')) %>%knitr::kable(., caption = "(\\#tab:admixCV) Admixture cross-validation (CV) error for different values of K.", format = 'latex') 
```

```{r}
create_gwas_cat_table <-function(){
  # data from rnotebooks/CoreExome/gwas_catalog.Rmd
  load('~/data/gwas_catalog/disease_ref_table-12-3-2018.RData')
  return(gwas_int_ref_table %>% select(-pmid, -PUBMEDID))
}

create_sel_summary_table <- function(s){
global_summary %>% ungroup%>% filter(stat == s) %>% left_join(., panel %>% select(pop, super_pop) %>% distinct(), by = 'pop') %>% arrange(super_pop) %>% select(super_pop, pop, mean, sd, min, lower_1,median,upper_99, max)  %>%  data.frame()
}
  
create_sel_stat_table <- function(sel_stat, cap){
  knitr::kable(chr_summary %>% filter(stat == sel_stat) %>% ungroup() %>%  select(pop, chrom, mean) %>% spread(.,chrom, mean) %>% left_join(., panel %>% select(pop, super_pop) %>% distinct(), "pop") %>% select(pop, super_pop, 2:23) %>% arrange(super_pop, pop) %>% head(), align = c('l','c',rep('r', 22)), col.names = c("Population", "Super Population", paste0('Chr',1:22)), caption = cap, digits =  3)
}
```


```{r, results = 'asis'}
m <- create_gwas_cat_table() %>% data.frame() %>%  arrange(DISEASE.TRAIT, ref) %>% filter(!grepl(x = ref, pattern = 'NA')) %>% select(-REPORTED.GENE.S., 'Category' = list, 'Disease Trait' = DISEASE.TRAIT, 'Reference' = ref) %>% mutate(Reference = sapply(Reference, function(x){paste0("\\citet{",substr(x, 2, nchar(x)),"}")})) %>% select(3,1,2)%>% distinct() %>% arrange(Category, `Disease Trait`, Reference)  #%>% filter(Category != "Kidney Disease")
#colnames(m) <- c("Disease\nTrait", "Reported\nGenes","Reference")  
#pander::pandoc.table(m, split.cells = 30 , caption = "(\\#tab:gwascatref) GWAS catalog studies used. Disease trait as specified in GWAS catalog", keep.line.breaks = TRUE)

kable(m %>% select(-Category), escape = FALSE, linesep = "", format = 'latex',caption = "(\\#tab:gwascatref) GWAS catalog studies used. Disease trait as specified in GWAS catalog", booktabs = TRUE, longtable = TRUE) %>% kable_styling(latex_options = c("repeat_header"), full_width = TRUE, font_size = 8) %>% 
  group_rows("Auto-immunity and Auto-inflammatory", 1, 128) %>% 
  group_rows("Gout and Urate",129,146) %>% group_rows("Kidney Disease",147,154) %>% 
  group_rows("Malaria", 155, 157) %>% 
  group_rows("Metabolic Syndrome", 158,161) %>% 
  group_rows("Neurological", 162, 191) %>% 
  group_rows("Obesity", 192, 224) %>% group_rows("T2D", 225, 262) #%>% group_rows("Metabolic Syndrome", 19,22) %>% group_rows("Obesity", 23, 55) %>% group_rows("T2D", 56, 94) 

#
```


```{r gwasgenes, message = FALSE}
# genelists
load('~/data/gwas_catalog/diseaseGR-12-3-2018.RData')
genelists <- data.frame(pheno = c(rep("obesity", length(obesity_GR$SYMBOL)), rep("urate", length( gc_urate_gout_GR$SYMBOL)), rep('t2d', length(t2d_GR$SYMBOL)), rep("kd", length(kd_GR$SYMBOL)), rep("metsyn", length(metsyn_GR$SYMBOL))),  genename = c(obesity_GR$SYMBOL, gc_urate_gout_GR$SYMBOL, t2d_GR$SYMBOL, kd_GR$SYMBOL, metsyn_GR$SYMBOL), stringsAsFactors = FALSE) %>% distinct() %>%  mutate(present = 1) %>% spread(pheno, present, 0)
genelists %>% select(genename, urate, obesity, t2d, kd, metsyn) %>% kable(., col.names = c("Gene name", "Urate and gout","Obesity","T2D","Kidney disease", "Metabolic syndrome"), caption = '(\\#tab:gwasgenes) Disease associated genes by category from the GWAS catalog. For references see Table \\ref{tab:gwascatref}.' , escape = FALSE, booktabs = TRUE, longtable = TRUE, linesep="") %>% kable_styling(latex_options = 'repeat_header', font_size = 8)
```



```{r, echo = FALSE}
# This is created by coreExome_100kb_windows_intra_filtered.Rmd
chr_summary <- readRDS("~/data/NZ_coreExome_1kgp/100kbWindow_intra/filtered/100kbwindows_chr_summary_popgenome_filtered_3ns_resized.31-8-2017.RDS")
pops <- list()
panel <- read.delim('/home/murraycadzow/data/NZ_coreExome_1kgp/nz_1kgp.panel', stringsAsFactors = FALSE)
for(p in unique(panel$pop)){
  pops[[p]] <- panel %>% filter(pop == p) %>%  .[["sample"]]
}

global_summary <- readRDS("~/data/NZ_coreExome_1kgp/100kbWindow_intra/filtered/100kbwindows_summary_popgenome_filtered_3ns_resized.31-8-2017.RDS") %>% filter(!pop %in% c("WPN","EPN","POL","NAD"))
global_super_summary <- readRDS('~/data/NZ_coreExome_1kgp/100kbWindow_intra/filtered/100kbwindows_summary_superpop_popgenome_filtered_3ns_resized.6-11-2017.RDS')


```


```{r gsTdTable, eval = TRUE, echo = FALSE}
gs_td_table <- create_sel_summary_table("Tajima.D")
gs_td_table %>% mutate_at(vars('mean','sd','min','lower_1','upper_99','median','max'), funs(sprintf("%.3f",. ))) %>% select(-super_pop) %>%  knitr::kable(., format = 'latex', booktabs = TRUE, col.names = c("Population","Mean","Std. Dev","Min","Lower 1%","Median","Upper 99%", "Max"), caption = "(\\#tab:gsTdTable) \\gls{td} summary statistics by population", align = c('l', rep('r', 8))) %>% kableExtra::group_rows("AFR", 1, 7) %>% kableExtra::group_rows("AMR", 8, 11) %>% kableExtra::group_rows("EAS", 12, 16) %>% kableExtra::group_rows("EUR", 17, 22) %>% kableExtra::group_rows("POL", 23, 26) %>% kableExtra::group_rows("SAS", 27,31)
```

```{r, eval = TRUE, echo = FALSE}
gs_fwh_table <-create_sel_summary_table("Fay.Wu.H")

gs_fwh_table %>%  mutate_at(vars('mean','sd','min','lower_1','upper_99','median','max'), funs(sprintf("%.3f",. ))) %>%  select(-super_pop) %>%  knitr::kable(., format = 'latex', booktabs = TRUE, col.names = c("Population","Mean","Std. Dev","Min","Lower 1%","Median","Upper 99%", "Max"), caption = "(\\#tab:gsFwhTable) \\gls{fwh} summary statistics by population", align = c('l', rep('r', 8))) %>% kableExtra::group_rows("AFR", 1, 7) %>% kableExtra::group_rows("AMR", 8, 11) %>% kableExtra::group_rows("EAS", 12, 16) %>% kableExtra::group_rows("EUR", 17, 22) %>% kableExtra::group_rows("POL", 23, 26) %>% kableExtra::group_rows("SAS", 27,31)
```

```{r, eval = TRUE, echo = FALSE}
gs_flf_table <- create_sel_summary_table("Fu.Li.F")

gs_flf_table %>% mutate_at(vars('mean','sd','min','lower_1','upper_99','median','max'), funs(sprintf("%.3f",. ))) %>% select(-super_pop) %>%  knitr::kable(., format = 'latex', booktabs = TRUE, col.names = c("Population","Mean","Std. Dev","Min","Lower 1%","Median","Upper 99%", "Max"), caption = "(\\#tab:gsFlfTable) \\gls{flf} summary statistics by population", align = c('l', rep('r', 8))) %>% kableExtra::group_rows("AFR", 1, 7) %>% kableExtra::group_rows("AMR", 8, 11) %>% kableExtra::group_rows("EAS", 12, 16) %>% kableExtra::group_rows("EUR", 17, 22) %>% kableExtra::group_rows("POL", 23, 26) %>% kableExtra::group_rows("SAS", 27,31)
```

```{r, eval = TRUE, echo = FALSE}
gs_ze_table <- create_sel_summary_table("Zeng.E")

gs_ze_table %>% mutate_at(vars('mean','sd','min','lower_1','upper_99','median','max'), funs(sprintf("%.3f",. ))) %>% select(-super_pop) %>%  knitr::kable(., format = 'latex', booktabs = TRUE, col.names = c("Population","Mean","Std. Dev","Min","Lower 1%","Median","Upper 99%", "Max"), caption = "(\\#tab:gsZeTable) \\gls{ze} summary statistics by population", align = c('l', rep('r', 8))) %>% kableExtra::group_rows("AFR", 1, 7) %>% kableExtra::group_rows("AMR", 8, 11) %>% kableExtra::group_rows("EAS", 12, 16) %>% kableExtra::group_rows("EUR", 17, 22) %>% kableExtra::group_rows("POL", 23, 26) %>% kableExtra::group_rows("SAS", 27,31)
```


#### Polynesian windows for clustering of the extremes

```{r}
eigen <- read.table('~/data/NZ_coreExome_1kgp/PCA/indiv_pops/NZ_1KGP_allchr_indep_eigen.pca.evec', stringsAsFactors = FALSE)

eigen <- eigen %>% select("sample" = V1, everything()) %>% left_join(., panel, by = 'sample') %>% filter(!pop %in% c('WPN','EPN','NAD'))
names(eigen)[2:11] <- paste0("PC",1:10)


```


```{r, message = FALSE}
read_csv('Appendices/04-td-pol-firstper-genes.csv', n_max = 0) %>% kable(format = 'latex', caption = '(\\#tab:polTdNegGenes) Genes intersecting the windows of the 1st percentile \\gls{td} in Polynesians') %>% footnote("Appendices/04-td-pol-firstper-genes.csv", threeparttable = TRUE)
```

```{r, message = FALSE}
read_csv('Appendices/04-td-pol-ninetyninthper-genes.csv', n_max = 0) %>% kable(format = 'latex', caption = '(\\#tab:polTdPosGenes) Genes intersecting the windows of the 99th percentile \\gls{td} in Polynesians') %>% footnote('Appendices/04-td-pol-ninetyninthper-genes.csv', threeparttable = TRUE)
```


```{r, message = FALSE}
read_csv('Appendices/04-fwh-pol-firstper-genes.csv', n_max = 0) %>% kable(format = 'latex', caption = '(\\#tab:polFwhNegGenes) Genes intersecting the windows of the 1st percentile \\gls{fwh} in Polynesians') %>% footnote('Appendices/04-fwh-pol-firstper-genes.csv', threeparttable = TRUE)
```

```{r, message = FALSE}
read_csv('Appendices/04-fwh-pol-ninetyninethper-genes.csv', n_max = 0) %>% kable(format = 'latex', caption = '(\\#tab:polFwhPosGenes) Genes intersecting the windows of the 99th percentile \\gls{fwh} in Polynesians') %>% footnote('Appendices/04-fwh-pol-ninetyninethper-genes.csv', threeparttable = TRUE)
```

```{r, message = FALSE}
read_csv('Appendices/04-flf-pol-firstper-genes.csv', n_max = 0) %>% kable(format = 'latex', caption = '(\\#tab:polFlfNegGenes) Genes intersecting the windows of the 1st percentile \\gls{flf} in Polynesians') %>% footnote('Appendices/04-flf-pol-firstper-genes.csv', threeparttable = TRUE)
```

```{r, message = FALSE}
read_csv('Appendices/04-flf-pol-ninetyninethper-genes.csv', n_max = 0) %>% kable(format = 'latex', caption = '(\\#tab:polFlfPosGenes) Genes intersecting the windows of the 99th percentile \\gls{flf} in Polynesians') %>% footnote('Appendices/04-flf-pol-ninetyninethper-genes.csv', threeparttable = TRUE)
```


```{r, message = FALSE}
read_csv('Appendices/04-ze-pol-firstper-genes.csv', n_max = 0) %>% kable(format = 'latex', caption = '(\\#tab:polZeNegGenes) Genes intersecting the windows of the 1st percentile \\gls{ze} in Polynesians') %>% footnote('Appendices/04-ze-pol-firstper-genes.csv', threeparttable = TRUE)
```

```{r, message = FALSE}
read_csv('Appendices/04-ze-pol-ninetyninethper-genes.csv', n_max = 0) %>% kable(format = 'latex', caption = '(\\#tab:polZePosGenes) Genes intersecting the windows of the 99th percentile \\gls{ze} in Polynesians') %>% footnote('Appendices/04-ze-pol-ninetyninethper-genes.csv', threeparttable = TRUE)
```

#### Clustered regions from significant markers for \gls{ihs} and \gls{nsl}

```{r}
clus_ihs <- readRDS('~/data/NZ_coreExome_1kgp/haplotype/ihs_clus_regions-15-12-2017.RDS')
clus_nsl <- readRDS('~/data/NZ_coreExome_1kgp/haplotype/nsl_clus_regions-15-12-2017.RDS')
```

```{r}
full_join(clus_ihs, clus_nsl, by = c('pop','chrom','ClusterID'), suffix = c(" iHS", " nSL")) %>% dplyr::rename(Population = pop) %T>% {write_csv(., path = 'Appendices/04-ihs-nsl-significant-marker-clusters.csv', col_names = TRUE)} %>% head(0) %>% kable(., caption = '(\\#tab:hapclustab) Positions of regions created by clustering significant markers for iHS or nSL by population') %>% footnote("Appendices/04-ihs-nsl-significant-marker-clusters.csv", threeparttable = TRUE)
```

\FloatBarrier

#### Clustered median centered metabolic disease genes


```{r load_gene_clus, include = FALSE}
# brings in objects called {gc_urate_gout,urate_gout,kd,metsyn,obesity,t2d}_GR, 
source("~/Git_repos/bookdown_thesis/scripts/Annotate_genes.R")
load('~/data/gwas_catalog/diseaseGR-12-3-2018.RData')
load('~/data/NZ_coreExome_1kgp/100kbWindow_intra/gene_clustering_filtered_3ns-16-4-2018.RData')# brings in objects called gout, obesity, t2d, metsyn
names(gout) <- c('mat_d','prop')
names(t2d) <-  c('mat_d','prop')
names(obesity) <- c('mat_d','prop')
names(metsyn) <- c('mat_d','prop')
names(kd) <- c('mat_d','prop')
```

##### Gout

```{r, message = FALSE}
s <- "Tajima.D"
gout$mat_d[[s]] %>%  mutate(chrom = paste0('chr', chrom)) %>% gather(pop, st, -contains("chrom")) %>% left_join(., gout$mat_d[[s]] %>% select(contains("chrom")) %>% mutate(chrom = paste0('chr', chrom)) %>% GRanges() %>% txdb_gene_annotate() , by = c('chrom' = "seqnames", "chrom_start" = 'start', "chrom_end" = "end")) %>% mutate(superpop = str_sub(pop, 1,3))%>% filter(SYMBOL %in% gc_urate_gout_GR$SYMBOL) %>% mutate(chr = str_sub(chrom, 4) %>% as.numeric()) %>% select(-width, -strand, -TXSTRAND, -superpop) %>% spread(pop, st) %>% arrange(chr, chrom_start) %>% select(-chr) %>% set_names(., c("Chrom","Start","End", "Gene name", names(.)[-1:-4] %>% str_sub(., 5)))  %>% kable(., longtable =  TRUE, caption = "(\\#tab:gouttdgeneclus) Tajima's D for windows at gout associated loci", digits = 3, booktabs = TRUE)  %>% column_spec(., 5:35, width = '2em') %>% column_spec(., 1, width = '3.5em') %>% column_spec(., 2:4, width = '3.5em')%>% add_header_above(., header = c(rep("", 4), "AFR" = 7, "AMR" = 4, "EAS" = 5, "EUR" = 6, "POL" = 4, "SAS" = 5)) %>% kable_styling(font_size = 5,latex_options = 'repeat_header') %>% landscape()
```

```{r, message = FALSE}
s <- "Fay.Wu.H"
gout$mat_d[[s]] %>%  mutate(chrom = paste0('chr', chrom)) %>% gather(pop, st, -contains("chrom")) %>% left_join(., gout$mat_d[[s]] %>% select(contains("chrom")) %>% mutate(chrom = paste0('chr', chrom)) %>% GRanges() %>% txdb_gene_annotate() , by = c('chrom' = "seqnames", "chrom_start" = 'start', "chrom_end" = "end")) %>% mutate(superpop = str_sub(pop, 1,3))%>% filter(SYMBOL %in% gc_urate_gout_GR$SYMBOL) %>% mutate(chr = str_sub(chrom, 4) %>% as.numeric()) %>% select(-width, -strand, -TXSTRAND, -superpop) %>% spread(pop, st) %>% arrange(chr, chrom_start) %>% select(-chr) %>% set_names(., c("Chrom","Start","End", "Gene name", names(.)[-1:-4] %>% str_sub(., 5)))  %>% kable(., longtable =  TRUE, caption = "(\\#tab:goutfwhgeneclus) Fay and Wu's H for windows at gout associated loci.", digits = 3, booktabs = TRUE)  %>% column_spec(., 5:35, width = '2em') %>% column_spec(., 1, width = '3.5em') %>% column_spec(., 2:4, width = '3.5em')%>% add_header_above(., header = c(rep("", 4), "AFR" = 7, "AMR" = 4, "EAS" = 5, "EUR" = 6, "POL" = 4, "SAS" = 5)) %>% kable_styling(font_size = 5,latex_options = 'repeat_header') %>% landscape()
```


```{r, message = FALSE}
s <- "Fu.Li.F"
gout$mat_d[[s]] %>%  mutate(chrom = paste0('chr', chrom)) %>% gather(pop, st, -contains("chrom")) %>% left_join(., gout$mat_d[[s]] %>% select(contains("chrom")) %>% mutate(chrom = paste0('chr', chrom)) %>% GRanges() %>% txdb_gene_annotate() , by = c('chrom' = "seqnames", "chrom_start" = 'start', "chrom_end" = "end")) %>% mutate(superpop = str_sub(pop, 1,3))%>% filter(SYMBOL %in% gc_urate_gout_GR$SYMBOL) %>% mutate(chr = str_sub(chrom, 4) %>% as.numeric()) %>% select(-width, -strand, -TXSTRAND, -superpop) %>% spread(pop, st) %>% arrange(chr, chrom_start) %>% select(-chr) %>% set_names(., c("Chrom","Start","End", "Gene name", names(.)[-1:-4] %>% str_sub(., 5)))  %>% kable(., longtable =  TRUE, caption = "(\\#tab:goutflfgeneclus) Fu and Li's F for windows at gout associated loci.", digits = 3, booktabs = TRUE)  %>% column_spec(., 5:35, width = '2em') %>% column_spec(., 1, width = '3.5em') %>% column_spec(., 2:4, width = '3.5em')%>% add_header_above(., header = c(rep("", 4), "AFR" = 7, "AMR" = 4, "EAS" = 5, "EUR" = 6, "POL" = 4, "SAS" = 5)) %>% kable_styling(font_size = 5,latex_options = 'repeat_header') %>% landscape()
```

```{r, message = FALSE}
s <- "Zeng.E"
gout$mat_d[[s]] %>%  mutate(chrom = paste0('chr', chrom)) %>% gather(pop, st, -contains("chrom")) %>% left_join(., gout$mat_d[[s]] %>% select(contains("chrom")) %>% mutate(chrom = paste0('chr', chrom)) %>% GRanges() %>% txdb_gene_annotate() , by = c('chrom' = "seqnames", "chrom_start" = 'start', "chrom_end" = "end")) %>% mutate(superpop = str_sub(pop, 1,3))%>% filter(SYMBOL %in% gc_urate_gout_GR$SYMBOL) %>% mutate(chr = str_sub(chrom, 4) %>% as.numeric()) %>% select(-width, -strand, -TXSTRAND, -superpop) %>% spread(pop, st) %>% arrange(chr, chrom_start) %>% select(-chr) %>% set_names(., c("Chrom","Start","End", "Gene name", names(.)[-1:-4] %>% str_sub(., 5)))  %>% kable(., longtable =  TRUE, caption = "(\\#tab:goutflfgeneclus) Zeng's E for windows at gout associated loci.", digits = 3, booktabs = TRUE)  %>% column_spec(., 5:35, width = '2em') %>% column_spec(., 1, width = '3.5em') %>% column_spec(., 2:4, width = '3.5em')%>% add_header_above(., header = c(rep("", 4), "AFR" = 7, "AMR" = 4, "EAS" = 5, "EUR" = 6, "POL" = 4, "SAS" = 5)) %>% kable_styling(font_size = 5,latex_options = 'repeat_header') %>% landscape()
```


##### Obesity

```{r, message = FALSE}
s <- "Tajima.D"
obesity$mat_d[[s]] %>%  mutate(chrom = paste0('chr', chrom)) %>% gather(pop, st, -contains("chrom")) %>% left_join(., obesity$mat_d[[s]] %>% select(contains("chrom")) %>% mutate(chrom = paste0('chr', chrom)) %>% GRanges() %>% txdb_gene_annotate() , by = c('chrom' = "seqnames", "chrom_start" = 'start', "chrom_end" = "end")) %>% mutate(superpop = str_sub(pop, 1,3))%>% filter(SYMBOL %in% obesity_GR$SYMBOL) %>% mutate(chr = str_sub(chrom, 4) %>% as.numeric()) %>% select(-width, -strand, -TXSTRAND, -superpop) %>% spread(pop, st) %>% arrange(chr, chrom_start) %>% select(-chr) %>% set_names(., c("Chrom","Start","End", "Gene name", names(.)[-1:-4] %>% str_sub(., 5)))  %>% kable(., longtable =  TRUE, caption = "(\\#tab:obesitytdgeneclus) Tajima's D for windows at obesity associated loci", digits = 3, booktabs = TRUE)  %>% column_spec(., 5:35, width = '2em') %>% column_spec(., 1, width = '3.5em') %>% column_spec(., 2:4, width = '3.5em')%>% add_header_above(., header = c(rep("", 4), "AFR" = 7, "AMR" = 4, "EAS" = 5, "EUR" = 6, "POL" = 4, "SAS" = 5)) %>% kable_styling(font_size = 5,latex_options = 'repeat_header') %>% landscape()
```

```{r, message = FALSE}
s <- "Fay.Wu.H"
obesity$mat_d[[s]] %>%  mutate(chrom = paste0('chr', chrom)) %>% gather(pop, st, -contains("chrom")) %>% left_join(., obesity$mat_d[[s]] %>% select(contains("chrom")) %>% mutate(chrom = paste0('chr', chrom)) %>% GRanges() %>% txdb_gene_annotate() , by = c('chrom' = "seqnames", "chrom_start" = 'start', "chrom_end" = "end")) %>% mutate(superpop = str_sub(pop, 1,3))%>% filter(SYMBOL %in% obesity_GR$SYMBOL) %>% mutate(chr = str_sub(chrom, 4) %>% as.numeric()) %>% select(-width, -strand, -TXSTRAND, -superpop) %>% spread(pop, st) %>% arrange(chr, chrom_start) %>% select(-chr) %>% set_names(., c("Chrom","Start","End", "Gene name", names(.)[-1:-4] %>% str_sub(., 5)))  %>% kable(., longtable =  TRUE, caption = "(\\#tab:obesityfwhgeneclus) Fay and Wu's H for windows at obesity associated loci.", digits = 3, booktabs = TRUE)  %>% column_spec(., 5:35, width = '2em') %>% column_spec(., 1, width = '3.5em') %>% column_spec(., 2:4, width = '3.5em')%>% add_header_above(., header = c(rep("", 4), "AFR" = 7, "AMR" = 4, "EAS" = 5, "EUR" = 6, "POL" = 4, "SAS" = 5)) %>% kable_styling(font_size = 5,latex_options = 'repeat_header') %>% landscape()
```


```{r, message = FALSE}
s <- "Fu.Li.F"
obesity$mat_d[[s]] %>%  mutate(chrom = paste0('chr', chrom)) %>% gather(pop, st, -contains("chrom")) %>% left_join(., obesity$mat_d[[s]] %>% select(contains("chrom")) %>% mutate(chrom = paste0('chr', chrom)) %>% GRanges() %>% txdb_gene_annotate() , by = c('chrom' = "seqnames", "chrom_start" = 'start', "chrom_end" = "end")) %>% mutate(superpop = str_sub(pop, 1,3))%>% filter(SYMBOL %in% obesity_GR$SYMBOL) %>% mutate(chr = str_sub(chrom, 4) %>% as.numeric()) %>% select(-width, -strand, -TXSTRAND, -superpop) %>% spread(pop, st) %>% arrange(chr, chrom_start) %>% select(-chr) %>% set_names(., c("Chrom","Start","End", "Gene name", names(.)[-1:-4] %>% str_sub(., 5)))  %>% kable(., longtable =  TRUE, caption = "(\\#tab:obesityflfgeneclus) Fu and Li's F for windows at obesity associated loci.", digits = 3, booktabs = TRUE)  %>% column_spec(., 5:35, width = '2em') %>% column_spec(., 1, width = '3.5em') %>% column_spec(., 2:4, width = '3.5em')%>% add_header_above(., header = c(rep("", 4), "AFR" = 7, "AMR" = 4, "EAS" = 5, "EUR" = 6, "POL" = 4, "SAS" = 5)) %>% kable_styling(font_size = 5,latex_options = 'repeat_header') %>% landscape()
```

```{r, message = FALSE}
s <- "Zeng.E"
obesity$mat_d[[s]] %>%  mutate(chrom = paste0('chr', chrom)) %>% gather(pop, st, -contains("chrom")) %>% left_join(., obesity$mat_d[[s]] %>% select(contains("chrom")) %>% mutate(chrom = paste0('chr', chrom)) %>% GRanges() %>% txdb_gene_annotate() , by = c('chrom' = "seqnames", "chrom_start" = 'start', "chrom_end" = "end")) %>% mutate(superpop = str_sub(pop, 1,3))%>% filter(SYMBOL %in% obesity_GR$SYMBOL) %>% mutate(chr = str_sub(chrom, 4) %>% as.numeric()) %>% select(-width, -strand, -TXSTRAND, -superpop) %>% spread(pop, st) %>% arrange(chr, chrom_start) %>% select(-chr) %>% set_names(., c("Chrom","Start","End", "Gene name", names(.)[-1:-4] %>% str_sub(., 5)))  %>% kable(., longtable =  TRUE, caption = "(\\#tab:obesityflfgeneclus) Zeng's E for windows at obesity associated loci.", digits = 3, booktabs = TRUE)  %>% column_spec(., 5:35, width = '2em') %>% column_spec(., 1, width = '3.5em') %>% column_spec(., 2:4, width = '3.5em')%>% add_header_above(., header = c(rep("", 4), "AFR" = 7, "AMR" = 4, "EAS" = 5, "EUR" = 6, "POL" = 4, "SAS" = 5)) %>% kable_styling(font_size = 5,latex_options = 'repeat_header') %>% landscape()
```

##### Type 2 diabetes

```{r, message = FALSE}
s <- "Tajima.D"
t2d$mat_d[[s]] %>%  mutate(chrom = paste0('chr', chrom)) %>% gather(pop, st, -contains("chrom")) %>% left_join(., t2d$mat_d[[s]] %>% select(contains("chrom")) %>% mutate(chrom = paste0('chr', chrom)) %>% GRanges() %>% txdb_gene_annotate() , by = c('chrom' = "seqnames", "chrom_start" = 'start', "chrom_end" = "end")) %>% mutate(superpop = str_sub(pop, 1,3))%>% filter(SYMBOL %in% t2d_GR$SYMBOL) %>% mutate(chr = str_sub(chrom, 4) %>% as.numeric()) %>% select(-width, -strand, -TXSTRAND, -superpop) %>% spread(pop, st) %>% arrange(chr, chrom_start) %>% select(-chr) %>% set_names(., c("Chrom","Start","End", "Gene name", names(.)[-1:-4] %>% str_sub(., 5)))  %>% kable(., longtable =  TRUE, caption = "(\\#tab:t2dtdgeneclus) Tajima's D for windows at t2d associated loci", digits = 3, booktabs = TRUE)  %>% column_spec(., 5:35, width = '2em') %>% column_spec(., 1, width = '3.5em') %>% column_spec(., 2:4, width = '3.5em')%>% add_header_above(., header = c(rep("", 4), "AFR" = 7, "AMR" = 4, "EAS" = 5, "EUR" = 6, "POL" = 4, "SAS" = 5)) %>% kable_styling(font_size = 5,latex_options = 'repeat_header') %>% landscape()
```

```{r, message = FALSE}
s <- "Fay.Wu.H"
t2d$mat_d[[s]] %>%  mutate(chrom = paste0('chr', chrom)) %>% gather(pop, st, -contains("chrom")) %>% left_join(., t2d$mat_d[[s]] %>% select(contains("chrom")) %>% mutate(chrom = paste0('chr', chrom)) %>% GRanges() %>% txdb_gene_annotate() , by = c('chrom' = "seqnames", "chrom_start" = 'start', "chrom_end" = "end")) %>% mutate(superpop = str_sub(pop, 1,3))%>% filter(SYMBOL %in% t2d_GR$SYMBOL) %>% mutate(chr = str_sub(chrom, 4) %>% as.numeric()) %>% select(-width, -strand, -TXSTRAND, -superpop) %>% spread(pop, st) %>% arrange(chr, chrom_start) %>% select(-chr) %>% set_names(., c("Chrom","Start","End", "Gene name", names(.)[-1:-4] %>% str_sub(., 5)))  %>% kable(., longtable =  TRUE, caption = "(\\#tab:t2dfwhgeneclus) Fay and Wu's H for windows at t2d associated loci.", digits = 3, booktabs = TRUE)  %>% column_spec(., 5:35, width = '2em') %>% column_spec(., 1, width = '3.5em') %>% column_spec(., 2:4, width = '3.5em')%>% add_header_above(., header = c(rep("", 4), "AFR" = 7, "AMR" = 4, "EAS" = 5, "EUR" = 6, "POL" = 4, "SAS" = 5)) %>% kable_styling(font_size = 5,latex_options = 'repeat_header') %>% landscape()
```


```{r, message = FALSE}
s <- "Fu.Li.F"
t2d$mat_d[[s]] %>%  mutate(chrom = paste0('chr', chrom)) %>% gather(pop, st, -contains("chrom")) %>% left_join(., t2d$mat_d[[s]] %>% select(contains("chrom")) %>% mutate(chrom = paste0('chr', chrom)) %>% GRanges() %>% txdb_gene_annotate() , by = c('chrom' = "seqnames", "chrom_start" = 'start', "chrom_end" = "end")) %>% mutate(superpop = str_sub(pop, 1,3))%>% filter(SYMBOL %in% t2d_GR$SYMBOL) %>% mutate(chr = str_sub(chrom, 4) %>% as.numeric()) %>% select(-width, -strand, -TXSTRAND, -superpop) %>% spread(pop, st) %>% arrange(chr, chrom_start) %>% select(-chr) %>% set_names(., c("Chrom","Start","End", "Gene name", names(.)[-1:-4] %>% str_sub(., 5)))  %>% kable(., longtable =  TRUE, caption = "(\\#tab:t2dflfgeneclus) Fu and Li's F for windows at t2d associated loci.", digits = 3, booktabs = TRUE)  %>% column_spec(., 5:35, width = '2em') %>% column_spec(., 1, width = '3.5em') %>% column_spec(., 2:4, width = '3.5em')%>% add_header_above(., header = c(rep("", 4), "AFR" = 7, "AMR" = 4, "EAS" = 5, "EUR" = 6, "POL" = 4, "SAS" = 5)) %>% kable_styling(font_size = 5,latex_options = 'repeat_header') %>% landscape()
```

```{r, message = FALSE}
s <- "Zeng.E"
t2d$mat_d[[s]] %>%  mutate(chrom = paste0('chr', chrom)) %>% gather(pop, st, -contains("chrom")) %>% left_join(., t2d$mat_d[[s]] %>% select(contains("chrom")) %>% mutate(chrom = paste0('chr', chrom)) %>% GRanges() %>% txdb_gene_annotate() , by = c('chrom' = "seqnames", "chrom_start" = 'start', "chrom_end" = "end")) %>% mutate(superpop = str_sub(pop, 1,3))%>% filter(SYMBOL %in% t2d_GR$SYMBOL) %>% mutate(chr = str_sub(chrom, 4) %>% as.numeric()) %>% select(-width, -strand, -TXSTRAND, -superpop) %>% spread(pop, st) %>% arrange(chr, chrom_start) %>% select(-chr) %>% set_names(., c("Chrom","Start","End", "Gene name", names(.)[-1:-4] %>% str_sub(., 5)))  %>% kable(., longtable =  TRUE, caption = "(\\#tab:t2dflfgeneclus) Zeng's E for windows at t2d associated loci.", digits = 3, booktabs = TRUE)  %>% column_spec(., 5:35, width = '2em') %>% column_spec(., 1, width = '3.5em') %>% column_spec(., 2:4, width = '3.5em')%>% add_header_above(., header = c(rep("", 4), "AFR" = 7, "AMR" = 4, "EAS" = 5, "EUR" = 6, "POL" = 4, "SAS" = 5)) %>% kable_styling(font_size = 5,latex_options = 'repeat_header') %>% landscape()
```
