```{r, messages = FALSE, warning=FALSE, include=FALSE}
#knitr::opts_knit$set(root.dir = '/home/murraycadzow/data/NZ_coreExome_1kgp/' )
knitr::opts_chunk$set(echo = FALSE)
library(tidyverse)
library(formattable)
library(knitr)
library(kableExtra)
library(phdFunctions)
#select <- dplyr::select


```
# Appendix E

## Electronic Supplements





```{r, results = 'asis', eval = FALSE}
# modify so that the full table comes out
pander::pandoc.table(chr_summary %>% mutate(mean = sapply(mean, three_dp), sd = sapply(sd, three_dp), lower_1 = sapply(lower_1, three_dp), upper_99 = sapply(upper_99, three_dp), median = sapply(median, three_dp), min = sapply(min,three_dp), max = sapply(max, three_dp)) %>% arrange(chrom,pop,stat) %>% head() %>% data.frame(),  caption = '(\\#tab:selStatSummary) Summary of the selection test statistics for all populations and all autosomes', split.tables = 80, style = 'multiline', justify = 'lccrrrrrrr')
```



```{r meanSegSitesTable, results='asis', echo = FALSE, eval=FALSE}
create_sel_stat_table("n.segregating.sites", "Mean Number of Segregating Sites")
```

<!-- Tajimas D Table -->
```{r meanTDtable, results='asis', echo = FALSE, eval = FALSE}
create_sel_stat_table("Tajima.D", "Mean Tajima's D")
```

<!-- Fay and Wu's H Table -->
```{r meanFWHtable, results='asis', echo = FALSE, eval = FALSE}
create_sel_stat_table("Fay.Wu.H", "Mean Fay and Wu's H")
```

<!-- Fu and Li D Table -->
```{r meanFLDtable, results='asis', echo = FALSE, eval = FALSE}
create_sel_stat_table("Fu.Li.D","Mean Fu and Li D")
```

<!-- Fu and Li F Table -->
```{r, results='asis', echo = FALSE, eval = FALSE}
create_sel_stat_table("Fu.Li.F","Mean Fu and Li F")
```

<!--Zeng E Table -->
```{r, results='asis', echo = FALSE, eval = FALSE}
create_sel_stat_table("Zeng.E", "Mean Zeng E")
```



```{r, echo = FALSE}
# This is created by coreExome_100kb_windows_intra_filtered.Rmd
chr_summary <- readRDS("~/data/NZ_coreExome_1kgp/100kbWindow_intra/filtered/100kbwindows_chr_summary_popgenome_filtered_3ns_resized.31-8-2017.RDS")
pops <- list()
panel <- read.delim('/home/murraycadzow/data/NZ_coreExome_1kgp/nz_1kgp.panel', stringsAsFactors = FALSE)
for(p in unique(panel$pop)){
  pops[[p]] <- panel %>% filter(pop == p) %>%  .[["sample"]]
}

global_summary <- readRDS("~/data/NZ_coreExome_1kgp/100kbWindow_intra/filtered/100kbwindows_summary_popgenome_filtered_3ns_resized.31-8-2017.RDS") %>% filter(!pop %in% c("WPN","EPN","POL","NAD"))

```


```{r, echo = FALSE}
global_summary  %>% left_join(., panel %>% select(pop, super_pop) %>% distinct(), by = 'pop') %>% arrange(super_pop) %>% select(stat, super_pop, pop, mean, sd, min, lower_1,median,upper_99, max)  %>% arrange(stat,super_pop, pop) %>%  data.frame() %T>% {write.csv(., file = '~/Git_repos/bookdown_thesis/Appendices/03-gsTable.csv', row.names = FALSE, quote = FALSE)} %>% head(0) %>% kable(., caption = '(\\#tab:gsTable) Global Summary Table', format = 'latex', booktabs = TRUE) %>% footnote(general = "Appendices/03-gsTable.csv")
```


```{r, warning =TRUE}
lower_sig_stats <- readRDS('~/data/NZ_coreExome_1kgp/100kbWindow_intra/filtered/100kbwindows_lower_sig_stat_genes_filtered_ns3_resized-31-8-2017.RDS')
lower_sig_stats <- bind_rows(lapply(names(lower_sig_stats), function(y){
  lower_sig_stats[[y]] <- bind_rows(
    lapply(names(lower_sig_stats[[y]]), function(x){
      lower_sig_stats[[y]][[x]] %>% mutate(pop = x, statname =y, chrom = as.character(chrom)) 
    }
    ))
})) %>% filter(!pop %in% c("NAD","EPN","WPN","POL"))

upper_sig_stats <-readRDS('~/data/NZ_coreExome_1kgp/100kbWindow_intra/filtered/100kbwindows_upper_sig_stat_genes_filtered_ns3_resized-31-8-2017.RDS')
upper_sig_stats <- bind_rows(lapply(names(upper_sig_stats), function(y){
  upper_sig_stats[[y]] <- bind_rows(
    lapply(names(upper_sig_stats[[y]]), function(x){
      upper_sig_stats[[y]][[x]] %>% mutate(pop = x, statname =y, chrom = as.character(chrom)) 
    }
    ))
})) %>% filter(!pop %in% c("NAD","EPN","WPN","POL"))

```

```{r intrasfsPol}
# create a table of all of the loci that had windows in the 1st percentile and < 0 in value
lower_sig_stats %>% filter( pop %in% c('CIM','NZM','SAM','TON')) %>% select(chrom, statname, SYMBOL, pop) %>% distinct() %>% spread(., key=pop, value = pop) %>% replace_na(list(CIM = "", NZM = "", SAM = "", TON = "")) %>% unite(., col = Pop, c('CIM','NZM','SAM','TON'), sep = ',')  %>% mutate(Pop = stringr::str_replace_all(Pop, "[,]+", ',') %>% stringr::str_replace_all(., pattern = '^[,]+|[,]$', replacement = "") %>% stringr::str_replace_all(., ",", ", ")) %>% mutate(chr = as.numeric(sapply(chrom, function(x){strsplit(x, split = 'r')[[1]][2]}))) %>% arrange(statname, chr) %>% select(statname, everything(), -chr) %>% filter(!is.na(SYMBOL)) %T>% {write.csv(., file = '~/Git_repos/bookdown_thesis/Appendices/03-intrasfsPol.csv', row.names = FALSE, quote = TRUE)} %>% head(0) %>% kable(., caption = '(\\#tab:intrasfsPol) All genes that had a window in the 1st percentile and value < 0 from a Polynesian population', booktabs = TRUE, format = 'latex') %>% footnote(general = "Electronic Supplement: Appendices/03-intrasfsPol.csv", escape = FALSE)
```

```{r}
sig_ihs_val <- readRDS('~/data/NZ_coreExome_1kgp/haplotype/sig_ihs_values_with_genes_14-11-2017.RDS') %>% filter(abs(statvalue) > 2.6 )
sig_ihs_val %>% filter(superpop == 'POL', !is.na(genename) & genename != "") %>% select(Pop = pop, chrom, chrom_start, chrom_end, genename, iHS = statvalue) %>% arrange(chrom, chrom_start, Pop) %T>% {write.csv(., file = "Appendices/03-ihs-Pol.csv", row.names = FALSE, quote = FALSE)}  %>% head(0) %>% kable(., booktabs = TRUE, format = 'latex', caption = '(\\#tab:ihsPolgenes) All genes that had a significant iHS value in Polynesian populations.') %>% kableExtra::footnote("Appendices/03-ihs-Pol.csv", threeparttable = TRUE)
rm(sig_ihs_val)
```

```{r}
sig_nsl_val <- readRDS('~/data/NZ_coreExome_1kgp/haplotype/sig_nsl_values_with_genes_14-11-2017.RDS') %>% filter(abs(statvalue) > 2.6 )
sig_nsl_val %>% filter(superpop == 'POL', !is.na(genename) & genename != "") %>% select(Pop = pop, chrom, chrom_start, chrom_end, genename, nSL = statvalue) %>% arrange(chrom, chrom_start, Pop) %T>% {write.csv(., file = "Appendices/03-nsl-Pol.csv", row.names = FALSE, quote = FALSE)}  %>% head(0) %>% kable(., booktabs = TRUE, format = 'latex', caption = '(\\#tab:nslPolgenes) All genes that had a significant nSL value in Polynesian populations.') %>% kableExtra::footnote("Appendices/03-nsl-Pol.csv", threeparttable = TRUE)
rm(sig_nsl_val)
```

```{r, eval = TRUE}
xpehh <- read_delim('~/data/NZ_coreExome_1kgp/haplotype/sig_xpehh_29-10-2017.csv', delim = ',') %>% filter(!pop1 %in% c("POL","EPN","WPN", "NAD"), !pop2 %in% c("POL","EPN","WPN", "NAD")) %>% filter(abs(xpehh_value) > 2.6)
xpehh %>% filter(!is.na(genename) & genename != "")%>% select(Pop1 = pop1,Pop2 = pop2, chrom, chrom_start, chrom_end,XPEHH = xpehh_value, Genename = genename)  %>% arrange(chrom, chrom_start, Pop1, Pop2) %T>% 
  {
  #write.csv(., file = "Appendices/03-xpehh-Pol.csv", row.names = FALSE, quote = FALSE)
  }  %>% head(0) %>% kable(., booktabs = TRUE, format = 'latex', caption = '(\\#tab:xpehhPolgenes) All genes that had a significant XP-EHH value in Polynesian populations.') %>% kableExtra::footnote("Appendices/03-xpehh-Pol.csv", threeparttable = TRUE)
rm(xpehh)
```


```{r tdPol, results = 'asis', warning = FALSE, eval = FALSE}
# write table to csv file and create just the header of the table in the pdf
lower_sig_stats %>% filter(statname == 'Tajima.D', pop %in% c('CIM','NZM','SAM','TON')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Tajima.D', !pop %in% c('CIM','NZM','SAM','TON')), by = "SYMBOL") %>% select(chrom, SYMBOL, pop) %>% distinct() %>% spread(., key=pop, value = pop) %>% replace_na(list(CIM = "", NZM = "", SAM = "", TON = "")) %>% unite(., col = Pop, c('CIM','NZM','SAM','TON'), sep = ',')  %>% mutate(Pop = stringr::str_replace_all(Pop, "[,]+", ',') %>% stringr::str_replace_all(., pattern = '^[,]+|[,]$', replacement = "") %>% stringr::str_replace_all(., ",", ", ")) %>% mutate(chr = as.numeric(sapply(chrom, function(x){strsplit(x, split = 'r')[[1]][2]}))) %>% arrange(chr) %>% select(-chr) %T>%  {write.csv(., file = '~/Git_repos/bookdown_thesis/Appendices/03_tdPol.csv', row.names = FALSE, quote = TRUE)} %>% head(0) %>%  kable(., booktabs = TRUE, caption = "(\\#tab:lowersigtdPol) Loci that had a Tajima's D in the lower tail that was only found in a Polynesian population", format = 'latex') %>% footnote(general = "Electronic Supplement: Appendices/03\\\\_tdPol.csv", escape = FALSE)
```

```{r fwhPol, results='asis', warning= FALSE, echo = FALSE, eval = FALSE}
lower_sig_stats %>% filter(statname == 'Fay.Wu.H', pop %in% c('CIM','NZM','SAM','TON')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Fay.Wu.H', !pop %in% c('CIM','NZM','SAM','TON')), by = "SYMBOL") %>% select(chrom, SYMBOL, pop) %>% distinct() %>% spread(., key=pop, value = pop) %>% replace_na(list(CIM = "", NZM = "", SAM = "", TON = "")) %>% unite(., col = Pop, c('CIM','NZM','SAM','TON'), sep = ',') %>% mutate(Pop = stringr::str_replace_all(Pop, "[,]+", ',') %>% stringr::str_replace_all(., pattern = '^[,]+|[,]$', replacement = "") %>% stringr::str_replace_all(., ",", ", ")) %>% mutate(chr = as.numeric(sapply(chrom, function(x){strsplit(x, split = 'r')[[1]][2]}))) %>% arrange(chr) %>% select(-chr)%>%  write.csv(., file = '~/Git_repos/bookdown_thesis/Appendices/03_fwhPol.csv', row.names = FALSE, quote = TRUE) # %>%  kable(., booktab = TRUE, longtable = TRUE,caption = "(\\#tab:lowersigfwhPol) Loci that had a Fay and Wu's H in the lower tail that was only found in a Polynesian population")
```

```{r fldPol, results = 'asis', warning = FALSE, echo = FALSE, eval = FALSE}
lower_sig_stats %>% filter(statname == 'Fu.Li.D', pop %in% c('CIM','NZM','SAM','TON')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Fu.Li.D', !pop %in% c('CIM','NZM','SAM','TON')), by = "SYMBOL") %>% select(chrom, SYMBOL, pop) %>% distinct() %>% spread(., key=pop, value = pop) %>% replace_na(list(CIM = "", NZM = "", SAM = "", TON = "")) %>% unite(., col = Pop, c('CIM','NZM','SAM','TON'), sep = ',') %>% mutate(Pop = stringr::str_replace_all(Pop, "[,]+", ',') %>% stringr::str_replace_all(., pattern = '^[,]+|[,]$', replacement = "") %>% stringr::str_replace_all(., ",", ", "))%>% mutate(chr = as.numeric(sapply(chrom, function(x){strsplit(x, split = 'r')[[1]][2]}))) %>% arrange(chr) %>% select(-chr) %>% write.csv(., file = '~/Git_repos/bookdown_thesis/Appendices/03_fldPol.csv', row.names = FALSE, quote = TRUE) # %>% kable(., booktab = TRUE, longtable = TRUE,caption = "(\\#tab:lowersigfldPol) Loci that had a Fu and Li's D in the lower tail that was only found in a Polynesian population")
```

```{r flfPol, results = 'asis', warning = FALSE, echo = FALSE, eval = FALSE}
lower_sig_stats %>% filter(statname == 'Fu.Li.F', pop %in% c('CIM','NZM','SAM','TON')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Fu.Li.F', !pop %in% c('CIM','NZM','SAM','TON')), by = "SYMBOL") %>% select(chrom, SYMBOL, pop) %>% distinct() %>% spread(., key=pop, value = pop) %>% replace_na(list(CIM = "", NZM = "", SAM = "", TON = "")) %>% unite(., col = Pop, c('CIM','NZM','SAM','TON'), sep = ',') %>% mutate(Pop = stringr::str_replace_all(Pop, "[,]+", ',') %>% stringr::str_replace_all(., pattern = '^[,]+|[,]$', replacement = "") %>% stringr::str_replace_all(., ",", ", "))%>% mutate(chr = as.numeric(sapply(chrom, function(x){strsplit(x, split = 'r')[[1]][2]}))) %>% arrange(chr) %>% select(-chr) %>% head() %>% kable(., booktab = TRUE, longtable = TRUE,caption = "(\\#tab:lowersigflfPol) Loci that had a Fu and Li's F in the lower tail that was only found in a Polynesian population")
```

```{r zePol, results='asis', eval = FALSE}
lower_sig_stats %>% filter(statname == 'Zeng.E', pop %in% c('CIM','NZM','SAM','TON')) %>% anti_join(., lower_sig_stats %>% filter(statname == 'Zeng.E', !pop %in% c('CIM','NZM','SAM','TON')), by = "SYMBOL") %>% select(chrom, SYMBOL, pop) %>% distinct() %>% spread(., key=pop, value = pop) %>% replace_na(list(CIM = "", NZM = "", SAM = "", TON = "")) %>% unite(., col = Pop, c('CIM','NZM','SAM','TON'), sep = ',') %>% mutate(Pop = stringr::str_replace_all(Pop, "[,]+", ',') %>% stringr::str_replace_all(., pattern = '^[,]+|[,]$', replacement = "") %>% stringr::str_replace_all(., ",", ", "))%>% mutate(chr = as.numeric(sapply(chrom, function(x){strsplit(x, split = 'r')[[1]][2]}))) %>% arrange(chr) %>% select(-chr) %>%head() %>%  kable(.,booktab = TRUE, longtable = TRUE, caption = "(\\#tab:lowersigzePol) Loci that had a Zeng's E in the lower tail that was only found in a Polynesian population")
```

## Chapter 4 Tables

```{r}
create_gwas_cat_table <-function(){
  # data from rnotebooks/CoreExome/gwas_catalog.Rmd
  load('~/data/gwas_catalog/disease_ref_table-25-7-2017.RData')
  return(gwas_int_ref_table %>% select(-pmid, -PUBMEDID))
}

create_sel_summary_table <- function(s){
global_summary %>% ungroup%>% filter(stat == s) %>% left_join(., panel %>% select(pop, super_pop) %>% distinct(), by = 'pop') %>% arrange(super_pop) %>% select(super_pop, pop, mean, sd, min, lower_1,median,upper_99, max)  %>%  data.frame()
}
  
create_sel_stat_table <- function(sel_stat, cap){
  knitr::kable(chr_summary %>% filter(stat == sel_stat) %>% ungroup() %>%  select(pop, chrom, mean) %>% spread(.,chrom, mean) %>% left_join(., panel %>% select(pop, super_pop) %>% distinct(), "pop") %>% select(pop, super_pop, 2:23) %>% arrange(super_pop, pop) %>% head(), align = c('l','c',rep('r', 22)), col.names = c("Population", "Super Population", paste0('Chr',1:22)), caption = cap, digits =  3)
}
```


```{r, results = 'asis'}
m <- create_gwas_cat_table() %>% data.frame() %>%  arrange(DISEASE.TRAIT, ref) %>% select(-REPORTED.GENE.S., 'Category' = list, 'Disease Trait' = DISEASE.TRAIT, 'Reference' = ref) %>% mutate(Reference = sapply(Reference, function(x){paste0("\\citet{",substr(x, 2, nchar(x)),"}")})) %>% select(3,1,2)%>% distinct() %>% arrange(Category, Reference) #%>% filter(Category != "Kidney Disease")
#colnames(m) <- c("Disease\nTrait", "Reported\nGenes","Reference")  
#pander::pandoc.table(m, split.cells = 30 , caption = "(\\#tab:gwascatref) GWAS catalog studies used. Disease trait as specified in GWAS catalog", keep.line.breaks = TRUE)

kable(m %>% select(-Category), escape = FALSE, format = 'latex',caption = "(\\#tab:gwascatref) GWAS catalog studies used. Disease trait as specified in GWAS catalog", booktabs = TRUE, longtable = TRUE) %>% kable_styling(latex_options = c("repeat_header"), full_width = TRUE) %>% group_rows("Gout and Urate",1,18) %>% group_rows("Kidney Disease",19,26) %>% group_rows("Metabolic Syndrome", 27,30) %>% group_rows("Obesity", 31, 63) %>% group_rows("T2D", 64, 102) #%>% group_rows("Metabolic Syndrome", 19,22) %>% group_rows("Obesity", 23, 55) %>% group_rows("T2D", 56, 94) 

#
```


```{r gwasgenes}
# genelists
load('~/data/gwas_catalog/diseaseGR-25-7-2017.RData')
genelists <- data.frame(pheno = c(rep("obesity", length(obesity_GR$SYMBOL)), rep("urate", length( urate_goutGR$SYMBOL)), rep('t2d', length(t2d_GR$SYMBOL)), rep("kd", length(kd_GR$SYMBOL)), rep("metsyn", length(metsyn_GR$SYMBOL))),  genename = c(obesity_GR$SYMBOL, urate_goutGR$SYMBOL, t2d_GR$SYMBOL, kd_GR$SYMBOL, metsyn_GR$SYMBOL), stringsAsFactors = FALSE) %>% distinct() %>%  mutate(present = 1) %>% spread(pheno, present, 0)
genelists %>% select(genename, urate, obesity, t2d, kd, metsyn) %>% kable(., col.names = c("Gene name", "Urate and gout","Obesity","T2D","Kidney disease", "Metabolic syndrome"), caption = '(\\#tab:gwasgenes) Disease associated genes by category from the GWAS catalog. For references see Table \\ref{tab:gwascatref}.' , escape = FALSE, booktabs = TRUE)
```



```{r, echo = FALSE}
# This is created by coreExome_100kb_windows_intra_filtered.Rmd
chr_summary <- readRDS("~/data/NZ_coreExome_1kgp/100kbWindow_intra/filtered/100kbwindows_chr_summary_popgenome_filtered_3ns_resized.31-8-2017.RDS")
pops <- list()
panel <- read.delim('/home/murraycadzow/data/NZ_coreExome_1kgp/nz_1kgp.panel', stringsAsFactors = FALSE)
for(p in unique(panel$pop)){
  pops[[p]] <- panel %>% filter(pop == p) %>%  .[["sample"]]
}

global_summary <- readRDS("~/data/NZ_coreExome_1kgp/100kbWindow_intra/filtered/100kbwindows_summary_popgenome_filtered_3ns_resized.31-8-2017.RDS") %>% filter(!pop %in% c("WPN","EPN","POL","NAD"))
global_super_summary <- readRDS('~/data/NZ_coreExome_1kgp/100kbWindow_intra/filtered/100kbwindows_summary_superpop_popgenome_filtered_3ns_resized.6-11-2017.RDS')


```

```{r chap4data}
prop_list <- readRDS('~/data/NZ_coreExome_1kgp/100kbWindow_intra/100kbwindows_filtered_3ns_prop_list-31-8-2017.RDS')

load('~/data/gwas_catalog/diseaseGR-25-7-2017.RData')# brings in objects called {gc_urate_gout,urate_gout,kd,metsyn,obesity,t2d}_GR, 
load('~/data/NZ_coreExome_1kgp/100kbWindow_intra/gene_clustering_filtered_3ns-31-8-2017.RData')# brings in objects called gout, obesity, t2d, metsyn
names(gout) <- c('mat_d','prop')
names(t2d) <-  c('mat_d','prop')
names(obesity) <-  c('mat_d','prop')
names(metsyn) <-  c('mat_d','prop')

```


```{r gsTdTable, eval = TRUE, echo = FALSE}
gs_td_table <- create_sel_summary_table("Tajima.D")
gs_td_table %>% mutate_at(vars('mean','sd','min','lower_1','upper_99','median','max'), funs(sprintf("%.3f",. ))) %>% select(-super_pop) %>%  knitr::kable(., format = 'latex', booktabs = TRUE, col.names = c("Population","Mean","Std. Dev","Min","Lower 1%","Median","Upper 99%", "Max"), caption = "(\\#tab:gsTdTable) \\gls{td} summary statistics by population", align = c('l', rep('r', 8))) %>% kableExtra::group_rows("AFR", 1, 7) %>% kableExtra::group_rows("AMR", 8, 11) %>% kableExtra::group_rows("EAS", 12, 16) %>% kableExtra::group_rows("EUR", 17, 22) %>% kableExtra::group_rows("POL", 23, 26) %>% kableExtra::group_rows("SAS", 27,31)
```

```{r, eval = TRUE, echo = FALSE}
gs_fwh_table <-create_sel_summary_table("Fay.Wu.H")

gs_fwh_table %>%  mutate_at(vars('mean','sd','min','lower_1','upper_99','median','max'), funs(sprintf("%.3f",. ))) %>%  select(-super_pop) %>%  knitr::kable(., format = 'latex', booktabs = TRUE, col.names = c("Population","Mean","Std. Dev","Min","Lower 1%","Median","Upper 99%", "Max"), caption = "(\\#tab:gsFwhTable) \\gls{fwh} summary statistics by population", align = c('l', rep('r', 8))) %>% kableExtra::group_rows("AFR", 1, 7) %>% kableExtra::group_rows("AMR", 8, 11) %>% kableExtra::group_rows("EAS", 12, 16) %>% kableExtra::group_rows("EUR", 17, 22) %>% kableExtra::group_rows("POL", 23, 26) %>% kableExtra::group_rows("SAS", 27,31)
```


```{r, eval = TRUE, echo = FALSE}
gs_fld_table <- create_sel_summary_table("Fu.Li.D")

gs_fld_table %>% mutate_at(vars('mean','sd','min','lower_1','upper_99','median','max'), funs(sprintf("%.3f",. ))) %>% select(-super_pop) %>%  knitr::kable(., format = 'latex', booktabs = TRUE, col.names = c("Population","Mean","Std. Dev","Min","Lower 1%","Median","Upper 99%", "Max"), caption = "(\\#tab:gsFldTable) \\gls{fld} summary statistics by population", align = c('l', rep('r', 8))) %>% kableExtra::group_rows("AFR", 1, 7) %>% kableExtra::group_rows("AMR", 8, 11) %>% kableExtra::group_rows("EAS", 12, 16) %>% kableExtra::group_rows("EUR", 17, 22) %>% kableExtra::group_rows("POL", 23, 26) %>% kableExtra::group_rows("SAS", 27,31)
```

```{r, eval = TRUE, echo = FALSE}
gs_flf_table <- create_sel_summary_table("Fu.Li.F")

gs_flf_table %>% mutate_at(vars('mean','sd','min','lower_1','upper_99','median','max'), funs(sprintf("%.3f",. ))) %>% select(-super_pop) %>%  knitr::kable(., format = 'latex', booktabs = TRUE, col.names = c("Population","Mean","Std. Dev","Min","Lower 1%","Median","Upper 99%", "Max"), caption = "(\\#tab:gsFlfTable) \\gls{flf} summary statistics by population", align = c('l', rep('r', 8))) %>% kableExtra::group_rows("AFR", 1, 7) %>% kableExtra::group_rows("AMR", 8, 11) %>% kableExtra::group_rows("EAS", 12, 16) %>% kableExtra::group_rows("EUR", 17, 22) %>% kableExtra::group_rows("POL", 23, 26) %>% kableExtra::group_rows("SAS", 27,31)
```

```{r, eval = TRUE, echo = FALSE}
gs_ze_table <- create_sel_summary_table("Zeng.E")

gs_ze_table %>% mutate_at(vars('mean','sd','min','lower_1','upper_99','median','max'), funs(sprintf("%.3f",. ))) %>% select(-super_pop) %>%  knitr::kable(., format = 'latex', booktabs = TRUE, col.names = c("Population","Mean","Std. Dev","Min","Lower 1%","Median","Upper 99%", "Max"), caption = "(\\#tab:gsZeTable) \\gls{ze} summary statistics by population", align = c('l', rep('r', 8))) %>% kableExtra::group_rows("AFR", 1, 7) %>% kableExtra::group_rows("AMR", 8, 11) %>% kableExtra::group_rows("EAS", 12, 16) %>% kableExtra::group_rows("EUR", 17, 22) %>% kableExtra::group_rows("POL", 23, 26) %>% kableExtra::group_rows("SAS", 27,31)
```



```{r proportion, results ='asis'}
# prop_list[['td_neg']]$exclusivity %>% data.frame%>% mutate(pop = rownames(.)) %>% gather("cluster", "prop", 1:6) %>% filter(prop > 0) %>% group_by(pop) %>% summarise(mean(prop))
# 
# prop_list[['td_neg']]$super_prop %>% data.frame() %>% spread(super, max_prop) 


bind_rows(list(
  bind_rows(lapply(names(prop_list), function(x){prop_list[[x]]$super_prop %>%  spread(super, max_prop) %>% mutate(stat = sapply(x, function(y){strsplit(y, '_')[[1]][1]}), Type = sapply(x, function(y){strsplit(y, '_')[[1]][2]}) ) %>% select(NCOL(.), 1:(NCOL(.)-1)) })), #%>% mutate_if(., is.numeric, funs(sprintf("%.3f",.))),
  
  bind_rows(lapply(names(gout[[2]]), function(x){gout[[2]][[x]]$super_prop %>% spread(super, max_prop)%>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1)) })) %>% mutate(Type = 'Urate/Gout'),
  
  bind_rows(lapply(names(t2d[[2]]), function(x){t2d[[2]][[x]]$super_prop %>% spread(super, max_prop)%>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1)) })) %>% mutate(Type = '\\gls{t2d}'),
  
  bind_rows(lapply(names(obesity[[2]]), function(x){obesity[[2]][[x]]$super_prop %>% spread(super, max_prop)%>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1)) })) %>% mutate(Type = 'Obesity'),
  
  bind_rows(lapply(names(kd[[2]]), function(x){kd[[2]][[x]]$super_prop %>% spread(super, max_prop)%>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1)) })) %>% mutate(Type = 'Kidney Disease'),
  
  bind_rows(lapply(names(metsyn[[2]]), function(x){metsyn[[2]][[x]]$super_prop %>% spread(super, max_prop) %>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1)) })) %>% mutate(Type = 'Metabolic Syndrome')
  
)) %>% mutate(stat = sapply(stat, function(x){short_to_latex(stat_to_short(x))}), Type = sapply(Type, function(x){switch(x, chr = "Chromosome", neg = "Lower Tail", pos ="Upper Tail",x)})) %>% mutate_if(., is.numeric, funs(sprintf('%.2f',.)))%>%  select(stat, Type, everything()) %>% data.frame %>% arrange(stat) %>%  data.frame %>% pander::pandoc.table(., caption = "(\\#tab:proportion) Proportion")#knitr::kable(., caption = "Proportion", booktabs = TRUE) %>% kableExtra::group_rows("Fu and Li's D", 1, 7) %>% kableExtra::group_rows("Fu and Li's F", 8, 14) %>% kableExtra::group_rows("Fay and Wu's H", 15, 21) %>% kableExtra::group_rows("Tajima's D", 22, 28) %>% kableExtra::group_rows("Zeng's E", 29, 35) %>% kableExtra::kable_styling(latex_options = "scale_down")

```




```{r exclusivity, results = 'asis'}
#exclusivity
bind_rows(list(
  bind_rows(lapply(names(prop_list), function(x){prop_list[[x]]$exclusivity %>% data.frame%>% mutate(pop = row.names(.)) %>% gather("cluster","prop",1:6) %>% filter(prop > 0) %>% group_by(pop) %>% summarise(mean_prop = mean(prop)) %>% spread(pop, mean_prop) %>% mutate(stat = sapply(x, function(y){strsplit(y, '_')[[1]][1]}), Type = sapply(x, function(y){strsplit(y, '_')[[1]][2]}) ) %>% select(NCOL(.), 1:(NCOL(.)-1)) })),
  
  bind_rows(lapply(names(gout[[2]]), function(x){gout[[2]][[x]]$exclusivity %>% data.frame%>% mutate(pop = row.names(.)) %>% gather("cluster","prop",1:6) %>% filter(prop > 0) %>% group_by(pop) %>% summarise(mean_prop = mean(prop)) %>% spread(pop, mean_prop) %>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1))})) %>% mutate(Type = 'Urate/Gout'),
  
  bind_rows(lapply(names(t2d[[2]]), function(x){t2d[[2]][[x]]$exclusivity %>% data.frame%>% mutate(pop = row.names(.)) %>% gather("cluster","prop",1:6) %>% filter(prop > 0) %>% group_by(pop) %>% summarise(mean_prop = mean(prop)) %>% spread(pop, mean_prop) %>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1))})) %>% mutate(Type = '\\gls{t2d}'),
  
  bind_rows(lapply(names(obesity[[2]]), function(x){obesity[[2]][[x]]$exclusivity %>% data.frame%>% mutate(pop = row.names(.)) %>% gather("cluster","prop",1:6) %>% filter(prop > 0) %>% group_by(pop) %>% summarise(mean_prop = mean(prop)) %>% spread(pop, mean_prop) %>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1))})) %>% mutate(Type = 'Obesity'),
  
  bind_rows(lapply(names(kd[[2]]), function(x){kd[[2]][[x]]$exclusivity %>% data.frame%>% mutate(pop = row.names(.)) %>% gather("cluster","prop",1:6) %>% filter(prop > 0) %>% group_by(pop) %>% summarise(mean_prop = mean(prop)) %>% spread(pop, mean_prop) %>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1))})) %>% mutate(Type = 'Kidney Disease'),
  
  bind_rows(lapply(names(metsyn[[2]]), function(x){metsyn[[2]][[x]]$exclusivity %>% data.frame%>% mutate(pop = row.names(.)) %>% gather("cluster","prop",1:6) %>% filter(prop > 0) %>% group_by(pop) %>% summarise(mean_prop = mean(prop)) %>% spread(pop, mean_prop) %>% mutate(stat = x) %>% select(NCOL(.), 1:(NCOL(.)-1))})) %>% mutate(Type = 'Metabolic Syndrome')    
  
  
)) %>% select(stat, Type, everything())  %>% mutate(stat = sapply(stat, function(x){short_to_latex(stat_to_short(x))}), Type = sapply(Type, function(x){switch(x, chr = "Chromosome", neg = "Lower Tail", pos ="Upper Tail",x)})) %>% mutate_if(., is.numeric, funs(sprintf("%.2f",.))) %>% arrange(Type) %>%  data.frame %>% pander::pandoc.table(., caption = "(\\#tab:exclusivity) Exclusivity")#knitr::kable(., caption = "Exclusivity", booktabs = TRUE) %>% kableExtra::group_rows("Fu and Li's D", 1, 7) %>% kableExtra::group_rows("Fu and Li's F", 8, 14) %>% kableExtra::group_rows("Fay and Wu's H", 15, 21) %>% kableExtra::group_rows("Tajima's D", 22, 28) %>% kableExtra::group_rows("Zeng's E", 29, 35) 
```
